--- odepkg-20170514/PKG_DEL.orig	2017-05-14 20:24:41.000000000 +0100
+++ odepkg-20170514/PKG_DEL	2017-05-16 21:46:37.869372514 +0100
@@ -5,8 +5,13 @@
 
 if (! exist (fullfile (dir, "inst"), "dir"))
   ## Run this if the package is installed
+  dir = fileparts (dir);
+  dir = fileparts (dir);
+  [dir, pname, ext] = fileparts (dir);
+  dir = pkg("prefix");
+
   for ii=1:length(dirlist)
-    rmpath (fullfile (dir, "..", dirlist{ii}))
+    rmpath (fullfile (dir, [pname ext]], dirlist{ii}))
   endfor
 else
   ## Run this if we are testing the package without installation        
--- odepkg-20170514/PKG_ADD.orig	2017-05-14 20:24:41.000000000 +0100
+++ odepkg-20170514/PKG_ADD	2017-05-16 21:46:00.038372779 +0100
@@ -5,8 +5,13 @@
 
 if (! exist (fullfile (dir, "inst"), "dir"))
   ## Run this if the package is installed
+  dir = fileparts (dir);
+  dir = fileparts (dir);
+  [dir, pname, ext] = fileparts (dir);
+  dir = pkg("prefix");
+
   for ii=1:length(dirlist)
-    addpath (fullfile (dir, "..", dirlist{ii}))
+    addpath (fullfile (dir, [pname ext], dirlist{ii}))
   endfor
 else
   ## Run this if we are testing the package without installation        
