--- odepkg-20170514/PKG_DEL.orig	2017-05-14 20:24:41.037366624 +0100
+++ odepkg-20170514/PKG_DEL	2017-05-18 19:59:09.144062145 +0100
@@ -5,8 +5,12 @@
 
 if (! exist (fullfile (dir, "inst"), "dir"))
   ## Run this if the package is installed
+  dir = fileparts (dir);
+  [dir, pname, ext] = fileparts (dir);
+  dir = pkg("-global", "prefix");
+
   for ii=1:length(dirlist)
-    rmpath (fullfile (dir, "..", dirlist{ii}))
+    rmpath (fullfile (dir, [pname ext], dirlist{ii}))
   endfor
 else
   ## Run this if we are testing the package without installation        
--- odepkg-20170514/PKG_ADD.orig	2017-05-14 20:24:41.037366624 +0100
+++ odepkg-20170514/PKG_ADD	2017-05-18 19:58:11.743058067 +0100
@@ -5,8 +5,12 @@
 
 if (! exist (fullfile (dir, "inst"), "dir"))
   ## Run this if the package is installed
+  dir = fileparts (dir);
+  [dir, pname, ext] = fileparts (dir);
+  dir = pkg("-global", "prefix");
+
   for ii=1:length(dirlist)
-    addpath (fullfile (dir, "..", dirlist{ii}))
+    addpath (fullfile (dir, [pname ext], dirlist{ii}))
   endfor
 else
   ## Run this if we are testing the package without installation        
