--- extractpdfmark-1.0.2/src/poppler-core/destname.cc.orig	2017-04-16 16:09:38.000000000 +0100
+++ extractpdfmark-1.0.2/src/poppler-core/destname.cc	2018-12-06 22:24:29.874757741 +0000
@@ -32,7 +32,7 @@
 
 inline std::string goo_to_string (GooString *goo)
 {
-  return std::string {goo->getCString (),
+  return std::string {goo->c_str (),
       static_cast<std::string::size_type>(goo->getLength ())};
 }
 
