--- aalib-1.4rc5/configure.orig	2001-04-26 15:44:54.000000000 +0100
+++ aalib-1.4rc5/configure	2014-04-21 20:44:31.274670195 +0100
@@ -4315,13 +4315,13 @@
 
 # Set runtime shared library paths as needed
 
-if test $ARCH = linux -o $ARCH = freebsd -o $ARCH = bsdi; then
-  AALIB_RLD_FLAGS="-Wl,-rpath,\${exec_prefix}/lib"
-fi
-
-if test $ARCH = solaris; then
-  AALIB_RLD_FLAGS="-R\${exec_prefix}/lib"
-fi
+#if test $ARCH = linux -o $ARCH = freebsd -o $ARCH = bsdi; then
+#  AALIB_RLD_FLAGS="-Wl,-rpath,\${exec_prefix}/lib"
+#fi
+#
+#if test $ARCH = solaris; then
+#  AALIB_RLD_FLAGS="-R\${exec_prefix}/lib"
+#fi
 
 
 
