--- abiword-extras-2.6.0/clipart/Makefile.am.orig	2008-03-18 22:18:48.000000000 +0000
+++ abiword-extras-2.6.0/clipart/Makefile.am	2008-03-20 09:34:34.000000000 +0000
@@ -1,5 +1,5 @@
 
-clipartdir = $(DESTDIR)/$(ABIWORD_EXTRAS_PKGDATADIR)/clipart
+clipartdir = $(ABIWORD_EXTRAS_PKGDATADIR)/clipart
 clipart_DATA = \
 	book.png \
 	chip.png \
--- abiword-extras-2.6.0/templates/Makefile.am.orig	2008-03-18 22:18:49.000000000 +0000
+++ abiword-extras-2.6.0/templates/Makefile.am	2008-03-20 09:35:07.000000000 +0000
@@ -1,5 +1,5 @@
 
-templatesdir = $(DESTDIR)/$(ABIWORD_EXTRAS_PKGDATADIR)/templates
+templatesdir = $(ABIWORD_EXTRAS_PKGDATADIR)/templates
 templates_DATA = \
 	A4.awt \
 	Business-Letter.awt \
--- abiword-extras-2.6.0/dictionary/Makefile.am.orig	2008-03-18 22:18:47.000000000 +0000
+++ abiword-extras-2.6.0/dictionary/Makefile.am	2008-03-20 09:34:53.000000000 +0000
@@ -1,5 +1,5 @@
 
-dictionarydir = $(DESTDIR)/$(ABIWORD_EXTRAS_PKGDATADIR)/dictionary
+dictionarydir = $(ABIWORD_EXTRAS_PKGDATADIR)/dictionary
 
 if BIGENDIAN
 hash = BigEndian32.american.hash
