--- Aegisub-3.3.3/src/command/video.cpp.orig	2022-07-03 18:27:30.000000000 +0100
+++ Aegisub-3.3.3/src/command/video.cpp	2023-05-06 15:33:44.436048685 +0100
@@ -475,7 +475,7 @@
 		// If where ever that is isn't defined, we can't save there
 		if ((basepath == "\\") || (basepath == "/")) {
 			// So save to the current user's home dir instead
-			basepath = wxGetHomeDir().c_str();
+			basepath = static_cast<const char *>(wxGetHomeDir().c_str());
 		}
 	}
 	// Actual fixed (possibly relative) path, decode it
--- Aegisub-3.3.3/src/dialog_attachments.cpp.orig	2022-07-03 18:27:30.000000000 +0100
+++ Aegisub-3.3.3/src/dialog_attachments.cpp	2023-05-06 15:34:34.317046305 +0100
@@ -161,7 +161,7 @@
 
 	// Multiple or single?
 	if (listView->GetNextSelected(i) != -1)
-		path = wxDirSelector(_("Select the path to save the files to:"), to_wx(OPT_GET("Path/Fonts Collector Destination")->GetString())).c_str();
+		path = static_cast<const char *>(wxDirSelector(_("Select the path to save the files to:"), to_wx(OPT_GET("Path/Fonts Collector Destination")->GetString())).c_str());
 	else {
 		path = SaveFileSelector(
 			_("Select the path to save the file to:"),
