--- akode-2.0.2/akode/plugins/ffmpeg_decoder/ffmpeg_decoder.cpp.orig	2008-02-04 12:14:23.000000000 +0000
+++ akode-2.0.2/akode/plugins/ffmpeg_decoder/ffmpeg_decoder.cpp	2008-02-04 12:15:31.000000000 +0000
@@ -26,10 +26,19 @@
 #include "decoder.h"
 
 #include <assert.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 #include <ffmpeg/avcodec.h>
 #include <ffmpeg/avformat.h>
 #include <ffmpeg/avio.h>
 
+#ifdef __cplusplus
+}
+#endif
+
 #include "ffmpeg_decoder.h"
 #include <iostream>
 
