--- AliceVision-2.4.0/src/dependencies/flann/src/cpp/CMakeLists.txt.orig	2021-03-09 16:47:48.000000000 +0000
+++ AliceVision-2.4.0/src/dependencies/flann/src/cpp/CMakeLists.txt	2021-03-09 16:50:53.024041896 +0000
@@ -14,7 +14,7 @@
 
 INSTALL(
   TARGETS flann_cpp_s
-  DESTINATION lib
+  DESTINATION lib${LIB_SUFFIX}
   EXPORT aliceVision-targets)
 
 INSTALL (
--- AliceVision-2.4.0/src/dependencies/flann/cmake/flann_utils.cmake.orig	2021-03-09 16:47:48.000000000 +0000
+++ AliceVision-2.4.0/src/dependencies/flann/cmake/flann_utils.cmake	2021-03-09 16:50:53.024041896 +0000
@@ -1,6 +1,6 @@
 macro(GET_OS_INFO)
     string(REGEX MATCH "Linux" OS_IS_LINUX ${CMAKE_SYSTEM_NAME})
-    set(FLANN_LIB_INSTALL_DIR "lib")
+    set(FLANN_LIB_INSTALL_DIR "lib${LIB_SUFFIX}")
     if(NOT FLANN_INCLUDE_INSTALL_DIR)
       set(FLANN_INCLUDE_INSTALL_DIR
         "include/${PROJECT_NAME_LOWER}-${FLANN_VERSION}")
--- AliceVision-2.4.0/src/dependencies/osi_clp/Clp/src/CMakeLists.txt.orig	2021-03-09 16:47:53.000000000 +0000
+++ AliceVision-2.4.0/src/dependencies/osi_clp/Clp/src/CMakeLists.txt	2021-03-09 16:50:53.024041896 +0000
@@ -36,7 +36,7 @@
 list(REMOVE_ITEM CLP_SRC_CPP ${REMOVE_CPP})
 
 ADD_LIBRARY(lib_clp ${CLP_SRC_HEADER} ${CLP_SRC_CPP})
-INSTALL(TARGETS lib_clp DESTINATION lib EXPORT aliceVision-targets)
+INSTALL(TARGETS lib_clp DESTINATION lib${LIB_SUFFIX} EXPORT aliceVision-targets)
 
 add_subdirectory(OsiClp)
 
--- AliceVision-2.4.0/src/dependencies/osi_clp/Clp/src/OsiClp/CMakeLists.txt.orig	2021-03-09 16:47:53.000000000 +0000
+++ AliceVision-2.4.0/src/dependencies/osi_clp/Clp/src/OsiClp/CMakeLists.txt	2021-03-09 16:50:53.024041896 +0000
@@ -10,4 +10,4 @@
 	*.cpp
 )
 ADD_LIBRARY(lib_OsiClpSolver ${OSI_HEADERS} ${OSI_SRCS})
-INSTALL(TARGETS lib_OsiClpSolver DESTINATION lib EXPORT aliceVision-targets)
+INSTALL(TARGETS lib_OsiClpSolver DESTINATION lib${LIB_SUFFIX} EXPORT aliceVision-targets)
--- AliceVision-2.4.0/src/dependencies/osi_clp/Osi/src/Osi/CMakeLists.txt.orig	2021-03-09 16:47:53.000000000 +0000
+++ AliceVision-2.4.0/src/dependencies/osi_clp/Osi/src/Osi/CMakeLists.txt	2021-03-09 16:50:53.024041896 +0000
@@ -11,4 +11,4 @@
 )
 
 ADD_LIBRARY(lib_Osi ${OSI_OSI_HEADER} ${OSI_OSI_CPP})
-INSTALL(TARGETS lib_Osi DESTINATION lib EXPORT aliceVision-targets)
+INSTALL(TARGETS lib_Osi DESTINATION lib${LIB_SUFFIX} EXPORT aliceVision-targets)
--- AliceVision-2.4.0/src/dependencies/osi_clp/CoinUtils/src/CMakeLists.txt.orig	2021-03-09 16:47:53.000000000 +0000
+++ AliceVision-2.4.0/src/dependencies/osi_clp/CoinUtils/src/CMakeLists.txt	2021-03-09 16:50:53.024041896 +0000
@@ -13,4 +13,4 @@
 
 
 ADD_LIBRARY(lib_CoinUtils ${COINUTILS_HEADER} ${COINUTILS_CPP})
-INSTALL(TARGETS lib_CoinUtils DESTINATION lib EXPORT aliceVision-targets)
+INSTALL(TARGETS lib_CoinUtils DESTINATION lib${LIB_SUFFIX} EXPORT aliceVision-targets)
--- AliceVision-2.4.0/src/dependencies/lemon/lemon/CMakeLists.txt.orig	2021-03-09 16:47:48.000000000 +0000
+++ AliceVision-2.4.0/src/dependencies/lemon/lemon/CMakeLists.txt	2021-03-09 16:50:53.024041896 +0000
@@ -61,7 +61,7 @@
 
 INSTALL(
   TARGETS lemon
-  DESTINATION lib
+  DESTINATION lib${LIB_SUFFIX}
   EXPORT aliceVision-targets
 )
 
--- AliceVision-2.4.0/src/CMakeLists.txt.orig	2021-03-09 16:47:48.000000000 +0000
+++ AliceVision-2.4.0/src/CMakeLists.txt	2021-10-31 14:47:52.186584172 +0000
@@ -41,7 +41,7 @@
 endfunction()
 
 # C++11
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 # ==============================================================================
@@ -209,7 +209,7 @@
 if(NOT CXX11_COMPILER)
   message(FATAL_ERROR "The compiler does not support the CXX11 standard.")
 endif(NOT CXX11_COMPILER)
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 # ==============================================================================
@@ -898,7 +898,7 @@
               PROPERTY FOLDER AliceVision
               )
 
-      install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/doc  DESTINATION ${CMAKE_INSTALL_DOCDIR})
+      install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/htmlDoc  DESTINATION ${CMAKE_INSTALL_DOCDIR})
 
     endif()
   endif()
--- AliceVision-2.4.0/src/software/export/main_exportMatlab.cpp.orig	2021-03-09 16:47:48.000000000 +0000
+++ AliceVision-2.4.0/src/software/export/main_exportMatlab.cpp	2022-04-18 17:15:19.085462684 +0100
@@ -20,6 +20,7 @@
 #include <iterator>
 #include <iomanip>
 #include <map>
+#include <fstream>
 
 // These constants define the current software version.
 // They must be updated when the command line is changed.
--- AliceVision-2.4.0/src/software/export/main_exportPMVS.cpp.orig	2021-03-09 16:47:48.000000000 +0000
+++ AliceVision-2.4.0/src/software/export/main_exportPMVS.cpp	2022-04-18 17:13:58.400466534 +0100
@@ -19,6 +19,7 @@
 #include <cmath>
 #include <iterator>
 #include <iomanip>
+#include <fstream>
 
 // These constants define the current software version.
 // They must be updated when the command line is changed.
--- AliceVision-2.4.0/src/software/export/main_exportMVE2.cpp.orig	2021-03-09 16:47:48.000000000 +0000
+++ AliceVision-2.4.0/src/software/export/main_exportMVE2.cpp	2022-04-18 17:14:42.106464449 +0100
@@ -19,6 +19,7 @@
 #include <cmath>
 #include <iterator>
 #include <iomanip>
+#include <fstream>
 
 // These constants define the current software version.
 // They must be updated when the command line is changed.
--- AliceVision-2.4.0/src/software/pipeline/main_prepareDenseScene.cpp.orig	2021-03-09 16:47:48.000000000 +0000
+++ AliceVision-2.4.0/src/software/pipeline/main_prepareDenseScene.cpp	2022-04-18 17:15:58.752460792 +0100
@@ -24,6 +24,7 @@
 #include <set>
 #include <iterator>
 #include <iomanip>
+#include <fstream>
 
 // These constants define the current software version.
 // They must be updated when the command line is changed.
--- AliceVision-2.4.0/src/software/pipeline/main_LdrToHdrSampling.cpp.orig	2021-03-09 16:47:48.000000000 +0000
+++ AliceVision-2.4.0/src/software/pipeline/main_LdrToHdrSampling.cpp	2022-04-18 17:13:03.928469133 +0100
@@ -34,6 +34,7 @@
 #include <boost/accumulators/statistics.hpp>
 
 #include <sstream>
+#include <fstream>
 
 
 // These constants define the current software version.
--- AliceVision-2.4.0/src/software/pipeline/main_LdrToHdrCalibration.cpp.orig	2021-03-09 16:47:48.000000000 +0000
+++ AliceVision-2.4.0/src/software/pipeline/main_LdrToHdrCalibration.cpp	2022-04-18 17:11:23.114473943 +0100
@@ -31,6 +31,7 @@
 #include <boost/filesystem.hpp>
 
 #include <sstream>
+#include <fstream>
 
 
 // These constants define the current software version.
--- AliceVision-2.4.0/src/aliceVision/hdr/rgbCurve.cpp.orig	2021-03-09 16:47:48.000000000 +0000
+++ AliceVision-2.4.0/src/aliceVision/hdr/rgbCurve.cpp	2021-08-16 18:15:30.598832125 +0100
@@ -10,6 +10,7 @@
 #include <iostream>
 #include <sstream>
 #include <numeric>
+#include <limits>
 
 #include <aliceVision/system/Logger.hpp>
 
--- AliceVision-2.4.0/src/cmake/FindOpenImageIO.cmake.orig	2021-03-09 16:47:48.000000000 +0000
+++ AliceVision-2.4.0/src/cmake/FindOpenImageIO.cmake	2021-10-31 14:56:15.174732629 +0000
@@ -106,6 +106,12 @@
 endif (NOT OPENIMAGEIO_LIBRARY OR
        NOT EXISTS ${OPENIMAGEIO_LIBRARY})
 
+find_library(OPENIMAGEIOUTIL_LIBRARY NAMES OpenImageIO_Util
+  HINTS
+        ${OPENIMAGEIO_LIBRARY_DIR_HINTS}
+        # ${OPENIMAGEIO_CHECK_LIBRARY_DIRS}
+  PATH_SUFFIXES ${OPENIMAGEIO_CHECK_LIBRARY_SUFFIXES})
+
 # Mark internally as found, then verify. OPENIMAGEIO_REPORT_NOT_FOUND() unsets
 # if called.
 set(OPENIMAGEIO_FOUND TRUE)
@@ -135,7 +141,7 @@
 # Set standard CMake FindPackage variables if found.
 if (OPENIMAGEIO_FOUND)
   set(OPENIMAGEIO_INCLUDE_DIRS ${OPENIMAGEIO_INCLUDE_DIR} ${OPENEXR_INCLUDE_DIR} ${ILMBASE_INCLUDE_DIR})
-  set(OPENIMAGEIO_LIBRARIES ${OPENIMAGEIO_LIBRARY} ${ILMBASE_LIBRARIES})
+  set(OPENIMAGEIO_LIBRARIES ${OPENIMAGEIO_LIBRARY} ${OPENIMAGEIOUTIL_LIBRARY} ${ILMBASE_LIBRARIES})
 endif (OPENIMAGEIO_FOUND)
 
 openimageio_reset_find_library_prefix()
