--- AliceVision-20220505.156c632/src/dependencies/osi_clp/Clp/src/CMakeLists.txt.orig	2022-05-05 08:19:15.000000000 +0100
+++ AliceVision-20220505.156c632/src/dependencies/osi_clp/Clp/src/CMakeLists.txt	2022-05-05 09:11:50.081367149 +0100
@@ -36,7 +36,7 @@
 list(REMOVE_ITEM CLP_SRC_CPP ${REMOVE_CPP})
 
 ADD_LIBRARY(lib_clp ${CLP_SRC_HEADER} ${CLP_SRC_CPP})
-INSTALL(TARGETS lib_clp DESTINATION lib EXPORT aliceVision-targets)
+INSTALL(TARGETS lib_clp DESTINATION lib${LIB_SUFFIX} EXPORT aliceVision-targets)
 
 add_subdirectory(OsiClp)
 
--- AliceVision-20220505.156c632/src/dependencies/osi_clp/Clp/src/OsiClp/CMakeLists.txt.orig	2022-05-05 08:19:15.000000000 +0100
+++ AliceVision-20220505.156c632/src/dependencies/osi_clp/Clp/src/OsiClp/CMakeLists.txt	2022-05-05 09:17:41.416350386 +0100
@@ -10,4 +10,4 @@
 	*.cpp
 )
 ADD_LIBRARY(lib_OsiClpSolver ${OSI_HEADERS} ${OSI_SRCS})
-INSTALL(TARGETS lib_OsiClpSolver DESTINATION lib EXPORT aliceVision-targets)
+INSTALL(TARGETS lib_OsiClpSolver DESTINATION lib${LIB_SUFFIX} EXPORT aliceVision-targets)
--- AliceVision-20220505.156c632/src/dependencies/osi_clp/Osi/src/Osi/CMakeLists.txt.orig	2022-05-05 08:19:15.000000000 +0100
+++ AliceVision-20220505.156c632/src/dependencies/osi_clp/Osi/src/Osi/CMakeLists.txt	2022-05-05 09:12:21.233365663 +0100
@@ -11,4 +11,4 @@
 )
 
 ADD_LIBRARY(lib_Osi ${OSI_OSI_HEADER} ${OSI_OSI_CPP})
-INSTALL(TARGETS lib_Osi DESTINATION lib EXPORT aliceVision-targets)
+INSTALL(TARGETS lib_Osi DESTINATION lib${LIB_SUFFIX} EXPORT aliceVision-targets)
--- AliceVision-20220505.156c632/src/dependencies/osi_clp/CoinUtils/src/CMakeLists.txt.orig	2022-05-05 08:19:15.000000000 +0100
+++ AliceVision-20220505.156c632/src/dependencies/osi_clp/CoinUtils/src/CMakeLists.txt	2022-05-05 09:12:47.955364388 +0100
@@ -13,4 +13,4 @@
 
 
 ADD_LIBRARY(lib_CoinUtils ${COINUTILS_HEADER} ${COINUTILS_CPP})
-INSTALL(TARGETS lib_CoinUtils DESTINATION lib EXPORT aliceVision-targets)
+INSTALL(TARGETS lib_CoinUtils DESTINATION lib${LIB_SUFFIX} EXPORT aliceVision-targets)
--- AliceVision-20220505.156c632/src/dependencies/lemon/lemon/CMakeLists.txt.orig	2022-05-05 08:19:11.000000000 +0100
+++ AliceVision-20220505.156c632/src/dependencies/lemon/lemon/CMakeLists.txt	2022-05-05 09:17:05.611352094 +0100
@@ -61,7 +61,7 @@
 
 INSTALL(
   TARGETS lemon
-  DESTINATION lib
+  DESTINATION lib${LIB_SUFFIX}
   EXPORT aliceVision-targets
 )
 
--- AliceVision-20220505.156c632/src/software/export/main_exportMatlab.cpp.orig	2022-05-05 08:19:11.000000000 +0100
+++ AliceVision-20220505.156c632/src/software/export/main_exportMatlab.cpp	2022-05-05 09:07:53.795378423 +0100
@@ -14,6 +14,7 @@
 #include <boost/filesystem.hpp>
 #include <boost/progress.hpp>
 
+#include <fstream>
 #include <stdlib.h>
 #include <stdio.h>
 #include <cmath>
--- AliceVision-20220505.156c632/src/software/export/main_exportPMVS.cpp.orig	2022-05-05 08:19:11.000000000 +0100
+++ AliceVision-20220505.156c632/src/software/export/main_exportPMVS.cpp	2022-05-05 09:06:11.552383301 +0100
@@ -14,6 +14,7 @@
 #include <boost/filesystem.hpp>
 #include <boost/progress.hpp>
 
+#include <fstream>
 #include <stdlib.h>
 #include <stdio.h>
 #include <cmath>
--- AliceVision-20220505.156c632/src/software/export/main_exportMVE2.cpp.orig	2022-05-05 08:19:11.000000000 +0100
+++ AliceVision-20220505.156c632/src/software/export/main_exportMVE2.cpp	2022-05-05 09:07:06.325380688 +0100
@@ -14,6 +14,7 @@
 #include <boost/filesystem.hpp>
 #include <boost/progress.hpp>
 
+#include <fstream>
 #include <stdlib.h>
 #include <stdio.h>
 #include <cmath>
--- AliceVision-20220505.156c632/src/software/pipeline/main_featureExtraction.cpp.orig	2022-05-05 08:19:11.000000000 +0100
+++ AliceVision-20220505.156c632/src/software/pipeline/main_featureExtraction.cpp	2022-11-09 16:31:52.761689889 +0000
@@ -27,6 +27,7 @@
 #include <boost/filesystem.hpp>
 #include <boost/progress.hpp>
 
+#include <iomanip>
 #include <string>
 #include <fstream>
 #include <sstream>
--- AliceVision-20220505.156c632/src/software/pipeline/main_prepareDenseScene.cpp.orig	2022-05-05 08:19:11.000000000 +0100
+++ AliceVision-20220505.156c632/src/software/pipeline/main_prepareDenseScene.cpp	2022-05-05 08:57:06.191409322 +0100
@@ -20,6 +20,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <cmath>
+#include <fstream>
 #include <vector>
 #include <set>
 #include <iterator>
--- AliceVision-20220505.156c632/src/software/pipeline/main_LdrToHdrMerge.cpp.orig	2022-05-05 08:19:11.000000000 +0100
+++ AliceVision-20220505.156c632/src/software/pipeline/main_LdrToHdrMerge.cpp	2022-11-09 16:30:31.246693778 +0000
@@ -24,6 +24,7 @@
 #include <boost/program_options.hpp>
 #include <boost/filesystem.hpp>
 #include <sstream>
+#include <iomanip>
 
 // These constants define the current software version.
 // They must be updated when the command line is changed.
--- AliceVision-20220505.156c632/src/software/pipeline/main_LdrToHdrSampling.cpp.orig	2022-05-05 08:19:11.000000000 +0100
+++ AliceVision-20220505.156c632/src/software/pipeline/main_LdrToHdrSampling.cpp	2022-05-05 09:00:51.326398580 +0100
@@ -33,6 +33,7 @@
 #include <boost/accumulators/accumulators.hpp>
 #include <boost/accumulators/statistics.hpp>
 
+#include <fstream>
 #include <sstream>
 
 
--- AliceVision-20220505.156c632/src/software/pipeline/main_LdrToHdrCalibration.cpp.orig	2022-05-05 08:19:11.000000000 +0100
+++ AliceVision-20220505.156c632/src/software/pipeline/main_LdrToHdrCalibration.cpp	2022-05-05 09:01:57.722395412 +0100
@@ -30,6 +30,7 @@
 #include <boost/program_options.hpp>
 #include <boost/filesystem.hpp>
 
+#include <fstream>
 #include <sstream>
 
 
--- AliceVision-20220505.156c632/src/aliceVision/keyframe/KeyframeSelector.cpp.orig	2022-05-05 08:19:11.000000000 +0100
+++ AliceVision-20220505.156c632/src/aliceVision/keyframe/KeyframeSelector.cpp	2022-11-09 16:29:31.961696607 +0000
@@ -12,6 +12,7 @@
 
 #include <boost/filesystem.hpp>
 
+#include <iomanip>
 #include <random>
 #include <tuple>
 #include <cassert>
--- AliceVision-20220505.156c632/src/aliceVision/fuseCut/DelaunayGraphCut.cpp.orig	2022-05-05 08:19:11.000000000 +0100
+++ AliceVision-20220505.156c632/src/aliceVision/fuseCut/DelaunayGraphCut.cpp	2022-11-09 16:34:47.315681561 +0000
@@ -1629,7 +1629,7 @@
     Point3d tempIntersectPt;
     const Point2d triangleUv = getLineTriangleIntersectBarycCoords(&tempIntersectPt, A, B, C, &originPt, &DirVec);
 
-    if (!isnormal(tempIntersectPt.x) || !isnormal(tempIntersectPt.y) || !isnormal(tempIntersectPt.z))
+    if (!std::isnormal(tempIntersectPt.x) || !std::isnormal(tempIntersectPt.y) || !std::isnormal(tempIntersectPt.z))
     {
         // This is not suppose to happen in real life, we log a warning instead of raising an exeption if we face a border case
         // ALICEVISION_LOG_WARNING("Invalid/notNormal intersection point found during rayIntersectTriangle.");
--- AliceVision-20220505.156c632/src/aliceVision/sfmDataIO/middlebury.cpp.orig	2022-05-05 08:19:11.000000000 +0100
+++ AliceVision-20220505.156c632/src/aliceVision/sfmDataIO/middlebury.cpp	2022-05-05 08:29:11.661548739 +0100
@@ -8,6 +8,7 @@
 #include <aliceVision/numeric/numeric.hpp>
 #include <aliceVision/image/io.hpp>
 
+#include <fstream>
 #include <boost/filesystem.hpp>
 
 namespace aliceVision {
