--- AliceVision-2.3.1/src/dependencies/flann/src/cpp/CMakeLists.txt.orig	2020-10-31 10:29:55.000000000 +0000
+++ AliceVision-2.3.1/src/dependencies/flann/src/cpp/CMakeLists.txt	2020-11-15 21:44:39.866579428 +0000
@@ -14,7 +14,7 @@
 
 INSTALL(
   TARGETS flann_cpp_s
-  DESTINATION lib
+  DESTINATION lib${LIB_SUFFIX}
   EXPORT aliceVision-targets)
 
 INSTALL (
--- AliceVision-2.3.1/src/dependencies/flann/cmake/flann_utils.cmake.orig	2020-10-31 10:29:55.000000000 +0000
+++ AliceVision-2.3.1/src/dependencies/flann/cmake/flann_utils.cmake	2020-11-15 21:48:43.070510404 +0000
@@ -1,6 +1,6 @@
 macro(GET_OS_INFO)
     string(REGEX MATCH "Linux" OS_IS_LINUX ${CMAKE_SYSTEM_NAME})
-    set(FLANN_LIB_INSTALL_DIR "lib")
+    set(FLANN_LIB_INSTALL_DIR "lib${LIB_SUFFIX}")
     if(NOT FLANN_INCLUDE_INSTALL_DIR)
       set(FLANN_INCLUDE_INSTALL_DIR
         "include/${PROJECT_NAME_LOWER}-${FLANN_VERSION}")
--- AliceVision-2.3.1/src/dependencies/osi_clp/Clp/src/CMakeLists.txt.orig	2020-10-31 10:30:03.000000000 +0000
+++ AliceVision-2.3.1/src/dependencies/osi_clp/Clp/src/CMakeLists.txt	2020-11-15 21:42:32.738585494 +0000
@@ -36,7 +36,7 @@
 list(REMOVE_ITEM CLP_SRC_CPP ${REMOVE_CPP})
 
 ADD_LIBRARY(lib_clp ${CLP_SRC_HEADER} ${CLP_SRC_CPP})
-INSTALL(TARGETS lib_clp DESTINATION lib EXPORT aliceVision-targets)
+INSTALL(TARGETS lib_clp DESTINATION lib${LIB_SUFFIX} EXPORT aliceVision-targets)
 
 add_subdirectory(OsiClp)
 
--- AliceVision-2.3.1/src/dependencies/osi_clp/Clp/src/OsiClp/CMakeLists.txt.orig	2020-10-31 10:30:03.000000000 +0000
+++ AliceVision-2.3.1/src/dependencies/osi_clp/Clp/src/OsiClp/CMakeLists.txt	2020-11-15 21:53:18.577439145 +0000
@@ -10,4 +10,4 @@
 	*.cpp
 )
 ADD_LIBRARY(lib_OsiClpSolver ${OSI_HEADERS} ${OSI_SRCS})
-INSTALL(TARGETS lib_OsiClpSolver DESTINATION lib EXPORT aliceVision-targets)
+INSTALL(TARGETS lib_OsiClpSolver DESTINATION lib${LIB_SUFFIX} EXPORT aliceVision-targets)
--- AliceVision-2.3.1/src/dependencies/osi_clp/Osi/src/Osi/CMakeLists.txt.orig	2020-10-31 10:30:03.000000000 +0000
+++ AliceVision-2.3.1/src/dependencies/osi_clp/Osi/src/Osi/CMakeLists.txt	2020-11-15 21:41:39.425588037 +0000
@@ -11,4 +11,4 @@
 )
 
 ADD_LIBRARY(lib_Osi ${OSI_OSI_HEADER} ${OSI_OSI_CPP})
-INSTALL(TARGETS lib_Osi DESTINATION lib EXPORT aliceVision-targets)
+INSTALL(TARGETS lib_Osi DESTINATION lib${LIB_SUFFIX} EXPORT aliceVision-targets)
--- AliceVision-2.3.1/src/dependencies/osi_clp/CoinUtils/src/CMakeLists.txt.orig	2020-10-31 10:30:03.000000000 +0000
+++ AliceVision-2.3.1/src/dependencies/osi_clp/CoinUtils/src/CMakeLists.txt	2020-11-15 21:41:02.346589807 +0000
@@ -13,4 +13,4 @@
 
 
 ADD_LIBRARY(lib_CoinUtils ${COINUTILS_HEADER} ${COINUTILS_CPP})
-INSTALL(TARGETS lib_CoinUtils DESTINATION lib EXPORT aliceVision-targets)
+INSTALL(TARGETS lib_CoinUtils DESTINATION lib${LIB_SUFFIX} EXPORT aliceVision-targets)
--- AliceVision-2.3.1/src/dependencies/lemon/lemon/CMakeLists.txt.orig	2020-10-31 10:29:55.000000000 +0000
+++ AliceVision-2.3.1/src/dependencies/lemon/lemon/CMakeLists.txt	2020-11-15 21:44:17.376580501 +0000
@@ -61,7 +61,7 @@
 
 INSTALL(
   TARGETS lemon
-  DESTINATION lib
+  DESTINATION lib${LIB_SUFFIX}
   EXPORT aliceVision-targets
 )
 
--- AliceVision-2.3.1/src/CMakeLists.txt.orig	2020-10-31 10:29:55.000000000 +0000
+++ AliceVision-2.3.1/src/CMakeLists.txt	2020-11-15 21:17:27.819657298 +0000
@@ -901,7 +901,7 @@
               PROPERTY FOLDER AliceVision
               )
 
-      install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/doc  DESTINATION ${CMAKE_INSTALL_DOCDIR})
+      install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/htmlDoc  DESTINATION ${CMAKE_INSTALL_DOCDIR})
 
     endif()
   endif()
--- AliceVision-2.3.1/src/nonFree/sift/CMakeLists.txt.orig	2020-10-31 10:29:55.000000000 +0000
+++ AliceVision-2.3.1/src/nonFree/sift/CMakeLists.txt	2020-11-15 21:51:38.342443928 +0000
@@ -25,7 +25,7 @@
 add_library(vlsift ${FEATS})
 
 install(TARGETS vlsift
-  DESTINATION lib
+  DESTINATION lib${LIB_SUFFIX}
   EXPORT aliceVision-targets
 )
 
