--- alsa-plugins-1.0.26/a52/pcm_a52.c.orig	2012-09-06 08:52:58.000000000 +0100
+++ alsa-plugins-1.0.26/a52/pcm_a52.c	2012-11-09 22:50:36.000000000 +0000
@@ -444,13 +444,13 @@
 #if LIBAVCODEC_VERSION_MAJOR > 52 || (LIBAVCODEC_VERSION_MAJOR == 52 && LIBAVCODEC_VERSION_MINOR >= 3)
 	switch (io->channels) {
 	case 2:
-		rec->avctx->channel_layout = CH_LAYOUT_STEREO;
+		rec->avctx->channel_layout = AV_CH_LAYOUT_STEREO;
 		break;
 	case 4:
-		rec->avctx->channel_layout = CH_LAYOUT_QUAD;
+		rec->avctx->channel_layout = AV_CH_LAYOUT_QUAD;
 		break;
 	case 6:
-		rec->avctx->channel_layout = CH_LAYOUT_5POINT1;
+		rec->avctx->channel_layout = AV_CH_LAYOUT_5POINT1;
 		break;
 	default:
 		break;
