--- antlr-3.4/tool/src/test/java/org/antlr/test/TestFastQueue.java.orig	2011-07-20 19:45:03.000000000 +0100
+++ antlr-3.4/tool/src/test/java/org/antlr/test/TestFastQueue.java	2014-03-23 21:18:06.323536631 +0000
@@ -36,7 +36,7 @@
 
 public class TestFastQueue {
     @Test public void testQueueNoRemove() throws Exception {
-        FastQueue<String> q = new FastQueue<String>();
+        FastQueue q = new FastQueue();
         q.add("a");
         q.add("b");
         q.add("c");
@@ -48,7 +48,7 @@
     }
 
     @Test public void testQueueThenRemoveAll() throws Exception {
-        FastQueue<String> q = new FastQueue<String>();
+        FastQueue q = new FastQueue();
         q.add("a");
         q.add("b");
         q.add("c");
@@ -56,7 +56,7 @@
         q.add("e");
         StringBuffer buf = new StringBuffer();
         while ( q.size()>0 ) {
-            String o = q.remove();
+            String o = (String) q.remove();
             buf.append(o);
             if ( q.size()>0 ) buf.append(" ");
         }
@@ -68,7 +68,7 @@
 
     @Test public void testQueueThenRemoveOneByOne() throws Exception {
         StringBuffer buf = new StringBuffer();
-        FastQueue<String> q = new FastQueue<String>();
+        FastQueue q = new FastQueue();
         q.add("a");
         buf.append(q.remove());
         q.add("b");
@@ -88,7 +88,7 @@
     // E r r o r s
 
     @Test public void testGetFromEmptyQueue() throws Exception {
-        FastQueue<String> q = new FastQueue<String>();
+        FastQueue q = new FastQueue();
         String msg = null;
         try { q.remove(); }
         catch (NoSuchElementException nsee) {
@@ -100,7 +100,7 @@
     }
 
     @Test public void testGetFromEmptyQueueAfterSomeAdds() throws Exception {
-        FastQueue<String> q = new FastQueue<String>();
+        FastQueue q = new FastQueue();
         q.add("a");
         q.add("b");
         q.remove();
@@ -116,7 +116,7 @@
     }
 
     @Test public void testGetFromEmptyQueueAfterClear() throws Exception {
-        FastQueue<String> q = new FastQueue<String>();
+        FastQueue q = new FastQueue();
         q.add("a");
         q.add("b");
         q.clear();
