--- ardour-2.8.14/libs/gtkmm2/gtk/gtkmm/textbuffer.cc.orig	2012-11-25 20:02:28.000000000 +0000
+++ ardour-2.8.14/libs/gtkmm2/gtk/gtkmm/textbuffer.cc	2012-11-25 22:38:10.000000000 +0000
@@ -24,7 +24,7 @@
  */
 
 #include <gtk/gtktextbuffer.h>
-#include <glib/gslist.h>
+#include <glib.h>
 #include <gtkmm/textmark.h>
 #include <gtkmm/textiter.h>
 
--- ardour-2.8.14/libs/glibmm2/glibmm/random.cc.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/random.cc	2012-11-25 20:47:52.000000000 +0000
@@ -21,7 +21,7 @@
  */
 
 #include <glibmm/random.h>
-#include <glib/grand.h>
+#include <glib.h>
 
 
 namespace Glib
--- ardour-2.8.14/libs/glibmm2/glibmm/containerhandle_shared.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/containerhandle_shared.h	2012-11-25 20:12:24.000000000 +0000
@@ -29,7 +29,7 @@
 #include <list>
 
 #include <glib-object.h>
-#include <glib/gmem.h>
+#include <glib.h>
 #include <glibmm/refptr.h>
 #include <glibmm/ustring.h>
 #include <glibmm/wrap.h>
--- ardour-2.8.14/libs/glibmm2/glibmm/thread.cc.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/thread.cc	2012-11-25 20:45:43.000000000 +0000
@@ -24,7 +24,7 @@
  */
 
 #include <glibmm/exceptionhandler.h>
-#include <glib/gmessages.h>
+#include <glib.h>
 
 
 namespace
--- ardour-2.8.14/libs/glibmm2/glibmm/debug.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/debug.h	2012-11-25 20:49:23.000000000 +0000
@@ -21,14 +21,13 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glibmmconfig.h>
 
 // Some stuff that's useful when debugging gtkmm internals:
 
 #ifdef GLIBMM_DEBUG_REFCOUNTING
 
-#include <glib/gmessages.h>
 
 /* We can't use G_GNUC_PRETTY_FUNCTION because it's always disabled in C++,
  * even though __PRETTY_FUNCTION__ works fine in C++ as well if you use it
--- ardour-2.8.14/libs/glibmm2/glibmm/timeval.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/timeval.h	2012-11-25 20:15:42.000000000 +0000
@@ -23,8 +23,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/gtypes.h>
-#include <glib/gtimer.h>
+#include <glib.h>
 
 namespace Glib
 {
--- ardour-2.8.14/libs/glibmm2/glibmm/timer.cc.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/timer.cc	2012-11-25 20:48:06.000000000 +0000
@@ -20,7 +20,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/gtimer.h>
+#include <glib.h>
 #include <glibmm/timer.h>
 
 
--- ardour-2.8.14/libs/glibmm2/glibmm/ustring.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/ustring.h	2012-11-25 20:12:47.000000000 +0000
@@ -21,8 +21,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/gmacros.h>
-#include <glib/gunicode.h>
+#include <glib.h>
 #include <glibmm/unicode.h>
 
 #include <iosfwd>
--- ardour-2.8.14/libs/glibmm2/glibmm/optionentry.cc.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/optionentry.cc	2012-11-25 20:47:17.000000000 +0000
@@ -24,8 +24,7 @@
  */
 
 #include <glibmm/utility.h>
-#include <glib/goption.h>
-#include <glib/gmem.h>
+#include <glib.h>
 
 namespace Glib
 {
--- ardour-2.8.14/libs/glibmm2/glibmm/wrap.cc.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/wrap.cc	2012-11-25 20:39:06.000000000 +0000
@@ -21,8 +21,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
-#include <glib/gmacros.h>
+#include <glib.h>
 
 #include <vector>
 #include <glibmm/object.h>
--- ardour-2.8.14/libs/glibmm2/glibmm/date.cc.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/date.cc	2012-11-25 20:46:16.000000000 +0000
@@ -23,11 +23,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/gmem.h>
-#include <glib/gmessages.h>
-#ifndef g_assert
-#include <glib/gtestutils.h>
-#endif
+#include <glib.h>
 
 #include <ctime>
 #include <algorithm>
--- ardour-2.8.14/libs/glibmm2/glibmm/main.cc.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/main.cc	2012-11-25 20:47:00.000000000 +0000
@@ -24,7 +24,7 @@
 #include <glibmm/wrap.h>
 #include <glibmm/iochannel.h>
 
-#include <glib/gmessages.h>
+#include <glib.h>
 #include <algorithm>
 
 GLIBMM_USING_STD(min)
--- ardour-2.8.14/libs/glibmm2/glibmm/unicode.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/unicode.h	2012-11-25 20:12:41.000000000 +0000
@@ -24,9 +24,7 @@
  */
 
 
-#include <glib/gmacros.h>
-#include <glib/gstrfuncs.h>
-#include <glib/gunicode.h>
+#include <glib.h>
 
 // Not used, but we want to get rid of possible <ctype.h> macros.
 #include <cctype>
--- ardour-2.8.14/libs/glibmm2/glibmm/listhandle.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/listhandle.h	2012-11-25 20:49:40.000000000 +0000
@@ -21,7 +21,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/glist.h>
+#include <glib.h>
 #include <glibmm/containerhandle_shared.h>
 
 
--- ardour-2.8.14/libs/glibmm2/glibmm/utility.cc.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/utility.cc	2012-11-25 20:48:15.000000000 +0000
@@ -20,7 +20,7 @@
  */
 
 #include <glibmm/utility.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 
 
 void Glib::append_canonical_typename(std::string& dest, const char* type_name)
--- ardour-2.8.14/libs/glibmm2/glibmm/error.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/error.h	2012-11-25 20:15:00.000000000 +0000
@@ -24,7 +24,7 @@
 extern "C" { typedef struct _GError GError; }
 #endif
 
-#include <glib/gquark.h>
+#include <glib.h>
 #include <glibmm/exception.h>
 
 
--- ardour-2.8.14/libs/glibmm2/glibmm/shell.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/shell.h	2012-11-25 20:24:02.000000000 +0000
@@ -24,7 +24,7 @@
  */
 
 
-#include <glib/gshell.h>
+#include <glib.h>
 #include <string>
 
 #include <glibmm/arrayhandle.h>
--- ardour-2.8.14/libs/glibmm2/glibmm/slisthandle.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/slisthandle.h	2012-11-25 20:50:04.000000000 +0000
@@ -21,7 +21,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/gslist.h>
+#include <glib.h>
 #include <glibmm/containerhandle_shared.h>
 
 
--- ardour-2.8.14/libs/glibmm2/glibmm/convert.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/convert.h	2012-11-25 20:31:20.000000000 +0000
@@ -24,7 +24,7 @@
  */
 
 
-#include <glib/gtypes.h> /* for gsize */
+#include <glib.h> /* for gsize */
 
 #include <glibmm/error.h>
 #include <glibmm/ustring.h>
--- ardour-2.8.14/libs/glibmm2/glibmm/containers.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/containers.h	2012-11-25 20:16:02.000000000 +0000
@@ -23,8 +23,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/glist.h>
-#include <glib/gslist.h>
+#include <glib.h>
 #include <glibmm/sarray.h> /* for backward compatibility */
 
 #include <iterator>
--- ardour-2.8.14/libs/glibmm2/glibmm/optioncontext.cc.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/optioncontext.cc	2012-11-25 20:35:50.000000000 +0000
@@ -24,7 +24,7 @@
  */
 
 #include <glibmm/utility.h>
-#include <glib/goption.h>
+#include <glib.h>
 
 namespace Glib
 {
--- ardour-2.8.14/libs/glibmm2/glibmm/quark.cc.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/quark.cc	2012-11-25 20:27:56.000000000 +0000
@@ -19,7 +19,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/gquark.h>
+#include <glib.h>
 #include <glibmm/quark.h>
 
 namespace Glib
--- ardour-2.8.14/libs/glibmm2/glibmm/quark.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/quark.h	2012-11-25 20:15:16.000000000 +0000
@@ -22,7 +22,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/gquark.h>
+#include <glib.h>
 #include <glibmm/ustring.h>
 
 
--- ardour-2.8.14/libs/glibmm2/glibmm/utility.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/utility.h	2012-11-25 20:21:40.000000000 +0000
@@ -20,7 +20,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glibmm/ustring.h>
 
 
--- ardour-2.8.14/libs/glibmm2/glibmm/optiongroup.cc.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/optiongroup.cc	2012-11-25 20:47:30.000000000 +0000
@@ -27,7 +27,7 @@
 #include <glibmm/optioncontext.h>
 #include <glibmm/utility.h>
 //#include <glibmm/containers.h>
-#include <glib/goption.h>
+#include <glib.h>
 
 namespace Glib
 {
--- ardour-2.8.14/libs/glibmm2/glibmm/timeval.cc.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/timeval.cc	2012-11-25 20:37:45.000000000 +0000
@@ -20,9 +20,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/gmain.h>
-#include <glib/gmessages.h>
-#include <glib/gtimer.h>
+#include <glib.h>
 
 #include <glibmm/timeval.h>
 
--- ardour-2.8.14/libs/glibmm2/glibmm/main.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/main.h	2012-11-25 20:32:17.000000000 +0000
@@ -21,8 +21,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/giochannel.h>
-#include <glib/gmain.h>
+#include <glib.h>
 
 #include <vector>
 #include <sigc++/sigc++.h>
--- ardour-2.8.14/libs/glibmm2/glibmm/convert.cc.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/convert.cc	2012-11-25 20:42:05.000000000 +0000
@@ -23,12 +23,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/gconvert.h>
-#include <glib/gmessages.h>
-#include <glib/gunicode.h>
-#ifndef g_assert
-#include <glib/gtestutils.h>
-#endif
+#include <glib.h>
 #include <glibmm/utility.h>
 
 
--- ardour-2.8.14/libs/glibmm2/glibmm/date.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/date.h	2012-11-25 20:31:46.000000000 +0000
@@ -26,8 +26,7 @@
 
 #include <glibmm/ustring.h>
 
-#include <glib/gdate.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 #ifndef DOXYGEN_SHOULD_SKIP_THIS
 extern "C" { struct tm; }
--- ardour-2.8.14/libs/glibmm2/glibmm/iochannel.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/iochannel.h	2012-11-25 20:32:31.000000000 +0000
@@ -29,7 +29,7 @@
 #include <glibmm/main.h>
 #include <glibmm/refptr.h>
 #include <glibmm/ustring.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 #include <string>
 #include <glibmmconfig.h>
--- ardour-2.8.14/libs/glibmm2/glibmm/fileutils.cc.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/fileutils.cc	2012-11-25 20:18:32.000000000 +0000
@@ -23,9 +23,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/gdir.h>
-#include <glib/gfileutils.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <glibmm/utility.h>
 
 
--- ardour-2.8.14/libs/glibmm2/glibmm/thread.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/thread.h	2012-11-25 20:14:35.000000000 +0000
@@ -24,7 +24,7 @@
  */
 
 
-#include <glib/gthread.h>
+#include <glib.h>
 #include <cstddef>
 
 #include <sigc++/sigc++.h>
--- ardour-2.8.14/libs/glibmm2/glibmm/pattern.cc.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/pattern.cc	2012-11-25 20:47:41.000000000 +0000
@@ -20,7 +20,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/gpattern.h>
+#include <glib.h>
 #include <glibmm/pattern.h>
 
 
--- ardour-2.8.14/libs/glibmm2/glibmm/error.cc.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/error.cc	2012-11-25 20:29:49.000000000 +0000
@@ -20,11 +20,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/gerror.h>
-#include <glib/gmessages.h>
-#ifndef g_assert
-#include <glib/gtestutils.h>
-#endif
+#include <glib.h>
 
 #include <map>
 #include <glibmmconfig.h>
--- ardour-2.8.14/libs/glibmm2/glibmm/spawn.cc.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/spawn.cc	2012-11-25 20:26:48.000000000 +0000
@@ -23,7 +23,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/gspawn.h>
+#include <glib.h>
 #include <glibmm/exceptionhandler.h>
 #include <glibmm/utility.h>
 
--- ardour-2.8.14/libs/glibmm2/glibmm/random.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/random.h	2012-11-25 20:49:54.000000000 +0000
@@ -23,7 +23,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 extern "C" { typedef struct _GRand GRand; }
 
--- ardour-2.8.14/libs/glibmm2/glibmm/exception.cc.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/exception.cc	2012-11-25 20:46:35.000000000 +0000
@@ -20,10 +20,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <glib/gmessages.h>
-#ifndef g_assert
-#include <glib/gtestutils.h>
-#endif
+#include <glib.h>
 #include <glibmm/exception.h>
 
 
--- ardour-2.8.14/libs/glibmm2/glibmm/optiongroup.h.orig	2012-11-25 20:02:20.000000000 +0000
+++ ardour-2.8.14/libs/glibmm2/glibmm/optiongroup.h	2012-11-25 20:32:53.000000000 +0000
@@ -27,7 +27,7 @@
 #include <glibmm/ustring.h>
 #include <map>
 #include <vector>
-#include <glib/goption.h> //TODO: Try to hide this.
+#include <glib.h> //TODO: Try to hide this.
 
 
 #ifndef DOXYGEN_SHOULD_SKIP_THIS
--- ardour-2.8.14/gtk2_ardour/SConscript.orig	2012-11-25 20:02:14.000000000 +0000
+++ ardour-2.8.14/gtk2_ardour/SConscript	2012-11-26 19:50:32.000000000 +0000
@@ -23,6 +23,7 @@
 #gtkardour.Append(CXXFLAGS="-DFLOWCANVAS_AA")
 gtkardour.Append(PACKAGE=domain)
 gtkardour.Append(POTFILE=domain + '.pot')
+gtkardour.Append(LINKFLAGS="-lboost_system-mt")
 
 if gtkardour['IS_OSX']:
 	gtkardour.Append (LINKFLAGS="-Xlinker -headerpad -Xlinker 2048")
