diff -Nur ardour-4.4/libs/ardour/config_text.cc Ardour-4.4.0/libs/ardour/config_text.cc
--- ardour-4.4/libs/ardour/config_text.cc	1970-01-01 01:00:00.000000000 +0100
+++ Ardour-4.4.0/libs/ardour/config_text.cc	2015-10-16 00:19:29.000000000 +0100
@@ -0,0 +1,52 @@
+#include "ardour/ardour.h"
+namespace ARDOUR {
+const char* const ardour_config_info = "\n\
+Build documentation: False\n\
+Debuggable build: True\n\
+Export all symbols (backtrace): False\n\
+Install prefix: /usr/local\n\
+Strict compiler flags: []\n\
+Internal Shared Libraries: True\n\
+Use External Libraries: False\n\
+Library exports hidden: True\n\
+ALSA Backend: False\n\
+ALSA DBus Reservation: True\n\
+Architecture flags: None\n\
+Aubio: True\n\
+AudioUnits: False\n\
+Free/Demo copy: False\n\
+Build target: x86_64\n\
+CoreAudio: False\n\
+CoreAudio/Midi Backend: False\n\
+Debug RT allocations: False\n\
+Debug Symbols: True\n\
+Dummy backend: False\n\
+Process thread timing: False\n\
+Denormal exceptions: False\n\
+FLAC: True\n\
+FPU optimization: True\n\
+Freedesktop files: False\n\
+JACK Backend: True\n\
+Libjack linking: link\n\
+Libjack metadata: True\n\
+LV2 UI embedding: True\n\
+LV2 support: True\n\
+LXVST support: True\n\
+OGG: True\n\
+Phone home: True\n\
+Program name: Ardour\n\
+Samplerate: True\n\
+PT format: False\n\
+Translation: True\n\
+Unit tests: False\n\
+Mac i386 Architecture: False\n\
+Mac ppc Architecture: False\n\
+Waves Backend: False\n\
+Windows VST support: False\n\
+Wiimote support: True\n\
+Windows key: Mod4><Super\n\
+C compiler flags: ['-I/usr/local/music/src/ardour/4.0', '-DWAF_BUILD', '-O0', '-g', '-std=c99', '-pedantic', '-Wshadow', '-Wall', '-Wcast-align', '-Wextra', '-Wwrite-strings', '-Wunsafe-loop-optimizations', '-Wlogical-op', '-fshow-column', '-DARCH_X86', '-mmmx', '-msse', '-mfpmath=sse', '-DUSE_XMMINTRIN', '-DBUILD_SSE_OPTIMIZATIONS', '-DLXVST_64BIT', '-Wall', '-Wpointer-arith', '-Wcast-qual', '-Wcast-align', '-Wno-unused-parameter', '-DBOOST_SYSTEM_NO_DEPRECATED', '-D_ISOC9X_SOURCE', '-D_LARGEFILE64_SOURCE', '-D_FILE_OFFSET_BITS=64', '-DENABLE_NLS', '-DPROGRAM_NAME=\"Ardour\"', u'-DPROGRAM_VERSION=\"4\"', '-Wstrict-prototypes', '-Wmissing-prototypes']\n\
+C++ compiler flags: ['-I/usr/local/music/src/ardour/4.0', '-DWAF_BUILD', '-O0', '-g', '-ansi', '-Wnon-virtual-dtor', '-Woverloaded-virtual', '-fstrict-overflow', '-Wall', '-Wcast-align', '-Wextra', '-Wwrite-strings', '-Wunsafe-loop-optimizations', '-Wlogical-op', '-fshow-column', '-DARCH_X86', '-mmmx', '-msse', '-mfpmath=sse', '-DUSE_XMMINTRIN', '-DBUILD_SSE_OPTIMIZATIONS', '-DLXVST_64BIT', '-Wall', '-Wpointer-arith', '-Wcast-qual', '-Wcast-align', '-Wno-unused-parameter', '-DBOOST_SYSTEM_NO_DEPRECATED', '-D_ISOC9X_SOURCE', '-D_LARGEFILE64_SOURCE', '-D_FILE_OFFSET_BITS=64', '-DENABLE_NLS', '-DPROGRAM_NAME=\"Ardour\"', u'-DPROGRAM_VERSION=\"4\"', '-Woverloaded-virtual', '-Wno-unused-local-typedefs', '-D__STDC_LIMIT_MACROS', '-D__STDC_FORMAT_MACROS', '-DCANVAS_COMPATIBILITY', '-DCANVAS_DEBUG']\n\
+Linker flags: ['']\n\
+";
+}
diff -Nur ardour-4.4/libs/ardour/revision.cc Ardour-4.4.0/libs/ardour/revision.cc
--- ardour-4.4/libs/ardour/revision.cc	1970-01-01 01:00:00.000000000 +0100
+++ Ardour-4.4.0/libs/ardour/revision.cc	2015-10-16 00:18:31.000000000 +0100
@@ -0,0 +1,2 @@
+#include "ardour/revision.h"
+namespace ARDOUR { const char* revision = "4.4"; }
diff -Nur ardour-4.4/mcp/faderport.device Ardour-4.4.0/mcp/faderport.device
--- ardour-4.4/mcp/faderport.device	1970-01-01 01:00:00.000000000 +0100
+++ Ardour-4.4.0/mcp/faderport.device	2015-10-03 16:38:52.000000000 +0100
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<MackieProtocolDevice>
+  <Name value="Presonus Faderport"/>
+  <DeviceType value="HUI"/>
+  <Strips value="1"/>
+  <MasterFader value="no"/>
+  <TimecodeDisplay value="no"/>
+  <TwoCharacterDisplay value="no"/>
+  <Extenders value="0"/>
+  <MasterPosition value="0"/>
+  <GlobalControls value="yes"/>
+  <JogWheel value="no"/>
+  <TouchSenseFaders value="yes"/>
+</MackieProtocolDevice>
