--- armagetronad-0.2.8.3.1/configure.orig	2010-02-20 17:42:27.000000000 +0000
+++ armagetronad-0.2.8.3.1/configure	2010-06-19 13:34:10.000000000 +0100
@@ -14513,9 +14513,9 @@
 CPPFLAGS="$CPPFLAGS `libpng-config --cflags`"
 LIBS="$LIBS $PNGLIBS"
 
-{ echo "$as_me:$LINENO: checking for png_check_sig" >&5
-echo $ECHO_N "checking for png_check_sig... $ECHO_C" >&6; }
-if test "${ac_cv_func_png_check_sig+set}" = set; then
+{ echo "$as_me:$LINENO: checking for png_sig_cmp" >&5
+echo $ECHO_N "checking for png_sig_cmp... $ECHO_C" >&6; }
+if test "${ac_cv_func_png_sig_cmp+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -14524,12 +14524,12 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-/* Define png_check_sig to an innocuous variant, in case <limits.h> declares png_check_sig.
+/* Define png_sig_cmp to an innocuous variant, in case <limits.h> declares png_sig_cmp.
    For example, HP-UX 11i <limits.h> declares gettimeofday.  */
-#define png_check_sig innocuous_png_check_sig
+#define png_sig_cmp innocuous_png_sig_cmp
 
 /* System header to define __stub macros and hopefully few prototypes,
-    which can conflict with char png_check_sig (); below.
+    which can conflict with char png_sig_cmp (); below.
     Prefer <limits.h> to <assert.h> if __STDC__ is defined, since
     <limits.h> exists even on freestanding compilers.  */
 
@@ -14539,7 +14539,7 @@
 # include <assert.h>
 #endif
 
-#undef png_check_sig
+#undef png_sig_cmp
 
 /* Override any GCC internal prototype to avoid an error.
    Use char because int might match the return type of a GCC
@@ -14547,18 +14547,18 @@
 #ifdef __cplusplus
 extern "C"
 #endif
-char png_check_sig ();
+char png_sig_cmp ();
 /* The GNU C library defines this for functions which it implements
     to always fail with ENOSYS.  Some functions are actually named
     something starting with __ and the normal name is an alias.  */
-#if defined __stub_png_check_sig || defined __stub___png_check_sig
+#if defined __stub_png_sig_cmp || defined __stub___png_sig_cmp
 choke me
 #endif
 
 int
 main ()
 {
-return png_check_sig ();
+return png_sig_cmp ();
   ;
   return 0;
 }
@@ -14581,20 +14581,20 @@
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext &&
        $as_test_x conftest$ac_exeext; then
-  ac_cv_func_png_check_sig=yes
+  ac_cv_func_png_sig_cmp=yes
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
 
-	ac_cv_func_png_check_sig=no
+	ac_cv_func_png_sig_cmp=no
 fi
 
 rm -f core conftest.err conftest.$ac_objext conftest_ipa8_conftest.oo \
       conftest$ac_exeext conftest.$ac_ext
 fi
-{ echo "$as_me:$LINENO: result: $ac_cv_func_png_check_sig" >&5
-echo "${ECHO_T}$ac_cv_func_png_check_sig" >&6; }
-if test $ac_cv_func_png_check_sig = yes; then
+{ echo "$as_me:$LINENO: result: $ac_cv_func_png_sig_cmp" >&5
+echo "${ECHO_T}$ac_cv_func_png_sig_cmp" >&6; }
+if test $ac_cv_func_png_sig_cmp = yes; then
   :
 else
   { { echo "$as_me:$LINENO: error: You need libpng to compile Armagetron." >&5
