--- astyle-3.4.10/build/cmake/InstallOptions.cmake.orig	2023-10-21 17:32:33.000000000 +0100
+++ astyle-3.4.10/build/cmake/InstallOptions.cmake	2023-12-08 17:46:27.815961028 +0000
@@ -5,7 +5,7 @@
 # Default Win32 install prefix is not used (C:/Program Files (x86))
 if(BUILD_SHARED_LIBS OR BUILD_STATIC_LIBS)
     if(NOT WIN32)
-        install(TARGETS astyle DESTINATION lib)
+        install(TARGETS astyle DESTINATION lib${LIB_SUFFIX})
     endif()
 else()
     if(WIN32)
