--- asymptote-2.73/LspCpp/Makefile.orig	2021-12-28 00:50:40.000000000 +0000
+++ asymptote-2.73/LspCpp/Makefile	2021-12-29 12:04:00.152719674 +0000
@@ -3,7 +3,7 @@
 CFLAGS = -Wall -fPIE
 INCLUDES=-I. -ILibLsp/lsp/extention/jdtls/ -ILibLsp/JsonRpc/ -ILibLsp/JsonRpc/lsp/extention/jdtls \
 	-Ithird_party/threadpool -Ithird_party/utfcpp/source -Ithird_party/rapidjson/include
-CXXFLAGS = -std=c++14
+CXXFLAGS =
 OPTFLAGS = -O3
 
 ALL_CXXFLAGS = $(CFLAGS) $(CXXFLAGS) $(CPPFLAGS) $(INCLUDES) $(OPTFLAGS)
--- asymptote-2.73/configure.orig	2021-12-28 00:50:43.000000000 +0000
+++ asymptote-2.73/configure	2021-12-29 12:15:42.071686183 +0000
@@ -5948,7 +5948,6 @@
    _|_system|_*[\\/]*)
       if test "x$ac_cv_use_gc" = "xsystem" ; then
          INCL="-I. -I$prefix/include/gc -I/usr/include/gc"
-         LIBS=$LIBS"-L$prefix/lib "
       else
          INCL="-I$ac_cv_use_gc/include/gc"
          LIBS=$LIBS"-L$ac_cv_use_gc/lib "
