--- atlas-3.10.0/makes/Make.lib.orig	2012-07-10 20:22:50.000000000 +0100
+++ atlas-3.10.0/makes/Make.lib	2012-07-17 20:30:34.000000000 +0100
@@ -112,12 +112,12 @@
 # These are a bunch of different ways to attempt to build a .so, try them all
 #
 LDTRY:
-	$(LD) $(LDFLAGS) -shared -soname $(LIBINSTdir)/$(outso) -o $(outso) \
+	$(LD) $(LDFLAGS) -shared -soname $(outso) -o $(outso) \
            -rpath-link $(LIBINSTdir)  \
            --whole-archive $(libas) --no-whole-archive $(LIBS)
 GCCTRY:
 	$(GOODGCC) -shared -o $(outso)  \
-           -Wl,"rpath-link $(LIBINSTdir)" \
+           -Wl,"-rpath-link $(LIBINSTdir)" \
            -Wl,--whole-archive $(libas) -Wl,--no-whole-archive $(LIBS)
 GCCTRY_norp:
 	$(GOODGCC) -shared -o $(outso)  \
