--- atlas-3.9.52/makes/Make.lib.orig	2011-09-29 17:32:55.000000000 +0100
+++ atlas-3.9.52/makes/Make.lib	2011-10-16 06:09:16.387918501 +0100
@@ -46,11 +46,11 @@
 # These are a bunch of different ways to attempt to build a .so, try them all
 #
 LDTRY:
-	$(LD) $(LDFLAGS) -shared -soname $(LIBINSTdir)/$(outso) -o $(outso) \
+	$(LD) $(LDFLAGS) -shared -soname $(outso) -o $(outso) \
            -rpath-link $(LIBINSTdir) \
            --whole-archive $(libas) --no-whole-archive $(LIBS)
 GCCTRY:
-	$(GOODGCC) -shared -o $(outso) -Wl,"rpath-link $(LIBINSTdir)" \
+	$(GOODGCC) -shared -o $(outso) -Wl,"-rpath-link,$(LIBINSTdir)" \
            -Wl,--whole-archive $(libas) -Wl,--no-whole-archive $(LIBS)
 #
 # TRYALL is going to just try a bunch of library combinations that may work
