--- atlas-3.10.3/makes/Make.Flp.orig	2016-07-28 20:42:59.000000000 +0100
+++ atlas-3.10.3/makes/Make.Flp	2021-12-13 19:46:24.089867020 +0000
@@ -746,11 +746,11 @@
 $(FLAdir)/make.inc : $(makeincex)
 	rm -f $(FLAdir)/make.inc
 	cat $(makeincex) | sed -e 's@PLAT *=.*@PLAT = _$(ARCH)@' \
-         -e 's@^FORTRAN *=.*@FORTRAN = $(F77)@' \
-         -e 's@^OPTS *=.*@OPTS = $(F77FLAGS)@' \
-         -e 's@^NOOPT *=.*@NOOPT = $(F77FLAGS) $(F77NOOPT)@' \
+         -e 's@^FC *=.*@FC = $(F77)@' \
+         -e 's@^FFLAGS *=.*@FFLAGS = $(F77FLAGS)@' \
+         -e 's@^FFLAGS_NOOPT *=.*@FFLAGS_NOOPT = $(F77FLAGS) $(F77NOOPT)@' \
          -e 's@^LOADER *=.*@LOADER = $(F77)@' \
-         -e 's@^LOADOPTS *=.*@LOADOPTS = $(F77FLAGS)@' \
+         -e 's@^LDFLAGS *=.*@LDFLAGS = $(F77FLAGS)@' \
          -e 's@^TIMER *=.*@TIMER = NONE@' \
          > $(FLAdir)/make.inc
 
