--- audacity-3.4.0/libraries/lib-files/PathList.cpp.orig	2023-11-02 11:22:32.000000000 +0000
+++ audacity-3.4.0/libraries/lib-files/PathList.cpp	2023-11-04 21:36:31.886883993 +0000
@@ -72,7 +72,7 @@
 
    FileNames::AddUniquePathToPathList(progPath, audacityPathList);
    // Add the path to modules:
-   FileNames::AddUniquePathToPathList(progPath + L"/lib/audacity", audacityPathList);
+   FileNames::AddUniquePathToPathList(progPath + L"/lib64/audacity", audacityPathList);
 
 #if !defined(__WXMSW__)
    // On Unix systems, the common directory structure is
@@ -82,8 +82,8 @@
 
    if (!progParentPath.IsEmpty())
    {
-      FileNames::AddUniquePathToPathList(progParentPath + L"/lib/audacity", audacityPathList);
-      FileNames::AddUniquePathToPathList(progParentPath + L"/lib", audacityPathList);
+      FileNames::AddUniquePathToPathList(progParentPath + L"/lib64/audacity", audacityPathList);
+      FileNames::AddUniquePathToPathList(progParentPath + L"/lib64", audacityPathList);
    }
 #endif
 
--- audacity-3.4.0/cmake-proxies/cmake-modules/Findopusfile.cmake.orig	2023-11-02 11:22:32.000000000 +0000
+++ audacity-3.4.0/cmake-proxies/cmake-modules/Findopusfile.cmake	2023-11-04 22:05:09.033515486 +0000
@@ -24,7 +24,7 @@
          add_library( opusfile::opusfile INTERFACE IMPORTED GLOBAL )
 
          target_include_directories( opusfile::opusfile INTERFACE ${opusfile_INCLUDE_DIR} "${opusfile_INCLUDE_DIR}/opus" )
-         target_link_libraries( opusfile::opusfile INTERFACE ${Oopusfile_LIBRARIES} )
+         target_link_libraries( opusfile::opusfile INTERFACE ${opusfile_LIBRARIES} )
       endif()
 
       mark_as_advanced(
