--- audacity-minsrc-2.4.1/cmake-proxies/wxWidgets/CMakeLists.txt.orig	2020-05-19 13:02:06.000000000 +0100
+++ audacity-minsrc-2.4.1/cmake-proxies/wxWidgets/CMakeLists.txt	2020-05-23 11:44:01.504321278 +0100
@@ -142,6 +142,11 @@
       list( REMOVE_DUPLICATES LIBRARIES )
    endforeach()
 
+   list( REMOVE_ITEM INCLUDES "value-NOTFOUND" )
+   list( REMOVE_ITEM DEFINES "value-NOTFOUND" )
+   list( REMOVE_ITEM COPTS "value-NOTFOUND" )
+   list( REMOVE_ITEM LIBRARIES "value-NOTFOUND" )
+
    # Do NOT split the generator expressions across multiple lines here.
    # CMake appears to have a bug and doesn't seem to handle it correctly
    # for target link libraries.
