--- audacity-minsrc-1.3.14/src/FFmpeg.h.orig	2011-12-08 08:13:13.000000000 +0000
+++ audacity-minsrc-1.3.14/src/FFmpeg.h	2011-12-23 16:53:36.000000000 +0000
@@ -505,12 +505,12 @@
       (AVCodecContext *avctx, AVCodec *codec),
       (avctx, codec);
    );
-   FFMPEG_FUNCTION_WITH_RETURN(
-      int,
-      avcodec_decode_audio2,
-      (AVCodecContext *avctx, int16_t *samples, int *frame_size_ptr, const uint8_t *buf, int buf_size),
-      (avctx, samples, frame_size_ptr, buf, buf_size)
-   );
+//   FFMPEG_FUNCTION_WITH_RETURN(
+//      int,
+//      avcodec_decode_audio2,
+//      (AVCodecContext *avctx, int16_t *samples, int *frame_size_ptr, const uint8_t *buf, int buf_size),
+//      (avctx, samples, frame_size_ptr, buf, buf_size)
+//   );
    FFMPEG_FUNCTION_WITH_RETURN(
       int,
       avcodec_decode_audio3,
@@ -559,7 +559,7 @@
    FFMPEG_FUNCTION_WITH_RETURN(
       void*,
       av_fast_realloc,
-      (void *ptr, unsigned int *size, unsigned int min_size),
+      (void *ptr, unsigned int *size, size_t min_size),
       (ptr, size, min_size)
    );
    FFMPEG_FUNCTION_WITH_RETURN(
@@ -747,7 +747,7 @@
    FFMPEG_FUNCTION_WITH_RETURN(
       void*,
       av_malloc,
-      (unsigned int size),
+      (size_t size),
       (size)
    );
    FFMPEG_FUNCTION_NO_RETURN(
