--- autopano-sift-C-2.5.1/CMakeLists.txt.orig	2009-10-06 21:51:08.000000000 +0100
+++ autopano-sift-C-2.5.1/CMakeLists.txt	2012-06-23 08:53:29.000000000 +0100
@@ -121,7 +121,7 @@
 
 ADD_LIBRARY(libsift STATIC ${LIBSIFT_SOURCES})
 
-set(all_libs libsift ${JPEG_LIBRARIES} ${TIFF_LIBRARIES}
+set(all_libs libsift libm.so ${JPEG_LIBRARIES} ${TIFF_LIBRARIES}
     ${PNG_LIBRARIES} ${ZLIB_LIBRARIES} ${PANO13_LIBRARIES} ${LIBXML2_LIBRARIES}) 
 
 add_executable(generatekeys GenerateKeys.c)
