--- autotrace-0.31.1/configure.in.orig	2002-11-08 17:49:23.000000000 +0000
+++ autotrace-0.31.1/configure.in	2016-07-16 16:57:30.881582092 +0100
@@ -53,23 +53,23 @@
   --without-magick  compile without maigck],,with_magick=yes)
 
 if test $with_magick = yes ; then
-	AC_CHECK_PROG(MAGICK_CONFIG, Magick-config, Magick-config, "")
-	if test "x$MAGICK_CONFIG" = "xMagick-config" ; then
+	AC_CHECK_PROG(MAGICK_CONFIG, MagickWand-config, MagickWand-config, "")
+	if test "x$MAGICK_CONFIG" = "xMagickWand-config" ; then
 		CPPFLAGS_BACKUP="$CPPFLAGS"
-		CPPFLAGS=`Magick-config --cppflags`
+		CPPFLAGS=`MagickWand-config --cppflags`
 		AC_CHECK_HEADER(magick/api.h,
 				magick_header_found=yes,
 				magick_header_found=no)
 		MAGICK_CFLAGS=${CPPFLAGS}
 		CPPFLAGS=${CPPFLAGS_BACKUP}
 		if test "x${magick_header_found}" = xyes ; then
-			MAGICK_LDFLAGS="`Magick-config --ldflags` `Magick-config --libs`"
+			MAGICK_LDFLAGS="`MagickWand-config --ldflags` `MagickWand-config --libs`"
 			dnl
 			dnl Checking version
 			dnl
 			AC_MSG_CHECKING([ImageMagick - version >= 5.2.1])
 			dnl Get version and split it into MAJOR, MINOR and MICRO
-			MAGICK_VERSION=`Magick-config --version`
+			MAGICK_VERSION=`MagickWand-config --version`
 			changequote(,)dnl
 			MAGICK_MAJOR_VERSION=`echo $MAGICK_VERSION \
 			| sed -e 's/^\([0-9]*\).[0-9]*.[0-9]*$/\1/'`
@@ -95,12 +95,12 @@
 				AC_DEFINE(HAVE_MAGICK)
 			fi
 		else
-			AC_MSG_WARN([*** Magick-config is found but magick/api.h is not found in $MAGICK_CFLAGS ***])
- 			AC_MSG_WARN([*** Check Magick-config.                                                   ***])
+			AC_MSG_WARN([*** MagickWand-config is found but magick/api.h is not found in $MAGICK_CFLAGS ***])
+ 			AC_MSG_WARN([*** Check MagickWand-config.                                                   ***])
 			AC_MSG_WARN([*** ImageMagick input handler is disabled.                                 ***])
 		fi
 	else
-		AC_MSG_WARN([*** Magick-config is not found.                      ***])
+		AC_MSG_WARN([*** MagickWand-config is not found.                      ***])
 		AC_MSG_WARN([*** ImageMagick input handler is disabled.           ***])		
 	fi
 fi
