--- configure.orig	2002-11-08 18:18:46.000000000 +0000
+++ configure	2005-07-29 14:55:00.000000000 +0100
@@ -7977,34 +7977,30 @@
 												echo "$as_me:$LINENO: checking ImageMagick - version >= 5.2.1" >&5
 echo $ECHO_N "checking ImageMagick - version >= 5.2.1... $ECHO_C" >&6
 						MAGICK_VERSION=`Magick-config --version`
-						MAGICK_MAJOR_VERSION=`echo $MAGICK_VERSION \
-			| sed -e 's/^\([0-9]*\).[0-9]*.[0-9]*$/\1/'`
-			MAGICK_MINOR_VERSION=`echo $MAGICK_VERSION \
-			| sed -e 's/^[0-9]*.\([0-9]*\).[0-9]*$/\1/'`
-			MAGICK_MICRO_VERSION=`echo $MAGICK_VERSION \
-			| sed -e 's/^[0-9]*.[0-9]*.\([0-9]*\)$/\1/'`
-						MAGICK_MAJOR_VERSION_CHECK=`expr $MAGICK_MAJOR_VERSION \>= 5`
-			MAGICK_MINOR_VERSION_CHECK=`expr $MAGICK_MINOR_VERSION \>= 2`
-						if test "$MAGICK_VERSION" = "5.2.0" \
-			   || test "$MAGICK_MAJOR_VERSION_CHECK" = 0 \
-			   || test "$MAGICK_MINOR_VERSION_CHECK" = 0 ; then
-			   echo "$as_me:$LINENO: result: no($MAGICK_VERSION)" >&5
+
+                                                                        case $MAGICK_VERSION in
+                          [01234] | [01234].* | 5.[01]* | 5.2.0* )
+
+                                echo "$as_me:$LINENO: result: no($MAGICK_VERSION)" >&5
 echo "${ECHO_T}no($MAGICK_VERSION)" >&6
-		   	   { echo "$as_me:$LINENO: WARNING: *** Version 5.2.1 or higher is required.             ***" >&5
+                                { echo "$as_me:$LINENO: WARNING: *** Version 5.2.1 or higher is required.             ***" >&5
 echo "$as_me: WARNING: *** Version 5.2.1 or higher is required.             ***" >&2;}
-			   { echo "$as_me:$LINENO: WARNING: *** ImageMagick input handler is disabled.           ***" >&5
+                                { echo "$as_me:$LINENO: WARNING: *** ImageMagick input handler is disabled.           ***" >&5
 echo "$as_me: WARNING: *** ImageMagick input handler is disabled.           ***" >&2;}
-			else
- 		   		echo "$as_me:$LINENO: result: yes($MAGICK_VERSION)" >&5
+                                ;;
+                           *)
+                                echo "$as_me:$LINENO: result: yes($MAGICK_VERSION)" >&5
 echo "${ECHO_T}yes($MAGICK_VERSION)" >&6
 
 
-				HAVE_MAGICK=yes
-				cat >>confdefs.h <<\_ACEOF
+                                HAVE_MAGICK=yes
+
+cat >>confdefs.h <<\_ACEOF
 #define HAVE_MAGICK 1
 _ACEOF
 
-			fi
+                                ;;
+                        esac
 		else
 			{ echo "$as_me:$LINENO: WARNING: *** Magick-config is found but magick/api.h is not found in $MAGICK_CFLAGS ***" >&5
 echo "$as_me: WARNING: *** Magick-config is found but magick/api.h is not found in $MAGICK_CFLAGS ***" >&2;}
