From 8ffe588e53ccdd8167421c29c80b3f02173df4be Mon Sep 17 00:00:00 2001
From: Vlad Rico <vlad.infosec@gmail.com>
Date: Mon, 18 Sep 2023 00:50:37 +0200
Subject: [PATCH] Adding -B ssh option to getopt

---
 autossh.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/autossh.c b/autossh.c
index e2e3b84..950fd51 100644
--- a/autossh.c
+++ b/autossh.c
@@ -109,7 +109,7 @@ const char *rcsid = "$Id: autossh.c,v 1.91 2019/01/05 01:23:39 harding Exp $";
 
 #define N_FAST_TRIES    5       /* try this many times fast before slowing */
 
-#define	OPTION_STRING "M:V1246ab:c:e:fgi:kl:m:no:p:qstvw:xyACD:E:F:GI:MJKL:NO:PQ:R:S:TW:XY"
+#define	OPTION_STRING "M:V1246ab:c:e:fgi:kl:m:no:p:qstvw:xyACD:E:F:GI:MJKL:NO:PQ:R:S:TW:XYB:"
 
 int	logtype  = L_SYSLOG;	/* default log to syslog */
 int	loglevel = LOG_INFO;	/* default loglevel */
