Index: avidemux/ADM_UIs/ADM_GTK/src/FAC_menu.cpp
===================================================================
--- avidemux/ADM_UIs/ADM_GTK/src/FAC_menu.cpp	(revision 6713)
+++ avidemux/ADM_UIs/ADM_GTK/src/FAC_menu.cpp	(working copy)
@@ -183,7 +183,7 @@
 {
   GtkWidget *widget=(GtkWidget *)myWidget;
   int rank = gtk_combo_box_get_active(GTK_COMBO_BOX(widget));
-  
+  if(rank==-1) rank=0; 
   if (!nbMenu)
 	  return;
 
@@ -226,6 +226,7 @@
   
   
   rank=gtk_combo_box_get_active(GTK_COMBO_BOX(widget));
+  if(-1==rank) rank=0;
   ADM_assert(rank<this->nbMenu);
   val=this->menu[rank]->val;
   /* Now search through the linked list to see if something happens ...*/
