--- avidemux-2.6.8/avidemux_plugins/ADM_videoEncoder/x264/qt4/Q_x264.cpp.orig	2014-03-12 06:15:23.000000000 +0000
+++ avidemux-2.6.8/avidemux_plugins/ADM_videoEncoder/x264/qt4/Q_x264.cpp	2014-10-12 18:31:51.717805616 +0100
@@ -77,6 +77,8 @@
     {64,45},
     {8,9},
     {32,27},
+    {16,11},
+    {32,17},
     
 };
 
--- avidemux-2.6.8/avidemux_plugins/ADM_videoEncoder/x264/qt4/x264ConfigDialog.ui.orig	2014-03-12 06:15:23.000000000 +0000
+++ avidemux-2.6.8/avidemux_plugins/ADM_videoEncoder/x264/qt4/x264ConfigDialog.ui	2014-10-12 18:32:19.097807562 +0100
@@ -3545,6 +3545,16 @@
                   <string>32:27 (NTSC 16:9)</string>
                  </property>
                 </item>
+                <item>
+                 <property name="text">
+                  <string>16:11 (DTB 16:9)</string>
+                 </property>
+                </item>
+                <item>
+                 <property name="text">
+                  <string>32:17 (DTB 16:9)</string>
+                 </property>
+                </item>
                </widget>
               </item>
               <item>
