--- avidemux-2.4.3/avidemux/ADM_codecs/ADM_x264.cpp.orig	2008-07-21 18:24:20.000000000 +0100
+++ avidemux-2.4.3/avidemux/ADM_codecs/ADM_x264.cpp	2008-10-06 22:11:00.000000000 +0100
@@ -109,8 +109,8 @@
   MKPARAM(i_bframe,MaxBFrame);
   MKPARAM(i_bframe_bias,Bias);
   MKPARAM( b_bframe_pyramid,BasReference );
-  MKPARAM(analyse. b_bidir_me,BidirME );
-  MKPARAM( b_bframe_adaptive, Adaptative);
+//  MKPARAM(analyse. b_bidir_me,BidirME );
+  MKPARAM( i_bframe_adaptive, Adaptative);
   MKPARAM( analyse.b_weighted_bipred, Weighted);
   MKPARAM( b_cabac , CABAC);
   MKPARAM( analyse.i_trellis, Trellis);
@@ -123,7 +123,7 @@
       parity=(zparam->PartitionDecision+1-MIN_RDO)&1;
       
       param.analyse.i_subpel_refine=rank;
-      param.analyse.b_bframe_rdo=parity;
+//      param.analyse.b_bframe_rdo=parity;
   }
   MKPARAM(analyse.b_chroma_me,ChromaME);
   MKPARAM(b_deblocking_filter,DeblockingFilter);
