--- avidemux-20250101/avidemux_plugins/ADM_videoEncoder/x265/ADM_x265Setup.cpp.orig	2024-07-19 12:37:29.000000000 +0100
+++ avidemux-20250101/avidemux_plugins/ADM_videoEncoder/x265/ADM_x265Setup.cpp	2025-01-01 10:44:26.582638352 +0000
@@ -216,13 +216,21 @@
                         {
                              param.rc.bStatWrite=1;
                              param.rc.bStatRead=0;
+#if X265_BUILD < 215
                              param.rc.statFileName=strdup(logFile);
+#else
+                             strncat(param.rc.statFileName, logFile, X265_MAX_STRING_SIZE-1);
+#endif
  
                         }else
                         {
                              param.rc.bStatWrite=0;
                              param.rc.bStatRead=1;
+#if X265_BUILD < 215
                              param.rc.statFileName=strdup(logFile);
+#else
+                             strncat(param.rc.statFileName, logFile, X265_MAX_STRING_SIZE-1);
+#endif
                              if(!ADM_fileExist(logFile))
                              {
                                    ADM_error("Logfile %s does not exist \n",logFile);
