--- avogadro-0.8.1/libavogadro/src/boost.h.orig	2008-05-15 02:10:06.000000000 +0100
+++ avogadro-0.8.1/libavogadro/src/boost.h	2008-07-05 23:23:39.000000000 +0100
@@ -46,7 +46,7 @@
 	void (OBMol::*ToInertialFrame)() = &OBMol::ToInertialFrame;
 	void (OBMol::*Center)() = &OBMol::Center;
 	bool (OBMol::*DeleteHydrogens)() = &OBMol::DeleteHydrogens;
-	bool (OBMol::*AddHydrogens)(bool, bool) = &OBMol::AddHydrogens;
+	bool (OBMol::*AddHydrogens)(bool, bool, double) = &OBMol::AddHydrogens;
 	OBBond * (OBMol::*GetBondIDX)(int)const = &OBMol::GetBond;
 	OBBond * (OBMol::*GetBondAtoms)(int, int)const = &OBMol::GetBond;
 
--- avogadro-0.8.1/CMakeLists.txt.orig	2008-06-06 19:04:57.000000000 +0100
+++ avogadro-0.8.1/CMakeLists.txt	2008-07-06 07:42:03.000000000 +0100
@@ -22,7 +22,7 @@
 	add_definitions (-D_BSD_SOURCE)
 	# Set some linker flags
 	if (CMAKE_SYSTEM_NAME MATCHES Linux)
-		set (CMAKE_SHARED_LINKER_FLAGS "-Wl,--fatal-warnings -Wl,--no-undefined -lc ${CMAKE_SHARED_LINKER_FLAGS}")
+		set (CMAKE_SHARED_LINKER_FLAGS "-Wl,--fatal-warnings -Wl,--no-undefined -lc -lutil ${CMAKE_SHARED_LINKER_FLAGS}")
 		set (CMAKE_SHARED_LINKER_FLAGS "-Wl,--enable-new-dtags ${CMAKE_SHARED_LINKER_FLAGS}")
 		set (CMAKE_EXE_LINKER_FLAGS "-Wl,--enable-new-dtags ${CMAKE_EXE_LINKER_FLAGS}")
 	endif (CMAKE_SYSTEM_NAME MATCHES Linux)
