--- avogadrolibs-1.90.0/CMakeLists.txt.orig	2016-12-01 21:17:48.000000000 +0000
+++ avogadrolibs-1.90.0/CMakeLists.txt	2018-08-18 16:02:06.668224073 +0100
@@ -19,6 +19,8 @@
 find_package(Git)
 determine_version(${AvogadroLibs_SOURCE_DIR} ${GIT_EXECUTABLE} "AvogadroLibs")
 
+find_package(Spglib)
+
 option(BUILD_SHARED_LIBS "Build with shared libraries" ON)
 
 # Before any plugins are defined, and before any add_subdirectory calls:
