--- avogadro-0.8.0/CMakeLists.txt.orig	2008-05-18 01:02:24.000000000 +0100
+++ avogadro-0.8.0/CMakeLists.txt	2008-05-20 22:12:23.000000000 +0100
@@ -21,7 +21,7 @@
 	add_definitions (-D_BSD_SOURCE)
 	# Set some linker flags
 	if (CMAKE_SYSTEM_NAME MATCHES Linux)
-		set (CMAKE_SHARED_LINKER_FLAGS "-Wl,--fatal-warnings -Wl,--no-undefined -lc ${CMAKE_SHARED_LINKER_FLAGS}")
+		set (CMAKE_SHARED_LINKER_FLAGS "-Wl,--fatal-warnings -Wl,--no-undefined -lc -lutil ${CMAKE_SHARED_LINKER_FLAGS}")
 		set (CMAKE_SHARED_LINKER_FLAGS "-Wl,--enable-new-dtags ${CMAKE_SHARED_LINKER_FLAGS}")
 		set (CMAKE_EXE_LINKER_FLAGS "-Wl,--enable-new-dtags ${CMAKE_EXE_LINKER_FLAGS}")
 	endif (CMAKE_SYSTEM_NAME MATCHES Linux)
