--- avogadro-1.1.1/libavogadro/src/pythoninterpreter.h.orig	2013-12-06 15:50:04.000000000 +0000
+++ avogadro-1.1.1/libavogadro/src/pythoninterpreter.h	2014-04-24 21:42:24.193086198 +0100
@@ -26,7 +26,9 @@
 #define PYTHONINTERPRETER_H
 
 #include <avogadro/global.h>
+#ifndef Q_MOC_RUN
 #include <boost/python.hpp>
+#endif
 #include <avogadro/primitive.h>
 #include <QString>
 
--- avogadro-1.1.1/libavogadro/src/pythonscript.h.orig	2013-12-06 15:50:04.000000000 +0000
+++ avogadro-1.1.1/libavogadro/src/pythonscript.h	2014-04-24 21:45:45.764095995 +0100
@@ -27,7 +27,9 @@
 #define PYTHONSCRIPT_H
 
 #include <avogadro/global.h>
+#ifndef Q_MOC_RUN
 #include <boost/python.hpp>
+#endif
 
 #include "pythonerror.h"
 
--- avogadro-1.1.1/libavogadro/src/colors/CMakeLists.txt.orig	2013-12-06 15:50:04.000000000 +0000
+++ avogadro-1.1.1/libavogadro/src/colors/CMakeLists.txt	2014-04-24 21:35:03.327064770 +0100
@@ -6,7 +6,7 @@
 
 set(DESTINATION_DIR ${Avogadro_PLUGIN_INSTALL_DIR}/colors)
 
-include_directories(${CMAKE_CURRENT_BINARY_DIR})
+include_directories(${CMAKE_CURRENT_BINARY_DIR} ${EIGEN2_INCLUDE_DIR})
 
 set(LINK_LIBS avogadro)
 set(PLUGIN_LABEL colors)
--- avogadro-1.1.1/libavogadro/src/pythonengine_p.h.orig	2013-12-06 15:50:04.000000000 +0000
+++ avogadro-1.1.1/libavogadro/src/pythonengine_p.h	2014-04-24 21:37:42.823072522 +0100
@@ -27,7 +27,9 @@
 
 #include <avogadro/global.h>
 #include <avogadro/engine.h>
+#ifndef Q_MOC_RUN
 #include <boost/python.hpp>
+#endif
 
 namespace Avogadro {
 
--- avogadro-1.1.1/libavogadro/src/pythonextension_p.h.orig	2013-12-06 15:50:04.000000000 +0000
+++ avogadro-1.1.1/libavogadro/src/pythonextension_p.h	2014-04-24 21:38:19.789074319 +0100
@@ -29,7 +29,9 @@
 #include <avogadro/extension.h>
 #include <avogadro/primitive.h>
 #include <avogadro/glwidget.h>
+#ifndef Q_MOC_RUN
 #include <boost/python.hpp>
+#endif
 
 #include <QWidget>
 #include <QList>
--- avogadro-1.1.1/libavogadro/src/pythontool_p.h.orig	2013-12-06 15:50:04.000000000 +0000
+++ avogadro-1.1.1/libavogadro/src/pythontool_p.h	2014-04-24 21:38:45.555075571 +0100
@@ -27,7 +27,9 @@
 
 #include <avogadro/global.h>
 #include <avogadro/tool.h>
+#ifndef Q_MOC_RUN
 #include <boost/python.hpp>
+#endif
 
 #include <QObject>
 #include <QAction>
