--- avogadro-1.0.0/libavogadro/src/CMakeLists.txt.orig	2009-10-23 20:06:32.000000000 +0100
+++ avogadro-1.0.0/libavogadro/src/CMakeLists.txt	2009-10-24 22:09:09.000000000 +0100
@@ -129,7 +129,7 @@
   )
 
   message(STATUS "Setting new boost python libraries")
-  set(AVO_LINK_LIBRARIES ${AVO_LINK_LIBRARIES} ${PYTHON_LIBRARIES}
+  set(AVO_LINK_LIBRARIES ${AVO_LINK_LIBRARIES} ${PYTHON_LIBRARIES} -lutil
                          ${Boost_LIBRARIES})
   set(libavogadro_HDRS ${libavogadro_HDRS} ${python_HDRS})
 endif(ENABLE_PYTHON AND ALL_PYTHON_FOUND)
