--- avogadro-1.2.0/libavogadro/src/extensions/symmetry/libmsym/CMakeLists.txt.orig	2016-06-08 15:19:45.000000000 +0100
+++ avogadro-1.2.0/libavogadro/src/extensions/symmetry/libmsym/CMakeLists.txt	2016-07-20 21:44:35.328506315 +0100
@@ -7,13 +7,13 @@
 set(LIBMSYM_PATCH_VERSION 0)
 set(LIBMSYM_VERSION ${LIBMSYM_MAJOR_VERSION}.${LIBMSYM_MINOR_VERSION}.${LIBMSYM_PATCH_VERSION})
 
-set(INSTALL_LIB_DIR lib CACHE PATH "Installation directory for msym library")
+set(INSTALL_LIB_DIR lib${LIB_SUFFIX} CACHE PATH "Installation directory for msym library")
 set(INSTALL_INCLUDE_DIR include CACHE PATH "Installation directory for header files")
 
 if(WIN32 AND NOT CYGWIN)
   set(DEF_INSTALL_CMAKE_DIR CMake)
 else()
-  set(DEF_INSTALL_CMAKE_DIR lib/cmake/libmsym)
+  set(DEF_INSTALL_CMAKE_DIR lib${LIB_SUFFIX}/cmake/libmsym)
 endif()
 set(INSTALL_CMAKE_DIR ${DEF_INSTALL_CMAKE_DIR} CACHE PATH "Installation directory for CMake files")
 
