--- avogadro-1.2.0/libavogadro/src/pythontool_p.h.orig	2016-06-08 15:19:45.000000000 +0100
+++ avogadro-1.2.0/libavogadro/src/pythontool_p.h	2016-07-20 21:29:41.944503023 +0100
@@ -31,7 +31,9 @@
 
 #include <avogadro/global.h>
 #include <avogadro/tool.h>
+#ifndef Q_MOC_RUN
 #include <boost/python.hpp>
+#endif
 
 #include <QObject>
 #include <QAction>
--- avogadro-1.2.0/libavogadro/src/pythonengine_p.h.orig	2016-06-08 15:19:45.000000000 +0100
+++ avogadro-1.2.0/libavogadro/src/pythonengine_p.h	2016-07-20 21:29:41.945503023 +0100
@@ -31,7 +31,9 @@
 
 #include <avogadro/global.h>
 #include <avogadro/engine.h>
+#ifndef Q_MOC_RUN
 #include <boost/python.hpp>
+#endif
 
 namespace Avogadro {
 
--- avogadro-1.2.0/libavogadro/src/pythonextension_p.h.orig	2016-06-08 15:19:45.000000000 +0100
+++ avogadro-1.2.0/libavogadro/src/pythonextension_p.h	2016-07-20 21:29:41.945503023 +0100
@@ -33,7 +33,9 @@
 #include <avogadro/extension.h>
 #include <avogadro/primitive.h>
 #include <avogadro/glwidget.h>
+#ifndef Q_MOC_RUN
 #include <boost/python.hpp>
+#endif
 
 #include <QWidget>
 #include <QList>
--- avogadro-1.2.0/libavogadro/src/python/CMakeLists.txt.orig	2016-06-08 15:19:45.000000000 +0100
+++ avogadro-1.2.0/libavogadro/src/python/CMakeLists.txt	2016-07-20 21:29:41.945503023 +0100
@@ -21,7 +21,7 @@
   target_link_libraries(python-module avogadro ${PYTHON_LIBRARIES})
 else()
   target_link_libraries(python-module stdc++ avogadro
-    ${QT_LIBRARIES} ${PYTHON_LIBRARIES} ${Boost_LIBRARIES})
+    ${QT_LIBRARIES} ${PYTHON_LIBRARIES} ${Boost_LIBRARIES} ${OPENBABEL2_LIBRARIES})
 endif()
 
 # Let's try to use this instead:
--- avogadro-1.2.0/libavogadro/src/extensions/symmetry/libmsym/CMakeLists.txt.orig	2016-06-08 15:19:45.000000000 +0100
+++ avogadro-1.2.0/libavogadro/src/extensions/symmetry/libmsym/CMakeLists.txt	2016-07-20 21:44:35.328506315 +0100
@@ -7,13 +7,13 @@
 set(LIBMSYM_PATCH_VERSION 0)
 set(LIBMSYM_VERSION ${LIBMSYM_MAJOR_VERSION}.${LIBMSYM_MINOR_VERSION}.${LIBMSYM_PATCH_VERSION})
 
-set(INSTALL_LIB_DIR lib CACHE PATH "Installation directory for msym library")
+set(INSTALL_LIB_DIR lib${LIB_SUFFIX} CACHE PATH "Installation directory for msym library")
 set(INSTALL_INCLUDE_DIR include CACHE PATH "Installation directory for header files")
 
 if(WIN32 AND NOT CYGWIN)
   set(DEF_INSTALL_CMAKE_DIR CMake)
 else()
-  set(DEF_INSTALL_CMAKE_DIR lib/cmake/libmsym)
+  set(DEF_INSTALL_CMAKE_DIR lib${LIB_SUFFIX}/cmake/libmsym)
 endif()
 set(INSTALL_CMAKE_DIR ${DEF_INSTALL_CMAKE_DIR} CACHE PATH "Installation directory for CMake files")
 
