--- avview-0.80.7/Makefile.in.orig	2005-11-12 13:26:25.000000000 +0000
+++ avview-0.80.7/Makefile.in	2007-11-17 23:18:34.000000000 +0000
@@ -954,7 +954,8 @@
 	echo 'Icon=$(datadir)/AVview/avview-48x48.png' >> start_avview.desktop
 
 install-data-hook: reminders
-	$(INSTALL_DATA) start_avview.desktop /usr/share/applications/ || \
+	$(mkinstalldirs) $(DESTDIR)/usr/share/applications
+	$(INSTALL_DATA) start_avview.desktop $(DESTDIR)/usr/share/applications/ || \
 	true	
 	@echo ""
 	@echo " ************************************************"
