--- banshee-0.12.1/docs/Makefile.in.orig	2007-04-01 06:44:00.000000000 +0100
+++ banshee-0.12.1/docs/Makefile.in	2007-04-04 20:40:41.000000000 +0100
@@ -328,7 +328,7 @@
 	$(top_builddir)/src/MusicBrainz/MusicBrainz.dll
 
 ASSEMBLIES_BUILD = $(foreach asm,$(ASSEMBLIES),$(addprefix -assembly:,$(asm)))
-@BUILD_DOCS_TRUE@monodocdir = $(DESTDIR)$(DOCDIR)
+@BUILD_DOCS_TRUE@monodocdir = $(DOCDIR)
 @BUILD_DOCS_TRUE@monodoc_DATA = \
 @BUILD_DOCS_TRUE@	banshee-docs.zip \
 @BUILD_DOCS_TRUE@	banshee-docs.tree \
