--- barcode-0.99/barcode.h.orig	2013-03-29 21:51:07.000000000 +0000
+++ barcode-0.99/barcode.h	2023-05-11 16:16:54.348112820 +0100
@@ -123,6 +123,6 @@
 }
 #endif
 
-int streaming;
+extern int streaming;
 
 #endif /* _BARCODE_H_ */
--- barcode-0.99/pcl.c.orig	2013-03-29 22:23:31.000000000 +0000
+++ barcode-0.99/pcl.c	2023-05-11 16:21:58.914098288 +0100
@@ -29,6 +29,7 @@
 
 #define SHRINK_AMOUNT 0.15  /* shrink the bars to account for ink spreading */
 
+int streaming;
 
 /*
  * How do the "partial" and "textinfo" strings work? See file "ps.c"
