--- bastard-0.17/Makefile.orig	2003-07-29 21:41:00.000000000 +0100
+++ bastard-0.17/Makefile	2007-02-01 15:39:12.000000000 +0000
@@ -30,7 +30,7 @@
 ifeq ($(OSTYPE), Linux)
 CC      = 	gcc
 MAKE    = 	make
-LSLIST  = 	`ls -1aA -ICVS`
+LSLIST  = 	`ls -1A -ICVS`
 TERMIO  = 	-lncurses
 endif
 
--- bastard-0.17/src/EiC/config/NormalNodeTargets.orig	2001-05-29 19:11:51.000000000 +0100
+++ bastard-0.17/src/EiC/config/NormalNodeTargets	2007-02-01 15:26:10.000000000 +0000
@@ -4,7 +4,7 @@
 		if [ $$i != "." ]  ; then \
 			(cd $$i ;\
 			${MAKE} showSRCS) ;\
-		fi\
+		fi \
 	done
 all::
 	@for i in $(DIRS) ; \
@@ -12,7 +12,7 @@
 		if [ $$i != "." ]  ; then \
 			(cd $$i ;\
 		 	${MAKE} all) ;\
-		fi\
+		fi \
 	done
 
 clean::
@@ -21,7 +21,7 @@
 		if [ $$i != "." ] ; then \
 			(cd $$i ;\
 			$(MAKE) clean) ;\
-		fi\
+		fi \
 	done
 
 install::
@@ -30,7 +30,7 @@
 		if [ $$i != "." ] ; then \
 			(cd $$i ;\
 			$(MAKE) install) ;\
-		fi\
+		fi \
 	done
 
 clobber::
@@ -39,7 +39,7 @@
 		if [ $$i != "." ] ; then \
 			(cd $$i ;\
 			$(MAKE) clobber) ;\
-		fi\
+		fi \
 	done
 
 headers::
@@ -48,7 +48,7 @@
 		if [ $$i != "." ] ; then \
 			(cd $$i ;\
 			$(MAKE) headers ) ;\
-		fi\
+		fi \
 	done
 
 
