--- beagle-0.2.10/tools/beagle-crawl-system.in.orig	2006-06-19 18:54:27.000000000 +0100
+++ beagle-0.2.10/tools/beagle-crawl-system.in	2006-09-19 21:20:40.000000000 +0100
@@ -64,7 +64,7 @@
                 IONICE="$IONICE -c 3"
             fi
 
-	    eval nice -n 19 $IONICE su -s /bin/bash $CRAWL_USER -c \"MONO_SHARED_DIR=$MONO_SHARED_DIR @sbindir@/beagle-build-index --target @localstatedir@/cache/beagle/indexes/$CRAWL_INDEX_NAME $OPTIONS $CRAWL_PATHS\" > /dev/null 2>&1
+	    eval nice -n 19 $IONICE su $CRAWL_USER -c \"MONO_SHARED_DIR=$MONO_SHARED_DIR @sbindir@/beagle-build-index --target @localstatedir@/cache/beagle/indexes/$CRAWL_INDEX_NAME $OPTIONS $CRAWL_PATHS\" > /dev/null 2>&1
 	fi
     fi
 done
