--- beecrypt-4.1.2/c++/beeyond/BeeCertificate.h.orig	2004-11-03 09:37:27.000000000 +0000
+++ beecrypt-4.1.2/c++/beeyond/BeeCertificate.h	2009-06-02 23:05:19.000000000 +0100
@@ -61,7 +61,7 @@
 		 */
 		class BEECRYPTCXXAPI BeeCertificate : public beecrypt::security::cert::Certificate, public beecrypt::lang::Cloneable
 		{
-			friend class BeeCertificateFactory;
+			friend class ::BeeCertificateFactory;
 
 		public:
 			static const Date FOREVER;
--- beecrypt-4.1.2/c++/security/Security.h.orig	2004-11-02 07:06:58.000000000 +0000
+++ beecrypt-4.1.2/c++/security/Security.h	2009-06-02 23:04:46.000000000 +0100
@@ -59,16 +59,16 @@
 		{
 			friend class AlgorithmParameterGenerator;
 			friend class AlgorithmParameters;
-			friend class CertificateFactory;
-			friend class CertPathValidator;
-			friend class Cipher;
-			friend class KeyAgreement;
+			friend class ::CertificateFactory;
+			friend class ::CertPathValidator;
+			friend class ::Cipher;
+			friend class ::KeyAgreement;
 			friend class KeyFactory;
 			friend class KeyPairGenerator;
 			friend class KeyStore;
-			friend class Mac;
+			friend class ::Mac;
 			friend class MessageDigest;
-			friend class SecretKeyFactory;
+			friend class ::SecretKeyFactory;
 			friend class SecureRandom;
 			friend class Signature;
 
