--- bind-9.8.1/lib/isc/unix/socket.c.orig	2011-07-29 03:19:20.000000000 +0100
+++ bind-9.8.1/lib/isc/unix/socket.c	2011-09-03 09:12:41.000000000 +0100
@@ -712,6 +712,8 @@
 }
 #endif
 
+#undef SO_BSDCOMPAT
+
 static void
 socket_log(isc__socket_t *sock, isc_sockaddr_t *address,
 	   isc_logcategory_t *category, isc_logmodule_t *module, int level,
--- bind-9.8.1/bin/tests/system/dlzexternal/Makefile.in.orig	2011-03-11 07:10:10.000000000 +0000
+++ bind-9.8.1/bin/tests/system/dlzexternal/Makefile.in	2011-09-03 16:55:49.000000000 +0100
@@ -49,7 +49,7 @@
 		-o $@ ${DLOPENOBJS} ${LIBS}
 
 driver.@SO@: ${SO_OBJS}
-	@SO_LD@ -o $@ driver.o
+	${LIBTOOL_MODE_LINK} ${PURIFY} ${CC} ${CFLAGS} ${LDFLAGS} -shared -o $@ driver.lo
 
 clean distclean::
 	rm -f ${TARGETS}
