--- blender-5.0.0/CMakeLists.txt.orig	2025-10-09 17:52:29.000000000 +0100
+++ blender-5.0.0/CMakeLists.txt	2025-11-19 09:16:46.798386964 +0000
@@ -2592,7 +2592,7 @@
   elseif((WITH_PYTHON_INSTALL AND WITH_PYTHON_INSTALL_NUMPY) OR WITH_PYTHON_NUMPY)
     if(("${PYTHON_NUMPY_PATH}" STREQUAL "") OR (${PYTHON_NUMPY_PATH} MATCHES NOTFOUND))
       set(_numpy_include "_core/include")
-      if(PYTHON_VERSION VERSION_LESS "3.13")
+      if(PYTHON_VERSION VERSION_LESS "3.12")
         set(_numpy_include "core/include")
       endif()
       find_python_package(numpy "${_numpy_include}")
