--- blender-2.41/intern/SoundSystem/openal/SND_OpenALDevice.cpp.orig	2006-01-07 00:14:36.000000000 +0000
+++ blender-2.41/intern/SoundSystem/openal/SND_OpenALDevice.cpp	2006-06-26 14:04:32.000000000 +0100
@@ -432,7 +432,7 @@
 #ifdef AL_VERSION_1_1					
 					alutLoadWAVMemory((ALbyte*)memlocation, &sampleformat, &data, &numberofsamples, &samplerate, &loop);//	openal_2.14+
 #else
-					 alutLoadWAVMemory((signed char*)memlocation, &sampleformat, &data, &numberofsamples, &samplerate, &loop);//  openal_2.14+
+					 alutLoadWAVMemory((ALbyte*)memlocation, &sampleformat, &data, &numberofsamples, &samplerate, &loop);//  openal_2.14+
 					 
 #endif 
 #endif
