Index: tools/btools.py
===================================================================
--- tools/btools.py	(revision 14549)
+++ tools/btools.py	(working copy)
@@ -2,7 +2,7 @@
 import os
 import os.path
 import SCons.Options
-import SCons.Options.BoolOption
+from SCons.Options import BoolOption
 try:
     import subprocess
 except ImportError:
@@ -13,7 +13,6 @@
 import sys
 
 Options = SCons.Options
-BoolOption = SCons.Options.BoolOption
 
 def print_arguments(args, bc):
     if len(args):
