--- blobwars-1.07/makefile.orig	2007-05-05 15:22:48.000000000 +0100
+++ blobwars-1.07/makefile	2007-06-15 20:31:26.000000000 +0100
@@ -8,13 +8,14 @@
 USEPAK = 1
 
 PREFIX=$(DESTDIR)/usr
-BINDIR = $(PREFIX)/games/
-DATADIR = $(PREFIX)/share/games/blobwars/
-DOCDIR = $(PREFIX)/share/doc/$(PROG)/
+BINDIR = $(PREFIX)/bin/
+DATADIR = $(PREFIX)/share/blobwars/
+DOCDIR = $(PREFIX)/share/$(PROG)/doc/
 ICONDIR = $(PREFIX)/share/icons/hicolor/
 DESKTOPDIR = $(PREFIX)/share/applications/
 LOCALEDIR = $(PREFIX)/share/locale/
 
+
 CXXFLAGS += `sdl-config --cflags` -DVERSION=$(VERSION) -DRELEASE=$(RELEASE) -DUSEPAK=$(USEPAK)
 CXXFLAGS += -DPAKNAME=\"$(PAKNAME)\" -DPAKLOCATION=\"$(DATADIR)\" -DUNIX -DGAMEPLAYMANUAL=\"$(DOCDIR)index.html\" -Wall
 CXXFLAGS += -DLOCALEDIR=\"$(LOCALEDIR)\"
@@ -77,9 +78,9 @@
 	mkdir -p $(ICONDIR)64x64/apps
 	mkdir -p $(DESKTOPDIR)
 
-	install -o root -g games -m 755 $(PROG) $(BINDIR)$(PROG)
-	install -o root -g games -m 644 $(PAKNAME) $(DATADIR)$(PAKNAME)
-	install -o root -g games -m 644 $(DOCS) $(DOCDIR)
+	install -m 755 $(PROG) $(BINDIR)$(PROG)
+	install -m 644 $(PAKNAME) $(DATADIR)$(PAKNAME)
+	install -m 644 $(DOCS) $(DOCDIR)
 	cp $(ICONS)$(PROG).png $(ICONDIR)32x32/apps/
 	cp $(ICONS)$(PROG)-mini.png $(ICONDIR)16x16/apps/$(PROG).png
 	cp $(ICONS)$(PROG)-large.png $(ICONDIR)64x64/apps/$(PROG).png
