--- bluefish-2.2.17/src/plugin_infbrowser/infb_gui.c.orig	2014-10-13 19:26:05.000000000 +0100
+++ bluefish-2.2.17/src/plugin_infbrowser/infb_gui.c	2025-06-01 22:23:49.636093181 +0100
@@ -140,13 +140,13 @@
 	      			if (g_str_has_prefix ((gchar*)aux,"http://")) {
 							gchar *pstr = g_strdup_printf("%s/bfish_%ld",g_get_tmp_dir(),(long int)time(NULL));
 							gchar *info;
-							if (xmlNanoHTTPFetch((const char *)aux,pstr,&info)==-1) {							
+//							if (xmlNanoHTTPFetch((const char *)aux,pstr,&info)==-1) {							
 								g_free(pstr);
 								infb_insert_error(GTK_TEXT_VIEW(widget),BAD_CAST _("Cannot load file from network"));
 	      					/*message_dialog_new(bfwin->main_window,
       								GTK_MESSAGE_INFO,GTK_BUTTONS_CLOSE,_("Cannot load file"),(gchar*)aux);*/
 								break;
-							}
+//							}
 							if (info) g_free(info);
 							doc = xmlReadFile(pstr,NULL,
 							       XML_PARSE_RECOVER | XML_PARSE_NOENT | XML_PARSE_NOBLANKS | 
@@ -200,13 +200,13 @@
       								if (text) {
 												gchar *pstr = g_strdup_printf("%s/bfish_%ld",g_get_tmp_dir(),(long int)time(NULL));
 												gchar *info;
-												if (xmlNanoHTTPFetch((const char *)text,pstr,&info)==-1) {							
+//												if (xmlNanoHTTPFetch((const char *)text,pstr,&info)==-1) {							
 													g_free(pstr);
 													infb_insert_error(GTK_TEXT_VIEW(widget),BAD_CAST _("Cannot load file from network"));
 	      										/*message_dialog_new(bfwin->main_window,
       																	GTK_MESSAGE_INFO,GTK_BUTTONS_CLOSE,_("Cannot load file from network"),(gchar*)text);*/
 													break;
-												}
+//												}
 												if (info) g_free(info);												
 												doc = xmlReadFile(pstr,NULL,
 														XML_PARSE_RECOVER | XML_PARSE_NOENT | XML_PARSE_NOBLANKS
