--- bluez-4.89/src/bluetooth.conf.orig	2010-10-04 12:41:01.000000000 +0100
+++ bluez-4.89/src/bluetooth.conf	2011-02-28 22:13:27.087000029 +0000
@@ -16,8 +16,12 @@
     <allow send_interface="org.bluez.Watcher"/>
   </policy>
 
-  <policy at_console="true">
+  <policy group="plugdev">
     <allow send_destination="org.bluez"/>
+    <allow send_interface="org.bluez.Agent"/>
+    <allow send_interface="org.bluez.Adapter"/>
+    <allow send_destination="org.bluez"
+           send_interface="org.freedesktop.DBus.Introspectable"/>
   </policy>
 
   <!-- allow users of lp group (printing subsystem) to 
