--- bombono-dvd-1.0.2/src/mbase/project/archieve.cpp.orig	2011-04-18 01:48:26.000000000 +0100
+++ bombono-dvd-1.0.2/src/mbase/project/archieve.cpp	2011-06-09 06:34:55.000000000 +0100
@@ -179,7 +179,7 @@
     // :KLUDGE: libxml++ не публикует в заголовках свою версию (а усложнение сборки
     // через pkg-config --modversion libxml++-2.6 не стоит делать,- игра не стоит свеч)
 #if IS_VERSION_GE(GLIBMM, 2, 27, 4)
-    xmlpp::Node::create_wrapper(children);
+//    xmlpp::Node::create_wrapper(children);
 #endif
     xmlpp::Node* child = reinterpret_cast<xmlpp::Node*>(children->_private);
     AcceptNode(ar, FindNextElement(child), true);
--- bombono-dvd-1.0.2/src/mgui/ffviewer.cpp.orig	2011-04-18 01:48:26.000000000 +0100
+++ bombono-dvd-1.0.2/src/mgui/ffviewer.cpp	2011-06-09 06:35:46.000000000 +0100
@@ -318,15 +318,15 @@
     {
         switch( av_res )
         {
-        case AVERROR_NOENT:
+//        case AVERROR_NOENT:
             // :TODO: решить, ставить в конце точки или нет (сообщения пользователю
             // показывается не HIG-ого)
-            err_str = _("No such file");
-            break;
-        case AVERROR_NOFMT:
-            err_str = _("Unknown file format");
-            break;
-        case AVERROR_UNKNOWN:
+//            err_str = _("No such file");
+//            break;
+//        case AVERROR_NOFMT:
+//            err_str = _("Unknown file format");
+//            break;
+//        case AVERROR_UNKNOWN:
         default:
             err_str = boost::format("FFmpeg unknown error: %1%") % av_res % bf::stop;
             break;
@@ -354,13 +354,13 @@
         {
             AVStream* strm = ic->streams[i];
             AVCodecContext* avctx = strm->codec;
-            if( SetIndex(video_idx, i, avctx->codec_type == CODEC_TYPE_VIDEO) )
+            if( SetIndex(video_idx, i, avctx->codec_type == AVMEDIA_TYPE_VIDEO) )
                 ;
             else
                 // для демиксера имеет значение только NONE и ALL
                 strm->discard = AVDISCARD_ALL;
 
-            SetIndex(audio_idx, i, avctx->codec_type == CODEC_TYPE_AUDIO);
+            SetIndex(audio_idx, i, avctx->codec_type == AVMEDIA_TYPE_AUDIO);
         }
 
         if( video_idx == -1 )
@@ -525,7 +525,7 @@
 
 static bool IsInHurry(AVCodecContext* dec)
 {
-    return dec->hurry_up != 0;
+    return dec->skip_frame != 0;
 }
 
 struct HurryModeEnabler
@@ -535,7 +535,7 @@
     HurryModeEnabler(AVCodecContext* dec_): dec(dec_)
     {
         // как признак (хоть и устаревший)
-        dec->hurry_up = 1;
+//        dec->hurry_up = 1;
         // Прирост скорости (h264): 
         // - AVDISCARD_NONREF: 2x
         // - AVDISCARD_BIDIR: для h264 (и других современных кодеков?) разница в скорости 
@@ -555,7 +555,7 @@
     }
    ~HurryModeEnabler()
     {
-        dec->hurry_up = 0;
+//        dec->hurry_up = 0;
         dec->skip_frame = AVDISCARD_DEFAULT;
         //dec->skip_idct = AVDISCARD_DEFAULT;
         //dec->skip_loop_filter = AVDISCARD_DEFAULT;
--- bombono-dvd-1.0.2/src/mgui/project/media-browser.cpp.orig	2011-04-18 01:48:26.000000000 +0100
+++ bombono-dvd-1.0.2/src/mgui/project/media-browser.cpp	2011-06-09 06:15:07.000000000 +0100
@@ -232,7 +232,7 @@
         for( int i=0; i < (int)ic->nb_streams; i++ )
         {
             AVCodecContext* avctx = ic->streams[i]->codec;
-            if( avctx->codec_type == CODEC_TYPE_AUDIO )
+            if( avctx->codec_type == AVMEDIA_TYPE_AUDIO )
                 rtc.audioNum++;
         }
 
