--- bonnie-1.4/Makefile.orig	2002-02-20 18:22:55.000000000 +0000
+++ bonnie-1.4/Makefile	2006-07-11 09:54:39.000000000 +0100
@@ -23,9 +23,9 @@
 # Added 99/07/20, <garloff@suse.de>
 install: Bonnie
 	install -d $(DESTDIR)$(PREFIX)/bin
-	install -s -m 0755 -o root -g root Bonnie $(DESTDIR)$(PREFIX)/bin/bonnie
+	install -s -m 0755 Bonnie $(DESTDIR)$(PREFIX)/bin/bonnie
 	install -d $(DESTDIR)$(MANDIR)/man1
-	install -m 0644 -o man -g root bonnie.1 $(DESTDIR)$(MANDIR)/man1/bonnie.1
+	install -m 0644 bonnie.1 $(DESTDIR)$(MANDIR)/man1/bonnie.1
 	gzip -9f $(DESTDIR)$(MANDIR)/man1/bonnie.1
 	
 clean: 
