--- boost-1_57_0/boost/signals2/trackable.hpp.orig	2014-09-26 23:30:52.000000000 +0100
+++ boost-1_57_0/boost/signals2/trackable.hpp	2015-01-25 08:28:12.144050664 +0000
@@ -18,6 +18,7 @@
 
 #include <boost/assert.hpp>
 #include <boost/shared_ptr.hpp>
+#include <boost/weak_ptr.hpp>
 
 namespace boost {
   namespace signals2 {
--- boost-1_57_0/libs/serialization/util/test.jam.orig	2014-09-26 18:50:50.000000000 +0100
+++ boost-1_57_0/libs/serialization/util/test.jam	2014-09-26 18:50:50.000000000 +0100
@@ -47,11 +47,8 @@
             # toolset optimizations
             <toolset>gcc:<cxxflags>"-ftemplate-depth-255"
             <toolset>clang:<cxxflags>"-ftemplate-depth-255"
-<<<<<<< HEAD
-=======
             <toolset>clang-darwin:<cxxflags>"-ftemplate-depth-255"
             <toolset>darwin:<cxxflags>"-ftemplate-depth-255"
->>>>>>> develop
             <toolset>msvc:<cxxflags>"-Gy"
             $(requirements)
         : # test name
