--- boson-all-0.13/cmake/modules/boson/FindBosonPythonLibs.cmake.orig	2006-06-17 14:36:11.000000000 +0100
+++ boson-all-0.13/cmake/modules/boson/FindBosonPythonLibs.cmake	2007-08-11 19:37:09.000000000 +0100
@@ -72,6 +72,6 @@
 
 	# PYTHON_LIBRARY is used for the cache entry only
 	# PYTHON_LIBRARIES is meant to be public
-	set(PYTHON_LIBRARIES "${PYTHON_LIBRARY} ${_python_dependency_libs}")
+	set(PYTHON_LIBRARIES ${PYTHON_LIBRARY} ${_python_dependency_libs})
 ENDIF (NOT PYTHON_LIBRARY)
 
--- boson-all-0.13/code/boson/info/boinfo_linux.cpp.orig	2005-07-26 22:53:07.000000000 +0100
+++ boson-all-0.13/code/boson/info/boinfo_linux.cpp	2007-08-11 18:05:34.000000000 +0100
@@ -28,6 +28,7 @@
 #include <qstringlist.h>
 
 #include <linux/version.h>
+#include <linux/utsrelease.h>
 
 QString readFile(const char* fileName)
 {
