--- boson-all-0.13/code/bogl/gl/extensions/arb_shader_objects_decl_p.h.orig	2006-09-05 04:19:13.000000000 +0100
+++ boson-all-0.13/code/bogl/gl/extensions/arb_shader_objects_decl_p.h	2008-05-11 06:26:41.000000000 +0100
@@ -37,7 +37,7 @@
 	typedef GLhandleARB (*_glCreateShaderObjectARB) (GLenum);
 	typedef GLvoid (*_glShaderSourceARB) (GLhandleARB, GLsizei, const GLcharARB**, const GLint*);
 	typedef GLvoid (*_glCompileShaderARB) (GLhandleARB);
-	typedef GLhandleARB (*_glCreateProgramObjectARB) ();
+	typedef GLhandleARB (*_glCreateProgramObjectARB) (void);
 	typedef GLvoid (*_glAttachObjectARB) (GLhandleARB, GLhandleARB);
 	typedef GLvoid (*_glLinkProgramARB) (GLhandleARB);
 	typedef GLvoid (*_glUseProgramObjectARB) (GLhandleARB);
--- boson-all-0.13/code/bogl/gl/bogl_2_0_decl_p.h.orig	2006-09-05 04:19:13.000000000 +0100
+++ boson-all-0.13/code/bogl/gl/bogl_2_0_decl_p.h	2008-05-11 06:26:41.000000000 +0100
@@ -290,7 +290,7 @@
 	typedef GLvoid (*_glShaderSource)(GLuint, GLsizei, const GLchar**, const GLint*);
 	typedef GLvoid (*_glCompileShader)(GLuint);
 	typedef GLvoid (*_glDeleteShader)(GLuint);
-	typedef GLuint (*_glCreateProgram)();
+	typedef GLuint (*_glCreateProgram)(void);
 	typedef GLvoid (*_glAttachShader)(GLuint, GLuint);
 	typedef GLvoid (*_glDetachShader)(GLuint, GLuint);
 	typedef GLvoid (*_glLinkProgram)(GLuint);
--- boson-all-0.13/cmake/modules/boson/FindBosonPythonLibs.cmake.orig	2006-06-17 14:36:11.000000000 +0100
+++ boson-all-0.13/cmake/modules/boson/FindBosonPythonLibs.cmake	2007-08-11 19:37:09.000000000 +0100
@@ -72,6 +72,6 @@
 
 	# PYTHON_LIBRARY is used for the cache entry only
 	# PYTHON_LIBRARIES is meant to be public
-	set(PYTHON_LIBRARIES "${PYTHON_LIBRARY} ${_python_dependency_libs}")
+	set(PYTHON_LIBRARIES ${PYTHON_LIBRARY} ${_python_dependency_libs})
 ENDIF (NOT PYTHON_LIBRARY)
 
--- boson-all-0.13/code/boson/info/boinfo_linux.cpp.orig	2005-07-26 22:53:07.000000000 +0100
+++ boson-all-0.13/code/boson/info/boinfo_linux.cpp	2007-08-11 18:05:34.000000000 +0100
@@ -28,6 +28,7 @@
 #include <qstringlist.h>
 
 #include <linux/version.h>
+#include <linux/utsrelease.h>
 
 QString readFile(const char* fileName)
 {
