--- bpftrace-20260307/CMakeLists.txt.orig	2026-03-02 10:45:51.000000000 +0000
+++ bpftrace-20260307/CMakeLists.txt	2026-03-07 22:43:32.780033647 +0000
@@ -163,7 +163,7 @@
   # releases.
   set(MAX_LLVM_MAJOR 999)
 else()
-  set(MAX_LLVM_MAJOR 21)
+  set(MAX_LLVM_MAJOR 22)
 endif()
 
 if((${LLVM_VERSION_MAJOR} VERSION_LESS ${MIN_LLVM_MAJOR}) OR (${LLVM_VERSION_MAJOR} VERSION_GREATER ${MAX_LLVM_MAJOR}))
--- bpftrace-20260307/src/btf/CMakeLists.txt.orig	2026-03-01 21:18:49.000000000 +0000
+++ bpftrace-20260307/src/btf/CMakeLists.txt	2026-03-07 22:42:53.877889194 +0000
@@ -5,4 +5,4 @@
   irbuilder.cpp
 )
 add_dependencies(btf parser util)
-target_link_libraries(btf util)
+target_link_libraries(btf util ${LIBBPF_LIBRARIES})
