--- bsdiff-4.2/Makefile.orig	2004-07-11 19:10:02.000000000 +0100
+++ bsdiff-4.2/Makefile	2006-12-02 21:04:31.000000000 +0000
@@ -1,7 +1,7 @@
 CFLAGS		+=	-O3
-.ifdef BZIP2
+ifdef BZIP2
 CFLAGS		+=	-DBZIP2=\"${BZIP2}\"
-.endif
+endif
 
 PREFIX		?=	/usr/local
 INSTALL_PROGRAM	?=	${INSTALL} -c -s -m 555
@@ -13,6 +13,6 @@
 
 install:
 	${INSTALL_PROGRAM} bsdiff bspatch ${PREFIX}/bin
-.ifndef WITHOUT_MAN
+ifndef WITHOUT_MAN
 	${INSTALL_MAN} bsdiff.1 bspatch.1 ${PREFIX}/man/man1
-.endif
+endif
