--- c-blosc-1.17.0/CMakeLists.txt.orig	2019-07-23 12:31:14.000000000 +0100
+++ c-blosc-1.17.0/CMakeLists.txt	2019-11-23 16:53:56.329255058 +0000
@@ -336,7 +336,7 @@
         "${CMAKE_CURRENT_BINARY_DIR}/blosc.pc"
         @ONLY)
     install(FILES "${CMAKE_CURRENT_BINARY_DIR}/blosc.pc"
-            DESTINATION lib/pkgconfig COMPONENT DEV)
+            DESTINATION lib${LIB_SUFFIX}/pkgconfig COMPONENT DEV)
 
     configure_file(
         "${CMAKE_CURRENT_SOURCE_DIR}/cmake_uninstall.cmake.in"
