--- cairo-1.8.10/configure.orig	2010-02-20 00:25:37.000000000 +0000
+++ cairo-1.8.10/configure	2010-06-04 06:23:56.000000000 +0100
@@ -22513,7 +22513,7 @@
 
   if test "x$png_REQUIRES" = x; then
     # libpng13 is GnuWin32's libpng-1.2.8 :-(
-    for l in libpng12 libpng13 libpng10 libpng; do
+    for l in libpng14 libpng12 libpng13 libpng10 libpng; do
       if $PKG_CONFIG --exists $l ; then
         png_REQUIRES=$l
         use_png=yes
