--- cantata-2.3.3/CMakeLists.txt.orig	2018-10-20 23:39:39.000000000 +0100
+++ cantata-2.3.3/CMakeLists.txt	2019-04-25 22:39:58.164333733 +0100
@@ -119,7 +119,7 @@
     if (CANTATA_HELPERS_LIB_DIR AND CMAKE_SIZEOF_VOID_P EQUAL 8)
         set(LINUX_LIB_DIR ${CANTATA_HELPERS_LIB_DIR})
     else ()
-        set(LINUX_LIB_DIR lib)
+        set(LINUX_LIB_DIR lib${LIB_SUFFIX})
     endif ()
 endif ()
 
@@ -560,7 +560,7 @@
     install(TARGETS cantata DESTINATION ${CMAKE_INSTALL_PREFIX})
 else ()
     ADD_EXECUTABLE(cantata ${CANTATA_SRCS} ${CANTATA_RC_SRCS} ${CANTATA_UI_HDRS})
-    install(TARGETS cantata RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+    install(TARGETS cantata RUNTIME DESTINATION bin LIBRARY DESTINATION lib${LIB_SUFFIX} ARCHIVE DESTINATION lib${LIB_SUFFIX})
 endif ()
 
 add_subdirectory(3rdparty/kcategorizedview)
