Index: config.h.in
===================================================================
RCS file: /cvsroot/ccache/config.h.in,v
retrieving revision 1.5
retrieving revision 1.7
diff -u -r1.5 -r1.7
--- config.h.in	7 Mar 2003 12:09:19 -0000	1.5
+++ config.h.in	25 Jul 2005 07:05:46 -0000	1.7
@@ -19,6 +19,9 @@
 /* Define to 1 if you have the `gethostname' function. */
 #undef HAVE_GETHOSTNAME
 
+/* Define to 1 if you have the `getpwuid' function. */
+#undef HAVE_GETPWUID
+
 /* Define to 1 if you have the <inttypes.h> header file. */
 #undef HAVE_INTTYPES_H
 
@@ -31,6 +34,9 @@
 /* Define to 1 if you have the <ndir.h> header file, and it defines `DIR'. */
 #undef HAVE_NDIR_H
 
+/* Define to 1 if you have the <pwd.h> header file. */
+#undef HAVE_PWD_H
+
 /* Define to 1 if you have the `realpath' function. */
 #undef HAVE_REALPATH
 
Index: util.c
===================================================================
RCS file: /cvsroot/ccache/util.c,v
retrieving revision 1.34
retrieving revision 1.37
diff -u -r1.34 -r1.37
--- util.c	6 Sep 2004 12:59:12 -0000	1.34
+++ util.c	17 Jul 2006 03:41:12 -0000	1.37
@@ -448,7 +444,7 @@
 		}
 	}
 #endif
-	fatal("Unable to determine home directory");
+	fprintf(stderr, "ccache: Unable to determine home directory");
 	return NULL;
 }
 
