--- celestia-1.6.0/src/celengine/overlay.h.orig	2009-06-22 16:44:08.000000000 +0100
+++ celestia-1.6.0/src/celengine/overlay.h	2010-06-15 21:16:23.000000000 +0100
@@ -10,6 +10,7 @@
 #ifndef _OVERLAY_H_
 #define _OVERLAY_H_
 
+#include <cstdio>
 #include <string>
 #include <iostream>
 #include <celtxf/texturefont.h>
--- celestia-1.6.0/src/celengine/star.cpp.orig	2009-06-22 16:44:08.000000000 +0100
+++ celestia-1.6.0/src/celengine/star.cpp	2010-06-15 21:37:38.000000000 +0100
@@ -8,6 +8,7 @@
 // of the License, or (at your option) any later version.
 
 #include <celmath/mathlib.h>
+#include <cstdio>
 #include <cstring>
 #include <cassert>
 #include "celestia.h"
--- celestia-1.6.0/src/celengine/image.cpp.orig	2009-06-22 16:44:08.000000000 +0100
+++ celestia-1.6.0/src/celengine/image.cpp	2010-06-15 21:18:21.000000000 +0100
@@ -737,7 +737,7 @@
 
     if (color_type == PNG_COLOR_TYPE_GRAY && bit_depth < 8)
     {
-        png_set_gray_1_2_4_to_8(png_ptr);
+        png_set_expand_gray_1_2_4_to_8(png_ptr);
     }
 
     if (png_get_valid(png_ptr, info_ptr, PNG_INFO_tRNS))
