--- ceres-solver-2.1.0/cmake/FindSuiteSparse.cmake.orig	2022-03-28 15:06:15.000000000 +0100
+++ ceres-solver-2.1.0/cmake/FindSuiteSparse.cmake	2023-09-17 17:15:59.270917550 +0100
@@ -394,19 +394,19 @@
   else (NOT EXISTS ${SuiteSparse_VERSION_FILE})
     file(READ ${SuiteSparse_VERSION_FILE} Config_CONTENTS)
 
-    string(REGEX MATCH "#define SUITESPARSE_MAIN_VERSION [0-9]+"
+    string(REGEX MATCH "#define SUITESPARSE_MAIN_VERSION +[0-9]+"
       SuiteSparse_VERSION_MAJOR "${Config_CONTENTS}")
-    string(REGEX REPLACE "#define SUITESPARSE_MAIN_VERSION ([0-9]+)" "\\1"
+    string(REGEX REPLACE "#define SUITESPARSE_MAIN_VERSION +([0-9]+)" "\\1"
       SuiteSparse_VERSION_MAJOR "${SuiteSparse_VERSION_MAJOR}")
 
-    string(REGEX MATCH "#define SUITESPARSE_SUB_VERSION [0-9]+"
+    string(REGEX MATCH "#define SUITESPARSE_SUB_VERSION +[0-9]+"
       SuiteSparse_VERSION_MINOR "${Config_CONTENTS}")
-    string(REGEX REPLACE "#define SUITESPARSE_SUB_VERSION ([0-9]+)" "\\1"
+    string(REGEX REPLACE "#define SUITESPARSE_SUB_VERSION +([0-9]+)" "\\1"
       SuiteSparse_VERSION_MINOR "${SuiteSparse_VERSION_MINOR}")
 
-    string(REGEX MATCH "#define SUITESPARSE_SUBSUB_VERSION [0-9]+"
+    string(REGEX MATCH "#define SUITESPARSE_SUBSUB_VERSION +[0-9]+"
       SuiteSparse_VERSION_PATCH "${Config_CONTENTS}")
-    string(REGEX REPLACE "#define SUITESPARSE_SUBSUB_VERSION ([0-9]+)" "\\1"
+    string(REGEX REPLACE "#define SUITESPARSE_SUBSUB_VERSION +([0-9]+)" "\\1"
       SuiteSparse_VERSION_PATCH "${SuiteSparse_VERSION_PATCH}")
 
     # This is on a single line s/t CMake does not interpret it as a list of
