--- clamav-0.93/clamav-milter/clamav-milter.c.orig	2008-04-09 16:39:18.000000000 +0100
+++ clamav-0.93/clamav-milter/clamav-milter.c	2008-04-18 15:07:49.000000000 +0100
@@ -5763,7 +5763,7 @@
 	daily = cli_malloc(strlen(dbdir) + 22);
 	sprintf(daily, "%s/daily.cvd", dbdir);
 	if(access(daily, R_OK) < 0)
-		sprintf(daily, "%s/daily.inc/daily.info", dbdir);
+		sprintf(daily, "%s/daily.cld", dbdir);
 
 	cli_dbgmsg("loadDatabase: check %s for updates\n", daily);
 
