--- Sources/Core/IOData/Unix/directory_scanner_unix.cpp
+++ Sources/Core/IOData/Unix/directory_scanner_unix.cpp
@@ -13,6 +13,7 @@
         ------------------------------------------------------------------------
 */
 
+#include <cassert>
 #include <API/Core/System/error.h>
 #include <iostream>
 #include <libgen.h>
--- Sources/Core/System/Generic/cl_assert.cpp
+++ Sources/Core/System/Generic/cl_assert.cpp
@@ -14,6 +14,7 @@
 
 #include "Core/precomp.h"
 
+#include <cassert>
 #include <cstdlib>
 #include <iostream>
 #include <cassert>
