--- ClanLib-0.8.1/Sources/GL/opengl.cpp.orig	2006-10-27 02:55:01.000000000 +0100
+++ ClanLib-0.8.1/Sources/GL/opengl.cpp	2010-06-19 20:52:36.000000000 +0100
@@ -27,6 +27,7 @@
 **    (if your name is missing here, please add it)
 */
 
+#include <cstring>
 #include "Display/display_precomp.h"
 #include "API/GL/opengl.h"
 #include "API/GL/opengl_wrap.h"
--- ClanLib-0.8.1/Sources/GL/canvas_opengl.cpp.orig	2007-10-21 01:14:01.000000000 +0100
+++ ClanLib-0.8.1/Sources/GL/canvas_opengl.cpp	2010-06-19 20:54:21.000000000 +0100
@@ -27,6 +27,7 @@
 **    (if your name is missing here, please add it)
 */
 
+#include <cstdio>
 #include "Display/display_precomp.h"
 #include "canvas_opengl.h"
 #include "API/Core/System/error.h"
--- ClanLib-0.8.1/Sources/GL/GLX/input_device_xinput.cpp.orig	2007-10-21 01:14:01.000000000 +0100
+++ ClanLib-0.8.1/Sources/GL/GLX/input_device_xinput.cpp	2010-06-19 21:07:31.000000000 +0100
@@ -29,6 +29,7 @@
 
 #include <iostream>
 #include <cstring>
+#include <cstdio>
 #include "API/Core/System/log.h"
 #include "API/Display/input_device.h"
 #include "API/Display/input_event.h"
--- ClanLib-0.8.1/Sources/GL/GLX/input_device_linuxevent.cpp.orig	2007-10-21 01:14:01.000000000 +0100
+++ ClanLib-0.8.1/Sources/GL/GLX/input_device_linuxevent.cpp	2010-06-19 21:04:13.000000000 +0100
@@ -27,6 +27,7 @@
 **    (if your name is missing here, please add it)
 */
 
+#include <cstdio>
 #include <assert.h>
 #include <linux/input.h>
 #include <sys/ioctl.h>
