--- CLAPACK-3.2.1/BLAS/dblat2.in.orig	2009-08-07 23:32:18.000000000 +0100
+++ CLAPACK-3.2.1/BLAS/dblat2.in	2010-11-28 21:03:24.000000000 +0000
@@ -4,7 +4,7 @@
 -1                UNIT NUMBER OF SNAPSHOT FILE (NOT USED IF .LT. 0)
 F        LOGICAL FLAG, T TO REWIND SNAPSHOT FILE AFTER EACH RECORD.
 F        LOGICAL FLAG, T TO STOP ON FAILURES.
-T        LOGICAL FLAG, T TO TEST ERROR EXITS.
+F        LOGICAL FLAG, T TO TEST ERROR EXITS.
 16.0     THRESHOLD VALUE OF TEST RATIO
 6                 NUMBER OF VALUES OF N
 0 1 2 3 5 9       VALUES OF N
--- CLAPACK-3.2.1/BLAS/zblat3.in.orig	2009-08-07 23:32:18.000000000 +0100
+++ CLAPACK-3.2.1/BLAS/zblat3.in	2010-11-28 21:03:53.000000000 +0000
@@ -4,7 +4,7 @@
 -1                UNIT NUMBER OF SNAPSHOT FILE (NOT USED IF .LT. 0)
 F        LOGICAL FLAG, T TO REWIND SNAPSHOT FILE AFTER EACH RECORD.
 F        LOGICAL FLAG, T TO STOP ON FAILURES.
-T        LOGICAL FLAG, T TO TEST ERROR EXITS.
+F        LOGICAL FLAG, T TO TEST ERROR EXITS.
 16.0     THRESHOLD VALUE OF TEST RATIO
 6                 NUMBER OF VALUES OF N
 0 1 2 3 5 9       VALUES OF N
--- CLAPACK-3.2.1/BLAS/zblat2.in.orig	2009-08-07 23:32:18.000000000 +0100
+++ CLAPACK-3.2.1/BLAS/zblat2.in	2010-11-28 21:03:48.000000000 +0000
@@ -4,7 +4,7 @@
 -1                UNIT NUMBER OF SNAPSHOT FILE (NOT USED IF .LT. 0)
 F        LOGICAL FLAG, T TO REWIND SNAPSHOT FILE AFTER EACH RECORD.
 F        LOGICAL FLAG, T TO STOP ON FAILURES.
-T        LOGICAL FLAG, T TO TEST ERROR EXITS.
+F        LOGICAL FLAG, T TO TEST ERROR EXITS.
 16.0     THRESHOLD VALUE OF TEST RATIO
 6                 NUMBER OF VALUES OF N
 0 1 2 3 5 9       VALUES OF N
--- CLAPACK-3.2.1/BLAS/cblat3.in.orig	2009-08-07 23:32:18.000000000 +0100
+++ CLAPACK-3.2.1/BLAS/cblat3.in	2010-11-28 21:03:18.000000000 +0000
@@ -4,7 +4,7 @@
 -1                UNIT NUMBER OF SNAPSHOT FILE (NOT USED IF .LT. 0)
 F        LOGICAL FLAG, T TO REWIND SNAPSHOT FILE AFTER EACH RECORD.
 F        LOGICAL FLAG, T TO STOP ON FAILURES.
-T        LOGICAL FLAG, T TO TEST ERROR EXITS.
+F        LOGICAL FLAG, T TO TEST ERROR EXITS.
 16.0     THRESHOLD VALUE OF TEST RATIO
 6                 NUMBER OF VALUES OF N
 0 1 2 3 5 9       VALUES OF N
--- CLAPACK-3.2.1/BLAS/sblat3.in.orig	2009-08-07 23:32:18.000000000 +0100
+++ CLAPACK-3.2.1/BLAS/sblat3.in	2010-11-28 21:03:41.000000000 +0000
@@ -4,7 +4,7 @@
 -1                UNIT NUMBER OF SNAPSHOT FILE (NOT USED IF .LT. 0)
 F        LOGICAL FLAG, T TO REWIND SNAPSHOT FILE AFTER EACH RECORD.
 F        LOGICAL FLAG, T TO STOP ON FAILURES.
-T        LOGICAL FLAG, T TO TEST ERROR EXITS.
+F        LOGICAL FLAG, T TO TEST ERROR EXITS.
 16.0     THRESHOLD VALUE OF TEST RATIO
 6                 NUMBER OF VALUES OF N
 0 1 2 3 5 9       VALUES OF N
--- CLAPACK-3.2.1/BLAS/cblat2.in.orig	2009-08-07 23:32:18.000000000 +0100
+++ CLAPACK-3.2.1/BLAS/cblat2.in	2010-11-28 21:03:12.000000000 +0000
@@ -4,7 +4,7 @@
 -1                UNIT NUMBER OF SNAPSHOT FILE (NOT USED IF .LT. 0)
 F        LOGICAL FLAG, T TO REWIND SNAPSHOT FILE AFTER EACH RECORD.
 F        LOGICAL FLAG, T TO STOP ON FAILURES.
-T        LOGICAL FLAG, T TO TEST ERROR EXITS.
+F        LOGICAL FLAG, T TO TEST ERROR EXITS.
 16.0     THRESHOLD VALUE OF TEST RATIO
 6                 NUMBER OF VALUES OF N
 0 1 2 3 5 9       VALUES OF N
--- CLAPACK-3.2.1/BLAS/dblat3.in.orig	2009-08-07 23:32:18.000000000 +0100
+++ CLAPACK-3.2.1/BLAS/dblat3.in	2010-11-28 21:03:30.000000000 +0000
@@ -4,7 +4,7 @@
 -1                UNIT NUMBER OF SNAPSHOT FILE (NOT USED IF .LT. 0)
 F        LOGICAL FLAG, T TO REWIND SNAPSHOT FILE AFTER EACH RECORD.
 F        LOGICAL FLAG, T TO STOP ON FAILURES.
-T        LOGICAL FLAG, T TO TEST ERROR EXITS.
+F        LOGICAL FLAG, T TO TEST ERROR EXITS.
 16.0     THRESHOLD VALUE OF TEST RATIO
 6                 NUMBER OF VALUES OF N
 0 1 2 3 5 9       VALUES OF N
--- CLAPACK-3.2.1/BLAS/sblat2.in.orig	2009-08-07 23:32:18.000000000 +0100
+++ CLAPACK-3.2.1/BLAS/sblat2.in	2010-11-28 21:03:35.000000000 +0000
@@ -4,7 +4,7 @@
 -1                UNIT NUMBER OF SNAPSHOT FILE (NOT USED IF .LT. 0)
 F        LOGICAL FLAG, T TO REWIND SNAPSHOT FILE AFTER EACH RECORD.
 F        LOGICAL FLAG, T TO STOP ON FAILURES.
-T        LOGICAL FLAG, T TO TEST ERROR EXITS.
+F        LOGICAL FLAG, T TO TEST ERROR EXITS.
 16.0     THRESHOLD VALUE OF TEST RATIO
 6                 NUMBER OF VALUES OF N
 0 1 2 3 5 9       VALUES OF N
--- CLAPACK-3.2.1/BLAS/WRAP/cblas.h.orig	2010-04-27 18:45:15.000000000 +0100
+++ CLAPACK-3.2.1/BLAS/WRAP/cblas.h	2010-11-28 19:59:18.000000000 +0000
@@ -70,10 +70,10 @@
  * Prototypes for level 0 BLAS routines
  * ===========================================================================
  */
-void cblas_srotg(float a, float b, float c, float s);
-void cblas_crotg(complex a, complex b, complex c, float s);
-void cblas_drotg(double a, double b, double c, double s);
-void cblas_zrotg(doublecomplex a, doublecomplex b, doublecomplex c, double s);
+void cblas_srotg(float *a, float *b, float *c, float *s);
+void cblas_crotg(complex *a, complex *b, complex *c, float *s);
+void cblas_drotg(double *a, double *b, double *c, double *s);
+void cblas_zrotg(doublecomplex *a, doublecomplex *b, doublecomplex *c, double *s);
 
 
 /*
--- CLAPACK-3.2.1/BLAS/WRAP/cblaswr.c.orig	2010-04-27 18:45:15.000000000 +0100
+++ CLAPACK-3.2.1/BLAS/WRAP/cblaswr.c	2010-11-28 20:00:06.000000000 +0000
@@ -209,7 +209,7 @@
 		  real* c,
 		  real* s)
 {
-    cblas_srotg(*a, *b, *c, *s);
+    cblas_srotg(a, b, c, s);
     return 0;
 }
 
@@ -219,7 +219,7 @@
           complex* C,
           real* S)
 {
-    cblas_crotg(*CA, *CB, *C, *S);
+    cblas_crotg(CA, CB, C, S);
     return 0;
 }
 
@@ -229,7 +229,7 @@
           doublereal* c,
           doublereal* s)
 {
-    cblas_drotg(*a, *b, *c, *s);
+    cblas_drotg(a, b, c, s);
     return 0;
 }
 int
@@ -238,7 +238,7 @@
           doublecomplex* C,
           doublereal* S)
 {
-    cblas_zrotg(*CA, *CB, *C, *S);
+    cblas_zrotg(CA, CB, C, S);
     return 0;
 }
 /*
