--- clucene-2.3.3.4/src/core/libclucene-core.pc.cmake.orig	2011-03-17 00:21:07.000000000 +0000
+++ clucene-2.3.3.4/src/core/libclucene-core.pc.cmake	2014-11-14 16:50:07.557875932 +0000
@@ -6,6 +6,5 @@
 Name: libclucene
 Description: CLucene - a C++ search engine, ported from the popular Apache Lucene
 Version: @CLUCENE_VERSION_MAJOR@.@CLUCENE_VERSION_MINOR@.@CLUCENE_VERSION_REVISION@.@CLUCENE_VERSION_PATCH@
-Libs: -L${prefix}/@LIB_DESTINATION@/ -lclucene-core
+Libs: -L${prefix}/@LIB_DESTINATION@/ -lclucene-core -lclucene-shared
 Cflags: -I${prefix}/include -I${prefix}/include/CLucene/ext
-~
