Description: convert old patches into DEP-3
Acked-By: Anton Gladky <gladk@debian.org>
Last-Update: 2013-03-03

--- a/include/Inventor/SbBasic.h
+++ b/include/Inventor/SbBasic.h
@@ -33,6 +33,7 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 \**************************************************************************/
 
+#include <Inventor/C/errors/debugerror.h>
 #include <Inventor/C/basic.h>
 
 /* ********************************************************************** */
