Description: Remove installation of exe-files
Author: Anton Gladky <gladk@debian.org>
Bug-Debian: https://bugs.debian.org/795241
Last-Update: 2015-08-12

Index: Coin3D-coin-abc9f50968c9/Makefile.in
===================================================================
--- Coin3D-coin-abc9f50968c9.orig/Makefile.in
+++ Coin3D-coin-abc9f50968c9/Makefile.in
@@ -1420,8 +1420,6 @@ EXTRA_DIST = \
 	cfg/errors.txt \
 	cfg/gendsp.pl.in \
 	cfg/ltconfig \
-	cfg/wrapmsvc.exe \
-	cfg/csubst.exe \
 	cfg/doxy4win.pl \
 	docs/coin.doxygen.in \
 	docs/oiki-launch.txt \
@@ -2242,12 +2240,10 @@ doxygen-docs: built-sources
 @BUILD_LIBRARY_TRUE@@MAC_FRAMEWORK_TRUE@  # Link to SoQt.framework to allow <Inventor/Qt/xxx.h> includes. 
 @BUILD_LIBRARY_TRUE@@MAC_FRAMEWORK_TRUE@	cd "$(DESTDIR)$(MAC_FRAMEWORK_PREFIX)/$(MAC_FRAMEWORK_NAME).framework/Versions/$(MAC_FRAMEWORK_VERSION)/Headers" && rm -f Qt && ln -s ../../../../SoQt.framework/Headers Qt && cd -;
 @BUILD_LIBRARY_TRUE@@MAC_FRAMEWORK_FALSE@install-data-local:
-@BUILD_LIBRARY_TRUE@@MAC_FRAMEWORK_FALSE@install-exec-local: $(srcdir)/cfg/wrapmsvc.exe
 @BUILD_LIBRARY_TRUE@@MAC_FRAMEWORK_FALSE@	@$(NORMAL_INSTALL)
 @BUILD_LIBRARY_TRUE@@MAC_FRAMEWORK_FALSE@	@if @BUILD_WITH_MSVC@; then \
 @BUILD_LIBRARY_TRUE@@MAC_FRAMEWORK_FALSE@	  echo " $(mkinstalldirs) $(DESTDIR)$(bindir)"; \
 @BUILD_LIBRARY_TRUE@@MAC_FRAMEWORK_FALSE@	  $(mkinstalldirs) $(DESTDIR)$(bindir); \
-@BUILD_LIBRARY_TRUE@@MAC_FRAMEWORK_FALSE@	  p=wrapmsvc.exe; \
 @BUILD_LIBRARY_TRUE@@MAC_FRAMEWORK_FALSE@	  echo " $(INSTALL_PROGRAM) $(srcdir)/cfg/$$p $(DESTDIR)$(bindir)/$$p"; \
 @BUILD_LIBRARY_TRUE@@MAC_FRAMEWORK_FALSE@	  $(INSTALL_PROGRAM) $(srcdir)/cfg/$$p $(DESTDIR)$(bindir)/$$p; \
 @BUILD_LIBRARY_TRUE@@MAC_FRAMEWORK_FALSE@	else :; fi
