--- colord-1.4.4/lib/compat/cd-compat-edid.c.orig	2019-03-08 12:43:10.000000000 +0000
+++ colord-1.4.4/lib/compat/cd-compat-edid.c	2020-07-16 21:54:55.701288583 +0100
@@ -331,9 +331,8 @@
 	/* get the default profile for the device */
 	profile = cd_device_get_default_profile (device);
 	if (profile == NULL) {
-		g_printerr ("No profile for %s: %s",
-			    cd_device_get_id (device),
-			    error->message);
+		g_printerr ("No profile for %s",
+			    cd_device_get_id (device));
 		return CD_EDID_ERROR_NO_PROFILE;
 	}
 
