--- cryptopp-5_6_5/CMakeLists.txt.orig	2016-10-11 05:02:23.000000000 +0100
+++ cryptopp-5_6_5/CMakeLists.txt	2017-01-20 21:06:42.273848384 +0000
@@ -399,8 +399,8 @@
 if(NOT CMAKE_VERSION VERSION_LESS 2.8.8)	# CMakePackageConfigHelpers is supported from 2.8.8
 	include(CMakePackageConfigHelpers)
 	write_basic_package_version_file("${PROJECT_BINARY_DIR}/cryptopp-config-version.cmake" VERSION ${cryptopp_VERSION_MAJOR}.${cryptopp_VERSION_MINOR}.${cryptopp_VERSION_PATCH} COMPATIBILITY SameMajorVersion)
-	install(FILES cryptopp-config.cmake ${PROJECT_BINARY_DIR}/cryptopp-config-version.cmake DESTINATION "lib/cmake/cryptopp")
-	install(EXPORT ${export_name} DESTINATION "lib/cmake/cryptopp")
+	install(FILES cryptopp-config.cmake ${PROJECT_BINARY_DIR}/cryptopp-config-version.cmake DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/cryptopp")
+	install(EXPORT ${export_name} DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/cryptopp")
 endif()
 
 # Tests
