--- crystalspace-src-1.0rc1/plugins/physics/bullet/bullet.cpp.orig	2006-11-29 07:12:28.000000000 +0000
+++ crystalspace-src-1.0rc1/plugins/physics/bullet/bullet.cpp	2006-11-29 20:13:47.000000000 +0000
@@ -31,7 +31,7 @@
 #include <CcdPhysics/CcdPhysicsController.h>
 #include <Dynamics/RigidBody.h>
 #include <CcdPhysics/CcdPhysicsEnvironment.h>
-#include <ConstraintSolver/SimpleConstraintSolver.h>
+//#include <ConstraintSolver/SimpleConstraintSolver.h>
 #include <BroadphaseCollision/SimpleBroadphase.h>
 #include <CollisionShapes/BoxShape.h>
 #include <CollisionShapes/EmptyShape.h>
