--- crystalspace-1.4.20090607/plugins/cscript/cspython/cspython.cpp.orig	2009-06-07 09:51:00.000000000 +0100
+++ crystalspace-1.4.20090607/plugins/cscript/cspython/cspython.cpp	2009-06-07 11:38:07.000000000 +0100
@@ -562,7 +562,7 @@
 }
 void* csPython::Object::GetPointer ()
 {
-  PySwigObject *sobj = SWIG_Python_GetSwigThis(self);
+  SwigPyObject *sobj = SWIG_Python_GetSwigThis(self);
   if (sobj)
     return sobj->ptr;
   else
