--- crystalspace-1.4.0/plugins/cscript/csjava/Jamfile.orig	2009-11-28 16:10:17.000000000 +0000
+++ crystalspace-1.4.0/plugins/cscript/csjava/Jamfile	2010-06-23 07:42:09.000000000 +0100
@@ -70,7 +70,7 @@
   Swig $(common) :
     [ DoSourceGrist cspace.i ] :
     java :
-    -c++ -java -package org.crystalspace3d :
+    -c++ -package org.crystalspace3d :
     [ ConcatDirs $(TOP) include ]
     [ ConcatDirs $(BUILDTOP) include ] :
     [ ConcatDirs $(TOP) include bindings ] :
