--- Csound-5.10.1/install.py.orig	2008-11-19 19:17:09.000000000 +0000
+++ Csound-5.10.1/install.py	2009-01-08 22:28:34.000000000 +0000
@@ -333,7 +333,7 @@
   for i in xmgList:
     makeDir(concatPath([xmgDir, i, 'LC_MESSAGES']))
     src = 'po/' + i + '/LC_MESSAGES/csound5.mo'
-    fileName = concatPath([xmgDir, i, 'LC_MESSAGES/csound5.mo'])
+    fileName = concatPath([instDir, xmgDir, i, 'LC_MESSAGES/csound5.mo'])
     err = runCmd(['install', '-p', '-m', '0644', src, fileName])
     if err == 0:
         addMD5(fileName, fileName)
