--- ctl-1.5.2/OpenEXR_CTL/CMakeLists.txt.orig	2014-06-03 02:11:24.000000000 +0100
+++ ctl-1.5.2/OpenEXR_CTL/CMakeLists.txt	2018-10-23 22:24:52.613729219 +0100
@@ -13,7 +13,7 @@
 
 if ( PKG_CONFIG_FOUND )
 configure_file(../config/OpenEXR_CTL.pc.in "${PROJECT_BINARY_DIR}/OpenEXR_CTL.pc" @ONLY)
-install( FILES "${PROJECT_BINARY_DIR}/OpenEXR_CTL.pc" DESTINATION lib/pkgconfig COMPONENT dev )
+install( FILES "${PROJECT_BINARY_DIR}/OpenEXR_CTL.pc" DESTINATION lib${LIB_SUFFIX}/pkgconfig COMPONENT dev )
 endif()
 
 # TODO CMake config
--- ctl-1.5.2/CMakeLists.txt.orig	2014-06-03 02:11:24.000000000 +0100
+++ ctl-1.5.2/CMakeLists.txt	2018-10-23 22:23:04.038734399 +0100
@@ -14,7 +14,7 @@
 if( WIN32 AND NOT CYGWIN )
   set(DEF_INSTALL_CMAKE_DIR CMake)
 else()
-  set(DEF_INSTALL_CMAKE_DIR lib/CMake/CTL)
+  set(DEF_INSTALL_CMAKE_DIR lib${LIB_SUFFIX}/CMake/CTL)
 endif()
 set(INSTALL_CMAKE_DIR ${DEF_INSTALL_CMAKE_DIR} CACHE PATH "Install directory for project CMake files" )
 
@@ -28,9 +28,9 @@
 # which point to directories outside the build tree to the install RPATH
 SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 # the RPATH to be used when installing, but only if it's not a system directory
-LIST(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_PREFIX}/lib" isSystemDir)
+LIST(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}" isSystemDir)
 IF("${isSystemDir}" STREQUAL "-1")
-  SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+  SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
 ENDIF("${isSystemDir}" STREQUAL "-1")
 
 
@@ -77,7 +77,7 @@
 
 if ( PKG_CONFIG_FOUND )
 configure_file(config/CTL.pc.in "${PROJECT_BINARY_DIR}/CTL.pc" @ONLY)
-install( FILES "${PROJECT_BINARY_DIR}/CTL.pc" DESTINATION lib/pkgconfig COMPONENT dev )
+install( FILES "${PROJECT_BINARY_DIR}/CTL.pc" DESTINATION lib${LIB_SUFFIX}/pkgconfig COMPONENT dev )
 endif()
 
 install( FILES
--- ctl-1.5.2/lib/IlmImfCtl/CMakeLists.txt.orig	2014-06-03 02:11:24.000000000 +0100
+++ ctl-1.5.2/lib/IlmImfCtl/CMakeLists.txt	2018-10-23 22:23:56.829731880 +0100
@@ -13,4 +13,4 @@
 
 export( TARGETS IlmImfCtl IlmCtl FILE "${PROJECT_BINARY_DIR}/CTLLibraryDepends.cmake" )
 
-install( TARGETS IlmImfCtl DESTINATION lib )
+install( TARGETS IlmImfCtl DESTINATION lib${LIB_SUFFIX} )
--- ctl-1.5.2/lib/IlmCtlSimd/CMakeLists.txt.orig	2014-06-03 02:11:24.000000000 +0100
+++ ctl-1.5.2/lib/IlmCtlSimd/CMakeLists.txt	2018-10-23 22:23:45.645732414 +0100
@@ -42,4 +42,4 @@
 
 install( FILES CtlSimdInterpreter.h DESTINATION include/CTL )
 
-install( TARGETS IlmCtlSimd DESTINATION lib )
+install( TARGETS IlmCtlSimd DESTINATION lib${LIB_SUFFIX} )
--- ctl-1.5.2/lib/IlmCtlMath/CMakeLists.txt.orig	2014-06-03 02:11:24.000000000 +0100
+++ ctl-1.5.2/lib/IlmCtlMath/CMakeLists.txt	2018-10-23 22:23:36.265732862 +0100
@@ -20,6 +20,6 @@
   CtlSparseMatrix.h
  DESTINATION include/CTL )
 
-install( TARGETS IlmCtlMath DESTINATION lib )
+install( TARGETS IlmCtlMath DESTINATION lib${LIB_SUFFIX} )
 
 # TODO Add Imath and Iex libadd dependency and version info
--- ctl-1.5.2/lib/IlmCtl/CMakeLists.txt.orig	2014-06-03 02:11:24.000000000 +0100
+++ ctl-1.5.2/lib/IlmCtl/CMakeLists.txt	2018-10-23 22:23:25.959733353 +0100
@@ -50,4 +50,4 @@
 	CtlVersion.h
  DESTINATION include/CTL )
 
-install( TARGETS IlmCtl DESTINATION lib )
+install( TARGETS IlmCtl DESTINATION lib${LIB_SUFFIX} )
