--- cups-filters-1.20.4/scripting/php/phpcups.h.orig	2018-02-07 22:08:45.000000000 +0000
+++ cups-filters-1.20.4/scripting/php/phpcups.h	2018-07-13 19:59:42.710420512 +0100
@@ -19,7 +19,7 @@
 
 #  include <cups/cups.h>
 #  include <cups/language.h>
-#  include <cups/debug-private.h>
+//#  include <cups/debug-private.h>
 #  include <fcntl.h>
 #  include <sys/stat.h>
 #  if defined(WIN32) || defined(__EMX__)
--- cups-filters-1.20.4/scripting/php/phpcups.c.orig	2018-02-07 22:08:45.000000000 +0000
+++ cups-filters-1.20.4/scripting/php/phpcups.c	2018-07-13 19:59:42.710420512 +0100
@@ -26,7 +26,7 @@
  * Include necessary headers...
  */
 
-#include <cups/string-private.h>
+//#include <cups/string-private.h>
 #include "php.h"
 #include "php_ini.h"
 #include "ext/standard/info.h"
--- cups-filters-1.20.4/filter/pdf.cxx.orig	2018-02-07 22:08:42.000000000 +0000
+++ cups-filters-1.20.4/filter/pdf.cxx	2018-07-20 19:53:21.560506301 +0100
@@ -734,7 +734,7 @@
         }
 
         FormField *ff = fm_text->getField();
-        GooString *field_name;
+        const GooString *field_name;
         field_name = ff->getFullyQualifiedName();
         if ( ! field_name )
             field_name = ff->getPartialName();
