--- cups-filters-1.21.4/scripting/php/phpcups.h.orig	2018-11-23 21:18:49.000000000 +0000
+++ cups-filters-1.21.4/scripting/php/phpcups.h	2018-11-24 09:15:30.912568253 +0000
@@ -19,7 +19,7 @@
 
 #  include <cups/cups.h>
 #  include <cups/language.h>
-#  include <cups/debug-private.h>
+//#  include <cups/debug-private.h>
 #  include <fcntl.h>
 #  include <sys/stat.h>
 #  if defined(WIN32) || defined(__EMX__)
--- cups-filters-1.21.4/scripting/php/phpcups.c.orig	2018-11-23 21:18:49.000000000 +0000
+++ cups-filters-1.21.4/scripting/php/phpcups.c	2018-11-24 09:15:30.912568253 +0000
@@ -26,7 +26,7 @@
  * Include necessary headers...
  */
 
-#include <cups/string-private.h>
+//#include <cups/string-private.h>
 #include "php.h"
 #include "php_ini.h"
 #include "ext/standard/info.h"
--- cups-filters-1.21.4/filter/pdftoraster.cxx.orig	2018-11-23 21:18:49.000000000 +0000
+++ cups-filters-1.21.4/filter/pdftoraster.cxx	2018-11-24 09:17:17.717563157 +0000
@@ -92,6 +92,10 @@
 #include <lcms2.h>
 #endif
 
+typedef bool GBool;
+#define gTrue true
+#define gFalse false
+
 #define MAX_CHECK_COMMENT_LINES	20
 #define MAX_BYTES_PER_PIXEL 32
 
