--- d4x-2.5.7.1/main/path.cc.orig	2008-05-09 11:36:59.000000000 +0100
+++ d4x-2.5.7.1/main/path.cc	2008-05-09 11:37:09.000000000 +0100
@@ -3,13 +3,15 @@
 using namespace d4x;
 
 void Path::normalize(){
-	size_type p=npos;
+	size_type p=0;
 	while((p=find("//",p))!=npos){
 		replace(p,2,"/");
 	};
+	p = 0;
 	while((p=find("/./",p))!=npos){
 		replace(p,3,"/");
 	};
+	p = 0;
 	while((p=find("/../"),p)!=npos){
 		size_type r=rfind('/',p-1);
 		if (r!=npos){
diff -ur d4x-2.5.7.1/configure d4x-2.5.7.1.new/configure
--- d4x-2.5.7.1/configure	2006-04-12 17:48:29.000000000 +0100
+++ d4x-2.5.7.1.new/configure	2008-01-22 11:25:45.000000000 +0000
@@ -18397,10 +18397,6 @@
 _ACEOF
 
 
-cat >>confdefs.h <<\_ACEOF
-#define GTK_DISABLE_DEPRECATED
-_ACEOF
-
 
 cat >>confdefs.h <<\_ACEOF
 #define _FILE_OFFSET_BITS 64
diff -ur d4x-2.5.7.1/main/face/list.cc d4x-2.5.7.1.new/main/face/list.cc
--- d4x-2.5.7.1/main/face/list.cc	2006-03-20 00:29:41.000000000 +0000
+++ d4x-2.5.7.1.new/main/face/list.cc	2008-01-22 10:49:08.000000000 +0000
@@ -8,7 +8,7 @@
  *	but WITHOUT ANY WARRANTY; without even the implied warranty of
  *	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
  */
-#define GTK_DISABLE_DEPRECATED
+//#define GTK_DISABLE_DEPRECATED
 
 #include <stdio.h>
 #include <gtk/gtk.h>
diff -ur d4x-2.5.7.1/main/httpd.cc d4x-2.5.7.1.new/main/httpd.cc
--- d4x-2.5.7.1/main/httpd.cc	2006-04-05 20:01:41.000000000 +0100
+++ d4x-2.5.7.1.new/main/httpd.cc	2008-01-23 15:29:40.000000000 +0000
@@ -124,10 +124,24 @@
 	if (tmp==NULL) return(NULL);
 	tmp=skip_spaces(tmp+1);
 	char *sqs=index(s,';');
-	if (sqs==NULL) sqs=tmp+strlen(tmp)-1;
-	else sqs-=1;
+	if (sqs==NULL) sqs=tmp+strlen(tmp);
 	int len=sqs-tmp;
+	while (len && isspace(tmp[len-1]))
+	  len--;
 	if (len<=0) return(NULL);
+	
+	if (*tmp == '"')
+	  {
+	    tmp++;
+	    len--;
+	    if (len<=0) return(NULL);
+	    if (tmp[len - 1] == '"')
+	      {
+	        len--;
+		if (len<=0) return(NULL);
+	      }
+	  }
+	
 	char *rval=copy_string2(tmp,len);
 	for (int i=len-1;i>=0;i++){
 		if (isspace(rval[i])) rval[i]=0;
@@ -198,6 +212,7 @@
 	ETag=NULL;
 	Auth=NULL;
 	D_FILE.type=T_FILE; //we don't know any other when download via http
+	D_FILE.date=time(NULL);
 	config.copy_ints(cfg);
 	if (cfg->split){
 		config.retry=0;
@@ -211,6 +226,7 @@
 	REQUESTED_URL=make_name();
 	if (!s)
 		return reconnect();
+reconnect();
 	HTTP->import_ctrl_socket(s);
 	RetrNum=1;
 	tDownloader::reconnect();
diff -ur d4x-2.5.7.1/main/locstr.cc d4x-2.5.7.1.new/main/locstr.cc
--- d4x-2.5.7.1/main/locstr.cc	2005-11-11 21:38:45.000000000 +0000
+++ d4x-2.5.7.1.new/main/locstr.cc	2008-01-22 10:49:08.000000000 +0000
@@ -1362,6 +1362,8 @@
 };
 
 int file_copy(char *from,char *to,char *buf,int size){
+	if (equal(from, to))
+		return 0;
 	FILE *src=fopen(from,"r");
 	if (src==NULL){
 		return(-1);
diff -ur d4x-2.5.7.1/main/sslsocket.cc d4x-2.5.7.1.new/main/sslsocket.cc
--- d4x-2.5.7.1/main/sslsocket.cc	2005-09-16 05:27:04.000000000 +0100
+++ d4x-2.5.7.1.new/main/sslsocket.cc	2008-01-23 14:25:15.000000000 +0000
@@ -23,8 +23,12 @@
 		return(SOCKET_CANT_CONNECT);
 	con_flag=0;
 	SSL_set_fd(Handler,fd);
-	if (SSL_connect(Handler)!=1)
+	int rv;
+	if ((rv = SSL_connect(Handler))!=1)
+	{
+//		::printf("ssl error %d %d\n", rv, SSL_get_error(Handler, rv));
 		return(SOCKET_CANT_CONNECT);
+	}
 	con_flag=1;
 	return 0;
 };
@@ -34,8 +38,12 @@
 		return(SOCKET_CANT_CONNECT);
 	con_flag=0;
 	SSL_set_fd(Handler,fd);
-	if (SSL_connect(Handler)!=1)
+	int rv;
+	if ((rv = SSL_connect(Handler))!=1)
+	{
+//		::printf("ssl error %d %d\n", rv, SSL_get_error(Handler, rv));
 		return(SOCKET_CANT_CONNECT);
+	}
 	con_flag=1;
 	return 0;
 };
