--- darktable-4.4.2/src/external/rawspeed/data/cameras.xsd.orig	2023-06-12 22:39:40.000000000 +0100
+++ darktable-4.4.2/src/external/rawspeed/data/cameras.xsd	2023-12-14 21:29:17.005290379 +0000
@@ -193,10 +193,7 @@
   </xs:simpleType>
   <xs:complexType name="CFA2Type">
     <xs:sequence>
-      <xs:choice>
-        <xs:element type="ColorRowType" name="ColorRow" minOccurs="2" maxOccurs="2"/>
-        <xs:element type="ColorRowType" name="ColorRow" minOccurs="6" maxOccurs="6"/>
-      </xs:choice>
+      <xs:element type="ColorRowType" name="ColorRow" minOccurs="2" maxOccurs="6"/>
     </xs:sequence>
     <xs:attribute type="CFA2TypeWidthType" name="width" use="required"/>
     <xs:attribute type="CFA2TypeHeightType" name="height" use="required"/>
