--- deja-dup-48.4/app/main.vala.orig	2025-07-26 20:10:06.000000000 +0100
+++ deja-dup-48.4/app/main.vala	2025-10-04 13:58:03.220613156 +0100
@@ -379,7 +379,7 @@
       <p>48.4</p>
       <ul>
         <li>Fix Rclone fresh-system backups from working without needing to edit
-            remote & folder in preferences dialog (i.e. stop having deja-dup ignore
+            remote &amp; folder in preferences dialog (i.e. stop having deja-dup ignore
             Rclone settings in the "restore from previous backup" dialog)</li>
         <li>Fix "last backup time" label from showing "Yesterday" instead of "Today"</li>
         <li>Restic: fix cache files accidentally being included in backup</li>
