--- DevIL-1.7.8/configure.orig	2009-03-08 20:43:20.000000000 +0000
+++ DevIL-1.7.8/configure	2010-06-06 06:47:35.000000000 +0100
@@ -19489,13 +19489,13 @@
 
 fi
 if test $ac_cv_header_png_h = yes; then
-  { echo "$as_me:$LINENO: checking for main in -lpng12" >&5
-echo $ECHO_N "checking for main in -lpng12... $ECHO_C" >&6; }
-if test "${ac_cv_lib_png12_main+set}" = set; then
+  { echo "$as_me:$LINENO: checking for main in -lpng14" >&5
+echo $ECHO_N "checking for main in -lpng14... $ECHO_C" >&6; }
+if test "${ac_cv_lib_png14_main+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lpng12  $LIBS"
+LIBS="-lpng14  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -19530,33 +19530,33 @@
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext &&
        $as_test_x conftest$ac_exeext; then
-  ac_cv_lib_png12_main=yes
+  ac_cv_lib_png14_main=yes
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
 
-	ac_cv_lib_png12_main=no
+	ac_cv_lib_png14_main=no
 fi
 
 rm -f core conftest.err conftest.$ac_objext conftest_ipa8_conftest.oo \
       conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-{ echo "$as_me:$LINENO: result: $ac_cv_lib_png12_main" >&5
-echo "${ECHO_T}$ac_cv_lib_png12_main" >&6; }
-if test $ac_cv_lib_png12_main = yes; then
-  IL_LIBS="-lpng12 $IL_LIBS"
-                                         have_png12="yes"
+{ echo "$as_me:$LINENO: result: $ac_cv_lib_png14_main" >&5
+echo "${ECHO_T}$ac_cv_lib_png14_main" >&6; }
+if test $ac_cv_lib_png14_main = yes; then
+  IL_LIBS="-lpng14 $IL_LIBS"
+                                         have_png14="yes"
 else
-  have_png12="no"
+  have_png14="no"
 fi
 
 else
-  have_png12="no"
+  have_png14="no"
 fi
 
 
-          if test "x$have_png12" = "xno"; then
+          if test "x$have_png14" = "xno"; then
   if test "${ac_cv_header_png_h+set}" = set; then
   { echo "$as_me:$LINENO: checking for png.h" >&5
 echo $ECHO_N "checking for png.h... $ECHO_C" >&6; }
@@ -19757,7 +19757,7 @@
 
                  lib_test_result="$have_png"
 else
-  lib_test_result="$have_png12"
+  lib_test_result="$have_png14"
 fi
 
           if test "x$lib_test_result" = "xyes"; then
--- DevIL-1.7.8/src-IL/src/il_png.c.orig	2009-03-08 07:10:09.000000000 +0000
+++ DevIL-1.7.8/src-IL/src/il_png.c	2010-06-06 06:51:45.000000000 +0100
@@ -278,7 +278,7 @@
 
 	// Expand low-bit-depth grayscale images to 8 bits
 	if (png_color_type == PNG_COLOR_TYPE_GRAY && bit_depth < 8) {
-		png_set_gray_1_2_4_to_8(png_ptr);
+		png_set_expand_gray_1_2_4_to_8(png_ptr);
 	}
 
 	// Expand RGB images with transparency to full alpha channels
--- DevIL-1.7.8/src-IL/src/il_icon.c.orig	2009-03-08 07:10:09.000000000 +0000
+++ DevIL-1.7.8/src-IL/src/il_icon.c	2010-06-06 06:51:25.000000000 +0100
@@ -525,7 +525,7 @@
 
 	// Expand low-bit-depth grayscale images to 8 bits
 	if (ico_color_type == PNG_COLOR_TYPE_GRAY && bit_depth < 8) {
-		png_set_gray_1_2_4_to_8(ico_png_ptr);
+		png_set_expand_gray_1_2_4_to_8(ico_png_ptr);
 	}
 
 	// Expand RGB images with transparency to full alpha channels
