--- DevIL-1.6.8-rc2/include/IL/ilut.h.orig	2006-07-04 15:43:23.000000000 +0100
+++ DevIL-1.6.8-rc2/include/IL/ilut.h	2007-11-12 22:31:46.000000000 +0000
@@ -234,7 +234,7 @@
 extern "C" {
 #endif
 
-	ILAPI BITMAP* ILAPIENTRY ilutAllegLoadImage(Lstring FileName);
+	ILAPI BITMAP* ILAPIENTRY ilutAllegLoadImage(ILstring FileName);
 	ILAPI BITMAP* ILAPIENTRY ilutConvertToAlleg(PALETTE Pal);
 #endif//ILUT_USE_ALLEGRO
 
