--- dia-0.96-pre1/app/Makefile.in-	2006-12-13 21:35:17.000000000 -0500
+++ dia-0.96-pre1/app/Makefile.in	2006-12-13 21:35:43.000000000 -0500
@@ -407,6 +407,8 @@
 	$(LIBART_CFLAGS) \
 	-DPREFIX=\""$(prefix)"\" \
 	-DSYSCONFDIR=\""$(sysconfdir)"\" \
+	-DDATADIR=\""$(datadir)"\" \
+	-DLIBDIR=\""$(libdir)"\" \
 	-DDIA_DATADIR=\""$(datadir)"\" \
 	-DDIA_LIBDIR=\""$(libdir)"\" \
 	-DLOCALEDIR=\"$(localedir)\" \
