--- doxygen-1.9.1/CMakeLists.txt.orig	2020-12-21 14:08:57.000000000 +0000
+++ doxygen-1.9.1/CMakeLists.txt	2021-08-08 13:49:14.337440429 +0100
@@ -86,6 +86,9 @@
 if(POLICY CMP0063)
   cmake_policy(SET CMP0063 NEW)
 endif()
+if(POLICY CMP0070)
+  cmake_policy(SET CMP0079 NEW)
+endif()
 set(CMAKE_CXX_VISIBILITY_PRESET hidden)
 set(CMAKE_C_VISIBILITY_PRESET hidden)
 set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
--- doxygen-1.9.1/src/CMakeLists.txt.orig	2020-12-24 14:02:11.000000000 +0000
+++ doxygen-1.9.1/src/CMakeLists.txt	2021-08-08 15:56:07.520345390 +0100
@@ -335,7 +335,7 @@
     if (static_libclang)
         set(CLANG_LIBS libclang clangTooling)
     else() # dynamically linked version of clang
-        llvm_config(doxymain USE_SHARED support)
+        llvm_config(doxymain support)
         set(CLANG_LIBS libclang clang-cpp)
     endif()
     target_compile_definitions(doxygen PRIVATE ${LLVM_DEFINITIONS})
--- doxygen-1.9.1/addon/doxyapp/CMakeLists.txt.orig	2021-01-04 16:10:06.000000000 +0000
+++ doxygen-1.9.1/addon/doxyapp/CMakeLists.txt	2021-08-08 14:52:00.978771978 +0100
@@ -28,7 +28,7 @@
     if (static_libclang)
         set(CLANG_LIBS libclang clangTooling)
     else() # dynamically linked version of clang
-        llvm_config(doxymain USE_SHARED support)
+        llvm_config(doxymain support)
         set(CLANG_LIBS libclang clang-cpp)
     endif()
     target_compile_definitions(doxyapp PRIVATE ${LLVM_DEFINITIONS})
--- doxygen-1.9.1/addon/doxyparse/CMakeLists.txt.orig	2020-09-16 19:09:04.000000000 +0100
+++ doxygen-1.9.1/addon/doxyparse/CMakeLists.txt	2021-08-08 14:52:11.079771496 +0100
@@ -18,7 +18,7 @@
     if (static_libclang)
         set(CLANG_LIBS libclang clangTooling ${llvm_libs})
     else()
-        set(CLANG_LIBS libclang clang-cpp ${llvm_libs})
+        set(CLANG_LIBS libclang clang-cpp)
     endif()
 endif()
 
