--- doxygen-1.9.2/src/CMakeLists.txt.orig	2021-05-12 19:47:52.000000000 +0100
+++ doxygen-1.9.2/src/CMakeLists.txt	2021-08-19 21:53:42.733604966 +0100
@@ -344,8 +344,8 @@
     if (static_libclang)
         set(CLANG_LIBS libclang clangTooling)
     else() # dynamically linked version of clang
-        llvm_config(doxymain USE_SHARED support)
-        set(CLANG_LIBS libclang clang-cpp)
+        llvm_config(doxymain support)
+        set(CLANG_LIBS libclang clangTooling)
     endif()
     target_compile_definitions(doxygen PRIVATE ${LLVM_DEFINITIONS})
 endif()
--- doxygen-1.9.2/addon/doxyapp/CMakeLists.txt.orig	2021-05-12 19:44:30.000000000 +0100
+++ doxygen-1.9.2/addon/doxyapp/CMakeLists.txt	2021-08-19 21:53:42.733604966 +0100
@@ -27,8 +27,8 @@
     if (static_libclang)
         set(CLANG_LIBS libclang clangTooling)
     else() # dynamically linked version of clang
-        llvm_config(doxymain USE_SHARED support)
-        set(CLANG_LIBS libclang clang-cpp)
+        llvm_config(doxymain support)
+        set(CLANG_LIBS libclang clangTooling)
     endif()
     target_compile_definitions(doxyapp PRIVATE ${LLVM_DEFINITIONS})
 endif()
--- doxygen-1.9.2/addon/doxyparse/CMakeLists.txt.orig	2021-05-12 19:44:30.000000000 +0100
+++ doxygen-1.9.2/addon/doxyparse/CMakeLists.txt	2021-08-19 21:53:42.733604966 +0100
@@ -15,9 +15,9 @@
 
 if (use_libclang)
     if (static_libclang)
-        set(CLANG_LIBS libclang clangTooling ${llvm_libs})
+        set(CLANG_LIBS libclang clangTooling)
     else()
-        set(CLANG_LIBS libclang clang-cpp ${llvm_libs})
+        set(CLANG_LIBS libclang clangTooling)
     endif()
 endif()
 
