--- dvb-apps-20090902/lib/libdvbapi/dvbaudio.c.orig	2009-09-02 21:47:32.000000000 +0100
+++ dvb-apps-20090902/lib/libdvbapi/dvbaudio.c	2009-09-02 22:13:55.000000000 +0100
@@ -25,6 +25,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <sys/ioctl.h>
+#include <stdint.h>
 #include <linux/dvb/audio.h>
 #include <errno.h>
 #include "dvbaudio.h"
--- dvb-apps-20090902/test/test_av.c.orig	2009-09-02 21:47:32.000000000 +0100
+++ dvb-apps-20090902/test/test_av.c	2009-09-02 22:15:05.000000000 +0100
@@ -30,6 +30,7 @@
 #include <time.h>
 #include <unistd.h>
 
+#include <stdint.h>
 #include <linux/types.h>
 #include <linux/dvb/audio.h>
 #include <linux/dvb/video.h>
--- dvb-apps-20090902/util/scan/diseqc.c.orig	2009-09-02 21:47:32.000000000 +0100
+++ dvb-apps-20090902/util/scan/diseqc.c	2009-09-02 22:16:53.000000000 +0100
@@ -1,6 +1,6 @@
-#include <linux/dvb/frontend.h>
 #include <sys/ioctl.h>
 #include <time.h>
+#include <linux/dvb/frontend.h>
 
 #include "scan.h"
 #include "diseqc.h"
