--- dvd+rw-tools-7.0/growisofs.c	2006-11-21 23:17:08.000000000 -0200
+++ dvd+rw-tools-7.0/growisofs.c.phys	2006-11-21 23:17:18.000000000 -0200
@@ -3096,12 +3096,12 @@ int main (int argc, char *argv[])
 
 #if defined(__unix) || defined(__unix__)
 
-#if defined(_SC_PAGESIZE) && defined(_SC_AVPHYS_PAGES)
-    { size_t phys_mem = (size_t)sysconf(_SC_AVPHYS_PAGES) *
+#if defined(_SC_PAGESIZE) && defined(_SC_PHYS_PAGES)
+    { size_t phys_mem = (size_t)sysconf(_SC_PHYS_PAGES) *
 			(size_t)sysconf(_SC_PAGESIZE);
 
 	if (phys_mem)
-	{   phys_mem /= 2;  /* normally AVPHYS is a bit smaller, so
+	{   phys_mem /= 2;  /* normally PHYS is a bit smaller, so
 			     *  we commonly land on 1/4 RAM */	
 	    while (the_buffer_size > phys_mem) the_buffer_size /= 2;
 	}
