--- DVDStyler-1.5.1_1/configure.orig	2007-07-20 07:02:48.000000000 +0100
+++ DVDStyler-1.5.1_1/configure	2007-08-10 22:35:54.000000000 +0100
@@ -4402,7 +4402,7 @@
       { echo "$as_me:$LINENO: result: yes (version $WX_VERSION)" >&5
 echo "${ECHO_T}yes (version $WX_VERSION)" >&6; }
 
-      WX_LIBS=`$WX_CONFIG_WITH_ARGS --libs core,base,html`
+      WX_LIBS=`$WX_CONFIG_WITH_ARGS --libs core,base,html,adv`
 
             wx_has_cppflags=""
       if test $wx_config_major_version -gt 2; then
