--- dvgrab-2.0/filehandler.cc.orig	2005-04-18 15:24:52.000000000 +0100
+++ dvgrab-2.0/filehandler.cc	2006-09-29 21:48:39.000000000 +0100
@@ -906,9 +906,10 @@
 {
 	if ( fd )
 	{
-		struct stat file_status;
-		fstat( fileno( fd->stream ), &file_status );
-		return file_status.st_size;
+//		struct stat file_status;
+//		fstat( fileno( fd->stream ), &file_status );
+//		return file_status.st_size;
+		return 0;
 	}
 	else
 		return 0;
