--- elogind-238.1/src/shared/musl_missing.c.orig	2018-06-25 15:41:00.000000000 +0100
+++ elogind-238.1/src/shared/musl_missing.c	2018-09-28 20:25:09.302546575 +0100
@@ -34,10 +34,10 @@
 
 static void elogind_free_program_name(void) {
 
-        if (program_invocation_name && (program_invocation_name != program_arg_name) && strlen(program_invocation_name))
-                program_invocation_name       = mfree(program_invocation_name);
-        if (program_invocation_short_name && (program_invocation_short_name != program_arg_name) && strlen(program_invocation_short_name))
-                program_invocation_short_name = mfree(program_invocation_short_name);
+//        if (program_invocation_name && (program_invocation_name != program_arg_name) && strlen(program_invocation_name))
+//                program_invocation_name       = mfree(program_invocation_name);
+//        if (program_invocation_short_name && (program_invocation_short_name != program_arg_name) && strlen(program_invocation_short_name))
+//                program_invocation_short_name = mfree(program_invocation_short_name);
 }
 
 void elogind_set_program_name(const char* pcall) {
