--- digitizer-4_0/src/digitmain.cpp.orig	2007-01-03 01:54:13.000000000 +0000
+++ digitizer-4_0/src/digitmain.cpp	2007-01-22 14:12:57.000000000 +0000
@@ -1881,7 +1881,7 @@
 
   // possible locations of user manual directory
 #ifndef DEBIAN_PACKAGE
-  const QDir helpPath1("usermanual");
+  const QDir helpPath1("/usr/share/engauge-digitizer/usermanual");
   const QDir helpPath2("../usermanual");
   const QDir helpPath3("../../usermanual");
 #else /* DEBIAN_PACKAGE */
--- digit-4_0/src/helpwindow.cpp.orig	2006-10-08 07:41:49.000000000 +0100
+++ digit-4_0/src/helpwindow.cpp	2007-01-22 15:02:25.000000000 +0000
@@ -171,23 +171,24 @@
   QMap<int, QString>::Iterator it = mHistory.begin();
   for ( ; it != mHistory.end(); ++it )
     history.append( *it );
-
+#if 0
   QFile f( QDir::currentDirPath() + "/.history" );
   f.open( IO_WriteOnly );
   QDataStream s( &f );
   s << history;
   f.close();
-
+#endif
   bookmarks.clear();
   QMap<int, QString>::Iterator it2 = mBookmarks.begin();
   for ( ; it2 != mBookmarks.end(); ++it2 )
     bookmarks.append( *it2 );
-
+#if 0
   QFile f2( QDir::currentDirPath() + "/.bookmarks" );
   f2.open( IO_WriteOnly );
   QDataStream s2( &f2 );
   s2 << bookmarks;
   f2.close();
+#endif
 }
 
 void HelpWindow::setBackwardAvailable( bool b)
