--- espeak-1.17/src/Makefile.orig	2006-11-16 22:59:58.000000000 +0000
+++ espeak-1.17/src/Makefile	2006-11-30 12:54:23.000000000 +0000
@@ -47,7 +47,7 @@
 
 
 x_%.o: %.cpp
-	$(CXX) $(CXXFLAGS) -Wall -fpic -fvisibility=hidden -pedantic \
+	$(CXX) $(CXXFLAGS) -Wall -fpic -pedantic \
 	-I. -D LIBRARY -c -fno-exceptions $<  -o x_$*.o
 
 $(LIB_NAME): $(OBJS2)
--- espeak-1.18/src/Makefile.orig	2007-01-14 22:41:54.000000000 +0000
+++ espeak-1.18/src/Makefile	2007-01-14 22:41:17.000000000 +0000
@@ -36,7 +36,7 @@
 
 SRCS3 = espeak.cpp
 OBJS3=$(patsubst %.cpp,%.o,$(SRCS3))
-LIBS3=-lstdc++ -lespeak
+LIBS3=-lstdc++ -L . -lespeak
 
 CXXFLAGS=-O2
 
