--- espeak-1.22/src/Makefile.orig	2007-04-04 13:39:47.000000000 +0100
+++ espeak-1.22/src/Makefile	2007-04-19 22:43:26.000000000 +0100
@@ -59,7 +59,7 @@
 
 
 x_%.o: %.cpp
-	$(CXX) $(CXXFLAGS) -Wall -fpic -fvisibility=hidden -pedantic \
+	$(CXX) $(CXXFLAGS) -Wall -fpic -pedantic \
 	-I. -D LIBRARY -c -fno-exceptions $<  -o x_$*.o
 
 $(LIB_NAME): $(OBJS2)
@@ -75,7 +75,7 @@
 
 install: all
 	# Create directories
-	rm -r $(DESTDIR)$(DATADIR)
+	-rm -r $(DESTDIR)$(DATADIR)
 	$(MKDIR) $(DESTDIR)$(BINDIR)
 	$(MKDIR) $(DESTDIR)$(LIBDIR)
 	$(MKDIR) $(DESTDIR)$(INCDIR)
