diff --speed-large-files --minimal -Npru evolution-sharp-0.10.2.orig/configure evolution-sharp-0.10.2/configure
--- evolution-sharp-0.10.2.orig/configure	2005-09-12 16:56:48.000000000 -0400
+++ evolution-sharp-0.10.2/configure	2006-01-19 14:47:21.000000000 -0500
@@ -19780,6 +19780,108 @@ fi
   else
      PKG_CONFIG_MIN_VERSION=0.9.0
      if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
+        echo "$as_me:$LINENO: checking for evolution-data-server-1.2 >= 1.5.0" >&5
+echo $ECHO_N "checking for evolution-data-server-1.2 >= 1.5.0... $ECHO_C" >&6
+
+        if $PKG_CONFIG --exists "evolution-data-server-1.2 >= 1.5.0" ; then
+            echo "$as_me:$LINENO: result: yes" >&5
+echo "${ECHO_T}yes" >&6
+            succeeded=yes
+
+            echo "$as_me:$LINENO: checking BASE_DEPENDENCIES_CFLAGS" >&5
+echo $ECHO_N "checking BASE_DEPENDENCIES_CFLAGS... $ECHO_C" >&6
+            BASE_DEPENDENCIES_CFLAGS=`$PKG_CONFIG --cflags "evolution-data-server-1.2 >= 1.5.0"`
+            echo "$as_me:$LINENO: result: $BASE_DEPENDENCIES_CFLAGS" >&5
+echo "${ECHO_T}$BASE_DEPENDENCIES_CFLAGS" >&6
+
+            echo "$as_me:$LINENO: checking BASE_DEPENDENCIES_LIBS" >&5
+echo $ECHO_N "checking BASE_DEPENDENCIES_LIBS... $ECHO_C" >&6
+            BASE_DEPENDENCIES_LIBS=`$PKG_CONFIG --libs "evolution-data-server-1.2 >= 1.5.0"`
+            echo "$as_me:$LINENO: result: $BASE_DEPENDENCIES_LIBS" >&5
+echo "${ECHO_T}$BASE_DEPENDENCIES_LIBS" >&6
+        else
+            BASE_DEPENDENCIES_CFLAGS=""
+            BASE_DEPENDENCIES_LIBS=""
+            ## If we have a custom action on failure, don't print errors, but
+            ## do set a variable so people can do so.
+            BASE_DEPENDENCIES_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "evolution-data-server-1.2 >= 1.5.0"`
+
+        fi
+
+
+
+     else
+        echo "*** Your version of pkg-config is too old. You need version $PKG_CONFIG_MIN_VERSION or newer."
+        echo "*** See http://www.freedesktop.org/software/pkgconfig"
+     fi
+  fi
+
+  if test $succeeded = yes; then
+     eds15=yes
+  else
+     eds15=no
+  fi
+
+if test "x$eds15" = "xyes"; then
+	EDS_VERSION=-1.2
+	EDS_SOVERSION=7
+	EBOOK_SOVERSION=5
+	ECAL_SOVERSION=3
+else
+
+  succeeded=no
+
+  if test -z "$PKG_CONFIG"; then
+    # Extract the first word of "pkg-config", so it can be a program name with args.
+set dummy pkg-config; ac_word=$2
+echo "$as_me:$LINENO: checking for $ac_word" >&5
+echo $ECHO_N "checking for $ac_word... $ECHO_C" >&6
+if test "${ac_cv_path_PKG_CONFIG+set}" = set; then
+  echo $ECHO_N "(cached) $ECHO_C" >&6
+else
+  case $PKG_CONFIG in
+  [\\/]* | ?:[\\/]*)
+  ac_cv_path_PKG_CONFIG="$PKG_CONFIG" # Let the user override the test with a path.
+  ;;
+  *)
+  as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
+for as_dir in $PATH
+do
+  IFS=$as_save_IFS
+  test -z "$as_dir" && as_dir=.
+  for ac_exec_ext in '' $ac_executable_extensions; do
+  if $as_executable_p "$as_dir/$ac_word$ac_exec_ext"; then
+    ac_cv_path_PKG_CONFIG="$as_dir/$ac_word$ac_exec_ext"
+    echo "$as_me:$LINENO: found $as_dir/$ac_word$ac_exec_ext" >&5
+    break 2
+  fi
+done
+done
+
+  test -z "$ac_cv_path_PKG_CONFIG" && ac_cv_path_PKG_CONFIG="no"
+  ;;
+esac
+fi
+PKG_CONFIG=$ac_cv_path_PKG_CONFIG
+
+if test -n "$PKG_CONFIG"; then
+  echo "$as_me:$LINENO: result: $PKG_CONFIG" >&5
+echo "${ECHO_T}$PKG_CONFIG" >&6
+else
+  echo "$as_me:$LINENO: result: no" >&5
+echo "${ECHO_T}no" >&6
+fi
+
+  fi
+
+  if test "$PKG_CONFIG" = "no" ; then
+     echo "*** The pkg-config script could not be found. Make sure it is"
+     echo "*** in your path, or set the PKG_CONFIG environment variable"
+     echo "*** to the full path to pkg-config."
+     echo "*** Or see http://www.freedesktop.org/software/pkgconfig to get pkg-config."
+  else
+     PKG_CONFIG_MIN_VERSION=0.9.0
+     if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
         echo "$as_me:$LINENO: checking for evolution-data-server-1.2 >= 1.3.5" >&5
 echo $ECHO_N "checking for evolution-data-server-1.2 >= 1.3.5... $ECHO_C" >&6
 
@@ -19822,12 +19924,12 @@ echo "${ECHO_T}$BASE_DEPENDENCIES_LIBS" 
      eds13=no
   fi
 
-if test "x$eds13" = "xyes"; then
-	EDS_VERSION=-1.2
-	EDS_SOVERSION=4
-	EBOOK_SOVERSION=5
-	ECAL_SOVERSION=3
-else
+	if test "x$eds13" = "xyes"; then
+		EDS_VERSION=-1.2
+		EDS_SOVERSION=4
+		EBOOK_SOVERSION=5
+		ECAL_SOVERSION=3
+	else
 
   succeeded=no
 
@@ -19925,15 +20027,16 @@ echo "${ECHO_T}$BASE_DEPENDENCIES_LIBS" 
   fi
 
 
-	if test "x$eds12" = "xyes"; then
-		EDS_VERSION=-1.2
-		EDS_SOVERSION=4
-		EBOOK_SOVERSION=3
-		ECAL_SOVERSION=2
-	else
-		{ { echo "$as_me:$LINENO: error: You need evolution-data-server 1.2.x or 1.3.5 or newer" >&5
+		if test "x$eds12" = "xyes"; then
+			EDS_VERSION=-1.2
+			EDS_SOVERSION=4
+			EBOOK_SOVERSION=3
+			ECAL_SOVERSION=2
+		else
+			{ { echo "$as_me:$LINENO: error: You need evolution-data-server 1.2.x or 1.3.5 or newer" >&5
 echo "$as_me: error: You need evolution-data-server 1.2.x or 1.3.5 or newer" >&2;}
    { (exit 1); exit 1; }; }
+		fi
 	fi
 fi
 
@@ -21344,23 +21447,23 @@ fi
   else
      PKG_CONFIG_MIN_VERSION=0.9.0
      if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
-        echo "$as_me:$LINENO: checking for \"evolution-shell-2.4\"" >&5
-echo $ECHO_N "checking for \"evolution-shell-2.4\"... $ECHO_C" >&6
+        echo "$as_me:$LINENO: checking for \"evolution-shell-2.6\"" >&5
+echo $ECHO_N "checking for \"evolution-shell-2.6\"... $ECHO_C" >&6
 
-        if $PKG_CONFIG --exists ""evolution-shell-2.4"" ; then
+        if $PKG_CONFIG --exists ""evolution-shell-2.6"" ; then
             echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6
             succeeded=yes
 
             echo "$as_me:$LINENO: checking EVO_SHELL_CFLAGS" >&5
 echo $ECHO_N "checking EVO_SHELL_CFLAGS... $ECHO_C" >&6
-            EVO_SHELL_CFLAGS=`$PKG_CONFIG --cflags ""evolution-shell-2.4""`
+            EVO_SHELL_CFLAGS=`$PKG_CONFIG --cflags ""evolution-shell-2.6""`
             echo "$as_me:$LINENO: result: $EVO_SHELL_CFLAGS" >&5
 echo "${ECHO_T}$EVO_SHELL_CFLAGS" >&6
 
             echo "$as_me:$LINENO: checking EVO_SHELL_LIBS" >&5
 echo $ECHO_N "checking EVO_SHELL_LIBS... $ECHO_C" >&6
-            EVO_SHELL_LIBS=`$PKG_CONFIG --libs ""evolution-shell-2.4""`
+            EVO_SHELL_LIBS=`$PKG_CONFIG --libs ""evolution-shell-2.6""`
             echo "$as_me:$LINENO: result: $EVO_SHELL_LIBS" >&5
 echo "${ECHO_T}$EVO_SHELL_LIBS" >&6
         else
@@ -21368,7 +21471,7 @@ echo "${ECHO_T}$EVO_SHELL_LIBS" >&6
             EVO_SHELL_LIBS=""
             ## If we have a custom action on failure, don't print errors, but
             ## do set a variable so people can do so.
-            EVO_SHELL_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors ""evolution-shell-2.4""`
+            EVO_SHELL_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors ""evolution-shell-2.6""`
 
         fi
 
diff --speed-large-files --minimal -Npru evolution-sharp-0.10.2.orig/configure.in evolution-sharp-0.10.2/configure.in
--- evolution-sharp-0.10.2.orig/configure.in	2005-09-12 16:55:58.000000000 -0400
+++ evolution-sharp-0.10.2/configure.in	2006-01-19 14:40:53.000000000 -0500
@@ -12,22 +12,30 @@ if test "x$PKG_CONFIG" = "xno"; then
 	AC_MSG_ERROR([You need to install pkg-config])
 fi
 
-PKG_CHECK_MODULES(BASE_DEPENDENCIES, evolution-data-server-1.2 >= 1.3.5, eds13=yes, eds13=no)
-if test "x$eds13" = "xyes"; then
+PKG_CHECK_MODULES(BASE_DEPENDENCIES, evolution-data-server-1.2 >= 1.5.0, eds15=yes, eds15=no)
+if test "x$eds15" = "xyes"; then
 	EDS_VERSION=-1.2
-	EDS_SOVERSION=4
+	EDS_SOVERSION=7
 	EBOOK_SOVERSION=5
 	ECAL_SOVERSION=3
 else
-	PKG_CHECK_MODULES(BASE_DEPENDENCIES, evolution-data-server-1.2 >= 1.2.0 evolution-data-server-1.2 < 1.3.0, eds12=yes, eds12=no)
-
-	if test "x$eds12" = "xyes"; then
+	PKG_CHECK_MODULES(BASE_DEPENDENCIES, evolution-data-server-1.2 >= 1.3.5, eds13=yes, eds13=no)
+	if test "x$eds13" = "xyes"; then
 		EDS_VERSION=-1.2
 		EDS_SOVERSION=4
-		EBOOK_SOVERSION=3
-		ECAL_SOVERSION=2
+		EBOOK_SOVERSION=5
+		ECAL_SOVERSION=3
 	else
-		AC_MSG_ERROR([You need evolution-data-server 1.2.x or 1.3.5 or newer])
+		PKG_CHECK_MODULES(BASE_DEPENDENCIES, evolution-data-server-1.2 >= 1.2.0 evolution-data-server-1.2 < 1.3.0, eds12=yes, eds12=no)
+
+		if test "x$eds12" = "xyes"; then
+			EDS_VERSION=-1.2
+			EDS_SOVERSION=4
+			EBOOK_SOVERSION=3
+			ECAL_SOVERSION=2
+		else
+			AC_MSG_ERROR([You need evolution-data-server 1.2.x or 1.3.5 or newer])
+		fi
 	fi
 fi
 
@@ -107,7 +115,7 @@ IDL_INCLUDES="-I "`pkg-config --variable
 AC_MSG_RESULT($IDL_INCLUDES)
 AC_SUBST(IDL_INCLUDES)
 
-PKG_CHECK_MODULES(EVO_SHELL, "evolution-shell-2.4",, [PKG_CHECK_MODULES(EVO_SHELL, "evolution-shell-2.2")])
+PKG_CHECK_MODULES(EVO_SHELL, "evolution-shell-2.6",, [PKG_CHECK_MODULES(EVO_SHELL, "evolution-shell-2.2")])
 AC_SUBST(EVO_SHELL_CFLAGS)
 AC_SUBST(EVO_SHELL_LIBS)
 
