--- f-spot-0.4.2/extensions/GalleryExport/Makefile.in.orig	2008-02-14 22:27:19.000000000 +0000
+++ f-spot-0.4.2/extensions/GalleryExport/Makefile.in	2008-02-15 21:20:28.000000000 +0000
@@ -538,7 +538,7 @@
 	$(CSC_LIB) -out:$@ $(PLUGIN_SOURCES) $(REFS) $(PKGS) $(ASSEMBLIES) $(RESOURCES)
 
 install-data-hook:
-	rm -f $(plugindir)/GalleryExport.addin.xml
+	-rm -f $(plugindir)/GalleryExport.addin.xml
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
