--- farstream-0.2.9/common-modified/gst-glib-gen.mak.orig	2021-01-26 20:26:22.147058249 +0000
+++ farstream-0.2.9/common-modified/gst-glib-gen.mak	2021-01-26 20:25:59.379059336 +0000
@@ -6,7 +6,7 @@
 #glib_gen_prefix=gst_color_balance
 #glib_gen_basename=colorbalance
 
-enum_headers=$(foreach h,$(glib_enum_headers),\n\#include \"$(h)\")
+enum_headers=$(foreach h,$(glib_enum_headers),\n#include \"$(h)\")
 
 # these are all the rules generating the relevant files
 $(glib_gen_basename)-enumtypes.h: $(glib_enum_headers)
