--- fec-3.0.1/makefile.in.orig	2008-03-28 17:04:57.000000000 +0000
+++ fec-3.0.1/makefile.in	2008-03-28 19:04:24.000000000 +0000
@@ -41,14 +41,14 @@
 	./vtest615
 
 install: all
-	mkdir -p @libdir@ 
-	install -m 644 -p $(SHARED_LIB) libfec.a @libdir@
-#	(cd @libdir@;ln -f -s $(SHARED_LIB) libfec.so)
-	@REBIND@
-	mkdir -p @includedir@
-	install -m 644 -p fec.h @includedir@
-	mkdir -m 0755 -p @mandir@/man3
-	install -m 644 -p simd-viterbi.3 rs.3 dsp.3 @mandir@/man3
+	mkdir -p $(DESTDIR)@libdir@ 
+	install -m 644 -p $(SHARED_LIB) libfec.a $(DESTDIR)@libdir@
+#	(cd $(DESTDIR)@libdir@;ln -f -s $(SHARED_LIB) libfec.so)
+	-@REBIND@
+	mkdir -p $(DESTDIR)@includedir@
+	install -m 644 -p fec.h $(DESTDIR)@includedir@
+	mkdir -m 0755 -p $(DESTDIR)@mandir@/man3
+	install -m 644 -p simd-viterbi.3 rs.3 dsp.3 $(DESTDIR)@mandir@/man3
 
 peaktest: peaktest.o libfec.a
 	gcc -g -o $@ $^
@@ -100,7 +100,7 @@
 
 # for Linux et al
 libfec.so: $(LIBS)
-	gcc -shared -Xlinker -soname=$@ -o $@ -Wl,-whole-archive $^ -Wl,-no-whole-archive -lc
+	gcc -shared -Xlinker -soname=$@ -o $@ -Wl,-whole-archive $^ -Wl,-no-whole-archive -lc -lm
 
 dotprod.o: dotprod.c fec.h
 
