diff -ur festival.orig/speech_tools/utils/Makefile festival/speech_tools/utils/Makefile
--- festival.orig/speech_tools/utils/Makefile	2004-05-29 14:33:14.000000000 -0400
+++ festival/speech_tools/utils/Makefile	2005-02-06 18:46:06.332800806 -0500
@@ -56,5 +56,5 @@
 include $(TOP)/config/common_make_rules
 
 EST_cutils.o: EST_cutils.c
-	$(CC_COMMAND) -DESTNAME='$(PROJECT_NAME)' -DESTDATE='$(PROJECT_DATE)' -DESTVERSION='$(PROJECT_VERSION)' -DESTLIBDIRC='$(EST_HOME)/lib' -DESTSTATE='$(PROJECT_STATE)' -DESTOSTYPE='$(SYSTEM_TYPE)' EST_cutils.c
+	$(CC_COMMAND) -DESTNAME='$(PROJECT_NAME)' -DESTDATE='$(PROJECT_DATE)' -DESTVERSION='$(PROJECT_VERSION)' -DESTLIBDIRC='/usr/share/festival' -DESTSTATE='$(PROJECT_STATE)' -DESTOSTYPE='$(SYSTEM_TYPE)' EST_cutils.c
 
diff -ur festival.orig/src/arch/festival/festival.cc festival/src/arch/festival/festival.cc
--- festival.orig/src/arch/festival/festival.cc	2004-05-24 16:14:33.000000000 -0400
+++ festival/src/arch/festival/festival.cc	2005-02-06 18:50:19.968101317 -0500
@@ -68,7 +68,7 @@
 #endif
 
 #ifndef FTLIBDIR
-#define FTLIBDIR "/projects/festival/lib/"
+#define FTLIBDIR "/usr/share/festival/"
 #endif
 #ifndef FTOSTYPE
 #define FTOSTYPE ""
diff -ur festival.orig/src/arch/festival/Makefile festival/src/arch/festival/Makefile
--- festival.orig/src/arch/festival/Makefile	2004-01-27 18:32:17.000000000 -0500
+++ festival/src/arch/festival/Makefile	2005-02-06 18:52:15.487123761 -0500
@@ -62,6 +62,6 @@
 LOCAL_INCLUDES += $(FESTIVAL_INCLUDES)
 
 festival.o: festival.cc
-	$(CXX_COMMAND_TEMPLATES) -DFTNAME='$(PROJECT_NAME)' -DFTLIBDIRC='$(FTLIBDIR)' -DFTVERSION='$(PROJECT_VERSION)' -DFTSTATE='$(PROJECT_STATE)'  -DFTDATE='$(PROJECT_DATE)' -DFTOSTYPE=\"$(SYSTEM_TYPE)\" festival.cc
+	$(CXX_COMMAND_TEMPLATES) -DFTNAME='$(PROJECT_NAME)' -DFTLIBDIRC='/usr/share/festival' -DFTVERSION='$(PROJECT_VERSION)' -DFTSTATE='$(PROJECT_STATE)'  -DFTDATE='$(PROJECT_DATE)' -DFTOSTYPE=\"$(SYSTEM_TYPE)\" festival.cc
 
 
