--- festival/speech_tools/config/compilers/gcc_defaults.mak.soname	2005-02-06 22:31:03.285679647 -0500
+++ festival/speech_tools/config/compilers/gcc_defaults.mak	2005-02-06 22:33:06.008528154 -0500
@@ -78,7 +78,7 @@
 SHARED_LINKFLAGS = -fno-shared-data
 
 ifndef GCC_MAKE_SHARED_LIB
-    MAKE_SHARED_LIB = $(CXX) -shared -fno-shared-data -o XXX
+    MAKE_SHARED_LIB = $(CXX) -shared -fno-shared-data -o XXX -Wl,-soname,XXX
 else
     MAKE_SHARED_LIB = $(GCC_MAKE_SHARED_LIB)
 endif
