--- ffmpeg-3.1.2/libavcodec/libopenjpegdec.c.orig	2016-06-27 00:54:29.000000000 +0100
+++ ffmpeg-3.1.2/libavcodec/libopenjpegdec.c	2016-08-10 21:16:40.631629404 +0100
@@ -24,8 +24,6 @@
  * JPEG 2000 decoder using libopenjpeg
  */
 
-#define  OPJ_STATIC
-
 #include "libavutil/common.h"
 #include "libavutil/imgutils.h"
 #include "libavutil/intreadwrite.h"
--- ffmpeg-3.1.2/libavcodec/libopenjpegenc.c.orig	2016-06-27 00:54:29.000000000 +0100
+++ ffmpeg-3.1.2/libavcodec/libopenjpegenc.c	2016-08-10 21:16:46.642629426 +0100
@@ -24,8 +24,6 @@
  * JPEG 2000 encoder using libopenjpeg
  */
 
-#define  OPJ_STATIC
-
 #include "libavutil/avassert.h"
 #include "libavutil/common.h"
 #include "libavutil/imgutils.h"
--- ffmpeg-3.1.2/configure.orig	2016-08-08 23:53:26.000000000 +0100
+++ ffmpeg-3.1.2/configure	2016-08-10 21:16:17.747629319 +0100
@@ -5669,10 +5669,10 @@
                                  require opencv opencv2/core/core_c.h cvCreateImageHeader -lopencv_core -lopencv_imgproc; } ||
                                require_pkg_config opencv opencv/cxcore.h cvCreateImageHeader; }
 enabled libopenh264       && require_pkg_config openh264 wels/codec_api.h WelsGetCodecVersion
-enabled libopenjpeg       && { check_lib openjpeg-2.1/openjpeg.h opj_version -lopenjp2 -DOPJ_STATIC ||
-                               check_lib openjpeg-2.0/openjpeg.h opj_version -lopenjp2 -DOPJ_STATIC ||
-                               check_lib openjpeg-1.5/openjpeg.h opj_version -lopenjpeg -DOPJ_STATIC ||
-                               check_lib openjpeg.h opj_version -lopenjpeg -DOPJ_STATIC ||
+enabled libopenjpeg       && { check_lib openjpeg-2.1/openjpeg.h opj_version -lopenjp2 ||
+                               check_lib openjpeg-2.0/openjpeg.h opj_version -lopenjp2 ||
+                               check_lib openjpeg-1.5/openjpeg.h opj_version -lopenjpeg ||
+                               check_lib openjpeg.h opj_version -lopenjpeg ||
                                die "ERROR: libopenjpeg not found"; }
 enabled libopus           && require_pkg_config opus opus_multistream.h opus_multistream_decoder_create
 enabled libpulse          && require_pkg_config libpulse pulse/pulseaudio.h pa_context_new
