--- ffmpeg-20100728/libavformat/rmdec.c.orig	2010-07-28 20:31:04.000000000 +0100
+++ ffmpeg-20100728/libavformat/rmdec.c	2010-08-01 07:35:15.000000000 +0100
@@ -305,7 +305,7 @@
             return ret;
 
 //        av_log(s, AV_LOG_DEBUG, "fps= %d fps2= %d\n", fps, fps2);
-        st->codec->time_base.den = fps * st->codec->time_base.num;
+        st->codec->time_base.den = (fps2 > 0 ? fps2 : fps) * st->codec->time_base.num;
         //XXX: do we really need that?
         switch(st->codec->extradata[4]>>4){
         case 1: st->codec->codec_id = CODEC_ID_RV10; break;
