--- ffmpeg-20070220/libavcodec/imgconvert.c.orig	2007-02-20 18:26:14.000000000 +0000
+++ ffmpeg-20070220/libavcodec/imgconvert.c	2007-02-20 19:07:46.000000000 +0000
@@ -2336,7 +2336,6 @@
     return 0;
 }
 
-#ifndef CONFIG_SWSCALER
 /* XXX: always use linesize. Return -1 if not supported */
 int img_convert(AVPicture *dst, int dst_pix_fmt,
                 const AVPicture *src, int src_pix_fmt,
@@ -2578,7 +2577,6 @@
     avpicture_free(tmp);
     return ret;
 }
-#endif
 
 /* NOTE: we scan all the pixels to have an exact information */
 static int get_alpha_info_pal8(const AVPicture *src, int width, int height)
