--- fido-0.3.00/src/libs/librmrtsp/rtsp/rtsp.c.orig	2006-11-29 06:17:25.000000000 +0000
+++ fido-0.3.00/src/libs/librmrtsp/rtsp/rtsp.c	2007-10-01 11:12:41.000000000 +0100
@@ -437,8 +437,8 @@
 		if (i < 4) 
 			return i;
 
-		while ((buffer[0]=='S') && (buffer[1]=='E') 
-		        &&(buffer[2]=='T') && (buffer[3]=='_'))
+		while (((buffer[0]=='S') && (buffer[1]=='E') &&(buffer[2]=='T') && (buffer[3]=='_')) ||
+		       ((buffer[0]=='O') && (buffer[1]=='P') &&(buffer[2]=='T') && (buffer[3]=='I')))
 		{
 			UT_LOG(MODNAME, "Got ping from server; ponging.");
 
