--- fido-0.3.00/src/libs/librmrtsp/real/real.c.orig	2006-11-29 06:03:21.000000000 +0000
+++ fido-0.3.00/src/libs/librmrtsp/real/real.c	2007-09-19 11:25:57.000000000 +0100
@@ -700,7 +700,7 @@
 	                    "GUID: 00000000-0000-0000-0000-000000000000");
 	rtsp_schedule_field(rtsp_session, "RegionData: 0");
 	rtsp_schedule_field(rtsp_session,
-	                    "ClientID: Linux_2.4_6.0.9.1235_play32_RN01_EN_586");
+	                    "ClientID: Linux_2.6_10.0.0.0_play32_RN01_EN_586");
 	rtsp_schedule_field(rtsp_session, "SupportsMaximumASMBandwidth: 1");
 	rtsp_schedule_field(rtsp_session, "Language: en-US");
 	rtsp_schedule_field(rtsp_session, 
--- fido-0.3.00/src/libs/librmrtsp/rtsp/rtsp.c.orig	2006-11-29 06:17:25.000000000 +0000
+++ fido-0.3.00/src/libs/librmrtsp/rtsp/rtsp.c	2007-10-01 11:12:41.000000000 +0100
@@ -508,8 +508,7 @@
 	if (user_agent)
 		s->user_agent = strdup(user_agent);
 	else
-		s->user_agent = strdup("User-Agent: RealMedia Player Version 6.0.9.1235"
-		                       " (linux-2.0-libc6-i386-gcc2.95)");
+		s->user_agent = strdup("User-Agent: RealMedia Player (HelixDNAClient)/10.0.0.0 (linux-2.2-libc6-gcc32-i586)");
 
 	slash = strchr(mrl, '/');
 	colon = strchr(mrl, ':');
@@ -556,12 +555,12 @@
 	/* now lets send an options request. */
 	rtsp_schedule_field(s, "CSeq: 1");
 	rtsp_schedule_field(s, s->user_agent);
-	rtsp_schedule_field(s, "ClientChallenge: 9e26d33f2984236010ef6253fb1887f7");
-	rtsp_schedule_field(s, "PlayerStarttime: [28/03/2003:22:50:23 00:00]");
-	rtsp_schedule_field(s, "CompanyID: KnKV4M4I/B2FjJ1TToLycw==");
+	rtsp_schedule_field(s, "ClientChallenge: ffb8cbb13436aba3c22c1c4e9d6680bc");
+	rtsp_schedule_field(s, "PlayerStarttime: [19/09/2007:09:47:56 00:00]");
+	rtsp_schedule_field(s, "CompanyID: jOCynekbYsJ2LQGJm+BPvA==");
 	rtsp_schedule_field(s, "GUID: 00000000-0000-0000-0000-000000000000");
 	rtsp_schedule_field(s, "RegionData: 0");
-	rtsp_schedule_field(s, "ClientID: Linux_2.4_6.0.9.1235_play32_RN01_EN_586");
+	rtsp_schedule_field(s, "ClientID: Linux_2.6_10.0.0.0_play32_RN01_EN_586");
 	/*rtsp_schedule_field(s, "Pragma: initiate-session");*/
 	rtsp_request_options(s, NULL);
 
