--- fido-0.3.00/src/db/db_XMLable.h.orig	2006-10-28 07:50:51.000000000 +0100
+++ fido-0.3.00/src/db/db_XMLable.h	2007-09-19 10:08:26.000000000 +0100
@@ -124,7 +124,7 @@
 			static const Element & lookupChildNode(const Element & parent,
 			                                       const char * xpath);
 
-			static bool XMLable::lookupChildren(
+			static bool lookupChildren(
 			                        const Node &    parent,
 			                        const NameVec & name_list,
 			                        ElementPtrVec & e_list,
--- fido-0.3.00/src/db/db_DatedStr.h.orig	2006-10-20 03:22:45.000000000 +0100
+++ fido-0.3.00/src/db/db_DatedStr.h	2007-09-19 10:10:21.000000000 +0100
@@ -101,7 +101,7 @@
 
 			/* Ctors to convert from STL and C strings. */
 			DatedStr(const Glib::ustring & str);
-			DatedStr::DatedStr(const char * str);
+			DatedStr(const char * str);
 
 			void clear(void);
 			void appendField(FieldType type, 
@@ -198,7 +198,7 @@
 			inline void assert_offset_valid(void) const;
 
 			/* Make printf format representing pad and width. */
-			static Glib::ustring DatedStr::make_fmt(size_type width,
+			static Glib::ustring make_fmt(size_type width,
 			                                        Pad       pad,
 			                                        size_type def_width,
 			                                        bool      def_zero_pad);
--- fido-0.3.00/src/io/test/echo_Client.h.orig	2006-10-28 08:19:08.000000000 +0100
+++ fido-0.3.00/src/io/test/echo_Client.h	2007-09-19 10:45:05.000000000 +0100
@@ -58,7 +58,7 @@
 		/*
 		 * Private functions.
 		 */
-		Glib::ustring EchoClient::getLine(bool & _eof);
+		Glib::ustring getLine(bool & _eof);
 
 		/* Try to send a message to the server; terminate the 
 		 * wait loop if the connection has died.
--- fido-0.3.00/src/be/be_Executer.h.orig	2006-12-18 03:49:31.000000000 +0000
+++ fido-0.3.00/src/be/be_Executer.h	2007-09-19 10:12:23.000000000 +0100
@@ -154,7 +154,7 @@
 			 *       some problem with the path.
 			 */
 			Glib::ustring 
-			Executer::gen_unique_orig_enc_fname(const Glib::ustring & pathfile);
+			gen_unique_orig_enc_fname(const Glib::ustring & pathfile);
 
 			/* The slave thread used to process the batch. */
 			Glib::Thread * slave_ptr;
--- fido-0.3.00/src/be/be_Agent.h.orig	2006-12-19 08:33:09.000000000 +0000
+++ fido-0.3.00/src/be/be_Agent.h	2007-09-19 10:12:45.000000000 +0100
@@ -151,14 +151,14 @@
 			db::BatchList * create_new_batch_list(const User & _user);
 
 			/* Calculate when the next event occurs from the BatchQueue. */
-			time_t Agent::calc_expiry_time(void);
+			time_t calc_expiry_time(void);
 
 			/* Try to find the batch list for the user 
 			 * (returns NULL if not found). 
 			 */
 			db::BatchList * find_batch_list(const User & _user);
 
-			void Agent::kill_connection(Connection & _conn);
+			void kill_connection(Connection & _conn);
 
 			/*
 			 * Signal handlers.
--- fido-0.3.00/src/pi/pi_Factory.h.orig	2006-10-29 02:04:38.000000000 +0000
+++ fido-0.3.00/src/pi/pi_Factory.h	2007-09-19 10:11:52.000000000 +0100
@@ -119,7 +119,7 @@
 		protected:
 			typedef FactoryConstIterator::PluginMap   PluginMap; 
 
-			void Factory::add_plugin_to_map(Plugin * plugin_ptr);
+			void add_plugin_to_map(Plugin * plugin_ptr);
 
 			/* 
 			 * A map of PluginInfo objects which allows rapid look up of a
--- fido-0.3.00/src/tr/tr_Transport.h.orig	2006-12-18 03:59:46.000000000 +0000
+++ fido-0.3.00/src/tr/tr_Transport.h	2007-09-19 10:11:31.000000000 +0100
@@ -120,7 +120,7 @@
 			virtual bool config(const ConfigMap & _cmap) = 0;
 
 			virtual void regProgressCB(progressCB * _cb, void * data);
-			virtual void Transport::unregProgressCB(void);
+			virtual void unregProgressCB(void);
 
 			/* Return human readable equivalent of error. */
 			static const char * strerror(Err err);
--- fido-0.3.00/src/si/test/SigRec.h.orig	2006-10-12 11:16:25.000000000 +0100
+++ fido-0.3.00/src/si/test/SigRec.h	2007-09-19 10:44:05.000000000 +0100
@@ -54,7 +54,7 @@
 			si::UserData *     udata;
 		};
 
-		SigRec::SigRec(void)
+		SigRec(void)
 			: num_conn_sigs(0), num_fatal_sigs(0) {};
 
 		/* Connect to a given instance of si::ServerInt. */
--- fido-0.3.00/src/clients/cli/CLIClient.h.orig	2006-12-23 03:14:00.000000000 +0000
+++ fido-0.3.00/src/clients/cli/CLIClient.h	2007-09-19 10:28:46.000000000 +0100
@@ -188,8 +188,8 @@
 		AddCmd __add_cmd;
 		RemoveCmd __remove_cmd;
 
-		static const char CLIClient::logfile[];
-		static const char CLIClient::prompt[];
-		static const char CLIClient::preamble[];
-		static const char CLIClient::help_cmd[];
+		static const char logfile[];
+		static const char prompt[];
+		static const char preamble[];
+		static const char help_cmd[];
 };
