--- Field3D-1.7.2/CMakeLists.txt.orig	2016-06-19 05:56:00.000000000 +0100
+++ Field3D-1.7.2/CMakeLists.txt	2019-04-21 08:35:23.196032920 +0100
@@ -201,7 +201,7 @@
   test/unit_tests/UnitTest.cpp
   )
 
-TARGET_LINK_LIBRARIES ( unitTest ${Field3D_BIN_Libraries} )
+TARGET_LINK_LIBRARIES ( unitTest ${Field3D_BIN_Libraries} Boost::thread )
 
 IF ( CMAKE_HOST_WIN32 )
   SET_TARGET_PROPERTIES( unitTest PROPERTIES COMPILE_FLAGS -bigobj )
@@ -212,35 +212,35 @@
   apps/f3dinfo/main.cpp
   )
 
-TARGET_LINK_LIBRARIES ( f3dinfo ${Field3D_BIN_Libraries} )
+TARGET_LINK_LIBRARIES ( f3dinfo ${Field3D_BIN_Libraries} Boost::regex Boost::program_options Boost::thread )
 
 # field3d - sparse_field_io
 ADD_EXECUTABLE ( sparse_field_io
   apps/sample_code/sparse_field_io/main.cpp
   )
 
-TARGET_LINK_LIBRARIES ( sparse_field_io ${Field3D_BIN_Libraries} )
+TARGET_LINK_LIBRARIES ( sparse_field_io ${Field3D_BIN_Libraries} Boost::thread )
 
 # field3d - read
 ADD_EXECUTABLE ( read
   apps/sample_code/read/main.cpp
   )
 
-TARGET_LINK_LIBRARIES ( read ${Field3D_BIN_Libraries} )
+TARGET_LINK_LIBRARIES ( read ${Field3D_BIN_Libraries} Boost::thread )
 
 # field3d - mixed_types
 ADD_EXECUTABLE ( mixed_types
   apps/sample_code/mixed_types/main.cpp
   )
 
-TARGET_LINK_LIBRARIES ( mixed_types ${Field3D_BIN_Libraries} )
+TARGET_LINK_LIBRARIES ( mixed_types ${Field3D_BIN_Libraries} Boost::thread )
 
 # field3d - create_and_write
 ADD_EXECUTABLE ( create_and_write
   apps/sample_code/create_and_write/main.cpp
   )
 
-TARGET_LINK_LIBRARIES ( create_and_write ${Field3D_BIN_Libraries} )
+TARGET_LINK_LIBRARIES ( create_and_write ${Field3D_BIN_Libraries} Boost::thread )
 
 IF (DOXYGEN_FOUND)
   ADD_CUSTOM_TARGET ( doc
