--- Field3D-1.7.3/CMakeLists.txt.orig	2022-05-04 10:59:34.917942811 +0100
+++ Field3D-1.7.3/CMakeLists.txt	2022-05-04 11:11:36.864908365 +0100
@@ -147,7 +147,7 @@
              ${MPI_LIBRARIES} )
   ENDIF ( MPI_FOUND )
   LIST ( APPEND Field3D_Libraries_Shared
-    Iex Half IlmThread Imath
+    Iex IlmThread Imath
     pthread dl z )
   SET ( Field3D_DSO_Libraries ${Field3D_Libraries_Shared} )
   SET ( Field3D_BIN_Libraries Field3D ${Field3D_Libraries_Shared}
@@ -155,7 +155,7 @@
 ENDIF ( )
 IF ( CMAKE_HOST_WIN32 )
   # Add OpenEXR and zlib release/debug
-  FOREACH ( lib Iex Half IlmThread Imath zdll )
+  FOREACH ( lib Iex IlmThread Imath zdll )
     LIST ( APPEND Field3D_Libraries_Shared
       optimized ${lib}
       debug ${lib}_d )
--- Field3D-1.7.3/test/unit_tests/UnitTest.cpp.orig	2020-03-11 15:54:36.000000000 +0000
+++ Field3D-1.7.3/test/unit_tests/UnitTest.cpp	2022-05-04 11:12:02.218907155 +0100
@@ -44,7 +44,7 @@
 #include <boost/thread/thread.hpp>
 #include <boost/thread/mutex.hpp>
 
-#include <OpenEXR/ImathFrustum.h>
+#include <Imath/ImathFrustum.h>
 
 #include "Field3D/DenseField.h"
 #include "Field3D/EmptyField.h"
--- Field3D-1.7.3/export/StdMathLib.h.orig	2020-03-11 15:54:36.000000000 +0000
+++ Field3D-1.7.3/export/StdMathLib.h	2022-05-04 11:02:57.586933141 +0100
@@ -38,17 +38,16 @@
 #ifndef _INCLUDED_Field3D_StdMathLib_H_
 #define _INCLUDED_Field3D_StdMathLib_H_
 
-#include <OpenEXR/ImathBox.h> 
-#include <OpenEXR/ImathBoxAlgo.h>
-#include <OpenEXR/ImathColor.h>
-#include <OpenEXR/ImathHalfLimits.h>
-#include <OpenEXR/ImathMatrix.h>
-#include <OpenEXR/ImathMatrixAlgo.h>
-#include <OpenEXR/ImathPlane.h>
-#include <OpenEXR/ImathRandom.h> 
-#include <OpenEXR/ImathRoots.h>
-#include <OpenEXR/ImathVec.h>
-#include <OpenEXR/half.h> 
+#include <Imath/ImathBox.h> 
+#include <Imath/ImathBoxAlgo.h>
+#include <Imath/ImathColor.h>
+#include <Imath/ImathMatrix.h>
+#include <Imath/ImathMatrixAlgo.h>
+#include <Imath/ImathPlane.h>
+#include <Imath/ImathRandom.h> 
+#include <Imath/ImathRoots.h>
+#include <Imath/ImathVec.h>
+#include <Imath/half.h> 
 
 //----------------------------------------------------------------------------//
 
--- Field3D-1.7.3/export/Curve.h.orig	2020-03-11 15:54:36.000000000 +0000
+++ Field3D-1.7.3/export/Curve.h	2022-05-04 11:04:38.110928345 +0100
@@ -53,8 +53,8 @@
 
 #include <boost/lexical_cast.hpp>
 
-#include <OpenEXR/ImathFun.h>
-#include <OpenEXR/ImathMatrix.h>
+#include <Imath/halfFunction.h>
+#include <Imath/ImathMatrix.h>
 
 //----------------------------------------------------------------------------//
 
--- Field3D-1.7.3/include/OgUtil.h.orig	2020-03-11 15:54:36.000000000 +0000
+++ Field3D-1.7.3/include/OgUtil.h	2022-05-04 11:05:12.257926715 +0100
@@ -10,7 +10,7 @@
 #include <iostream>
 #include <string>
 
-#include <OpenEXR/ImathVec.h>
+#include <Imath/ImathVec.h>
 
 #include "All.h"
 #include "UtilFoundation.h"
--- Field3D-1.7.3/include/OgIAttribute.h.orig	2020-03-11 15:54:36.000000000 +0000
+++ Field3D-1.7.3/include/OgIAttribute.h	2022-05-04 11:07:16.996920764 +0100
@@ -9,7 +9,7 @@
 
 #include "OgUtil.h"
 
-#include <OpenEXR/ImathMatrix.h>
+#include <Imath/ImathMatrix.h>
 
 //----------------------------------------------------------------------------//
 
--- Field3D-1.7.3/include/UtilFoundation.h.orig	2020-03-11 15:54:36.000000000 +0000
+++ Field3D-1.7.3/include/UtilFoundation.h	2022-05-04 11:06:55.135921807 +0100
@@ -68,7 +68,7 @@
 
 #include <memory>
 
-#include <half.h>
+#include <Imath/half.h>
 
 #include <iomanip>
 #include <iostream>
--- Field3D-1.7.3/Field3D/StdMathLib.h.orig	2020-03-11 15:54:36.000000000 +0000
+++ Field3D-1.7.3/Field3D/StdMathLib.h	2022-05-04 11:02:57.000000000 +0100
@@ -38,17 +38,16 @@
 #ifndef _INCLUDED_Field3D_StdMathLib_H_
 #define _INCLUDED_Field3D_StdMathLib_H_
 
-#include <OpenEXR/ImathBox.h> 
-#include <OpenEXR/ImathBoxAlgo.h>
-#include <OpenEXR/ImathColor.h>
-#include <OpenEXR/ImathHalfLimits.h>
-#include <OpenEXR/ImathMatrix.h>
-#include <OpenEXR/ImathMatrixAlgo.h>
-#include <OpenEXR/ImathPlane.h>
-#include <OpenEXR/ImathRandom.h> 
-#include <OpenEXR/ImathRoots.h>
-#include <OpenEXR/ImathVec.h>
-#include <OpenEXR/half.h> 
+#include <Imath/ImathBox.h> 
+#include <Imath/ImathBoxAlgo.h>
+#include <Imath/ImathColor.h>
+#include <Imath/ImathMatrix.h>
+#include <Imath/ImathMatrixAlgo.h>
+#include <Imath/ImathPlane.h>
+#include <Imath/ImathRandom.h> 
+#include <Imath/ImathRoots.h>
+#include <Imath/ImathVec.h>
+#include <Imath/half.h> 
 
 //----------------------------------------------------------------------------//
 
--- Field3D-1.7.3/Field3D/Curve.h.orig	2020-03-11 15:54:36.000000000 +0000
+++ Field3D-1.7.3/Field3D/Curve.h	2022-05-04 11:04:38.000000000 +0100
@@ -53,8 +53,8 @@
 
 #include <boost/lexical_cast.hpp>
 
-#include <OpenEXR/ImathFun.h>
-#include <OpenEXR/ImathMatrix.h>
+#include <Imath/halfFunction.h>
+#include <Imath/ImathMatrix.h>
 
 //----------------------------------------------------------------------------//
 
