--- Field3D-1.3.2/CMakeLists.txt.orig	2012-02-04 17:37:22.000000000 +0000
+++ Field3D-1.3.2/CMakeLists.txt	2012-12-22 17:18:48.000000000 +0000
@@ -39,7 +39,7 @@
 
 FIND_PACKAGE (Doxygen)
 FIND_PACKAGE (HDF5)
-FIND_PACKAGE (Boost COMPONENTS thread program_options)
+FIND_PACKAGE (Boost COMPONENTS thread program_options system)
 FIND_PACKAGE (ILMBase)
 
 OPTION (INSTALL_DOCS "Automatically install documentation." ON)
