--- fityk-1.3.1/wxgui/app.cpp.orig	2016-05-13 20:46:46.000000000 +0100
+++ fityk-1.3.1/wxgui/app.cpp	2020-11-01 13:59:36.659322175 +0000
@@ -8,10 +8,10 @@
 #include <wx/filesys.h>
 #include <wx/tooltip.h>
 
-#ifdef __WXGTK3__
-#error "Not everything is working with wxGTK3. Use default wxGTK instead, " \
-       "based on GTK+2. If you want to test it, just remove this #error."
-#endif
+//#ifdef __WXGTK3__
+//#error "Not everything is working with wxGTK3. Use default wxGTK instead, " \
+//       "based on GTK+2. If you want to test it, just remove this #error."
+//#endif
 
 #include <stdio.h>
 #include <stdlib.h>
