--- flac-1.1.3/src/libFLAC/Makefile.in.orig	2006-11-27 06:41:11.000000000 +0000
+++ flac-1.1.3/src/libFLAC/Makefile.in	2007-04-02 22:21:04.000000000 +0100
@@ -268,7 +268,7 @@
 @FLaC__HAS_OGG_TRUE@	ogg_mapping.c
 
 # see 'http://www.gnu.org/software/libtool/manual.html#Libtool-versioning' for numbering convention
-libFLAC_la_LDFLAGS = -version-info 8:0:0 -lm $(LOCAL_EXTRA_LDFLAGS)
+libFLAC_la_LDFLAGS = -version-info 8:0:0 -lm $(LOCAL_EXTRA_LDFLAGS) -logg
 libFLAC_la_SOURCES = \
 	bitbuffer.c \
 	bitmath.c \
