--- flann-1.9.2/CMakeLists.txt.orig	2021-04-12 08:15:28.000000000 +0100
+++ flann-1.9.2/CMakeLists.txt	2024-05-03 11:25:17.644470618 +0100
@@ -164,7 +164,7 @@
 endif()
 
 # install and export variables
-set(config_install_dir "lib/cmake/${PROJECT_NAME}")
+set(config_install_dir "lib${LIB_SUFFIX}/cmake/${PROJECT_NAME}")
 set(generated_dir "${CMAKE_CURRENT_BINARY_DIR}/generated")
 set(version_config "${generated_dir}/flann-config-version.cmake")
 set(project_config "${generated_dir}/flann-config.cmake")
--- flann-1.9.2/src/python/CMakeLists.txt.orig	2021-04-12 08:15:28.000000000 +0100
+++ flann-1.9.2/src/python/CMakeLists.txt	2024-05-03 11:17:06.048437613 +0100
@@ -7,6 +7,6 @@
 # python instalation
 if (PYTHON_EXECUTABLE)
     install(CODE "execute_process(
-        COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/setup.py install
+        COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/setup.py install --root=${PKGROOT}
         WORKING_DIRECTORY \"${CMAKE_CURRENT_SOURCE_DIR}\")")
 endif()
