--- flann-1.8.4/src/python/CMakeLists.txt.orig	2013-01-15 22:38:32.000000000 +0000
+++ flann-1.8.4/src/python/CMakeLists.txt	2015-12-19 20:56:40.923467346 +0000
@@ -7,6 +7,6 @@
 # python instalation
 if (PYTHON_EXECUTABLE)
     install(CODE "execute_process(
-        COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/setup.py install
+        COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/setup.py install --root=\$DESTDIR
         WORKING_DIRECTORY \"${CMAKE_CURRENT_SOURCE_DIR}\")")
 endif()
--- flann-1.8.4/cmake/flann_utils.cmake.orig	2013-01-15 22:38:32.000000000 +0000
+++ flann-1.8.4/cmake/flann_utils.cmake	2015-12-19 20:50:30.962441063 +0000
@@ -1,6 +1,6 @@
 macro(GET_OS_INFO)
     string(REGEX MATCH "Linux" OS_IS_LINUX ${CMAKE_SYSTEM_NAME})
-    set(FLANN_LIB_INSTALL_DIR "lib")
+    set(FLANN_LIB_INSTALL_DIR "lib${LIB_SUFFIX}")
     set(FLANN_INCLUDE_INSTALL_DIR
         "include/${PROJECT_NAME_LOWER}-${FLANN_MAJOR_VERSION}.${FLANN_MINOR_VERSION}")
 endmacro(GET_OS_INFO)
