--- flex-2.6.0/src/skel.c.orig	2015-11-17 16:17:39.000000000 +0000
+++ flex-2.6.0/src/skel.c	2016-07-20 19:56:03.669125269 +0100
@@ -2561,7 +2561,7 @@
   "		 * scanner will even need a stack. We use 2 instead of 1 to avoid an",
   "		 * immediate realloc on the next call.",
   "         */",
-  "		num_to_alloc = 1; // After all that talk, this was set to 1 anyways...",
+  "		num_to_alloc = 1; /* After all that talk, this was set to 1 anyways... */",
   "		YY_G(yy_buffer_stack) = (struct yy_buffer_state**)yyalloc",
   "								(num_to_alloc * sizeof(struct yy_buffer_state*)",
   "								M4_YY_CALL_LAST_ARG);",
