--- flightgear-3.2.0/utils/fgviewer/CMakeLists.txt.orig	2014-08-17 14:40:24.000000000 +0100
+++ flightgear-3.2.0/utils/fgviewer/CMakeLists.txt	2016-01-31 15:20:14.035561247 +0000
@@ -43,6 +43,7 @@
 
 add_executable(fgviewer ${FGVIEWER_SOURCES} ${FGVIEWER_RTI_SOURCES})
 target_link_libraries(fgviewer
+	X11
 	${SIMGEAR_LIBRARIES}
 	${OPENSCENEGRAPH_LIBRARIES}
 	${OPENGL_LIBRARIES}
--- flightgear-3.2.0/src/Main/CMakeLists.txt.orig	2014-08-17 14:40:24.000000000 +0100
+++ flightgear-3.2.0/src/Main/CMakeLists.txt	2016-01-31 15:27:04.193562759 +0000
@@ -132,6 +132,7 @@
 endif()
 
 target_link_libraries(fgfs
+	X11
 	${SQLITE3_LIBRARY}
 	${SIMGEAR_LIBRARIES}
 	${OPENSCENEGRAPH_LIBRARIES}
