--- fontforge-20080302/fontforge/tottf.c.orig	2008-02-29 22:35:14.000000000 +0000
+++ fontforge-20080302/fontforge/tottf.c	2008-03-09 22:07:51.000000000 +0000
@@ -4723,7 +4723,7 @@
 	format2  = Needs816Enc(sf,&cjklen,map,&apple2,&applecjklen);
 	format14 = NeedsVariationSequenceTable(sf,&vslen,map);
     } else
-	format12 = format2 = format14 = NULL;
+	format12 = format2 = format14 = apple2 = NULL;
 
     /* Two/Three/Four encoding table pointers, one for ms, one for mac */
     /*  usually one for mac big, just a copy of ms */
