--- fontconfig-2.5.0/fonts.conf.in.orig	2007-10-25 22:14:27.000000000 +0100
+++ fontconfig-2.5.0/fonts.conf.in	2008-03-11 11:38:28.000000000 +0000
@@ -23,10 +23,22 @@
 
 <!-- Font directory list -->
 
-	<dir>@FC_DEFAULT_FONTS@</dir>
-	@FC_FONTPATH@
+	<dir>/usr/share/fonts/OTF</dir>
+	<dir>/usr/share/fonts/TTF</dir>
+	<dir>/usr/share/fonts/Type1</dir>
+	<dir>/usr/share/fonts/Speedo</dir>
+	<dir>/usr/share/fonts/cyrillic</dir>
+	<dir>/usr/share/fonts/misc</dir>
+	<dir>/usr/share/fonts/truetype</dir>
 	<dir>~/.fonts</dir>
 
+<!--    Commented out font directories:
+	These are not indexed by default, as fontconfig then seems
+	to prefer bitmapped fonts in some cases...
+	<dir>/usr/share/fonts/100dpi</dir>
+	<dir>/usr/share/fonts/75dpi</dir>
+-->
+
 <!--
   Accept deprecated 'mono' alias, replacing it with 'monospace'
 -->
