--- fontconfig-2.13.0/fonts.conf.in.orig	2017-11-20 12:02:20.000000000 +0000
+++ fontconfig-2.13.0/fonts.conf.in	2018-03-12 20:19:27.898997724 +0000
@@ -28,8 +28,12 @@
 
 <!-- Font directory list -->
 
-	<dir>@FC_DEFAULT_FONTS@</dir>
-	@FC_FONTPATH@
+        <dir>/usr/share/fonts/OTF</dir>
+        <dir>/usr/share/fonts/TTF</dir>
+        <dir>/usr/share/fonts/Type1</dir>
+        <dir>/usr/share/fonts/Speedo</dir>
+        <dir>/usr/share/fonts/cyrillic</dir>
+        <dir>/usr/share/fonts/misc</dir>
 	<dir prefix="xdg">fonts</dir>
 	<!-- the following element will be removed in the future -->
 	<dir>~/.fonts</dir>
