--- ./conf.d/60-latin.conf.orig	2006-09-09 22:30:00.000000000 -0500
+++ ./conf.d/60-latin.conf	2006-09-30 01:17:48.000000000 -0500
@@ -4,8 +4,8 @@
 	<alias>
 		<family>serif</family>
 		<prefer>
-			<family>Bitstream Vera Serif</family>
 			<family>DejaVu Serif</family>
+			<family>Bitstream Vera Serif</family>
 			<family>Times New Roman</family>
 			<family>Thorndale AMT</family>
 			<family>Luxi Serif</family>
@@ -16,8 +16,8 @@
 	<alias>
 		<family>sans-serif</family>
 		<prefer>
-			<family>Bitstream Vera Sans</family>
 			<family>DejaVu Sans</family>
+			<family>Bitstream Vera Sans</family>
 			<family>Verdana</family>
 			<family>Arial</family>
 			<family>Albany AMT</family>
@@ -29,8 +29,8 @@
 	<alias>
 		<family>monospace</family>
 		<prefer>
-			<family>Bitstream Vera Sans Mono</family>
 			<family>DejaVu Sans Mono</family>
+			<family>Bitstream Vera Sans Mono</family>
 			<family>Andale Mono</family>
 			<family>Courier New</family>
 			<family>Cumberland AMT</family>
--- ./conf.d/40-generic.conf.orig	2006-09-09 22:33:43.000000000 -0500
+++ ./conf.d/40-generic.conf	2006-09-30 01:17:22.000000000 -0500
@@ -10,8 +10,8 @@
   Serif faces
  -->
 	<alias>
-		<family>Bitstream Vera Serif</family>
 		<family>DejaVu Serif</family>
+		<family>Bitstream Vera Serif</family>
 		<family>Times New Roman</family>
 		<family>Times</family>
 		<family>Nimbus Roman No9 L</family>
@@ -29,8 +29,8 @@
   Sans-serif faces
  -->
 	<alias>
-		<family>Bitstream Vera Sans</family>
 		<family>DejaVu Sans</family>
+		<family>Bitstream Vera Sans</family>
 		<family>Helvetica</family>
 		<family>Arial</family>
 		<family>Verdana</family>
@@ -51,8 +51,8 @@
   Monospace faces
  -->
  	<alias>
-		<family>Bitstream Vera Sans Mono</family>
 		<family>DejaVu Sans Mono</family>
+		<family>Bitstream Vera Sans Mono</family>
 		<family>Courier</family>
 		<family>Courier New</family>
 		<family>Andale Mono</family>
