--- fontconfig-2.10.1/fonts.conf.in.orig	2012-11-11 07:34:25.000000000 +0000
+++ fontconfig-2.10.1/fonts.conf.in	2012-11-11 07:38:02.000000000 +0000
@@ -23,8 +23,12 @@
 
 <!-- Font directory list -->
 
-	<dir>@FC_DEFAULT_FONTS@</dir>
-	@FC_FONTPATH@
+        <dir>/usr/share/fonts/OTF</dir>
+        <dir>/usr/share/fonts/TTF</dir>
+        <dir>/usr/share/fonts/Type1</dir>
+        <dir>/usr/share/fonts/Speedo</dir>
+        <dir>/usr/share/fonts/cyrillic</dir>
+        <dir>/usr/share/fonts/misc</dir>
 	<dir prefix="xdg">fonts</dir>
 	<!-- the following element will be removed in the future -->
 	<dir>~/.fonts</dir>
