--- fontconfig-2.13.0/conf.d/60-latin.conf.orig	2017-11-20 12:02:20.000000000 +0000
+++ fontconfig-2.13.0/conf.d/60-latin.conf	2018-03-12 20:19:27.887997724 +0000
@@ -9,8 +9,8 @@
 	<alias>
 		<family>serif</family>
 		<prefer>
-			<family>Bitstream Vera Serif</family>
 			<family>DejaVu Serif</family>
+			<family>Bitstream Vera Serif</family>
 			<family>Times New Roman</family>
 			<family>Thorndale AMT</family>
 			<family>Luxi Serif</family>
@@ -22,8 +22,8 @@
 	<alias>
 		<family>sans-serif</family>
 		<prefer>
-			<family>Bitstream Vera Sans</family>
 			<family>DejaVu Sans</family>
+			<family>Bitstream Vera Sans</family>
 			<family>Verdana</family>
 			<family>Arial</family>
 			<family>Albany AMT</family>
@@ -39,8 +39,8 @@
 	<alias>
 		<family>monospace</family>
 		<prefer>
-			<family>Bitstream Vera Sans Mono</family>
 			<family>DejaVu Sans Mono</family>
+			<family>Bitstream Vera Sans Mono</family>
 			<family>Inconsolata</family>
 			<family>Andale Mono</family>
 			<family>Courier New</family>
--- fontconfig-2.13.0/conf.d/45-latin.conf.orig	2017-11-20 12:02:20.000000000 +0000
+++ fontconfig-2.13.0/conf.d/45-latin.conf	2018-03-12 20:25:11.107998989 +0000
@@ -15,7 +15,7 @@
   Serif faces
  -->
 	<alias>
-		<family>Bitstream Vera Serif</family>
+		<family>DejaVu Serif</family>
 		<default><family>serif</family></default>
 	</alias>
 	<alias>
@@ -27,7 +27,7 @@
 		<default><family>serif</family></default>
 	</alias>
 	<alias>
-		<family>DejaVu Serif</family>
+		<family>Liberation Serif</family>
 		<default><family>serif</family></default>
 	</alias>
 	<alias>
@@ -43,7 +43,7 @@
 		<default><family>serif</family></default>
 	</alias>
 	<alias>
-		<family>Liberation Serif</family>
+		<family>Bitstream Vera Serif</family>
 		<default><family>serif</family></default>
 	</alias>
 	<alias>
@@ -102,7 +102,7 @@
 		<default><family>sans-serif</family></default>
 	</alias>
 	<alias>
-		<family>Bitstream Vera Sans</family>
+		<family>DejaVu Sans</family>
 		<default><family>sans-serif</family></default>
 	</alias>
 	<alias>
@@ -126,7 +126,7 @@
 		<default><family>sans-serif</family></default>
 	</alias>
 	<alias>
-		<family>DejaVu Sans</family>
+		<family>Liberation Sans</family>
 		<default><family>sans-serif</family></default>
 	</alias>
 	<alias>
@@ -138,7 +138,7 @@
 		<default><family>sans-serif</family></default>
 	</alias>
 	<alias>
-		<family>Liberation Sans</family>
+		<family>Bitstream Vera Sans</family>
 		<default><family>sans-serif</family></default>
 	</alias>
 	<alias>
@@ -180,10 +180,6 @@
 		<family>Andale Mono</family>
 		<default><family>monospace</family></default>
 	</alias>
- 	<alias>
-		<family>Bitstream Vera Sans Mono</family>
-		<default><family>monospace</family></default>
-	</alias>
 	<alias>
 		<family>Consolas</family>
 		<default><family>monospace</family></default>
@@ -212,6 +208,10 @@
 		<family>Fixedsys</family>
 		<default><family>monospace</family></default>
 	</alias>
+ 	<alias>
+		<family>Bitstream Vera Sans Mono</family>
+		<default><family>monospace</family></default>
+	</alias>
 	<alias>
 		<family>Inconsolata</family>
 		<default><family>monospace</family></default>
