--- fontconfig-2.5.0/conf.d/45-latin.conf.orig	2007-11-05 23:38:03.000000000 +0000
+++ fontconfig-2.5.0/conf.d/45-latin.conf	2008-03-11 11:29:37.000000000 +0000
@@ -10,8 +10,8 @@
   Serif faces
  -->
 	<alias>
-		<family>Bitstream Vera Serif</family>
 		<family>DejaVu Serif</family>
+		<family>Bitstream Vera Serif</family>
 		<family>Liberation Serif</family>
 		<family>Times New Roman</family>
 		<family>Times</family>
@@ -25,8 +25,8 @@
   Sans-serif faces
  -->
 	<alias>
-		<family>Bitstream Vera Sans</family>
 		<family>DejaVu Sans</family>
+		<family>Bitstream Vera Sans</family>
 		<family>Liberation Sans</family>
 		<family>Arial</family>
 		<family>Helvetica</family>
@@ -41,8 +41,8 @@
   Monospace faces
  -->
  	<alias>
-		<family>Bitstream Vera Sans Mono</family>
 		<family>DejaVu Sans Mono</family>
+		<family>Bitstream Vera Sans Mono</family>
 		<family>Liberation Mono</family>
 		<family>Courier New</family>
 		<family>Courier</family>
--- fontconfig-2.5.0/fonts.conf.in.orig	2007-10-25 22:14:27.000000000 +0100
+++ fontconfig-2.5.0/fonts.conf.in	2008-03-11 11:38:28.000000000 +0000
@@ -23,10 +23,22 @@
 
 <!-- Font directory list -->
 
-	<dir>@FC_DEFAULT_FONTS@</dir>
-	@FC_FONTPATH@
+	<dir>/usr/share/fonts/OTF</dir>
+	<dir>/usr/share/fonts/TTF</dir>
+	<dir>/usr/share/fonts/Type1</dir>
+	<dir>/usr/share/fonts/Speedo</dir>
+	<dir>/usr/share/fonts/cyrillic</dir>
+	<dir>/usr/share/fonts/misc</dir>
+	<dir>/usr/share/fonts/truetype</dir>
 	<dir>~/.fonts</dir>
 
+<!--    Commented out font directories:
+	These are not indexed by default, as fontconfig then seems
+	to prefer bitmapped fonts in some cases...
+	<dir>/usr/share/fonts/100dpi</dir>
+	<dir>/usr/share/fonts/75dpi</dir>
+-->
+
 <!--
   Accept deprecated 'mono' alias, replacing it with 'monospace'
 -->
