--- fotoxx-10.8.4/Makefile.orig	2010-08-18 12:23:51.000000000 +0100
+++ fotoxx-10.8.4/Makefile	2010-08-19 09:34:40.000000000 +0100
@@ -47,9 +47,9 @@
 
 menu:
 	cp -f desktop $(DESTDIR)$(DESKTOP)
-	echo Exec=$(DESTDIR)$(BINDIR)/$(PROGRAM) >> $(DESTDIR)$(DESKTOP)
-	echo Icon=$(DESTDIR)$(ICONDIR)/$(PROGRAM).png >> $(DESTDIR)$(DESKTOP)
-	xdg-desktop-menu install $(DESTDIR)$(DESKTOP)
+	echo Exec=$(BINDIR)/$(PROGRAM) >> $(DESTDIR)$(DESKTOP)
+	echo Icon=$(ICONDIR)/$(PROGRAM).png >> $(DESTDIR)$(DESKTOP)
+#	xdg-desktop-menu install $(DESTDIR)$(DESKTOP)
 
 manpage:
 	cp -f doc/$(PROGRAM).man $(PROGRAM).1
