--- FreeCAD-0.18.4/src/Gui/DAGView/DAGView.cpp.orig	2019-10-22 17:53:35.000000000 +0100
+++ FreeCAD-0.18.4/src/Gui/DAGView/DAGView.cpp	2020-05-10 13:40:12.737256150 +0100
@@ -39,6 +39,7 @@
 
 using namespace Gui;
 using namespace DAG;
+using namespace boost::placeholders;
 
 DAG::DockWindow::DockWindow(Gui::Document* gDocumentIn, QWidget* parent): Gui::DockWindow(gDocumentIn, parent)
 {
--- FreeCAD-0.18.4/src/Mod/Sketcher/Gui/TaskSketcherGeneral.cpp.orig	2019-10-22 17:53:35.000000000 +0100
+++ FreeCAD-0.18.4/src/Mod/Sketcher/Gui/TaskSketcherGeneral.cpp	2020-05-10 14:10:49.912794147 +0100
@@ -40,6 +40,7 @@
 
 using namespace SketcherGui;
 using namespace Gui::TaskView;
+using namespace boost::placeholders;
 
 SketcherGeneralWidget::SketcherGeneralWidget(QWidget *parent)
   : QWidget(parent), ui(new Ui_TaskSketcherGeneral)
