--- FreeCAD-0.21.2/src/Mod/Part/App/Geometry.cpp.orig	2023-11-14 02:33:39.000000000 +0000
+++ FreeCAD-0.21.2/src/Mod/Part/App/Geometry.cpp	2024-08-16 10:23:23.253185157 +0100
@@ -138,6 +138,8 @@
 #include "SurfaceOfRevolutionPy.h"
 #include "Tools.h"
 #include "ToroidPy.h"
+
+#include <boost/random.hpp>
 
 
 #if OCC_VERSION_HEX >= 0x070600
--- FreeCAD-0.21.2/src/Mod/TechDraw/App/CenterLine.cpp.orig	2023-11-14 02:33:39.000000000 +0000
+++ FreeCAD-0.21.2/src/Mod/TechDraw/App/CenterLine.cpp	2024-08-16 10:49:28.084290218 +0100
@@ -42,6 +42,8 @@
 #include "GeometryObject.h"
 #include "CenterLinePy.h"
 
+#include <boost/random.hpp>
+
 using namespace TechDraw;
 using DU = DrawUtil;
 
--- FreeCAD-0.21.2/src/Mod/TechDraw/App/Cosmetic.cpp.orig	2023-11-14 02:33:39.000000000 +0000
+++ FreeCAD-0.21.2/src/Mod/TechDraw/App/Cosmetic.cpp	2024-08-16 10:52:03.061300623 +0100
@@ -40,6 +40,7 @@
 #include "LineGroup.h"
 #include "Preferences.h"
 
+#include <boost/random.hpp>
 
 using namespace TechDraw;
 using namespace std;
--- FreeCAD-0.21.2/src/Mod/TechDraw/App/CosmeticVertex.cpp.orig	2023-11-14 02:33:39.000000000 +0000
+++ FreeCAD-0.21.2/src/Mod/TechDraw/App/CosmeticVertex.cpp	2024-08-16 10:52:53.852304033 +0100
@@ -36,6 +36,8 @@
 #include "LineGroup.h"
 #include "Preferences.h"
 
+#include <boost/random.hpp>
+
 using namespace TechDraw;
 using namespace std;
 
--- FreeCAD-0.21.2/src/Mod/TechDraw/App/Geometry.cpp.orig	2023-11-14 02:33:39.000000000 +0000
+++ FreeCAD-0.21.2/src/Mod/TechDraw/App/Geometry.cpp	2024-08-16 10:47:26.135282030 +0100
@@ -81,6 +81,7 @@
 #include "GeometryObject.h"
 #include "DrawUtil.h"
 
+#include <boost/random.hpp>
 
 using namespace TechDraw;
 using namespace Part;
--- FreeCAD-0.21.2/src/Mod/Sketcher/App/Constraint.cpp.orig	2023-11-14 02:33:39.000000000 +0000
+++ FreeCAD-0.21.2/src/Mod/Sketcher/App/Constraint.cpp	2024-08-16 10:46:32.216278410 +0100
@@ -32,6 +32,8 @@
 
 #include "Constraint.h"
 #include "ConstraintPy.h"
+
+#include <boost/random.hpp>
 
 
 using namespace Sketcher;
--- FreeCAD-0.21.2/src/Gui/PreferencePackManager.cpp.orig	2023-11-14 02:33:39.000000000 +0000
+++ FreeCAD-0.21.2/src/Gui/PreferencePackManager.cpp	2024-04-18 16:26:00.828115601 +0100
@@ -221,7 +221,7 @@
     auto savedPreferencePacksDirectory =
         fs::path(App::Application::getUserAppDataDir()) / "SavedPreferencePacks";
     auto cfgFilename = savedPreferencePacksDirectory / packName / (packName + ".cfg");
-    fs::copy_file(path, cfgFilename, fs::copy_option::overwrite_if_exists);
+    fs::copy_file(path, cfgFilename, fs::copy_options::overwrite_existing);
     rescan();
 }
 
--- FreeCAD-0.21.2/src/3rdParty/salomesmesh/src/SMESH/DriverGMF.cpp.orig	2023-11-14 02:33:39.000000000 +0000
+++ FreeCAD-0.21.2/src/3rdParty/salomesmesh/src/SMESH/DriverGMF.cpp	2024-04-17 22:31:01.253265551 +0100
@@ -55,7 +55,7 @@
 
   bool isExtensionCorrect( const std::string& fileName )
   {
-    std::string ext  = boost::filesystem::extension(fileName);
+    std::string ext  = boost::filesystem::path(fileName).extension().string();
     switch ( ext.size() ) {
     case 5: return ( ext == ".mesh" || ext == ".solb" );
     case 6: return ( ext == ".meshb" );
