--- FreeCAD-0.20.2/src/3rdParty/libE57Format/include/E57Format.h.orig	2022-12-07 02:35:37.000000000 +0000
+++ FreeCAD-0.20.2/src/3rdParty/libE57Format/include/E57Format.h	2023-07-20 22:01:17.313078047 +0100
@@ -32,6 +32,7 @@
 //! @file  E57Format.h header file for the E57 API
 
 #include <cfloat>
+#include <cstdint>
 #include <memory>
 #include <vector>
 
--- FreeCAD-0.20.2/src/Mod/Part/App/OCCError.h.orig	2022-12-07 02:35:37.000000000 +0000
+++ FreeCAD-0.20.2/src/Mod/Part/App/OCCError.h	2023-07-21 18:39:27.692992381 +0100
@@ -50,7 +50,6 @@
 # include <Standard_Overflow.hxx>
 # include <Standard_ProgramError.hxx>
 # include <Standard_RangeError.hxx>
-# include <Standard_TooManyUsers.hxx>
 # include <Standard_TypeMismatch.hxx>
 # include <Standard_Underflow.hxx>
 
