--- FreeCAD-0.17/src/Mod/Path/libarea/CMakeLists.txt.orig	2018-04-06 09:48:37.000000000 +0100
+++ FreeCAD-0.17/src/Mod/Path/libarea/CMakeLists.txt	2018-04-16 21:16:01.188805702 +0100
@@ -19,10 +19,10 @@
         if(NOT PYTHON_VERSION_MAJOR LESS 3)
             find_package( Boost COMPONENTS python3)
             if (NOT Boost_PYTHON3_FOUND)
-                find_package( Boost COMPONENTS python REQUIRED)
+                find_package( Boost COMPONENTS python27 REQUIRED)
             endif()
         else()
-            find_package( Boost COMPONENTS python REQUIRED)  # find BOOST and boost-python
+            find_package( Boost COMPONENTS python27 REQUIRED)  # find BOOST and boost-python
         endif()
 
         if(Boost_FOUND)
