--- FreeCAD-1.0.2/cMake/FreeCAD_Helpers/SetupBoost.cmake.orig	2025-08-10 10:38:54.000000000 +0100
+++ FreeCAD-1.0.2/cMake/FreeCAD_Helpers/SetupBoost.cmake	2025-12-12 09:36:53.535585629 +0000
@@ -3,7 +3,7 @@
 
     set(_boost_TEST_VERSIONS ${Boost_ADDITIONAL_VERSIONS})
 
-    set (BOOST_COMPONENTS filesystem program_options regex system thread date_time)
+    set (BOOST_COMPONENTS filesystem program_options regex thread date_time)
     find_package(Boost ${BOOST_MIN_VERSION}
         COMPONENTS ${BOOST_COMPONENTS} REQUIRED)
 
--- FreeCAD-1.0.2/src/Mod/Sketcher/App/SolverGeometryExtension.h.orig	2025-08-10 10:38:57.000000000 +0100
+++ FreeCAD-1.0.2/src/Mod/Sketcher/App/SolverGeometryExtension.h	2025-12-12 10:30:40.695924199 +0000
@@ -23,6 +23,8 @@
 #ifndef SKETCHER_SOLVERGEOMETRYEXTENSION_H
 #define SKETCHER_SOLVERGEOMETRYEXTENSION_H
 
+#include <algorithm>
+
 #include <Base/Exception.h>
 #include <Mod/Part/App/GeometryExtension.h>
 #include <Mod/Sketcher/SketcherGlobal.h>
