--- freetype-2.3.4/include/freetype/config/ftoption.h.orig	2007-01-05 16:18:45.000000000 +0000
+++ freetype-2.3.4/include/freetype/config/ftoption.h	2007-04-27 09:18:31.000000000 +0100
@@ -92,7 +92,7 @@
   /* This is done to allow FreeType clients to run unmodified, forcing     */
   /* them to display normal gray-level anti-aliased glyphs.                */
   /*                                                                       */
-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
 
 
   /*************************************************************************/
@@ -456,7 +456,7 @@
   /*   Do not #undef this macro here, since the build system might         */
   /*   define it for certain configurations only.                          */
   /*                                                                       */
-/* #define TT_CONFIG_OPTION_BYTECODE_INTERPRETER */
+#define TT_CONFIG_OPTION_BYTECODE_INTERPRETER
 
 
   /*************************************************************************/
@@ -468,7 +468,7 @@
   /* FT_PARAM_TAG_UNPATENTED_HINTING; or when the debug hook               */
   /* FT_DEBUG_HOOK_UNPATENTED_HINTING is globally activated.               */
   /*                                                                       */
-#define TT_CONFIG_OPTION_UNPATENTED_HINTING
+/* #define TT_CONFIG_OPTION_UNPATENTED_HINTING */
 
 
   /*************************************************************************/
--- freetype-2.3.4/include/freetype/freetype.h.orig	2007-04-10 04:58:39.000000000 +0100
+++ freetype-2.3.4/include/freetype/freetype.h	2007-04-27 09:18:20.000000000 +0100
@@ -16,13 +16,20 @@
 /***************************************************************************/
 
 
-#ifndef FT_FREETYPE_H
-#error "`ft2build.h' hasn't been included yet!"
-#error "Please always use macros to include FreeType header files."
-#error "Example:"
-#error "  #include <ft2build.h>"
-#error "  #include FT_FREETYPE_H"
-#endif
+/***************************************************************************/
+/*                                                                         */
+/* If you think breaking almost 100% of all source that links with         */
+/* freetype (anything using a configure file for sure) is a good idea,     */
+/* then feel free to uncomment this block.                                 */
+/*                                                                         */
+/* #ifndef FT_FREETYPE_H                                                   */
+/* #error "`ft2build.h' hasn't been included yet!"                         */
+/* #error "Please always use macros to include FreeType header files."     */
+/* #error "Example:"                                                       */
+/* #error "  #include <ft2build.h>"                                        */
+/* #error "  #include FT_FREETYPE_H"                                       */
+/* #endif                                                                  */
+/***************************************************************************/
 
 
   /*************************************************************************/
