--- freetype-2.9.1/builds/unix/unix-cc.in.orig	2018-04-22 10:41:36.000000000 +0100
+++ freetype-2.9.1/builds/unix/unix-cc.in	2018-05-02 20:51:36.590295885 +0100
@@ -92,10 +92,10 @@
 
 # Resource compiler to use on Cygwin/MinGW, usually windres.
 #
-RCraw := @RC@
-ifneq ($(RCraw),)
-  RC := $(LIBTOOL) --tag=RC --mode=compile $(RCraw)
-endif
+#RCraw := @RC@
+#ifneq ($(RCraw),)
+#  RC := $(LIBTOOL) --tag=RC --mode=compile $(RCraw)
+#endif
 
 # Linker flags.
 #
--- freetype-2.9.1/include/freetype/config/ftoption.h.orig	2018-04-22 10:41:36.000000000 +0100
+++ freetype-2.9.1/include/freetype/config/ftoption.h	2018-05-02 20:51:05.083297388 +0100
@@ -122,7 +122,7 @@
   /* rendering technology that produces excellent output without LCD       */
   /* filtering.                                                            */
   /*                                                                       */
-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
 
 
   /*************************************************************************/
--- freetype-2.9.1/include/freetype/freetype.h.orig	2018-05-01 11:34:32.000000000 +0100
+++ freetype-2.9.1/include/freetype/freetype.h	2018-05-02 20:51:05.083297388 +0100
@@ -20,14 +20,20 @@
 #define FREETYPE_H_
 
 
-#ifndef FT_FREETYPE_H
-#error "`ft2build.h' hasn't been included yet!"
-#error "Please always use macros to include FreeType header files."
-#error "Example:"
-#error "  #include <ft2build.h>"
-#error "  #include FT_FREETYPE_H"
-#endif
-
+/***************************************************************************/
+/*                                                                         */
+/* If you think breaking almost 100% of all source that links with         */
+/* freetype (anything using a configure file for sure) is a good idea,     */
+/* then feel free to uncomment this block.                                 */
+/*                                                                         */
+/* #ifndef FT_FREETYPE_H                                                   */
+/* #error "`ft2build.h' hasn't been included yet!"                         */
+/* #error "Please always use macros to include FreeType header files."     */
+/* #error "Example:"                                                       */
+/* #error "  #include <ft2build.h>"                                        */
+/* #error "  #include FT_FREETYPE_H"                                       */
+/* #endif                                                                  */
+/***************************************************************************/
 
 #include <ft2build.h>
 #include FT_CONFIG_CONFIG_H
