--- freetype-2.3.2/include/freetype/config/ftoption.h.orig	2007-01-05 16:18:45.000000000 +0000
+++ freetype-2.3.2/include/freetype/config/ftoption.h	2007-03-16 21:07:49.000000000 +0000
@@ -456,7 +456,7 @@
   /*   Do not #undef this macro here, since the build system might         */
   /*   define it for certain configurations only.                          */
   /*                                                                       */
-/* #define TT_CONFIG_OPTION_BYTECODE_INTERPRETER */
+#define TT_CONFIG_OPTION_BYTECODE_INTERPRETER
 
 
   /*************************************************************************/
--- freetype-2.3.2/include/freetype/freetype.h.orig	2007-03-08 19:01:15.000000000 +0000
+++ freetype-2.3.2/include/freetype/freetype.h	2007-03-16 21:09:53.000000000 +0000
@@ -15,7 +15,7 @@
 /*                                                                         */
 /***************************************************************************/
 
-
+#if 0
 #ifndef FT_FREETYPE_H
 #error "`ft2build.h' hasn't been included yet!"
 #error "Please always use macros to include FreeType header files."
@@ -23,7 +23,7 @@
 #error "  #include <ft2build.h>"
 #error "  #include FT_FREETYPE_H"
 #endif
-
+#endif
 
   /*************************************************************************/
   /*                                                                       */
