--- freetype-2.5.3/include/freetype.h.orig	2014-03-06 18:10:31.000000000 +0000
+++ freetype-2.5.3/include/freetype.h	2014-04-18 15:39:25.890945439 +0100
@@ -20,14 +20,20 @@
 #define __FREETYPE_H__
 
 
-#ifndef FT_FREETYPE_H
-#error "`ft2build.h' hasn't been included yet!"
-#error "Please always use macros to include FreeType header files."
-#error "Example:"
-#error "  #include <ft2build.h>"
-#error "  #include FT_FREETYPE_H"
-#endif
-
+/***************************************************************************/
+/*                                                                         */
+/* If you think breaking almost 100% of all source that links with         */
+/* freetype (anything using a configure file for sure) is a good idea,     */
+/* then feel free to uncomment this block.                                 */
+/*                                                                         */
+/* #ifndef FT_FREETYPE_H                                                   */
+/* #error "`ft2build.h' hasn't been included yet!"                         */
+/* #error "Please always use macros to include FreeType header files."     */
+/* #error "Example:"                                                       */
+/* #error "  #include <ft2build.h>"                                        */
+/* #error "  #include FT_FREETYPE_H"                                       */
+/* #endif                                                                  */
+/***************************************************************************/
 
 #include <ft2build.h>
 #include FT_CONFIG_CONFIG_H
--- freetype-2.5.3/include/config/ftoption.h.orig	2014-03-01 11:27:34.000000000 +0000
+++ freetype-2.5.3/include/config/ftoption.h	2014-04-18 15:37:42.655956163 +0100
@@ -92,7 +92,7 @@
   /* This is done to allow FreeType clients to run unmodified, forcing     */
   /* them to display normal gray-level anti-aliased glyphs.                */
   /*                                                                       */
-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
 
 
   /*************************************************************************/
