--- freetype-2.4.0/include/freetype/config/ftoption.h.orig	2010-07-05 05:38:55.000000000 +0100
+++ freetype-2.4.0/include/freetype/config/ftoption.h	2010-07-14 06:06:56.000000000 +0100
@@ -93,7 +93,7 @@
   /* This is done to allow FreeType clients to run unmodified, forcing     */
   /* them to display normal gray-level anti-aliased glyphs.                */
   /*                                                                       */
-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
 
 
   /*************************************************************************/
--- freetype-2.4.0/include/freetype/freetype.h.orig	2010-07-12 20:27:14.000000000 +0100
+++ freetype-2.4.0/include/freetype/freetype.h	2010-07-14 06:06:56.000000000 +0100
@@ -17,13 +17,20 @@
 /***************************************************************************/
 
 
-#ifndef FT_FREETYPE_H
-#error "`ft2build.h' hasn't been included yet!"
-#error "Please always use macros to include FreeType header files."
-#error "Example:"
-#error "  #include <ft2build.h>"
-#error "  #include FT_FREETYPE_H"
-#endif
+/***************************************************************************/
+/*                                                                         */
+/* If you think breaking almost 100% of all source that links with         */
+/* freetype (anything using a configure file for sure) is a good idea,     */
+/* then feel free to uncomment this block.                                 */
+/*                                                                         */
+/* #ifndef FT_FREETYPE_H                                                   */
+/* #error "`ft2build.h' hasn't been included yet!"                         */
+/* #error "Please always use macros to include FreeType header files."     */
+/* #error "Example:"                                                       */
+/* #error "  #include <ft2build.h>"                                        */
+/* #error "  #include FT_FREETYPE_H"                                       */
+/* #endif                                                                  */
+/***************************************************************************/
 
 
 #ifndef __FREETYPE_H__
