--- FreeCAD-1.0.2/cMake/FreeCAD_Helpers/SetupBoost.cmake.orig	2025-08-10 10:38:54.000000000 +0100
+++ FreeCAD-1.0.2/cMake/FreeCAD_Helpers/SetupBoost.cmake	2025-08-17 17:01:32.477391165 +0100
@@ -3,7 +3,7 @@
 
     set(_boost_TEST_VERSIONS ${Boost_ADDITIONAL_VERSIONS})
 
-    set (BOOST_COMPONENTS filesystem program_options regex system thread date_time)
+    set (BOOST_COMPONENTS filesystem program_options regex thread date_time)
     find_package(Boost ${BOOST_MIN_VERSION}
         COMPONENTS ${BOOST_COMPONENTS} REQUIRED)
 
