--- freetype-2.1.9/include/freetype/config/ftoption.h	2004-04-25 21:15:09.000000000 +0100
+++ freetype-2.1.9/include/freetype/config/ftoption.h	2004-09-22 22:28:14.000000000 +0100
@@ -436,7 +436,7 @@
   /*   Do not #undef this macro here, since the build system might         */
   /*   define it for certain configurations only.                          */
   /*                                                                       */
-/* #define  TT_CONFIG_OPTION_BYTECODE_INTERPRETER */
+#define  TT_CONFIG_OPTION_BYTECODE_INTERPRETER
 
 
   /*************************************************************************/
