--- FreeCAD-0.19/cMake/FreeCAD_Helpers/SetupPython.cmake.orig	2021-02-25 14:08:19.000000000 +0000
+++ FreeCAD-0.19/cMake/FreeCAD_Helpers/SetupPython.cmake	2021-02-28 17:28:54.325407283 +0000
@@ -3,7 +3,7 @@
 
     # http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=677598
     # Acceptable versions of Python
-    set(Python_ADDITIONAL_VERSIONS "2.7")
+    set(Python_ADDITIONAL_VERSIONS 3.10 3.9 3.8 3.7 3.6 3.5 3.4 "2.7")
 
     # For building on OS X
     if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND NOT BUILD_WITH_CONDA)
