--- fritzing-app-0.9.6/pri/libgit2detect.pri.orig	2021-02-21 09:53:04.000000000 +0000
+++ fritzing-app-0.9.6/pri/libgit2detect.pri	2021-05-30 11:43:46.105835954 +0100
@@ -14,10 +14,7 @@
 # ********************************************************************
 
 
-packagesExist(libgit2) {
-} else {
-    LIBGIT_STATIC = true
-}
+LIBGIT_STATIC = !packagesExist(libgit2) 
 
 if ($$LIBGIT_STATIC) {
     LIBGIT2INCLUDE = "$$_PRO_FILE_PWD_/../libgit2/include"
