--- ftgl-2.4.0/CMakeLists.txt.orig	2019-02-07 21:39:58.000000000 +0000
+++ ftgl-2.4.0/CMakeLists.txt	2020-03-22 22:20:33.832787965 +0000
@@ -67,7 +67,7 @@
 if (WIN32)
   set(cmakedir cmake)
 else ()
-  set(cmakedir lib/cmake/FTGL)
+  set(cmakedir lib${LIB_SUFFIX}/cmake/FTGL)
 endif ()
 
 install(EXPORT FTGL-targets DESTINATION "${cmakedir}")
--- ftgl-2.4.0/src/CMakeLists.txt.orig	2019-02-07 21:39:58.000000000 +0000
+++ ftgl-2.4.0/src/CMakeLists.txt	2020-03-22 22:19:42.954790393 +0000
@@ -26,7 +26,7 @@
     FTGlyphContainer.h
     FTInternals.h
     FTLibrary.cpp
-    FTLibrary.h
+    FTGL/FTLibrary.h
     FTList.h
     FTPoint.cpp
     FTSize.cpp
@@ -138,8 +138,8 @@
 INSTALL(TARGETS   ftgl                         	  
     EXPORT  FTGL-targets
 	RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX}
 )
 
 FILE(GLOB files "${CMAKE_CURRENT_SOURCE_DIR}/FTGL/*.h")
