--- fuse-3.2.6/util/install_helper.sh.orig	2018-08-31 19:34:41.400620227 +0100
+++ fuse-3.2.6/util/install_helper.sh	2018-08-31 19:39:04.887607656 +0100
@@ -22,17 +22,17 @@
     DESTDIR="${DESTDIR%/}"
 fi
 
-chown root:root "${DESTDIR}${bindir}/fusermount3"
-chmod u+s "${DESTDIR}${bindir}/fusermount3"
+#chown root:root "${DESTDIR}${bindir}/fusermount3"
+#chmod u+s "${DESTDIR}${bindir}/fusermount3"
 
 install -D -m 644 "${MESON_SOURCE_ROOT}/util/fuse.conf" \
 	"${DESTDIR}${sysconfdir}/fuse.conf"
 
 
-if test ! -e "${DESTDIR}/dev/fuse"; then
-    mkdir -p "${DESTDIR}/dev"
-    mknod "${DESTDIR}/dev/fuse" -m 0666 c 10 229
-fi
+#if test ! -e "${DESTDIR}/dev/fuse"; then
+#    mkdir -p "${DESTDIR}/dev"
+#    mknod "${DESTDIR}/dev/fuse" -m 0666 c 10 229
+#fi
 
 install -D -m 644 "${MESON_SOURCE_ROOT}/util/udev.rules" \
         "${DESTDIR}${udevrulesdir}/99-fuse3.rules"
