--- fwupd-2.0.7/meson.build.orig	2025-03-25 16:15:38.000000000 +0000
+++ fwupd-2.0.7/meson.build	2025-03-26 15:40:13.461284266 +0000
@@ -514,6 +514,11 @@
   endif
 endif
 
+elogind = dependency('systemd', 'libelogind', required: get_option('elogind'))
+if elogind.found()
+  conf.set('HAVE_LOGIND' , '1')
+endif
+
 supported_build = get_option('supported_build').disable_auto_if(not tag).allowed()
 if supported_build
     conf.set('SUPPORTED_BUILD', '1')
@@ -698,6 +703,7 @@
   'dbus_socket_address': get_option('dbus_socket_address'),
   'vendor_ids_dir': vendor_ids_dir,
   'docs': build_docs,
+  'elogind': elogind,
   'gnutls': gnutls,
   'introspection': introspection.allowed(),
   'libblkid': libblkid,
--- fwupd-2.0.7/meson_options.txt.orig	2025-03-25 16:15:38.000000000 +0000
+++ fwupd-2.0.7/meson_options.txt	2025-03-26 15:37:49.284274586 +0000
@@ -44,6 +44,10 @@
   type: 'string',
   description: 'the hardcoded name of OS directory in ESP, e.g. fedora',
 )
+option('elogind',
+  type: 'feature',
+  description: 'elogind support',
+)
 option('firmware-packager',
   type: 'boolean',
   value: true,
--- fwupd-2.0.7/data/meson.build.orig	2025-03-25 16:15:38.000000000 +0000
+++ fwupd-2.0.7/data/meson.build	2025-03-26 15:37:49.284274586 +0000
@@ -167,7 +167,7 @@
     )
   endif
 
-  if libsystemd.found()
+  if libsystemd.found() or elogind.found()
     con2 = configuration_data()
     con2.set('libexecdir', libexecdir)
 
--- fwupd-2.0.7/plugins/logind/meson.build.orig	2025-03-25 16:15:38.000000000 +0000
+++ fwupd-2.0.7/plugins/logind/meson.build	2025-03-26 15:37:49.284274586 +0000
@@ -1,4 +1,4 @@
-if libsystemd.found()
+if libsystemd.found() or elogind.found()
 cargs = ['-DG_LOG_DOMAIN="FuPluginLogind"']
 plugins += {meson.current_source_dir().split('/')[-1]: true}
 
