--- gambas-1.0.15/Makefile.in.orig	2006-03-24 21:56:54.000000000 +0000
+++ gambas-1.0.15/Makefile.in	2006-03-27 15:35:27.000000000 +0100
@@ -706,14 +706,11 @@
 
 	@if test "$(DESTDIR)$(bindir)" != "$(ROOT)/usr/bin" && test "$(DESTDIR)$(bindir)" != "$(ROOT)/usr/bin/"; then \
 	 echo "Making symbolic links in /usr/bin..."; \
-	 rm -f $(ROOT)/usr/bin/gbx; \
-	 ln -s $(DESTDIR)$(bindir)/gbx $(ROOT)/usr/bin/gbx; \
-	 rm -f $(ROOT)/usr/bin/gbc; \
-	 ln -s $(DESTDIR)$(bindir)/gbc $(ROOT)/usr/bin/gbc; \
-	 rm -f $(ROOT)/usr/bin/gba; \
-	 ln -s $(DESTDIR)$(bindir)/gba $(ROOT)/usr/bin/gba; \
-	 rm -f $(ROOT)/usr/bin/gbi; \
-	 ln -s $(DESTDIR)$(bindir)/gbi $(ROOT)/usr/bin/gbi; \
+	 mkdir -p $(DESTDIR)/usr/bin; \
+	 ln -s $(bindir)/gbx $(DESTDIR)/usr/bin/gbx; \
+	 ln -s $(bindir)/gbc $(DESTDIR)/usr/bin/gbc; \
+	 ln -s $(bindir)/gba $(DESTDIR)/usr/bin/gba; \
+	 ln -s $(bindir)/gbi $(DESTDIR)/usr/bin/gbi; \
 	fi
 
 	@echo
@@ -729,8 +726,7 @@
 	@(cd $(srcdir)/app; d=`pwd`; for p in *; do echo "Compiling $$p..."; cd $$d/$$p; \
 	  $(DESTDIR)$(bindir)/gbc -ag; $(DESTDIR)$(bindir)/gba; rm -rf .gambas; $(INSTALL) $$p $(DESTDIR)$(bindir); done)
 	@if test "$(bindir)" != "$(ROOT)/usr/bin" && test "$(bindir)" != "$(ROOT)/usr/bin/"; then \
-	 rm -f $(ROOT)/usr/bin/gambas; \
-	 ln -s $(DESTDIR)$(bindir)/gambas $(ROOT)/usr/bin/gambas; \
+	 ln -s $(bindir)/gambas $(DESTDIR)/usr/bin/gambas; \
         fi
 
 	@echo
