--- ./gcc/Makefile.in.orig	2007-10-31 11:32:06.000000000 -0500
+++ ./gcc/Makefile.in	2008-02-21 19:02:11.000000000 -0600
@@ -3199,9 +3199,9 @@
 	cp $(UNWIND_H) include/unwind.h
 	chmod a+r include/limits.h
 # Install the README
-	rm -f include/README
-	cp $(srcdir)/../fixincludes/README-fixinc include/README
-	chmod a+r include/README
+#	rm -f include/README
+#	cp $(srcdir)/../fixincludes/README-fixinc include/README
+#	chmod a+r include/README
 	$(STAMP) $@
 
 .PHONY: install-gcc-tooldir
@@ -3253,10 +3253,7 @@
 	else \
 	  (TARGET_MACHINE='$(target)'; srcdir=`cd $(srcdir); ${PWD_COMMAND}`; \
 	    SHELL='$(SHELL)'; MACRO_LIST=`${PWD_COMMAND}`/macro_list ; \
-	    export TARGET_MACHINE srcdir SHELL MACRO_LIST && \
-	    cd $(build_objdir)/fixincludes && \
-	    $(SHELL) ./fixinc.sh ../../gcc/include \
-	      $(SYSTEM_HEADER_DIR) $(OTHER_FIXINCLUDES_DIRS) ); \
+	    export TARGET_MACHINE srcdir SHELL MACRO_LIST ); \
 	  rm -f include/syslimits.h; \
 	  if [ -f include/limits.h ]; then \
 	    mv include/limits.h include/syslimits.h; \
