--- geda-gaf-1.8.1/gschem/lib/system-gschemrc.scm.orig	2012-11-23 21:09:39.000000000 +0000
+++ geda-gaf-1.8.1/gschem/lib/system-gschemrc.scm	2013-05-04 16:52:44.914877940 +0100
@@ -529,7 +529,7 @@
 ; the proper aspect ratio.  All units are in inches. (use output-orientation
 ; to get portrait mode)
 ;
-(paper-size 11.0 8.5) ; letter
+;(paper-size 11.0 8.5) ; letter
 ;(paper-size 14.0 8.5) ; legal 
 ;(paper-size 8.5 5.5) ; statement
 ;(paper-size 17.0 11.0) ; tabloid
@@ -547,7 +547,7 @@
 ;(paper-size 33.11 23.39) ; A1
 ;(paper-size 23.39 16.54) ; A2
 ;(paper-size 16.54 11.69) ; A3
-;(paper-size 11.69 8.27) ; A4
+(paper-size 11.69 8.27) ; A4
 ;(paper-size 8.27 5.83) ; A5
 ;(paper-size 5.83 4.13) ; A6
 ;(paper-size 4.13 2.91) ; A7 
@@ -569,6 +569,19 @@
 ; You MUST change this list in this file (gschem-systemrc) if you want
 ; any reordering changes to take effect.
 ;
+(paper-sizes "A0  84.10 cm x 118.90 cm" 46.81 33.11)
+(paper-sizes "A1  59.40 cm x 84.10 cm" 33.11 23.39)
+(paper-sizes "A2  42.00 cm x 59.40 cm" 23.39 16.54)
+(paper-sizes "A3  29.70 cm x 42.00 cm" 16.54 11.69)
+(paper-sizes "A4  21.00 cm x 29.70 cm" 11.69 8.27)
+(paper-sizes "A5  14.80 cm x 21.00 cm" 8.27 5.83)
+(paper-sizes "A6  10.50 cm x 14.80 cm" 5.83 4.13)
+(paper-sizes "A7  7.40 cm x 10.50 cm" 4.13 2.91)
+(paper-sizes "A8  5.20 cm x 7.40 cm" 2.91 2.05)
+(paper-sizes "A9  3.70 cm x 5.20 cm" 2.05 1.46)
+(paper-sizes "A10 2.60 cm x 3.70 cm" 1.46 1.02)
+(paper-sizes "A11 1.80 cm x 2.60 cm" 1.02 0.71)
+(paper-sizes "A12 1.30 cm x 1.80 cm" 0.71 0.51)
 (paper-sizes "Letter : 8.5 in x 11 in" 11.0 8.5)
 (paper-sizes "Legal : 8.5 in x 14 in" 14.0 8.5)
 (paper-sizes "Statement : 5.5 in x 8.5 in" 8.5 5.5)
@@ -583,19 +596,6 @@
 (paper-sizes "C : 17 in x 22 in" 22.0 17.0)
 (paper-sizes "D : 22 in x 34 in" 34.0 22.0)
 (paper-sizes "E : 34 in x 44 in" 44.0 34.0)
-(paper-sizes "A0  84.10 cm x 118.90 cm" 46.81 33.11)
-(paper-sizes "A1  59.40 cm x 84.10 cm" 33.11 23.39)
-(paper-sizes "A2  42.00 cm x 59.40 cm" 23.39 16.54)
-(paper-sizes "A3  29.70 cm x 42.00 cm" 16.54 11.69)
-(paper-sizes "A4  21.00 cm x 29.70 cm" 11.69 8.27)
-(paper-sizes "A5  14.80 cm x 21.00 cm" 8.27 5.83)
-(paper-sizes "A6  10.50 cm x 14.80 cm" 5.83 4.13)
-(paper-sizes "A7  7.40 cm x 10.50 cm" 4.13 2.91)
-(paper-sizes "A8  5.20 cm x 7.40 cm" 2.91 2.05)
-(paper-sizes "A9  3.70 cm x 5.20 cm" 2.05 1.46)
-(paper-sizes "A10 2.60 cm x 3.70 cm" 1.46 1.02)
-(paper-sizes "A11 1.80 cm x 2.60 cm" 1.02 0.71)
-(paper-sizes "A12 1.30 cm x 1.80 cm" 0.71 0.51)
 
 ; print-command string 
 ;
@@ -941,7 +941,7 @@
 ;   (define default-titleblock '())
 ;   (define default-titleblock #f)
 ;
-(define default-titleblock "title-B.sym")
+(define default-titleblock "title-A3.sym")
 
 ; Load the regular expressions module
 (if (provided? 'regex)
