--- geda-gaf-1.10.0/configure.ac.orig	2020-02-10 22:03:32.991863107 +0000
+++ geda-gaf-1.10.0/configure.ac	2020-02-10 22:03:41.921862681 +0000
@@ -76,7 +76,7 @@
 
 PKG_PROG_PKG_CONFIG
 
-AX_CHECK_GUILE([2.0.0])
+AX_CHECK_GUILE([2.2.0])
 
 PKG_CHECK_MODULES(GLIB, [glib-2.0 >= 2.20.0], ,
   AC_MSG_ERROR([GLib 2.20.0 or later is required.]))
--- geda-gaf-1.10.0/xorn/configure.ac.orig	2019-10-08 19:16:33.000000000 +0100
+++ geda-gaf-1.10.0/xorn/configure.ac	2020-02-10 22:26:14.744798134 +0000
@@ -30,7 +30,7 @@
 
 AM_PATH_PYTHON([2.7])
 PKG_CHECK_MODULES([PYTHON], [python2 >= 2.7])
-PKG_CHECK_MODULES([GUILE], [guile-2.0 >= 2.0.10])
+PKG_CHECK_MODULES([GUILE], [guile-2.2 >= 2.2.0])
 
 AM_GNU_GETTEXT([external])
 AM_GNU_GETTEXT_VERSION([0.18.2])
