--- geeqie-20190712/configure.ac.orig	2019-07-12 20:03:18.000000000 +0100
+++ geeqie-20190712/configure.ac	2019-07-12 20:07:27.943423492 +0100
@@ -251,7 +251,7 @@
 AC_DEFINE_UNQUOTED([GQ_APP_DIR], "$appdir", [Location of application data])
 AC_SUBST(appdir)
 
-eval "eval gq_bindir=${prefix}/lib/${PACKAGE}"
+eval "eval gq_bindir=${libdir}/${PACKAGE}"
 AC_DEFINE_UNQUOTED([GQ_BIN_DIR], "$gq_bindir", [Location of helper scripts and executables])
 AC_SUBST(gq_bindir)
 
