--- geogram-1.7.6/CMakeLists.txt.orig	2020-09-04 18:38:51.000000000 +0100
+++ geogram-1.7.6/CMakeLists.txt	2020-11-15 13:54:11.329301400 +0000
@@ -87,10 +87,10 @@
 
 if(VORPALINE_BUILD_DYNAMIC)
    set(CMAKE_MACOSX_RPATH 1)
-   set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+   set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
 endif()
 
-set(CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib")
+set(CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
 
 # uninstall target
 # Needs to be created before add_subdirectory() because GLFw has
@@ -146,7 +146,7 @@
 
 # FindGeogram.cmake
 
-install(FILES cmake/FindGeogram.cmake DESTINATION lib/cmake/modules COMPONENT devkit)
+install(FILES cmake/FindGeogram.cmake DESTINATION lib${LIB_SUFFIX}/cmake/modules COMPONENT devkit)
 
 # Configure CPack
 
--- geogram-1.7.6/cmake/utilities.cmake.orig	2020-09-04 18:38:51.000000000 +0100
+++ geogram-1.7.6/cmake/utilities.cmake	2020-11-15 13:56:40.395294288 +0000
@@ -241,8 +241,8 @@
         TARGETS ${ARGN}
         COMPONENT runtime
         RUNTIME DESTINATION bin
-        LIBRARY DESTINATION lib
-        ARCHIVE DESTINATION lib
+        LIBRARY DESTINATION lib${LIB_SUFFIX}
+        ARCHIVE DESTINATION lib${LIB_SUFFIX}
     )
 endfunction()
 
@@ -270,8 +270,8 @@
             TARGETS ${ARGN}
             COMPONENT ${component}
             RUNTIME DESTINATION bin
-            LIBRARY DESTINATION lib
-            ARCHIVE DESTINATION lib
+            LIBRARY DESTINATION lib${LIB_SUFFIX}
+            ARCHIVE DESTINATION lib${LIB_SUFFIX}
         )
     endforeach()
 endfunction()
