--- a/src/bin/geomutil/math2oogl/math2oogl.c
+++ b/src/bin/geomutil/math2oogl/math2oogl.c
@@ -345,7 +345,7 @@ int main(int ac,char **av)
     if (!lastline)
       lastline = lines = malloc(sizeof(struct line));
     else {
-      if (toss) lastline = prev;
+      if (toss && prev) lastline = prev;
       toss = 0;
       prev = lastline;
       lastline = lastline->next = malloc (sizeof(struct line));
