Description: Fix the build with 64-bit time_t on 32-bit
Author: Adrian Bunk <bunk@debian.org>
Bug-Debian: https://bugs.debian.org/1088201

--- a/src/bin/geomutil/ucd/anytoucd.c
+++ b/src/bin/geomutil/ucd/anytoucd.c
@@ -372,9 +372,8 @@
 	}
     {
     char *timestring;
-    long mytime;
     time_t myt;
-    myt = time(&mytime);
+    myt = time(NULL);
     timestring = ctime(&myt);
     fprintf(fp,"#  Created by anytoucd on %s \n",timestring);
     }
