--- a/src/bin/geomview/x11/Makefile.am
+++ b/src/bin/geomview/x11/Makefile.am
@@ -2,7 +2,7 @@ INCLUDES = $(default_includes) $(MOTIFIN
 SUBDIRS = bitmaps interface
 
 gvx_LDADD = ../common/libgvcommon.a buildinfo.o \
- $(MOTIFUILIB) $(MOTIFLIBS) $(XMULIBS) $(OOGLLIB) $(XLIBS)
+ $(MOTIFUILIB) $(MOTIFLIBS) $(XMULIBS) $(OOGLLIB) $(XLIBS) $(OPENGLLIBS)
 gvx_DEPENDENCIES = ../common/libgvcommon.a buildinfo.o $(OOGLLIB)
 #gvx_LDFLAGS = -rpath $(libdir)
 
