--- gerbv-2.6.1/man/Makefile.am.orig	2013-07-22 20:09:16.000000000 +0100
+++ gerbv-2.6.1/man/Makefile.am	2013-12-08 12:59:51.135008999 +0000
@@ -19,8 +19,7 @@
 ##  along with this program; if not, write to the Free Software
 ##  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111 USA
 
-man_MANS = gerbv.1 gerbv.ru.1
-PO_FILES= gerbv.ru.1.in.po
+man_MANS = gerbv.1
 
 MOSTLYCLEANFILES = *~
 CLEANFILES = *~  *.pot $(man_MANS)
@@ -37,13 +36,6 @@
 .in.po.in:
 	po4a-translate -k 70 -M UTF-8 -f man -m gerbv.1.in -p $< -l $@
 
-$(PO_FILES): gerbv.1.in.pot
-	msgmerge --width=76 $@ $< -o $@.new
-	mv $@.new $@
-
 gerbv.1.in.pot: gerbv.1.in
 	po4a-gettextize -M UTF-8 -f man -m $< -p $@
 
-update-po:
-	@touch gerbv.1.in.pot
-	make $(PO_FILES)
--- gerbv-2.6.1/po/LINGUAS.orig	2013-12-08 13:00:02.193008472 +0000
+++ gerbv-2.6.1/po/LINGUAS	2013-12-08 12:56:34.274018392 +0000
@@ -1 +0,0 @@
-ru
