--- gerbv-2.7.0/src/callbacks.c.orig	2023-07-16 11:43:14.978832989 +0100
+++ gerbv-2.7.0/src/callbacks.c	2023-07-16 11:42:42.990834515 +0100
@@ -110,6 +110,8 @@
 
 static void update_selected_object_message (gboolean userTriedToSelect);
 
+enum CALLBACKS CALLBACKS_SAVE_FILE_TYPE;
+enum TOGGLE_LAYER toggle_layer;
 
 gchar *utf8_strncpy(gchar *dst, const gchar *src, gsize byte_len)
 {
--- gerbv-2.7.0/src/callbacks.h.orig	2019-01-22 09:01:00.000000000 +0000
+++ gerbv-2.7.0/src/callbacks.h	2023-07-16 11:42:27.179835269 +0100
@@ -26,7 +26,7 @@
     \ingroup gerbv
 */
 
-enum {
+enum CALLBACKS {
 	CALLBACKS_SAVE_PROJECT_AS,
 	CALLBACKS_SAVE_FILE_PS,
 	CALLBACKS_SAVE_FILE_PDF,
@@ -41,13 +41,17 @@
 	CALLBACKS_SAVE_LAYER_AS,
 	CALLBACKS_SAVE_FILE_IDRILL
 	
-} CALLBACKS_SAVE_FILE_TYPE;
+};
 
-enum {
+extern enum CALLBACKS CALLBACKS_SAVE_FILE_TYPE;
+
+enum TOGGLE_LAYER {
 	LAYER_SELECTED =	-1,
 	LAYER_ALL_ON =		-2,
 	LAYER_ALL_OFF =		-3,
-} toggle_layer;
+};
+
+extern enum TOGGLE_LAYER toggle_layer;
 
 void
 callbacks_new_project_activate                (GtkMenuItem     *menuitem,
