--- get_iplayer-20100703/get_iplayer.orig	2010-07-03 09:37:58.000000000 +0100
+++ get_iplayer-20100703/get_iplayer	2010-09-23 20:33:11.000000000 +0100
@@ -5847,7 +5847,7 @@
 				my $url = ${media_stream_live_prefix}."?server=$cattribs->{server}&identifier=$cattribs->{identifier}&kind=$cattribs->{kind}&application=$cattribs->{application}";
 				my $xml = main::request_url_retry( main::create_ua( 'desktop' ), $url, 3, undef, undef, 1 );
 				main::logger "\n$xml\n" if $opt->{debug};
-				$cattribs->{authString} = 'auth='.$1 if $xml =~ m{<token>(.+?)</token>};
+				$cattribs->{authString} = $1 if $xml =~ m{<token>(.+?)</token>};
 				$conn->{authstring} = $cattribs->{authString};
 			}
 
