--- gftp-2.0.19/lib/misc.c.orig	2008-03-28 11:44:39.000000000 +0000
+++ gftp-2.0.19/lib/misc.c	2009-02-03 22:46:58.000000000 +0000
@@ -63,7 +63,7 @@
       return (dest_str);
     }
 
-  len += len / 3;
+  len += len / 3 + 1;
 
   if (dest_str != NULL && len > dest_len)
     {
--- gftp-2.0.19/acinclude.m4.orig	2007-01-10 01:24:59.000000000 +0000
+++ gftp-2.0.19/acinclude.m4	2009-02-04 19:28:42.000000000 +0000
@@ -27,9 +27,9 @@
                #include <string.h>
 
                int main(void) {
-                 char buf[20];
+                 char buf[22];
 #if defined (_LARGEFILE_SOURCE)
-                 sprintf (buf, "%'lld", (long) 1);
+                 sprintf (buf, "%'lld", (long long) 1);
 #else
                  sprintf (buf, "%'ld", (long) 1);
 #endif
