--- ./Resource/Init/gs_res.ps.orig	2007-02-20 17:54:19.000000000 -0600
+++ ./Resource/Init/gs_res.ps	2008-03-03 18:32:55.000000000 -0600
@@ -317,9 +317,9 @@
 } if
 pssystemparams begin
   .default_resource_dir
-  /FontResourceDir (Font) .resource_dir_name
+  /FontResourceDir (/usr/share/ghostscript/8.63/Resource/CIDFont/) .resource_dir_name
      readonly .forcedef	% pssys'params is r-o
-  /GenericResourceDir () .resource_dir_name
+  /GenericResourceDir (/usr/share/ghostscript/8.63/Resource/) .resource_dir_name
      readonly .forcedef	% pssys'params is r-o
   pop % .default_resource_dir
   /GenericResourcePathSep
