--- gimagereader-3.3.0/qt/src/hocr/HOCRTextExporter.cc.orig	2018-09-26 22:06:06.000000000 +0100
+++ gimagereader-3.3.0/qt/src/hocr/HOCRTextExporter.cc	2020-01-19 22:27:26.713089090 +0000
@@ -28,6 +28,7 @@
 #include <QFileInfo>
 #include <QMessageBox>
 #include <QTextStream>
+#include <QUrl>
 
 bool HOCRTextExporter::run(const HOCRDocument* hocrdocument, QString& filebasename) {
 	QString suggestion = filebasename;
--- gimagereader-3.3.0/qt/src/hocr/HOCROdtExporter.cc.orig	2018-09-26 22:06:06.000000000 +0100
+++ gimagereader-3.3.0/qt/src/hocr/HOCROdtExporter.cc	2020-01-19 22:26:52.094090742 +0000
@@ -30,6 +30,7 @@
 #include <QFileInfo>
 #include <QMessageBox>
 #include <QUuid>
+#include <QUrl>
 #include <QXmlStreamWriter>
 #include <quazipfile.h>
 
