--- gimagereader-3.3.1/CMakeLists.txt.orig	2019-07-28 21:59:40.000000000 +0100
+++ gimagereader-3.3.1/CMakeLists.txt	2020-10-03 21:43:24.374184449 +0100
@@ -1,4 +1,5 @@
 CMAKE_MINIMUM_REQUIRED(VERSION 2.8.11)
+cmake_policy(SET CMP0100 NEW)
 LIST(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 INCLUDE(GSettings)
 SET(ENV{LC_ALL} "C")
--- gimagereader-3.3.0/qt/src/hocr/HOCRTextExporter.cc.orig	2018-09-26 22:06:06.000000000 +0100
+++ gimagereader-3.3.0/qt/src/hocr/HOCRTextExporter.cc	2020-01-19 22:27:26.713089090 +0000
@@ -28,6 +28,7 @@
 #include <QFileInfo>
 #include <QMessageBox>
 #include <QTextStream>
+#include <QUrl>
 
 bool HOCRTextExporter::run(const HOCRDocument* hocrdocument, QString& filebasename) {
 	QString suggestion = filebasename;
--- gimagereader-3.3.0/qt/src/hocr/HOCROdtExporter.cc.orig	2018-09-26 22:06:06.000000000 +0100
+++ gimagereader-3.3.0/qt/src/hocr/HOCROdtExporter.cc	2020-01-19 22:26:52.094090742 +0000
@@ -30,6 +30,7 @@
 #include <QFileInfo>
 #include <QMessageBox>
 #include <QUuid>
+#include <QUrl>
 #include <QXmlStreamWriter>
 #include <quazipfile.h>
 
--- gimagereader-3.3.1/CMakeLists.txt.orig	2020-10-03 21:59:15.172085004 +0100
+++ gimagereader-3.3.1/CMakeLists.txt	2020-11-04 19:30:19.077204097 +0000
@@ -177,8 +177,11 @@
     FIND_PACKAGE(Qt5Widgets REQUIRED)
     FIND_PACKAGE(Qt5Xml REQUIRED)
     FIND_PACKAGE(Qt5PrintSupport REQUIRED)
-    FIND_PACKAGE(QuaZip5)
-    IF(NOT QUAZIP_FOUND)
+    FIND_PACKAGE(QuaZip-Qt5)
+    IF(QuaZip-Qt5_FOUND)
+        set(QUAZIP_INCLUDE_DIR QuaZip::QuaZip)
+        set(QUAZIP_LIBRARIES QuaZip::QuaZip)
+    ELSE()
         FIND_PATH(QUAZIP_INCLUDE_DIR NAME quazipfile.h PATH_SUFFIXES quazip5)
         FIND_LIBRARY(QUAZIP_LIBRARIES quazip5)
         IF(NOT QUAZIP_INCLUDE_DIR OR NOT QUAZIP_LIBRARIES)
