--- gimp-gap-2.6.0/extern_libs/Makefile.in.orig	2009-06-04 21:40:59.000000000 +0100
+++ gimp-gap-2.6.0/extern_libs/Makefile.in	2010-01-01 11:36:12.000000000 +0000
@@ -458,9 +458,10 @@
 @BUILD_LIBMPEG3_LIB_TRUE@	$(EXGMAKE)
 
 install-data-hook:
-	rm -f $(GAPLIBDIR)/libavformat.a \
-	rm -f $(GAPLIBDIR)/libavcodec.a \
-	rm -f $(GAPLIBDIR)/libmpeg3.a
+	rm -f $(DESTDIR)$(GAPLIBDIR)/libavutil.a \
+	rm -f $(DESTDIR)$(GAPLIBDIR)/libavformat.a \
+	rm -f $(DESTDIR)$(GAPLIBDIR)/libavcodec.a \
+	rm -f $(DESTDIR)$(GAPLIBDIR)/libmpeg3.a
 
 # eof (without this line Makefile generation produces syntax error)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
