--- gimp-2.4.0-rc1/plug-ins/print/print-page-layout.c.orig	2007-08-15 22:57:03.000000000 +0100
+++ gimp-2.4.0-rc1/plug-ins/print/print-page-layout.c	2007-08-23 11:25:28.000000000 +0100
@@ -547,6 +547,13 @@
 
   setup = gtk_print_operation_get_default_page_setup (info->data->operation);
 
+  if (setup == NULL)
+    {
+      *page_width = 0.0;
+      *page_height = 0.0;
+      return;
+    }
+
   if (info->data->use_full_page)
     {
       *page_width = gtk_page_setup_get_paper_width (setup, unit);
