--- gimp-2.10.38/plug-ins/metadata/metadata-editor.c.orig	2024-05-03 01:33:35.000000000 +0100
+++ gimp-2.10.38/plug-ins/metadata/metadata-editor.c	2024-09-16 19:58:00.058709060 +0100
@@ -2140,7 +2140,7 @@
                             }
                           else
                             {
-                              if (! g_strv_contains (values, equiv_values[evi]))
+                              if (! g_strv_contains ((const gchar * const *) values, equiv_values[evi]))
                                 {
                                   gchar *tmpvalue;
 
--- gimp-2.10.38/plug-ins/file-tiff/file-tiff-load.c.orig	2024-05-03 01:33:35.000000000 +0100
+++ gimp-2.10.38/plug-ins/file-tiff/file-tiff-load.c	2024-09-16 19:55:39.005699590 +0100
@@ -1301,8 +1301,8 @@
 
       /* any resolution info in the file? */
       {
-        gfloat   xres = 72.0;
-        gfloat   yres = 72.0;
+        gdouble  xres = 72.0;
+        gdouble  yres = 72.0;
         gushort  read_unit;
         GimpUnit unit = GIMP_UNIT_PIXEL; /* invalid unit */
 
--- gimp-2.10.38/libgimpconfig/gimpconfig-serialize.c.orig	2024-05-03 01:33:35.000000000 +0100
+++ gimp-2.10.38/libgimpconfig/gimpconfig-serialize.c	2025-08-16 17:38:58.037359264 +0100
@@ -367,10 +367,10 @@
 {
   if (G_VALUE_HOLDS_BOOLEAN (value))
     {
-      gboolean bool;
+      gboolean v_bool;
 
-      bool = g_value_get_boolean (value);
-      g_string_append (str, bool ? "yes" : "no");
+      v_bool = g_value_get_boolean (value);
+      g_string_append (str, v_bool ? "yes" : "no");
       return TRUE;
     }
 
