--- gimp-sharp-0.11/plug-ins/AverageBlur/Makefile.in.orig	2007-01-05 18:59:56.000000000 +0000
+++ gimp-sharp-0.11/plug-ins/AverageBlur/Makefile.in	2007-01-09 14:35:52.000000000 +0000
@@ -505,8 +505,9 @@
 	$(INSTALL) AverageBlur $(DESTDIR)$(GIMPPLUGINDIR)
 	@for lang in `ls po | grep .mo`;\
 		do localedir=`echo $$lang | sed -e's/.mo//'`;\
-		echo "Copying po/$$lang to $(GIMPLOCALE)/$$localedir/LC_MESSAGES/AverageBlur.mo";\
-		$(INSTALL_DATA) po/$$lang $(GIMPLOCALE)/$$localedir/LC_MESSAGES/AverageBlur.mo;\
+		echo "Copying po/$$lang to $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/AverageBlur.mo";\
+		$(INSTALL) -d $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES;\
+		$(INSTALL_DATA) po/$$lang $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/AverageBlur.mo;\
 		done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- gimp-sharp-0.11/plug-ins/Forge/Makefile.in.orig	2007-01-05 18:59:58.000000000 +0000
+++ gimp-sharp-0.11/plug-ins/Forge/Makefile.in	2007-01-09 14:39:13.000000000 +0000
@@ -505,8 +505,9 @@
 	$(INSTALL) Forge $(DESTDIR)$(GIMPPLUGINDIR)
 	@for lang in `ls po | grep .mo`;\
 		do localedir=`echo $$lang | sed -e's/.mo//'`;\
-		echo "Copying po/$$lang to $(GIMPLOCALE)/$$localedir/LC_MESSAGES/Forge.mo";\
-		$(INSTALL_DATA) po/$$lang $(GIMPLOCALE)/$$localedir/LC_MESSAGES/Forge.mo;\
+		echo "Copying po/$$lang to $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/Forge.mo";\
+		$(INSTALL) -d $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES;\
+		$(INSTALL_DATA) po/$$lang $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/Forge.mo;\
 		done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- gimp-sharp-0.11/plug-ins/Fragment/Makefile.in.orig	2007-01-05 18:59:58.000000000 +0000
+++ gimp-sharp-0.11/plug-ins/Fragment/Makefile.in	2007-01-09 14:40:30.000000000 +0000
@@ -505,8 +505,9 @@
 	$(INSTALL) Fragment $(DESTDIR)$(GIMPPLUGINDIR)
 	@for lang in `ls po | grep .mo`;\
 		do localedir=`echo $$lang | sed -e's/.mo//'`;\
-		echo "Copying po/$$lang to $(GIMPLOCALE)/$$localedir/LC_MESSAGES/Fragment.mo";\
-		$(INSTALL_DATA) po/$$lang $(GIMPLOCALE)/$$localedir/LC_MESSAGES/Fragment.mo;\
+		echo "Copying po/$$lang to $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/Fragment.mo";\
+		$(INSTALL) -d $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES;\
+		$(INSTALL_DATA) po/$$lang $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/Fragment.mo;\
 		done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- gimp-sharp-0.11/plug-ins/KoalaPaint/Makefile.in.orig	2007-01-05 18:59:59.000000000 +0000
+++ gimp-sharp-0.11/plug-ins/KoalaPaint/Makefile.in	2007-01-09 14:41:23.000000000 +0000
@@ -504,8 +504,9 @@
 	$(INSTALL) KoalaPaint $(DESTDIR)$(GIMPPLUGINDIR)
 	@for lang in `ls po | grep .mo`;\
 		do localedir=`echo $$lang | sed -e's/.mo//'`;\
-		echo "Copying po/$$lang to $(GIMPLOCALE)/$$localedir/LC_MESSAGES/KoalaPaint.mo";\
-		$(INSTALL_DATA) po/$$lang $(GIMPLOCALE)/$$localedir/LC_MESSAGES/KoalaPaint.mo;\
+		echo "Copying po/$$lang to $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/KoalaPaint.mo";\
+		$(INSTALL) -d $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/;\
+		$(INSTALL_DATA) po/$$lang $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/KoalaPaint.mo;\
 		done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- gimp-sharp-0.11/plug-ins/Ministeck/Makefile.in.orig	2007-01-05 18:59:59.000000000 +0000
+++ gimp-sharp-0.11/plug-ins/Ministeck/Makefile.in	2007-01-09 14:43:37.000000000 +0000
@@ -518,8 +518,8 @@
 	$(INSTALL) Ministeck $(DESTDIR)$(GIMPPLUGINDIR)
 	@for lang in `ls po | grep .mo`;\
 		do localedir=`echo $$lang | sed -e's/.mo//'`;\
-		echo "Copying po/$$lang to $(GIMPLOCALE)/$$localedir/LC_MESSAGES/Ministeck.mo";\
-		$(INSTALL_DATA) po/$$lang $(GIMPLOCALE)/$$localedir/LC_MESSAGES/Ministeck.mo;\
+		echo "Copying po/$$lang to $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/Ministeck.mo";\
+		$(INSTALL_DATA) po/$$lang $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/Ministeck.mo;\
 		done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- gimp-sharp-0.11/plug-ins/ncp/Makefile.in.orig	2007-01-05 19:00:05.000000000 +0000
+++ gimp-sharp-0.11/plug-ins/ncp/Makefile.in	2007-01-09 14:43:52.000000000 +0000
@@ -505,8 +505,8 @@
 	$(INSTALL) ncp $(DESTDIR)$(GIMPPLUGINDIR)
 	@for lang in `ls po | grep .mo`;\
 		do localedir=`echo $$lang | sed -e's/.mo//'`;\
-		echo "Copying po/$$lang to $(GIMPLOCALE)/$$localedir/LC_MESSAGES/ncp.mo";\
-		$(INSTALL_DATA) po/$$lang $(GIMPLOCALE)/$$localedir/LC_MESSAGES/ncp.mo;\
+		echo "Copying po/$$lang to $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/ncp.mo";\
+		$(INSTALL_DATA) po/$$lang $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/ncp.mo;\
 		done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- gimp-sharp-0.11/plug-ins/neo/Makefile.in.orig	2007-01-05 19:00:05.000000000 +0000
+++ gimp-sharp-0.11/plug-ins/neo/Makefile.in	2007-01-09 14:43:57.000000000 +0000
@@ -505,8 +505,8 @@
 	$(INSTALL) neo $(DESTDIR)$(GIMPPLUGINDIR)
 	@for lang in `ls po | grep .mo`;\
 		do localedir=`echo $$lang | sed -e's/.mo//'`;\
-		echo "Copying po/$$lang to $(GIMPLOCALE)/$$localedir/LC_MESSAGES/neo.mo";\
-		$(INSTALL_DATA) po/$$lang $(GIMPLOCALE)/$$localedir/LC_MESSAGES/neo.mo;\
+		echo "Copying po/$$lang to $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/neo.mo";\
+		$(INSTALL_DATA) po/$$lang $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/neo.mo;\
 		done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- gimp-sharp-0.11/plug-ins/PicturePackage/Makefile.in.orig	2007-01-05 19:00:01.000000000 +0000
+++ gimp-sharp-0.11/plug-ins/PicturePackage/Makefile.in	2007-01-09 14:44:13.000000000 +0000
@@ -529,8 +529,8 @@
 	$(INSTALL) PicturePackage $(DESTDIR)$(GIMPPLUGINDIR)
 	@for lang in `ls po | grep .mo`;\
 		do localedir=`echo $$lang | sed -e's/.mo//'`;\
-		echo "Copying po/$$lang to $(GIMPLOCALE)/$$localedir/LC_MESSAGES/PicturePackage.mo";\
-		$(INSTALL_DATA) po/$$lang $(GIMPLOCALE)/$$localedir/LC_MESSAGES/PicturePackage.mo;\
+		echo "Copying po/$$lang to $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/PicturePackage.mo";\
+		$(INSTALL_DATA) po/$$lang $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/PicturePackage.mo;\
 		done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- gimp-sharp-0.11/plug-ins/Pointillize/Makefile.in.orig	2007-01-05 19:00:01.000000000 +0000
+++ gimp-sharp-0.11/plug-ins/Pointillize/Makefile.in	2007-01-09 14:44:19.000000000 +0000
@@ -510,8 +510,8 @@
 	$(INSTALL) Pointillize $(DESTDIR)$(GIMPPLUGINDIR)
 	@for lang in `ls po | grep .mo`;\
 		do localedir=`echo $$lang | sed -e's/.mo//'`;\
-		echo "Copying po/$$lang to $(GIMPLOCALE)/$$localedir/LC_MESSAGES/Pointillize.mo";\
-		$(INSTALL_DATA) po/$$lang $(GIMPLOCALE)/$$localedir/LC_MESSAGES/Pointillize.mo;\
+		echo "Copying po/$$lang to $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/Pointillize.mo";\
+		$(INSTALL_DATA) po/$$lang $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/Pointillize.mo;\
 		done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- gimp-sharp-0.11/plug-ins/Raindrops/Makefile.in.orig	2007-01-06 10:15:37.000000000 +0000
+++ gimp-sharp-0.11/plug-ins/Raindrops/Makefile.in	2007-01-09 14:44:25.000000000 +0000
@@ -508,8 +508,8 @@
 	$(INSTALL) Raindrops $(DESTDIR)$(GIMPPLUGINDIR)
 	@for lang in `ls po | grep .mo`;\
 		do localedir=`echo $$lang | sed -e's/.mo//'`;\
-		echo "Copying po/$$lang to $(GIMPLOCALE)/$$localedir/LC_MESSAGES/Raindrops.mo";\
-		$(INSTALL_DATA) po/$$lang $(GIMPLOCALE)/$$localedir/LC_MESSAGES/Raindrops.mo;\
+		echo "Copying po/$$lang to $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/Raindrops.mo";\
+		$(INSTALL_DATA) po/$$lang $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/Raindrops.mo;\
 		done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- gimp-sharp-0.11/plug-ins/SliceTool/Makefile.in.orig	2007-01-05 19:00:02.000000000 +0000
+++ gimp-sharp-0.11/plug-ins/SliceTool/Makefile.in	2007-01-09 14:44:46.000000000 +0000
@@ -548,8 +548,8 @@
 	$(INSTALL) SliceTool $(DESTDIR)$(GIMPPLUGINDIR)
 	@for lang in `ls po | grep .mo`;\
 		do localedir=`echo $$lang | sed -e's/.mo//'`;\
-		echo "Copying po/$$lang to $(GIMPLOCALE)/$$localedir/LC_MESSAGES/SliceTool.mo";\
-		$(INSTALL_DATA) po/$$lang $(GIMPLOCALE)/$$localedir/LC_MESSAGES/SliceTool.mo;\
+		echo "Copying po/$$lang to $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/SliceTool.mo";\
+		$(INSTALL_DATA) po/$$lang $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/SliceTool.mo;\
 		done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- gimp-sharp-0.11/plug-ins/Splitter/Makefile.in.orig	2007-01-05 19:00:03.000000000 +0000
+++ gimp-sharp-0.11/plug-ins/Splitter/Makefile.in	2007-01-09 14:44:49.000000000 +0000
@@ -510,8 +510,8 @@
 	$(INSTALL) Splitter $(DESTDIR)$(GIMPPLUGINDIR)
 	@for lang in `ls po | grep .mo`;\
 		do localedir=`echo $$lang | sed -e's/.mo//'`;\
-		echo "Copying po/$$lang to $(GIMPLOCALE)/$$localedir/LC_MESSAGES/Splitter.mo";\
-		$(INSTALL_DATA) po/$$lang $(GIMPLOCALE)/$$localedir/LC_MESSAGES/Splitter.mo;\
+		echo "Copying po/$$lang to $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/Splitter.mo";\
+		$(INSTALL_DATA) po/$$lang $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/Splitter.mo;\
 		done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- gimp-sharp-0.11/plug-ins/Swirlies/Makefile.in.orig	2007-01-05 19:00:03.000000000 +0000
+++ gimp-sharp-0.11/plug-ins/Swirlies/Makefile.in	2007-01-09 14:44:52.000000000 +0000
@@ -505,8 +505,8 @@
 	$(INSTALL) Swirlies $(DESTDIR)$(GIMPPLUGINDIR)
 	@for lang in `ls po | grep .mo`;\
 		do localedir=`echo $$lang | sed -e's/.mo//'`;\
-		echo "Copying po/$$lang to $(GIMPLOCALE)/$$localedir/LC_MESSAGES/Swirlies.mo";\
-		$(INSTALL_DATA) po/$$lang $(GIMPLOCALE)/$$localedir/LC_MESSAGES/Swirlies.mo;\
+		echo "Copying po/$$lang to $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/Swirlies.mo";\
+		$(INSTALL_DATA) po/$$lang $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/Swirlies.mo;\
 		done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- gimp-sharp-0.11/plug-ins/UpdateCheck/Makefile.in.orig	2007-01-05 19:00:04.000000000 +0000
+++ gimp-sharp-0.11/plug-ins/UpdateCheck/Makefile.in	2007-01-09 14:44:57.000000000 +0000
@@ -508,8 +508,8 @@
 	$(INSTALL) UpdateCheck $(DESTDIR)$(GIMPPLUGINDIR)
 	@for lang in `ls po | grep .mo`;\
 		do localedir=`echo $$lang | sed -e's/.mo//'`;\
-		echo "Copying po/$$lang to $(GIMPLOCALE)/$$localedir/LC_MESSAGES/UpdateCheck.mo";\
-		$(INSTALL_DATA) po/$$lang $(GIMPLOCALE)/$$localedir/LC_MESSAGES/UpdateCheck.mo;\
+		echo "Copying po/$$lang to $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/UpdateCheck.mo";\
+		$(INSTALL_DATA) po/$$lang $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/UpdateCheck.mo;\
 		done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- gimp-sharp-0.11/plug-ins/wbmp/Makefile.in.orig	2007-01-05 19:00:06.000000000 +0000
+++ gimp-sharp-0.11/plug-ins/wbmp/Makefile.in	2007-01-09 14:45:00.000000000 +0000
@@ -505,8 +505,8 @@
 	$(INSTALL) wbmp $(DESTDIR)$(GIMPPLUGINDIR)
 	@for lang in `ls po | grep .mo`;\
 		do localedir=`echo $$lang | sed -e's/.mo//'`;\
-		echo "Copying po/$$lang to $(GIMPLOCALE)/$$localedir/LC_MESSAGES/wbmp.mo";\
-		$(INSTALL_DATA) po/$$lang $(GIMPLOCALE)/$$localedir/LC_MESSAGES/wbmp.mo;\
+		echo "Copying po/$$lang to $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/wbmp.mo";\
+		$(INSTALL_DATA) po/$$lang $(DESTDIR)$(GIMPLOCALE)/$$localedir/LC_MESSAGES/wbmp.mo;\
 		done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- gimp-sharp-0.11/plug-ins/Colorize/Colorize.cs.orig	2007-01-06 14:40:44.000000000 +0000
+++ gimp-sharp-0.11/plug-ins/Colorize/Colorize.cs	2007-01-10 21:37:58.000000000 +0000
@@ -156,7 +156,7 @@
       table.Attach(useChroma, 0, 1, 4, 5);
 
       CheckButton useEntireImage = 
-	new CheckButton(_("Unselected areas are mask"));
+	new CheckButton(_("Use the entire image, even when the selected area is smaller"));
       useEntireImage.Active = _useEntireImage;
       useEntireImage.Toggled += delegate(object sender, EventArgs args) {
 	_useEntireImage = useEntireImage.Active;
