--- GKlib-20230917/CMakeLists.txt.orig	2023-09-17 15:47:41.000000000 +0100
+++ GKlib-20230917/CMakeLists.txt	2023-09-17 21:12:01.815945902 +0100
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 2.8)
 project(GKlib C)
 
-option(BUILD_SHARED_LIBS "Build shared libraries (.dll/.so) instead of static ones (.lib/.a)" OFF)
+option(BUILD_SHARED_LIBS "Build shared libraries (.dll/.so) instead of static ones (.lib/.a)" ON)
 
 get_filename_component(abs "." ABSOLUTE)
 set(GKLIB_PATH ${abs})
@@ -26,6 +26,6 @@
 add_subdirectory("test")
 
 install(TARGETS GKlib
-  ARCHIVE DESTINATION lib/${LINSTALL_PATH}
-  LIBRARY DESTINATION lib/${LINSTALL_PATH})
+  ARCHIVE DESTINATION lib64/${LINSTALL_PATH}
+  LIBRARY DESTINATION lib64/${LINSTALL_PATH})
 install(FILES ${GKlib_includes} DESTINATION include/${HINSTALL_PATH})
