--- a/src/metadata.c	2009-11-09 20:54:46.000000000 +0100
+++ b/src/metadata.c	2009-11-09 20:56:04.000000000 +0100
@@ -1018,7 +1018,7 @@
 bool
 init_metadata(void)
 {
-    magic_cookie = magic_open(MAGIC_SYMLINK|MAGIC_MIME|MAGIC_ERROR);
+    magic_cookie = magic_open(MAGIC_SYMLINK|MAGIC_MIME_TYPE|MAGIC_ERROR);
     if (magic_cookie == NULL) {
         warn(_("cannot initialize magic library\n"));
         return false;
