--- gmic-1.6.0.4/src/Makefile.orig	2015-02-06 10:20:54.000000000 +0000
+++ gmic-1.6.0.4/src/Makefile	2015-02-26 22:20:08.939851780 +0000
@@ -68,7 +68,7 @@
 ifeq ($(OSTYPE),msys)
 EXE = .exe
 endif
-LIB=lib
+LIB=lib64
 BIN=bin
 INCLUDE=include
 
@@ -143,13 +143,13 @@
 # This requires the presence of the X11 include and library files.
 # (package 'libx11-dev' on Debian).
 X11_CFLAGS = -Dcimg_display=1 -Dcimg_appname=\\\"gmic\\\" -I/usr/X11R6/include #-Dcimg_use_xrandr
-X11_LIBS = -L/usr/X11R6/lib -lX11 -lpthread #-lXrandr
+X11_LIBS = -L/usr/X11R6/lib64 -lX11 -lpthread #-lXrandr
 
 # Flags to enable fast display, using XShm.
 # This requires the presence of the X11 extension include and library files.
 # (package 'libx11-dev' on Debian).
 XSHM_CFLAGS = -Dcimg_use_xshm
-XSHM_LIBS = -L$(USR)/X11R6/lib -lXext
+XSHM_LIBS = -L$(USR)/X11R6/lib64 -lXext
 
 # Flags to enable image display, using GDI32.
 # This requires the presence of the GDI32 include and library files.
@@ -362,7 +362,7 @@
 	$(MAKE) "CFLAGS=$(STD_UNIX_CFLAGS) $(OPT_CFLAGS)" "LIBS=$(STD_UNIX_LIBS)" "STRIP_EXE=1" gmic
 
 solaris:
-	$(MAKE) "CFLAGS=$(STD_UNIX_CFLAGS) $(OPT_CFLAGS)" "LIBS=$(STD_UNIX_LIBS) -R$(USR)/X11R6/lib -lrt -lnsl -lsocket" "STRIP_EXE=1" gmic
+	$(MAKE) "CFLAGS=$(STD_UNIX_CFLAGS) $(OPT_CFLAGS)" "LIBS=$(STD_UNIX_LIBS) -R$(USR)/X11R6/lib64 -lrt -lnsl -lsocket" "STRIP_EXE=1" gmic
 
 macosx:
 	$(MAKE) "CFLAGS=$(STD_MACOSX_CFLAGS) $(OPT_CFLAGS)" "LIBS=$(STD_MACOSX_LIBS) $(OPT_LIBS)" gmic
