--- gmic-2.1.3/src/Makefile.orig	2017-10-27 12:38:43.000000000 +0100
+++ gmic-2.1.3/src/Makefile	2017-11-02 21:40:28.513171825 +0000
@@ -112,7 +112,7 @@
 ifneq (,$(wildcard ../../gmic/src))
 ZART_GMIC_PATH=../../gmic/src
 else
-ZART_GMIC_PATH=../src
+ZART_GMIC_PATH=../../src
 endif
 
 ifneq (,$(wildcard ../../gmic/src))
@@ -455,7 +455,7 @@
 	$(STRIP) gmic$(EXE)
 
 zart: zart_files CImg.h gmic_stdlib.h
-	@if [ ! -d ../zart ]; then ln -fs ../gmic-community/zart ..; fi
+	@if [ ! -d ../zart ]; then ln -fs gmic-community/zart ..; fi
 ifeq ($(OS),Darwin)
 	cd ../zart && qmake CONFIG+=release GMIC_PATH=$(ZART_GMIC_PATH) zart.pro && $(MAKE) && $(STRIP) zart.app/Contents/MacOS/zart
 else
@@ -482,9 +482,9 @@
 	    echo "** Warning: folder 'gmic-qt' was not found !"; \
 	    echo "** It is mandatory to compile the G'MIC plug-in for GIMP and Krita ('gmic_gimp_qt' and 'gmic_krita_qt'),"; \
 	    echo "** as well as the standalone Qt-based interface 'gmic_qt'."; \
-            echo "** Trying to retrieve it, with: $$ cd ../../ && git clone https://github.com/c-koi/gmic-qt.git"; \
+            echo "** Trying to retrieve it, with: $$ cd ../ && git clone https://github.com/c-koi/gmic-qt.git"; \
 	    echo "**"; \
-	    if cd ../../ && git clone https://github.com/c-koi/gmic-qt.git; then \
+	    if cd ../ && git clone https://github.com/c-koi/gmic-qt.git; then \
 	      echo "**"; \
               echo "** Success !"; \
 	      echo "**"; \
@@ -499,13 +499,13 @@
 
 zart_files:
 	@if [ ! -d ../zart ]; then \
-	  if [ ! -d ../../gmic-community/zart ]; then \
+	  if [ ! -d ../gmic-community/zart ]; then \
 	    echo "**"; \
 	    echo "** Warning: folder 'gmic-community/zart' was not found !"; \
 	    echo "** It is mandatory to compile 'zart'."; \
-            echo "** Trying to retrieve it, with: $$ cd ../../ && git clone https://github.com/dtschump/gmic-community.git"; \
+            echo "** Trying to retrieve it, with: $$ cd ../ && git clone https://github.com/dtschump/gmic-community.git"; \
 	    echo "**"; \
-	    if cd ../../ && git clone https://github.com/dtschump/gmic-community.git; then \
+	    if cd ../ && git clone https://github.com/dtschump/gmic-community.git; then \
 	      echo "**"; \
               echo "** Success !"; \
 	      echo "**"; \
@@ -515,7 +515,7 @@
               echo "**"; \
 	    fi; \
 	  fi; \
-	  if [ -d ../../gmic-community/zart ]; then ln -fs ../gmic-community/zart ..; fi \
+	  if [ -d ../gmic-community/zart ]; then ln -fs gmic-community/zart ..; fi \
 	fi
 
 # Internal rules to build compilation modules.
