--- gmic-3.7.3/src/Makefile.orig	2026-03-16 14:40:46.081986712 +0000
+++ gmic-3.7.3/src/Makefile	2026-03-16 14:41:03.676590356 +0000
@@ -88,7 +88,7 @@
 EXE =
 WGET = wget --no-check-certificate --quiet -O
 GIMPTOOL = $(shell command -v gimptool-3.2 >/dev/null 2>&1 && echo gimptool-3.2 || \
-           command -v gimptool-3.0 >/dev/null 2>&1 && echo gimptool-3.0 || \
+           (command -v gimptool-3.0 >/dev/null 2>&1 && echo gimptool-3.0) || \
            echo gimptool-2.0)
 PLUGIN = $(shell $(GIMPTOOL) --gimpplugindir)/plug-ins
 SHARE = $(PREFIX)/share
