--- gmic-2.0.2/src/Makefile.orig	2017-06-22 13:41:18.000000000 +0100
+++ gmic-2.0.2/src/Makefile	2017-07-23 16:33:13.158664040 +0100
@@ -112,7 +112,7 @@
 ifneq (,$(wildcard ../../gmic/src))
 ZART_GMIC_PATH=../../gmic/src
 else
-ZART_GMIC_PATH=../src
+ZART_GMIC_PATH=../../src
 endif
 
 ifneq (,$(wildcard ../../gmic/src))
@@ -459,7 +459,7 @@
 	$(STRIP) gmic$(EXE)
 
 zart: CImg.h gmic_stdlib.h gmic_community_path
-	@if [ ! -d ../zart ]; then ln -fs ../gmic-community/zart ..; fi
+	@if [ ! -d ../zart ]; then ln -fs gmic-community/zart ..; fi
 ifeq ($(OS),Darwin)
 	cd ../zart && qmake CONFIG+=release GMIC_PATH=$(ZART_GMIC_PATH) zart.pro && $(MAKE) && $(STRIP) zart.app/Contents/MacOS/zart
 else
@@ -484,7 +484,7 @@
 	    echo "** as well as the standalone Qt-based interface 'gmic_qt'."; \
             echo "** Trying to retrieve it, with: $$ cd ../../ && git clone https://github.com/c-koi/gmic-qt.git"; \
 	    echo "**"; \
-	    if cd ../../ && git clone https://github.com/c-koi/gmic-qt.git; then \
+	    if (cd ../ && git clone https://github.com/c-koi/gmic-qt.git); then \
 	      echo "**"; \
               echo "** Success !"; \
 	      echo "**"; \
@@ -499,13 +499,13 @@
 
 gmic_community_path:
 	@if [ ! -d ../zart ]; then \
-	  if [ ! -d ../../gmic-community/zart ]; then \
+	  if [ ! -d ../../gmic-community ]; then \
 	    echo "**"; \
 	    echo "** Warning: folder 'gmic-community/zart' was not found !"; \
 	    echo "** It is mandatory to compile 'zart'."; \
             echo "** Trying to retrieve it, with: $$ cd ../../ && git clone https://github.com/dtschump/gmic-community.git"; \
 	    echo "**"; \
-	    if cd ../../ && git clone https://github.com/dtschump/gmic-community.git; then \
+	    if (cd ../ && git clone https://github.com/dtschump/gmic-community.git); then \
 	      echo "**"; \
               echo "** Success !"; \
 	      echo "**"; \
