--- gmsh-2.5.0/CMakeLists.txt.orig	2010-10-15 14:35:00.000000000 +0100
+++ gmsh-2.5.0/CMakeLists.txt	2011-02-12 20:01:20.000000000 +0000
@@ -224,7 +224,7 @@
       # on Linux also try to find ATLAS without a Fortran compiler,
       # because cmake ships with a buggy FindBLAS e.g. on Ubuntu Lucid
       # Lynx
-      set(ATLAS_LIBS_REQUIRED lapack f77blas cblas atlas)
+      set(ATLAS_LIBS_REQUIRED satlas)
       find_all_libraries(LAPACK_LIBRARIES ATLAS_LIBS_REQUIRED "" "")
       if(LAPACK_LIBRARIES)
         set_config_option(HAVE_BLAS "Blas(ATLAS)")
