--- gmsh-4.11.1/src/mesh/meshGFacePack.cpp.orig	2022-09-28 19:01:00.000000000 +0100
+++ gmsh-4.11.1/src/mesh/meshGFacePack.cpp	2023-07-20 20:51:22.451746898 +0100
@@ -4,6 +4,7 @@
 // Please report all issues on https://gitlab.onelab.info/gmsh/gmsh/issues.
 
 #include <stdio.h>
+#include <stdint.h>
 #include <stdlib.h>
 #include <queue>
 #include <map>
--- gmsh-4.11.1/contrib/QuadMeshingTools/qmtMeshGeometryOptimization.h.orig	2022-09-28 19:00:54.000000000 +0100
+++ gmsh-4.11.1/contrib/QuadMeshingTools/qmtMeshGeometryOptimization.h	2023-07-20 20:57:34.472729148 +0100
@@ -7,6 +7,7 @@
 
 #pragma once
 
+#include <stdint.h>
 #include <float.h>
 #include "qmtMeshUtils.h"
 
