--- gnome-python-extras-2.14.1/configure.orig	2006-07-10 23:34:56.000000000 +0100
+++ gnome-python-extras-2.14.1/configure	2006-07-11 21:19:37.000000000 +0100
@@ -19275,7 +19275,7 @@
 
 echo "$as_me:$LINENO: checking for gnome-python defs" >&5
 echo $ECHO_N "checking for gnome-python defs... $ECHO_C" >&6
-PYGTK_DEFSDIR=`$PKG_CONFIG --variable=defsdir gnome-python-2.0`
+GNOME_PYTHON_DEFSDIR=`$PKG_CONFIG --variable=defsdir gnome-python-2.0`
 
 echo "$as_me:$LINENO: result: $GNOME_PYTHON_DEFSDIR" >&5
 echo "${ECHO_T}$GNOME_PYTHON_DEFSDIR" >&6
