--- gnome-subtitles-1.8/subprojects/gstreamer-sharp.wrap.orig	2022-06-24 21:43:23.000000000 +0100
+++ gnome-subtitles-1.8/subprojects/gstreamer-sharp.wrap	2024-02-07 21:26:23.194515705 +0000
@@ -1,6 +1,6 @@
 [wrap-git]
 directory=gstreamer-sharp
 url=https://gitlab.freedesktop.org/noup/gstreamer-sharp.git
-revision=tags/1.18.0-gnome-subtitles
+revision=tags/1.19.2
 #url=https://gitlab.freedesktop.org/gstreamer/gstreamer-sharp.git
 #revision=tags/1.18.0
