--- gnucash-docs-5.10/cmake/AddXdgHelpTarget.cmake.orig	2024-09-23 00:25:40.000000000 +0100
+++ gnucash-docs-5.10/cmake/AddXdgHelpTarget.cmake	2024-12-17 17:56:20.540041192 +0000
@@ -90,7 +90,8 @@
         COMPONENT "xdghelp")
 
     # Add links to make our documentation visible for KDE's help system
-    set(doc_install_dir_kde_base "\$ENV{DESTDIR}${CMAKE_INSTALL_DATADIR}/doc/HTML")
+    set(doc_install_dir_kde_base "\$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATADIR}/doc/HTML")
+    message("SM: ${doc_install_dir_kde_base}")
     install(CODE "execute_process (COMMAND ${CMAKE_COMMAND} -E make_directory \"${doc_install_dir_kde_base}\")"
         COMPONENT "xdghelp")
     install(CODE "execute_process (COMMAND ${CMAKE_COMMAND} -E create_symlink \"../../../${doc_install_dir_base}\" \"${doc_install_dir_kde_base}/${kde_lang}\")"
