--- gnucash-5.15/CMakeLists.txt.orig	2026-03-28 21:58:52.000000000 +0000
+++ gnucash-5.15/CMakeLists.txt	2026-08-14 22:24:39.582321622 +0100
@@ -605,7 +605,7 @@
  if (Boost_VERSION VERSION_LESS "1.89.0")
   find_package(Boost 1.67.0  COMPONENTS date_time filesystem locale program_options regex system)
  else()
-  if (Boost_VERSION VERSION_LESS "1.90.0")
+  if (Boost_VERSION VERSION_LESS "1.94.0")
     find_package(Boost 1.67.0  COMPONENTS date_time filesystem locale program_options regex)
   else()
     find_package(Boost 1.67.0  COMPONENTS date_time locale program_options regex)
@@ -642,8 +642,8 @@
 set(CMAKE_C_EXTENSIONS ON)
 
 if (UNIX)
-  set( CMAKE_C_FLAGS "-Werror -Wall -Wmissing-prototypes -Wmissing-declarations ${CMAKE_C_FLAGS}")
-  set( CMAKE_CXX_FLAGS "-Werror -Wall -Wmissing-declarations ${CMAKE_CXX_FLAGS}")
+  set( CMAKE_C_FLAGS "-Wall -Wmissing-prototypes -Wmissing-declarations ${CMAKE_C_FLAGS}")
+  set( CMAKE_CXX_FLAGS "-Wall -Wmissing-declarations ${CMAKE_CXX_FLAGS}")
   set( CMAKE_C_FLAGS_RELEASE "-O3 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 ${CMAKE_C_FLAGS}")
 endif()
 if (MINGW)
