--- gnucash-3.5/CMakeLists.txt.orig	2019-03-30 21:31:24.000000000 +0000
+++ gnucash-3.5/CMakeLists.txt	2019-04-21 14:16:38.981186160 +0100
@@ -526,9 +526,12 @@
 if (NOT DEFINED ${BOOST_ROOT})
   set(BOOST_ROOT $ENV{BOOST_ROOT})
 endif()
+set(BUILD_SHARED_LIBS on)
 find_package (Boost 1.54.0 REQUIRED COMPONENTS date_time regex locale filesystem system)
+set(BUILD_SHARED_LIBS off)
 
 if (Boost_FOUND)
+  set(Boost_LIBRARIES Boost::date_time Boost::regex Boost::locale Boost::filesystem Boost::system)
   include_directories(${Boost_INCLUDE_DIRS})
   set(HAVE_BOOST 1)
 else (Boost_FOUND)
