--- gnucash-2.0.4/src/app-utils/file-utils.h.orig	2007-01-02 06:15:37.000000000 +0000
+++ gnucash-2.0.4/src/app-utils/file-utils.h	2007-01-05 11:53:49.000000000 +0000
@@ -64,7 +64,7 @@
 
 /* Definitions shared by file-utils.c and gnc-main-window.c */
 #define STATE_FILE_TOP		"Top"
-#define STATE_FILE_BOOK_GUID	"Book Guid"
+#define STATE_FILE_BOOK_GUID	"BookGuid"
 
 /** Find the state file that corresponds to this URL and guid.  The
  *  URL is used to compute the base name of the file (which will be in
--- gnucash-2.0.4/src/report/report-gnome/gnc-plugin-page-report.c.orig	2007-01-02 06:15:43.000000000 +0000
+++ gnucash-2.0.4/src/report/report-gnome/gnc-plugin-page-report.c	2007-01-05 11:51:28.000000000 +0000
@@ -644,8 +644,8 @@
 
 /** The key name used it the state file for storing the report
  *  options. */
-#define SCHEME_OPTIONS   "Scheme Options"
-#define SCHEME_OPTIONS_N "Scheme Options %d"
+#define SCHEME_OPTIONS   "SchemeOptions"
+#define SCHEME_OPTIONS_N "SchemeOptions%d"
 
 
 /** Save enough information about this report page that it can be
--- gnucash-2.0.4/src/gnome/gnc-plugin-page-budget.c.orig	2007-01-02 06:15:47.000000000 +0000
+++ gnucash-2.0.4/src/gnome/gnc-plugin-page-budget.c	2007-01-05 12:08:03.000000000 +0000
@@ -459,7 +459,7 @@
     LEAVE("widget destroyed");
 }
 
-#define BUDGET_GUID "Budget GUID"
+#define BUDGET_GUID "BudgetGUID"
 
 /** Save enough information about this plugin page that it can
  *  be recreated next time the user starts gnucash.
--- gnucash-2.0.4/src/gnome/gnc-plugin-page-register.c.orig	2007-01-02 06:15:47.000000000 +0000
+++ gnucash-2.0.4/src/gnome/gnc-plugin-page-register.c	2007-01-05 12:09:59.000000000 +0000
@@ -829,10 +829,10 @@
   NULL
 };
 
-#define KEY_REGISTER_TYPE	"Register Type"
-#define KEY_ACCOUNT_NAME	"Account Name"
-#define KEY_REGISTER_STYLE	"Register Style"
-#define KEY_DOUBLE_LINE		"Double Line Mode"
+#define KEY_REGISTER_TYPE	"RegisterType"
+#define KEY_ACCOUNT_NAME	"AccountName"
+#define KEY_REGISTER_STYLE	"RegisterStyle"
+#define KEY_DOUBLE_LINE		"DoubleLineMode"
 
 #define LABEL_ACCOUNT		"Account"
 #define LABEL_SUBACCOUNT	"SubAccount"
--- gnucash-2.0.4/src/gnome-utils/gnc-main-window.c.orig	2007-01-02 06:15:40.000000000 +0000
+++ gnucash-2.0.4/src/gnome-utils/gnc-main-window.c	2007-01-05 12:30:42.000000000 +0000
@@ -390,19 +390,19 @@
 /************************************************************
  *                                                          *
  ************************************************************/
-#define WINDOW_COUNT		"Window Count"
+#define WINDOW_COUNT		"WindowCount"
 #define WINDOW_STRING		"Window %d"
-#define WINDOW_GEOMETRY		"Window Geometry"
-#define WINDOW_POSITION		"Window Position"
-#define WINDOW_MAXIMIZED	"Window Maximized"
-#define TOOLBAR_VISIBLE		"Toolbar Visible"
-#define STATUSBAR_VISIBLE	"Statusbar Visible"
-#define SUMMARYBAR_VISIBLE	"Summarybar Visible"
-#define WINDOW_FIRSTPAGE	"First Page"
-#define WINDOW_PAGECOUNT	"Page Count"
-#define WINDOW_PAGEORDER	"Page Order"
-#define PAGE_TYPE		"Page Type"
-#define PAGE_NAME		"Page Name"
+#define WINDOW_GEOMETRY		"WindowGeometry"
+#define WINDOW_POSITION		"WindowPosition"
+#define WINDOW_MAXIMIZED	"WindowMaximized"
+#define TOOLBAR_VISIBLE		"ToolbarVisible"
+#define STATUSBAR_VISIBLE	"StatusbarVisible"
+#define SUMMARYBAR_VISIBLE	"SummarybarVisible"
+#define WINDOW_FIRSTPAGE	"FirstPage"
+#define WINDOW_PAGECOUNT	"PageCount"
+#define WINDOW_PAGEORDER	"PageOrder"
+#define PAGE_TYPE		"PageType"
+#define PAGE_NAME		"PageName"
 #define PAGE_STRING		"Page %d"
 
 typedef struct {
--- gnucash-2.0.4/src/gnome-utils/gnc-tree-view-account.c.orig	2007-01-02 06:15:40.000000000 +0000
+++ gnucash-2.0.4/src/gnome-utils/gnc-tree-view-account.c	2007-01-05 12:07:15.000000000 +0000
@@ -1888,12 +1888,12 @@
   LEAVE(" ");
 }
 
-#define ACCT_COUNT    "Number of Open Accounts"
-#define ACCT_OPEN     "Open Account %d"
-#define ACCT_SELECTED "Selected Account"
-#define SHOW_HIDDEN   "Show Hidden"
-#define SHOW_ZERO     "Show Zero Total"
-#define ACCT_TYPES    "Account Types"
+#define ACCT_COUNT    "NumberOfOpenAccounts"
+#define ACCT_OPEN     "OpenAccount%d"
+#define ACCT_SELECTED "SelectedAccount"
+#define SHOW_HIDDEN   "ShowHidden"
+#define SHOW_ZERO     "ShowZeroTotal"
+#define ACCT_TYPES    "AccountTypes"
 
 typedef struct foo {
   GKeyFile *key_file;
--- gnucash-2.0.4/src/business/business-gnome/dialog-invoice.c.orig	2007-01-02 06:15:55.000000000 +0000
+++ gnucash-2.0.4/src/business/business-gnome/dialog-invoice.c	2007-01-05 11:56:58.000000000 +0000
@@ -1684,10 +1684,10 @@
   return iw;
 }
 
-#define KEY_INVOICE_TYPE	"Invoice Type"
-#define KEY_INVOICE_GUID	"Invoice GUID"
-#define KEY_OWNER_TYPE		"Owner Type"
-#define KEY_OWNER_GUID		"Owner GUID"
+#define KEY_INVOICE_TYPE	"InvoiceType"
+#define KEY_INVOICE_GUID	"InvoiceGUID"
+#define KEY_OWNER_TYPE		"OwnerType"
+#define KEY_OWNER_GUID		"OwnerGUID"
 
 GncPluginPage *
 gnc_invoice_recreate_page (GKeyFile *key_file,
