--- gnucash-docs-5.10/cmake/AddXdgHelpTarget.cmake.orig	2025-01-07 17:12:18.980289097 +0000
+++ gnucash-docs-5.10/cmake/AddXdgHelpTarget.cmake	2025-01-07 17:10:13.430280668 +0000
@@ -90,9 +90,9 @@
         COMPONENT "xdghelp")
 
     # Add links to make our documentation visible for KDE's help system
-    set(doc_install_dir_kde_base "\$ENV{DESTDIR}${CMAKE_INSTALL_DATADIR}/doc/HTML")
-    install(CODE "execute_process (COMMAND ${CMAKE_COMMAND} -E make_directory \"${doc_install_dir_kde_base}\")"
+    set(doc_install_dir_kde_base "\$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATADIR}/doc/HTML")
+    install(CODE "execute_process (COMMAND ${CMAKE_COMMAND} -E make_directory \"${doc_install_dir_kde_base}/${kde_lang}\")"
         COMPONENT "xdghelp")
-    install(CODE "execute_process (COMMAND ${CMAKE_COMMAND} -E create_symlink \"../../../${doc_install_dir_base}\" \"${doc_install_dir_kde_base}/${kde_lang}\")"
+    install(CODE "execute_process (COMMAND ${CMAKE_COMMAND} -E create_symlink \"../../../../${doc_install_dir_base}/${docname}\" \"${doc_install_dir_kde_base}/${kde_lang}/${docname}\")"
         COMPONENT "xdghelp")
 endfunction()
