--- gnucash-5.15/CMakeLists.txt.orig	2026-03-28 21:58:52.000000000 +0000
+++ gnucash-5.15/CMakeLists.txt	2026-03-30 09:56:46.435930797 +0100
@@ -605,7 +605,7 @@
  if (Boost_VERSION VERSION_LESS "1.89.0")
   find_package(Boost 1.67.0  COMPONENTS date_time filesystem locale program_options regex system)
  else()
-  if (Boost_VERSION VERSION_LESS "1.90.0")
+  if (Boost_VERSION VERSION_LESS "1.91.0")
     find_package(Boost 1.67.0  COMPONENTS date_time filesystem locale program_options regex)
   else()
     find_package(Boost 1.67.0  COMPONENTS date_time locale program_options regex)
