--- gnucash-5.13/CMakeLists.txt.orig	2025-09-27 23:10:46.000000000 +0100
+++ gnucash-5.13/CMakeLists.txt	2025-09-29 09:06:31.460149752 +0100
@@ -599,7 +599,7 @@
 if (NOT DEFINED ${BOOST_ROOT})
   set(BOOST_ROOT $ENV{BOOST_ROOT})
 endif()
-find_package (Boost 1.67.0 COMPONENTS date_time filesystem locale program_options regex system)
+find_package (Boost 1.67.0 COMPONENTS date_time filesystem locale program_options regex)
 
 if (Boost_FOUND)
   include_directories(${Boost_INCLUDE_DIRS})
