--- gnumeric-1.7.0/plugins/gda/plugin-gda.c-	2006-09-06 10:25:19.000000000 -0400
+++ gnumeric-1.7.0/plugins/gda/plugin-gda.c	2006-09-06 10:26:31.000000000 -0400
@@ -66,11 +66,11 @@
 	for (row = 0; row < rowcount; row++) {
 		for (col = 0; col < fieldcount; col++) {
 			gchar *str;
-			const GdaValue *value;
+			const GValue *value;
 
 			value = gda_data_model_get_value_at (GDA_DATA_MODEL (recset),
 							     col, row);
-			str = gda_value_stringify ((GdaValue *) value);
+			str = gda_value_stringify (value);
 			value_array_set (array,
 					 col,
 					 row,
