--- gnusound-0.7.5/modules/file_ffmpeg.c.orig	2008-07-05 17:00:26.000000000 +0100
+++ gnusound-0.7.5/modules/file_ffmpeg.c	2008-10-05 22:31:38.000000000 +0100
@@ -37,8 +37,8 @@
 #warning "Not building ffmpeg support."
 #else
 
-#include <ffmpeg/avformat.h>
-#include <ffmpeg/avcodec.h>
+#include <libavformat/avformat.h>
+#include <libavcodec/avcodec.h>
 
 struct ffmpeg_data {
     AVFormatContext *format_context;
--- gnusound-0.7.5/configure.orig	2008-10-05 22:13:18.000000000 +0100
+++ gnusound-0.7.5/configure	2008-10-05 22:29:16.000000000 +0100
@@ -8556,8 +8556,8 @@
 
 
 if test "${ac_cv_header_ffmpeg_avformat_h+set}" = set; then
-  { echo "$as_me:$LINENO: checking for ffmpeg/avformat.h" >&5
-echo $ECHO_N "checking for ffmpeg/avformat.h... $ECHO_C" >&6; }
+  { echo "$as_me:$LINENO: checking for libavformat/avformat.h" >&5
+echo $ECHO_N "checking for libavformat/avformat.h... $ECHO_C" >&6; }
 if test "${ac_cv_header_ffmpeg_avformat_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 fi
@@ -8565,8 +8565,8 @@
 echo "${ECHO_T}$ac_cv_header_ffmpeg_avformat_h" >&6; }
 else
   # Is the header compilable?
-{ echo "$as_me:$LINENO: checking ffmpeg/avformat.h usability" >&5
-echo $ECHO_N "checking ffmpeg/avformat.h usability... $ECHO_C" >&6; }
+{ echo "$as_me:$LINENO: checking libavformat/avformat.h usability" >&5
+echo $ECHO_N "checking libavformat/avformat.h usability... $ECHO_C" >&6; }
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -8574,7 +8574,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 $ac_includes_default
-#include <ffmpeg/avformat.h>
+#include <libavformat/avformat.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (ac_try="$ac_compile"
@@ -8606,15 +8606,15 @@
 echo "${ECHO_T}$ac_header_compiler" >&6; }
 
 # Is the header present?
-{ echo "$as_me:$LINENO: checking ffmpeg/avformat.h presence" >&5
-echo $ECHO_N "checking ffmpeg/avformat.h presence... $ECHO_C" >&6; }
+{ echo "$as_me:$LINENO: checking libavformat/avformat.h presence" >&5
+echo $ECHO_N "checking libavformat/avformat.h presence... $ECHO_C" >&6; }
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <ffmpeg/avformat.h>
+#include <libavformat/avformat.h>
 _ACEOF
 if { (ac_try="$ac_cpp conftest.$ac_ext"
 case "(($ac_try" in
@@ -8647,30 +8647,30 @@
 # So?  What about this header?
 case $ac_header_compiler:$ac_header_preproc:$ac_c_preproc_warn_flag in
   yes:no: )
-    { echo "$as_me:$LINENO: WARNING: ffmpeg/avformat.h: accepted by the compiler, rejected by the preprocessor!" >&5
-echo "$as_me: WARNING: ffmpeg/avformat.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
-    { echo "$as_me:$LINENO: WARNING: ffmpeg/avformat.h: proceeding with the compiler's result" >&5
-echo "$as_me: WARNING: ffmpeg/avformat.h: proceeding with the compiler's result" >&2;}
+    { echo "$as_me:$LINENO: WARNING: libavformat/avformat.h: accepted by the compiler, rejected by the preprocessor!" >&5
+echo "$as_me: WARNING: libavformat/avformat.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
+    { echo "$as_me:$LINENO: WARNING: libavformat/avformat.h: proceeding with the compiler's result" >&5
+echo "$as_me: WARNING: libavformat/avformat.h: proceeding with the compiler's result" >&2;}
     ac_header_preproc=yes
     ;;
   no:yes:* )
-    { echo "$as_me:$LINENO: WARNING: ffmpeg/avformat.h: present but cannot be compiled" >&5
-echo "$as_me: WARNING: ffmpeg/avformat.h: present but cannot be compiled" >&2;}
-    { echo "$as_me:$LINENO: WARNING: ffmpeg/avformat.h:     check for missing prerequisite headers?" >&5
-echo "$as_me: WARNING: ffmpeg/avformat.h:     check for missing prerequisite headers?" >&2;}
-    { echo "$as_me:$LINENO: WARNING: ffmpeg/avformat.h: see the Autoconf documentation" >&5
-echo "$as_me: WARNING: ffmpeg/avformat.h: see the Autoconf documentation" >&2;}
-    { echo "$as_me:$LINENO: WARNING: ffmpeg/avformat.h:     section \"Present But Cannot Be Compiled\"" >&5
-echo "$as_me: WARNING: ffmpeg/avformat.h:     section \"Present But Cannot Be Compiled\"" >&2;}
-    { echo "$as_me:$LINENO: WARNING: ffmpeg/avformat.h: proceeding with the preprocessor's result" >&5
-echo "$as_me: WARNING: ffmpeg/avformat.h: proceeding with the preprocessor's result" >&2;}
-    { echo "$as_me:$LINENO: WARNING: ffmpeg/avformat.h: in the future, the compiler will take precedence" >&5
-echo "$as_me: WARNING: ffmpeg/avformat.h: in the future, the compiler will take precedence" >&2;}
+    { echo "$as_me:$LINENO: WARNING: libavformat/avformat.h: present but cannot be compiled" >&5
+echo "$as_me: WARNING: libavformat/avformat.h: present but cannot be compiled" >&2;}
+    { echo "$as_me:$LINENO: WARNING: libavformat/avformat.h:     check for missing prerequisite headers?" >&5
+echo "$as_me: WARNING: libavformat/avformat.h:     check for missing prerequisite headers?" >&2;}
+    { echo "$as_me:$LINENO: WARNING: libavformat/avformat.h: see the Autoconf documentation" >&5
+echo "$as_me: WARNING: libavformat/avformat.h: see the Autoconf documentation" >&2;}
+    { echo "$as_me:$LINENO: WARNING: libavformat/avformat.h:     section \"Present But Cannot Be Compiled\"" >&5
+echo "$as_me: WARNING: libavformat/avformat.h:     section \"Present But Cannot Be Compiled\"" >&2;}
+    { echo "$as_me:$LINENO: WARNING: libavformat/avformat.h: proceeding with the preprocessor's result" >&5
+echo "$as_me: WARNING: libavformat/avformat.h: proceeding with the preprocessor's result" >&2;}
+    { echo "$as_me:$LINENO: WARNING: libavformat/avformat.h: in the future, the compiler will take precedence" >&5
+echo "$as_me: WARNING: libavformat/avformat.h: in the future, the compiler will take precedence" >&2;}
 
     ;;
 esac
-{ echo "$as_me:$LINENO: checking for ffmpeg/avformat.h" >&5
-echo $ECHO_N "checking for ffmpeg/avformat.h... $ECHO_C" >&6; }
+{ echo "$as_me:$LINENO: checking for libavformat/avformat.h" >&5
+echo $ECHO_N "checking for libavformat/avformat.h... $ECHO_C" >&6; }
 if test "${ac_cv_header_ffmpeg_avformat_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
