--- gpgme-1.8.0/lang/cpp/src/GpgmeppConfig.cmake.in.in.orig	2016-11-16 09:17:20.000000000 +0000
+++ gpgme-1.8.0/lang/cpp/src/GpgmeppConfig.cmake.in.in	2016-12-12 20:59:43.139022462 +0000
@@ -63,7 +63,7 @@
 
 set_target_properties(Gpgmepp PROPERTIES
   INTERFACE_INCLUDE_DIRECTORIES "@resolved_includedir@/gpgme++;@resolved_includedir@"
-  INTERFACE_LINK_LIBRARIES "pthread;@resolved_libdir@/libgpgme@libsuffix@;@LIBASSUAN_LIBS@"
+  INTERFACE_LINK_LIBRARIES "pthread;@resolved_libdir@/libgpgme.so;@LIBASSUAN_LIBS@"
   IMPORTED_LOCATION "@resolved_libdir@/libgpgmepp.so"
 )
 
