--- gpsd-3.13/ntpshmread.c.orig	2015-02-26 18:58:34.000000000 +0000
+++ gpsd-3.13/ntpshmread.c	2015-02-28 20:03:26.838083442 +0000
@@ -99,7 +99,7 @@
      */
     if (shmcopy.mode > 0 && cnt != shm->count) {
 	shm_stat->status = CLASH;
-	return shm_stat->status;
+	return (enum segstat_t) shm_stat->status;
     }
 
     shm_stat->status = OK;
@@ -172,7 +172,7 @@
     shm_stat->leap = shmcopy.leap;
     shm_stat->precision = shmcopy.precision;
 
-    return shm_stat->status;
+    return (enum segstat_t) shm_stat->status;
 }
 
 /* end */
