From fec62d6c5fdc0cd3ec3808b7eb3519ea4cd7c3d2 Mon Sep 17 00:00:00 2001
From: Simon Munton <simon.j.munton@gmail.com>
Date: Mon, 11 May 2020 20:13:36 +0100
Subject: [PATCH 2/4] Fix detection of subtree row

---
 gui/gui_breadboard.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gui/gui_breadboard.cc b/gui/gui_breadboard.cc
index fa6ea61..e1cf42b 100644
--- a/gui/gui_breadboard.cc
+++ b/gui/gui_breadboard.cc
@@ -1035,7 +1035,7 @@ static void treeselect_cb(GtkTreeSelection *selection, gpointer p)
     }
 
   } else {
-    if (strlen(spath) > 1) {
+    if (strchr(spath, ':')) {
       GuiPin *gp;
       gtk_tree_model_get(model, &iter, 1, &gp, -1);
       treeselect_stimulus(gp);
-- 
2.26.2

