--- gqcam-0.9/greyscale.c.orig	2012-07-28 16:34:31.764969058 +0100
+++ gqcam-0.9/greyscale.c	2012-07-28 16:34:43.966969251 +0100
@@ -10,7 +10,8 @@
 #include <pthread.h>
 #include <linux/types.h> 
 #include <gtk/gtk.h>
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
+#include <linux/videodev2.h>
 #include <png.h>
 #include "gqcam.h"
 #include "frontend.h"
--- gqcam-0.9/frontend.c.orig	2012-07-28 16:34:31.758969058 +0100
+++ gqcam-0.9/frontend.c	2012-07-28 16:34:43.965969255 +0100
@@ -12,7 +12,8 @@
 #include <linux/types.h> 
 #include <gtk/gtk.h>
 #include <glib.h>
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
+#include <linux/videodev2.h>
 #include <png.h>
 #include "gqcam.h"
 #include "frontend.h"
--- gqcam-0.9/timer.c.orig	2012-07-28 16:34:31.746969057 +0100
+++ gqcam-0.9/timer.c	2012-07-28 16:34:43.963969264 +0100
@@ -10,7 +10,8 @@
 #include <pthread.h>
 #include <gtk/gtk.h>
 #include <linux/types.h> 
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
+#include <linux/videodev2.h>
 #include <stdio.h>
 #include <jpeglib.h>
 #include <png.h>
@@ -63,7 +64,7 @@
 
   camera->on_timer = 1;
   
-  if(strnlen(gtk_entry_get_text(GTK_ENTRY(camera->timer_struct.commandentry))) != 0){
+  if(strlen(gtk_entry_get_text(GTK_ENTRY(camera->timer_struct.commandentry))) != 0){
     strcpy(camera->timer_struct.command, gtk_entry_get_text(GTK_ENTRY(camera->timer_struct.commandentry)));
     camera->timer_struct.iscommand = 1;
   }
--- gqcam-0.9/gqcam.c.orig	2012-07-28 16:34:31.746969057 +0100
+++ gqcam-0.9/gqcam.c	2012-07-28 17:24:35.879016819 +0100
@@ -31,7 +31,9 @@
 #include <semaphore.h>
 #include <gtk/gtk.h>
 #include <linux/types.h> 
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
+#include <linux/videodev2.h>
+#include <libv4l1.h>
 #include <signal.h>
 #include <png.h>
 #include "gqcam.h"
@@ -79,10 +81,10 @@
 
 void set_cam_info(struct Camera *camera)
 {
-  if (ioctl (camera->dev, VIDIOCSPICT, &camera->vid_pic) == -1) {
+  if (v4l1_ioctl (camera->dev, VIDIOCSPICT, &camera->vid_pic) == -1) {
     perror ("ioctl (VIDIOCSPICT)");
   }
-  if (ioctl (camera->dev, VIDIOCSWIN, &camera->vid_win) == -1) {
+  if (v4l1_ioctl (camera->dev, VIDIOCSWIN, &camera->vid_win) == -1) {
     perror ("ioctl (VIDIOCSWIN)");
   }
 }
@@ -92,9 +94,12 @@
   int i;
   struct video_clip vid_clips[32];
 
-  ioctl(camera->dev, VIDIOCGCAP, &camera->vid_caps);
-  ioctl(camera->dev, VIDIOCGWIN, &camera->vid_win);
-  ioctl(camera->dev, VIDIOCGPICT, &camera->vid_pic);
+  if (v4l1_ioctl(camera->dev, VIDIOCGCAP, &camera->vid_caps) < 0)
+    perror("VIDIOCGCAP");
+  if (v4l1_ioctl(camera->dev, VIDIOCGWIN, &camera->vid_win) < 0)
+    perror("VIDIOCGWIN");
+  if (v4l1_ioctl(camera->dev, VIDIOCGPICT, &camera->vid_pic) < 0)
+    perror("VIDIOCGPICT");
 
   for (i = 0; i < 32; i++) {
     vid_clips[i].x      = 0;
@@ -108,7 +113,7 @@
   if (camera->vid_caps.type & VID_TYPE_MONOCHROME) {
     camera->greyscale = 1;
     camera->pic = realloc( camera->pic, camera->vid_caps.maxwidth*camera->vid_caps.maxheight);
-    camera->picbuff = realloc( camera->picbuff, camera->vid_caps.maxwidth*camera->vid_caps.maxheight );
+    camera->picbuff = realloc( camera->picbuff, camera->vid_caps.maxwidth*camera->vid_caps.maxheight);
   }
   else {
     camera->greyscale = 0;
@@ -189,7 +194,7 @@
 void open_cam(struct Camera *camera) 
 {
   if((camera->dev<=0)){
-    camera->dev = open(camera->devname, O_RDWR);
+    camera->dev = v4l1_open(camera->devname, O_RDWR);
     //    printf("Opening: %d\n", camera->dev);
     if (camera->dev < 0) {
       perror("/dev/video");
@@ -204,7 +209,7 @@
   int debug = 0;
   pthread_mutex_lock( &camera->iscam_mutex );
   if(camera->dev > 0){
-    close(camera->dev);
+    v4l1_close(camera->dev);
     camera->dev = 0;
   }
 }
@@ -291,7 +296,7 @@
       pthread_mutex_unlock( &camera->pref_mutex );
       
       if( camera->dev )
-	len = read (camera->dev, camera->picbuff, camera->vid_caps.maxwidth * camera->vid_caps.maxheight*3);
+	len = v4l1_read (camera->dev, camera->picbuff, camera->vid_caps.maxwidth * camera->vid_caps.maxheight*3);
       if (len <= 0)
 	fprintf(stderr, "Error reading image...\n");      
     }
@@ -491,6 +496,8 @@
   //camera.vid_pic.contrast = contrast*256;
   //camera.vid_pic.whiteness = whiteness*256;
   //set_cam_info(&camera);
+  camera.vid_pic.palette = VIDEO_PALETTE_RGB24;
+  set_cam_info(&camera);
   create_frontend(&camera);
 
   if( !camera.speed_fastest )
@@ -565,6 +572,7 @@
   camera->vid_pic.brightness = brightness*256;
   camera->vid_pic.contrast = contrast*256;
   camera->vid_pic.whiteness = whiteness*256;
+  camera->vid_pic.palette = VIDEO_PALETTE_RGB24;
   set_cam_info(camera);
   sem_post( &s_grab1 );
   sem_post( &s_grab2 );
--- gqcam-0.9/color.c.orig	2012-07-28 16:34:31.752969058 +0100
+++ gqcam-0.9/color.c	2012-07-28 16:34:43.964969259 +0100
@@ -10,7 +10,8 @@
 #include <pthread.h>
 #include <linux/types.h> 
 #include <gtk/gtk.h>
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
+#include <linux/videodev2.h>
 #include <png.h>
 #include "gqcam.h"
 #include "frontend.h"
--- gqcam-0.9/save.c.orig	2012-07-28 16:34:31.770969058 +0100
+++ gqcam-0.9/save.c	2012-07-28 16:34:43.967969249 +0100
@@ -10,7 +10,8 @@
 #include <pthread.h>
 #include <gtk/gtk.h>
 #include <linux/types.h> 
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
+#include <linux/videodev2.h>
 #include <stdio.h>
 #include <jpeglib.h>
 #include <png.h>
@@ -171,7 +172,7 @@
   /* Set error handling.  REQUIRED if you aren't supplying your own
    * error hadnling functions in the png_create_write_struct() call.
    */
-  if (setjmp(png_ptr->jmpbuf))
+  if (setjmp(png_jmpbuf(png_ptr)))
     {
       /* If we get here, we had a problem reading the file */
       fclose(fp);
@@ -314,12 +315,12 @@
   png_write_end(png_ptr, info_ptr);
   
   /* if you malloced the palette, free it here */
-  free(info_ptr->palette);
+//  free(info_ptr->palette);
   
   /* if you allocated any text comments, free them here */
   
   /* clean up after the write, and free any memory allocated */
-  png_destroy_write_struct(&png_ptr, (png_infopp)NULL);
+  png_destroy_write_struct(&png_ptr, &info_ptr);
   
   /* close the file */
   fclose(fp);
--- gqcam-0.9/preferences.c.orig	2012-07-28 16:34:31.757969058 +0100
+++ gqcam-0.9/preferences.c	2012-07-28 16:34:43.965969255 +0100
@@ -12,7 +12,8 @@
 #include <linux/types.h> 
 #include <gtk/gtk.h>
 #include <glib.h>
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
+#include <linux/videodev2.h>
 #include <png.h>
 #include "gqcam.h"
 #include "frontend.h"
--- gqcam-0.9/filters.c.orig	2012-07-28 16:34:31.723969052 +0100
+++ gqcam-0.9/filters.c	2012-07-28 16:34:43.963969264 +0100
@@ -12,7 +12,8 @@
 #include <semaphore.h>
 #include <gtk/gtk.h>
 #include <linux/types.h> 
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
+#include <linux/videodev2.h>
 #include <signal.h>
 #include <png.h>
 #include "gqcam.h"
--- gqcam-0.9/Makefile.orig	2012-07-28 16:34:31.770969058 +0100
+++ gqcam-0.9/Makefile	2012-07-28 16:34:43.966969251 +0100
@@ -2,7 +2,7 @@
 CC = gcc
 LD = $(CC)
 
-LDFLAGS = `gtk-config --libs gthread` -ljpeg -lpthread -lpng
+LDFLAGS = `gtk-config --libs gthread` -ljpeg -lpthread -lpng -lv4l1
 
 PROG = gqcam
 VERSION = 0.9
