--- graphviz-2.49.0/CMakeLists.txt.orig	2021-08-28 18:03:42.000000000 +0100
+++ graphviz-2.49.0/CMakeLists.txt	2021-08-30 19:39:54.145646987 +0100
@@ -49,11 +49,11 @@
 
 # ================== Convenient values for CMake configuration =================
 set(BINARY_INSTALL_DIR      bin)
-set(LIBRARY_INSTALL_DIR     lib)
+set(LIBRARY_INSTALL_DIR     lib${LIB_SUFFIX})
 set(PLUGIN_INSTALL_DIR      ${LIBRARY_INSTALL_DIR}/graphviz)
 set(HEADER_INSTALL_DIR      include/graphviz)
 set(MAN_INSTALL_DIR         share/man)
-set(PKGCONFIG_DIR	    lib/pkgconfig)
+set(PKGCONFIG_DIR	    lib${LIB_SUFFIX}/pkgconfig)
 # TODO: Find a way to check for groff and ps2pdf for manpage pdf generation
 # set(MAN_PDF_INSTALL_DIR share/graphviz/doc/pdf)
 set(TOP_SOURCE_DIR          "${CMAKE_CURRENT_SOURCE_DIR}")
@@ -78,7 +78,7 @@
     list(APPEND CMAKE_PREFIX_PATH ${WINDOWS_DEPENDENCY_DIR})
     list(APPEND CMAKE_PREFIX_PATH ${WINDOWS_DEPENDENCY_VCPKG_DIR})
 else()
-    list(APPEND CMAKE_INCLUDE_PATH /usr/lib /usr/local/lib /usr/lib/x86_64-linux-gnu)
+    list(APPEND CMAKE_INCLUDE_PATH /usr/lib${LIB_SUFFIX} /usr/local/lib${LIB_SUFFIX} /usr/lib${LIB_SUFFIX}/x86_64-linux-gnu)
 endif()
 
 find_package(ANN)
@@ -260,7 +260,7 @@
 set(VERSION "${GRAPHVIZ_VERSION_STRING}")
 set(prefix "${CMAKE_INSTALL_PREFIX}")
 set(exec_prefix "${CMAKE_INSTALL_PREFIX}")
-set(libdir "${CMAKE_INSTALL_PREFIX}/lib")
+set(libdir "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
 set(includedir "${CMAKE_INSTALL_PREFIX}/include")
 set(PACKAGE "graphviz")
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/lib/cdt/libcdt.pc.in ${CMAKE_CURRENT_SOURCE_DIR}/libcdt.pc @ONLY)
