--- graphviz-7.0.1/tclpkg/Makefile.in.orig	2022-11-09 15:10:02.000000000 +0000
+++ graphviz-7.0.1/tclpkg/Makefile.in	2022-11-09 22:33:17.789936170 +0000
@@ -954,14 +954,14 @@
 @ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@		echo "Warning: $(PHP_INSTALL_DIR) is not writable."; \
 @ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@		echo "Skipping system installation of php binding."; \
 @ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@	fi
-@ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@	-mkdir -p $(DESTDIR)$(PHP_INSTALL_DATADIR);
-@ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@	if test -w $(DESTDIR)$(PHP_INSTALL_DATADIR); then \
-@ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@		(cd $(DESTDIR)$(PHP_INSTALL_DATADIR); \
-@ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@			cp -f $(DESTDIR)$(pkgphpdir)/gv.php gv.php;) \
-@ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@	else \
-@ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@		echo "Warning: $(PHP_INSTALL_DATADIR) is not writable."; \
-@ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@		echo "Skipping system installation of php binding."; \
-@ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@	fi
+@ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@@WITH_SWIG_PHP_DATA_TRUE@	-mkdir -p $(DESTDIR)$(PHP_INSTALL_DATADIR);
+@ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@@WITH_SWIG_PHP_DATA_TRUE@	if test -w $(DESTDIR)$(PHP_INSTALL_DATADIR); then \
+@ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@@WITH_SWIG_PHP_DATA_TRUE@		(cd $(DESTDIR)$(PHP_INSTALL_DATADIR); \
+@ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@@WITH_SWIG_PHP_DATA_TRUE@			cp -f $(DESTDIR)$(pkgphpdir)/gv.php gv.php;) \
+@ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@@WITH_SWIG_PHP_DATA_TRUE@	else \
+@ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@@WITH_SWIG_PHP_DATA_TRUE@		echo "Warning: $(PHP_INSTALL_DATADIR) is not writable."; \
+@ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@@WITH_SWIG_PHP_DATA_TRUE@		echo "Skipping system installation of php binding."; \
+@ENABLE_SHARED_TRUE@@WITH_PHP_TRUE@@WITH_SWIG_PHP_DATA_TRUE@	fi
 @ENABLE_SHARED_TRUE@@WITH_PYTHON_TRUE@	-mkdir -p $(DESTDIR)$(PYTHON_INSTALL_DIR);
 @ENABLE_SHARED_TRUE@@WITH_PYTHON_TRUE@	if test -w $(DESTDIR)$(PYTHON_INSTALL_DIR); then \
 @ENABLE_SHARED_TRUE@@WITH_PYTHON_TRUE@		(cd $(DESTDIR)$(PYTHON_INSTALL_DIR); \
--- graphviz-7.0.1/tclpkg/Makefile.am.orig	2022-11-09 15:09:45.000000000 +0000
+++ graphviz-7.0.1/tclpkg/Makefile.am	2022-11-09 22:31:44.604940616 +0000
@@ -63,6 +63,7 @@
 		echo "Warning: $(PHP_INSTALL_DIR) is not writable."; \
 		echo "Skipping system installation of php binding."; \
 	fi
+if WITH_SWIG_PHP_DATA
 	-mkdir -p $(DESTDIR)$(PHP_INSTALL_DATADIR);
 	if test -w $(DESTDIR)$(PHP_INSTALL_DATADIR); then \
 		(cd $(DESTDIR)$(PHP_INSTALL_DATADIR); \
@@ -72,6 +73,7 @@
 		echo "Skipping system installation of php binding."; \
 	fi
 endif
+endif
 if WITH_PYTHON
 	-mkdir -p $(DESTDIR)$(PYTHON_INSTALL_DIR);
 	if test -w $(DESTDIR)$(PYTHON_INSTALL_DIR); then \
