--- grip-3.3.1/configure.orig	2016-07-16 14:36:18.000000000 +0100
+++ grip-3.3.1/configure	2016-07-16 14:36:29.906984939 +0100
@@ -11275,7 +11275,7 @@
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-  . conftest.id3
+  . ./conftest.id3
    echo "${ID3LIB_MAJOR}.${ID3LIB_MINOR}.${ID3LIB_PATCH}"
 else
   echo "$as_me: program exited with status $ac_status" >&5
