--- gscanbus-0.7.1/gscanbus.c.orig	2001-07-11 11:53:51.000000000 +0100
+++ gscanbus-0.7.1/gscanbus.c	2006-12-02 20:36:28.000000000 +0000
@@ -109,14 +109,14 @@
 
 static GdkPixmap *pixmap = NULL;
 
-const char not_compatible[] = "\
-This libraw1394 does not work with your version of Linux. You need a different
-version that matches your kernel (see kernel help text for the raw1394 option to
-find out which is the correct version).\n";
-
-const char not_loaded[] = "\
-This probably means that you don't have raw1394 support in the kernel or that
-you haven't loaded the raw1394 module.\n";
+const char not_compatible[] = 
+"This libraw1394 does not work with your version of Linux. You need a different\n"
+"version that matches your kernel (see kernel help text for the raw1394 option to\n"
+"find out which is the correct version).\n";
+
+const char not_loaded[] = 
+"This probably means that you don't have raw1394 support in the kernel or that\n"
+"you haven't loaded the raw1394 module.\n";
 
 /*---------------------------------------------------------------------------
  * Drawing routines
--- gscanbus-0.7.1/Makefile.in.orig	2001-07-11 11:54:09.000000000 +0100
+++ gscanbus-0.7.1/Makefile.in	2006-12-02 20:38:33.000000000 +0000
@@ -328,8 +328,8 @@
 
 
 install-data-local:
-	$(INSTALL_DATA) $(srcdir)/guid-resolv.conf /etc/guid-resolv.conf
-	$(INSTALL_DATA) $(srcdir)/oui-resolv.conf /etc/oui-resolv.conf
+	$(INSTALL_DATA) $(srcdir)/guid-resolv.conf $(DESTDIR)/etc/guid-resolv.conf
+	$(INSTALL_DATA) $(srcdir)/oui-resolv.conf $(DESTDIR)/etc/oui-resolv.conf
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
