--- gspiceui-v1.2.87/src/main/FrmMain.cpp.orig	2020-09-17 16:36:05.000000000 +0100
+++ gspiceui-v1.2.87/src/main/FrmMain.cpp	2020-11-05 20:43:17.961078678 +0000
@@ -583,7 +583,7 @@
   poSzr->Add( &m_oLblCpnts, oPosn, oSpan, iFlags, iBorder+5 );
 
   // Add the nodes list box
-  iFlags = (wxALL & ~wxRIGHT) | wxALIGN_CENTER | wxEXPAND;
+  iFlags = /*(wxALL & ~wxRIGHT) | wxALIGN_CENTER |*/ wxEXPAND;
   oPosn.SetCol( 0 ); oSpan.SetColspan( 1 );
   oPosn.SetRow( 1 );
   if( m_uiFrmLayout == 0 ) oSpan.SetRowspan( 14 );
@@ -592,7 +592,7 @@
   poSzr->Add( &m_oLbxNodes, oPosn, oSpan, iFlags, iBorder );
 
   // Add the components list box
-  iFlags = wxTOP | wxBOTTOM | wxALIGN_CENTER | wxEXPAND;
+  iFlags = /*wxTOP | wxBOTTOM | wxALIGN_CENTER |*/ wxEXPAND;
   oPosn.SetCol( 1 ); oSpan.SetColspan( 1 );
   oPosn.SetRow( 1 );
   if( m_uiFrmLayout == 0 ) oSpan.SetRowspan( 14 );
@@ -601,13 +601,13 @@
   poSzr->Add( &m_oLbxCpnts, oPosn, oSpan, iFlags, iBorder );
 
   // Add the simulator notebook
-  iFlags = wxTOP | wxRIGHT | wxALIGN_CENTER | wxEXPAND;
+  iFlags = /*wxTOP | wxRIGHT | wxALIGN_CENTER |*/ wxEXPAND;
   oPosn.SetCol( 2 ); oSpan.SetColspan( 5 );
   oPosn.SetRow( 0 ); oSpan.SetRowspan( 7 );
   poSzr->Add( m_poNbkSimEng, oPosn, oSpan, iFlags, iBorder );
 
   // Add the console notebook
-  iFlags = wxBOTTOM | wxRIGHT | wxALIGN_CENTER | wxEXPAND;
+  iFlags = /*wxBOTTOM | wxRIGHT | wxALIGN_CENTER |*/ wxEXPAND;
   oPosn.SetRow( 7 ); oSpan.SetRowspan( 8 );
   if( m_uiFrmLayout == 0 ) { oPosn.SetCol( 2 ); oSpan.SetColspan( 5 ); }
   else                     { oPosn.SetCol( 0 ); oSpan.SetColspan( 7 ); iFlags |= wxLEFT; }
--- gspiceui-v1.2.87/src/main/DlgPrefs.cpp.orig	2020-08-20 05:53:03.000000000 +0100
+++ gspiceui-v1.2.87/src/main/DlgPrefs.cpp	2020-11-03 19:37:31.125018849 +0000
@@ -331,7 +331,7 @@
 
   // Layout the buttons
   oFlags.Border( wxTOP | wxBOTTOM, 10 );
-  oFlags.Align( wxALIGN_RIGHT );
+  oFlags.Align( wxALIGN_NOT );
   poSzrBtns->Add( &m_oBtnOk    , oFlags );
   poSzrBtns->AddSpacer( 10 );
   oFlags.Align( wxALIGN_LEFT );
--- gspiceui-v1.2.87/src/gnucap/dialogs/DlgGnuCapCfgOPT.cpp.orig	2020-08-17 16:58:42.000000000 +0100
+++ gspiceui-v1.2.87/src/gnucap/dialogs/DlgGnuCapCfgOPT.cpp	2020-11-03 19:43:14.575002462 +0000
@@ -454,7 +454,7 @@
   poSzrRHS1->Add( &m_oCbxTRACEL  , oFlags );
 
   // Layout the buttons panel
-  oFlags.Align( wxALIGN_RIGHT  );
+  oFlags.Align( wxALIGN_NOT  );
   oFlags.Border( wxTOP | wxBOTTOM        , 12 );
   poSzrBtns->Add( &m_oBtnOk      , oFlags );
   poSzrBtns->AddSpacer( 10 );
--- gspiceui-v1.2.87/src/gnucap/dialogs/DlgGnuCapCfgGEN.cpp.orig	2020-08-17 17:00:51.000000000 +0100
+++ gspiceui-v1.2.87/src/gnucap/dialogs/DlgGnuCapCfgGEN.cpp	2020-11-03 19:42:53.204003482 +0000
@@ -222,7 +222,7 @@
   poSzrPulse->Add( &m_oPnlPulPeriod, oFlags );
 
   // Layout the button panel
-  oFlags.Align( wxALIGN_RIGHT  );
+  oFlags.Align( wxALIGN_NOT  );
   oFlags.Border( wxTOP | wxBOTTOM, 10 );
   poSzrBtns->Add( &m_oBtnOk    , oFlags );
   poSzrBtns->AddSpacer( 10 );
--- gspiceui-v1.2.87/src/TypeDefs.hpp.orig	2020-08-18 16:57:04.000000000 +0100
+++ gspiceui-v1.2.87/src/TypeDefs.hpp	2020-11-05 20:23:38.995134929 +0000
@@ -86,7 +86,7 @@
 //**************************************************************************************************
 // Useful #defines
 
-#define  GUI_CTRL_HT  24         // The height of the PnlValue control in pixels
+#define  GUI_CTRL_HT  28         // The height of the PnlValue control in pixels
 
 // The following #define values where determined using the test utility test_CnvtType
 #define  NOVAL_UINT   UINT_MAX   // UINT_MAX  =  4294967295
--- gspiceui-v1.2.87/src/ngspice/dialogs/DlgNgSpiceCfgOPT.cpp.orig	2020-08-19 14:35:08.000000000 +0100
+++ gspiceui-v1.2.87/src/ngspice/dialogs/DlgNgSpiceCfgOPT.cpp	2020-11-03 19:40:22.607010667 +0000
@@ -391,7 +391,7 @@
   poSzrRHS1->Add( &m_oCbxWARN    , oFlags );
 
   // Layout the buttons panel
-  oFlags.Align( wxALIGN_RIGHT  );
+  oFlags.Align( wxALIGN_NOT  );
   oFlags.Border( wxTOP | wxBOTTOM        , 12 );
   poSzrBtns->Add( &m_oBtnOk      , oFlags );
   poSzrBtns->AddSpacer( 10 );
--- gspiceui-v1.2.87/src/ngspice/dialogs/DlgNgSpiceCfgSrc.cpp.orig	2020-08-17 16:58:17.000000000 +0100
+++ gspiceui-v1.2.87/src/ngspice/dialogs/DlgNgSpiceCfgSrc.cpp	2020-11-03 19:40:01.626011668 +0000
@@ -203,7 +203,7 @@
   poSzrPulse->Add( &m_oPnlPulPeriod, oFlags );
 
   // Layout the button panel
-  oFlags.Align( wxALIGN_RIGHT  );
+  oFlags.Align( wxALIGN_NOT  );
   oFlags.Border( wxTOP | wxBOTTOM, 10 );
   poSzrBtns->Add( &m_oBtnOk    , oFlags );
   poSzrBtns->AddSpacer( 10 );
