--- gspiceui-v1.2.36/src/main/FrmMain.cpp.orig	2018-11-12 22:23:59.000000000 +0000
+++ gspiceui-v1.2.36/src/main/FrmMain.cpp	2020-04-04 17:41:40.698172040 +0100
@@ -548,7 +548,7 @@
   iFlags = wxBOTTOM | wxRIGHT | wxALIGN_CENTER | wxEXPAND;
   oPosn.SetRow( 7 );          oSpan.SetRowspan( 13 );
   if( m_uiFrmLayout == 0 )
-       { oPosn.SetCol( 2 );   oSpan.SetColspan(  1 ); }
+       { oPosn.SetCol( 2 );   oSpan.SetColspan(  5 ); }
   else { oPosn.SetCol( 0 );   oSpan.SetColspan(  7 ); iFlags |= wxLEFT; }
   poSzr->Add( &m_oNbkTxtCtls, oPosn, oSpan, iFlags, iBorder );
 #ifndef LAYOUT_MNGR
--- gspiceui-v1.2.36/src/TypeDefs.hpp.orig	2018-10-26 21:27:22.000000000 +0100
+++ gspiceui-v1.2.36/src/TypeDefs.hpp	2020-04-04 17:08:19.867267505 +0100
@@ -114,7 +114,7 @@
   #define  BIN_GNETLIST  wxT("gnetlist")          // gEDA       netlist program
   #define  BIN_LSCHEM    wxT("lepton-schematic")  // Lepton EDA schematic capture program
   #define  BIN_GSCHEM    wxT("gschem")            // gEDA       schematic capture program
-  #define  BIN_GWAVE     wxT("gwave2")
+  #define  BIN_GWAVE     wxT("gwave")
   #define  BIN_GAW       wxT("gaw")
   #define  BIN_GNUCAP    wxT("gnucap")
   #define  BIN_NGSPICE   wxT("ngspice")
--- gspiceui-v1.2.36/gspiceui.1.orig	2014-01-22 04:43:56.000000000 +0000
+++ gspiceui-v1.2.36/gspiceui.1	2020-04-04 17:07:56.281268630 +0100
@@ -1,4 +1,4 @@
-.TH "gspiceui" "22" "January 2014" "Mike Waters" "User Commands"
+.TH GSPICEUI 1 "January 22, 2014" "Mike Waters" "User Commands"
 .SH "NAME"
 gspiceui \- GUI frontend for various electronic circuit simulation engines
 
@@ -58,7 +58,7 @@
 the command line.
 
 .TP 12
-NOTE : Option -s should come before -a if both are specified.
+NOTE : Option \-s should come before \-a if both are specified.
 
 .SH "REPORTING BUGS"
 Report bugs to <mwaters517_AT_gmail_DOT_com>.
