--- gst-plugins-bad-0.10.4/configure.orig	2006-12-22 09:21:12.000000000 +0000
+++ gst-plugins-bad-0.10.4/configure	2006-12-22 08:28:35.000000000 +0000
@@ -37339,7 +37339,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lfaad -lm $LIBS"
+LIBS="-lfaad -lmpeg4ip_faad -lm $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -37579,7 +37579,7 @@
 
 
     if test "x$HAVE_FAAD" = "xyes"; then
-            FAAD_LIBS="-lfaad -lm"
+            FAAD_LIBS="-lfaad -lmpeg4ip_faad -lm"
     else
       HAVE_FAAD="no"
     fi
