--- gst-plugins-good-0.10.6/ext/taglib/gstid3v2mux.cc.orig	2007-06-13 10:21:10.000000000 +0100
+++ gst-plugins-good-0.10.6/ext/taglib/gstid3v2mux.cc	2008-01-19 16:41:04.000000000 +0000
@@ -544,7 +544,7 @@
   GST_TAG_MUSICBRAINZ_ALBUMARTISTID, add_musicbrainz_tag, "\002"}, {
   GST_TAG_MUSICBRAINZ_TRMID, add_musicbrainz_tag, "\003"}, {
   GST_TAG_MUSICBRAINZ_TRACKID, add_unique_file_id_tag, ""}, {
-  GST_TAG_MUSICBRAINZ_SORTNAME, add_text_tag, "TSOP"}, {
+  GST_TAG_ARTIST_SORTNAME, add_text_tag, "TSOP"}, {
   GST_TAG_TRACK_NUMBER, add_count_or_num_tag, "TRCK"}, {
   GST_TAG_TRACK_COUNT, add_count_or_num_tag, "TRCK"}, {
   GST_TAG_ALBUM_VOLUME_NUMBER, add_count_or_num_tag, "TPOS"}, {
