--- gthumb-2.8.0/src/rotation-utils.h.orig	2005-12-24 13:08:28.000000000 +0000
+++ gthumb-2.8.0/src/rotation-utils.h	2006-11-25 17:35:47.000000000 +0000
@@ -54,6 +54,7 @@
 
 void			update_orientation_field       (const char      *filename, 
 							RotationData    *rot_data);
+void			reset_orientation_field	       (const char *path);
 gboolean		swap_fields                    (ExifContent     *content,
 							ExifTag          tag1,
 							ExifTag          tag2);
