--- gtk+-2.10.5/gtk/Makefile.in.orig	2006-10-02 06:49:35.000000000 +0100
+++ gtk+-2.10.5/gtk/Makefile.in	2006-10-02 21:18:22.000000000 +0100
@@ -2935,7 +2935,7 @@
 	   && $(LN_S) gtk-goto-first-ltr.png gtk-goto-last-rtl.png		\
 	   && $(RM) gtk-media-forward-rtl.png 					\
 	   && $(LN_S) gtk-media-rewind-ltr.png gtk-media-forward-rtl.png	\
-	   && $(RM) gtk-mdedia-next-rtl.png					\
+	   && $(RM) gtk-media-next-rtl.png					\
 	   && $(LN_S) gtk-media-previous-ltr.png gtk-media-next-rtl.png		\
 	   && $(RM) gtk-media-previous-rtl.png 					\
 	   && $(LN_S) gtk-media-next-ltr.png gtk-media-previous-rtl.png		\
