--- gtk-wave-cleaner-0.22-04/gwc.c.orig	2019-04-16 06:27:31.000000000 +0100
+++ gtk-wave-cleaner-0.22-04/gwc.c	2020-06-10 22:15:18.366919172 +0100
@@ -2543,8 +2543,8 @@
   { "Filter", "filter_icon", "Apply DSP Frequency Filters", NULL, "lowpass, highpass, notch or bandpass biquad filtering", G_CALLBACK(filter_cb) },
   { "PinkNoise", "pinknoise_icon", "Generate Pink Noise", NULL, "Replace current view or selection with pink noise", G_CALLBACK(pinknoise_cb) },
   { "Amplify", "amplify_icon", "Amplify", NULL, "Amplify or attenuate the current view or selection", G_CALLBACK(amplify) },
-  { "DeclickStrong", "declick_icon", "Declick Strong", NULL, "Automatically find and repair pops/clicks in current view or selection", G_CALLBACK(declick) },
-  { "DeclickWeak", "declick_w_icon", "Declick Weak", NULL, "Automatically find and repair weaker pops/clicks in current view or selection", G_CALLBACK(declick_weak) },
+  { "DeclickStrong", "declick_icon", "Declick Strong", "S", "Automatically find and repair pops/clicks in current view or selection", G_CALLBACK(declick) },
+  { "DeclickWeak", "declick_w_icon", "Declick Weak", "W", "Automatically find and repair weaker pops/clicks in current view or selection", G_CALLBACK(declick_weak) },
   { "DeclickManual", "declick_m_icon", "Declick Manual", NULL, "Apply LSAR signal estimation to repair a single selected/viewed click", G_CALLBACK(manual_declick) },
   { "Decrackle", "decrackle_icon", "Decrackle", "C", "Remove crackle from old, deteriorated vinyl", G_CALLBACK(decrackle) },
   { "Estimate", "estimate_icon", "Estimate", NULL, "Estimate signal (> 300 samples) in current view or selection", G_CALLBACK(estimate) },
