--- gurlchecker-0.10.2/src/ucclam.c.orig	2007-04-25 06:41:44.000000000 +0100
+++ gurlchecker-0.10.2/src/ucclam.c	2008-04-15 22:40:47.000000000 +0100
@@ -34,7 +34,7 @@
   clam->limits.maxfiles = 1;
   clam->limits.maxfilesize = 10 * 1048576;
   clam->limits.maxreclevel = 1;
-  clam->limits.maxratio = 200;
+  clam->limits.maxscansize = 20 * 1048576;
   clam->limits.archivememlim = 0;
   if (cl_loaddbdir (cl_retdbdir (), &clam->root, &clam->no) ||
       cl_build (clam->root))
--- gurlchecker-0.10.2/configure.orig	2007-12-13 17:06:53.000000000 +0000
+++ gurlchecker-0.10.2/configure	2007-12-22 20:31:44.000000000 +0000
@@ -23602,7 +23602,7 @@
 _ACEOF
 
 	      have_tidy=yes
-	      TIDY_FLAGS="$TIDY_FLAGS -DHAVE_TIDY_H_IN_SUBDIR -DENABLE_TIDY"
+	      TIDY_FLAGS="$TIDY_FLAGS -DENABLE_TIDY"
         TIDY_LIBS="-ltidy"
         LDFLAGS="$LDFLAGS $want_tidy_path"
 
