--- gwc-0.21-08/audio_alsa.c.orig	2007-02-25 17:46:59.000000000 +0000
+++ gwc-0.21-08/audio_alsa.c	2007-11-08 11:53:34.000000000 +0000
@@ -152,6 +152,9 @@
 /* recover underrun and suspend */
 static int recover_snd_handle(int err)
 {
+    if (handle == NULL)
+	return err;
+
     if (err == -EPIPE) { /* underrun */
         err = snd_pcm_prepare(handle);
         if (err < 0)
@@ -240,6 +243,7 @@
 
 	if (recover_snd_handle(err) < 0) {
 	    fprintf(stderr, "audio_device_processed_bytes: could not recover handle\n");
+	    return 0;
 	}
 
 	return snd_pcm_frames_to_bytes(handle, written_frames);
