--- gxine-0.5.905/Makefile.in.orig	2010-01-01 01:02:27.000000000 +0000
+++ gxine-0.5.905/Makefile.in	2010-02-14 23:06:27.000000000 +0000
@@ -887,7 +887,7 @@
 	$(MAKE) -C $(top_builddir)/misc/po update-po
 
 @XINE_LIST_TRUE@gxine.desktop: gxine.desktop.in @XINE_LIST@
-@XINE_LIST_TRUE@	exec > $@; $(XINE_MIME_PROCESS) $<; @XINE_LIST@; cat $(srcdir)/mime.extra
+@XINE_LIST_TRUE@	exec > $@; $(XINE_MIME_PROCESS) $<; @XINE_LIST@
 @XINE_LIST_FALSE@gxine.desktop: gxine.desktop.in $(MIME)
 @XINE_LIST_FALSE@	exec > $@; \
 @XINE_LIST_FALSE@	$(XINE_MIME_PROCESS) $<; \
