--- hal-0.5.7.1/hal.conf.in.orig	2006-01-10 19:03:45.000000000 +0000
+++ hal-0.5.7.1/hal.conf.in	2006-10-18 20:28:35.000000000 +0100
@@ -49,5 +49,12 @@
     <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
   </policy>
 
+  <policy group="100">
+    <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
+    <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
+    <allow send_interface="org.freedesktop.Hal.Device.Volume"/>
+    <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
+  </policy>
+
 </busconfig>
 
