Index: tools/hal-system-power-set-power-save
===================================================================
RCS file: /cvs/hal/hal/tools/hal-system-power-set-power-save,v
retrieving revision 1.4
diff -u -p -r1.4 hal-system-power-set-power-save
--- hal-0.5.7/tools/hal-system-power-set-power-save	29 Nov 2005 17:47:54 -0000	1.4
+++ hal-0.5.7/tools/hal-system-power-set-power-save	24 Feb 2006 16:45:15 -0000
@@ -25,10 +25,10 @@ if [ -f /etc/altlinux-release ] || [ -f 
 elif [ -f /etc/redhat-release ] || [ -f /etc/fedora-release ] ; then
 	if [ -x "/usr/sbin/pm-powersave" ] ; then
 		if [ $value = "true" ]; then
-			/usr/sbin/pm-powersave on
+			/usr/sbin/pm-powersave true
 			RET=$?
 		elif [ $value = "false" ]; then
-			/usr/sbin/pm-powersave off
+			/usr/sbin/pm-powersave false
 			RET=$?
 		fi
 	else
