--- hal-0.5.8.1/hal.conf.in.orig	2006-09-19 21:23:25.000000000 +0100
+++ hal-0.5.8.1/hal.conf.in	2006-11-14 22:11:38.000000000 +0000
@@ -63,5 +63,13 @@
     <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
   </policy>
 
+  <policy group="100">
+    <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
+    <allow send_interface="org.freedesktop.Hal.Device.VideoAdapterPM"/>
+    <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
+    <allow send_interface="org.freedesktop.Hal.Device.Volume"/>
+    <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
+  </policy>
+
 </busconfig>
 
