--- HandBrake-0.9.9/gtk/config.h.in.orig	2013-12-28 16:47:44.000000000 +0000
+++ HandBrake-0.9.9/gtk/config.h.in	2013-12-28 16:57:26.244586107 +0000
@@ -0,0 +1,4 @@
+
+
+/* gettext package name */
+#undef GETTEXT_PACKAGE
--- HandBrake-0.9.9/libhb/dvdnav.c.orig	2013-03-09 22:27:58.000000000 +0000
+++ HandBrake-0.9.9/libhb/dvdnav.c	2014-08-24 22:06:05.980843322 +0100
@@ -387,7 +387,7 @@
                     "is not before last_sector (0x%x) - skipping title", i,
                     ifo->vts_c_adt->cell_adr_table[i].start_sector,
                     ifo->vts_c_adt->cell_adr_table[i].last_sector );
-            goto fail;
+//            goto fail;
         }
     }
 
--- HandBrake-0.9.9/libhb/dvd.c.orig	2013-03-09 22:27:58.000000000 +0000
+++ HandBrake-0.9.9/libhb/dvd.c	2014-08-24 22:06:23.764843388 +0100
@@ -241,7 +241,7 @@
                     "is not before last_sector (0x%x) - skipping title", i,
                     vts->vts_c_adt->cell_adr_table[i].start_sector,
                     vts->vts_c_adt->cell_adr_table[i].last_sector );
-            goto fail;
+//            goto fail;
         }
     }
 
