--- hdf5-1.6.4/fortran/config/linux-gnulibc1.BAD	2005-04-03 12:55:55.000000000 -0500
+++ hdf5-1.6.4/fortran/config/linux-gnulibc1	2005-04-03 12:56:50.000000000 -0500
@@ -25,7 +25,11 @@
 # The default Fortran 90 compiler
 if test "X-" = "X-$F9X"; then
     case $CC_BASENAME in
-        gcc*|pgcc*)
+        gcc*)
+            F9X=gfortran
+            F9X_BASENAME=gfortran    
+            ;;
+        pgcc*)
             F9X=pgf90
             F9X_BASENAME=pgf90
             ;;
