--- hdf5-1.6.4/config/commence.in.BAD	2005-04-03 13:52:00.000000000 -0500
+++ hdf5-1.6.4/config/commence.in	2005-04-03 13:52:53.000000000 -0500
@@ -35,11 +35,11 @@
 TRACE=:
 
 ## Installation points
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-bindir=@bindir@
-libdir=@libdir@
-includedir=@includedir@
-docdir=@exec_prefix@/doc
+prefix=$(DESTDIR)@prefix@
+exec_prefix=$(DESTDIR)@exec_prefix@
+bindir=$(DESTDIR)@bindir@
+libdir=$(DESTDIR)@libdir@
+includedir=$(DESTDIR)@includedir@
+docdir=$(DESTDIR)@exec_prefix@/doc
 PUB_LIB=$(LIB)
 
--- hdf5-1.6.4/fortran/config/commence.in.BAD	2005-04-03 14:08:27.000000000 -0500
+++ hdf5-1.6.4/fortran/config/commence.in	2005-04-03 14:08:48.000000000 -0500
@@ -40,11 +40,11 @@
 TRACE=:
 
 ## Installation points
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-bindir=@bindir@
-libdir=@libdir@
-includedir=@includedir@
-docdir=@exec_prefix@/doc
+prefix=$(DESTDIR)@prefix@
+exec_prefix=$(DESTDIR)@exec_prefix@
+bindir=$(DESTDIR)@bindir@
+libdir=$(DESTDIR)@libdir@
+includedir=$(DESTDIR)@includedir@
+docdir=$(DESTDIR)@exec_prefix@/doc
 PUB_LIB=$(LIB)
 
--- hdf5-1.6.4/c++/config/commence.in.BAD	2005-04-03 14:07:10.000000000 -0500
+++ hdf5-1.6.4/c++/config/commence.in	2005-04-03 14:07:35.000000000 -0500
@@ -32,11 +32,11 @@
 TRACE=:
 
 ## Installation points
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-bindir=@bindir@
-libdir=@libdir@
-includedir=@includedir@
-docdir=@exec_prefix@/doc
+prefix=$(DESTDIR)@prefix@
+exec_prefix=$(DESTDIR)@exec_prefix@
+bindir=$(DESTDIR)@bindir@
+libdir=$(DESTDIR)@libdir@
+includedir=$(DESTDIR)@includedir@
+docdir=$(DESTDIR)@exec_prefix@/doc
 PUB_LIB=$(LIB)
 
