--- hdf5-1.6.5/bin/ltmain.sh.sort	2007-03-02 16:20:05.000000000 -0700
+++ hdf5-1.6.5/bin/ltmain.sh	2007-03-02 16:20:12.000000000 -0700
@@ -3329,7 +3329,7 @@
 	    fi
 
 	    # Try sorting and uniquifying the output.
-	    if grep -v "^: " < "$nlist" | sort +2 | uniq > "$nlist"S; then
+	    if grep -v "^: " < "$nlist" | sort -k 2 | uniq > "$nlist"S; then
 	      :
 	    else
 	      grep -v "^: " < "$nlist" > "$nlist"S
