--- hdf5-1.6.5/config/gnu-flags.orig	2005-10-28 14:15:26.000000000 -0600
+++ hdf5-1.6.5/config/gnu-flags	2006-03-13 10:59:58.000000000 -0700
@@ -107,25 +107,6 @@
   gcc*|egcs*|pgcc*)
     # Architecture-specific flags
     arch=
-    case "$host_os-$host_cpu" in
-        # FreeBSD sets the information from "uname -m" to the general machine
-        # architecture, not the specific CPU for the machine, so even our
-        # Pentium II Xeon server is set to "i386".  Once we know we are on a FreeBSD
-        # machine, use the "sysctl" command to get the CPU hardware model.
-        freebsd*-i386)
-            host_cpu_model=`sysctl -n hw.model`
-            case "$host_cpu_model" in
-                # Hmm.. this might not catch Celerons, but it won't hurt them either...
-                *Pro*|*II*|*III*|*IV*|*Athlon*)
-                    arch="-march=i686"
-                ;;
-            esac
-            ;;
-
-        *-i686)
-            arch="-march=i686"
-            ;;
-    esac
 
     # Host-specific flags
     case "`hostname`" in
