--- hp2xx-3.4.4/sources/Makefile.orig	2003-06-21 16:16:34.000000000 +0100
+++ hp2xx-3.4.4/sources/Makefile	2021-05-03 13:20:46.229355098 +0100
@@ -331,14 +331,14 @@
 
 install-bin:	$(PROGRAM)
 		-$(STRIP) $(PROGRAM)
-		-$(CP) $(PROGRAM) $(bindir)
-		-$(CHMOD) 755 $(bindir)/$(PROGRAM)
+		-$(CP) $(PROGRAM) $(DESTDIR)$(bindir)
+		-$(CHMOD) 755 $(DESTDIR)$(bindir)/$(PROGRAM)
 
 install-info:	hp2xx.info
-		-$(CP) hp2xx.info $(infodir)
-		-$(CHMOD) 644 $(infodir)/hp2xx.info
+		-$(CP) hp2xx.info $(DESTDIR)$(infodir)
+		-$(CHMOD) 644 $(DESTDIR)$(infodir)/hp2xx.info
 
 install-man:	../doc/hp2xx.1
-		-$(CP) ../doc/hp2xx.1 $(man1dir)
-		-$(CHMOD) 644 $(man1dir)/hp2xx.1
+		-$(CP) ../doc/hp2xx.1 $(DESTDIR)$(man1dir)
+		-$(CHMOD) 644 $(DESTDIR)$(man1dir)/hp2xx.1
 
