--- hp2xx-5e1f411/src/Makefile.am.orig	2021-05-03 22:10:11.000000000 +0100
+++ hp2xx-5e1f411/src/Makefile.am	2021-05-04 18:02:12.391125265 +0100
@@ -1,6 +1,6 @@
 # what flags you want to pass to the C compiler & linker
-AM_CFLAGS = --pedantic -Wall -std=c99 -O2
-AM_LDFLAGS =
+AM_CFLAGS = --pedantic -Wall -std=c99 -O2 -fcommon
+AM_LDFLAGS = -lm
 
 # this lists the binaries to produce, the (non-PHONY, binary) targets in
 # the previous manual Makefile
