Description: upstream patch
Author: Christian T. Steigies <cts@debian.org>
Last-Update: 2018-12-04
--- hp2xx.orig/sources/hpgl.c~	2003-06-21 19:54:45.000000000 +0200
+++ hp2xx/sources/hpgl.c	2010-02-10 14:22:10.585264000 +0100
@@ -2622,9 +2622,6 @@
 					fprintf(stderr,
 						"only fixed fonts available\n");
 				break;
-			case 3:	/* font pitch */
-			case 4:	/* font height */
-			case 5:	/* posture */
 			case 6:	/* stroke weight */
 				if (read_float(&ftmp, hd))
 					par_err_exit(2, cmd, hd);
@@ -2636,6 +2633,9 @@
 					tp->astrokewidth = 0.11 + ftmp / 70.;	/* 0.01 ... 0.21 mm */
 				}
 				break;
+			case 3:	/* font pitch */
+			case 4:	/* font height */
+			case 5:	/* posture */
 			case 7:	/* typeface */
 				if (read_float(&csfont, hd))
 					par_err_exit(2, cmd, hd);
@@ -3720,9 +3720,6 @@
 					fprintf(stderr,
 						"only fixed fonts available\n");
 				break;
-			case 3:	/* font pitch */
-			case 4:	/* font height */
-			case 5:	/* posture */
 			case 6:	/* stroke weight */
 				if (read_float(&ftmp, hd))
 					par_err_exit(2, cmd, hd);
@@ -3734,6 +3731,9 @@
 					tp->sstrokewidth = 0.11 + ftmp / 70.;	/* 0.01 ... 0.21 mm */
 				}
 				break;
+			case 3:	/* font pitch */
+			case 4:	/* font height */
+			case 5:	/* posture */
 			case 7:	/* typeface */
 				if (read_float(&csfont, hd))
 					par_err_exit(2, cmd, hd);
