--- hplip-3.16.9/prnt/hpijs/hpcupsfax.cpp.orig	2016-09-19 14:03:05.000000000 +0100
+++ hplip-3.16.9/prnt/hpijs/hpcupsfax.cpp	2016-09-22 20:53:42.521092757 +0100
@@ -45,6 +45,7 @@
 #include <cups/cups.h>
 #include <cups/raster.h>
 #include <string>
+#include <cups/ppd.h>
 #ifdef FALSE
 #undef FALSE
 #endif
--- hplip-3.16.9/prnt/hpcups/HPCupsFilter.h.orig	2016-09-19 14:03:17.000000000 +0100
+++ hplip-3.16.9/prnt/hpcups/HPCupsFilter.h	2016-09-22 20:52:05.465092400 +0100
@@ -31,6 +31,8 @@
 #ifndef HP_CUPSFILTER_H
 #define HP_CUPSFILTER_H
 
+#include <cups/ppd.h>
+
 #include "CommonDefinitions.h"
 #include "SystemServices.h"
 #include "Compressor.h"
