--- hplip-3.18.7/Makefile.in.orig	2018-08-03 22:44:10.642108321 +0100
+++ hplip-3.18.7/Makefile.in	2018-08-03 22:49:48.972201839 +0100
@@ -9329,14 +9329,14 @@
 
 install-data-hook:
 @HPLIP_BUILD_TRUE@	if [ \( "$(UNAME)" = "x86_64" -a  -d "$(libdir)/" \) ]; then \
-@HPLIP_BUILD_TRUE@		cp prnt/hpcups/libImageProcessor-x86_64.so $(libdir)/ ; \
-@HPLIP_BUILD_TRUE@		chmod 775 $(libdir)/libImageProcessor-x86_64.so ; \
-@HPLIP_BUILD_TRUE@		ln -sf $(libdir)/libImageProcessor-x86_64.so $(libdir)/libImageProcessor.so ; \
+@HPLIP_BUILD_TRUE@		cp prnt/hpcups/libImageProcessor-x86_64.so $(DESTDIR)$(libdir)/ ; \
+@HPLIP_BUILD_TRUE@		chmod 775 $(DESTDIR)$(libdir)/libImageProcessor-x86_64.so ; \
+@HPLIP_BUILD_TRUE@		ln -sf $(libdir)/libImageProcessor-x86_64.so $(DESTDIR)$(libdir)/libImageProcessor.so ; \
 @HPLIP_BUILD_TRUE@	fi; \
 @HPLIP_BUILD_TRUE@	if [ \( \( "$(UNAME)" = "i686" -o "$(UNAME)" = "i386" \) -a -d "$(libdir)/" \) ]; then \
-@HPLIP_BUILD_TRUE@		cp prnt/hpcups/libImageProcessor-x86_32.so $(libdir)/ ; \
-@HPLIP_BUILD_TRUE@		chmod 775 $(libdir)/libImageProcessor-x86_32.so ; \
-@HPLIP_BUILD_TRUE@		ln -sf $(libdir)/libImageProcessor-x86_32.so $(libdir)/libImageProcessor.so ; \
+@HPLIP_BUILD_TRUE@		cp prnt/hpcups/libImageProcessor-x86_32.so $(DESTDIR)$(libdir)/ ; \
+@HPLIP_BUILD_TRUE@		chmod 775 $(DESTDIR)$(libdir)/libImageProcessor-x86_32.so ; \
+@HPLIP_BUILD_TRUE@		ln -sf $(libdir)/libImageProcessor-x86_32.so $(DESTDIR)$(libdir)/libImageProcessor.so ; \
 @HPLIP_BUILD_TRUE@	fi
 #	   If scanner build, add hpaio entry to sane dll.conf.
 @HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@	if [ "$(scan_build)" = "yes" ]; then \
@@ -9349,14 +9349,14 @@
 @HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@		  echo hpaio >>$(DESTDIR)/etc/sane.d/dll.conf ; \
 @HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@	   fi; \
 @HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@ 	   if [ \( "$(UNAME)" = "x86_64" -a  -d "$(libdir)/x86_64-linux-gnu/sane" \) ]; then \
-@HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@	  	ln -sf $(libdir)/sane/libsane-hpaio.so $(libdir)/x86_64-linux-gnu/sane/ ; \
-@HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@	  	ln -sf $(libdir)/sane/libsane-hpaio.so.1 $(libdir)/x86_64-linux-gnu/sane/ ; \
+@HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@	  	ln -sf $(libdir)/sane/libsane-hpaio.so $(DESTDIR)$(libdir)/x86_64-linux-gnu/sane/ ; \
+@HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@	  	ln -sf $(libdir)/sane/libsane-hpaio.so.1 $(DESTDIR)$(libdir)/x86_64-linux-gnu/sane/ ; \
 @HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@	   fi; \
 @HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@	   if [ \( \( "$(UNAME)" = "i686" -o "$(UNAME)" = "i386" \) -a -d "$(libdir)/i386-linux-gnu" \) ]; then \
-@HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@	      ln -sf $(libdir)/libhpmud.so.0.0.6  $(libdir)/i386-linux-gnu/libhpmud.so ; \
-@HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@	      ln -sf $(libdir)/libhpmud.so.0.0.6  $(libdir)/i386-linux-gnu/libhpmud.so.0 ; \
-@HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@	      ln -sf $(libdir)/sane/libsane-hpaio.so.1.0.0 $(libdir)/i386-linux-gnu/sane/libsane-hpaio.so.1 ; \
-@HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@	      ln -sf $(libdir)/sane/libsane-hpaio.so.1.0.0 $(libdir)/i386-linux-gnu/sane/libsane-hpaio.so ; \
+@HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@	      ln -sf $(libdir)/libhpmud.so.0.0.6  $(DESTDIR)$(libdir)/i386-linux-gnu/libhpmud.so ; \
+@HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@	      ln -sf $(libdir)/libhpmud.so.0.0.6  $(DESTDIR)$(libdir)/i386-linux-gnu/libhpmud.so.0 ; \
+@HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@	      ln -sf $(libdir)/sane/libsane-hpaio.so.1.0.0 $(DESTDIR)$(libdir)/i386-linux-gnu/sane/libsane-hpaio.so.1 ; \
+@HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@	      ln -sf $(libdir)/sane/libsane-hpaio.so.1.0.0 $(DESTDIR)$(libdir)/i386-linux-gnu/sane/libsane-hpaio.so ; \
 @HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@	   fi \
 @HPLIP_BUILD_TRUE@@HPLIP_CLASS_DRIVER_FALSE@	fi
 #	   Create hp-xxx commands in bindir.
--- hplip-3.18.7/prnt/hpijs/hpcupsfax.cpp.orig	2018-07-15 21:08:01.000000000 +0100
+++ hplip-3.18.7/prnt/hpijs/hpcupsfax.cpp	2018-08-01 19:12:43.100779336 +0100
@@ -45,6 +45,7 @@
 #include <cups/cups.h>
 #include <cups/raster.h>
 #include <string>
+#include <cups/ppd.h>
 #ifdef FALSE
 #undef FALSE
 #endif
--- hplip-3.18.7/prnt/hpcups/HPCupsFilter.h.orig	2018-07-15 21:08:03.000000000 +0100
+++ hplip-3.18.7/prnt/hpcups/HPCupsFilter.h	2018-08-01 19:12:43.100779336 +0100
@@ -31,6 +31,8 @@
 #ifndef HP_CUPSFILTER_H
 #define HP_CUPSFILTER_H
 
+#include <cups/ppd.h>
+
 #include "CommonDefinitions.h"
 #include "SystemServices.h"
 #include "Compressor.h"
--- hplip-3.18.7/installer/distros.dat.orig	2018-07-15 21:08:17.000000000 +0100
+++ hplip-3.18.7/installer/distros.dat	2018-08-01 19:12:43.087779335 +0100
@@ -2173,7 +2173,7 @@
 # ****************************************
 [slackware]
 index=8
-versions=9.0,9.1,10.0,10.1,10.2,11,12,12.1
+versions=9.0,9.1,10.0,10.1,10.2,11,12,12.1,14.0
 display_name=Slackware Linux
 alt_names=
 display=0
@@ -2189,6 +2189,7 @@
 ppd_dir=/usr/share/ppd/HP
 fix_ppd_symlink=0
 drv_dir=/usr/share/cups/drv/HP
+add_user_to_group=-a -Glp
 
 # ********************
 [slackware:9.0]
@@ -2344,6 +2345,26 @@
 release_date=
 notes=
 ppd_install=ppd
+udev_mode_fix=1
+ppd_dir=/usr/share/ppd/HP
+fix_ppd_symlink=0
+drv_dir=/usr/share/cups/drv/HP
+ui_toolkit=none
+
+# ********************
+[slackware:14.0]
+code_name=
+supported=0
+scan_supported=0
+fax_supported=0
+pcard_supported=0
+network_supported=0
+parallel_supported=0
+usb_supported=0
+packaged_version=3.12.9
+release_date=
+notes=
+ppd_install=ppd
 udev_mode_fix=1
 ppd_dir=/usr/share/ppd/HP
 fix_ppd_symlink=0
--- hplip-3.18.7/base/utils.py.orig	2018-07-15 21:08:37.000000000 +0100
+++ hplip-3.18.7/base/utils.py	2018-08-01 19:12:43.086779335 +0100
@@ -284,8 +284,8 @@
                         except ValueError:
                             yield fullname
 
-        #if os.path.islink(fullname):
-        #    fullname = os.path.realpath(os.readlink(fullname))
+        if os.path.islink(fullname):
+            fullname = os.path.realpath(os.readlink(fullname))
 
         if recurse and os.path.isdir(fullname): # or os.path.islink(fullname):
             for f in walkFiles(fullname, recurse, abs_paths, return_folders, pattern, path):
