--- HTK-3.4.1/HLMTools/Makefile.in.orig	2009-03-11 10:07:18.000000000 +0000
+++ HTK-3.4.1/HLMTools/Makefile.in	2013-05-21 21:57:39.478100265 +0100
@@ -74,6 +74,6 @@
 	for program in $(PROGS) ; do $(INSTALL) -m 755 $${program}@BINARY_EXTENSION@ $(bindir) ; done
 
 mkinstalldir:
-        if [ ! -d $(bindir) -a X_@TRADHTK@ = X_yes ] ; then mkdir -p $(bindir) ; fi
+	if [ ! -d $(bindir) -a X_@TRADHTK@ = X_yes ] ; then mkdir -p $(bindir) ; fi
 
 .PHONY: all strip clean cleanup distclean install mkinstalldir
--- HTK-3.4.1/configure.orig	2013-05-21 21:59:12.781090573 +0100
+++ HTK-3.4.1/configure	2013-05-21 21:55:53.212111304 +0100
@@ -5504,7 +5504,7 @@
 
 case "$host" in
 		*x86_64*linux*)
-				CFLAGS="-m32 -ansi -D_SVID_SOURCE -DOSS_AUDIO -D'ARCH=\"$host_cpu\"' $CFLAGS"
+				CFLAGS=" -ansi -D_SVID_SOURCE -DOSS_AUDIO -D'ARCH=\"$host_cpu\"' $CFLAGS"
 				LDFLAGS="-L/usr/X11R6/lib $LDFLAGS"
 				ARCH=linux
 				trad_bin_dir=linux
