--- html2text-2.0.0/html2text.cpp.orig	2020-06-18 07:11:22.000000000 +0100
+++ html2text-2.0.0/html2text.cpp	2020-09-28 19:18:57.542487515 +0100
@@ -147,6 +147,8 @@
 			extarg = &to_encoding;
 		} else if (!strcmp(arg, "-ascii")) {
 			to_encoding = "ASCII//TRANSLIT";  /* create things like (c) */
+		} else if (!strcmp(arg, "-utf8")) {
+			/* ignore for bacward compatibility */
 		} else {
 			std::cerr
 				<< "Unrecognized command line option \""
