--- httping-2.3.1/Makefile.orig	2013-05-09 10:34:54.000000000 +0100
+++ httping-2.3.1/Makefile	2013-05-09 21:24:36.814594659 +0100
@@ -109,13 +109,15 @@
 	$(INSTALLBIN) $(TARGET) $(DESTDIR)/$(BINDIR)
 	$(INSTALLDIR) $(DESTDIR)/$(MANDIR)/man1
 	$(INSTALLMAN) $(MAN_EN) $(DESTDIR)/$(MANDIR)/man1
+	$(INSTALLDIR) $(DESTDIR)/$(MANDIR)/nl/man1
 	$(INSTALLMAN) $(MAN_NL) $(DESTDIR)/$(MANDIR)/nl/man1
 	$(INSTALLDIR) $(DESTDIR)/$(DOCDIR)
 	$(INSTALLDOC) $(DOCS) $(DESTDIR)/$(DOCDIR)
 ifneq ($(DEBUG),yes)
 	$(STRIP) $(DESTDIR)/$(BINDIR)/$(TARGET)
 endif
-	cp nl.mo $(PREFIX)/share/locale/nl/LC_MESSAGES/httping.mo
+	$(INSTALLDIR) $(DESTDIR)/$(PREFIX)/share/locale/nl/LC_MESSAGES
+	cp nl.mo $(DESTDIR)/$(PREFIX)/share/locale/nl/LC_MESSAGES/httping.mo
 
 makefile.inc:
 	./configure
