--- hugin-2019.0.0/CMakeLists.txt.orig	2019-04-22 10:57:57.545598822 +0100
+++ hugin-2019.0.0/CMakeLists.txt	2019-04-22 10:58:46.473596487 +0100
@@ -342,7 +342,10 @@
   ENDIF()
 ENDIF(WIN32)
 
+set(BUILD_SHARED_LIBS ON)
 FIND_PACKAGE(Boost 1.47 COMPONENTS filesystem system)
+set(BUILD_SHARED_LIBS OFF)
+SET(Boost_LIBRARIES Boost::filesystem Boost::system ${Boost_LIBRARIES})
 
 IF (NOT Boost_FOUND)
   IF(WIN32)
--- hugin-2019.0.0/src/hugin_base/panodata/SrcPanoImage.cpp.orig	2019-02-16 08:18:55.000000000 +0000
+++ hugin-2019.0.0/src/hugin_base/panodata/SrcPanoImage.cpp	2019-07-14 20:34:26.979752462 +0100
@@ -43,6 +43,7 @@
 #include <exiv2/exif.hpp>
 #include <exiv2/image.hpp>
 #include <exiv2/easyaccess.hpp>
+#include <exiv2/error.hpp>
 #include <lensdb/LensDB.h>
 #include "Exiv2Helper.h"
 
--- hugin-2019.0.0/src/hugin1/hugin/RawImport.cpp.orig	2019-02-02 10:11:01.000000000 +0000
+++ hugin-2019.0.0/src/hugin1/hugin/RawImport.cpp	2019-07-14 20:41:14.192733032 +0100
@@ -36,6 +36,7 @@
 #define HAVE_PID_T 1
 #endif
 #include <exiv2/exif.hpp>
+#include <exiv2/error.hpp>
 #include <exiv2/image.hpp>
 #include <exiv2/easyaccess.hpp>
 #include <exiv2/xmpsidecar.hpp>
