--- hugin-2025.0.1/src/hugin_script_interface/hpi_classes.cpp.orig	2015-05-10 07:52:11.000000000 +0100
+++ hugin-2025.0.1/src/hugin_script_interface/hpi_classes.cpp	2026-09-06 10:37:16.497026192 +0100
@@ -112,7 +112,7 @@
         if ( pValue != NULL )
         {
             // the function should have returned an integer
-            result = PyInt_AsLong ( pValue ) ;
+            result = PyLong_AsLong ( pValue ) ;
             Py_DECREF(pValue);
         }
     }
