--- hugin-2023.0.0/CMakeModules/FindLAPACK.cmake.orig	2020-08-22 09:25:19.000000000 +0100
+++ hugin-2023.0.0/CMakeModules/FindLAPACK.cmake	2023-11-18 20:24:45.115565457 +0000
@@ -8,14 +8,14 @@
 #  LAPACK_FOUND, If != "YES", do not try to use PANO13.
 
 FIND_LIBRARY(LAPACK_LAPACK_LIBRARY
-  NAMES lapack
+  NAMES openblas
   PATHS ${SYSTEM_LIB_DIRS}
         ${SOURCE_BASE_DIR}/
         ${SOURCE_BASE_DIR}/lapack
   )
 
 FIND_LIBRARY(LAPACK_BLAS_LIBRARY
-  NAMES blas openblas
+  NAMES openblas
   PATHS ${SYSTEM_LIB_DIRS}
         ${SOURCE_BASE_DIR}/
         ${SOURCE_BASE_DIR}/lapack
