--- hugin-2023.0.0/CMakeModules/FindLAPACK.cmake.orig	2020-08-22 09:25:19.000000000 +0100
+++ hugin-2023.0.0/CMakeModules/FindLAPACK.cmake	2023-11-18 20:24:45.115565457 +0000
@@ -8,14 +8,14 @@
 #  LAPACK_FOUND, If != "YES", do not try to use PANO13.
 
 FIND_LIBRARY(LAPACK_LAPACK_LIBRARY
-  NAMES lapack
+  NAMES openblas
   PATHS ${SYSTEM_LIB_DIRS}
         ${SOURCE_BASE_DIR}/
         ${SOURCE_BASE_DIR}/lapack
   )
 
 FIND_LIBRARY(LAPACK_BLAS_LIBRARY
-  NAMES blas openblas
+  NAMES openblas
   PATHS ${SYSTEM_LIB_DIRS}
         ${SOURCE_BASE_DIR}/
         ${SOURCE_BASE_DIR}/lapack
--- hugin-2023.0.0/src/hugin_base/hugin_utils/filesystem.h.orig	2023-05-08 19:19:14.000000000 +0100
+++ hugin-2023.0.0/src/hugin_base/hugin_utils/filesystem.h	2024-04-17 18:31:00.008097016 +0100
@@ -64,6 +64,6 @@
     #endif
     #include <boost/filesystem.hpp>
     namespace fs = boost::filesystem;
-    #define OVERWRITE_EXISTING boost::filesystem::copy_option::overwrite_if_exists
+    #define OVERWRITE_EXISTING boost::filesystem::copy_options::overwrite_existing
 #endif
 #endif // _HUGIN_UTILS_FILESYSTEM_H
