--- hugin-2023.0.0/CMakeModules/FindLAPACK.cmake.orig	2020-08-22 09:25:19.000000000 +0100
+++ hugin-2023.0.0/CMakeModules/FindLAPACK.cmake	2023-11-18 20:24:45.115565457 +0000
@@ -8,14 +8,14 @@
 #  LAPACK_FOUND, If != "YES", do not try to use PANO13.
 
 FIND_LIBRARY(LAPACK_LAPACK_LIBRARY
-  NAMES lapack
+  NAMES openblas
   PATHS ${SYSTEM_LIB_DIRS}
         ${SOURCE_BASE_DIR}/
         ${SOURCE_BASE_DIR}/lapack
   )
 
 FIND_LIBRARY(LAPACK_BLAS_LIBRARY
-  NAMES blas openblas
+  NAMES openblas
   PATHS ${SYSTEM_LIB_DIRS}
         ${SOURCE_BASE_DIR}/
         ${SOURCE_BASE_DIR}/lapack
--- hugin-2024.0.1/CMakeLists.txt.orig	2024-11-19 15:51:02.000000000 +0000
+++ hugin-2024.0.1/CMakeLists.txt	2025-09-07 21:39:00.892230365 +0100
@@ -373,7 +373,7 @@
   ENDIF()
 ENDIF(WIN32)
 
-FIND_PACKAGE(Boost 1.47 COMPONENTS filesystem system)
+FIND_PACKAGE(Boost 1.47 COMPONENTS filesystem)
 
 IF (NOT Boost_FOUND)
   IF(WIN32)
