--- hugin-2019.0.0/CMakeLists.txt.orig	2019-04-22 10:57:57.545598822 +0100
+++ hugin-2019.0.0/CMakeLists.txt	2019-04-22 10:58:46.473596487 +0100
@@ -342,7 +342,10 @@
   ENDIF()
 ENDIF(WIN32)
 
+set(BUILD_SHARED_LIBS ON)
 FIND_PACKAGE(Boost 1.47 COMPONENTS filesystem system)
+set(BUILD_SHARED_LIBS OFF)
+SET(Boost_LIBRARIES Boost::filesystem Boost::system ${Boost_LIBRARIES})
 
 IF (NOT Boost_FOUND)
   IF(WIN32)
