--- id3v2-0.1.12/id3v2.cpp.orig	2010-03-26 23:26:46.000000000 +0000
+++ id3v2-0.1.12/id3v2.cpp	2012-06-03 23:35:30.000000000 +0100
@@ -460,7 +460,7 @@
             if (*currentTrackNum == '/') 
             {
               newTrackNum = (char *)malloc(strlen(currentTrackNum) 
-                                   + strlen(frameList[ii].data)); 
+                                   + strlen(frameList[ii].data) + 1); 
               strcpy(newTrackNum, frameList[ii].data);
               strcat(newTrackNum, currentTrackNum);
             }
