--- ImageMagick-6.7.8-1/configure.orig	2012-07-05 01:04:24.000000000 +0100
+++ ImageMagick-6.7.8-1/configure	2012-07-15 10:49:58.339513530 +0100
@@ -29459,8 +29459,8 @@
     if test -n "$freetype_config"; then
       freetype_prefix=`${freetype_config} --prefix`
       freetype_exec_prefix=`${freetype_config} --exec-prefix`
-      LDFLAGS="$LDFLAGS -L${freetype_exec_prefix}/lib"
-      CPPFLAGS="$CPPFLAGS -I${freetype_prefix}/include/freetype2"
+      LDFLAGS="$LDFLAGS `${freetype_config} --libs`"
+      CPPFLAGS="$CPPFLAGS `${freetype_config} --cflags`"
     fi
 
     if test "$FREETYPE_LIBS" = ''; then
