--- ImageMosaicWall-20231130/CMakeLists.txt.orig	2023-11-30 22:08:18.000000000 +0000
+++ ImageMosaicWall-20231130/CMakeLists.txt	2023-12-02 15:12:07.208940206 +0000
@@ -74,7 +74,7 @@
 
 include(i18nUtils)
 
-include_directories($<TARGET_PROPERTY:exiv2lib,INTERFACE_INCLUDE_DIRECTORIES>
+include_directories($<TARGET_PROPERTY:Exiv2::exiv2lib,INTERFACE_INCLUDE_DIRECTORIES>
                     $<TARGET_PROPERTY:Qt${QT_VERSION_MAJOR}::Widgets,INTERFACE_INCLUDE_DIRECTORIES>
                     $<TARGET_PROPERTY:Qt${QT_VERSION_MAJOR}::Core,INTERFACE_INCLUDE_DIRECTORIES>
                     $<TARGET_PROPERTY:Qt${QT_VERSION_MAJOR}::Gui,INTERFACE_INCLUDE_DIRECTORIES>
@@ -138,7 +138,7 @@
                       Qt${QT_VERSION_MAJOR}::Gui
                       Qt${QT_VERSION_MAJOR}::Concurrent
 
-                      exiv2lib
+                      Exiv2::exiv2lib
                       ${CMAKE_THREAD_LIBS_INIT}
 )
 
--- ImageMosaicWall-20231130/dplugin/CMakeLists.txt.orig	2023-11-30 22:08:18.000000000 +0000
+++ ImageMosaicWall-20231130/dplugin/CMakeLists.txt	2023-12-02 15:12:44.305938436 +0000
@@ -36,7 +36,7 @@
                       Qt${QT_VERSION_MAJOR}::Gui
                       Qt${QT_VERSION_MAJOR}::Concurrent
 
-                      exiv2lib
+                      Exiv2::exiv2lib
 )
 
 MACRO_ADD_PLUGIN_INSTALL_TARGET(Generic_ImageMosaicWall_Plugin generic)
