--- ImageMosaicWall-20231028/dplugin/CMakeLists.txt.orig	2023-10-28 11:21:49.000000000 +0100
+++ ImageMosaicWall-20231028/dplugin/CMakeLists.txt	2023-10-28 16:36:46.698915134 +0100
@@ -12,7 +12,7 @@
                        DESCRIPTION "digiKam core library"
 )
 
-include_directories($<TARGET_PROPERTY:Digikam::digikamcore,INTERFACE_INCLUDE_DIRECTORIES>
+include_directories($<TARGET_PROPERTY:Digikam::digikamcore,INTERFACE_INCLUDE_DIRECTORIES>/digikam
                     ${CMAKE_CURRENT_BINARY_DIR}
                     ${CMAKE_CURRENT_SOURCE_DIR}
                     ${CMAKE_BINARY_DIR}
