--- ImageMosaicWall-20220302/imageprocessing.cpp.orig	2022-03-02 22:02:51.000000000 +0000
+++ ImageMosaicWall-20220302/imageprocessing.cpp	2023-10-28 11:24:38.436980560 +0100
@@ -323,7 +323,7 @@
 {
     try
     {
-        Exiv2::Image::AutoPtr image = Exiv2::ImageFactory::open(filename.toStdString());
+        Exiv2::Image::UniquePtr image = Exiv2::ImageFactory::open(filename.toStdString());
 
         if (image.get() == nullptr || !image.get()->good())
             return QImage();
