--- inkscape-1.0beta2_2019-12-03_2b71d25d45/CMakeLists.txt.orig	2019-12-03 20:17:39.000000000 +0000
+++ inkscape-1.0beta2_2019-12-03_2b71d25d45/CMakeLists.txt	2019-12-12 19:53:05.762349294 +0000
@@ -28,12 +28,12 @@
 # avoid having empty buildtype
 set(CMAKE_BUILD_TYPE_INIT "Release")
 
+project(inkscape)
+
 include(CMakeScripts/HelperFunctions.cmake)
 include(CMakeScripts/ConfigEnv.cmake)
 include(GNUInstallDirs) # for the CMAKE_INSTALL_LIBDIR variable
 
-project(inkscape)
-
 include(CMakeScripts/ConfigPaths.cmake)
 
 set(PROJECT_NAME inkscape)
