--- inkscape-1.1/src/3rdparty/2geom/CMakeLists.txt.orig	2021-05-17 20:25:55.000000000 +0100
+++ inkscape-1.1/src/3rdparty/2geom/CMakeLists.txt	2021-05-25 16:01:28.561828410 +0100
@@ -144,4 +144,4 @@
 
 configure_file( ${CMAKE_CURRENT_SOURCE_DIR}/2geom.pc.in
                 ${CMAKE_CURRENT_BINARY_DIR}/2geom.pc @ONLY IMMEDIATE )
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/2geom.pc" DESTINATION lib/pkgconfig)
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/2geom.pc" DESTINATION lib${LIB_SUFFIX}/pkgconfig)
--- inkscape-1.1/src/3rdparty/2geom/2geom.pc.in.orig	2021-05-17 20:25:55.000000000 +0100
+++ inkscape-1.1/src/3rdparty/2geom/2geom.pc.in	2021-05-25 16:02:52.381824410 +0100
@@ -1,6 +1,6 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=${exec_prefix}/lib@LIB_SUFFIX@
 includedir=${prefix}/include
 
 Name: 2geom
