--- inkscape-1.0rc1_2020-04-09_09960d6f05/CMakeLists.txt.orig	2020-04-09 10:21:56.000000000 +0100
+++ inkscape-1.0rc1_2020-04-09_09960d6f05/CMakeLists.txt	2020-04-14 19:54:12.455404127 +0100
@@ -28,11 +28,11 @@
 # avoid having empty buildtype
 set(CMAKE_BUILD_TYPE_INIT "Release")
 
+project(inkscape)
 include(CMakeScripts/HelperFunctions.cmake)
 include(CMakeScripts/ConfigEnv.cmake)
 include(GNUInstallDirs) # for the CMAKE_INSTALL_LIBDIR variable
 
-project(inkscape)
 set(CMAKE_INSTALL_DEFAULT_COMPONENT_NAME inkscape) # needs to be before any install() commands
 
 include(CMakeScripts/ConfigPaths.cmake)
