--- inkscape-1.0.2/src/3rdparty/autotrace/autotrace.h.orig	2021-01-15 10:04:25.000000000 +0000
+++ inkscape-1.0.2/src/3rdparty/autotrace/autotrace.h	2021-04-02 15:57:13.885494838 +0100
@@ -23,10 +23,6 @@
 
 #include <stdio.h>
 
-#ifdef __cplusplus
-extern "C" {
-#endif                          /* __cplusplus */
-
 #ifndef N_
 #define N_(x) x
 #endif                          /* Not def: N_ */
@@ -38,6 +34,10 @@
 #include "types.h"
 #include "color.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif                          /* __cplusplus */
+
 /* Third degree is the highest we deal with.  */
   enum _at_polynomial_degree {
     AT_LINEARTYPE = 1,
--- inkscape-1.0.2/src/3rdparty/autotrace/spline.h.orig	2021-01-15 10:04:25.000000000 +0000
+++ inkscape-1.0.2/src/3rdparty/autotrace/spline.h	2021-04-02 15:58:26.844491357 +0100
@@ -8,6 +8,10 @@
 #include "autotrace.h"
 #include "output.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif                          /* __cplusplus */
+
 typedef at_polynomial_degree polynomial_degree;
 typedef at_spline_type spline_type;
 
@@ -85,4 +89,8 @@
 extern void append_spline_list(spline_list_array_type *, spline_list_type);
 extern void free_spline_list_array(spline_list_array_type *);
 
+#ifdef __cplusplus
+}
+#endif                          /* __cplusplus */
+
 #endif /* not SPLINE_H */
--- inkscape-1.0.2/src/3rdparty/autotrace/color.h.orig	2021-01-15 10:04:25.000000000 +0000
+++ inkscape-1.0.2/src/3rdparty/autotrace/color.h	2021-04-02 16:53:22.974106231 +0100
@@ -24,6 +24,10 @@
 #include <glib.h>
 #include <glib-object.h>
 
+#ifdef __cplusplus
+extern "C" {
+#endif                          /* __cplusplus */
+
 typedef struct _at_color at_color;
 struct _at_color {
   guint8 r;
@@ -43,4 +47,7 @@
 GType at_color_get_type(void);
 #define AT_TYPE_COLOR (at_color_get_type ())
 
+#ifdef __cplusplus
+}
+#endif                          /* __cplusplus */
 #endif /* not AT_COLOR_H */
--- inkscape-1.0.2/src/trace/autotrace/inkscape-autotrace.cpp.orig	2021-01-15 12:09:08.000000000 +0000
+++ inkscape-1.0.2/src/trace/autotrace/inkscape-autotrace.cpp	2021-04-02 15:55:18.471500345 +0100
@@ -14,11 +14,9 @@
 
 #include "inkscape-autotrace.h"
 
-extern "C" {
 #include "3rdparty/autotrace/autotrace.h"
 #include "3rdparty/autotrace/output.h"
 #include "3rdparty/autotrace/spline.h"
-}
 
 #include <glibmm/i18n.h>
 #include <gtkmm/main.h>
