--- inkscape-1.2_2022-05-15_dc2aedaf03/src/ui/dialog/dialog-container.cpp.orig	2022-05-15 22:32:46.000000000 +0100
+++ inkscape-1.2_2022-05-15_dc2aedaf03/src/ui/dialog/dialog-container.cpp	2022-05-18 09:52:31.409116951 +0100
@@ -11,6 +11,10 @@
  * Released under GNU GPL v2+, read the file 'COPYING' for more information.
  */
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"  // only include where actually required!
+#endif
+
 #include "dialog-container.h"
 
 #include <glibmm/i18n.h>
--- inkscape-1.2_2022-05-15_dc2aedaf03/src/ui/dialog/dialog-data.h.orig	2022-05-15 15:53:38.000000000 +0100
+++ inkscape-1.2_2022-05-15_dc2aedaf03/src/ui/dialog/dialog-data.h	2022-05-18 09:47:24.329950747 +0100
@@ -18,6 +18,10 @@
  * so we need a place to keep this information.
  */
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"  // only include where actually required!
+#endif
+
 #include <map>
 #include <string>
 
