--- irrlicht-r3643/source/Irrlicht/Makefile.orig	2011-04-17 08:12:49.000000000 +0100
+++ irrlicht-r3643/source/Irrlicht/Makefile	2011-04-17 08:47:40.000000000 +0100
@@ -142,7 +142,7 @@
 	mkdir -p $(INSTALL_DIR)/../include/irrlicht
 	cp ../../include/*.h $(INSTALL_DIR)/../include/irrlicht/
 	cp $(LIB_PATH)/$(SHARED_LIB).$(VERSION) $(INSTALL_DIR)
-	cd $(INSTALL_DIR) && ln -s -f $(SHARED_LIB).$(VERSION) $(COMPATIBILITY_VERSION)
+	cd $(INSTALL_DIR) && ln -s -f $(SHARED_LIB).$(VERSION) $(SHARED_LIB).$(COMPATIBILITY_VERSION)
 
 #	ldconfig -n $(INSTALL_DIR)
 
