--- itk3.3/configure.in.orig	2007-12-19 17:28:55.000000000 -0800
+++ itk3.3/configure.in	2007-12-19 17:34:21.000000000 -0800
@@ -223,6 +223,39 @@
 AC_SUBST(itk_LIB_FILE)
 
 #--------------------------------------------------------------------
+# These are for itkConfig.sh
+#--------------------------------------------------------------------
+
+# pkglibdir must be a fully qualified path and (not ${exec_prefix}/lib)
+eval pkglibdir="${libdir}/${PACKAGE_NAME}${PACKAGE_VERSION}"
+if test "${TCL_LIB_VERSIONS_OK}" = "ok"; then
+    eval ITK_LIB_FLAG="-litk${PACKAGE_VERSION}${DBGX}"
+    eval ITK_STUB_LIB_FLAG="-litkstub${PACKAGE_VERSION}${DBGX}"
+else
+    eval ITK_LIB_FLAG="-litk`echo ${PACKAGE_VERSION} | tr -d .`${DBGX}"
+    eval ITK_STUB_LIB_FLAG="-litkstub`echo ${PACKAGE_VERSION} | tr -d .`${DBGX}"
+fi
+ITK_BUILD_LIB_SPEC="-L`pwd` ${itk_LIB_FLAG}"
+ITK_LIB_SPEC="-L${libdir} ${itk_LIB_FLAG}"
+
+ITK_BUILD_STUB_LIB_SPEC="-L`pwd` ${ITK_STUB_LIB_FLAG}"
+ITK_STUB_LIB_SPEC="-L${pkglibdir} ${ITK_STUB_LIB_FLAG}"
+ITK_BUILD_STUB_LIB_PATH="`pwd`/${ITK_STUB_LIB_FILE}"
+ITK_STUB_LIB_PATH="${pkglibdir}/${ITK_STUB_LIB_FILE}"
+
+AC_SUBST(ITK_BUILD_LIB_SPEC)
+AC_SUBST(ITK_LIB_SPEC)
+AC_SUBST(ITK_BUILD_STUB_LIB_SPEC)
+AC_SUBST(ITK_STUB_LIB_SPEC)
+AC_SUBST(ITK_BUILD_STUB_LIB_PATH)
+AC_SUBST(ITK_STUB_LIB_PATH)
+
+# ITK_SRC_DIR must be a fully qualified path
+eval ITK_SRC_DIR="$srcdir"
+ITK_SRC_DIR=`cd "${ITK_SRC_DIR}"; pwd`
+AC_SUBST(ITK_SRC_DIR)
+
+#--------------------------------------------------------------------
 # Find tclsh so that we can run pkg_mkIndex to generate the pkgIndex.tcl
 # file during the install process.  Don't run the TCLSH_PROG through
 # ${CYGPATH} because it's being used directly by make.
--- itk3.3/configure.orig	2007-12-19 17:28:52.000000000 -0800
+++ itk3.3/configure	2007-12-19 17:41:28.000000000 -0800
@@ -10227,6 +10227,32 @@
 #--------------------------------------------------------------------
 
 
+#--------------------------------------------------------------------
+# These are for itkConfig.sh
+#--------------------------------------------------------------------
+
+# pkglibdir must be a fully qualified path and (not ${exec_prefix}/lib)
+eval pkglibdir="${libdir}/${PACKAGE_NAME}${PACKAGE_VERSION}"
+if test "${TCL_LIB_VERSIONS_OK}" = "ok"; then
+    eval ITK_LIB_FLAG="-litk${PACKAGE_VERSION}${DBGX}"
+    eval ITK_STUB_LIB_FLAG="-litkstub${PACKAGE_VERSION}${DBGX}"
+else
+    eval ITK_LIB_FLAG="-litk`echo ${PACKAGE_VERSION} | tr -d .`${DBGX}"
+    eval ITK_STUB_LIB_FLAG="-litkstub`echo ${PACKAGE_VERSION} | tr -d .`${DBGX}"
+fi
+ITK_BUILD_LIB_SPEC="-L`pwd` ${itk_LIB_FLAG}"
+ITK_LIB_SPEC="-L${libdir} ${itk_LIB_FLAG}"
+
+ITK_BUILD_STUB_LIB_SPEC="-L`pwd` ${ITK_STUB_LIB_FLAG}"
+ITK_STUB_LIB_SPEC="-L${pkglibdir} ${ITK_STUB_LIB_FLAG}"
+ITK_BUILD_STUB_LIB_PATH="`pwd`/${ITK_STUB_LIB_FILE}"
+ITK_STUB_LIB_PATH="${pkglibdir}/${ITK_STUB_LIB_FILE}"
+
+
+# ITK_SRC_DIR must be a fully qualified path
+eval ITK_SRC_DIR="$srcdir"
+ITK_SRC_DIR=`cd "${ITK_SRC_DIR}"; pwd`
+
 
 
 #--------------------------------------------------------------------
@@ -11006,6 +11032,13 @@
 s,@TK_STUB_LIB_SPEC@,$TK_STUB_LIB_SPEC,;t t
 s,@TK_LIBS@,$TK_LIBS,;t t
 s,@TK_XINCLUDES@,$TK_XINCLUDES,;t t
+s,@ITK_BUILD_LIB_SPEC@,$ITK_BUILD_LIB_SPEC,;t t
+s,@ITK_LIB_SPEC@,$ITK_LIB_SPEC,;t t
+s,@ITK_BUILD_STUB_LIB_SPEC@,$ITK_BUILD_STUB_LIB_SPEC,;t t
+s,@ITK_STUB_LIB_SPEC@,$ITK_STUB_LIB_SPEC,;t t
+s,@ITK_BUILD_STUB_LIB_PATH@,$ITK_BUILD_STUB_LIB_PATH,;t t
+s,@ITK_STUB_LIB_PATH@,$ITK_STUB_LIB_PATH,;t t
+s,@ITK_SRC_DIR@,$ITK_SRC_DIR,;t t
 s,@itcl_VERSION@,$itcl_VERSION,;t t
 s,@itcl_BIN_DIR@,$itcl_BIN_DIR,;t t
 s,@itcl_SRC_DIR@,$itcl_SRC_DIR,;t t
--- itk3.3/Makefile.in.orig	2007-12-19 17:29:14.000000000 -0800
+++ itk3.3/Makefile.in	2007-12-19 17:34:44.000000000 -0800
@@ -366,13 +366,15 @@
 	@mkdir -p $(DESTDIR)$(pkglibdir)
 	@list='$(lib_BINARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
-	    echo " $(INSTALL_PROGRAM) $$p $(DESTDIR)$(pkglibdir)/$$p"; \
-	    $(INSTALL_PROGRAM) $$p $(DESTDIR)$(pkglibdir)/$$p; \
 	    stub=`echo $$p|sed -e "s/.*\(stub\).*/\1/"`; \
 	    if test "x$$stub" = "xstub"; then \
+		echo " $(INSTALL_PROGRAM) $$p $(DESTDIR)$(pkglibdir)/$$p"; \
+		$(INSTALL_PROGRAM) $$p $(DESTDIR)$(pkglibdir)/$$p; \
 		echo " $(RANLIB_STUB) $(DESTDIR)$(pkglibdir)/$$p"; \
 		$(RANLIB_STUB) $(DESTDIR)$(pkglibdir)/$$p; \
 	    else \
+		echo " $(INSTALL_PROGRAM) $$p $(DESTDIR)$(libdir)/$$p"; \
+		$(INSTALL_PROGRAM) $$p $(DESTDIR)$(libdir)/$$p; \
 		echo " $(RANLIB) $(DESTDIR)$(pkglibdir)/$$p"; \
 		$(RANLIB) $(DESTDIR)$(pkglibdir)/$$p; \
 	    fi; \
--- itk3.3/pkgIndex.tcl.in.orig	2007-12-19 17:29:10.000000000 -0800
+++ itk3.3/pkgIndex.tcl.in	2007-12-19 17:30:35.000000000 -0800
@@ -1,3 +1,3 @@
 # Tcl package index file, version 1.0
 
-package ifneeded Itk @PACKAGE_VERSION@ [list load [file join $dir "@PKG_LIB_FILE@"] Itk]
+package ifneeded Itk @PACKAGE_VERSION@ [list load [file join $dir .. "@PKG_LIB_FILE@"] Itk]
