--- jami-20200309.1.e2d6791/lrc/src/callbackshandler.h.orig	2020-03-10 03:46:09.000000000 +0000
+++ jami-20200309.1.e2d6791/lrc/src/callbackshandler.h	2020-03-11 20:41:43.607868325 +0000
@@ -300,6 +300,7 @@
      * @param payloads of the interaction
      */
     void slotNewAccountMessage(const QString& accountId,
+                               const QString& messageId,
                                const QString& from,
                                const QMap<QString,QString>& payloads);
     /**
--- jami-20200309.1.e2d6791/lrc/src/callbackshandler.cpp.orig	2020-03-10 03:46:09.000000000 +0000
+++ jami-20200309.1.e2d6791/lrc/src/callbackshandler.cpp	2020-03-11 20:41:37.746868605 +0000
@@ -246,6 +246,7 @@
 
 void
 CallbacksHandler::slotNewAccountMessage(const QString& accountId,
+                                        const QString& messageId,
                                         const QString& from,
                                         const QMap<QString,QString>& payloads)
 {
