--- jami-20210409.1.3e6a4fb/daemon/contrib/src/libgit2/rules.mak.orig	2021-04-09 21:57:32.000000000 +0100
+++ jami-20210409.1.3e6a4fb/daemon/contrib/src/libgit2/rules.mak	2021-04-10 21:37:42.982850609 +0100
@@ -24,6 +24,6 @@
 .libgit2: libgit2
 	cd $< && mkdir -p build && cd build \
     && $(CMAKE) -DCMAKE_C_FLAGS=-fPIC -DCMAKE_INSTALL_PREFIX=$(PREFIX) -DUSE_HTTPS=OFF -DCURL=OFF -DUSE_HTTP_PARSER=system -DBUILD_CLAR=OFF -DBUILD_SHARED_LIBS=OFF -DUSE_SSH=OFF -DREGEX_BACKEND=regcomp .. \
-    && $(CMAKE) --build . \
+    && $(MAKE) \
     && $(MAKE) install
 	touch $@
