--- jsoncpp-0.6.0-rc2/SConstruct.orig	2010-03-13 10:59:50.000000000 +0000
+++ jsoncpp-0.6.0-rc2/SConstruct	2013-08-17 06:36:28.970135410 +0100
@@ -120,6 +120,7 @@
 elif platform.startswith('linux-gcc'):
     env.Tool( 'default' )
     env.Append( LIBS = ['pthread'], CCFLAGS = "-Wall" )
+    env.Append( LINKFLAGS='-Wl,-soname,libjsoncpp.so.0')
     env['SHARED_LIB_ENABLED'] = True
 else:
     print "UNSUPPORTED PLATFORM."
