--- jtag-0.5.1/libbrux/cmd/cmd_detectflash.c.orig	2003-08-19 09:10:50.000000000 +0100
+++ jtag-0.5.1/libbrux/cmd/cmd_detectflash.c	2008-09-08 20:05:34.000000000 +0100
@@ -40,7 +40,7 @@
 		return 1;
 	}
 
-	detectflash( bus );
+	detectflash( bus , 0);
 
 	return 1;
 }
--- jtag-0.5.1/libbrux/flash/detectflash.c.orig	2003-09-24 05:40:12.000000000 +0100
+++ jtag-0.5.1/libbrux/flash/detectflash.c	2008-09-08 20:08:04.000000000 +0100
@@ -43,7 +43,7 @@
 int jedec_detect( bus_t *bus, uint32_t adr, cfi_array_t **cfi_array );
 
 void
-detectflash( bus_t *bus )
+detectflash( bus_t *bus, uint32_t adr )
 {
 	cfi_array_t *cfi_array = NULL;
 	cfi_query_structure_t *cfi;
