--- julia-1.12.0/Makefile.orig	2025-09-28 04:13:12.000000000 +0100
+++ julia-1.12.0/Makefile	2025-10-09 18:08:02.339140468 +0100
@@ -308,7 +308,7 @@
 # Note that we disable MSYS2's path munging here, as otherwise
 # it replaces our `:`-separated list as a `;`-separated one.
 define stringreplace
-	MSYS2_ARG_CONV_EXCL='*' $(build_depsbindir)/stringreplace $$(strings -t x - '$1' | grep '$2' | awk '{print $$1;}') '$3' 255 '$(call cygpath_w,$1)'
+	MSYS2_ARG_CONV_EXCL='*' $(build_depsbindir)/stringreplace $$(strings -t x - "$1" | grep '$2' | awk '{print $$1;}') '$3' 255 "$(call cygpath_w,$1)"
 endef
 
 
--- julia-1.12.0/src/debuginfo.cpp.orig	2025-09-28 04:13:13.000000000 +0100
+++ julia-1.12.0/src/debuginfo.cpp	2025-10-09 17:48:19.656481509 +0100
@@ -2,6 +2,8 @@
 
 #include "platform.h"
 
+#include <cstdint>
+
 #include "llvm-version.h"
 #include <llvm/DebugInfo/DIContext.h>
 #include <llvm/DebugInfo/DWARF/DWARFContext.h>
--- julia-1.12.0/src/aotcompile.cpp.orig	2025-09-28 04:13:13.000000000 +0100
+++ julia-1.12.0/src/aotcompile.cpp	2025-10-09 17:50:51.157341719 +0100
@@ -3,6 +3,8 @@
 #include "llvm-version.h"
 #include "platform.h"
 
+#include <cstdint>
+
 // target support
 #include <llvm/TargetParser/Triple.h>
 #include "llvm/Support/CodeGen.h"
--- julia-1.12.0/src/disasm.cpp.orig	2025-09-28 04:13:13.000000000 +0100
+++ julia-1.12.0/src/disasm.cpp	2025-10-09 17:48:47.588297191 +0100
@@ -51,6 +51,7 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include <cstdint>
 #include <map>
 #include <set>
 #include <string>
--- julia-1.12.0/src/cgmemmgr.cpp.orig	2025-09-28 04:13:13.000000000 +0100
+++ julia-1.12.0/src/cgmemmgr.cpp	2025-10-09 17:49:12.787037519 +0100
@@ -3,6 +3,8 @@
 #include "llvm-version.h"
 #include "platform.h"
 
+#include <cstdint>
+
 #include <llvm/ExecutionEngine/SectionMemoryManager.h>
 #include "julia.h"
 #include "julia_internal.h"
