--- julia-1.0.1/Makefile.orig	2018-09-29 20:08:15.000000000 +0100
+++ julia-1.0.1/Makefile	2018-10-25 22:52:20.119046751 +0100
@@ -313,7 +313,7 @@
 $(eval $(call std_dll,atomic-1))
 endif
 define stringreplace
-	$(build_depsbindir)/stringreplace $$(strings -t x - $1 | grep '$2' | awk '{print $$1;}') '$3' 255 "$(call cygpath_w,$1)"
+	$(build_depsbindir)/stringreplace $$(strings-GNU -t x - $1 | grep '$2' | awk '{print $$1;}') '$3' 255 "$(call cygpath_w,$1)"
 endef
 
 install: $(build_depsbindir)/stringreplace $(BUILDROOT)/doc/_build/html/en/index.html
