--- k3d-0.8.0.5/CMakeLists.txt.orig	2015-11-07 13:00:59.000000000 +0000
+++ k3d-0.8.0.5/CMakeLists.txt	2018-01-31 22:02:18.341503716 +0000
@@ -362,7 +362,7 @@
 # Setup output directories ...
 SET(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${k3d_BINARY_DIR}/bin)
 
-SET(K3D_LIBDIR lib) # Allows us to handle 64-bit libs if/when it becomes necessary.
+SET(K3D_LIBDIR lib${LIB_SUFFIX}) # Allows us to handle 64-bit libs if/when it becomes necessary.
 
 IF(APPLE AND K3D_ENABLE_OSX_BUNDLE)
 	SET(K3D_BUNDLE_DIRECTORY ${k3d_BINARY_DIR}/K-3D.app/Contents)
