--- k3d-0.8.0.5/CMakeLists.txt.orig	2018-02-26 19:50:27.229075172 +0000
+++ k3d-0.8.0.5/CMakeLists.txt	2018-04-16 21:33:11.003756566 +0100
@@ -245,7 +245,7 @@
 # Find Boost - this logic must follow all user-definable options 
 SET(K3D_BOOST_COMPONENTS date_time program_options regex unit_test_framework)
 IF(K3D_ENABLE_PYTHON)
-	LIST(APPEND K3D_BOOST_COMPONENTS python)
+	LIST(APPEND K3D_BOOST_COMPONENTS python27)
 ENDIF()
 IF(K3D_BUILD_COLLADA_IO_MODULE)
 	LIST(APPEND K3D_BOOST_COMPONENTS filesystem system)
--- k3d-0.8.0.5/k3dsdk/python/CMakeLists.txt.orig	2015-11-07 13:00:59.000000000 +0000
+++ k3d-0.8.0.5/k3dsdk/python/CMakeLists.txt	2018-04-16 22:12:27.255504445 +0100
@@ -17,7 +17,7 @@
 TARGET_LINK_LIBRARIES(k3dsdk-python-arrays
 	k3dsdk
 	${K3D_PYTHON_LIBS}
-	${Boost_PYTHON_LIBRARY}
+	${Boost_PYTHON27_LIBRARY}
 	)
 
 K3D_ADD_LIBRARY(k3dsdk-python-const-arrays SHARED
@@ -28,7 +28,7 @@
 TARGET_LINK_LIBRARIES(k3dsdk-python-const-arrays
 	k3dsdk
 	${K3D_PYTHON_LIBS}
-	${Boost_PYTHON_LIBRARY}
+	${Boost_PYTHON27_LIBRARY}
 	)
 
 K3D_ADD_LIBRARY(k3dsdk-python-values SHARED
@@ -70,7 +70,7 @@
 K3D_GENERATE_DEF_FILE(k3dsdk-python-values)
 TARGET_LINK_LIBRARIES(k3dsdk-python-values
 	k3dsdk
-	${Boost_PYTHON_LIBRARY}
+	${Boost_PYTHON27_LIBRARY}
 	${K3D_PYTHON_LIBS}
 	)
 
@@ -224,7 +224,7 @@
 K3D_GENERATE_DEF_FILE(k3dsdk-python)
 TARGET_LINK_LIBRARIES(k3dsdk-python
 	k3dsdk-python-primitives
-	${Boost_PYTHON_LIBRARY}
+	${Boost_PYTHON27_LIBRARY}
 	)
 
 INSTALL(TARGETS k3dsdk-python-arrays
@@ -277,7 +277,7 @@
 	K3D_GENERATE_DEF_FILE(k3dsdk-python-atk)
 	TARGET_LINK_LIBRARIES(k3dsdk-python-atk
 		k3dsdk
-		${Boost_PYTHON_LIBRARY}
+		${Boost_PYTHON27_LIBRARY}
 		${K3D_PYTHON_LIBS}
 		${K3D_GTKMM_LIBS}
 		)
