--- k9copy-2.3.8/src/import/k9avidecode.h.orig	2011-12-09 21:08:41.000000000 +0000
+++ k9copy-2.3.8/src/import/k9avidecode.h	2012-06-06 21:58:42.000000000 +0100
@@ -29,7 +29,7 @@
 //typedef dvd_file_t * (*DVDOpenFile_t) ( dvd_reader_t *, int, dvd_read_domain_t );
 typedef void (*av_register_all_t) (void);
 typedef int64_t (*av_gettime_t) (void);
-typedef int (*av_open_input_file_t)(AVFormatContext **, const char *,AVInputFormat *,int, AVFormatParameters *);
+typedef int (*avformat_open_input_t)(AVFormatContext **, const char *,AVInputFormat *, AVDictionary **options);
 typedef int (*av_find_stream_info_t)(AVFormatContext *);
 typedef AVCodec* (*avcodec_find_decoder_t)(enum CodecID);
 typedef int (*avcodec_open_t)(AVCodecContext *, AVCodec *);
@@ -77,7 +77,7 @@
 	
 private:
     av_register_all_t av_register_all;
-    av_open_input_file_t av_open_input_file;
+    avformat_open_input_t avformat_open_input;
     av_find_stream_info_t av_find_stream_info;
     avcodec_find_decoder_t avcodec_find_decoder;
     avcodec_open_t avcodec_open;
--- k9copy-2.3.8/src/import/k9avidecode.cpp.orig	2011-12-09 21:08:41.000000000 +0000
+++ k9copy-2.3.8/src/import/k9avidecode.cpp	2012-06-06 22:15:43.000000000 +0100
@@ -63,7 +63,7 @@
 #endif
 
     av_register_all = (av_register_all_t)dlsym(FormatHandle,"av_register_all");
-    av_open_input_file = (av_open_input_file_t)dlsym(FormatHandle,"av_open_input_file");
+    avformat_open_input = (avformat_open_input_t)dlsym(FormatHandle,"avformat_open_input");
     av_find_stream_info = (av_find_stream_info_t)dlsym(FormatHandle,"av_find_stream_info");
     avcodec_find_decoder =(avcodec_find_decoder_t) dlsym(CodecHandle,"avcodec_find_decoder");
     avcodec_open = (avcodec_open_t)dlsym(CodecHandle,"avcodec_open");
@@ -134,7 +134,7 @@
         close();
 
     // Open video file
-    if (av_open_input_file(&m_FormatCtx, _fileName.toUtf8(), NULL, 0, NULL)!=0) {
+    if (avformat_open_input(&m_FormatCtx, _fileName.toUtf8(), NULL, NULL)!=0) {
         m_error=i18n("Couldn't open the file %1").arg(_fileName);
         return false; // Couldn't open file}
     }
@@ -252,8 +252,8 @@
             if (frameFinished) {
 //            if (m_Frame->pts >=fspos)
 		int64_t cur_dts=fspos;
-		if (m_FormatCtx->cur_st)
-		    cur_dts=	m_FormatCtx->cur_st->cur_dts;
+//		if (m_FormatCtx->cur_st)
+//		    cur_dts=	m_FormatCtx->cur_st->cur_dts;
                 if (cur_dts >=fspos) {
                     bFound=true;
 #ifndef HAVE_SWSCALE
