--- kalarm-2.0.1/autostart/autostart.cpp.orig	2008-07-21 00:44:22.000000000 +0100
+++ kalarm-2.0.1/autostart/autostart.cpp	2008-09-03 07:13:30.000000000 +0100
@@ -53,12 +53,14 @@
 		KAboutData::License_GPL,
 		"Copyright 2002,2008 David Jarvie", 0, "http://www.astrojar.org.uk/kalarm");
 	aboutData.addAuthor("David Jarvie", I18N_NOOP("Maintainer"), "software@astrojar.org.uk");
-	KGlobal::locale()->insertCatalogue("kalarm");
 
 	KCmdLineArgs::init(argc, argv, &aboutData);
 	KCmdLineArgs::addCmdLineOptions(options);
 
 	AutostartApp app;
+
+	KGlobal::locale()->insertCatalogue("kalarm");
+
 	return app.exec();
 }
 
