--- Kalpa-0.0.4.2/Src/Libs/Glan/GlanLib/md5.cpp.orig	2005-05-19 12:06:41.000000000 +0100
+++ Kalpa-0.0.4.2/Src/Libs/Glan/GlanLib/md5.cpp	2007-10-25 14:19:24.000000000 +0100
@@ -83,7 +83,7 @@
  (a) += (b); \
 }
 
-void Glan::make_digest( char * md5str, unsigned char * digest )
+void make_digest( char * md5str, unsigned char * digest )
 {
     int i;
 
@@ -96,7 +96,7 @@
     *md5str = '\0';
 }
 
-void Glan::MD5Init( MD5_CTX *context )
+void MD5Init( MD5_CTX *context )
 {
     context->count[0] = context->count[1] = 0;
     /* Load magic initialization constants.
@@ -107,7 +107,7 @@
     context->state[3] = 0x10325476;
 }
 
-void Glan::MD5Update( MD5_CTX *context, const unsigned char *input, unsigned int inputLen)
+void MD5Update( MD5_CTX *context, const unsigned char *input, unsigned int inputLen)
 {
     unsigned int i, index, partLen;
 
@@ -144,7 +144,7 @@
      inputLen - i);
 }
 
-void Glan::MD5Final( unsigned char digest[16], MD5_CTX *context)
+void MD5Final( unsigned char digest[16], MD5_CTX *context)
 {
     unsigned char bits[8];
     unsigned int index, padLen;
--- Kalpa-0.0.4.2/Src/Libs/ShtrihDriver/ShtrihDriver.cpp.orig	2005-06-16 10:11:45.000000000 +0100
+++ Kalpa-0.0.4.2/Src/Libs/ShtrihDriver/ShtrihDriver.cpp	2007-10-25 16:27:29.000000000 +0100
@@ -11,6 +11,9 @@
 #include <qdatastream.h>
 #include <qtextcodec.h>
 
+extern "C" {
+extern long long int llround(double);
+};
 
 ECRError::ECRError(uint Code):Glan::Exception(),ErrorCode(Code)
 {
