--- kicad-5.0.2/qa/geometry/CMakeLists.txt.orig	2018-11-26 22:59:28.000000000 +0000
+++ kicad-5.0.2/qa/geometry/CMakeLists.txt	2019-04-21 16:22:59.702203252 +0100
@@ -19,7 +19,9 @@
 # or you may write to the Free Software Foundation, Inc.,
 # 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
 
+set(BUILD_SHARED_LIBS ON)
 find_package(Boost COMPONENTS unit_test_framework REQUIRED)
+set(BUILD_SHARED_LIBS OFF)
 find_package( wxWidgets 3.0.0 COMPONENTS gl aui adv html core net base xml stc REQUIRED )
 
 
@@ -40,7 +42,7 @@
 target_link_libraries( qa_geometry
     common
     polygon
-    ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY}
+    ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY} Boost::unit_test_framework
     ${wxWidgets_LIBRARIES}
 )
 
--- kicad-5.0.2/qa/shape_poly_set_refactor/CMakeLists.txt.orig	2018-11-26 22:59:28.000000000 +0000
+++ kicad-5.0.2/qa/shape_poly_set_refactor/CMakeLists.txt	2019-04-21 16:24:18.478199493 +0100
@@ -21,7 +21,9 @@
 # or you may write to the Free Software Foundation, Inc.,
 # 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
 
+set(BUILD_SHARED_LIBS ON)
 find_package(Boost COMPONENTS unit_test_framework REQUIRED)
+set(BUILD_SHARED_LIBS OFF)
 find_package( wxWidgets 3.0.0 COMPONENTS gl aui adv html core net base xml stc REQUIRED )
 
 add_definitions(-DBOOST_TEST_DYN_LINK)
@@ -47,9 +49,9 @@
     common
     polygon
     bitmaps
-    ${Boost_FILESYSTEM_LIBRARY}
-    ${Boost_SYSTEM_LIBRARY}
-    ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY}
+    ${Boost_FILESYSTEM_LIBRARY} Boost::filesystem
+    ${Boost_SYSTEM_LIBRARY} Boost::system
+    ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY} Boost::unit_test_framework
     ${wxWidgets_LIBRARIES}
 )
 
--- kicad-5.0.2/qa/pcb_test_window/CMakeLists.txt.orig	2018-11-26 22:59:28.000000000 +0000
+++ kicad-5.0.2/qa/pcb_test_window/CMakeLists.txt	2019-04-21 16:25:04.683197288 +0100
@@ -22,7 +22,9 @@
 # or you may write to the Free Software Foundation, Inc.,
 # 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
 
+set(BUILD_SHARED_LIBS ON)
 find_package(Boost COMPONENTS unit_test_framework REQUIRED)
+set(BUILD_SHARED_LIBS OFF)
 find_package( wxWidgets 3.0.0 COMPONENTS gl aui adv html core net base xml stc REQUIRED )
 
 add_definitions(-DBOOST_TEST_DYN_LINK -DPCBNEW)
@@ -91,8 +93,8 @@
     ${GITHUB_PLUGIN_LIBRARIES}
     common
     pcbcommon
-    ${Boost_FILESYSTEM_LIBRARY}
-    ${Boost_SYSTEM_LIBRARY}
-    ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY}
+    ${Boost_FILESYSTEM_LIBRARY} Boost::filesystem
+    ${Boost_SYSTEM_LIBRARY} Boost::system
+    ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY} Boost::unit_test_framework
     ${wxWidgets_LIBRARIES}
 )
--- kicad-5.0.2/eeschema/qa/CMakeLists.txt.orig	2018-11-26 22:59:28.000000000 +0000
+++ kicad-5.0.2/eeschema/qa/CMakeLists.txt	2019-04-21 16:22:31.785204584 +0100
@@ -21,7 +21,9 @@
 # or you may write to the Free Software Foundation, Inc.,
 # 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
 
+set(BUILD_SHARED_LIBS ON)
 find_package( Boost COMPONENTS unit_test_framework REQUIRED )
+set(BUILD_SHARED_LIBS OFF)
 
 include_directories( BEFORE ${INC_BEFORE} )
 include_directories(
@@ -42,6 +44,6 @@
 target_link_libraries( qa_eagle_plugin
     common
     eeschema_kiface
-    ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY}
+    ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY} Boost::unit_test_framework
     ${wxWidgets_LIBRARIES}
     )
