--- kicad-4.0.7/CMakeModules/CheckCXXSymbolExists.cmake.orig	2017-08-28 15:31:31.000000000 +0100
+++ kicad-4.0.7/CMakeModules/CheckCXXSymbolExists.cmake	2018-04-15 20:44:31.300711106 +0100
@@ -38,5 +38,6 @@
 include(CheckSymbolExists)
 
 macro(CHECK_CXX_SYMBOL_EXISTS SYMBOL FILES VARIABLE)
-  _CHECK_SYMBOL_EXISTS("${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/CMakeTmp/CheckSymbolExists.cxx" "${SYMBOL}" "${FILES}" "${VARIABLE}" )
+  check_symbol_exists( "${SYMBOL}" "${FILES}" "${VARIABLE}" )
+#  _CHECK_SYMBOL_EXISTS("${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/CMakeTmp/CheckSymbolExists.cxx" "${SYMBOL}" "${FILES}" "${VARIABLE}" )
 endmacro()
