--- kicad-5.1.10/common/gal/opengl/opengl_gal.cpp.orig	2021-04-23 18:42:36.000000000 +0100
+++ kicad-5.1.10/common/gal/opengl/opengl_gal.cpp	2021-07-13 21:36:39.432953627 +0100
@@ -2025,7 +2025,7 @@
 
     GLenum err = glewInit();
 
-    if( GLEW_OK != err )
+    if( GLEW_OK != err && GLEW_ERROR_NO_GLX_DISPLAY != err)
         throw std::runtime_error( (const char*) glewGetErrorString( err ) );
 
     // Check the OpenGL version (minimum 2.1 is required)
