--- kinoplus-0.3.5/src/Makefile.in.orig	2005-02-06 18:36:45.000000000 +0000
+++ kinoplus-0.3.5/src/Makefile.in	2006-09-29 22:05:01.000000000 +0100
@@ -550,7 +550,7 @@
 	uninstall-libLTLIBRARIES
 
 
-#libkinoplus_la_LIBADD = $(PACKAGE_LIBS)
+libkinoplus_la_LIBADD = $(PACKAGE_LIBS)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
