--- smilutils-0.3.0/libkino/filehandler.cc.orig	2003-08-26 11:31:04.000000000 +0100
+++ smilutils-0.3.0/libkino/filehandler.cc	2006-10-01 22:20:30.000000000 +0100
@@ -730,9 +730,10 @@
 
 off_t QtHandler::GetFileSize()
 {
-	struct stat file_status;
-	fstat(fileno(fd->stream), &file_status);
-	return file_status.st_size;
+//	struct stat file_status;
+//	fstat(fileno(fd->stream), &file_status);
+//	return file_status.st_size;
+	return 0;
 }
 
 
--- smilutils-0.3.0/apps/ppmfilter/ppmfilter.cc.orig	2003-10-04 20:21:30.000000000 +0100
+++ smilutils-0.3.0/apps/ppmfilter/ppmfilter.cc	2006-10-01 22:39:37.000000000 +0100
@@ -249,7 +249,7 @@
 			name( _name ),
 			description( _description )
 		{
-			Parse( initial );
+			property::Parse( initial );
 		}
 
 		// Constructor with default double value
@@ -260,7 +260,7 @@
 			char o[ 100 ];
 			sprintf( o, "%f", initial );
 			string t( o );
-			Parse( t );
+			property::Parse( t );
 		}
 
 		// Return the name of the property
