--- KoboDeluxe-0.5.1/sound/a_midi.c.orig	2003-06-08 19:17:14.000000000 +0100
+++ KoboDeluxe-0.5.1/sound/a_midi.c	2008-09-16 22:40:00.000000000 +0100
@@ -188,8 +188,7 @@
 
 	mon = &monitor_midisock;
 #ifdef HAVE_ALSA
-	if(snd_rawmidi_open(&m_in_fd, 0, 0, SND_RAWMIDI_OPEN_INPUT |
-			SND_RAWMIDI_OPEN_NONBLOCK))
+	if(snd_rawmidi_open(&m_in_fd, 0, 0, SND_RAWMIDI_NONBLOCK))
 	{
 		log_printf(ELOG, "Failed to open MIDI input device!\n");
 		return -1;
@@ -231,12 +230,12 @@
 #ifdef HAVE_ALSA
 	if(m_in_fd)
 	{
-		snd_rawmidi_input_flush(m_in_fd);
+//		snd_rawmidi_input_flush(m_in_fd);
 		snd_rawmidi_close(m_in_fd);
 	}
 	if(m_out_fd)
 	{
-		snd_rawmidi_output_flush(m_out_fd);
+//		snd_rawmidi_output_flush(m_out_fd);
 		snd_rawmidi_close(m_out_fd);
 	}
 	m_in_fd = m_out_fd = NULL;
