--- kpodder-0.6/src/kpodder.sh.orig	2006-03-19 00:09:18.000000000 +0000
+++ kpodder-0.6/src/kpodder.sh	2007-10-16 14:22:27.000000000 +0100
@@ -89,7 +89,7 @@
     while read line; do
         sep=
         for(( i=0; i<${#line}; ++i )); do
-            if [ ${line:i:8} = "<kpname>" ]; then
+            if [ "${line:i:8}" = "<kpname>" ]; then
                 sep=i
                 break
             fi
