--- kQOAuth-0.98/src/src.pro.orig	2013-09-08 11:01:33.000000000 +0100
+++ kQOAuth-0.98/src/src.pro	2014-08-19 21:30:16.973349432 +0100
@@ -75,7 +75,7 @@
     system( ./pcfile.sh $${INSTALL_PREFIX} $${VERSION} )
     pkgconfig.files = kqoauth.pc
     
-    target.path = $${INSTALL_PREFIX}/lib
+    target.path = $${INSTALL_PREFIX}/lib$${LIB_SUFFIX}
 
     headers.path = $${INSTALL_PREFIX}/include/QtKOAuth
     docs.path = $${INSTALL_PREFIX}/share/doc/$${TARGET}-$${VERSION}/html
