--- krystaldrop-0.7.2/Sources/KDpp/Resources/ArchiveManager.cpp.orig	2003-08-26 20:14:19.000000000 +0100
+++ krystaldrop-0.7.2/Sources/KDpp/Resources/ArchiveManager.cpp	2008-12-21 09:39:52.000000000 +0000
@@ -5,6 +5,8 @@
 #include <ctype.h>
 #endif
 
+#include <assert.h>
+
 std::map<std::string,KD_ArchiveReader*> KD_ArchiveManager::opened_archives;
 std::map<std::string,T_ArchiveReaderFactory> KD_ArchiveManager::known_suffixes;
 
