--- ktp-call-ui-0.6.1/libqtf/CMakeLists.txt.orig	2013-04-16 19:03:39.000000000 +0100
+++ ktp-call-ui-0.6.1/libqtf/CMakeLists.txt	2013-05-17 22:29:58.744682444 +0100
@@ -2,7 +2,7 @@
 # since gnome people like to include as many headers as possible
 # in their public headers instead of doing forward declarations.
 find_package(PkgConfig REQUIRED)
-pkg_check_modules(TP_FARSTREAM REQUIRED telepathy-farstream)
+pkg_check_modules(TP_FARSTREAM REQUIRED telepathy-farstream-0.4)
 pkg_check_modules(FARSTREAM REQUIRED farstream-0.1)
 
 include_directories(
