--- laidout-0.097/src/Makefile.orig	2018-11-25 20:23:08.000000000 +0000
+++ laidout-0.097/src/Makefile	2023-03-11 22:13:19.121987351 +0000
@@ -9,7 +9,7 @@
 
 
 LD=g++
-LDFLAGS= $(EXTRA_LDFLAGS) -L/usr/local/lib -L/usr/X11R6/lib -rdynamic -lXi -lXext -lX11 -lm -lpng `imlib2-config --libs` `freetype-config --libs`\
+LDFLAGS= $(EXTRA_LDFLAGS) -L/usr/local/lib -L/usr/X11R6/lib -rdynamic -lXi -lXext -lX11 -lm -lpng `pkg-config --libs imlib2` `freetype-config --libs`\
 		 `cups-config --libs` -ldl -lXft
 DEBUGFLAGS= -g -gdwarf-2
 CPPFLAGS= $(HIDEGARBAGE) -Wall $(DEBUGFLAGS) $(EXTRA_CPPFLAGS)  -I$(LAXDIR)/.. `freetype-config --cflags` -I$(POLYPTYCHBASEDIR)
--- laidout-0.097/src/interfaces/pagerangeinterface.cc.orig	2018-11-25 20:23:13.000000000 +0000
+++ laidout-0.097/src/interfaces/pagerangeinterface.cc	2023-03-11 22:11:36.049992268 +0000
@@ -567,7 +567,7 @@
 	if (part && pos>=0) *part=PART_Position;
 	if (position) *position=pos;
 	
-	if (pos>=0 || r>=0 || part>0) return 1;
+	if (pos>=0 || r>=0 || part) return 1;
 	return 0;
 }
 
