--- lapack-3.6.1/CBLAS/CMakeLists.txt.orig	2016-06-18 23:15:11.000000000 +0100
+++ lapack-3.6.1/CBLAS/CMakeLists.txt	2016-06-20 20:15:15.257018842 +0100
@@ -49,7 +49,7 @@
 set(_cblas_config_install_guard_target "")
 if(ALL_TARGETS)
   install(EXPORT cblas-targets
-    DESTINATION lib/cmake/cblas-${LAPACK_VERSION})
+    DESTINATION ${LIBRARY_DIR}/cmake/cblas-${LAPACK_VERSION})
   # Choose one of the cblas targets to use as a guard for
   # cblas-config.cmake to load targets from the install tree.
   list(GET ALL_TARGETS 0 _cblas_config_install_guard_target)
@@ -82,7 +82,7 @@
 install(FILES
   ${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/cblas-config.cmake
   ${LAPACK_BINARY_DIR}/cblas-config-version.cmake
-  DESTINATION lib/cmake/cblas-${LAPACK_VERSION}
+  DESTINATION ${LIBRARY_DIR}/cmake/cblas-${LAPACK_VERSION}
   )
 
 #install(EXPORT cblas-targets
--- lapack-3.6.1/LAPACKE/CMakeLists.txt.orig	2016-06-18 23:15:11.000000000 +0100
+++ lapack-3.6.1/LAPACKE/CMakeLists.txt	2016-06-20 20:16:11.087019047 +0100
@@ -85,4 +85,4 @@
   )
 
 install(EXPORT lapacke-targets
-  DESTINATION lib/cmake/lapacke-${LAPACK_VERSION})
+  DESTINATION ${LIBRARY_DIR}/cmake/lapacke-${LAPACK_VERSION})
