--- lapack-3.1.1/SRC/Makefile.orig	2007-02-23 21:33:05.000000000 +0000
+++ lapack-3.1.1/SRC/Makefile	2008-05-07 14:01:22.000000000 +0100
@@ -297,6 +297,9 @@
 	$(ARCH) $(ARCHFLAGS) $@ $(ALLOBJ)
 	$(RANLIB) $@
 
+shared: $(ALLOBJ)
+	gcc $(CFLAGS) -shared -Wl,-soname,liblapack.so.3 -o liblapack.so.3.1.1 $(ALLOBJ) -lblas -lm -lgfortran -lc
+
 single: $(SLASRC) $(ALLAUX) $(SCLAUX) 
 	$(ARCH) $(ARCHFLAGS) ../$(LAPACKLIB) $(SLASRC) $(ALLAUX) \
 	$(SCLAUX)
--- lapack-3.1.1/make.inc.orig	2007-02-23 20:07:35.000000000 +0000
+++ lapack-3.1.1/make.inc	2008-05-07 14:42:41.000000000 +0100
@@ -10,7 +10,7 @@
 #
 #  The machine (platform) identifier to append to the library names
 #
-PLAT = _LINUX
+PLAT = 
 #  
 #  Modify the FORTRAN and OPTS definitions to refer to the
 #  compiler and desired compiler options for your machine.  NOOPT
@@ -18,21 +18,21 @@
 #  selected.  Define LOADER and LOADOPTS to refer to the loader and 
 #  desired load options for your machine.
 #
-FORTRAN  = g77 
-OPTS     = -funroll-all-loops -O3
+FORTRAN  = gfortran
+OPTS     = -O2
 DRVOPTS  = $(OPTS)
-NOOPT    =
-LOADER   = g77
+NOOPT    = -O0
+LOADER   = gfortran
 LOADOPTS =
 #
 # Timer for the SECOND and DSECND routines
 #
 # Default : SECOND and DSECND will use a call to the EXTERNAL FUNCTION ETIME
-TIMER    = EXT_ETIME
+# TIMER    = EXT_ETIME
 # For RS6K : SECOND and DSECND will use a call to the EXTERNAL FUNCTION ETIME_
 # TIMER    = EXT_ETIME_
 # For gfortran compiler: SECOND and DSECND will use a call to the INTERNAL FUNCTION ETIME
-# TIMER    = INT_ETIME 
+TIMER    = INT_ETIME
 # If your Fortran compiler does not provide etime (like Nag Fortran Compiler, etc...)
 # SECOND and DSECND will use a call to the INTERNAL FUNCTION CPU_TIME 
 # TIMER    = INT_CPU_TIME
