--- lcdproc-0.5.6/LCDd.conf.orig	2012-08-19 15:29:07.000000000 +0100
+++ lcdproc-0.5.6/LCDd.conf	2013-03-14 21:02:08.282676308 +0000
@@ -34,7 +34,7 @@
 #            the driver modules and will thus not be able to
 #            function properly.
 # NOTE: Always place a slash as last character !
-DriverPath=server/drivers/
+DriverPath=/usr/lib/lcdproc/
 
 # Tells the server to load the given drivers. Multiple lines can be given.
 # The name of the driver is case sensitive and determines the section
@@ -50,7 +50,7 @@
 #   mdm166a, ms6931, mtc_s16209x, MtxOrb, mx5000, NoritakeVFD, picolcd,
 #   pyramid, sdeclcd, sed1330, sed1520, serialPOS, serialVFD, shuttleVFD, sli,
 #   stv5730, svga, t6963, text, tyan, ula200, vlsys_m428, xosd
-Driver=curses
+Driver=imonlcd
 
 # Tells the driver to bind to the given interface
 Bind=127.0.0.1
@@ -642,7 +642,7 @@
 # [legal: 0, 1; default: 0]
 # Choose 0 for 15c2:ffdc device,
 # Choose 1 for 15c2:0038 device
-Protocol=0
+Protocol=1
 
 # Set the exit behavior [legal: 0-2; default: 1]
 # 0 means leave shutdown message,
