--- /usr/local/etc/LCDd.conf.orig	2008-04-04 19:08:21.000000000 -0400
+++ /usr/local/etc/LCDd.conf	2008-04-04 19:09:08.000000000 -0400
@@ -41,7 +41,7 @@
 #   irman, joy, lb216, lcdm001, lcterm, lirc, MD8800, ms6931, mtc_s16209x,
 #   MtxOrb, NoritakeVFD, picolcd, pyramid, sed1330, sed1520, serialPOS,
 #   serialVFD, sli, stv5730, svga, t6963, text, tyan, ula200, xosd
-Driver=curses
+Driver=imonlcd
 
 # Tells the driver to bind to the given interface
 Bind=127.0.0.1
@@ -76,7 +76,7 @@
 #            the driver modules and will thus not be able to
 #            function properly.
 # NOTE: Always place a slash as last character !
-DriverPath=server/drivers/
+DriverPath=/usr/local/lib/lcdproc/
 
 # GoodBye message: each entry represents a display line; default: builtin
 #GoodBye="Thanks for using"
@@ -953,4 +953,8 @@
 # font to use, in XLFD format, as given by "xfontsel"
 Font=-*-terminus-*-r-*-*-*-320-*-*-*-*-*
 
+[imonlcd]
+Device=/dev/lcd0
+Contrast=200
+
 # EOF
