--- leptonica-1.78.0/CMakeLists.txt.orig	2019-02-22 20:56:12.000000000 +0000
+++ leptonica-1.78.0/CMakeLists.txt	2019-10-13 20:16:19.962973580 +0100
@@ -174,7 +174,7 @@
 get_target_property(leptonica_VERSION leptonica VERSION)
 get_target_property(leptonica_OUTPUT_NAME leptonica OUTPUT_NAME)
 configure_file(lept.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/lept.pc @ONLY)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/lept.pc DESTINATION lib/pkgconfig)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/lept.pc DESTINATION lib${LIB_SUFFIX}/pkgconfig)
 install(FILES
     ${CMAKE_CURRENT_BINARY_DIR}/LeptonicaConfig.cmake
     ${CMAKE_CURRENT_BINARY_DIR}/LeptonicaConfig-version.cmake
--- leptonica-1.78.0/src/CMakeLists.txt.orig	2018-05-31 23:08:00.000000000 +0100
+++ leptonica-1.78.0/src/CMakeLists.txt	2019-10-13 20:15:28.173976051 +0100
@@ -79,7 +79,7 @@
     export(TARGETS leptonica APPEND FILE ${CMAKE_BINARY_DIR}/LeptonicaTargets.cmake)
 endif()
 
-install(TARGETS leptonica EXPORT LeptonicaTargets RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+install(TARGETS leptonica EXPORT LeptonicaTargets RUNTIME DESTINATION bin LIBRARY DESTINATION lib${LIB_SUFFIX} ARCHIVE DESTINATION lib)
 install(EXPORT LeptonicaTargets DESTINATION cmake)
 install(FILES
     ${hdr}
