--- libarea-51e6778/CMakeLists.txt.orig	2016-12-05 20:57:32.000000000 +0000
+++ libarea-51e6778/CMakeLists.txt	2019-04-22 08:32:01.606661149 +0100
@@ -40,8 +40,11 @@
 include_directories(${Python_Includes})
 include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 
+set(BUILD_SHARED_LIBS ON)
 find_package( Boost COMPONENTS python REQUIRED)  # find BOOST and boost-python
+set(BUILD_SHARED_LIBS OFF)
 if(Boost_FOUND)
+    set(Boost_LIBRARIES ${Boost_LIBRARIES}  Boost::python)
     include_directories(${Boost_INCLUDE_DIRS})
     MESSAGE(STATUS "found Boost: " ${Boost_LIB_VERSION})
     MESSAGE(STATUS "boost-incude dirs are: " ${Boost_INCLUDE_DIRS})
@@ -124,8 +127,8 @@
 #    set(CMAKE_INSTALL_LIBDIR ${CMAKE_INSTALL_PREFIX}/lib)
 #endif(DEFINED CMAKE_INSTALL_LIBDIR)
 
-install(TARGETS libarea LIBRARY DESTINATION lib/ COMPONENT libraries)
-message(STATUS "Library will be installed to: " ${CMAKE_INSTALL_PREFIX}/lib)
+install(TARGETS libarea LIBRARY DESTINATION lib${LIB_SUFFIX}/ COMPONENT libraries)
+message(STATUS "Library will be installed to: " ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX})
 
 
 # this makes the Python module
