--- libarea-51e6778/CMakeLists.txt.orig	2016-12-05 20:57:32.000000000 +0000
+++ libarea-51e6778/CMakeLists.txt	2016-12-07 11:45:33.474635370 +0000
@@ -124,8 +124,8 @@
 #    set(CMAKE_INSTALL_LIBDIR ${CMAKE_INSTALL_PREFIX}/lib)
 #endif(DEFINED CMAKE_INSTALL_LIBDIR)
 
-install(TARGETS libarea LIBRARY DESTINATION lib/ COMPONENT libraries)
-message(STATUS "Library will be installed to: " ${CMAKE_INSTALL_PREFIX}/lib)
+install(TARGETS libarea LIBRARY DESTINATION lib${LIB_SUFFIX}/ COMPONENT libraries)
+message(STATUS "Library will be installed to: " ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX})
 
 
 # this makes the Python module
