--- libcanberra-0.30/src/Makefile.in.orig	2012-09-25 01:27:23.000000000 +0100
+++ libcanberra-0.30/src/Makefile.in	2016-06-22 19:43:02.279356753 +0100
@@ -78,7 +78,7 @@
 @HAVE_CACHE_TRUE@	cache.c cache.h
 
 @HAVE_CACHE_TRUE@am__append_3 = \
-@HAVE_CACHE_TRUE@	$(TDB_CLFAGS) \
+@HAVE_CACHE_TRUE@	$(TDB_CFLAGS) \
 @HAVE_CACHE_TRUE@	-DCA_MACHINE_ID=\"$(localstatedir)/lib/dbus/machine-id\"
 
 @HAVE_CACHE_TRUE@am__append_4 = \
