--- libcdr-0.1.4/src/lib/CDRTransforms.h.orig	2017-04-15 11:17:20.000000000 +0100
+++ libcdr-0.1.4/src/lib/CDRTransforms.h	2019-07-14 09:24:27.194568995 +0100
@@ -24,6 +24,7 @@
   CDRTransform();
   CDRTransform(double v0, double v1, double x0, double v3, double v4, double y0);
   CDRTransform(const CDRTransform &trafo);
+  CDRTransform& operator=(const CDRTransform&) = default;
 
   void applyToPoint(double &x, double &y) const;
   void applyToArc(double &rx, double &ry, double &rotation, bool &sweep, double &endx, double &endy) const;
@@ -53,6 +54,7 @@
   CDRTransforms();
   CDRTransforms(const CDRTransforms &trafos);
   ~CDRTransforms();
+  CDRTransforms& operator=(const CDRTransforms&) = default;
 
   void append(double v0, double v1, double x0, double v3, double v4, double y0);
   void append(const CDRTransform &trafo);
--- libcdr-0.1.4/src/lib/CDRTypes.h.orig	2017-04-15 11:17:20.000000000 +0100
+++ libcdr-0.1.4/src/lib/CDRTypes.h	2019-07-14 07:22:16.123347156 +0100
@@ -370,6 +370,7 @@
     : m_name(name), m_encoding(encoding) {}
   CDRFont(const CDRFont &font)
     : m_name(font.m_name), m_encoding(font.m_encoding) {}
+  CDRFont& operator=(const CDRFont&) = default;
   librevenge::RVNGString m_name;
   unsigned short m_encoding;
 };
