--- libdmtx-0.6.0/util/dmtxwrite/dmtxwrite.c.orig	2008-11-07 18:37:36.000000000 +0000
+++ libdmtx-0.6.0/util/dmtxwrite/dmtxwrite.c	2010-06-19 12:25:37.000000000 +0100
@@ -425,7 +425,7 @@
    if(infoPtr == NULL) {
       if(fp != stdout)
          fclose(fp);
-      png_destroy_write_struct(&pngPtr,  png_infopp_NULL);
+      png_destroy_write_struct(&pngPtr,  NULL);
       perror(programName);
    }
 
