--- libde265-1.0.3/sherlock265/VideoDecoder.cc.orig	2018-04-18 15:35:25.000000000 +0100
+++ libde265-1.0.3/sherlock265/VideoDecoder.cc	2018-04-25 22:05:05.866466571 +0100
@@ -237,7 +237,7 @@
     }
     width = img->get_width();
     height = img->get_height();
-    sws = sws_getContext(width, height, PIX_FMT_YUV420P, width, height, PIX_FMT_BGRA, SWS_FAST_BILINEAR, NULL, NULL, NULL);
+    sws = sws_getContext(width, height, AV_PIX_FMT_YUV420P, width, height, AV_PIX_FMT_BGRA, SWS_FAST_BILINEAR, NULL, NULL, NULL);
   }
 
   int stride[3];
