--- libdbusmenu-12.10.2/tools/testapp/main.c.orig	2012-10-03 16:54:22.000000000 +0100
+++ libdbusmenu-12.10.2/tools/testapp/main.c	2015-05-10 21:51:34.056388586 +0100
@@ -135,8 +135,6 @@
 
 int main (int argc, char ** argv)
 {
-	g_type_init();
-
 	if (argc != 2) {
 		g_warning(USAGE);
 		return 1;
--- libdbusmenu-12.10.2/tools/dbusmenu-dumper.c.orig	2012-10-03 16:54:22.000000000 +0100
+++ libdbusmenu-12.10.2/tools/dbusmenu-dumper.c	2015-05-10 21:52:11.073385833 +0100
@@ -388,7 +388,6 @@
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
 	GError * error = NULL;
 	GOptionContext * context;
 
