--- libglpng-1.45/src/glpng.c.orig	2000-07-10 20:27:10.000000000 +0100
+++ libglpng-1.45/src/glpng.c	2008-12-31 16:14:52.000000000 +0000
@@ -29,7 +29,7 @@
 #include <GL/gl.h>
 #include <stdlib.h>
 #include <math.h>
-#include "png/png.h"
+#include <png.h>
 
 /* Used to decide if GL/gl.h supports the paletted extension */
 #ifdef GL_COLOR_INDEX1_EXT
--- libglpng-1.45/Example/Test.c.orig	2000-07-10 20:23:42.000000000 +0100
+++ libglpng-1.45/Example/Test.c	2008-12-31 16:19:17.000000000 +0000
@@ -6,8 +6,8 @@
  * http://www.wyatt100.freeserve.co.uk/download.htm
  */
 
-#include <gl/glpng.h>
-#include <gl/glut.h>
+#include <GL/glpng.h>
+#include <GL/glut.h>
 #include <stdlib.h>
 
 int angle = 0;
--- libglpng-1.45/Makefile.orig	2008-12-31 16:17:53.000000000 +0000
+++ libglpng-1.45/Makefile	2008-12-31 16:17:00.000000000 +0000
@@ -0,0 +1,35 @@
+CFLAGS=-Wall -g -O3 -fPIC -I$(shell pwd)/include
+LDFLAGS=-lpng -lGL
+SHAREDLIBFLAGS=-shared
+DESTDIR=/usr/local
+DOCDIR=$(DESTDIR)/share/doc/libglpng
+
+all: libglpng.a libglpng.so.1.45
+
+libglpng.a: glpng.o
+	ar rv $@ $<
+
+libglpng.so.1.45: glpng.o
+	gcc $(CFLAGS) $(SHAREDLIBFLAGS) -Wl,-soname=libglpng.so.1 -Wl,--whole-archive $< -Wl,--no-whole-archive $(LDFLAGS) -o $@
+
+glpng.o: src/glpng.c
+	gcc $(CFLAGS) -c $<
+
+clean:
+	rm glpng.o libglpng.*
+
+install: libglpng.a libglpng.so.1.45
+	for i in include include/GL lib; do \
+		install -m 755 -d $(DESTDIR)/$$i; \
+	done
+	for i in $(DOCDIR) $(DOCDIR)/examples; do \
+		install -m 755 -d $$i; \
+	done
+	install -m 644 glpng.htm $(DOCDIR)/glpng.html
+	install -m 644 Example/Stunt.png Example/Test.c $(DOCDIR)/examples
+	install -m 644 include/GL/glpng.h $(DESTDIR)/include/GL
+	install -m 644 libglpng.* $(DESTDIR)/lib
+	ln -s libglpng.so.1.45 $(DESTDIR)/lib/libglpng.so.1
+	ln -s libglpng.so.1.45 $(DESTDIR)/lib/libglpng.so
+
+.PHONY: clean install
