--- libgpod-0.6.0/tools/ipod-scsi-inquiry.c.orig	2007-11-10 09:40:27.000000000 +0000
+++ libgpod-0.6.0/tools/ipod-scsi-inquiry.c	2008-06-29 07:40:21.000000000 +0100
@@ -32,6 +32,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#include <stdint.h>
 #include <scsi/sg_cmds.h>
 #include <sys/stat.h>
 
--- libgpod-0.6.0/configure.orig	2007-11-10 11:16:17.000000000 +0000
+++ libgpod-0.6.0/configure	2008-06-29 07:39:02.000000000 +0100
@@ -20204,13 +20204,13 @@
     eval echo $libdir
 )`
 
-{ echo "$as_me:$LINENO: checking for sg_ll_inquiry in -lsgutils" >&5
-echo $ECHO_N "checking for sg_ll_inquiry in -lsgutils... $ECHO_C" >&6; }
+{ echo "$as_me:$LINENO: checking for sg_ll_inquiry in -lsgutils2" >&5
+echo $ECHO_N "checking for sg_ll_inquiry in -lsgutils2... $ECHO_C" >&6; }
 if test "${ac_cv_lib_sgutils_sg_ll_inquiry+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lsgutils  $LIBS"
+LIBS="-lsgutils2  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -20266,7 +20266,7 @@
 { echo "$as_me:$LINENO: result: $ac_cv_lib_sgutils_sg_ll_inquiry" >&5
 echo "${ECHO_T}$ac_cv_lib_sgutils_sg_ll_inquiry" >&6; }
 if test $ac_cv_lib_sgutils_sg_ll_inquiry = yes; then
-  SGUTILS_LIBS="-lsgutils"; have_sgutils=yes
+  SGUTILS_LIBS="-lsgutils2"; have_sgutils=yes
 else
   have_sgutils=no
 fi
