--- libgphoto2-2.1.6/configure.orig	2005-06-23 09:42:55.000000000 +0100
+++ libgphoto2-2.1.6/configure	2006-04-25 11:48:26.000000000 +0100
@@ -29068,7 +29068,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 $ac_includes_default
-#include <exif-ifd.h>
+#include <libexif/exif-ifd.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
@@ -29111,7 +29111,7 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <exif-ifd.h>
+#include <libexif/exif-ifd.h>
 _ACEOF
 if { (eval echo "$as_me:$LINENO: \"$ac_cpp conftest.$ac_ext\"") >&5
   (eval $ac_cpp conftest.$ac_ext) 2>conftest.er1
