--- libgdiplus-2.6.4/configure.orig	2010-04-13 12:30:18.000000000 +0100
+++ libgdiplus-2.6.4/configure	2010-06-12 08:39:45.000000000 +0100
@@ -14427,14 +14427,14 @@
 
 fi
 
-    { $as_echo "$as_me:$LINENO: checking for libpng12" >&5
-$as_echo_n "checking for libpng12... " >&6; }
-    if $PKG_CONFIG --exists libpng12 ; then
+    { $as_echo "$as_me:$LINENO: checking for libpng14" >&5
+$as_echo_n "checking for libpng14... " >&6; }
+    if $PKG_CONFIG --exists libpng14 ; then
         { $as_echo "$as_me:$LINENO: result: yes" >&5
 $as_echo "yes" >&6; }
         PNG='png'
-	PNG_DEP_CFLAGS_PACKAGES=libpng12
-	LIBPNG=`$PKG_CONFIG --libs libpng12`
+	PNG_DEP_CFLAGS_PACKAGES=libpng14
+	LIBPNG=`$PKG_CONFIG --libs libpng14`
     else
       { $as_echo "$as_me:$LINENO: result: no" >&5
 $as_echo "no" >&6; }
@@ -14639,8 +14639,8 @@
 
 
 else
-  { { $as_echo "$as_me:$LINENO: error: *** libpng12 not found. See http://www.libpng.org/pub/png/libpng.html." >&5
-$as_echo "$as_me: error: *** libpng12 not found. See http://www.libpng.org/pub/png/libpng.html." >&2;}
+  { { $as_echo "$as_me:$LINENO: error: *** libpng14 not found. See http://www.libpng.org/pub/png/libpng.html." >&5
+$as_echo "$as_me: error: *** libpng14 not found. See http://www.libpng.org/pub/png/libpng.html." >&2;}
    { (exit 1); exit 1; }; }
 fi
 
@@ -14694,13 +14694,13 @@
         if test "$png_ok" = yes; then
           PNG='png'; LIBPNG='-lpng -lz'
         else
-          { { $as_echo "$as_me:$LINENO: error: *** libpng12 found" >&5
-$as_echo "$as_me: error: *** libpng12 found" >&2;}
+          { { $as_echo "$as_me:$LINENO: error: *** libpng14 found" >&5
+$as_echo "$as_me: error: *** libpng14 found" >&2;}
    { (exit but it is too old. See http://www.libpng.org/pub/png/libpng.html.); exit but it is too old. See http://www.libpng.org/pub/png/libpng.html.; }; }
         fi
       else
-          { { $as_echo "$as_me:$LINENO: error: *** libpng12 not found. See http://www.libpng.org/pub/png/libpng.html." >&5
-$as_echo "$as_me: error: *** libpng12 not found. See http://www.libpng.org/pub/png/libpng.html." >&2;}
+          { { $as_echo "$as_me:$LINENO: error: *** libpng14 not found. See http://www.libpng.org/pub/png/libpng.html." >&5
+$as_echo "$as_me: error: *** libpng14 not found. See http://www.libpng.org/pub/png/libpng.html." >&2;}
    { (exit 1); exit 1; }; }
       fi
     fi
