--- libgda-1.9.103/providers/mysql/gda-mysql-recordset.c.orig	2006-08-29 14:23:29.000000000 +0100
+++ libgda-1.9.103/providers/mysql/gda-mysql-recordset.c	2006-09-12 07:30:32.000000000 +0100
@@ -123,7 +123,9 @@
 	switch (type) {
 	case FIELD_TYPE_DECIMAL :
 	case FIELD_TYPE_DOUBLE :
+#if NDB_VERSION_MAJOR >= 5
 	case FIELD_TYPE_NEWDECIMAL :
+#endif
 		gda_value_reset_with_type (gda_value, G_TYPE_DOUBLE);
 		g_value_set_double (gda_value, atof (value));
 		break;
