--- libgphoto2-2.4.0/libgphoto2_port/usb/libusb.c.orig	2007-07-27 01:36:15.000000000 +0100
+++ libgphoto2-2.4.0/libgphoto2_port/usb/libusb.c	2007-10-28 14:17:58.000000000 +0000
@@ -289,7 +289,7 @@
 			port->settings.usb.interface);
 		return (GP_ERROR_IO);
 	}
-
+#if 0
 	/* This is only for our very special Canon cameras which need a good
 	 * whack after close, otherwise they get timeouts on reconnect.
 	 */
@@ -299,7 +299,7 @@
 			return (GP_ERROR_IO);
 		}
 	}
-
+#endif
 	if (usb_close (port->pl->dh) < 0) {
 		gp_port_set_error (port, _("Could not close USB port (%m)."));
 		return (GP_ERROR_IO);
