--- libiff-20230711/src/iffpp/Makefile.am.orig	2023-07-11 18:24:02.000000000 +0100
+++ libiff-20230711/src/iffpp/Makefile.am	2023-07-11 19:47:13.680535784 +0100
@@ -1,4 +1,4 @@
-iffpp.1: main.c
+iffpp.1: iffpp
 	$(HELP2MAN) --output=$@ --no-info --name 'IFF pretty printer' --libtool ./iffpp
 
 AM_CPPFLAGS = -DHAVE_GETOPT_H=$(HAVE_GETOPT_H)
@@ -9,6 +9,6 @@
 
 iffpp_SOURCES = main.c pp.c
 iffpp_LDADD = ../libiff/libiff.la
-iffpp_CFLAGS = -I../libiff
+iffpp_CFLAGS = -I$(top_srcdir)/src/libiff -I$(top_builddir)/src/libiff
 
 EXTRA_DIST = iffpp.1
--- libiff-20230711/src/iffjoin/Makefile.am.orig	2023-07-11 18:24:02.000000000 +0100
+++ libiff-20230711/src/iffjoin/Makefile.am	2023-07-11 19:46:55.101973658 +0100
@@ -1,4 +1,4 @@
-iffjoin.1: main.c
+iffjoin.1: iffjoin
 	$(HELP2MAN) --output=$@ --no-info --name 'Joins an arbitrary number of IFF files into a single concatenation IFF file' --libtool ./iffjoin
 
 AM_CPPFLAGS = -DHAVE_GETOPT_H=$(HAVE_GETOPT_H)
@@ -9,6 +9,6 @@
 
 iffjoin_SOURCES = main.c join.c
 iffjoin_LDADD = ../libiff/libiff.la
-iffjoin_CFLAGS = -I../libiff
+iffjoin_CFLAGS = -I$(top_srcdir)/src/libiff -I$(top_builddir)/src/libiff
 
 EXTRA_DIST = iffjoin.1
