--- libindicate-0.4.1/bindings/python/Makefile.in.orig	2010-06-04 20:35:27.000000000 +0100
+++ libindicate-0.4.1/bindings/python/Makefile.in	2010-06-08 21:24:33.000000000 +0100
@@ -313,7 +313,7 @@
 pkgindicate_PYTHON = __init__.py
 indicatedir = $(pkgpyexecdir)/indicate
 indicate_LTLIBRARIES = _indicate.la
-_indicate_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_indicate -lpyglib-2.0-python2.6 -lpython2.6
+_indicate_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_indicate -lpyglib-2.0-python -lpython2.6
 _indicate_la_LIBADD = $(INDICATE_PYTHON_LIBS)  -L$(top_builddir)/libindicate/.libs/ -lindicate -L$(top_builddir)/libindicate-gtk/.libs -lindicate-gtk $(LIBINDICATEGTK_LIBS)
 _indicate_la_SOURCES = indicatemodule.c
 nodist__indicate_la_SOURCES = indicate.c
