--- libilbm-20230710/src/ilbmpp/Makefile.am.orig	2023-07-10 22:30:03.000000000 +0100
+++ libilbm-20230710/src/ilbmpp/Makefile.am	2023-07-11 19:53:37.367517477 +0100
@@ -1,4 +1,4 @@
-ilbmpp.1: main.c
+ilbmpp.1: ilbmpp
 	$(HELP2MAN) --output=$@ --no-info --name ' Displays a textual representation of an IFF/ILBM file on the standard output' --libtool ./ilbmpp
 
 AM_CPPFLAGS = -DHAVE_GETOPT_H=$(HAVE_GETOPT_H)
@@ -9,6 +9,6 @@
 
 ilbmpp_SOURCES = main.c pp.c
 ilbmpp_LDADD = ../libilbm/libilbm.la
-ilbmpp_CFLAGS = -I../libilbm $(LIBIFF_CFLAGS)
+ilbmpp_CFLAGS = -I$(top_srcdir)/src/libilbm -I$(top_builddir)/src/libilbm $(LIBIFF_CFLAGS)
 
 EXTRA_DIST = ilbmpp.1
--- libilbm-20230710/src/acbm2ilbm/Makefile.am.orig	2023-07-10 22:30:03.000000000 +0100
+++ libilbm-20230710/src/acbm2ilbm/Makefile.am	2023-07-11 19:51:41.840522989 +0100
@@ -1,4 +1,4 @@
-acbm2ilbm.1: main.c
+acbm2ilbm.1: acbm2ilbm
 	$(HELP2MAN) --output=$@ --no-info --name 'Convert all ACBM images inside an IFF file to ILBM images' --libtool ./acbm2ilbm
 
 AM_CPPFLAGS = -DHAVE_GETOPT_H=$(HAVE_GETOPT_H)
@@ -9,6 +9,6 @@
 
 acbm2ilbm_SOURCES = main.c acbm2ilbm.c
 acbm2ilbm_LDADD = ../libilbm/libilbm.la
-acbm2ilbm_CFLAGS = -I../libilbm $(LIBIFF_CFLAGS)
+acbm2ilbm_CFLAGS = -I$(top_srcdir)/src/libilbm -I$(top_builddir)/src/libilbm $(LIBIFF_CFLAGS)
 
 EXTRA_DIST = acbm2ilbm.1
--- libilbm-20230710/src/ilbmpack/Makefile.am.orig	2023-07-10 22:30:03.000000000 +0100
+++ libilbm-20230710/src/ilbmpack/Makefile.am	2023-07-11 19:53:12.127518681 +0100
@@ -1,4 +1,4 @@
-ilbmpack.1: main.c
+ilbmpack.1: ilbmpack
 	$(HELP2MAN) --output=$@ --no-info --name 'Compress or decompress ILBM images inside an IFF file' --libtool ./ilbmpack
 
 AM_CPPFLAGS = -DHAVE_GETOPT_H=$(HAVE_GETOPT_H)
@@ -9,6 +9,6 @@
 
 ilbmpack_SOURCES = main.c pack.c
 ilbmpack_LDADD = ../libilbm/libilbm.la
-ilbmpack_CFLAGS = -I../libilbm $(LIBIFF_CFLAGS)
+ilbmpack_CFLAGS = -I$(top_srcdir)/src/libilbm -I$(top_builddir)/src/libilbm $(LIBIFF_CFLAGS)
 
 EXTRA_DIST = ilbmpack.1
--- libilbm-20230710/src/ilbm2acbm/Makefile.am.orig	2023-07-10 22:30:03.000000000 +0100
+++ libilbm-20230710/src/ilbm2acbm/Makefile.am	2023-07-11 19:52:44.850519983 +0100
@@ -1,4 +1,4 @@
-ilbm2acbm.1: main.c
+ilbm2acbm.1: ilbm2acbm
 	$(HELP2MAN) --output=$@ --no-info --name 'Convert all ILBM images inside an IFF file to ACBM images' --libtool ./ilbm2acbm
 
 AM_CPPFLAGS = -DHAVE_GETOPT_H=$(HAVE_GETOPT_H)
@@ -9,6 +9,6 @@
 
 ilbm2acbm_SOURCES = main.c ilbm2acbm.c
 ilbm2acbm_LDADD = ../libilbm/libilbm.la
-ilbm2acbm_CFLAGS = -I../libilbm $(LIBIFF_CFLAGS)
+ilbm2acbm_CFLAGS = -I$(top_srcdir)/src/libilbm -I$(top_builddir)/src/libilbm $(LIBIFF_CFLAGS)
 
 EXTRA_DIST = ilbm2acbm.1
