--- libinklevel-0.7.0/Makefile.orig	2007-03-13 22:30:45.000000000 +0000
+++ libinklevel-0.7.0/Makefile	2007-03-14 21:19:53.000000000 +0000
@@ -5,15 +5,14 @@
 
 install: all
 	mkdir -p $(DESTDIR)/include
-	install -m 644 -o root -g root inklevel.h $(DESTDIR)/include/inklevel.h
+	install -m 644 inklevel.h $(DESTDIR)/include/inklevel.h
 	mkdir -p $(DESTDIR)/lib
-	install -m 755 -o root -g root libinklevel.so.4.0.7.0 \
+	install -m 755 libinklevel.so.4.0.7.0 \
 	$(DESTDIR)/lib/libinklevel.so.4.0.7.0
 	cd $(DESTDIR)/lib && rm -fr libinklevel.so && \
 	ln -s libinklevel.so.4.0.7.0 libinklevel.so
 	cd $(DESTDIR)/lib && rm -fr libinklevel.so.4 && \
 	ln -s libinklevel.so.4.0.7.0 libinklevel.so.4
-	/sbin/ldconfig $(DESTDIR)/lib
 
 clean:
 	rm -fr *.o *.so.4.0.7.0 *~
