--- libiodbc-3.52.9/samples/Makefile.in.orig	2014-04-15 12:01:55.000000000 +0100
+++ libiodbc-3.52.9/samples/Makefile.in	2015-02-15 09:31:29.343830914 +0000
@@ -283,11 +283,11 @@
 #
 iodbctest_SOURCES = iodbctest.c
 iodbctest_LDADD = ../iodbc/libiodbc.la
-iodbctest_LDFLAGS = -static
+iodbctest_LDFLAGS =
 iodbctestw_CFLAGS = -DUNICODE
 iodbctestw_SOURCES = iodbctest.c
 iodbctestw_LDADD = ../iodbc/libiodbc.la
-iodbctestw_LDFLAGS = -static
+iodbctestw_LDFLAGS =
 EXTRA_DIST = $(srcdir)/Makefile.sample 
 all: all-am
 
