--- libical-3.0.15/src/libical/icalparser.c.orig	2022-10-09 16:52:46.242775425 +0100
+++ libical-3.0.15/src/libical/icalparser.c	2022-10-09 17:42:54.624631887 +0100
@@ -630,7 +630,7 @@
                                 icalparser_line_gen_func line_gen_func)
 {
     char *line;
-    unsigned int cnt = 0;
+
     icalcomponent *c = 0;
     icalcomponent *root = 0;
     icalerrorstate es = icalerror_get_error_state(ICAL_MALFORMEDDATA_ERROR);
@@ -641,7 +641,6 @@
     icalerror_set_error_state(ICAL_MALFORMEDDATA_ERROR, ICAL_ERROR_NONFATAL);
 
     do {
-        cnt++;
         line = icalparser_get_line(parser, line_gen_func);
 
         printf("got: %s\n", line);
@@ -683,7 +682,7 @@
             icalmemory_free_buffer(line);
             cont = 1;
         }
-    } while (cont && cnt < TMP_BUF_SIZE);
+    } while (cont);
 
     icalerror_set_error_state(ICAL_MALFORMEDDATA_ERROR, es);
 
