--- libical-0.44/src/libicalss/icalspanlist.h.orig	2009-09-27 03:38:52.000000000 +0100
+++ libical-0.44/src/libicalss/icalspanlist.h	2010-07-04 15:47:10.000000000 +0100
@@ -26,7 +26,7 @@
 #define ICALSPANLIST_H
 
 #include <libical/ical.h>
-#include <icalset.h>
+#include "icalset.h"
 
 /** @file icalspanlist.h
  *  @brief Code that supports collections of free/busy spans of time
--- libical-0.44/src/libicalss/icaldirset.h.orig	2009-09-27 03:38:52.000000000 +0100
+++ libical-0.44/src/libicalss/icaldirset.h	2010-07-04 15:47:10.000000000 +0100
@@ -30,9 +30,9 @@
 #define ICALDIRSET_H
 
 #include <libical/ical.h>
-#include <icalset.h>
-#include <icalcluster.h>
-#include <icalgauge.h>
+#include "icalset.h"
+#include "icalcluster.h"
+#include "icalgauge.h"
 
 /* icaldirset Routines for storing, fetching, and searching for ical
  * objects in a database */
--- libical-0.44/src/libicalss/icalcalendar.h.orig	2009-09-27 03:38:52.000000000 +0100
+++ libical-0.44/src/libicalss/icalcalendar.h	2010-07-04 15:47:10.000000000 +0100
@@ -30,7 +30,7 @@
 #define ICALCALENDAR_H
 
 #include <libical/ical.h>
-#include <icalset.h>
+#include "icalset.h"
 
 /* icalcalendar
  * Routines for storing calendar data in a file system. The calendar 
--- libical-0.44/src/libicalss/icalfileset.h.orig	2009-09-27 03:38:52.000000000 +0100
+++ libical-0.44/src/libicalss/icalfileset.h	2010-07-04 15:47:10.000000000 +0100
@@ -30,9 +30,9 @@
 #define ICALFILESET_H
 
 #include <libical/ical.h>
-#include <icalset.h>
-#include <icalcluster.h>
-#include <icalgauge.h>
+#include "icalset.h"
+#include "icalcluster.h"
+#include "icalgauge.h"
 #include <sys/types.h> /* For open() flags and mode */
 #include <sys/stat.h> /* For open() flags and mode */
 #include <fcntl.h> /* For open() flags and mode */
--- libical-0.44/src/libicalss/icalgauge.h.orig	2009-09-27 03:38:52.000000000 +0100
+++ libical-0.44/src/libicalss/icalgauge.h	2010-07-04 15:47:10.000000000 +0100
@@ -29,7 +29,7 @@
 #ifndef ICALGAUGE_H
 #define ICALGAUGE_H
 
-#include <icalcomponent.h>
+#include "icalcomponent.h"
 
 /** @file icalgauge.h
  *  @brief Routines implementing a filter for ical components
--- libical-0.44/src/libicalss/icalcluster.h.orig	2009-09-27 03:38:52.000000000 +0100
+++ libical-0.44/src/libicalss/icalcluster.h	2010-07-04 15:47:10.000000000 +0100
@@ -30,7 +30,7 @@
 #define ICALCLUSTER_H
 
 #include <libical/ical.h>
-#include <icalset.h>
+#include "icalset.h"
 
 typedef struct icalcluster_impl icalcluster;
 
--- libical-0.44/src/libicalss/icalclassify.h.orig	2009-09-27 03:38:52.000000000 +0100
+++ libical-0.44/src/libicalss/icalclassify.h	2010-07-04 15:47:10.000000000 +0100
@@ -28,8 +28,8 @@
 #define ICALCLASSIFY_H
 
 #include <libical/ical.h>
-#include <icalset.h>
-#include <icalcomponent.h>
+#include "icalset.h"
+#include "icalcomponent.h"
 
 icalproperty_xlicclass icalclassify(icalcomponent* c,icalcomponent* match, 
 			      const char* user);
--- libical-0.44/src/libicalss/icalset.h.orig	2009-09-27 03:38:52.000000000 +0100
+++ libical-0.44/src/libicalss/icalset.h	2010-07-04 15:47:10.000000000 +0100
@@ -41,7 +41,7 @@
 
 #include <limits.h> /* For PATH_MAX */
 #include <libical/ical.h>
-#include <icalgauge.h>
+#include "icalgauge.h"
 
 #ifdef PATH_MAX
 #define ICAL_PATH_MAX PATH_MAX
--- libical-0.44/src/libicalcap/icalcap_message_impl.h.orig	2009-09-27 03:38:49.000000000 +0100
+++ libical-0.44/src/libicalcap/icalcap_message_impl.h	2010-07-04 15:47:10.000000000 +0100
@@ -3,7 +3,7 @@
 
 #ifdef WITH_RR
 
-#include <ical.h>
+#include <libical/ical.h>
 #include <librr/rr.h>
 
 #define	ICALCAP_MESSAGE_CMD	1
--- libical-0.44/src/libicalcap/icalcap.h.orig	2009-09-27 03:38:49.000000000 +0100
+++ libical-0.44/src/libicalcap/icalcap.h	2010-07-04 15:47:10.000000000 +0100
@@ -1,7 +1,7 @@
 #ifndef __ICALCAP_H__
 #define __ICALCAP_H__
 
-#include <ical.h>
+#include <libical/ical.h>
 
 /*
  * Opaque objects
--- libical-0.44/src/libical/icaltz-util.c.orig	2009-09-27 03:38:51.000000000 +0100
+++ libical-0.44/src/libical/icaltz-util.c	2010-07-04 15:47:10.000000000 +0100
@@ -81,8 +81,8 @@
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
-#include <libical/icalerror.h>
-#include <icaltz-util.h>
+#include "icalerror.h"
+#include "icaltz-util.h"
 
 typedef struct 
 {
