--- libipoddevice-0.5.3/src/hal-ipod-info.c.orig	2007-03-26 18:24:12.000000000 +0100
+++ libipoddevice-0.5.3/src/hal-ipod-info.c	2008-06-29 07:36:05.000000000 +0100
@@ -6,6 +6,7 @@
 #include <errno.h>
 #include <sys/types.h>
 #include <fcntl.h>
+#include <stdint.h>
 #include <scsi/sg_cmds.h>
 #include <libhal.h>
 #include <glib.h>
--- libipoddevice-0.5.3/configure.orig	2007-04-01 05:57:33.000000000 +0100
+++ libipoddevice-0.5.3/configure	2008-06-29 07:32:33.000000000 +0100
@@ -20863,13 +20863,13 @@
 
 fi
 
-{ echo "$as_me:$LINENO: checking for sg_ll_inquiry in -lsgutils" >&5
-echo $ECHO_N "checking for sg_ll_inquiry in -lsgutils... $ECHO_C" >&6; }
+{ echo "$as_me:$LINENO: checking for sg_ll_inquiry in -lsgutils2" >&5
+echo $ECHO_N "checking for sg_ll_inquiry in -lsgutils2... $ECHO_C" >&6; }
 if test "${ac_cv_lib_sgutils_sg_ll_inquiry+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lsgutils  $LIBS"
+LIBS="-lsgutils2  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -20941,7 +20941,7 @@
 { echo "$as_me:$LINENO: result: $ac_cv_lib_sgutils_sg_ll_inquiry" >&5
 echo "${ECHO_T}$ac_cv_lib_sgutils_sg_ll_inquiry" >&6; }
 if test $ac_cv_lib_sgutils_sg_ll_inquiry = yes; then
-  HAL_IPOD_INFO_LIBS="-lsgutils"
+  HAL_IPOD_INFO_LIBS="-lsgutils2"
 else
   { { echo "$as_me:$LINENO: error: Error! You need to have libsgutils." >&5
 echo "$as_me: error: Error! You need to have libsgutils." >&2;}
