--- libmsym-0.2.3-paper/CMakeLists.txt.orig	2016-06-15 20:43:21.000000000 +0100
+++ libmsym-0.2.3-paper/CMakeLists.txt	2020-04-24 17:39:41.866689836 +0100
@@ -14,13 +14,13 @@
 set(LIBMSYM_PATCH_VERSION 4)
 set(LIBMSYM_VERSION ${LIBMSYM_MAJOR_VERSION}.${LIBMSYM_MINOR_VERSION}.${LIBMSYM_PATCH_VERSION})
 
-set(INSTALL_LIB_DIR lib CACHE PATH "Installation directory for msym library")
+set(INSTALL_LIB_DIR lib${LIB_SUFFIX} CACHE PATH "Installation directory for msym library")
 set(INSTALL_INCLUDE_DIR include CACHE PATH "Installation directory for header files")
 
 if(WIN32 AND NOT CYGWIN)
   set(DEF_INSTALL_CMAKE_DIR CMake)
 else()
-  set(DEF_INSTALL_CMAKE_DIR lib/cmake/libmsym)
+  set(DEF_INSTALL_CMAKE_DIR lib${LIB_SUFFIX}/cmake/libmsym)
 endif()
 set(INSTALL_CMAKE_DIR ${DEF_INSTALL_CMAKE_DIR} CACHE PATH "Installation directory for CMake files")
 
