--- libmovtar-0.1.3/movtar_play.c.orig	2001-07-13 22:38:03.000000000 +0100
+++ libmovtar-0.1.3/movtar_play.c	2010-07-03 16:47:02.000000000 +0100
@@ -102,7 +102,7 @@
 }
 
 GLOBAL(void)
-jpeg_mem_src (j_decompress_ptr cinfo,void * buff, int size)
+my_jpeg_mem_src (j_decompress_ptr cinfo,void * buff, int size)
 {
   mem_src_ptr src;
 
@@ -805,7 +805,7 @@
     ComplainAndExit(); 
   cinfo.err = jpeg_std_error(&jerr);	
   jpeg_create_decompress(&cinfo);
-  jpeg_mem_src(&cinfo, readbuffer, 200000);
+  my_jpeg_mem_src(&cinfo, readbuffer, 200000);
 
   sprintf(wintitle, "movtar_play %s", argv[1]);
   SDL_WM_SetCaption(wintitle, "0000000");  
--- libmovtar-0.1.3/rtjpeg/RTjpeg.c.orig	2001-06-15 10:13:39.000000000 +0100
+++ libmovtar-0.1.3/rtjpeg/RTjpeg.c	2008-09-14 19:54:48.000000000 +0100
@@ -129,7 +129,7 @@
  register int ci, co=1, tmp;
  register __s16 ZZvalue;
 
- (__u8)strm[0]=(__u8)(data[RTjpeg_ZZ[0]]>254) ? 254:((data[RTjpeg_ZZ[0]]<0)?0:data[RTjpeg_ZZ[0]]);
+ strm[0]=(__u8)(data[RTjpeg_ZZ[0]]>254) ? 254:((data[RTjpeg_ZZ[0]]<0)?0:data[RTjpeg_ZZ[0]]);
  
  for(ci=1; ci<=bt8; ci++) 
  {
--- libmovtar-0.1.3/rtjpeg/RTjpeg.h.orig	2001-06-15 10:13:39.000000000 +0100
+++ libmovtar-0.1.3/rtjpeg/RTjpeg.h	2008-09-14 19:59:29.000000000 +0100
@@ -25,19 +25,21 @@
 #ifndef __RTjpeg_h__
 #define __RTjpeg_h__
 
+#include <asm/types.h>
+
 #ifndef _I386_TYPES_H
 typedef unsigned char __u8;
 typedef unsigned short __u16;
-typedef unsigned long __u32;
+//typedef unsigned long __u32;
 typedef unsigned long long __u64;
 typedef signed char __s8;
 typedef signed short __s16;
-typedef signed long __s32;
+//typedef signed long __s32;
 #endif
 
 extern void RTjpeg_init_Q(__u8 Q);
-extern void RTjpeg_init_compress(long unsigned int *buf, int width, int height, __u8 Q);
-extern void RTjpeg_init_decompress(long unsigned int *buf, int width, int height);
+extern void RTjpeg_init_compress(__u32 *buf, int width, int height, __u8 Q);
+extern void RTjpeg_init_decompress(__u32 *buf, int width, int height);
 extern int RTjpeg_compressYUV420(__s8 *sp, unsigned char *bp);
 extern int RTjpeg_compressYUV422(__s8 *sp, unsigned char *bp);
 extern void RTjpeg_decompressYUV420(__s8 *sp, __u8 *bp);
