--- libmpeg3-1.7/Makefile.orig	2006-02-15 05:09:31.000000000 +0000
+++ libmpeg3-1.7/Makefile	2008-01-10 22:26:22.000000000 +0000
@@ -149,7 +149,7 @@
 
 LIBS = -lm -lpthread
 
-$(shell if ! test -d $(OBJDIR) \; then mkdir -p $(OBJDIR) \; fi )
+$(shell if ! test -d $(OBJDIR) ; then mkdir -p $(OBJDIR) ; fi )
 
 $(shell echo $(CFLAGS) > $(OBJDIR)/c_flags)
 $(shell echo $(A52CFLAGS) > $(OBJDIR)/a52_flags)
@@ -192,8 +192,8 @@
 
 install: 
 	cp $(UTILS) $(PREFIX)/bin
-#	cp $(OUTPUT) $(PREFIX)/lib
-#	cp libmpeg3.h mpeg3private.h $(PREFIX)/include
+	cp $(OUTPUT) $(PREFIX)/lib
+	cp libmpeg3.h mpeg3private.h mpeg3protos.h $(PREFIX)/include
 
 clean:
 	rm -rf $(OBJDIR)
