--- libnfs-4.0.0/CMakeLists.txt.orig	2019-02-12 21:38:04.000000000 +0000
+++ libnfs-4.0.0/CMakeLists.txt	2019-02-17 21:14:54.206131746 +0000
@@ -6,11 +6,11 @@
 
 set(SOVERSION 11.0.0 CACHE STRING "" FORCE)
 
-set(INSTALL_LIB_DIR "${CMAKE_INSTALL_PREFIX}/lib" CACHE PATH "Installation directory for libraries")
+set(INSTALL_LIB_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}" CACHE PATH "Installation directory for libraries")
 set(INSTALL_INC_DIR "${CMAKE_INSTALL_PREFIX}/include" CACHE PATH "Installation directory for headers")
 set(INSTALL_MAN_DIR "${CMAKE_INSTALL_PREFIX}/share/man" CACHE PATH "Installation directory for manual pages")
-set(INSTALL_PKGCONFIG_DIR "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig" CACHE PATH "Installation directory for pkgconfig (.pc) files")
-set(INSTALL_CMAKE_DIR "${CMAKE_INSTALL_PREFIX}/lib/cmake/libnfs" CACHE PATH "Installation directory for cmake (.cmake) files")
+set(INSTALL_PKGCONFIG_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/pkgconfig" CACHE PATH "Installation directory for pkgconfig (.pc) files")
+set(INSTALL_CMAKE_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/cmake/libnfs" CACHE PATH "Installation directory for cmake (.cmake) files")
 
 option(BUILD_SHARED_LIBS "Build shared libraries" ON)
 option(ENABLE_TESTS "Build and run test programs" OFF)
--- libnfs-4.0.0/lib/CMakeLists.txt.orig	2019-02-12 21:38:04.000000000 +0000
+++ libnfs-4.0.0/lib/CMakeLists.txt	2019-02-17 21:16:39.272126733 +0000
@@ -18,5 +18,5 @@
 
 install(TARGETS nfs EXPORT nfs
                     RUNTIME DESTINATION bin
-                    ARCHIVE DESTINATION lib
-                    LIBRARY DESTINATION lib)
+                    ARCHIVE DESTINATION lib${LIB_SUFFIX}
+                    LIBRARY DESTINATION lib${LIB_SUFFIX})
