--- libnet-1.1.4/src/libnet_if_addr.c.orig	2009-06-05 01:31:44.000000000 +0100
+++ libnet-1.1.4/src/libnet_if_addr.c	2009-07-14 22:14:42.000000000 +0100
@@ -240,6 +240,7 @@
         {
             /* fix memory leak */
             free(al->device);
+	    al->device = NULL;
         }
         if ((al->device = strdup(device)) == NULL)
         {
@@ -406,6 +407,7 @@
     for (i = 0; i < c; i++)
     {
         free(al[i].device);
+	al[i].device = NULL;
     }
     return (1);
 
@@ -413,6 +415,7 @@
     for (i = 0; i < c; i++)
     {
         free(al[i].device);
+	al[i].device = NULL;
     }
     return (-1);
 }
