--- libofa-0.9.4-test/examples/example.cpp.orig	2008-02-21 15:16:20.000000000 +0000
+++ libofa-0.9.4-test/examples/example.cpp	2011-02-25 19:30:00.284999300 +0000
@@ -7,6 +7,8 @@
 
 -------------------------------------------------------------------*/
 
+#include <cstring>
+#include <cstdio>
 #include <ctype.h>
 #include "protocol.h"
 
--- libofa-0.9.4-test/examples/protocol.cpp.orig	2008-02-21 15:16:20.000000000 +0000
+++ libofa-0.9.4-test/examples/protocol.cpp	2011-02-25 19:30:24.443002099 +0000
@@ -8,6 +8,7 @@
 -------------------------------------------------------------------*/
 #include <stdio.h>
 #include <stdlib.h>
+#include <cstring>
 #include <string>
 #include <map>
 #include <expat.h>
--- libofa-0.9.4-test/examples/protocol.cpp.orig	2019-07-14 13:55:28.452885799 +0100
+++ libofa-0.9.4-test/examples/protocol.cpp	2019-07-14 14:20:48.878813255 +0100
@@ -13,7 +13,6 @@
 #include <map>
 #include <expat.h>
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 using namespace std;
