--- liborigin-20070115/Makefile.LINUX.orig	2007-01-08 19:38:05.000000000 +0000
+++ liborigin-20070115/Makefile.LINUX	2007-06-06 11:41:27.000000000 +0100
@@ -15,9 +15,9 @@
 	rm -f *~ *.o opj2dat $(TARGET).so*
 
 install:
-	chmod ugo-x $(TARGET).so.0.0.0
+	chmod ugo-x $(TARGET).so.0.0.1
 	mkdir -p $(DESTDIR)/usr/lib
-	cp $(TARGET).so.0.0.0 $(DESTDIR)/usr/lib/
+	cp $(TARGET).so.0.0.1 $(DESTDIR)/usr/lib/
 	(cd $(DESTDIR)/usr/lib/;\
 	ln -s $(TARGET).so.0.0.1 $(TARGET).so;\
 	ln -s $(TARGET).so.0.0.1 $(TARGET).so.0;\
