--- libofx-0.9.1/ofxdump/ofxdump.cpp.orig	2006-08-02 05:19:22.000000000 +0100
+++ libofx-0.9.1/ofxdump/ofxdump.cpp	2010-06-22 05:58:34.000000000 +0100
@@ -32,6 +32,8 @@
 #include <string>
 #include "libofx.h"
 #include <stdio.h>		/* for printf() */
+#include <string.h>
+#include <stdlib.h>
 #include <config.h>		/* Include config constants, e.g., VERSION TF */
 #include <errno.h>
 
--- libofx-0.9.1/lib/ofx_request_statement.cpp.orig	2007-10-28 03:01:38.000000000 +0000
+++ libofx-0.9.1/lib/ofx_request_statement.cpp	2010-06-22 05:57:02.000000000 +0100
@@ -22,6 +22,7 @@
 #endif
 
 #include <string>
+#include <stdlib.h>
 #include "libofx.h"
 #include "ofx_utilities.hh"
 #include "ofx_request_statement.hh"
--- libofx-0.9.1/lib/ofx_request_accountinfo.cpp.orig	2007-10-28 03:01:38.000000000 +0000
+++ libofx-0.9.1/lib/ofx_request_accountinfo.cpp	2010-06-22 05:56:39.000000000 +0100
@@ -22,6 +22,7 @@
 #endif
 
 #include <string>
+#include <stdlib.h>
 #include "libofx.h"
 #include "ofx_request_accountinfo.hh"
 
--- libofx-0.9.1/lib/ofx_container_transaction.cpp.orig	2007-11-09 23:13:21.000000000 +0000
+++ libofx-0.9.1/lib/ofx_container_transaction.cpp	2010-06-22 05:55:33.000000000 +0100
@@ -22,6 +22,7 @@
 #endif
 
 #include <string>
+#include <stdlib.h>
 #include "messages.hh"
 #include "libofx.h"
 #include "ofx_containers.hh"
--- libofx-0.9.1/lib/ofx_request.cpp.orig	2007-10-28 03:01:38.000000000 +0000
+++ libofx-0.9.1/lib/ofx_request.cpp	2010-06-22 05:56:02.000000000 +0100
@@ -22,6 +22,7 @@
 #endif
 
 #include <string>
+#include <string.h>
 #include "messages.hh"
 #include "libofx.h"
 #include "ofx_request.hh"
--- libofx-0.9.1/lib/ofx_utilities.cpp.orig	2007-10-27 13:15:58.000000000 +0100
+++ libofx-0.9.1/lib/ofx_utilities.cpp	2010-06-22 05:54:34.000000000 +0100
@@ -21,6 +21,7 @@
 
 #include "ParserEventGeneratorKit.h"
 #include "SGMLApplication.h"
+#include <stdlib.h>
 #include <time.h>
 #include <string>
 #include <locale.h>
@@ -113,7 +114,7 @@
   char time_zone_specified = false;
 
   time_t temptime;
-  std::time(&temptime);
+  ::time(&temptime);
   local_offset = difftime(mktime(localtime(&temptime)), mktime(gmtime(&temptime)));
   
   if(ofxdate.size()!=0){
