--- libopensync-plugin-gnokii-0.22/src/gnokii_contact_format.c.orig	2007-03-27 12:49:51.000000000 +0100
+++ libopensync-plugin-gnokii-0.22/src/gnokii_contact_format.c	2008-04-27 22:15:03.000000000 +0100
@@ -142,7 +142,6 @@
 			case GN_PHONEBOOK_ENTRY_LogoSwitch:	
 			case GN_PHONEBOOK_ENTRY_RingtoneAdv:	
 			// TODO support new CVS entries
-			/*	
 			case GN_PHONEBOOK_ENTRY_Location:
 			case GN_PHONEBOOK_ENTRY_Image:
 			case GN_PHONEBOOK_ENTRY_UserID:
@@ -161,7 +160,6 @@
 			case GN_PHONEBOOK_ENTRY_Company:
 			case GN_PHONEBOOK_ENTRY_Nickname:
 			case GN_PHONEBOOK_ENTRY_Birthday:
-			*/
 				break;
 		}
 
