--- libpng-1.2.34/pngconf.h.orig	2008-12-18 11:45:16.000000000 +0000
+++ libpng-1.2.34/pngconf.h	2008-12-18 19:01:06.000000000 +0000
@@ -323,8 +323,8 @@
      /* If you encounter a compiler error here, see the explanation
       * near the end of INSTALL.
       */
-         __pngconf.h__ already includes setjmp.h;
-         __dont__ include it again.;
+/*         __pngconf.h__ already includes setjmp.h;
+         __dont__ include it again.; */
 #    endif
 #  endif /* __linux__ */
 
