--- libpagemaker-0.0.4/src/lib/PMDParser.cpp.orig	2018-01-08 12:27:29.000000000 +0000
+++ libpagemaker-0.0.4/src/lib/PMDParser.cpp	2019-07-13 22:24:34.991239748 +0100
@@ -852,7 +852,7 @@
     *tocLength = readU16(m_input, m_bigEndian);
     PMD_DEBUG_MSG(("[Header] TOC length is %d\n", *tocLength));
   }
-  catch (PMDStreamException)
+  catch (PMDStreamException&)
   {
     throw PMDParseException("Can't find the table of contents length in the header.");
   }
@@ -862,7 +862,7 @@
     *tocOffset = readU32(m_input, m_bigEndian);
     PMD_DEBUG_MSG(("[Header] TOC offset is 0x%x\n", *tocOffset));
   }
-  catch (PMDStreamException)
+  catch (PMDStreamException&)
   {
     throw PMDParseException("Can't find the table of contents offset in the header.");
   }
