--- libphonenumber-8.10.3/cpp/CMakeLists.txt.orig	2019-04-22 09:07:32.315954958 +0100
+++ libphonenumber-8.10.3/cpp/CMakeLists.txt	2019-04-22 17:11:37.658079277 +0100
@@ -97,9 +97,13 @@
   if (WIN32)
     set (Boost_USE_STATIC_LIBS ON)
   endif ()
+  set(BUILD_SHARED_LIBS ON)
   find_package (Boost 1.40.0 COMPONENTS date_time system thread)
+  set(BUILD_SHARED_LIBS OFF)
   if (NOT Boost_FOUND)
     print_error ("Boost Date_Time/System/Thread" "Boost")
+  else ()
+    set(Boost_LIBRARIES ${Boost_LIBRARIES} Boost::date_time Boost::system Boost::thread )
   endif ()
   include_directories (${Boost_INCLUDE_DIRS})
 endif ()
@@ -513,7 +517,7 @@
   list (APPEND TEST_LIBS pthread)
 endif ()
 
-target_link_libraries (libphonenumber_test ${TEST_LIBS})
+target_link_libraries (libphonenumber_test ${TEST_LIBS} Boost::thread)
 
 # Unfortunately add_custom_target() can't accept a single command provided as a
 # list of commands.
@@ -611,7 +615,7 @@
     geocoding_test_program
     "test/phonenumbers/geocoding/geocoding_test_program.cc"
   )
-  target_link_libraries (geocoding_test_program geocoding phonenumber)
+  target_link_libraries (geocoding_test_program geocoding phonenumber Boost::thread)
 endif ()
 
 # Build an RPM
