--- libphonenumber-9.0.9/cpp/CMakeLists.txt.orig	2025-07-07 08:32:50.000000000 +0100
+++ libphonenumber-9.0.9/cpp/CMakeLists.txt	2025-08-17 15:25:45.346387726 +0100
@@ -145,7 +145,7 @@
   if (WIN32)
     set (Boost_USE_STATIC_LIBS ON)
   endif ()
-  find_package (Boost 1.40.0 COMPONENTS date_time system thread)
+  find_package (Boost 1.40.0 COMPONENTS date_time thread)
   if (NOT Boost_FOUND)
     print_error ("Boost Date_Time/System/Thread" "Boost")
   endif ()
@@ -418,7 +418,7 @@
 endif ()
 
 if (NOT WIN32)
-  add_definitions ("-Wall -Werror")
+  add_definitions ("-Wall")
 endif ()
 
 include_directories ("src")
