--- libpng-1.2.10/scripts/makefile.linux.orig	2006-04-23 04:45:06.000000000 +0100
+++ libpng-1.2.10/scripts/makefile.linux	2006-04-25 22:43:36.000000000 +0100
@@ -110,12 +110,12 @@
 	$(LN_SF) $(LIBSOVER) $(LIBSOMAJ)
 
 $(LIBSOVER): $(OBJSDLL)
-	$(CC) -shared -Wl,-soname,$(LIBSOMAJ) -o $(LIBSOVER) $(OBJSDLL)
+	$(CC) -shared -Wl,-soname,$(LIBSOMAJ) -o $(LIBSOVER) $(OBJSDLL)  -lz -lm
 
 $(OLDSOVER): $(OBJSDLL)
 	$(CC) -shared -Wl,-soname,$(OLDSOMAJ) \
 	-o $(OLDSOVER) \
-	$(OBJSDLL)
+	$(OBJSDLL)  -lz -lm
 
 pngtest: pngtest.o $(LIBSO)
 	$(CC) -o pngtest $(CFLAGS) pngtest.o $(LDFLAGS)
