--- libpst-0.6.72/src/Makefile.in.orig	2018-08-01 16:16:00.000000000 +0100
+++ libpst-0.6.72/src/Makefile.in	2019-04-22 14:34:00.865673701 +0100
@@ -355,7 +355,7 @@
 
 @STATIC_TOOLS_FALSE@libpst_la_LDFLAGS = $(NO_UNDEFINED) -version-info @LIBPST_VERSION_INFO@
 libpst_la_SOURCES = $(common_source) $(common_header)
-libpst_la_LIBADD = $(LTLIBICONV)
+libpst_la_LIBADD = $(LTLIBICONV) $(ZLIB_LIBS)
 EXTRA_DIST = $(am__append_4)
 
 # set the include path found by configure
--- libpst-0.6.72/configure.orig	2018-08-01 16:16:01.000000000 +0100
+++ libpst-0.6.72/configure	2019-04-22 14:00:11.188770543 +0100
@@ -23478,7 +23478,7 @@
 $as_echo_n "checking for python build information... " >&6; }
 { $as_echo "$as_me:$LINENO: result: " >&5
 $as_echo "" >&6; }
-for python in python3.7 python3.6 python3.5 python3.4 python3.3 python3.2 python3.1 python3.0 python2.7 python2.6 python2.5 python2.4 python2.3 python2.2 python2.1 python; do
+for python in python2.7 python3.7 python3.6 python3.5 python3.4 python3.3 python3.2 python3.1 python3.0 python2.7 python2.6 python2.5 python2.4 python2.3 python2.2 python2.1 python; do
 for ac_prog in $python
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
