--- scripts/makefile.linux.orig	2004-12-03 00:14:31.000000000 +0000
+++ scripts/makefile.linux	2005-08-01 19:59:53.000000000 +0100
@@ -99,12 +99,12 @@
 $(LIBNAME).so.$(PNGVER): $(OBJSDLL)
 	$(CC) -shared -Wl,-soname,$(LIBNAME).so.$(PNGMAJ) \
 	-o $(LIBNAME).so.$(PNGVER) \
-	$(OBJSDLL)
+	$(OBJSDLL) -lz -lm
 
 libpng.so.3.$(PNGMIN): $(OBJSDLL)
 	$(CC) -shared -Wl,-soname,libpng.so.3 \
 	-o libpng.so.3.$(PNGMIN) \
-	$(OBJSDLL)
+	$(OBJSDLL) -lz -lm
 
 pngtest: pngtest.o $(LIBNAME).so
 	$(CC) -o pngtest $(CFLAGS) pngtest.o $(LDFLAGS)
