--- libproxy-0.4.4/libproxy/libproxy-1.0.pc.in.orig	2010-06-13 16:07:48.000000000 +0100
+++ libproxy-0.4.4/libproxy/libproxy-1.0.pc.in	2010-06-24 06:52:43.000000000 +0100
@@ -1,6 +1,6 @@
-prefix=@prefix@
-libdir=@libdir@
-includedir=@includedir@
+prefix=@CMAKE_INSTALL_PREFIX@
+libdir=@LIB_INSTALL_DIR@
+includedir=@INCLUDE_INSTALL_DIR@
 
 Name: libproxy-1.0
 Description: Proxy Configuration Library
