--- libquicktime-1.0.3/plugins/x264/x264.c.orig	2008-07-01 23:27:11.000000000 +0100
+++ libquicktime-1.0.3/plugins/x264/x264.c	2008-10-06 07:04:42.000000000 +0100
@@ -776,7 +776,7 @@
   INTPARAM("x264_i_keyint_min", codec->params.i_keyint_min);
   INTPARAM("x264_i_scenecut_threshold", codec->params.i_scenecut_threshold);
   INTPARAM("x264_i_bframe", codec->params.i_bframe);
-  INTPARAM("x264_b_bframe_adaptive", codec->params.b_bframe_adaptive);
+  INTPARAM("x264_b_bframe_adaptive", codec->params.i_bframe_adaptive);
   INTPARAM("x264_i_bframe_bias", codec->params.i_bframe_bias);
   INTPARAM("x264_b_bframe_pyramid", codec->params.b_bframe_pyramid);
 
@@ -867,12 +867,12 @@
   ENUMPARAM("x264_i_me_method", codec->params.analyse.i_me_method, me_methods);
 
   INTPARAM("x264_i_subpel_refine", codec->params.analyse.i_subpel_refine);
-  INTPARAM("x264_b_bframe_rdo", codec->params.analyse.b_bframe_rdo);
+//  INTPARAM("x264_b_bframe_rdo", codec->params.analyse.b_bframe_rdo);
   INTPARAM("x264_i_me_range", codec->params.analyse.i_me_range);
   INTPARAM("x264_i_frame_reference", codec->params.i_frame_reference);
   INTPARAM("x264_b_chroma_me", codec->params.analyse.b_chroma_me);
   INTPARAM("x264_b_mixed_references", codec->params.analyse.b_mixed_references);
-  INTPARAM("x264_b_bidir_me", codec->params.analyse.b_bidir_me);
+//  INTPARAM("x264_b_bidir_me", codec->params.analyse.b_bidir_me);
   INTPARAM("x264_b_weighted_bipred", codec->params.analyse.b_weighted_bipred);
     
   ENUMPARAM("x264_i_direct_mv_pred", codec->params.analyse.i_direct_mv_pred, direct_modes);
--- libquicktime-1.0.3/plugins/ffmpeg/audio.c.orig	2007-11-11 00:38:12.000000000 +0000
+++ libquicktime-1.0.3/plugins/ffmpeg/audio.c	2008-10-06 06:45:26.000000000 +0100
@@ -671,7 +671,7 @@
     //  priv->ctx->block_align     = s->data.audio.block_align;
     //  priv->ctx->bit_rate        = s->codec_bitrate;
 
-    codec->avctx->bits_per_sample = quicktime_audio_bits(file, track);
+    codec->avctx->bits_per_coded_sample = quicktime_audio_bits(file, track);
 
     /* Some codecs need extra stuff */
 
--- libquicktime-1.0.3/plugins/ffmpeg/video.c.orig	2008-05-28 22:57:45.000000000 +0100
+++ libquicktime-1.0.3/plugins/ffmpeg/video.c	2008-10-06 06:45:46.000000000 +0100
@@ -377,7 +377,7 @@
     {
     codec->avctx->width           = width;
     codec->avctx->height          = height;
-    codec->avctx->bits_per_sample = quicktime_video_depth(file, track);
+    codec->avctx->bits_per_coded_sample = quicktime_video_depth(file, track);
 
     /* Set extradata: It's done differently for each codec */
 
--- libquicktime-1.0.3/plugins/ffmpeg/params.c.orig	2008-05-14 21:50:41.000000000 +0100
+++ libquicktime-1.0.3/plugins/ffmpeg/params.c	2008-10-06 06:49:27.000000000 +0100
@@ -241,7 +241,7 @@
   PARAM_FLAG("ff_flag_emu_edge",CODEC_FLAG_EMU_EDGE);
   PARAM_FLAG("ff_flag_normalize_aqp",CODEC_FLAG_NORMALIZE_AQP);
   PARAM_FLAG("ff_flag_alt_scan",CODEC_FLAG_ALT_SCAN);
-  PARAM_FLAG("ff_flag_trellis_quant",CODEC_FLAG_TRELLIS_QUANT);
+//  PARAM_FLAG("ff_flag_trellis_quant",CODEC_FLAG_TRELLIS_QUANT);
   PARAM_FLAG("ff_flag_bitexact",CODEC_FLAG_BITEXACT);
   PARAM_FLAG("ff_flag_ac_pred",CODEC_FLAG_AC_PRED);
   PARAM_FLAG("ff_flag_h263p_umv",CODEC_FLAG_H263P_UMV);
