--- libquicktime-20180425/utils/lqtplay.c.orig	2018-04-25 21:49:51.000000000 +0100
+++ libquicktime-20180425/utils/lqtplay.c	2026-06-20 18:07:18.761900462 +0100
@@ -1477,7 +1477,7 @@
     while(!done) {
         /* Decode new audio samples */
         if(!qt_audio_samples_in_buffer)
-          decode_audio(AUDIO_BLOCK_SIZE);
+          decode_audio();
         ret = snd_pcm_writei(pcm_handle, (void *)(qt_audio_ptr), qt_audio_samples_in_buffer);
         if (ret == -EAGAIN) { 
             ret = 0;
@@ -1515,7 +1515,7 @@
     int rc, n, revents;
 
     if(!qt_audio_samples_in_buffer)
-      decode_audio(AUDIO_BLOCK_SIZE);
+      decode_audio();
 
     /* this code is absolutely horrible.  do not follow this example. */
 
@@ -1557,7 +1557,7 @@
     int rc;
 
     if(!qt_audio_samples_in_buffer)
-      decode_audio(AUDIO_BLOCK_SIZE);
+      decode_audio();
     
     rc = write(oss_fd,qt_audio_ptr,qt_audio_samples_in_buffer * qt_channels * sizeof(*qt_audio));
     switch (rc) {
