--- libQuotient-0.9.5/CMakeLists.txt.orig	2025-09-20 12:57:20.000000000 +0100
+++ libQuotient-0.9.5/CMakeLists.txt	2025-10-08 22:40:55.284120580 +0100
@@ -370,7 +370,7 @@
 
 target_link_libraries(${QUOTIENT_LIB_NAME}
     PUBLIC ${Qt}::Core ${Qt}::Network ${Qt}::Gui qt${${Qt}Core_VERSION_MAJOR}keychain Olm::Olm ${Qt}::Sql
-    PRIVATE OpenSSL::Crypto ${Qt}::CorePrivate)
+    PRIVATE OpenSSL::Crypto $<BUILD_INTERFACE:${Qt}::CorePrivate>)
 
 configure_file(${PROJECT_NAME}.pc.in ${CMAKE_CURRENT_BINARY_DIR}/${QUOTIENT_LIB_NAME}.pc @ONLY NEWLINE_STYLE UNIX)
 
