--- libqalculate-0.9.7/configure.in.orig	2010-01-05 15:15:27.000000000 +0000
+++ libqalculate-0.9.7/configure.in	2013-08-09 20:12:32.723456529 +0100
@@ -8,7 +8,7 @@
 
 AC_INIT(configure.in)
 AM_INIT_AUTOMAKE(libqalculate, 0.9.7)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AM_MAINTAINER_MODE
 
 AC_ARG_ENABLE(textport, 
@@ -29,7 +29,6 @@
 AC_ISC_POSIX
 AC_PROG_CC
 AC_PROG_CXX
-AM_PROG_CC_STDC
 AC_PROG_LN_S
 AC_HEADER_STDC
 AC_PROG_INTLTOOL
--- libqalculate-0.9.7/data/currencies.xml.in.orig	2009-12-02 20:24:25.000000000 +0000
+++ libqalculate-0.9.7/data/currencies.xml.in	2013-08-09 20:10:44.037461714 +0100
@@ -23,7 +23,7 @@
       <_names>ar:GBP,au:&#xA3;</_names>
     </builtin_unit>
     <builtin_unit name="THB">
-      <_title>Thai Bat</_title>
+      <_title>Thai Baht</_title>
       <_names>ar:THB</_names>
     </builtin_unit>
     <builtin_unit name="SEK">
@@ -51,7 +51,7 @@
       <_names>ar:RUB,ruble</_names>
     </builtin_unit>
     <builtin_unit name="PHP">
-      <_title>Phillipine Peso</_title>
+      <_title>Philippine Peso</_title>
       <_names>ar:PHP</_names>
     </builtin_unit>
     <builtin_unit name="MYR">
@@ -63,7 +63,7 @@
       <_names>ar:HRK</_names>
     </builtin_unit>
     <builtin_unit name="CNY">
-      <_title>Yuan Renmimbi (PR China)</_title>
+      <_title>Yuan Renminbi (PR China)</_title>
       <_names>ar:CNY</_names>
     </builtin_unit>
     <builtin_unit name="IDR">
@@ -219,7 +219,7 @@
       </base>
     </unit>
     <unit type="alias">
-      <_title>Deutche Mark</_title>
+      <_title>Deutsche Mark</_title>
       <_names>ar:DEM,mark</_names>
       <base>
         <unit>EUR</unit>
--- libqalculate-0.9.7/data/units.xml.in.orig	2010-01-05 15:15:27.000000000 +0000
+++ libqalculate-0.9.7/data/units.xml.in	2013-08-09 20:11:23.661459824 +0100
@@ -1097,8 +1097,8 @@
     </unit>
     <unit type="alias">
       <system>SI</system>
-      <_title>Degree Celcius</_title>
-      <_names>ar:oC,au:&#xB0;C,r:celcius</_names>
+      <_title>Degree Celsius</_title>
+      <_names>ar:oC,au:&#xB0;C,r:celsius</_names>
       <base>
         <unit>K</unit>
         <relation>\x + 273.15</relation>
@@ -2508,7 +2508,7 @@
       </base>
     </unit>
     <unit type="alias">
-      <_title>Calorie (15 degrees Celcius)</_title>
+      <_title>Calorie (15 degrees Celsius)</_title>
       <_names>ars:cal_fifteen</_names>
       <base>
         <unit>J</unit>
