--- libquicktime-1.1.3/plugins/x264/x264.c.orig	2009-03-11 14:25:50.000000000 +0000
+++ libquicktime-1.1.3/plugins/x264/x264.c	2010-01-01 11:02:00.000000000 +0000
@@ -67,7 +67,7 @@
   lqt_dump("  i_bframe:                    %d\n", params->i_bframe);          // 0.. X264_BFRAME_MAX
   lqt_dump("  b_bframe_adaptive:           %d\n", params->b_bframe_adaptive);
   lqt_dump("  i_bframe_bias:               %d\n", params->i_bframe_bias);
-  lqt_dump("  b_bframe_pyramid:            %d\n", params->b_bframe_pyramid);
+  lqt_dump("  i_bframe_pyramid:            %d\n", params->i_bframe_pyramid);
   
   lqt_dump("  b_deblocking_filter:         %d\n", params->b_deblocking_filter);
   lqt_dump("  i_deblocking_filter_alphac0: %d\n", params->i_deblocking_filter_alphac0); // -6..6
@@ -782,7 +782,7 @@
   ENUMPARAM("x264_i_bframe_adaptive", codec->params.i_bframe_adaptive, bframe_adaptives);
 #endif
   INTPARAM("x264_i_bframe_bias", codec->params.i_bframe_bias);
-  INTPARAM("x264_b_bframe_pyramid", codec->params.b_bframe_pyramid);
+  INTPARAM("x264_i_bframe_pyramid", codec->params.i_bframe_pyramid);
 
   ENUMPARAM("x264_i_rc_method", codec->params.rc.i_rc_method, rc_methods);
   INTPARAM("x264_i_bitrate", codec->params.rc.i_bitrate);
