--- libsndfile-1.0.18pre24h/configure.orig	2008-10-27 20:06:01.000000000 +0000
+++ libsndfile-1.0.18pre24h/configure	2008-12-13 23:18:01.000000000 +0000
@@ -24288,7 +24288,7 @@
 
 { echo "$as_me:$LINENO: checking for OCTAVE_VERSION in $OCTAVE" >&5
 echo $ECHO_N "checking for OCTAVE_VERSION in $OCTAVE... $ECHO_C" >&6; }
-OCTAVE_VERSION=`echo "disp(OCTAVE_VERSION)" | $OCTAVE -qfH`
+OCTAVE_VERSION=`echo "disp(OCTAVE_VERSION)" | $OCTAVE -qfH | sed -e 's/.*[a-z]//'`
 { echo "$as_me:$LINENO: result: $OCTAVE_VERSION" >&5
 echo "${ECHO_T}$OCTAVE_VERSION" >&6; }
 
