--- libsndfile-1.0.25/M4/octave.m4.orig	2010-03-16 14:38:46.000000000 +0000
+++ libsndfile-1.0.25/M4/octave.m4	2012-01-24 21:16:55.000000000 +0000
@@ -119,7 +119,7 @@
 			2.*)
 				AC_MSG_WARN([Octave version 2.X is not supported.])
 				;;
-			3.0.*)
+			3.*.*)
 				OCTAVE_DEST_ODIR=`$OCTAVE_CONFIG --oct-site-dir | sed 's%^/usr%${prefix}%'`
 				OCTAVE_DEST_MDIR=`$OCTAVE_CONFIG --m-site-dir | sed 's%^/usr%${prefix}%'`
 
