--- libsbsms-1.7.0/src/convert.cpp.orig	2008-11-13 05:34:52.000000000 +0000
+++ libsbsms-1.7.0/src/convert.cpp	2009-01-29 07:04:42.000000000 +0000
@@ -74,6 +74,7 @@
   int channels;
   long samplesToProcess;
   long samplesIn;
+  long samplesOut;
   sbsmsInfo si;
   sbsms *sbsmser = NULL;
   Resampler *resampler = NULL;
@@ -174,7 +175,7 @@
     stretch2 = 1.0/stretch0;
   else
     stretch2 = log(stretch1/stretch0)/(stretch1-stretch0);
-  long samplesOut = samplesToProcess * stretch2;
+  samplesOut = samplesToProcess * stretch2;
   si.samplesToProcess = samplesToProcess;
   si.samplesToGenerate = samplesOut;
   si.stretch0 = stretch0;
