--- libtheora-1.0alpha4/doc/Makefile.in.orig	2005-01-07 11:07:57.000000000 +0000
+++ libtheora-1.0alpha4/doc/Makefile.in	2005-01-07 11:06:20.000000000 +0000
@@ -545,14 +545,14 @@
 	fi
 
 install-data-local: doxygen-build.stamp
-	$(mkinstalldirs) $(docdir)
+	$(mkinstalldirs) $(DESTDIR)$(docdir)
 	if test -d libtheora; then \
 	  for dir in libtheora/*; do \
 	    if test -d $$dir; then \
 	      b=`basename $$dir`; \
-	      $(mkinstalldirs) $(docdir)/$$b; \
+	      $(mkinstalldirs) $(DESTDIR)$(docdir)/$$b; \
 	      for f in $$dir/*; do \
-		$(INSTALL_DATA) $$f $(docdir)/$$b; \
+		$(INSTALL_DATA) $$f $(DESTDIR)$(docdir)/$$b; \
 	      done \
 	    fi \
 	  done \
