--- libtunepimp-0.5.3/lib/protocol.cpp.orig	2011-10-29 10:40:33.000000000 +0100
+++ libtunepimp-0.5.3/lib/protocol.cpp	2011-10-29 11:40:36.000000000 +0100
@@ -13,7 +13,6 @@
 #include <map>
 #include <expat.h>
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 using namespace std;
 
--- libtunepimp-0.5.3/plugins/mp4/mp4.cpp.orig	2006-11-18 10:51:08.000000000 +0000
+++ libtunepimp-0.5.3/plugins/mp4/mp4.cpp	2011-10-29 12:09:26.000000000 +0100
@@ -231,8 +231,7 @@
   
   strcpy(mdata->fileFormat, fileName + strlen(fileName) - 3); 
   
-  if (!MP4Close(mp4file))
-    return 0;
+  MP4Close(mp4file);
   
   return 1;
 }
@@ -315,8 +314,7 @@
   sprintf(temp, "%d", mdata->nonAlbum);  
   MP4SetMetadataFreeForm(mp4file, "MusicBrainz Non-Album", (u_int8_t *)temp, strlen(temp) + 1);
   
-  if (!MP4Close(mp4file))
-    return 0;
+  MP4Close(mp4file);
 
 #ifndef WIN32
   if (!MP4Optimize(utf8ToEncoding(fileName, encoding).c_str()))
--- libtunepimp-0.5.3/plugins/vorbis/vorbis_meta.cpp.orig	2006-11-18 10:51:20.000000000 +0000
+++ libtunepimp-0.5.3/plugins/vorbis/vorbis_meta.cpp	2011-02-25 19:50:18.925000411 +0000
@@ -40,6 +40,7 @@
 #	include <unistd.h>
 #endif
 
+#include <cstring>
 #include <stdio.h>
 #include <math.h>
 #include <assert.h>
--- libtunepimp-0.5.3/plugins/wma/wma.cpp.orig	2006-11-18 10:51:28.000000000 +0000
+++ libtunepimp-0.5.3/plugins/wma/wma.cpp	2011-02-25 19:45:05.225999758 +0000
@@ -24,7 +24,7 @@
    $Id: wma.cpp 8497 2006-09-29 21:26:44Z luks $
 
 ----------------------------------------------------------------------------*/
-
+#include <cstdlib>
 #include <string.h>
 #include <stdio.h>
 #include <wmafile.h>
--- libtunepimp-0.5.3/plugins/wma/wmafile.cpp.orig	2006-11-18 10:51:28.000000000 +0000
+++ libtunepimp-0.5.3/plugins/wma/wmafile.cpp	2011-02-25 19:45:30.957997942 +0000
@@ -18,7 +18,7 @@
  *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
  *   USA                                                                   *
  ***************************************************************************/
-
+#include <cstring>
 #include <tstring.h>
 #include <wmafile.h>
 #include <wmatag.h>
--- libtunepimp-0.5.3/plugins/mp3/id3_2_3_meta.cpp.orig	2006-11-18 10:51:07.000000000 +0000
+++ libtunepimp-0.5.3/plugins/mp3/id3_2_3_meta.cpp	2011-02-25 19:48:16.346003211 +0000
@@ -25,6 +25,8 @@
 
 ----------------------------------------------------------------------------*/
 
+#include <cstdlib>
+#include <cstring>
 #include <stdio.h>
 #include <assert.h>
 #include <ctype.h>
--- libtunepimp-0.5.3/plugins/mp3/id3_meta.cpp.orig	2006-11-18 10:51:07.000000000 +0000
+++ libtunepimp-0.5.3/plugins/mp3/id3_meta.cpp	2011-02-25 19:47:41.503000333 +0000
@@ -24,7 +24,8 @@
    $Id: id3_meta.cpp 8498 2006-09-29 21:31:25Z luks $
 
 ----------------------------------------------------------------------------*/
-
+#include <cstdlib>
+#include <cstring>
 #include <stdio.h>
 #include <time.h>
 #include <ctype.h>
--- libtunepimp-0.5.3/plugins/flac/flac_meta.cpp.orig	2006-11-18 10:51:35.000000000 +0000
+++ libtunepimp-0.5.3/plugins/flac/flac_meta.cpp	2011-02-25 19:50:51.838000901 +0000
@@ -41,6 +41,7 @@
 #       include <unistd.h>
 #endif
 
+#include <cstring>
 #include <stdio.h>
 #include <math.h>
 #include <assert.h>
--- libtunepimp-0.5.3/lib/analyzer.cpp.orig	2006-11-18 10:52:33.000000000 +0000
+++ libtunepimp-0.5.3/lib/analyzer.cpp	2011-02-25 19:37:57.804999217 +0000
@@ -31,6 +31,7 @@
 #endif
 
 #include <stdio.h>
+#include <cstring>
 #ifndef WIN32
 #include <unistd.h>
 #endif
--- libtunepimp-0.5.3/lib/track.cpp.orig	2006-11-18 10:52:33.000000000 +0000
+++ libtunepimp-0.5.3/lib/track.cpp	2011-02-25 19:39:21.283000303 +0000
@@ -24,6 +24,7 @@
    $Id: track.cpp 7216 2006-04-14 23:10:49Z robert $
 
 ----------------------------------------------------------------------------*/
+#include <cstring>
 #include "../config.h"
 #include "track.h"
 #include "tunepimp.h"
--- libtunepimp-0.5.3/lib/write.cpp.orig	2006-11-18 10:52:33.000000000 +0000
+++ libtunepimp-0.5.3/lib/write.cpp	2011-02-25 19:40:13.643999953 +0000
@@ -24,6 +24,7 @@
    $Id: write.cpp 8421 2006-08-14 20:47:24Z luks $
 
 ----------------------------------------------------------------------------*/
+#include <cstring>
 #ifdef WIN32
 #  if _MSC_VER == 1200
 #       pragma warning(disable:4786)
--- libtunepimp-0.5.3/lib/c_wrapper.cpp.orig	2006-11-18 10:52:33.000000000 +0000
+++ libtunepimp-0.5.3/lib/c_wrapper.cpp	2011-02-25 19:40:56.610999704 +0000
@@ -24,6 +24,8 @@
    $Id: c_wrapper.cpp 8360 2006-08-07 21:05:12Z luks $
 
 ----------------------------------------------------------------------------*/
+#include <cstring>
+#include <cstdlib>
 #include "tunepimp.h"
 #include "mutex.h"
 #include "tp_c.h"
--- libtunepimp-0.5.3/lib/utf8/utf8util.cpp.orig	2006-11-18 10:52:26.000000000 +0000
+++ libtunepimp-0.5.3/lib/utf8/utf8util.cpp	2011-02-25 19:35:34.419999984 +0000
@@ -26,6 +26,8 @@
 ----------------------------------------------------------------------------*/
 
 #include <stdio.h>
+#include <cstring>
+#include <cstdlib>
 #include "utf8util.h"
 #include "utf8.h"
 #ifdef WIN32
--- libtunepimp-0.5.3/lib/tunepimp.cpp.orig	2006-11-18 10:52:33.000000000 +0000
+++ libtunepimp-0.5.3/lib/tunepimp.cpp	2011-02-25 19:36:57.655998765 +0000
@@ -31,6 +31,8 @@
 #endif
 
 #include <stdio.h>
+#include <cstring>
+#include <cstdlib>
 #include <map>
 using namespace std;
 
--- libtunepimp-0.5.3/lib/metadata.cpp.orig	2006-11-18 10:52:33.000000000 +0000
+++ libtunepimp-0.5.3/lib/metadata.cpp	2011-02-25 19:38:54.215000158 +0000
@@ -26,6 +26,8 @@
 ----------------------------------------------------------------------------*/
 #include <math.h>
 #include <stdio.h>
+#include <cstdlib>
+#include <cstring>
 #include "metadata.h"
 #include "../config.h"
 using namespace std;
--- libtunepimp-0.5.3/lib/protocol.cpp.orig	2006-11-28 20:25:04.000000000 +0000
+++ libtunepimp-0.5.3/lib/protocol.cpp	2011-02-25 19:44:01.827999143 +0000
@@ -6,6 +6,7 @@
    No rights reserved.
 
 -------------------------------------------------------------------*/
+#include <cstring>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string>
--- libtunepimp-0.5.3/lib/readmeta.cpp.orig	2006-11-18 10:52:33.000000000 +0000
+++ libtunepimp-0.5.3/lib/readmeta.cpp	2011-02-25 19:41:59.437000724 +0000
@@ -30,6 +30,8 @@
 #endif
 #endif
 
+#include <cstring>
+#include <cstdlib>
 #include <stdio.h>
 #ifndef WIN32
 #include <unistd.h>
--- libtunepimp-0.5.3/lib/fileio.cpp.orig	2006-11-18 10:52:33.000000000 +0000
+++ libtunepimp-0.5.3/lib/fileio.cpp	2011-02-25 19:43:36.611000063 +0000
@@ -24,7 +24,8 @@
    $Id: fileio.cpp 1416 2005-07-03 06:48:17Z robert $
 
 ----------------------------------------------------------------------------*/
-
+#include <cstring>
+#include <cstdlib>
 #include <assert.h>
 #include <errno.h>
 #include <string>
@@ -125,7 +126,7 @@
     char *ptr, *temp;
 
     temp = (char *)malloc(strlen(path) + 32);
-    ptr = strrchr(path, dirSepChar);
+    ptr = strrchr((char *)path, dirSepChar);
     if (ptr)
     {
         int len = (int)(ptr - path);
--- libtunepimp-0.5.3/lib/filecache.cpp.orig	2006-11-18 10:52:33.000000000 +0000
+++ libtunepimp-0.5.3/lib/filecache.cpp	2011-02-25 19:37:30.390000426 +0000
@@ -29,6 +29,7 @@
 #pragma warning(disable:4786)
 #endif
 #endif
+#include <cstring>
 
 #include "filecache.h"
 #include "tunepimp.h"
