--- libultrahdr-1.4.0/lib/include/ultrahdr/gainmapmath.h.orig	2025-01-10 19:31:48.000000000 +0000
+++ libultrahdr-1.4.0/lib/include/ultrahdr/gainmapmath.h	2026-01-19 16:43:47.598982805 +0000
@@ -19,6 +19,7 @@
 
 #include <array>
 #include <cmath>
+#include <cstdint>
 #include <cstring>
 #include <functional>
 
--- libultrahdr-1.4.0/lib/include/ultrahdr/ultrahdrcommon.h.orig	2025-01-10 19:31:48.000000000 +0000
+++ libultrahdr-1.4.0/lib/include/ultrahdr/ultrahdrcommon.h	2026-01-19 16:45:00.221074701 +0000
@@ -24,6 +24,7 @@
 #include <GLES3/gl3.h>
 #endif
 
+#include <cstdint>
 #include <deque>
 #include <map>
 #include <memory>
