--- libv4l-0.5.98/libv4lconvert/Makefile.orig	2009-05-25 21:35:01.000000000 +0100
+++ libv4l-0.5.98/libv4lconvert/Makefile	2009-05-27 22:53:54.000000000 +0100
@@ -3,7 +3,7 @@
 CFLAGS := -g -O1
 CFLAGS += -Wall -Wno-unused -Wpointer-arith -Wstrict-prototypes -Wmissing-prototypes
 
-LIBS_libv4lconvert = -lrt
+LIBS_libv4lconvert = -lrt -lm
 
 ifeq ($(LINKTYPE),static)
 CONVERT_LIB   = libv4lconvert.a
