--- LibVNCServer-0.9.10/CMakeLists.txt.orig	2014-10-21 16:57:11.000000000 +0100
+++ LibVNCServer-0.9.10/CMakeLists.txt	2015-02-15 10:27:21.620558324 +0000
@@ -366,8 +366,11 @@
   target_link_libraries(client_examples/${test} vncclient ${CMAKE_THREAD_LIBS_INIT} ${X11_LIBRARIES} ${SDL_LIBRARY} ${FFMPEG_LIBRARIES})
 endforeach(test ${LIBVNCCLIENT_TESTS})
 
-install_targets(/lib vncserver)
-install_targets(/lib vncclient)
+set(LIB_INSTALL_DIR      "lib${LIB_SUFFIX}"
+  CACHE STRING "Install dir for libraries" FORCE)
+
+install_targets(/${LIB_INSTALL_DIR} vncserver)
+install_targets(/${LIB_INSTALL_DIR} vncclient)
 install_files(/include/rfb FILES
     rfb/keysym.h
     rfb/rfb.h
