--- libxslt-1.1.16/libxslt/Makefile.in.orig	2006-05-01 22:33:06.000000000 +0100
+++ libxslt-1.1.16/libxslt/Makefile.in	2006-05-02 21:31:57.000000000 +0100
@@ -662,7 +662,7 @@
 	@(cd ../xsltproc ; $(MAKE))
 
 install-exec-hook:
-	$(mkinstalldirs) "$(libdir)/libxslt-plugins"
+	$(mkinstalldirs) "$(DESTDIR)$(libdir)/libxslt-plugins"
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
