--- linphone-3.10.2/configure.orig	2016-08-30 15:20:13.000000000 +0100
+++ linphone-3.10.2/configure	2017-01-20 22:21:10.916422753 +0000
@@ -21164,9 +21164,9 @@
 
 	CPPFLAGS="$CPPFLAGS $POLARSSL_CFLAGS"
 	LIBS="$LIBS $POLARSSL_LIBS"
-	for ac_header in polarssl/gcm.h
+	for ac_header in mbedtls/gcm.h
 do :
-  ac_fn_c_check_header_mongrel "$LINENO" "polarssl/gcm.h" "ac_cv_header_polarssl_gcm_h" "$ac_includes_default"
+  ac_fn_c_check_header_mongrel "$LINENO" "mbedtls/gcm.h" "ac_cv_header_polarssl_gcm_h" "$ac_includes_default"
 if test "x$ac_cv_header_polarssl_gcm_h" = xyes; then :
   cat >>confdefs.h <<_ACEOF
 #define HAVE_POLARSSL_GCM_H 1
