--- linphone-3.5.2/mediastreamer2/src/h264dec.c.orig	2012-02-22 11:54:58.000000000 +0000
+++ linphone-3.5.2/mediastreamer2/src/h264dec.c	2013-01-19 21:56:31.000000000 +0000
@@ -44,7 +44,6 @@
 static void ffmpeg_init(){
 	static bool_t done=FALSE;
 	if (!done){
-		avcodec_init();
 		avcodec_register_all();
 		done=TRUE;
 	}
--- linphone-3.5.2/mediastreamer2/src/videoenc.c.orig	2012-02-22 11:54:58.000000000 +0000
+++ linphone-3.5.2/mediastreamer2/src/videoenc.c	2013-01-19 21:49:33.000000000 +0000
@@ -53,7 +53,6 @@
 
 void ms_ffmpeg_check_init(){
 	if(!avcodec_initialized){
-		avcodec_init();
 		avcodec_register_all();
 		avcodec_initialized=TRUE;
 #ifdef ENABLE_LOG_FFMPEG
@@ -300,10 +299,10 @@
 	if (s->profile==0){
 		s->codec=CODEC_ID_H263;
 	}else{
+		/*
 		c->flags|=CODEC_FLAG_H263P_UMV;
 		c->flags|=CODEC_FLAG_AC_PRED;
 		c->flags|=CODEC_FLAG_H263P_SLICE_STRUCT;
-		/*
 		c->flags|=CODEC_FLAG_OBMC;
 		c->flags|=CODEC_FLAG_AC_PRED;
 		*/
@@ -372,7 +371,7 @@
 	// assume video size is CIF or QCIF
 	if (context->width == 352 && context->height == 288) header->b_wptr[1] = 0x60;
 	else header->b_wptr[1] = 0x40;
-	if (context->coded_frame->pict_type != FF_I_TYPE) header->b_wptr[1] |= 0x10;
+	if (context->coded_frame->pict_type != AV_PICTURE_TYPE_I) header->b_wptr[1] |= 0x10;
 	header->b_wptr += 4;
 	header->b_cont = *packet;
 	*packet = header;
@@ -779,7 +778,7 @@
 		s->req_vfu=TRUE;
 	}
 	if (s->req_vfu){
-		pict.pict_type=FF_I_TYPE;
+		pict.pict_type=AV_PICTURE_TYPE_I;
 		s->req_vfu=FALSE;
 	}
 	comp_buf->b_rptr=comp_buf->b_wptr=comp_buf->b_datap->db_base;
@@ -799,7 +798,7 @@
 		if (s->framenum==1){
 			video_starter_first_frame (&s->starter,f->ticker->time);
 		}
-		if (c->coded_frame->pict_type==FF_I_TYPE){
+		if (c->coded_frame->pict_type==AV_PICTURE_TYPE_I){
 			ms_message("Emitting I-frame");
 		}
 		comp_buf->b_wptr+=error;
--- linphone-3.5.2/gtk/setupwizard.c.orig	2012-02-22 11:54:29.000000000 +0000
+++ linphone-3.5.2/gtk/setupwizard.c	2013-01-19 22:17:14.000000000 +0000
@@ -270,7 +270,7 @@
 	linphone_account_creator_set_username(creator, gtk_entry_get_text(username));
 
 	if (g_regex_match_simple("^[a-zA-Z]+[a-zA-Z0-9.\\-_]{3,}$", gtk_entry_get_text(username), 0, 0)) {
-		g_thread_create(check_username_availability, (void*)w, FALSE, NULL);
+		g_thread_new("check_username_availability", check_username_availability, (void*)w);
 	}
 	else {
 		if (gtk_entry_get_text_length(username) < LOGIN_MIN_SIZE) {
@@ -534,7 +534,6 @@
 	ok = create_pixbuf(linphone_gtk_get_ui_config("ok","ok.png"));
 	notok = create_pixbuf(linphone_gtk_get_ui_config("notok","notok.png"));
 
-	g_thread_init (NULL);
 	gdk_threads_init ();
 
 	GtkWidget *p1=create_intro();
--- linphone-3.5.2/gtk/main.c.orig	2012-02-22 14:10:53.000000000 +0000
+++ linphone-3.5.2/gtk/main.c	2013-01-19 22:14:01.000000000 +0000
@@ -1643,7 +1643,6 @@
 	GdkPixbuf *pbuf;
 	const char *app_name="Linphone";
 
-	g_thread_init(NULL);
 	gdk_threads_init();
 	
 	progpath = strdup(argv[0]);
