--- linssid-3.6/linssid-app/MainForm.cpp.orig	2018-06-23 17:36:57.000000000 +0100
+++ linssid-3.6/linssid-app/MainForm.cpp	2026-02-04 21:40:43.985284343 +0000
@@ -28,6 +28,7 @@
 #include <qwt_plot_curve.h>
 #include <qwt_plot_grid.h>
 #include <qwt_symbol.h>
+#include <qwt_text.h>
 #include <qwt_plot_marker.h>
 #include <QPointF>
 #include <qwt_scale_draw.h>
@@ -1036,7 +1037,7 @@
                     MainForm::cellDataRay[tbi].signal ) );
             }
         } else {
-            MainForm::cellDataRay[tbi].pBandCurve->setSamples(0, 0, 0);
+            MainForm::cellDataRay[tbi].pBandCurve->setSamples((float*) 0, (float*) 0, 0);
         }
         // now the signal history plot
         int ixStart;
@@ -1055,7 +1056,7 @@
                     &(MainForm::cellDataRay[tbi].pHistory->signal[ixStart]), ixLength);
         } else {
 
-            MainForm::cellDataRay[tbi].pTimeCurve->setSamples(0, 0, 0);
+            MainForm::cellDataRay[tbi].pTimeCurve->setSamples((float*) 0, (float*) 0, 0);
         }
     }
     MainForm::mainFormWidget.chan24Plot->replot();
--- linssid-3.6/linssid-app/linssid-app.pro.orig	2018-06-23 02:01:15.000000000 +0100
+++ linssid-3.6/linssid-app/linssid-app.pro	2026-02-04 21:42:57.176277902 +0000
@@ -22,7 +22,7 @@
 INCLUDEPATH += /usr/include/qt5
 INCLUDEPATH += /usr/include/qwt
 LIBS += -lboost_regex
-LIBS += /usr/lib/libqwt-qt5.so.6
+LIBS += /usr/lib64/libqwt-qt5.so.6
 QMAKE_CXXFLAGS += -std=c++11
 #
 TARGET = linssid
