--- LiVES-0.9.9.4/lives-plugins/weed-plugins/frei0r.c.orig	2008-07-22 22:55:52.000000000 +0100
+++ LiVES-0.9.9.4/lives-plugins/weed-plugins/frei0r.c	2008-12-09 22:47:16.000000000 +0000
@@ -290,10 +290,6 @@
 
       do {
 
-	snprintf(dir1,512,"%s/%s",vdir1,vendor_name);
-	snprintf(dir2,512,"%s/%s",vdir2,vendor_name);
-	snprintf(dir3,512,"%s/%s",vdir3,vendor_name);
-	
 	vdirent=readdir(curvdir);
 
 	if (vdirent==NULL) {
@@ -303,8 +299,15 @@
 	  break;
 	}
 
+	if (strcmp (vdirent->d_name, "..") == 0)
+	  continue;
+
 	snprintf(vendor_name,256,"%s",vdirent->d_name);
 
+	snprintf(dir1,512,"%s/%s",vdir1,vendor_name);
+	snprintf(dir2,512,"%s/%s",vdir2,vendor_name);
+	snprintf(dir3,512,"%s/%s",vdir3,vendor_name);
+	
 	if (vdirval==1) {
 	  curdir=opendir(dir1);
 	  if (curdir==NULL) continue;
