--- llvm-3.0/bindings/ocaml/llvm/Makefile.orig	2011-10-14 21:38:02.000000000 +0100
+++ llvm-3.0/bindings/ocaml/llvm/Makefile	2012-04-12 22:34:30.000000000 +0100
@@ -30,7 +30,7 @@
 $(OcamlDir)/META.llvm: META.llvm
 	$(Verb) $(CP) -f $< $@
 
-install-meta:: $(ObjDir)/META.llvm
+install-meta:: $(OcamlDir)/META.llvm
 	$(Echo) "Install $(BuildMode) $(DestMETA)"
 	$(Verb) $(MKDIR) $(PROJ_libocamldir)
 	$(Verb) $(DataInstall) META.llvm "$(DestMETA)"
