--- llvm-3.2/tools/llvm-config/llvm-config.cpp.orig	2013-02-23 07:56:36.625627236 +0000
+++ llvm-3.2/tools/llvm-config/llvm-config.cpp	2013-02-23 07:57:49.663619649 +0000
@@ -231,15 +231,15 @@
     switch (DevelopmentTreeLayout) {
     case MakefileStyle:
       ActiveBinDir = ActiveObjRoot + "/" + LLVM_BUILDMODE + "/bin";
-      ActiveLibDir = ActiveObjRoot + "/" + LLVM_BUILDMODE + "/lib";
+      ActiveLibDir = ActiveObjRoot + "/" + LLVM_BUILDMODE + "/lib64";
       break;
     case CMakeStyle:
       ActiveBinDir = ActiveObjRoot + "/bin";
-      ActiveLibDir = ActiveObjRoot + "/lib";
+      ActiveLibDir = ActiveObjRoot + "/lib64";
       break;
     case CMakeBuildModeStyle:
       ActiveBinDir = ActiveObjRoot + "/bin/" + LLVM_BUILDMODE;
-      ActiveLibDir = ActiveObjRoot + "/lib/" + LLVM_BUILDMODE;
+      ActiveLibDir = ActiveObjRoot + "/lib64/" + LLVM_BUILDMODE;
       break;
     }
 
@@ -250,7 +250,7 @@
     ActivePrefix = CurrentExecPrefix;
     ActiveIncludeDir = ActivePrefix + "/include";
     ActiveBinDir = ActivePrefix + "/bin";
-    ActiveLibDir = ActivePrefix + "/lib";
+    ActiveLibDir = ActivePrefix + "/lib64";
     ActiveIncludeOption = "-I" + ActiveIncludeDir;
   }
 
--- llvm-3.2/Makefile.rules.orig	2012-10-12 22:48:14.000000000 +0100
+++ llvm-3.2/Makefile.rules	2013-02-24 19:56:08.403526125 +0000
@@ -475,10 +475,10 @@
 
 ObjRootDir  := $(PROJ_OBJ_DIR)/$(BuildMode)
 ObjDir      := $(ObjRootDir)
-LibDir      := $(PROJ_OBJ_ROOT)/$(BuildMode)/lib
+LibDir      := $(PROJ_OBJ_ROOT)/$(BuildMode)/lib64
 ToolDir     := $(PROJ_OBJ_ROOT)/$(BuildMode)/bin
 ExmplDir    := $(PROJ_OBJ_ROOT)/$(BuildMode)/examples
-LLVMLibDir  := $(LLVM_OBJ_ROOT)/$(BuildMode)/lib
+LLVMLibDir  := $(LLVM_OBJ_ROOT)/$(BuildMode)/lib64
 LLVMToolDir := $(LLVM_OBJ_ROOT)/$(BuildMode)/bin
 LLVMExmplDir:= $(LLVM_OBJ_ROOT)/$(BuildMode)/examples
 
@@ -637,7 +637,7 @@
 ifneq ($(HOST_OS), $(filter $(HOST_OS), Cygwin MingW))
   ifneq ($(HOST_OS), Darwin)
     ifdef TOOLNAME
-      LD.Flags += $(RPATH) -Wl,'$$ORIGIN/../lib'
+      LD.Flags += $(RPATH) -Wl,'$$ORIGIN/../lib64'
       ifdef EXAMPLE_TOOL
         LD.Flags += $(RPATH) -Wl,$(ExmplDir) $(DynamicFlag)
       else
@@ -646,7 +646,7 @@
   endif
 else
   ifneq ($(DARWIN_MAJVERS),4)
-    LD.Flags += $(RPATH) -Wl,@executable_path/../lib
+    LD.Flags += $(RPATH) -Wl,@executable_path/../lib64
   endif
   ifeq ($(RC_BUILDIT),YES)
     TempFile := $(shell mkdir -p ${OBJROOT}/dSYMs ; mktemp ${OBJROOT}/dSYMs/llvm-lto.XXXXXX)
@@ -777,7 +777,7 @@
 TableGen.Flags= -I $(call SYSPATH, $(PROJ_SRC_DIR)) \
                 -I $(call SYSPATH, $(LLVM_SRC_ROOT)/include) \
                 -I $(call SYSPATH, $(PROJ_SRC_ROOT)/include) \
-                -I $(call SYSPATH, $(PROJ_SRC_ROOT)/lib/Target)
+                -I $(call SYSPATH, $(PROJ_SRC_ROOT)/lib64/Target)
 LLVMTableGen  = $(LLVM_TBLGEN) $(TableGen.Flags)
 
 Archive       = $(AR) $(AR.Flags)
