--- llvm-3.2/tools/llvm-config/llvm-config.cpp.orig	2013-02-23 07:56:36.625627236 +0000
+++ llvm-3.2/tools/llvm-config/llvm-config.cpp	2013-02-23 07:57:49.663619649 +0000
@@ -231,15 +231,15 @@
     switch (DevelopmentTreeLayout) {
     case MakefileStyle:
       ActiveBinDir = ActiveObjRoot + "/" + LLVM_BUILDMODE + "/bin";
-      ActiveLibDir = ActiveObjRoot + "/" + LLVM_BUILDMODE + "/lib";
+      ActiveLibDir = ActiveObjRoot + "/" + LLVM_BUILDMODE + "/lib64";
       break;
     case CMakeStyle:
       ActiveBinDir = ActiveObjRoot + "/bin";
-      ActiveLibDir = ActiveObjRoot + "/lib";
+      ActiveLibDir = ActiveObjRoot + "/lib64";
       break;
     case CMakeBuildModeStyle:
       ActiveBinDir = ActiveObjRoot + "/bin/" + LLVM_BUILDMODE;
-      ActiveLibDir = ActiveObjRoot + "/lib/" + LLVM_BUILDMODE;
+      ActiveLibDir = ActiveObjRoot + "/lib64/" + LLVM_BUILDMODE;
       break;
     }
 
@@ -250,7 +250,7 @@
     ActivePrefix = CurrentExecPrefix;
     ActiveIncludeDir = ActivePrefix + "/include";
     ActiveBinDir = ActivePrefix + "/bin";
-    ActiveLibDir = ActivePrefix + "/lib";
+    ActiveLibDir = ActivePrefix + "/lib64";
     ActiveIncludeOption = "-I" + ActiveIncludeDir;
   }
 
