--- lrmi-0.10/Makefile.orig	2005-12-04 23:33:57.000000000 +0000
+++ lrmi-0.10/Makefile	2008-05-01 20:53:47.000000000 +0100
@@ -39,8 +39,7 @@
 	rm -f $(LIBDIR)/$(LIBNAME).so
 	ln -sf $(LIBNAME).so.$(VERSION) $(LIBDIR)/$(LIBNAME).so.$(MAJOR)
 	ln -sf $(LIBNAME).so.$(MAJOR) $(LIBDIR)/$(LIBNAME).so
-	install -m 644 -s -p lrmi.h $(INCDIR)/lrmi.h
-	ldconfig
+	install -m 644 -p lrmi.h $(INCDIR)/lrmi.h
 
 .PHONY: clean
 clean:
