From 5c37171a18bddfbc716d4f3da8b008a844eea4f7 Mon Sep 17 00:00:00 2001
From: Petr Machata <pmachata@redhat.com>
Date: Tue, 5 Feb 2013 01:52:37 +0100
Subject: Add elf_can_read_next

---
 ltrace-elf.c |   10 +++++-----
 ltrace-elf.h |    3 +++
 2 files changed, 8 insertions(+), 5 deletions(-)

Index: b/ltrace-elf.c
===================================================================
--- a/ltrace-elf.c
+++ b/ltrace-elf.c
@@ -202,23 +202,23 @@ elf_get_section_named(struct ltelf *lte,
 				  &name_p, &data);
 }
 
-static int
-need_data(Elf_Data *data, GElf_Xword offset, GElf_Xword size)
+int
+elf_can_read_next(Elf_Data *data, GElf_Xword offset, GElf_Xword size)
 {
 	assert(data != NULL);
 	if (data->d_size < size || offset > data->d_size - size) {
 		debug(1, "Not enough data to read %"PRId64"-byte value"
 		      " at offset %"PRId64".", size, offset);
-		return -1;
+		return 0;
 	}
-	return 0;
+	return 1;
 }
 
 #define DEF_READER(NAME, SIZE)						\
 	int								\
 	NAME(Elf_Data *data, GElf_Xword offset, uint##SIZE##_t *retp)	\
 	{								\
-		if (need_data(data, offset, SIZE / 8) < 0)		\
+		if (!elf_can_read_next(data, offset, SIZE / 8))		\
 			return -1;					\
 									\
 		if (data->d_buf == NULL) /* NODATA section */ {		\
Index: b/ltrace-elf.h
===================================================================
--- a/ltrace-elf.h
+++ b/ltrace-elf.h
@@ -116,6 +116,9 @@ int elf_read_next_u16(Elf_Data *data, GE
 int elf_read_next_u32(Elf_Data *data, GElf_Xword *offset, uint32_t *retp);
 int elf_read_next_u64(Elf_Data *data, GElf_Xword *offset, uint64_t *retp);
 
+/* Return whether there's AMOUNT more bytes after OFFSET in DATA.  */
+int elf_can_read_next(Elf_Data *data, GElf_Xword offset, GElf_Xword amount);
+
 #if __WORDSIZE == 32
 #define PRI_ELF_ADDR		PRIx32
 #define GELF_ADDR_CAST(x)	(void *)(uint32_t)(x)
