--- luminance-hdr-2.5.1/src/TransplantExif/TransplantExifDialog.cpp.orig	2017-05-10 20:14:06.000000000 +0100
+++ luminance-hdr-2.5.1/src/TransplantExif/TransplantExifDialog.cpp	2018-12-21 08:15:52.613385424 +0000
@@ -26,6 +26,7 @@
 #include <QFileInfo>
 #include <image.hpp>
 #include <exif.hpp>
+#include <exiv2/version.hpp>
 
 #include "Common/global.h"
 #include "Common/config.h"
--- luminance-hdr-2.5.1/CMakeLists.txt.orig	2017-05-10 20:14:06.000000000 +0100
+++ luminance-hdr-2.5.1/CMakeLists.txt	2019-04-21 11:55:39.756897436 +0100
@@ -115,7 +115,9 @@
 IF(WIN32)
     SET(Boost_USE_STATIC_LIBS ON)
 ENDIF()
+SET(BUILD_SHARED_LIBS ON)
 FIND_PACKAGE(Boost REQUIRED COMPONENTS program_options thread chrono system date_time)
+SET(BUILD_SHARED_LIBS OFF)
 INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIR})
 
 FIND_PACKAGE(exiv2 REQUIRED)
@@ -276,7 +278,7 @@
     TARGET_LINK_LIBRARIES(luminance-hdr -Xlinker --start-group ${LUMINANCE_MODULES_GUI} -Xlinker --end-group)
     TARGET_LINK_LIBRARIES(luminance-hdr-cli -Xlinker --start-group ${LUMINANCE_MODULES_CLI} -Xlinker --end-group)
     TARGET_LINK_LIBRARIES(luminance-hdr ${LIBS})
-    TARGET_LINK_LIBRARIES(luminance-hdr-cli ${LIBS})
+    TARGET_LINK_LIBRARIES(luminance-hdr-cli ${LIBS} Boost::program_options)
 
     # make install directivies
     INSTALL(TARGETS luminance-hdr RUNTIME DESTINATION bin) # main GUI exec
