--- luminance-hdr-634b489/CMakeLists.txt.orig	2025-08-16 16:34:47.000000000 +0100
+++ luminance-hdr-634b489/CMakeLists.txt	2025-08-16 17:18:48.242353794 +0100
@@ -4,6 +4,7 @@
 CMAKE_POLICY(SET CMP0054 NEW)
 CMAKE_POLICY(SET CMP0053 NEW)
 CMAKE_POLICY(SET CMP0070 NEW)
+CMAKE_POLICY(SET CMP0167 OLD)
 PROJECT("Luminance HDR")
 
 string(TOLOWER ${CMAKE_BUILD_TYPE} LOWERCASE_CMAKE_BUILD_TYPE)
@@ -180,7 +181,7 @@
 IF(WIN32)
     SET(Boost_USE_STATIC_LIBS ON)
 ENDIF()
-FIND_PACKAGE(Boost REQUIRED COMPONENTS program_options thread chrono system date_time atomic)
+FIND_PACKAGE(Boost REQUIRED COMPONENTS program_options thread chrono date_time atomic)
 INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIR})
 
 FIND_PACKAGE(exiv2 REQUIRED 0.27.0)
--- luminance-hdr-634b489/src/Libpfs/manip/resize.hxx.orig	2025-08-16 16:34:47.000000000 +0100
+++ luminance-hdr-634b489/src/Libpfs/manip/resize.hxx	2025-08-16 16:35:18.242938155 +0100
@@ -29,6 +29,7 @@
 #endif
 
 #include <boost/math/constants/constants.hpp>
+#include <boost/numeric/conversion/bounds.hpp>
 #include "copy.h"
 #include "resize.h"
 #include "../../sleef.c"
