--- luminance-hdr-2.4.0/src/HelpBrowser/CMakeLists.txt.orig	2014-01-27 20:20:46.000000000 +0000
+++ luminance-hdr-2.4.0/src/HelpBrowser/CMakeLists.txt	2015-02-15 14:32:46.994542413 +0000
@@ -29,7 +29,7 @@
 
 
 ADD_LIBRARY(helpbrowser ${FILES_H} ${FILES_CPP} ${FILES_MOC} ${FILES_UI_H} ${FILES_HXX})
-qt5_use_modules(helpbrowser Core Concurrent Gui Widgets Xml WebKit WebKitWidgets)
+qt5_use_modules(helpbrowser Core Concurrent Gui Widgets Xml WebKit WebKitWidgets PrintSupport)
 
 SET(FILES_TO_TRANSLATE ${FILES_TO_TRANSLATE} ${FILES_CPP} ${FILES_H} ${FILES_UI} ${FILES_HXX} PARENT_SCOPE)
 SET(LUMINANCE_MODULES_GUI ${LUMINANCE_MODULES_GUI} helpbrowser PARENT_SCOPE)
--- luminance-hdr-2.4.0/CMakeLists.txt.orig	2014-01-27 20:20:46.000000000 +0000
+++ luminance-hdr-2.4.0/CMakeLists.txt	2015-02-15 14:33:23.511539697 +0000
@@ -29,6 +29,7 @@
 find_package(Qt5Sql)
 find_package(Qt5Network)
 find_package(Qt5LinguistTools)
+find_package(Qt5PrintSupport)
 IF(WIN32)
 	find_package(Qt5WinExtras)
 ENDIF()
