--- luz-20110731/input-manager/Makefile.orig	2011-07-31 06:19:27.000000000 +0100
+++ luz-20110731/input-manager/Makefile	2011-08-13 10:31:31.000000000 +0100
@@ -1,4 +1,4 @@
 all: input-manager
 
 input-manager: *.cc
-	gcc -o input-manager -pthread -lcwiimote -lportmidi -lporttime -I/usr/include/libcwiimote `pkg-config gtkmm-2.4 unique-1.0 liblo sdl bluez --cflags --libs` *.cc
+	gcc *.cc -o input-manager -pthread -lcwiimote -lportmidi -I/usr/include/libcwiimote `pkg-config gtkmm-2.4 unique-1.0 liblo sdl bluez --cflags --libs`
