--- LVM2-2.02.81/configure.orig	2011-01-10 14:28:18.000000000 +0000
+++ LVM2-2.02.81/configure	2011-01-18 20:55:33.000000000 +0000
@@ -8773,7 +8773,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_udev_udev_queue_get_udev_is_active" >&5
 $as_echo "$ac_cv_lib_udev_udev_queue_get_udev_is_active" >&6; }
 if test "x$ac_cv_lib_udev_udev_queue_get_udev_is_active" = x""yes; then :
-  UDEV_PC="libudev"; UDEV_LIBS="-ludev"
+  UDEV_PC="libudev"; UDEV_LIBS="-ludev -lrt"
 else
   as_fn_error $? "bailing out... libudev library is required" "$LINENO" 5
 fi
