--- mail-notification-5.4/jbsrc/jb.c.orig	2016-07-26 22:24:25.000000000 +0100
+++ mail-notification-5.4/jbsrc/jb.c	2016-07-26 22:26:39.340715977 +0100
@@ -422,7 +422,7 @@
        */
       jb_compile_options_add_cflags(object->compile_options, "-std=c99");
       jb_compile_options_add_cppflags(object->compile_options, "-D_BSD_SOURCE -D_POSIX_C_SOURCE=199309L");
-      jb_compile_options_add_libs(object->compile_options, "-lbsd-compat");
+//      jb_compile_options_add_libs(object->compile_options, "-lbsd-compat");
     }
 
   jb_compile_options_add_string_defines(object->compile_options,
