--- jbsrc/jb.c.orig	2011-02-08 23:11:32.767114666 +0100
+++ jbsrc/jb.c	2011-02-08 23:18:48.870910628 +0100
@@ -443,6 +443,9 @@
    */
   jb_compile_options_add_ldflags(object->compile_options, "-Wl,--export-dynamic");
 
+  jb_compile_options_add_cppflags(object->compile_options, "-O0");
+  jb_compile_options_add_ldflags(object->compile_options, "-O0");
+
   jb_compile_options_add_libs(object->compile_options, "-lm");
 
   jb_compile_options_add_package(object->compile_options, "gettext");
