--- mame-0146s/makefile.orig	2012-05-21 21:48:30.000000000 +0100
+++ mame-0146s/makefile	2012-05-26 19:51:45.000000000 +0100
@@ -207,16 +207,16 @@
 # BIGENDIAN = 1
 
 # uncomment next line to build expat as part of MAME build
-BUILD_EXPAT = 1
+#BUILD_EXPAT = 1
 
 # uncomment next line to build zlib as part of MAME build
-BUILD_ZLIB = 1
+#BUILD_ZLIB = 1
 
 # uncomment next line to build libflac as part of MAME build
-BUILD_FLAC = 1
+#BUILD_FLAC = 1
 
 # uncomment next line to build jpeglib as part of MAME build
-BUILD_JPEGLIB = 1
+#BUILD_JPEGLIB = 1
 
 # uncomment next line to include the symbols
 # SYMBOLS = 1
