--- mapnik-v3.1.0/src/parse_path.cpp.orig	2021-01-08 11:23:24.000000000 +0000
+++ mapnik-v3.1.0/src/parse_path.cpp	2023-03-08 21:46:50.994885538 +0000
@@ -20,6 +20,8 @@
  *
  *****************************************************************************/
 
+#define BOOST_PHOENIX_STL_TUPLE_H_
+
 #include <mapnik/parse_path.hpp>
 #include <mapnik/path_expression_grammar.hpp>
 #include <mapnik/path_expression_grammar_impl.hpp>
--- mapnik-v3.1.0/src/dasharray_parser.cpp.orig	2021-01-08 11:23:24.000000000 +0000
+++ mapnik-v3.1.0/src/dasharray_parser.cpp	2023-03-08 21:59:49.247553668 +0000
@@ -20,6 +20,8 @@
  *
  *****************************************************************************/
 
+#define BOOST_PHOENIX_STL_TUPLE_H_
+
 // mapnik
 #include <mapnik/util/dasharray_parser.hpp>
 
--- mapnik-v3.1.0/src/json/mapnik_json_feature_collection_grammar.cpp.orig	2021-01-08 11:23:24.000000000 +0000
+++ mapnik-v3.1.0/src/json/mapnik_json_feature_collection_grammar.cpp	2023-03-08 22:16:06.356507047 +0000
@@ -20,6 +20,8 @@
  *
  *****************************************************************************/
 
+#define BOOST_PHOENIX_STL_TUPLE_H_
+
 #include <mapnik/feature.hpp>
 #include <mapnik/json/feature_collection_grammar_impl.hpp>
 #include <string>
--- mapnik-v3.1.0/src/json/mapnik_json_feature_grammar.cpp.orig	2021-01-08 11:23:24.000000000 +0000
+++ mapnik-v3.1.0/src/json/mapnik_json_feature_grammar.cpp	2023-03-08 22:16:23.238506242 +0000
@@ -20,6 +20,8 @@
  *
  *****************************************************************************/
 
+#define BOOST_PHOENIX_STL_TUPLE_H_
+
 #include <mapnik/feature.hpp>
 #include <mapnik/json/feature_grammar_impl.hpp>
 #include <string>
--- mapnik-v3.1.0/src/json/mapnik_json_positions_grammar.cpp.orig	2021-01-08 11:23:24.000000000 +0000
+++ mapnik-v3.1.0/src/json/mapnik_json_positions_grammar.cpp	2023-03-08 22:05:16.425538057 +0000
@@ -20,6 +20,8 @@
  *
  *****************************************************************************/
 
+#define BOOST_PHOENIX_STL_TUPLE_H_
+
 #include <mapnik/json/positions_grammar_impl.hpp>
 #include <string>
 
--- mapnik-v3.1.0/src/json/generic_json.cpp.orig	2021-01-08 11:23:24.000000000 +0000
+++ mapnik-v3.1.0/src/json/generic_json.cpp	2023-03-08 22:04:52.146539216 +0000
@@ -20,6 +20,8 @@
  *
  *****************************************************************************/
 
+#define BOOST_PHOENIX_STL_TUPLE_H_
+
 #include <mapnik/json/generic_json.hpp>
 
 namespace mapnik { namespace json {
--- mapnik-v3.1.0/src/json/mapnik_json_geometry_grammar.cpp.orig	2021-01-08 11:23:24.000000000 +0000
+++ mapnik-v3.1.0/src/json/mapnik_json_geometry_grammar.cpp	2023-03-08 21:59:07.639555653 +0000
@@ -20,6 +20,8 @@
  *
  *****************************************************************************/
 
+#define BOOST_PHOENIX_STL_TUPLE_H_
+
 #include <mapnik/json/geometry_grammar_impl.hpp>
 #include <string>
 
--- mapnik-v3.1.0/src/json/mapnik_json_geometry_parser.cpp.orig	2021-01-08 11:23:24.000000000 +0000
+++ mapnik-v3.1.0/src/json/mapnik_json_geometry_parser.cpp	2023-03-08 21:57:32.631560186 +0000
@@ -20,6 +20,7 @@
  *
  *****************************************************************************/
 
+#define BOOST_PHOENIX_STL_TUPLE_H_
 
 #include <mapnik/json/geometry_parser.hpp>
 #include <mapnik/json/geometry_grammar.hpp>
--- mapnik-v3.1.0/src/text/placements/simple.cpp.orig	2021-01-08 11:23:24.000000000 +0000
+++ mapnik-v3.1.0/src/text/placements/simple.cpp	2023-03-08 22:25:50.267479187 +0000
@@ -20,6 +20,8 @@
  *
  *****************************************************************************/
 
+#define BOOST_PHOENIX_STL_TUPLE_H_
+
 // mapnik
 #include <mapnik/debug.hpp>
 #include <mapnik/text/placements/simple.hpp>
--- mapnik-v3.1.0/src/wkt/mapnik_wkt_grammar.cpp.orig	2021-01-08 11:23:24.000000000 +0000
+++ mapnik-v3.1.0/src/wkt/mapnik_wkt_grammar.cpp	2023-03-08 22:05:44.097536737 +0000
@@ -20,6 +20,8 @@
  *
  *****************************************************************************/
 
+#define BOOST_PHOENIX_STL_TUPLE_H_
+
 #include <mapnik/wkt/wkt_grammar_impl.hpp>
 #include <string>
 
--- mapnik-v3.1.0/SConstruct.orig	2021-01-08 11:23:24.000000000 +0000
+++ mapnik-v3.1.0/SConstruct	2022-12-22 21:28:41.625934078 +0000
@@ -1912,7 +1912,7 @@
         # faster compile
         # http://www.boost.org/doc/libs/1_47_0/libs/spirit/doc/html/spirit/what_s_new/spirit_2_5.html#spirit.what_s_new.spirit_2_5.breaking_changes
         env.Append(CPPDEFINES = '-DBOOST_SPIRIT_NO_PREDEFINED_TERMINALS=1')
-        env.Append(CPPDEFINES = '-DBOOST_PHOENIX_NO_PREDEFINED_TERMINALS=1')
+        #env.Append(CPPDEFINES = '-DBOOST_PHOENIX_NO_PREDEFINED_TERMINALS=1')
         # c++11 support / https://github.com/mapnik/mapnik/issues/1683
         #  - upgrade to PHOENIX_V3 since that is needed for c++11 compile
         env.Append(CPPDEFINES = '-DBOOST_SPIRIT_USE_PHOENIX_V3=1')
--- mapnik-v3.1.0/plugins/input/topojson/topojson_featureset.cpp.orig	2021-01-08 11:23:24.000000000 +0000
+++ mapnik-v3.1.0/plugins/input/topojson/topojson_featureset.cpp	2023-03-08 22:21:00.855492996 +0000
@@ -20,6 +20,8 @@
  *
  *****************************************************************************/
 
+#define BOOST_PHOENIX_STL_TUPLE_H_
+
 // mapnik
 #include <mapnik/json/topology.hpp>
 #include <mapnik/json/topojson_utils.hpp>
--- mapnik-v3.1.0/plugins/input/topojson/topojson_datasource.cpp.orig	2021-01-08 11:23:24.000000000 +0000
+++ mapnik-v3.1.0/plugins/input/topojson/topojson_datasource.cpp	2023-03-08 22:21:25.136491838 +0000
@@ -20,6 +20,8 @@
  *
  *****************************************************************************/
 
+#define BOOST_PHOENIX_STL_TUPLE_H_
+
 #include "topojson_datasource.hpp"
 #include "topojson_featureset.hpp"
 
--- mapnik-v3.1.0/plugins/input/csv/csv_utils.cpp.orig	2021-01-08 11:23:24.000000000 +0000
+++ mapnik-v3.1.0/plugins/input/csv/csv_utils.cpp	2023-03-08 21:55:48.174565170 +0000
@@ -20,6 +20,8 @@
  *
  *****************************************************************************/
 
+#define BOOST_PHOENIX_STL_TUPLE_H_
+
 // mapnik
 #include <mapnik/debug.hpp>
 #include <mapnik/geometry.hpp>
--- mapnik-v3.1.0/plugins/input/geojson/geojson_memory_index_featureset.cpp.orig	2021-01-08 11:23:24.000000000 +0000
+++ mapnik-v3.1.0/plugins/input/geojson/geojson_memory_index_featureset.cpp	2023-03-08 22:17:24.067503340 +0000
@@ -20,6 +20,8 @@
  *
  *****************************************************************************/
 
+#define BOOST_PHOENIX_STL_TUPLE_H_
+
 // mapnik
 #include <mapnik/feature.hpp>
 #include <mapnik/feature_factory.hpp>
--- mapnik-v3.1.0/plugins/input/geojson/geojson_datasource.cpp.orig	2021-01-08 11:23:24.000000000 +0000
+++ mapnik-v3.1.0/plugins/input/geojson/geojson_datasource.cpp	2023-03-08 22:17:04.832504257 +0000
@@ -20,6 +20,8 @@
  *
  *****************************************************************************/
 
+#define BOOST_PHOENIX_STL_TUPLE_H_
+
 #include "geojson_datasource.hpp"
 #include "geojson_featureset.hpp"
 #include "geojson_index_featureset.hpp"
