--- mapnik-v3.1.0/plugins/input/csv/csv_datasource.hpp.orig	2021-01-08 11:23:24.000000000 +0000
+++ mapnik-v3.1.0/plugins/input/csv/csv_datasource.hpp	2021-08-16 22:35:21.508129154 +0100
@@ -38,6 +38,7 @@
 #include <mapnik/warning_ignore.hpp>
 #include <boost/optional.hpp>
 #include <boost/version.hpp>
+#include <boost/geometry/strategies/disjoint.hpp>
 #include <boost/geometry/index/rtree.hpp>
 #pragma GCC diagnostic pop
 
--- mapnik-v3.1.0/plugins/input/geojson/geojson_datasource.hpp.orig	2021-01-08 11:23:24.000000000 +0000
+++ mapnik-v3.1.0/plugins/input/geojson/geojson_datasource.hpp	2021-08-16 22:25:16.937158000 +0100
@@ -37,6 +37,7 @@
 #include <mapnik/warning_ignore.hpp>
 #include <boost/optional.hpp>
 #include <boost/version.hpp>
+#include <boost/geometry/strategies/disjoint.hpp>
 #include <boost/geometry/index/rtree.hpp>
 #pragma GCC diagnostic pop
 
