--- MaterialX-1.39.3/_BuilD_/I/opt/materialx/include/MaterialXGenShader/TypeDesc.h.orig	2025-08-17 09:16:38.000000000 +0100
+++ MaterialX-1.39.3/_BuilD_/I/opt/materialx/include/MaterialXGenShader/TypeDesc.h	2025-08-17 09:20:47.000000000 +0100
@@ -14,6 +14,7 @@
 #include <MaterialXCore/Document.h>
 
 #include <string_view>
+#include <cstdint>
 
 MATERIALX_NAMESPACE_BEGIN
 
--- MaterialX-1.39.3/_BuilD_/I/opt/materialx/include/MaterialXRender/Types.h.orig	2025-08-17 09:16:38.000000000 +0100
+++ MaterialX-1.39.3/_BuilD_/I/opt/materialx/include/MaterialXRender/Types.h	2025-08-17 09:21:47.000000000 +0100
@@ -13,6 +13,8 @@
 
 #include <MaterialXCore/Types.h>
 
+#include <cstdint>
+
 MATERIALX_NAMESPACE_BEGIN
 
 /// @class Quaternion
--- MaterialX-1.39.3/_BuilD_/I/opt/materialx/include/MaterialXRender/Mesh.h.orig	2025-08-17 09:16:38.000000000 +0100
+++ MaterialX-1.39.3/_BuilD_/I/opt/materialx/include/MaterialXRender/Mesh.h	2025-08-17 09:22:11.000000000 +0100
@@ -12,6 +12,9 @@
 #include <MaterialXCore/Types.h>
 #include <MaterialXRender/Export.h>
 
+#include <cstdint>
+
+
 MATERIALX_NAMESPACE_BEGIN
 
 /// Geometry index buffer
--- MaterialX-1.39.3/python/CMakeLists.txt.orig	2025-08-17 09:16:37.000000000 +0100
+++ MaterialX-1.39.3/python/CMakeLists.txt	2025-08-17 09:19:39.991613206 +0100
@@ -19,7 +19,7 @@
 endif()
 
 if(MATERIALX_INSTALL_PYTHON AND PYTHON_EXECUTABLE AND NOT SKBUILD)
-  set(SETUP_PY "${CMAKE_INSTALL_PREFIX}/python/setup.py")
+  set(SETUP_PY "${CMAKE_BINARY_DIR}/python/setup.py")
   configure_file("${CMAKE_CURRENT_SOURCE_DIR}/setup.py.in" "${SETUP_PY}")
   install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ${SETUP_PY} install clean --all)")
 endif()
--- MaterialX-1.39.3/source/MaterialXGenShader/TypeDesc.h.orig	2025-08-17 09:16:38.000000000 +0100
+++ MaterialX-1.39.3/source/MaterialXGenShader/TypeDesc.h	2025-08-17 09:20:47.777331468 +0100
@@ -14,6 +14,7 @@
 #include <MaterialXCore/Document.h>
 
 #include <string_view>
+#include <cstdint>
 
 MATERIALX_NAMESPACE_BEGIN
 
--- MaterialX-1.39.3/source/MaterialXRender/Types.h.orig	2025-08-17 09:16:38.000000000 +0100
+++ MaterialX-1.39.3/source/MaterialXRender/Types.h	2025-08-17 09:21:47.820030349 +0100
@@ -13,6 +13,8 @@
 
 #include <MaterialXCore/Types.h>
 
+#include <cstdint>
+
 MATERIALX_NAMESPACE_BEGIN
 
 /// @class Quaternion
--- MaterialX-1.39.3/source/MaterialXRender/Mesh.h.orig	2025-08-17 09:16:38.000000000 +0100
+++ MaterialX-1.39.3/source/MaterialXRender/Mesh.h	2025-08-17 09:22:11.037639235 +0100
@@ -12,6 +12,9 @@
 #include <MaterialXCore/Types.h>
 #include <MaterialXRender/Export.h>
 
+#include <cstdint>
+
+
 MATERIALX_NAMESPACE_BEGIN
 
 /// Geometry index buffer
