--- mc-20090606/edit/editcmd.c.orig	2009-06-06 23:01:26.000000000 +0100
+++ mc-20090606/edit/editcmd.c	2009-06-06 23:32:24.000000000 +0100
@@ -1326,6 +1326,7 @@
 		    break;
 		case B_CANCEL:
 		    replace_yes = 0;
+		    edit->replace_mode=-1;
 		    break;
 		}
 	    }
@@ -1375,7 +1376,7 @@
 			      D_NORMAL, 1, _("&OK"));
 	    edit->replace_mode = -1;
 	}
-    } while (edit->replace_mode >0);
+    } while (edit->replace_mode >= 0);
 
     edit->force = REDRAW_COMPLETELY;
     edit_scroll_screen_over_cursor (edit);
