--- mc-4.8.11/src/filemanager/filegui.c.orig	2013-11-29 18:27:07.000000000 +0000
+++ mc-4.8.11/src/filemanager/filegui.c	2014-01-05 19:58:41.261149495 +0000
@@ -1312,6 +1312,10 @@
         else
             ctx->search_handle->search_type = MC_SEARCH_T_REGEX;
 
+        tmp = dest_dir; 
+        dest_dir = tilde_expand (tmp); 
+        g_free (tmp);
+
         vpath = vfs_path_from_str (dest_dir);
 
         ctx->dest_mask = strrchr (dest_dir, PATH_SEP);
