--- mc-4.7.0-pre2/src/util.c.orig	2009-08-24 22:12:17.000000000 +0100
+++ mc-4.7.0-pre2/src/util.c	2009-09-04 10:49:39.000000000 +0100
@@ -906,7 +906,7 @@
     /* Support for LZMA (only utils format with magic in header).
      * This is the default format of LZMA utils 4.32.1 and later. */
 
-    if (mc_read(fd, (char *) magic+4, 1) == 1)
+    if (mc_read(fd, (char *) magic+4, 2) == 2)
     {
         /* LZMA utils format */
         if
@@ -919,6 +919,7 @@
         )
             return COMPRESSION_LZMA;
     }
+    mc_lseek (fd, 4, SEEK_SET); /* seek back to let following test work */
 
     /* XZ compression magic */
     if (magic[0] == 0xFD
--- mc-4.7.0-pre2/edit/edit.c.orig	2009-08-23 21:43:25.000000000 +0100
+++ mc-4.7.0-pre2/edit/edit.c	2009-09-08 10:59:39.000000000 +0100
@@ -2724,10 +2724,12 @@
 	break;
     case CK_Word_Left:
     case CK_Word_Left_Highlight:
+	edit->over_col = 0;
 	edit_left_word_move_cmd (edit);
 	break;
     case CK_Word_Right:
     case CK_Word_Right_Highlight:
+        edit->over_col = 0;
 	edit_right_word_move_cmd (edit);
 	break;
     case CK_Up_Alt_Highlight:
--- mc-4.7.0-pre2/misc/mc.menu.orig	2009-08-12 20:13:20.000000000 +0100
+++ mc-4.7.0-pre2/misc/mc.menu	2009-09-08 10:53:57.000000000 +0100
@@ -229,7 +229,7 @@
 		PRG="bunzip2 -c %f"
 		;;
 	    *.tar.gz|*.tar.z|*.tgz|*.tpz|*.tar.Z)
-		PRG="gzup -dc %f"
+		PRG="gzip -dc %f"
 		;;
 	    *.tar.lzma)
 		PRG="lzma -dc %f"
