--- mc-4.7.0/edit/editwidget.c.orig	2009-12-25 13:53:32.000000000 +0000
+++ mc-4.7.0/edit/editwidget.c	2009-12-30 22:14:40.000000000 +0000
@@ -313,6 +313,12 @@
 	    int cmd, ch;
 	    cb_ret_t ret = MSG_NOT_HANDLED;
 
+	    if (parm == KEY_BTAB) {
+		edit_execute_key_command (e, cmd, '\t');
+		edit_update_screen (e);
+		return MSG_HANDLED;
+            }
+
 	    /* The user may override the access-keys for the menu bar. */
 	    if (edit_translate_key (e, parm, &cmd, &ch)) {
 		edit_execute_key_command (e, cmd, ch);
