--- mc-4.6.3/edit/editcmd.c.orig	2008-12-01 01:02:54.000000000 +0000
+++ mc-4.6.3/edit/editcmd.c	2009-05-31 07:08:44.000000000 +0100
@@ -1643,11 +1643,11 @@
 	mc_wchar_t *buf;
 	mc_wchar_t mbuf[MAX_REPL_LEN * 2 + 3];
 
-	replace_scanf = (!replace_regexp);	/* can't have both */
+	//replace_scanf = (!replace_regexp);	/* can't have both */
 
 	buf = mbuf;
 
-	if (replace_scanf) {
+	if (!replace_regexp) {
 	    mc_wchar_t e[MAX_REPL_LEN];
 	    if (n >= NUM_REPL_ARGS) {
 #ifdef UTF8
