--- mc-4.8.9/src/viewer/display.c.orig	2013-07-09 13:41:32.000000000 +0100
+++ mc-4.8.9/src/viewer/display.c	2013-07-14 06:39:51.042896938 +0100
@@ -166,8 +166,24 @@
                         "");
         }
     }
+    if (width > 62)
+    {
+        char buffer[BUF_TINY];
+        widget_move (view, top, width - 54);
+        if (!view->hex_mode)
+        {
+            off_t line, col;
+            mcview_offset_to_coord (view, &line, &col, view->dpy_start);
+            size_trunc_len (buffer, 5, mcview_get_filesize (view), 0, panels_options.kilobyte_si);
+            tty_printf ("Line %" PRIuMAX " Col %" PRIuMAX , (uintmax_t) line+1,
+                        (uintmax_t) (view->text_wrap_mode ? col : view->dpy_text_column)
+                );
+        }
+    }
     widget_move (view, top, left);
-    if (width > 40)
+    if (width > 62)
+        tty_print_string (str_fit_to_term (file_label, width - 56, J_LEFT_FIT));
+    else if (width > 40)
         tty_print_string (str_fit_to_term (file_label, width - 34, J_LEFT_FIT));
     else
         tty_print_string (str_fit_to_term (file_label, width - 5, J_LEFT_FIT));
--- mc-4.8.9/src/filemanager/panel.c.orig	2013-07-09 13:41:32.000000000 +0100
+++ mc-4.8.9/src/filemanager/panel.c	2013-07-14 06:39:51.043896938 +0100
@@ -3535,7 +3535,7 @@
     format_e *format;
     gchar *title;
 
-    for (i = 0, format = panel->format; format != NULL; format = format->next)
+    for (i = 1, format = panel->format; format != NULL; format = format->next)
     {
         i += format->field_len;
         if (x < i + 1)
--- mc-4.8.9/src/editor/editwidget.c.orig	2013-07-09 13:41:32.000000000 +0100
+++ mc-4.8.9/src/editor/editwidget.c	2013-07-14 06:39:51.043896938 +0100
@@ -1113,6 +1113,18 @@
             int cmd, ch;
             cb_ret_t ret = MSG_NOT_HANDLED;
 
+            if (parm == KEY_BTAB) {
+                edit_execute_key_command (e, cmd, '\t');
+                edit_update_screen (e);
+                return MSG_HANDLED;
+             }
+
+            if ((parm & 0xff) == '\n') {
+                edit_execute_key_command (e, cmd, '\n');
+                edit_update_screen (e);
+                return MSG_HANDLED;
+            }
+
             /* The user may override the access-keys for the menu bar. */
             if (macro_index == -1 && edit_execute_macro (e, parm))
             {
--- mc-4.8.9/misc/syntax/php.syntax.orig	2012-04-20 08:21:37.000000000 +0100
+++ mc-4.8.9/misc/syntax/php.syntax	2013-07-14 06:39:51.044896938 +0100
@@ -3151,4 +3151,5 @@
 
 context ' ' brightgreen
     spellcheck
+    keyword \\\\ brightcyan
     keyword \\' brightcyan
