--- mc-4.7.0.7/src/editor/editcmd.c.orig	2010-07-05 20:12:51.000000000 +0100
+++ mc-4.7.0.7/src/editor/editcmd.c	2010-07-15 16:29:33.000000000 +0100
@@ -1847,7 +1847,7 @@
                 edit->search_start--;
             else
             {
-                edit->search_start += repl_str->len;
+                edit->search_start += repl_str ? repl_str->len : 1;
 
                 if (edit->search_start >= edit->last_byte)
                     break;
