--- mc-4.7.0.1/vfs/fish.c.orig	2010-01-14 10:50:30.000000000 +0000
+++ mc-4.7.0.1/vfs/fish.c	2010-01-14 11:19:41.000000000 +0000
@@ -481,7 +481,7 @@ fish_dir_load(struct vfs_class *me, stru
 				  "if test \"$m\" = \"0\" ; then \n"
 					  "s=$d; m=$y; d=$n y=$n2; n=$n3\n"
 				  "else\n"
-					  "n=$n\" \"$n2\" \"$n3\n"
+					  "if [ \"$n2\" != \"\" ] ; then n=$n\" \"$n2\" \"$n3 ; fi\n"
 				  "fi\n"
 				  "if [ $ADD  = 0 ]; then\n"
 					  "echo \"P$p $u.$g\nS$s\nd$m $d $y\n:$n\n\"\n"
@@ -498,7 +498,7 @@ fish_dir_load(struct vfs_class *me, stru
 				  "if test \"$a\" = \"0\" ; then \n"
 					  "a=$m; i=$d; m=$y; d=$n y=$n2; n=$n3\n"
 				  "else\n"
-					  "n=$n\" \"$n2\" \"$n3\n"
+					  "if [ \"$n2\" != \"\" ] ; then n=$n\" \"$n2\" \"$n3 ; fi\n"
 				  "fi\n"
 				  "if [ $ADD = 0 ]; then\n"
 					  "echo \"P$p $u.$g\nE$a$i\nd$m $d $y\n:$n\n\"\n"
