Index: med-fichier/tests/unittests/c/Unittest_MEDlibrary_1.c
===================================================================
--- med-fichier.orig/tests/unittests/c/Unittest_MEDlibrary_1.c
+++ med-fichier/tests/unittests/c/Unittest_MEDlibrary_1.c
@@ -29,7 +29,7 @@ int main (int argc, char **argv)
 {
   med_int major,minor,release;
   med_int majorFromStr, minorFromStr, releaseFromStr;
-  char version[11];
+  char version[12];
 
   /* Get library version numbers */
   if (MEDlibraryNumVersion(&major,&minor,&release) < 0) {
