Description: fix HDF5 linkage
Last-Update: 2014-04-24

--- med-fichier-3.0.6.orig/tests/parallel/Makefile.am
+++ med-fichier-3.0.6/tests/parallel/Makefile.am
@@ -29,7 +29,7 @@ generateFilterArray.c \
 getBlocksOfEntitiesPartition.c
 
 
-LDADD = $(top_builddir)/src/libmedC.la
+LDADD = $(top_builddir)/src/libmedC.la @HDF5_LDFLAGS@ @HDF5_LIBS@
 
 nodist_EXTRA_filterBlockOfentities_SOURCES = dummy.cxx
 
