--- med-fichier-4.1.0+repack/python/Makefile.in.orig	2025-11-11 16:18:38.548033797 +0000
+++ med-fichier-4.1.0+repack/python/Makefile.in	2025-11-11 22:12:50.278439339 +0000
@@ -1323,11 +1323,7 @@
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(pkgpyexecdir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(pkgpyexecdir)" || exit $$?; \
-	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  $(am__py_compile) --destdir "$(DESTDIR)" \
-	                    --basedir "$(pkgpyexecdir)" $$dlist; \
-	else :; fi
+	done || exit $$?;
 
 uninstall-nodist_pkgpyexecPYTHON:
 	@$(NORMAL_UNINSTALL)
@@ -1364,11 +1360,7 @@
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(pkgpyexecdir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(pkgpyexecdir)" || exit $$?; \
-	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  $(am__py_compile) --destdir "$(DESTDIR)" \
-	                    --basedir "$(pkgpyexecdir)" $$dlist; \
-	else :; fi
+	done || exit $$?;
 
 uninstall-pkgpyexecPYTHON:
 	@$(NORMAL_UNINSTALL)
