--- Mesa-7.0.2/src/glw/glw.pc.in.orig	2007-12-02 22:23:11.000000000 +0000
+++ Mesa-7.0.2/src/glw/glw.pc.in	2007-12-02 22:23:26.000000000 +0000
@@ -0,0 +1,11 @@
+prefix=@INSTALL_DIR@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/@LIB_DIR@
+includedir=${prefix}/include
+
+Name: glw
+Description: Mesa OpenGL widget library
+Requires: gl
+Version: @VERSION@
+Libs: -L${libdir} -lGLU
+Cflags: -I${includedir}
