--- Meshlab-2022.02/src/common/globals.cpp.orig	2022-12-11 09:09:41.000000000 +0000
+++ Meshlab-2022.02/src/common/globals.cpp	2022-12-11 17:02:55.942332091 +0000
@@ -90,7 +90,7 @@
 #ifdef Q_OS_LINUX
 	else if (pluginsDir.dirName() == "bin") {
 		pluginsDir.cdUp();
-		pluginsDir.cd("lib");
+		pluginsDir.cd("lib64");
 		pluginsDir.cd("meshlab");
 		if (pluginsDir.exists("plugins")) {
 			pluginsDir.cd("plugins");
