--- meshroom-2020.1.1/meshroom/ui/app.py.orig	2020-10-14 09:23:10.000000000 +0100
+++ meshroom-2020.1.1/meshroom/ui/app.py	2020-11-20 22:34:42.359485610 +0000
@@ -109,7 +109,7 @@
         self.engine.setWatching(os.environ.get("MESHROOM_INSTANT_CODING", False))
         # whether to output qml warnings to stderr (disable by default)
         self.engine.setOutputWarningsToStandardError(MessageHandler.outputQmlWarnings)
-        qInstallMessageHandler(MessageHandler.handler)
+        # qInstallMessageHandler(MessageHandler.handler)
 
         self.engine.addImportPath(qmlDir)
         components.registerTypes()
