--- Meshlab-2022.02/src/common/globals.cpp.orig	2022-12-11 09:09:41.000000000 +0000
+++ Meshlab-2022.02/src/common/globals.cpp	2022-12-11 17:02:55.942332091 +0000
@@ -90,7 +90,7 @@
 #ifdef Q_OS_LINUX
 	else if (pluginsDir.dirName() == "bin") {
 		pluginsDir.cdUp();
-		pluginsDir.cd("lib");
+		pluginsDir.cd("lib64");
 		pluginsDir.cd("meshlab");
 		if (pluginsDir.exists("plugins")) {
 			pluginsDir.cd("plugins");
--- Meshlab-2022.02/src/external/e57/include/E57Format.h.orig	2022-12-11 09:09:41.000000000 +0000
+++ Meshlab-2022.02/src/external/e57/include/E57Format.h	2023-08-15 22:27:14.003179261 +0100
@@ -32,6 +32,7 @@
 //! @file  E57Format.h header file for the E57 API
 
 #include <cfloat>
+#include <cstdint>
 #include <memory>
 #include <vector>
 
--- Meshlab-2022.02/src/external/nexus/src/corto/include/corto/tunstall.h.orig	2022-12-11 09:09:51.000000000 +0000
+++ Meshlab-2022.02/src/external/nexus/src/corto/include/corto/tunstall.h	2023-08-15 22:26:23.984181648 +0100
@@ -19,6 +19,7 @@
 #ifndef CRT_TUNSTALL_H
 #define CRT_TUNSTALL_H
 
+#include <cstdint>
 #include <vector>
 
 
