--- METIS-5.2.1/CMakeLists.txt.orig	2022-12-05 13:27:57.000000000 +0000
+++ METIS-5.2.1/CMakeLists.txt	2023-09-17 21:15:23.030936302 +0100
@@ -43,8 +43,8 @@
 
 # List of paths that the compiler will search for library files.
 # i.e., the -L equivalent
-link_directories(${GKLIB_PATH}/lib)
-link_directories(${CMAKE_INSTALL_PREFIX}/lib)
+link_directories(${GKLIB_PATH}/lib64)
+link_directories(${CMAKE_INSTALL_PREFIX}/lib64)
 
 # Recursively look for CMakeLists.txt in subdirs.
 add_subdirectory("build/xinclude")
--- METIS-5.2.1/libmetis/CMakeLists.txt.orig	2022-12-05 13:27:57.000000000 +0000
+++ METIS-5.2.1/libmetis/CMakeLists.txt	2023-09-17 21:19:33.087924371 +0100
@@ -7,9 +7,11 @@
 # Build libmetis.
 add_library(metis ${METIS_LIBRARY_TYPE} ${metis_sources})
 
+target_link_libraries(metis GKlib)
+
 if(METIS_INSTALL)
   install(TARGETS metis
-    LIBRARY DESTINATION lib
-    RUNTIME DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    LIBRARY DESTINATION lib64
+    RUNTIME DESTINATION lib64
+    ARCHIVE DESTINATION lib64)
 endif()
