--- mindia-0.97.3/configure.orig	2009-02-02 14:32:30.000000000 +0000
+++ mindia-0.97.3/configure	2009-02-02 14:30:06.000000000 +0000
@@ -6394,7 +6394,7 @@
 if test -z ""2.3""; then
   version="1.5"
 else
-  version=""2.3""
+  version=""2.5""
 fi
 
 { $as_echo "$as_me:$LINENO: checking for Python$version" >&5
--- mindia-0.97.3/src/iocontainer.h.orig	2003-08-15 17:38:22.000000000 +0100
+++ mindia-0.97.3/src/iocontainer.h	2009-02-02 12:14:19.000000000 +0000
@@ -73,8 +73,8 @@
 
 		aTree.SetAsContainer( true );
 
-		typename vector< minHandle<Type> >::const_iterator aIter = begin();
-		while( aIter != end() )
+		typename vector< minHandle<Type> >::const_iterator aIter = vector< minHandle<Type> >::begin();
+		while( aIter != vector< minHandle<Type> >::end() )
 		{
 			aTree.PushTag( (*aIter)->GetXMLTree() );
 
