--- mixxx-2.2.4/build/depends.py.orig	2020-05-17 09:48:42.226283023 +0100
+++ mixxx-2.2.4/build/depends.py	2020-05-17 10:33:40.859154264 +0100
@@ -615,7 +615,7 @@
             env.Append(CPPPATH=[self.SOUNDTOUCH_INTERNAL_PATH])
 
             # Prevents circular import.
-            from features import Optimize
+            from .features import Optimize
 
             # If we do not want optimizations then disable them.
             optimize = (build.flags['optimize'] if 'optimize' in build.flags
