--- mjpegtools-1.8.0/lavtools/lav_io.c.orig	2008-02-20 15:50:39.000000000 +0000
+++ mjpegtools-1.8.0/lavtools/lav_io.c	2008-02-20 15:51:21.000000000 +0000
@@ -1367,7 +1367,7 @@
          break;
 #ifdef HAVE_LIBQUICKTIME
       case 'q':
-         res = fileno(((quicktime_t *)lav_file->qt_fd)->stream);
+	 res = lqt_fileno((quicktime_t *)lav_file->qt_fd);
          break;
 #endif
       default:
--- mjpegtools-1.8.0/y4mdenoise/Region2D.hh.orig	2004-07-18 07:43:01.000000000 +0100
+++ mjpegtools-1.8.0/y4mdenoise/Region2D.hh	2008-02-20 15:51:59.000000000 +0000
@@ -15,6 +15,7 @@
 // code that will work with any implementation of regions.
 
 #include "Status_t.h"
+#include <cassert>
 #include <iostream>
 
 
