--- mjpegtools-2.0.0/configure.orig	2011-05-21 13:45:05.000000000 +0100
+++ mjpegtools-2.0.0/configure	2012-06-21 07:15:42.000000000 +0100
@@ -16392,7 +16392,7 @@
 
 case $host in
   *-*-linux*)
-  ac_fn_c_check_header_mongrel "$LINENO" "linux/videodev.h" "ac_cv_header_linux_videodev_h" "$ac_includes_default"
+  ac_fn_c_check_header_mongrel "$LINENO" "libv4l1-videodev.h" "ac_cv_header_linux_videodev_h" "$ac_includes_default"
 if test "x$ac_cv_header_linux_videodev_h" = xyes; then :
   have_video4linux=true
 
--- mjpegtools-2.0.0/lavtools/liblavplay.c.orig	2010-03-07 11:42:27.000000000 +0000
+++ mjpegtools-2.0.0/lavtools/liblavplay.c	2012-06-21 07:23:41.000000000 +0100
@@ -68,7 +68,9 @@
  * errors here, check your linux/time.h && sys/time.h header setup.
  */
 #define _LINUX_TIME_H
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
+#include <linux/videodev2.h>
+#define BASE_VIDIOCPRIVATE BASE_VIDIOC_PRIVATE
 #else
 #define VIDEO_MODE_PAL		0
 #define VIDEO_MODE_NTSC		1
--- mjpegtools-2.0.0/lavtools/lavvideo.c.orig	2004-07-29 21:26:00.000000000 +0100
+++ mjpegtools-2.0.0/lavtools/lavvideo.c	2012-06-21 07:21:44.000000000 +0100
@@ -63,7 +63,8 @@
  * errors here, check your linux/time.h && sys/time.h header setup.
  */
 #define _LINUX_TIME_H
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
+#include <linux/videodev2.h>
 
 #include <videodev_mjpeg.h>
 #include <frequencies.h>
--- mjpegtools-2.0.0/lavtools/liblavrec.c.orig	2006-05-18 16:50:42.000000000 +0100
+++ mjpegtools-2.0.0/lavtools/liblavrec.c	2012-06-21 07:23:23.000000000 +0100
@@ -63,7 +63,9 @@
  * errors here, check your linux/time.h && sys/time.h header setup.
  */
 #define _LINUX_TIME_H
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
+#include <linux/videodev2.h>
+#define BASE_VIDIOCPRIVATE BASE_VIDIOC_PRIVATE
 #ifdef HAVE_SYS_SOUNDCARD_H
 #include <sys/soundcard.h>
 #endif
--- mjpegtools-2.0.0/lavtools/testrec.c.orig	2004-07-29 21:26:00.000000000 +0100
+++ mjpegtools-2.0.0/lavtools/testrec.c	2012-06-21 07:21:08.000000000 +0100
@@ -89,7 +89,8 @@
  * errors here, check your linux/time.h && sys/time.h header setup.
  */
 #define _LINUX_TIME_H
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
+#include <linux/videodev2.h>
 #include <linux/soundcard.h>
 
 /* These are explicit prototypes for the compiler, to prepare separation of audiolib.c */
--- mjpegtools-2.0.0/y4mdenoise/MotionSearcher.hh.orig	2010-07-25 21:06:56.000000000 +0100
+++ mjpegtools-2.0.0/y4mdenoise/MotionSearcher.hh	2012-06-21 07:26:30.000000000 +0100
@@ -2196,7 +2196,7 @@
 	{
 		// This region is too small to be bothered with.
 		// Just get rid of it.
-		DeleteRegion (a_pRegion);
+		this->DeleteRegion (a_pRegion);
 	}
 	else
 	{
