--- mkvtoolnix-94.0/build-config.in.orig	2025-07-27 17:23:09.000000000 +0100
+++ mkvtoolnix-94.0/build-config.in	2025-08-17 16:01:14.221071176 +0100
@@ -78,7 +78,7 @@
 BOOST_CPPFLAGS = @BOOST_CPPFLAGS@
 BOOST_FILESYSTEM_LIB = @BOOST_FILESYSTEM_LIB@
 BOOST_LDFLAGS = @BOOST_LDFLAGS@
-BOOST_SYSTEM_LIB = @BOOST_SYSTEM_LIB@
+BOOST_SYSTEM_LIB = 
 CMARK_CFLAGS = @CMARK_CFLAGS@
 CMARK_LIBS = @CMARK_LIBS@
 COMPILER_TYPE = @COMPILER_TYPE@
--- mkvtoolnix-94.0/configure.ac.orig	2025-07-27 17:23:09.000000000 +0100
+++ mkvtoolnix-94.0/configure.ac	2025-08-17 15:29:33.186032028 +0100
@@ -39,7 +39,6 @@
 m4_include(ac/ax_boost_base.m4)
 m4_include(ac/ax_boost_check_headers.m4)
 m4_include(ac/ax_boost_filesystem.m4)
-m4_include(ac/ax_boost_system.m4)
 m4_include(ac/boost.m4)
 m4_include(ac/gtest.m4)
 m4_include(ac/benchmark.m4)
--- mkvtoolnix-94.0/ac/boost.m4.orig	2025-07-27 17:23:09.000000000 +0100
+++ mkvtoolnix-94.0/ac/boost.m4	2025-08-17 15:30:51.959360862 +0100
@@ -36,9 +36,6 @@
 fi
 AC_MSG_RESULT([yes])
 
-# boost::system must be present.
-AX_BOOST_SYSTEM()
-
 # boost::filesystem must be present.
 AX_BOOST_FILESYSTEM()
 
@@ -46,10 +43,6 @@
   AC_MSG_ERROR(The Boost Filesystem Library was not found.)
 fi
 
-if test x"$ax_cv_boost_system" != "xyes"; then
-  AC_MSG_ERROR(The Boost System Library was not found.)
-fi
-
 AX_BOOST_CHECK_HEADERS([boost/operators.hpp],,[
   AC_MSG_ERROR([Boost's Operators library is required but wasn't found])
 ])
