--- mlt-0.5.10/src/modules/core/filter_crop.c.orig	2010-09-14 07:00:09.000000000 +0100
+++ mlt-0.5.10/src/modules/core/filter_crop.c	2010-09-22 06:22:17.000000000 +0100
@@ -88,8 +88,8 @@
 		// Subsampled YUV is messy and less precise.
 		if ( *format == mlt_image_yuv422 )
 		{
-			*format = mlt_image_rgb24;
-			mlt_frame_get_image( this, image, format, width, height, writable );
+			mlt_image_format requested_format = mlt_image_rgb24;
+			this->convert_image( this, image, format, requested_format );
 		}
 	
 		mlt_log_debug( NULL, "[filter crop] %s %dx%d -> %dx%d\n", mlt_image_format_name(*format),
