--- mlt-0.7.8/src/modules/avformat/consumer_avformat.c.orig	2012-02-14 06:02:46.000000000 +0000
+++ mlt-0.7.8/src/modules/avformat/consumer_avformat.c	2012-02-19 07:54:17.000000000 +0000
@@ -257,7 +257,7 @@
 		mlt_properties_set_data( properties, "vcodec", codecs, 0, (mlt_destructor) mlt_properties_close, NULL );
 		mlt_properties_set_data( doc, "video_codecs", codecs, 0, NULL, NULL );
 		while ( ( codec = av_codec_next( codec ) ) )
-#if LIBAVCODEC_VERSION_INT >= ((53<<16)+(34<<8)+0)
+#if LIBAVCODEC_VERSION_INT >= ((53<<16)+(51<<8)+0)
 			if ( (codec->encode || codec->encode2) && codec->type == CODEC_TYPE_VIDEO )
 #else
 			if ( codec->encode && codec->type == CODEC_TYPE_VIDEO )
