--- monkey-bubble-20060615/src/audio/sound-manager.c.orig	2006-06-14 18:21:24.000000000 +0100
+++ monkey-bubble-20060615/src/audio/sound-manager.c	2006-06-21 14:01:55.000000000 +0100
@@ -186,12 +186,12 @@
 	}
 
 	if(PRIVATE(m)->main_bin) {
-		gst_element_set_state(PRIVATE(m)->main_bin, GST_STATE_PAUSED);
+		gst_element_set_state(PRIVATE(m)->main_bin, GST_STATE_NULL);
 		g_object_unref(PRIVATE(m)->main_bin);
 		PRIVATE(m)->main_bin = NULL;
 	}
 
-	g_object_unref( G_OBJECT(PRIVATE(m)->main_bin ));
+//	g_object_unref( G_OBJECT(PRIVATE(m)->main_bin ));
 	  
 	PRIVATE(m)->output = NULL;
 	PRIVATE(m)->main_bin = NULL;
