--- moon-0.8.1/src/pipeline-ffmpeg.cpp.orig	2008-08-18 18:24:12.000000000 +0100
+++ moon-0.8.1/src/pipeline-ffmpeg.cpp	2008-10-07 22:53:46.000000000 +0100
@@ -142,7 +142,7 @@
 		VideoStream *vs = (VideoStream*) stream;
 		context->width = vs->width;
 		context->height = vs->height;
-		context->bits_per_sample = vs->bits_per_sample;
+		context->bits_per_coded_sample = vs->bits_per_sample;
 		context->codec_type = CODEC_TYPE_VIDEO;
 	} else if (stream->GetType () == MediaTypeAudio) {
 		AudioStream *as = (AudioStream*) stream;
