--- motion-3.2.10.1/configure.orig	2008-06-10 23:39:40.000000000 +0100
+++ motion-3.2.10.1/configure	2008-10-06 06:36:55.000000000 +0100
@@ -4321,9 +4321,9 @@
 	    s//\1/
 	    q
 	  }
-	  s/.*/./; q'`/include/ffmpeg/avformat.h; then
-		{ echo "$as_me:$LINENO: result: found ${FFMPEG}/include/ffmpeg/avformat.h" >&5
-echo "${ECHO_T}found ${FFMPEG}/include/ffmpeg/avformat.h" >&6; }
+	  s/.*/./; q'`/include/libavformat/avformat.h; then
+		{ echo "$as_me:$LINENO: result: found ${FFMPEG}/include/libavformat/avformat.h" >&5
+echo "${ECHO_T}found ${FFMPEG}/include/libavformat/avformat.h" >&6; }
 		FFMPEG_CFLAGS="-I`$as_dirname -- ${FFMPEG} ||
 $as_expr X${FFMPEG} : 'X\(.*[^/]\)//*[^/][^/]*/*$' \| \
 	 X${FFMPEG} : 'X\(//\)[^/]' \| \
--- motion-3.2.10.1/ffmpeg.c.orig	2008-06-10 23:39:40.000000000 +0100
+++ motion-3.2.10.1/ffmpeg.c	2008-10-06 06:50:47.000000000 +0100
@@ -122,7 +122,7 @@
  * 
  */ 
    
-#include "avstring.h"
+#include "libavutil/avstring.h"
 
 static int file_open(URLContext *h, const char *filename, int flags)
 {
@@ -159,7 +159,7 @@
 	return write(fd, buf, size);
 }
 
-static offset_t file_seek(URLContext *h, offset_t pos, int whence)
+static off_t file_seek(URLContext *h, off_t pos, int whence)
 {
 	int fd = (size_t)h->priv_data;
 	return lseek(fd, pos, whence);
--- motion-3.2.10.1/ffmpeg.h.orig	2008-06-10 23:39:40.000000000 +0100
+++ motion-3.2.10.1/ffmpeg.h	2008-10-06 06:40:31.000000000 +0100
@@ -3,7 +3,7 @@
 
 #ifdef HAVE_FFMPEG
 #include <errno.h>
-#include <avformat.h>
+#include <libavformat/avformat.h>
 
 #ifndef AVERROR /* 0.4.8 & 0.4.9-pre1 */
 
