--- MPlayer-cvs-20060405/libmpdemux/demux_mov.c.orig	2006-03-27 22:37:18.000000000 +0100
+++ MPlayer-cvs-20060405/libmpdemux/demux_mov.c	2006-04-07 13:33:53.000000000 +0100
@@ -1911,9 +1911,9 @@
     {
 	mp_msg(MSGT_DEMUX, MSGL_DBG2, "WARNING! Samplesize(%d) != 1\n",
 	    trak->samplesize);
-//	x=trak->chunks[trak->pos].size*trak->samplesize;
+	x=trak->chunks[trak->pos].size*trak->samplesize;
     }
-//    else
+    else
 	x=trak->chunks[trak->pos].size;
 //    printf("X = %d\n", x);
     /* the following stuff is audio related */
