--- MPlayer-1.0pre7try2/libmpdemux/demux_mov.c.orig	2005-03-30 23:10:07.000000000 +0100
+++ MPlayer-1.0pre7try2/libmpdemux/demux_mov.c	2005-12-21 10:52:59.000000000 +0000
@@ -874,6 +874,8 @@
 			trak->samplebytes=sh->samplesize=1;
 			trak->nchannels=sh->channels=1;
 			sh->samplerate=8000;
+			trak->stdata[24]=sh->samplerate >> 8;
+			trak->stdata[25]=sh->samplerate & 0xff;
 			break;
 			
 		    case 0x62776173: /* sawb */
@@ -881,6 +883,8 @@
 			trak->samplebytes=sh->samplesize=1;
 			trak->nchannels=sh->channels=1;
 			sh->samplerate=16000;
+			trak->stdata[24]=sh->samplerate >> 8;
+			trak->stdata[25]=sh->samplerate & 0xff;
 			break;
 
 		    default:
