--- MPlayer-1.0pre7try2/libdha/Makefile.orig	2005-01-01 14:45:11.000000000 +0000
+++ MPlayer-1.0pre7try2/libdha/Makefile	2005-09-02 11:54:14.000000000 +0100
@@ -82,11 +82,6 @@
 	install -m 755 -s -p $(LIBNAME) $(LIBDIR)/$(LIBNAME)
 	rm -f $(LIBDIR)/libdha.so
 	ln -sf $(LIBNAME) $(LIBDIR)/$(SHORTNAME)
-ifeq ($(TARGET_OS),OpenBSD)
-	ldconfig -R
-else
-	ldconfig
-endif
 
 uninstall:
 	rm -f $(LIBDIR)/libdha.so $(LIBDIR)/$(SHORTNAME) $(LIBDIR)/$(LIBNAME)
