--- muine-0.8.7/libmuine/metadata.c.orig	2007-01-19 22:14:07.000000000 +0000
+++ muine-0.8.7/libmuine/metadata.c	2007-06-02 20:10:19.000000000 +0100
@@ -19,6 +19,7 @@
 
 #include <libgnomevfs/gnome-vfs.h>
 #include <vorbis/vorbisfile.h>
+#include <FLAC/export.h>
 #include <FLAC/metadata.h>
 #include <FLAC/stream_decoder.h>
 #include <glib.h>
@@ -829,6 +830,7 @@
 
 	flac_decoder = FLAC__stream_decoder_new ();
 
+#if !defined(FLAC_API_VERSION_CURRENT) || FLAC_API_VERSION_CURRENT <= 7
 	FLAC__stream_decoder_set_read_callback (flac_decoder, FLAC_read_callback);
 	FLAC__stream_decoder_set_write_callback (flac_decoder, FLAC_write_callback);
 	FLAC__stream_decoder_set_metadata_callback (flac_decoder, FLAC_metadata_callback);
@@ -845,6 +847,27 @@
 	FLAC__stream_decoder_set_metadata_respond (flac_decoder, FLAC__METADATA_TYPE_VORBIS_COMMENT);
 
 	FLAC__stream_decoder_init (flac_decoder);
+#else
+	callback_data = g_new0 (CallbackData, 1);
+	callback_data->handle = handle;
+	callback_data->comment = comment;
+
+	/* by default, only the STREAMINFO block is parsed and passed to
+	 * the metadata callback.  Here we instruct the decoder to also
+	 * pass us the VORBISCOMMENT block if there is one. */
+	FLAC__stream_decoder_set_metadata_respond (flac_decoder, FLAC__METADATA_TYPE_VORBIS_COMMENT);
+
+	FLAC__stream_decoder_init_stream (flac_decoder,
+					  FLAC_read_callback,
+					  NULL,
+					  NULL,
+					  NULL,
+					  NULL,
+					  FLAC_write_callback,
+					  FLAC_metadata_callback,
+					  FLAC_error_callback,
+					  callback_data);
+#endif
 
 	/* this runs the decoding process, calling the callbacks as appropriate */
 	if (FLAC__stream_decoder_process_until_end_of_metadata (flac_decoder) == 0) {
