--- musepack-r475/libmpcdec/CMakeLists.txt.orig	2010-08-21 16:55:01.000000000 +0100
+++ musepack-r475/libmpcdec/CMakeLists.txt	2012-01-08 19:22:57.391772586 +0000
@@ -1,6 +1,8 @@
 include_directories(${libmpc_SOURCE_DIR}/include)
 if(SHARED)
   add_library(mpcdec SHARED huffman mpc_decoder mpc_reader streaminfo mpc_bits_reader mpc_demux requant synth_filter ${libmpc_SOURCE_DIR}/common/crc32)
+  find_library(M_LIB m)
+  target_link_libraries(mpcdec ${M_LIB})
 else(SHARED)
   add_library(mpcdec_static STATIC huffman mpc_decoder mpc_reader streaminfo mpc_bits_reader mpc_demux requant synth_filter ${libmpc_SOURCE_DIR}/common/crc32)
 endif(SHARED)
