--- mpcsv7-src-1.15v/mpp.h.orig	2004-11-20 15:03:58.000000000 +0000
+++ mpcsv7-src-1.15v/mpp.h	2006-06-13 22:14:21.000000000 +0100
@@ -54,12 +54,12 @@
 // Sound support for SGI Irix
 // If you have a SGI Workstation running IRIX, you can output to /dev/audio and
 // instead of writing a file the program plays the file via this sound device.
-#define USE_IRIX_AUDIO
+//#define USE_IRIX_AUDIO
 
 // Audio support for Windows (WAVE OUT) (only Windows)
 // If you have a Windows based system and if you want also play files directly instead of only writing audio files,
 // then define the next item
-#define USE_WIN_AUDIO
+//#define USE_WIN_AUDIO
 
 // Buffersize for Windows Audio in 4.5 KByte units
 // Only needed for Windows+USE_WIN_AUDIO
--- mpcsv7-src-1.15v/Makefile.orig	2005-01-28 14:56:12.000000000 +0000
+++ mpcsv7-src-1.15v/Makefile	2006-06-13 22:18:16.000000000 +0100
@@ -98,7 +98,7 @@
 
 ifndef MINGW
 LDADD    = -lm
-#LDADD   += -lesd
+LDADD   += -lesd
 else
 LDADD    = -lwinmm -lws2_32
 endif
@@ -322,21 +322,21 @@
 	make clean
 	BITS=16 make mppdec
 	$(MV_F) mppdec mppdec16
-	$(MV_F) mppdec-static mppdec16-static
+#	$(MV_F) mppdec-static mppdec16-static
 
 
 $(MPPDEC_TARGET)24: $(MPPDEC_OBJ)
 	make clean
 	BITS=24 make mppdec
 	$(MV_F) mppdec mppdec24
-	$(MV_F) mppdec-static mppdec24-static
+#	$(MV_F) mppdec-static mppdec24-static
 
 
 $(MPPDEC_TARGET)32: $(MPPDEC_OBJ)
 	make clean
 	BITS=32 make mppdec
 	$(MV_F) mppdec mppdec32
-	$(MV_F) mppdec-static mppdec32-static
+#	$(MV_F) mppdec-static mppdec32-static
 
 
 $(MPPENC_TARGET): $(MPPENC_OBJ)
