--- mythtv-0.23_r24674/programs/mythbackend/scheduler.cpp.orig	2010-05-15 06:18:54.000000000 +0100
+++ mythtv-0.23_r24674/programs/mythbackend/scheduler.cpp	2010-05-15 12:57:19.000000000 +0100
@@ -2029,7 +2029,6 @@
         {
             MythEvent me("SCHEDULE_CHANGE");
             gContext->dispatch(me);
-            idleSince = QDateTime();
         }
 
         // if idletimeout is 0, the user disabled the auto-shutdown feature
@@ -2062,8 +2061,11 @@
                     }
                     resetIdleTime_lock.unlock();
 
-                    if (!idleSince.isValid())
+                    if (statuschanged || !idleSince.isValid())
                     {
+                        if (!idleSince.isValid())
+                            idleSince = curtime;
+
                         RecIter idleIter = reclist.begin();
                         for ( ; idleIter != reclist.end(); idleIter++)
                             if ((*idleIter)->recstatus == rsWillRecord)
@@ -2072,17 +2074,16 @@
                         if (idleIter != reclist.end())
                         {
                             if (curtime.secsTo((*idleIter)->recstartts) -
-                                prerollseconds >
+                                prerollseconds <
                                 (idleWaitForRecordingTime * 60) +
                                 idleTimeoutSecs)
                             {
-                                idleSince = curtime;
+                                idleSince = QDateTime();;
                             }
                         }
-                        else
-                            idleSince = curtime;
                     }
-                    else
+
+                    if (idleSince.isValid())
                     {
                         // is the machine already idling the timeout time?
                         if (idleSince.addSecs(idleTimeoutSecs) < curtime)
