--- mythtv-0.22_r22969/programs/mythbackend/scheduler.cpp.orig	2009-11-14 14:38:30.000000000 +0000
+++ mythtv-0.22_r22969/programs/mythbackend/scheduler.cpp	2009-12-20 09:32:43.000000000 +0000
@@ -1968,7 +1968,6 @@
         {
             MythEvent me("SCHEDULE_CHANGE");
             gContext->dispatch(me);
-            idleSince = QDateTime();
         }
 
         // if idletimeout is 0, the user disabled the auto-shutdown feature
@@ -2001,8 +2000,11 @@
                     }
                     resetIdleTime_lock.unlock();
 
-                    if (!idleSince.isValid())
+                    if (statuschanged || !idleSince.isValid())
                     {
+                        if (!idleSince.isValid())
+                            idleSince = curtime;
+
                         RecIter idleIter = reclist.begin();
                         for ( ; idleIter != reclist.end(); idleIter++)
                             if ((*idleIter)->recstatus == rsWillRecord)
@@ -2011,15 +2013,14 @@
                         if (idleIter != reclist.end())
                         {
                             if (curtime.secsTo((*idleIter)->recstartts) -
-                                prerollseconds > idleWaitForRecordingTime * 60)
+                                prerollseconds < idleWaitForRecordingTime * 60)
                             {
-                                idleSince = curtime;
+                                idleSince = QDateTime();
                             }
                         }
-                        else
-                            idleSince = curtime;
                     }
-                    else
+
+                    if (idleSince.isValid())
                     {
                         // is the machine already idling the timeout time?
                         if (idleSince.addSecs(idleTimeoutSecs) < curtime)
