--- nasm-2.13.01/misc/omfdump.c.orig	2017-05-02 06:18:37.000000000 +0100
+++ nasm-2.13.01/misc/omfdump.c	2017-10-09 16:23:26.000000000 +0100
@@ -20,14 +20,27 @@
 
 static const char *record_types[256] =
 {
+    [0x6e] = "RHEADR",
+    [0x70] = "REGINT",
+    [0x72] = "REDATA",
+    [0x74] = "RIDATA",
+    [0x76] = "OVLDEF",
+    [0x78] = "ENDREC",
+    [0x7a] = "BLKDEF",
+    [0x7c] = "BLKEND",
+    [0x7e] = "DEBSYM",
     [0x80] = "THEADR",
     [0x82] = "LHEADR",
+    [0x84] = "PEDATA",
+    [0x86] = "PIDATA",
     [0x88] = "COMENT",
     [0x8a] = "MODEND16",
     [0x8b] = "MODEND32",
     [0x8c] = "EXTDEF",
+    [0x8e] = "TYPDEF",
     [0x90] = "PUBDEF16",
     [0x91] = "PUBDEF32",
+    [0x92] = "LOCSYM",
     [0x94] = "LINNUM16",
     [0x95] = "LINNUM32",
     [0x96] = "LNAMES",
@@ -40,6 +53,8 @@
     [0xa1] = "LEDATA32",
     [0xa2] = "LIDATA16",
     [0xa3] = "LIDATA32",
+    [0xa8] = "LIBLOC",
+    [0xaa] = "LIBDIC",
     [0xb0] = "COMDEF",
     [0xb2] = "BAKPAT16",
     [0xb3] = "BAKPAT32",
