Index: src/netdaemon.c
================================================================================
--- src/netdaemon.c
+++ src/netdaemon.c
@@ -955,7 +956,7 @@
 			char ap_addr[128];
 
 			memset (ap_addr, 0, 128);
-			iw_pr_ether (ap_addr, wrq.u.ap_addr.sa_data);
+			iw_saether_ntop (&wrq.u.ap_addr, ap_addr);
 
 			if (!is_valid_ap_address (ap_addr)) {
 				error = TRUE;
@@ -1093,7 +1094,7 @@
 		char ap_addr[128];
 
 		memset (ap_addr, 0, 128);
-		iw_pr_ether (ap_addr, wrq.u.ap_addr.sa_data);
+		iw_saether_ntop (&wrq.u.ap_addr, ap_addr);
 
 		if (!is_valid_ap_address (ap_addr)) {
 			/* This is kind of ugly... */
