Index: src/netapplet.c
===================================================================
RCS file: /cvs/gnome/netapplet/src/netapplet.c,v
retrieving revision 1.11
diff -u -r1.11 netapplet.c
--- src/netapplet.c	13 Mar 2005 23:39:22 -0000	1.11
+++ src/netapplet.c	19 Mar 2005 13:57:49 -0000
@@ -1376,6 +1376,9 @@
 	Accesspoint *new_active = NULL;
 	gboolean changed = FALSE;
 
+	if (strcmp (netapplet->active->type, "Wireless") != 0)
+		return;
+
 	for (iter = &args[2]; *iter != NULL; iter += 3) {
 		Accesspoint *ap;
 
