--- netsniff-ng-0.6.4/mausezahn/Makefile.orig	2018-01-05 11:02:22.000000000 +0000
+++ netsniff-ng-0.6.4/mausezahn/Makefile	2018-06-23 15:23:56.185023685 +0100
@@ -1,4 +1,4 @@
-mausezahn-libs =	-lcli \
+mausezahn-libs =	-lciscocli \
 			-lnet \
 			-lpcap \
 			-lrt \
--- netsniff-ng-0.6.4/staging/cli.h.orig	2018-01-05 11:02:22.000000000 +0000
+++ netsniff-ng-0.6.4/staging/cli.h	2018-06-23 15:23:02.287026256 +0100
@@ -22,7 +22,7 @@
 #define __MAUSEZAHN_CLI__
 
 #include <sys/time.h>
-#include <libcli.h>
+#include <libciscocli.h>
 #include "mops.h"
 
 #define CLI_DEBUG_PACKET      0x0001
--- netsniff-ng-0.6.4/configure.orig	2018-01-05 11:02:22.000000000 +0000
+++ netsniff-ng-0.6.4/configure	2018-06-23 15:23:41.773024372 +0100
@@ -649,7 +649,7 @@
 
 	cat > $TMPDIR/clitest.c << EOF
 #include <sys/time.h>
-#include <libcli.h>
+#include <libciscocli.h>
 
 int main(void)
 {
@@ -657,7 +657,7 @@
 }
 EOF
 
-	$CC -o $TMPDIR/clitest $TMPDIR/clitest.c -lcli >> config.log 2>&1
+	$CC -o $TMPDIR/clitest $TMPDIR/clitest.c -lciscocli >> config.log 2>&1
 	if [ ! -x $TMPDIR/clitest ] ; then
 		echo "[NO]"
 		MISSING_DEFS=1
