--- newt-0.52.13/configure.orig	2011-06-27 13:27:40.000000000 +0100
+++ newt-0.52.13/configure	2011-10-23 21:45:33.000000000 +0100
@@ -3455,7 +3455,7 @@
 $as_echo "skipped" >&6; }
   PYTHONVERS=
 else
-  PYTHONVERS=$(ls /usr/include/python*/Python.h 2> /dev/null | sed "s|/usr/include/||g"| sed "s|/Python.h||g" 2> /dev/null)
+  PYTHONVERS=$(ls /usr/include/python*/Python.h 2> /dev/null | sed "s|/usr/include/||g"| sed "s|/Python.h||g" | tr "\012" " " 2> /dev/null)
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: $PYTHONVERS" >&5
 $as_echo "$PYTHONVERS" >&6; }
 fi
