--- ./configure.orig	2007-05-10 22:43:34.000000000 -0500
+++ ./configure	2007-05-17 17:17:22.000000000 -0500
@@ -3931,7 +3931,7 @@
                     CPPFLAGS="-I${with_tcpw} ${CPPFLAGS}"
                 fi
             fi
-            LIBWRAP="-lwrap"
+            LIBWRAP="-lwrap -lnsl"
             LIBS="$LIBWRAP $LIBS"
             { echo "$as_me:$LINENO: checking for libwrap" >&5
 echo $ECHO_N "checking for libwrap... $ECHO_C" >&6; }
