--- nfs-utils-1.1.4/configure.orig	2008-10-17 15:24:00.000000000 +0100
+++ nfs-utils-1.1.4/configure	2008-10-22 12:10:42.000000000 +0100
@@ -3945,7 +3945,7 @@
                     CPPFLAGS="-I${with_tcpw} ${CPPFLAGS}"
                 fi
             fi
-            LIBWRAP="-lwrap"
+            LIBWRAP="-lwrap -lnsl"
             LIBS="$LIBWRAP $LIBS"
             { echo "$as_me:$LINENO: checking for libwrap" >&5
 echo $ECHO_N "checking for libwrap... $ECHO_C" >&6; }
@@ -22994,7 +22994,7 @@
 { echo "$as_me:$LINENO: result: $ac_cv_lib_blkid_blkid_get_library_version" >&5
 echo "${ECHO_T}$ac_cv_lib_blkid_blkid_get_library_version" >&6; }
 if test $ac_cv_lib_blkid_blkid_get_library_version = yes; then
-  LIBBLKID="-lblkid"
+  LIBBLKID="-lblkid -luuid"
 else
   { { echo "$as_me:$LINENO: error: libblkid needed" >&5
 echo "$as_me: error: libblkid needed" >&2;}
