--- ngplant-2020413/ngplant/p3dmedit.cpp.orig	2020-04-13 08:28:03.000000000 +0100
+++ ngplant-2020413/ngplant/p3dmedit.cpp	2021-12-05 22:42:42.621612849 +0000
@@ -1684,7 +1684,7 @@
    }
 
   UpdateControlSizer->Add(AutoUpdateCheckBox,1,wxALIGN_LEFT | wxALIGN_CENTER_VERTICAL,0);
-  UpdateControlSizer->Add(UpdateButton,0,wxALIGN_RIGHT | wxALIGN_CENTER_VERTICAL,0);
+  UpdateControlSizer->Add(UpdateButton,0,/*wxALIGN_RIGHT |*/ wxALIGN_CENTER_VERTICAL,0);
 
   TopSizer->Add(UpdateControlSizer,0,wxGROW | wxALL,2);
 
--- ngplant-2020413/ngplant/p3dcanvas3d.cpp.orig	2021-12-01 20:58:58.238867207 +0000
+++ ngplant-2020413/ngplant/p3dcanvas3d.cpp	2021-12-06 19:11:22.689026151 +0000
@@ -104,7 +104,7 @@
  {
   #if defined(__WXGTK__)
    {
-    Render();
+    Refresh();
    }
   #else
    {
