--- ngplant-2020413/sctool/SConcheck.py.orig	2020-11-01 14:14:39.566279094 +0000
+++ ngplant-2020413/sctool/SConcheck.py	2021-12-01 20:51:03.560889855 +0000
@@ -147,7 +147,7 @@
     return result
 
 def GetEnvKeyOpt(Env,Key,Def = None):
-    if Env.has_key(Key):
+    if Key in Env:
         return Env[Key]
     else:
         return Def
@@ -196,14 +196,14 @@
 def ConfigureGLU(Context):
     Context.Message('Checking GLU presence and usability ... ')
 
-    if   not Context.env.has_key('GLU_INC'):
+    if   'GLU_INC' not in Context.env:
         Context.env.Append(GLU_INC=[])
     elif Context.env['GLU_INC'] != '':
         Context.env.Replace(GLU_INC=Split(Context.env['GLU_INC']))
     else:
         Context.env.Replace(GLU_INC=[])
 
-    if   not Context.env.has_key('GLU_LIBPATH'):
+    if   'GLU_LIBPATH' not in Context.env:
         Context.env.Append(GLU_LIBPATH=[])
     elif Context.env['GLU_LIBPATH'] != '':
         Context.env.Replace(GLU_LIBPATH=Split(Context.env['GLU_LIBPATH']))
@@ -215,7 +215,7 @@
     else:
         DefaultGLULibs=[]
 
-    if  not Context.env.has_key('GLU_LIBS'):
+    if  'GLU_LIBS' not in Context.env:
         Context.env.Append(GLU_LIBS=DefaultGLULibs)
     elif Context.env['GLU_LIBS'] != '':
         Context.env.Replace(GLU_LIBS=Split(Context.env['GLU_LIBS']))
@@ -244,21 +244,21 @@
 def ConfigureGLEW(Context):
     Context.Message('Checking GLEW presence and usability ... ')
 
-    if   not Context.env.has_key('GLEW_INC'):
+    if   'GLEW_INC' not in Context.env:
         Context.env.Append(GLEW_INC=[])
     elif Context.env['GLEW_INC'] != '':
         Context.env.Replace(GLEW_INC=Split(Context.env['GLEW_INC']))
     else:
         Context.env.Replace(GLEW_INC=[])
 
-    if   not Context.env.has_key('GLEW_LIBPATH'):
+    if   'GLEW_LIBPATH' not in Context.env:
         Context.env.Append(GLEW_LIBPATH=[])
     elif Context.env['GLEW_LIBPATH'] != '':
         Context.env.Replace(GLEW_LIBPATH=Split(Context.env['GLEW_LIBPATH']))
     else:
         Context.env.Replace(GLEW_LIBPATH=[])
 
-    if  not Context.env.has_key('GLEW_LIBS'):
+    if  'GLEW_LIBS' not in Context.env:
         Context.env.Append(GLEW_LIBS=['GLEW'])
     elif Context.env['GLEW_LIBS'] != '':
         Context.env.Replace(GLEW_LIBS=Split(Context.env['GLEW_LIBS']))
--- ngplant-2020413/ngplant/SConscript.orig	2020-11-01 14:14:39.567279094 +0000
+++ ngplant-2020413/ngplant/SConscript	2021-12-01 20:52:22.906886070 +0000
@@ -59,10 +59,10 @@
 if TimingsEnabled:
     NGPlantEnv.Append(CPPDEFINES=[('P3D_TIMINGS_ENABLED','1')])
 
-if NGPlantEnv.has_key('PLUGINS_DIR') and NGPlantEnv['PLUGINS_DIR'] is not None:
+if 'PLUGINS_DIR' in NGPlantEnv and NGPlantEnv['PLUGINS_DIR'] is not None:
     NGPlantEnv.Append(CPPDEFINES=[('PLUGINS_DIR','\\"' + NGPlantEnv['PLUGINS_DIR'] + '\\"')])
 
-if NGPlantEnv.has_key('EXTRA_VERSION') and NGPlantEnv['EXTRA_VERSION'] is not None:
+if 'EXTRA_VERSION' in NGPlantEnv and NGPlantEnv['EXTRA_VERSION'] is not None:
     NGPlantEnv.Append(CPPDEFINES=[('EXTRA_VERSION','\\"' + NGPlantEnv['EXTRA_VERSION'] + '\\"')])
 
 if CrossCompileMode:
--- ngplant-2020413/SConstruct.orig	2020-11-01 14:14:39.567279094 +0000
+++ ngplant-2020413/SConstruct	2021-12-01 20:47:35.478738436 +0000
@@ -130,7 +130,7 @@
     PythonIncPath = GetPythonIncPath()
 
     if PythonIncPath is not None:
-        if BaseConf.env.has_key('CPPPATH'):
+        if 'CPPPATH' in BaseConf.env:
             lastCPPPATH = BaseConf.env['CPPPATH']
         else:
             lastCPPPATH = None
