--- nload-0.7.4/docs/Makefile.am.orig	2002-08-01 18:23:47.000000000 +0100
+++ nload-0.7.4/docs/Makefile.am	2023-05-19 17:27:00.740535580 +0100
@@ -4,7 +4,7 @@
 
 install:
 	$(mkinstalldirs) $(DESTDIR)$(mandir)/man1
-	$(INSTALL_DATA) $(top_srcdir)/docs/$(PACKAGE).1 $(DESTDIR)$(mandir)/man1
+	$(INSTALL_DATA) $(man1_MAN) $(DESTDIR)$(mandir)/man1
 	gzip -f $(DESTDIR)$(mandir)/man1/$(PACKAGE).1
 
 uninstall:
