--- nmap-6.25/nping/configure.orig	2012-11-16 00:54:16.000000000 +0000
+++ nmap-6.25/nping/configure	2013-06-15 09:50:58.654145622 +0100
@@ -4854,7 +4854,8 @@
 
 fi
 
-
+if false
+then
 # libpcap can require libnl
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for library containing nl_handle_alloc" >&5
 $as_echo_n "checking for library containing nl_handle_alloc... " >&6; }
@@ -4911,7 +4912,7 @@
   test "$ac_res" = "none required" || LIBS="$ac_res $LIBS"
 
 fi
-
+fi
 
 # We test whether they specified openssl desires explicitly
 use_openssl="yes"
