--- nut-12.5/Makefile.orig	2007-01-29 05:06:37.000000000 +0000
+++ nut-12.5/Makefile	2007-04-27 22:29:24.000000000 +0100
@@ -1,6 +1,6 @@
 objects := $(patsubst %.c,%.o,$(wildcard *.c))
 
-CFLAGS= -Os -s -Wall -DNUTDIR=\".nutdb\" -DFOODDIR=\"/usr/local/lib/nut\" -ansi -pedantic
+CFLAGS= -Os -s -Wall -DNUTDIR=\".nutdb\" -DFOODDIR=\"/usr/share/nutrition\" -ansi -pedantic
 
 nut:	$(objects)
 	gcc ${CFLAGS} -lm -o nut $(objects)
