--- obs-studio-32.0.2/frontend/plugins/frontend-tools/CMakeLists.txt.orig	2025-10-23 17:50:30.000000000 +0100
+++ obs-studio-32.0.2/frontend/plugins/frontend-tools/CMakeLists.txt	2025-10-23 17:57:26.280409202 +0100
@@ -65,8 +65,8 @@
     Qt::Widgets
     "$<$<PLATFORM_ID:Darwin>:$<LINK_LIBRARY:FRAMEWORK,Cocoa>>"
     $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>:X11::X11>
-    $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>:Qt::GuiPrivate>
 )
+#    $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>:Qt::GuiPrivate>
 
 add_subdirectory("${CMAKE_SOURCE_DIR}/shared/obs-scripting" "${CMAKE_BINARY_DIR}/shared/obs-scripting")
 
--- obs-studio-32.0.2/frontend/cmake/os-linux.cmake.orig	2025-10-23 17:50:30.000000000 +0100
+++ obs-studio-32.0.2/frontend/cmake/os-linux.cmake	2025-10-23 18:05:52.842842968 +0100
@@ -3,7 +3,8 @@
   obs-studio
   PRIVATE OBS_INSTALL_PREFIX="${OBS_INSTALL_PREFIX}" $<$<BOOL:${ENABLE_PORTABLE_CONFIG}>:ENABLE_PORTABLE_CONFIG>
 )
-target_link_libraries(obs-studio PRIVATE Qt::GuiPrivate Qt::DBus)
+target_link_libraries(obs-studio PRIVATE Qt::DBus)
+# Qt::GuiPrivate
 
 find_package(Libpci REQUIRED)
 target_link_libraries(obs-studio PRIVATE Libpci::pci)
