--- ocropus-0.4.3/SConstruct.orig	2010-01-10 13:35:53.000000000 +0000
+++ ocropus-0.4.3/SConstruct	2010-01-10 14:33:20.000000000 +0000
@@ -100,6 +100,8 @@
 else:
     print "NOTE: compiling for development (slower but safer)"
 
+env.Append(CPPPATH=["include","ocr-utils"])
+
 ################################################################
 ### libraries
 ################################################################
@@ -152,7 +154,7 @@
 
 ### gsl
 
-env.Append(LIBS=["gsl","blas"])
+env.Append(LIBS=["gsl","gslcblas"])
 
 if re.search("-O3",env["opt"]):
     env.Append(CXXFLAGS=["-fopenmp"])
@@ -164,8 +166,6 @@
 ### main targets
 ################################################################
 
-env.Append(CPPPATH=["include","ocr-utils"])
-
 #env.Append(CPPPATH=glob("ocr-*"))
 env.Append(LIBPATH=['.'])
 
