--- ocropus-0.4.3/ocr-utils/xml-entities.cc.orig	2010-01-10 13:35:54.000000000 +0000
+++ ocropus-0.4.3/ocr-utils/xml-entities.cc	2010-06-16 06:43:13.000000000 +0100
@@ -115,10 +115,10 @@
         CHECK_ARG(src);
         dest.clear();
         const char *begin = src;
-        char *p;
+        const char *p;
         while((p = strchr(begin, '&'))) {
             append_to_nustring(dest, begin, p - begin);
-            begin = append_XML_entity(dest, p);
+            begin = append_XML_entity(dest, (char *)p);
             CHECK_CONDITION(begin);
         }
         append_to_nustring(dest, begin);
--- ocropus-0.4.3/SConstruct.orig	2010-01-10 13:35:53.000000000 +0000
+++ ocropus-0.4.3/SConstruct	2010-01-10 14:33:20.000000000 +0000
@@ -100,6 +100,8 @@
 else:
     print "NOTE: compiling for development (slower but safer)"
 
+env.Append(CPPPATH=["include","ocr-utils"])
+
 ################################################################
 ### libraries
 ################################################################
@@ -152,7 +154,7 @@
 
 ### gsl
 
-env.Append(LIBS=["gsl","blas"])
+env.Append(LIBS=["gsl","gslcblas"])
 
 if re.search("-O3",env["opt"]):
     env.Append(CXXFLAGS=["-fopenmp"])
@@ -164,8 +166,6 @@
 ### main targets
 ################################################################
 
-env.Append(CPPPATH=["include","ocr-utils"])
-
 #env.Append(CPPPATH=glob("ocr-*"))
 env.Append(LIBPATH=['.'])
 
