--- octave-4.2.1/libgui/src/octave-qt-link.cc.orig	2017-02-22 18:01:55.000000000 +0000
+++ octave-4.2.1/libgui/src/octave-qt-link.cc	2017-03-01 08:56:20.218150443 +0000
@@ -67,6 +67,8 @@
 
 octave_qt_link::~octave_qt_link (void)
 {
+  main_thread->quit ();
+  main_thread->wait ();
   delete command_interpreter;
   delete main_thread;
 }
@@ -102,7 +104,7 @@
 
   // Could wait for a while and then timeout, but for now just
   // assume the GUI application exit will be without problems.
-  return true;
+  return false;
 }
 
 bool
