--- octave-6.4.0/scripts/pkg/private/install.m.orig	2021-10-30 15:20:24.000000000 +0100
+++ octave-6.4.0/scripts/pkg/private/install.m	2022-03-20 21:51:59.602543475 +0000
@@ -68,7 +68,10 @@
     for i = 1:length (files)
       tgz = files{i};
 
-      if (exist (tgz, "file"))
+      ## The filename pointed to an uncompressed package to begin with.
+      if (exist (tgz, "dir"))
+        dirlist = {".", "..", tgz};
+      elseif (exist (tgz, "file"))
         ## Create a temporary directory.
         tmpdir = tempname ();
         tmpdirs{end+1} = tmpdir;
@@ -103,17 +106,16 @@
         endif
       endif
 
-      ## The filename pointed to an uncompressed package to begin with.
-      if (isfolder (tgz))
-        dirlist = {".", "..", tgz};
-      endif
-
-      if (exist (tgz, "file") || isfolder (tgz))
+      if (exist (tgz, "file"))
         ## The two first entries of dirlist are "." and "..".
-        if (exist (tgz, "file"))
-          packdir = fullfile (tmpdir, dirlist{3});
+        if (exist (tgz, "dir"))
+          if (tgz(1) == '/')
+            packdir = tgz;
+          else
+            packdir = fullfile (pwd (), dirlist{3});
+          endif
         else
-          packdir = fullfile (pwd (), dirlist{3});
+          packdir = fullfile (tmpdir, dirlist{3});
         endif
         packdirs{end+1} = packdir;
 
