--- octave-4.2.1/libgui/src/octave-qt-link.cc.orig	2017-02-22 18:01:55.000000000 +0000
+++ octave-4.2.1/libgui/src/octave-qt-link.cc	2017-03-01 08:56:20.218150443 +0000
@@ -67,6 +67,8 @@
 
 octave_qt_link::~octave_qt_link (void)
 {
+  main_thread->quit ();
+  main_thread->wait ();
   delete command_interpreter;
   delete main_thread;
 }
@@ -102,7 +104,7 @@
 
   // Could wait for a while and then timeout, but for now just
   // assume the GUI application exit will be without problems.
-  return true;
+  return false;
 }
 
 bool
--- octave-4.2.1/configure.orig	2017-02-22 18:18:36.000000000 +0000
+++ octave-4.2.1/configure	2017-04-14 11:49:04.567749125 +0100
@@ -34701,7 +34701,7 @@
     ;;
     5)
       QT_MODULES="Qt5Core Qt5Gui Qt5Network Qt5OpenGL Qt5PrintSupport"
-      octave_qscintilla_libnames="qscintilla2-qt5 qt5scintilla2"
+      octave_qscintilla_libnames="qscintilla2-qt5 qscintilla2_qt5 qt5scintilla2"
     ;;
     *)
       as_fn_error $? "Unrecognized Qt version $qt_version" "$LINENO" 5
