--- octave-3.8.2/doc/interpreter/Makefile.in.orig	2014-08-28 23:20:14.127115489 +0100
+++ octave-3.8.2/doc/interpreter/Makefile.in	2014-08-28 23:27:27.455146274 +0100
@@ -1855,7 +1855,7 @@
 null = 
 TEXINFO_TEX = ../texinfo.tex
 TEXINPUTS := "$(PATH_SEPARATOR)$(srcdir)$(PATH_SEPARATOR)$(TEXINPUTS)$(PATH_SEPARATOR)"
-TEXMFCNF := "..$(PATH_SEPARATOR)$(srcdir)/..$(PATH_SEPARATOR)$(TEXMFCNF)$(PATH_SEPARATOR)"
+TEXMFCNF := ..$(PATH_SEPARATOR)$(abs_srcdir)/..$(PATH_SEPARATOR)$(TEXMFCNF)$(PATH_SEPARATOR)
 dist_man_MANS = \
   mkoctfile.1 \
   octave-cli.1 \
--- octave-3.8.2/configure.orig	2014-08-06 20:06:36.000000000 +0100
+++ octave-3.8.2/configure	2014-08-28 20:32:27.023339075 +0100
@@ -70047,7 +70047,7 @@
 
   if test $build_gui = yes; then
     if test -n "$ac_tool_prefix"; then
-  for ac_prog in moc-qt5 moc-qt4 moc
+  for ac_prog in moc-qt4 moc
   do
     # Extract the first word of "$ac_tool_prefix$ac_prog", so it can be a program name with args.
 set dummy $ac_tool_prefix$ac_prog; ac_word=$2
@@ -70091,7 +70091,7 @@
 fi
 if test -z "$MOC"; then
   ac_ct_MOC=$MOC
-  for ac_prog in moc-qt5 moc-qt4 moc
+  for ac_prog in moc-qt4 moc
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog; ac_word=$2
@@ -70147,7 +70147,7 @@
 fi
 
     if test -n "$ac_tool_prefix"; then
-  for ac_prog in uic-qt5 uic-qt4 uic
+  for ac_prog in uic-qt4 uic
   do
     # Extract the first word of "$ac_tool_prefix$ac_prog", so it can be a program name with args.
 set dummy $ac_tool_prefix$ac_prog; ac_word=$2
@@ -70191,7 +70191,7 @@
 fi
 if test -z "$UIC"; then
   ac_ct_UIC=$UIC
-  for ac_prog in uic-qt5 uic-qt4 uic
+  for ac_prog in uic-qt4 uic
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog; ac_word=$2
@@ -70247,7 +70247,7 @@
 fi
 
     if test -n "$ac_tool_prefix"; then
-  for ac_prog in rcc-qt5 rcc-qt4 rcc
+  for ac_prog in rcc-qt4 rcc
   do
     # Extract the first word of "$ac_tool_prefix$ac_prog", so it can be a program name with args.
 set dummy $ac_tool_prefix$ac_prog; ac_word=$2
@@ -70291,7 +70291,7 @@
 fi
 if test -z "$RCC"; then
   ac_ct_RCC=$RCC
-  for ac_prog in rcc-qt5 rcc-qt4 rcc
+  for ac_prog in rcc-qt4 rcc
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog; ac_word=$2
@@ -70347,7 +70347,7 @@
 fi
 
     if test -n "$ac_tool_prefix"; then
-  for ac_prog in lrelease-qt5 lrelease-qt4 lrelease
+  for ac_prog in lrelease-qt4 lrelease
   do
     # Extract the first word of "$ac_tool_prefix$ac_prog", so it can be a program name with args.
 set dummy $ac_tool_prefix$ac_prog; ac_word=$2
@@ -70391,7 +70391,7 @@
 fi
 if test -z "$LRELEASE"; then
   ac_ct_LRELEASE=$LRELEASE
-  for ac_prog in lrelease-qt5 lrelease-qt4 lrelease
+  for ac_prog in lrelease-qt4 lrelease
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog; ac_word=$2
