--- openexr-1.6.1/exrenvmap/main.cpp.orig	2007-04-25 02:07:51.000000000 +0100
+++ openexr-1.6.1/exrenvmap/main.cpp	2011-06-27 11:21:52.625737548 +0100
@@ -45,6 +45,7 @@
 #include <iostream>
 #include <exception>
 #include <stdlib.h>
+#include <cstring>
 
 using namespace Imf;
 using namespace std;
--- openexr-1.6.1/exrmaketiled/main.cpp.orig	2007-04-25 02:08:45.000000000 +0100
+++ openexr-1.6.1/exrmaketiled/main.cpp	2011-06-27 11:21:12.145737690 +0100
@@ -46,6 +46,7 @@
 #include <exception>
 #include <string>
 #include <stdlib.h>
+#include <cstring>
 
 using namespace Imf;
 using namespace std;
