--- opencv-1.0.0/otherlibs/highgui/cvcap_ffmpeg.cpp.orig	2006-07-24 23:27:39.000000000 +0100
+++ opencv-1.0.0/otherlibs/highgui/cvcap_ffmpeg.cpp	2008-02-21 13:32:52.000000000 +0000
@@ -39,6 +39,9 @@
 //
 //M*/
 
+#define __STDC_CONSTANT_MACROS
+#include <stdint.h>
+
 #include "_highgui.h"
 
 
@@ -781,7 +784,7 @@
 
 	if (!(mywriter->fmt->flags & AVFMT_NOFILE)) {
 		/* close the output file */
-		url_fclose(&mywriter->oc->pb);
+		url_fclose(mywriter->oc->pb);
 	}
 
 	/* free the stream */
