--- openvdb-6.2.1/CMakeLists.txt.orig	2019-10-01 19:05:29.000000000 +0100
+++ openvdb-6.2.1/CMakeLists.txt	2019-11-23 17:11:47.720203939 +0000
@@ -280,7 +280,7 @@
     cmake/OpenVDBMayaSetup.cmake
     cmake/OpenVDBUtils.cmake
   )
-  install(FILES ${OPENVDB_CMAKE_MODULES} DESTINATION lib/cmake/OpenVDB)
+  install(FILES ${OPENVDB_CMAKE_MODULES} DESTINATION lib${LIB_SUFFIX}/cmake/OpenVDB)
 endif()
 
 # Configure DCC installation if necessary
--- openvdb-6.2.1/openvdb/CMakeLists.txt.orig	2019-10-01 19:05:29.000000000 +0100
+++ openvdb-6.2.1/openvdb/CMakeLists.txt	2019-11-23 17:12:53.555200798 +0000
@@ -492,7 +492,7 @@
   install(TARGETS
     openvdb_static
     DESTINATION
-    lib
+    lib${LIB_SUFFIX}
   )
 endif()
 
@@ -507,7 +507,7 @@
     install(TARGETS
       openvdb_shared
       DESTINATION
-      lib
+      lib${LIB_SUFFIX}
     )
   endif()
 endif()
--- openvdb-6.2.1/openvdb/python/CMakeLists.txt.orig	2019-10-01 19:05:29.000000000 +0100
+++ openvdb-6.2.1/openvdb/python/CMakeLists.txt	2019-11-23 17:13:37.911198682 +0000
@@ -162,7 +162,7 @@
 
 if(NOT DEFINED PYOPENVDB_INSTALL_DIRECTORY)
   set(PYOPENVDB_INSTALL_DIRECTORY
-    lib/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}
+    lib${LIB_SUFFIX}/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}
     CACHE STRING "The directory to install the pyopenvdb.so module."
   )
 endif()
--- openvdb-6.2.1/cmake/FindIlmBase.cmake.orig	2019-10-01 19:05:29.000000000 +0100
+++ openvdb-6.2.1/cmake/FindIlmBase.cmake	2019-11-23 17:04:36.114224532 +0000
@@ -307,7 +307,6 @@
 
   set(IlmBase_INCLUDE_DIRS)
   list(APPEND IlmBase_INCLUDE_DIRS
-    ${IlmBase_INCLUDE_DIR}/../
     ${IlmBase_INCLUDE_DIR}
   )
   set(IlmBase_DEFINITIONS ${PC_IlmBase_CFLAGS_OTHER})
--- openvdb-6.2.1/cmake/FindOpenEXR.cmake.orig	2019-10-01 19:05:29.000000000 +0100
+++ openvdb-6.2.1/cmake/FindOpenEXR.cmake	2019-11-23 17:03:37.563227326 +0000
@@ -300,7 +300,6 @@
 
   set(OpenEXR_INCLUDE_DIRS)
   list(APPEND OpenEXR_INCLUDE_DIRS
-    ${OpenEXR_INCLUDE_DIR}/../
     ${OpenEXR_INCLUDE_DIR}
   )
   set(OpenEXR_DEFINITIONS ${PC_OpenEXR_CFLAGS_OTHER})
