--- openconnect-7.07/gnutls.c.orig	2016-07-26 21:06:28.231909168 +0100
+++ openconnect-7.07/gnutls.c	2016-07-26 21:09:31.384909843 +0100
@@ -2292,7 +2292,7 @@
 	default_prio = DEFAULT_PRIO ":%COMPAT";
 #else
 	if (gtls_ver(3,2,9)) {
-		default_prio = "NORMAL:-VERS-SSL3.0:%COMPAT";
+		default_prio = "NORMAL:-VERS-SSL3.0:%COMPAT:+ARCFOUR-128";
 	} else if (gtls_ver(3,0,0)) {
 		default_prio = "NORMAL:-VERS-TLS-ALL:+VERS-TLS1.0:" \
 			"%COMPAT:%DISABLE_SAFE_RENEGOTIATION:%LATEST_RECORD_VERSION" \
