--- openvrml-0.16.2/src/openvrml-gtkplug/Makefile.in.orig	2006-11-28 21:05:54.000000000 +0000
+++ openvrml-0.16.2/src/openvrml-gtkplug/Makefile.in	2006-12-10 09:57:13.000000000 +0000
@@ -264,7 +264,7 @@
         @GTK_LIBS@ \
         @GL_LIBS@ \
         -lXmu -lXt @X_PRE_LIBS@ @X_LIBS@ @X_EXTRA_LIBS@ \
-        -lboost_thread \
+        -lboost_thread@BOOST_LIB_SUFFIX@ \
         @PTHREAD_LIBS@
 
 LDADD = $(top_builddir)/src/libopenvrml-gl/libopenvrml-gl.la \
