--- opentimelineio-0.18.1/CMakeLists.txt.orig	2025-11-08 23:02:42.000000000 +0000
+++ opentimelineio-0.18.1/CMakeLists.txt	2025-12-12 15:19:12.284813223 +0000
@@ -106,7 +106,7 @@
     endif()
 
 else()
-    set(OTIO_RESOLVED_CXX_DYLIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib")
+    set(OTIO_RESOLVED_CXX_DYLIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
     message(STATUS "OTIO C++ installing to ${CMAKE_INSTALL_PREFIX}")
 endif()
 
