--- OpenColorIO-2.0.0/CMakeLists.txt.orig	2021-01-28 22:07:14.000000000 +0000
+++ OpenColorIO-2.0.0/CMakeLists.txt	2021-03-24 22:02:25.261752265 +0000
@@ -38,6 +38,7 @@
 
 enable_testing()
 
+include(GNUInstallDirs)
 
 ###############################################################################
 # Forbid in-source build.
--- OpenColorIO-2.0.0/src/OpenColorIO/CMakeLists.txt.orig	2021-01-28 22:07:14.000000000 +0000
+++ OpenColorIO-2.0.0/src/OpenColorIO/CMakeLists.txt	2021-03-24 22:03:51.526748149 +0000
@@ -295,7 +295,7 @@
 endif()
 
 install(TARGETS OpenColorIO
-	LIBRARY DESTINATION lib
+	LIBRARY DESTINATION lib${LIB_SUFFIX}
 	RUNTIME DESTINATION bin
-	ARCHIVE DESTINATION lib
+	ARCHIVE DESTINATION lib${LIB_SUFFIX}
 )
--- OpenColorIO-2.0.0/src/libutils/oiiohelpers/CMakeLists.txt.orig	2021-01-28 22:07:14.000000000 +0000
+++ OpenColorIO-2.0.0/src/libutils/oiiohelpers/CMakeLists.txt	2021-03-24 22:08:15.387735559 +0000
@@ -36,8 +36,8 @@
 )
 
 install(TARGETS oiiohelpers
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX}
 )
 
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/oiiohelpers.h
--- OpenColorIO-2.0.0/src/libutils/oglapphelpers/CMakeLists.txt.orig	2021-01-28 22:07:14.000000000 +0000
+++ OpenColorIO-2.0.0/src/libutils/oglapphelpers/CMakeLists.txt	2021-03-24 22:08:00.984736246 +0000
@@ -84,8 +84,8 @@
 endif()
 
 install(TARGETS oglapphelpers 
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX}
 )
 
 install(FILES ${INCLUDES}
