--- a/src/libopenvrml/openvrml/local/error.cpp
+++ b/src/libopenvrml/openvrml/local/error.cpp
@@ -19,6 +19,7 @@
 //
 
 # include "error.h"
+# include <boost/mpl/if.hpp>
 # include <boost/multi_index/detail/scope_guard.hpp>
 # include <boost/ref.hpp>
 
--- a/src/openvrml-xembed/browser.cpp
+++ b/src/openvrml-xembed/browser.cpp
@@ -18,6 +18,7 @@
 // with this library; if not, see <http://www.gnu.org/licenses/>.
 //
 
+# include <boost/mpl/if.hpp>
 # include <boost/multi_index/detail/scope_guard.hpp>
 // Must include before X11 headers.
 # include <boost/numeric/conversion/converter.hpp>
