--- openscad-2021.01/src/FileModule.cc.orig	2021-01-31 20:17:41.000000000 +0000
+++ openscad-2021.01/src/FileModule.cc	2024-04-18 16:42:47.186183166 +0100
@@ -65,7 +65,7 @@
 	auto ext = fs::path(path).extension().generic_string();
 
 	if (boost::iequals(ext, ".otf") || boost::iequals(ext, ".ttf")) {
-		if (fs::is_regular(path)) {
+		if (fs::is_regular_file(path)) {
 			FontCache::instance()->register_font_file(path);
 		} else {
 			LOG(message_group::Error,Location::NONE,"","Can't read font with path '%1$s'",path);
--- openscad-2021.01/CMakeLists.txt.orig	2021-01-31 20:17:41.000000000 +0000
+++ openscad-2021.01/CMakeLists.txt	2025-08-17 17:18:42.140813162 +0100
@@ -80,7 +80,7 @@
   endif()
 endif()
 
-set(BOOST_DIRECTLY_REQUIRED_LIBRARIES filesystem system regex program_options)
+set(BOOST_DIRECTLY_REQUIRED_LIBRARIES filesystem regex program_options)
 
 find_package(PkgConfig)
 
