--- opencv-3.4.1/CMakeLists.txt.orig	2018-02-23 08:38:33.000000000 +0000
+++ opencv-3.4.1/CMakeLists.txt	2018-05-05 22:34:52.221857982 +0100
@@ -439,7 +439,7 @@
     ocv_update(OPENCV_CONFIG_INSTALL_PATH  ".")
   else()
     include(GNUInstallDirs)
-    ocv_update(OPENCV_LIB_INSTALL_PATH     ${CMAKE_INSTALL_LIBDIR}${LIB_SUFFIX})
+    ocv_update(OPENCV_LIB_INSTALL_PATH     ${CMAKE_INSTALL_LIBDIR})
     ocv_update(OPENCV_3P_LIB_INSTALL_PATH  share/OpenCV/3rdparty/${OPENCV_LIB_INSTALL_PATH})
     ocv_update(OPENCV_SAMPLES_SRC_INSTALL_PATH    share/OpenCV/samples)
     ocv_update(OPENCV_JAR_INSTALL_PATH share/OpenCV/java)
@@ -448,7 +448,7 @@
     if(NOT DEFINED OPENCV_CONFIG_INSTALL_PATH)
       math(EXPR SIZEOF_VOID_P_BITS "8 * ${CMAKE_SIZEOF_VOID_P}")
       if(LIB_SUFFIX AND NOT SIZEOF_VOID_P_BITS EQUAL LIB_SUFFIX)
-        ocv_update(OPENCV_CONFIG_INSTALL_PATH ${CMAKE_INSTALL_LIBDIR}${LIB_SUFFIX}/cmake/opencv)
+        ocv_update(OPENCV_CONFIG_INSTALL_PATH ${CMAKE_INSTALL_LIBDIR}/cmake/opencv)
       else()
         ocv_update(OPENCV_CONFIG_INSTALL_PATH share/OpenCV)
       endif()
