--- openbabel-3.1.1/external/maeparser-v1.2.3/maeparser/CMakeLists.txt.orig	2020-03-16 20:26:47.000000000 +0000
+++ openbabel-3.1.1/external/maeparser-v1.2.3/maeparser/CMakeLists.txt	2020-11-01 14:27:49.899241385 +0000
@@ -65,13 +65,13 @@
 
 install(TARGETS maeparser
     EXPORT maeparser-targets
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX}
     RUNTIME DESTINATION bin)
 
 INSTALL(EXPORT maeparser-targets
     FILE ${PROJECT_NAME}-config.cmake
-    DESTINATION lib/cmake)
+    DESTINATION lib${LIB_SUFFIX}/cmake)
 
 file(GLOB mae_headers "*.hpp")
 install(FILES ${mae_headers} DESTINATION include/maeparser)
--- openbabel-3.1.1/external/coordgen-master/coordgen/CMakeLists.txt.orig	2020-05-04 17:00:07.000000000 +0100
+++ openbabel-3.1.1/external/coordgen-master/coordgen/CMakeLists.txt	2020-11-01 14:27:33.608242163 +0000
@@ -88,9 +88,9 @@
 # Install configuration
 install(TARGETS coordgen
     EXPORT coordgen-targets
-    ARCHIVE DESTINATION lib
+    ARCHIVE DESTINATION lib${LIB_SUFFIX}
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib)
+    LIBRARY DESTINATION lib${LIB_SUFFIX})
 
 install(FILES
     CoordgenConfig.hpp
@@ -117,7 +117,7 @@
 
 install(EXPORT coordgen-targets
     FILE ${PROJECT_NAME}-config.cmake
-    DESTINATION lib/cmake)
+    DESTINATION lib${LIB_SUFFIX}/cmake)
 
 # Example
 if(COORDGEN_BUILD_EXAMPLE)
