--- OpenSceneGraph-2.8.3/src/osgPlugins/ffmpeg/ReaderWriterFFmpeg.cpp.orig	2010-03-19 09:04:40.000000000 +0000
+++ OpenSceneGraph-2.8.3/src/osgPlugins/ffmpeg/ReaderWriterFFmpeg.cpp	2010-06-20 09:36:16.000000000 +0100
@@ -11,6 +11,7 @@
  * OpenSceneGraph Public License for more details.
 */
 
+#define __STDC_CONSTANT_MACROS
 #include <osgDB/Registry>
 #include <osgDB/FileNameUtils>
 #include <osgDB/FileUtils>
--- OpenSceneGraph-2.8.3/src/osgPlugins/ffmpeg/FFmpegImageStream.cpp.orig	2010-03-19 09:14:18.000000000 +0000
+++ OpenSceneGraph-2.8.3/src/osgPlugins/ffmpeg/FFmpegImageStream.cpp	2010-06-20 09:32:30.000000000 +0100
@@ -1,4 +1,5 @@
 
+#define __STDC_CONSTANT_MACROS
 #include "FFmpegImageStream.hpp"
 #include "FFmpegAudioStream.hpp"
 
--- OpenSceneGraph-2.8.3/src/osgPlugins/ffmpeg/FFmpegAudioStream.cpp.orig	2010-03-19 08:48:34.000000000 +0000
+++ OpenSceneGraph-2.8.3/src/osgPlugins/ffmpeg/FFmpegAudioStream.cpp	2010-06-20 09:33:36.000000000 +0100
@@ -1,4 +1,5 @@
 
+#define __STDC_CONSTANT_MACROS
 #include "FFmpegAudioStream.hpp"
 
 #include <OpenThreads/ScopedLock>
