Index: b/OpenGTL/GTLCore/VirtualMachine_p.cpp
===================================================================
--- a/OpenGTL/GTLCore/VirtualMachine_p.cpp
+++ b/OpenGTL/GTLCore/VirtualMachine_p.cpp
@@ -61,8 +61,6 @@ VirtualMachine::VirtualMachine() : d(new
   
   llvm::install_fatal_error_handler(gtl_llvm_error_handler);
   
-  llvm::llvm_start_multithreaded();
-
   std::string errorMessage;
   llvm::LLVMContext& dummyContext = llvm::getGlobalContext(); // TODO someday (llvm 2.7 ?) use own context
   llvm::InitializeNativeTarget();
@@ -93,7 +91,6 @@ VirtualMachine::VirtualMachine() : d(new
 
 VirtualMachine::~VirtualMachine()
 {
-  llvm::llvm_stop_multithreaded();
 #ifndef LLVM_27_OR_28
   delete d->executionEngine;
 #endif
