--- oiio-Release-1.8.17/src/python/CMakeLists.txt.orig	2018-12-01 16:42:08.000000000 +0000
+++ oiio-Release-1.8.17/src/python/CMakeLists.txt	2018-12-13 22:02:00.148758907 +0000
@@ -17,7 +17,7 @@
     # Finding the python component for boost is a little tricky, since it has
     # different names on different systems. Try the most common ones.
     foreach (_py_lib python-${PYTHON_VERSION_FOUND} python
-             python${PYTHON_VERSION_MAJOR}
+             python${PYTHON_VERSION_MAJOR} python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}
              python-py${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR})
         find_package (Boost QUIET COMPONENTS ${_py_lib})
         string (TOUPPER ${_py_lib} _py_lib_name)
