--- OpenColorIO-1.1.0/src/core/CMakeLists.txt.orig	2018-01-05 01:38:27.000000000 +0000
+++ OpenColorIO-1.1.0/src/core/CMakeLists.txt	2018-12-13 17:57:06.109940329 +0000
@@ -24,7 +24,7 @@
         set(EXTERNAL_COMPILE_FLAGS "${EXTERNAL_COMPILE_FLAGS} /WX")
     endif()
 else()
-    set(EXTERNAL_COMPILE_FLAGS "${EXTERNAL_COMPILE_FLAGS} -Werror")
+    set(EXTERNAL_COMPILE_FLAGS "${EXTERNAL_COMPILE_FLAGS}")
 endif()
 
 # SHARED
--- OpenColorIO-1.1.0/src/pyglue/PyAllocationTransform.cpp.orig	2018-01-05 01:38:27.000000000 +0000
+++ OpenColorIO-1.1.0/src/pyglue/PyAllocationTransform.cpp	2018-12-13 17:58:14.732937055 +0000
@@ -53,7 +53,7 @@
         ///
         
         int PyOCIO_AllocationTransform_init(PyOCIO_Transform * self, PyObject * args, PyObject * kwds);
-        PyObject * PyOCIO_AllocationTransform_equals(PyObject * self,  PyObject * args);
+//        PyObject * PyOCIO_AllocationTransform_equals(PyObject * self,  PyObject * args);
         PyObject * PyOCIO_AllocationTransform_getAllocation(PyObject * self);
         PyObject * PyOCIO_AllocationTransform_setAllocation(PyObject * self,  PyObject * args);
         PyObject * PyOCIO_AllocationTransform_getNumVars(PyObject * self);
--- OpenColorIO-1.1.0/src/apps/ociodisplay/main.cpp.orig	2018-01-05 01:38:27.000000000 +0000
+++ OpenColorIO-1.1.0/src/apps/ociodisplay/main.cpp	2019-01-24 22:14:25.657822481 +0000
@@ -55,7 +55,7 @@
 #else
 #include <GL/glew.h>
 #include <GL/gl.h>
-#include <GL/glext.h>
+//#include <GL/glext.h>
 #include <GL/glut.h>
 #endif
 
