--- OpenSubdiv-3_4_0.x/CMakeLists.txt.orig	2019-06-28 21:55:48.000000000 +0100
+++ OpenSubdiv-3_4_0.x/CMakeLists.txt	2019-11-23 17:21:56.208174907 +0000
@@ -71,7 +71,7 @@
 endif()
 
 if (NOT DEFINED CMAKE_LIBDIR_BASE)
-    set( CMAKE_LIBDIR_BASE lib )
+    set( CMAKE_LIBDIR_BASE lib${LIB_SUFFIX} )
 endif()
 
 if (NOT DEFINED CMAKE_FRAMEWORKDIR_BASE)
@@ -137,7 +137,7 @@
     # sequences of ':' that this command causes. The consequence is that examples
     # built and installed using icc will not have an rpath pointing to the built
     # OSD library which they depend on and will have to set LD_LIBRARY_PATH instead.
-    list(APPEND CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+    list(APPEND CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
 endif()
 
 # add the automatically determined parts of the RPATH
