--- openswan-3.0.0/Makefile.common.orig	2021-01-22 18:25:50.000000000 +0000
+++ openswan-3.0.0/Makefile.common	2021-02-12 22:27:38.485877857 +0000
@@ -2,7 +2,7 @@
 # some files have issues with cpp, and need -Wno-error=cpp, but only for
 # newer versions of gcc.
 # Check if GCC's version is recent enough
-GCC_VERSION_GE_46:=$(shell test `${CC} -dumpversion | sed 's/^\([0-9]\)\.\([0-9]\).*/\1\2/'` -ge 46 && echo true || echo false)
+GCC_VERSION_GE_46:=$(shell test `${CC} -dumpversion | sed 's/^\([0-9]*\)\.\([0-9]*\).*/\1\2/'` -ge 46 && echo true || echo false)
 ifeq ($(GCC_VERSION_GE_46),true)
 GCC_CPP_FLAG=-Wno-error=cpp
 endif
