--- OpenSceneGraph-3.6.5/packaging/ld.so.conf.d/openscenegraph.conf.in.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/packaging/ld.so.conf.d/openscenegraph.conf.in	2020-01-31 22:27:10.370267751 +0000
@@ -1,2 +1,2 @@
 # openscenegraph library configuration
-${CMAKE_INSTALL_PREFIX}/lib${LIB_POSTFIX}
+${CMAKE_INSTALL_PREFIX}/${OSG_INSTALL_LIBDIR}
--- OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osg.pc.in.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osg.pc.in	2020-01-31 22:28:11.641264828 +0000
@@ -2,7 +2,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib@LIB_POSTFIX@
+libdir=${exec_prefix}/@OSG_INSTALL_LIBDIR@
 includedir=${prefix}/include
 
 Name: openscenegraph-osg
--- OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgWidget.pc.in.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgWidget.pc.in	2020-01-31 22:28:11.642264828 +0000
@@ -2,7 +2,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib@LIB_POSTFIX@
+libdir=${exec_prefix}/@OSG_INSTALL_LIBDIR@
 includedir=${prefix}/include
 
 Name: openscenegraph-osgWidget
--- OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgUtil.pc.in.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgUtil.pc.in	2020-01-31 22:28:11.642264828 +0000
@@ -2,7 +2,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib@LIB_POSTFIX@
+libdir=${exec_prefix}/@OSG_INSTALL_LIBDIR@
 includedir=${prefix}/include
 
 Name: openscenegraph-osgUtil
--- OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgViewer.pc.in.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgViewer.pc.in	2020-01-31 22:28:11.642264828 +0000
@@ -2,7 +2,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib@LIB_POSTFIX@
+libdir=${exec_prefix}/@OSG_INSTALL_LIBDIR@
 includedir=${prefix}/include
 
 Name: openscenegraph-osgViewer
--- OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgManipulator.pc.in.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgManipulator.pc.in	2020-01-31 22:28:11.641264828 +0000
@@ -2,7 +2,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib@LIB_POSTFIX@
+libdir=${exec_prefix}/@OSG_INSTALL_LIBDIR@
 includedir=${prefix}/include
 
 Name: openscenegraph-osgManipulator
--- OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph.pc.in.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph.pc.in	2020-01-31 22:28:11.642264828 +0000
@@ -2,7 +2,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib@LIB_POSTFIX@
+libdir=${exec_prefix}/@OSG_INSTALL_LIBDIR@
 includedir=${prefix}/include
 
 Name: openscenegraph
--- OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgShadow.pc.in.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgShadow.pc.in	2020-01-31 22:28:11.641264828 +0000
@@ -2,7 +2,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib@LIB_POSTFIX@
+libdir=${exec_prefix}/@OSG_INSTALL_LIBDIR@
 includedir=${prefix}/include
 
 Name: openscenegraph-osgShadow
--- OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgDB.pc.in.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgDB.pc.in	2020-01-31 22:28:11.641264828 +0000
@@ -2,7 +2,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib@LIB_POSTFIX@
+libdir=${exec_prefix}/@OSG_INSTALL_LIBDIR@
 includedir=${prefix}/include
 
 Name: openscenegraph-osgDB
--- OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgVolume.pc.in.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgVolume.pc.in	2020-01-31 22:28:11.642264828 +0000
@@ -2,7 +2,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib@LIB_POSTFIX@
+libdir=${exec_prefix}/@OSG_INSTALL_LIBDIR@
 includedir=${prefix}/include
 
 Name: openscenegraph-osgVolume
--- OpenSceneGraph-3.6.5/packaging/pkgconfig/openthreads.pc.in.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/packaging/pkgconfig/openthreads.pc.in	2020-01-31 22:28:11.642264828 +0000
@@ -2,7 +2,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib@LIB_POSTFIX@
+libdir=${exec_prefix}/@OSG_INSTALL_LIBDIR@
 includedir=${prefix}/include
 
 Name: openthreads
--- OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgAnimation.pc.in.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgAnimation.pc.in	2020-01-31 22:28:11.641264828 +0000
@@ -2,7 +2,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib@LIB_POSTFIX@
+libdir=${exec_prefix}/@OSG_INSTALL_LIBDIR@
 includedir=${prefix}/include
 
 Name: openscenegraph-osgAnimation
--- OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgParticle.pc.in.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgParticle.pc.in	2020-01-31 22:28:11.641264828 +0000
@@ -2,7 +2,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib@LIB_POSTFIX@
+libdir=${exec_prefix}/@OSG_INSTALL_LIBDIR@
 includedir=${prefix}/include
 
 Name: openscenegraph-osgParticle
--- OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgSim.pc.in.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgSim.pc.in	2020-01-31 22:28:11.641264828 +0000
@@ -2,7 +2,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib@LIB_POSTFIX@
+libdir=${exec_prefix}/@OSG_INSTALL_LIBDIR@
 includedir=${prefix}/include
 
 Name: openscenegraph-osgSim
--- OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgFX.pc.in.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgFX.pc.in	2020-01-31 22:28:11.641264828 +0000
@@ -2,7 +2,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib@LIB_POSTFIX@
+libdir=${exec_prefix}/@OSG_INSTALL_LIBDIR@
 includedir=${prefix}/include
 
 Name: openscenegraph-osgFX
--- OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgGA.pc.in.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgGA.pc.in	2020-01-31 22:28:11.641264828 +0000
@@ -2,7 +2,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib@LIB_POSTFIX@
+libdir=${exec_prefix}/@OSG_INSTALL_LIBDIR@
 includedir=${prefix}/include
 
 Name: openscenegraph-osgGA
--- OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgTerrain.pc.in.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgTerrain.pc.in	2020-01-31 22:28:11.642264828 +0000
@@ -2,7 +2,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib@LIB_POSTFIX@
+libdir=${exec_prefix}/@OSG_INSTALL_LIBDIR@
 includedir=${prefix}/include
 
 Name: openscenegraph-osgTerrain
--- OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgText.pc.in.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/packaging/pkgconfig/openscenegraph-osgText.pc.in	2020-01-31 22:28:11.642264828 +0000
@@ -2,7 +2,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib@LIB_POSTFIX@
+libdir=${exec_prefix}/@OSG_INSTALL_LIBDIR@
 includedir=${prefix}/include
 
 Name: openscenegraph-osgText
--- OpenSceneGraph-3.6.5/src/OpenThreads/CMakeLists.txt.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/src/OpenThreads/CMakeLists.txt	2020-01-31 22:19:40.411289220 +0000
@@ -113,4 +113,4 @@
     @ONLY
     )
 
-INSTALL(FILES ${PROJECT_BINARY_DIR}/packaging/pkgconfig/openthreads.pc DESTINATION lib${LIB_POSTFIX}/pkgconfig  COMPONENT libopenthreads-dev)
+INSTALL(FILES ${PROJECT_BINARY_DIR}/packaging/pkgconfig/openthreads.pc DESTINATION ${OSG_INSTALL_LIBDIR}/pkgconfig  COMPONENT libopenthreads-dev)
--- OpenSceneGraph-3.6.5/src/OpenThreads/pthreads/CMakeLists.txt.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/src/OpenThreads/pthreads/CMakeLists.txt	2020-01-31 22:20:27.227286986 +0000
@@ -157,8 +157,8 @@
 
     INSTALL(
         TARGETS OpenThreads
-        ARCHIVE DESTINATION lib${LIB_POSTFIX} COMPONENT libopenthreads-dev
-        LIBRARY DESTINATION lib${LIB_POSTFIX} COMPONENT libopenthreads
+        ARCHIVE DESTINATION ${OSG_INSTALL_LIBDIR} COMPONENT libopenthreads-dev
+        LIBRARY DESTINATION ${OSG_INSTALL_LIBDIR} COMPONENT libopenthreads
         RUNTIME DESTINATION bin COMPONENT libopenthreads
     )
 
--- OpenSceneGraph-3.6.5/CMakeModules/ModuleInstall.cmake.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/CMakeModules/ModuleInstall.cmake	2020-01-31 22:14:16.005304698 +0000
@@ -28,7 +28,7 @@
 INSTALL(
     TARGETS ${LIB_NAME}
     RUNTIME DESTINATION ${INSTALL_BINDIR} COMPONENT libopenscenegraph
-    LIBRARY DESTINATION ${INSTALL_LIBDIR} COMPONENT libopenscenegraph
+    LIBRARY DESTINATION ${OSG_INSTALL_LIBDIR} COMPONENT libopenscenegraph
     ARCHIVE DESTINATION ${INSTALL_ARCHIVEDIR} COMPONENT libopenscenegraph-dev    
 )
 
--- OpenSceneGraph-3.6.5/CMakeModules/OsgMacroUtils.cmake.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/CMakeModules/OsgMacroUtils.cmake	2020-01-31 22:26:33.039269532 +0000
@@ -355,8 +355,8 @@
     ELSE(WIN32)
         INSTALL(TARGETS ${TARGET_TARGETNAME}
             RUNTIME DESTINATION bin COMPONENT ${PACKAGE_COMPONENT}
-            ARCHIVE DESTINATION lib${LIB_POSTFIX}/${OSG_PLUGINS} COMPONENT libopenscenegraph-dev
-            LIBRARY DESTINATION lib${LIB_POSTFIX}/${OSG_PLUGINS} COMPONENT ${PACKAGE_COMPONENT})
+            ARCHIVE DESTINATION ${OSG_INSTALL_LIBDIR}/${OSG_PLUGINS} COMPONENT libopenscenegraph-dev
+            LIBRARY DESTINATION ${OSG_INSTALL_LIBDIR}/${OSG_PLUGINS} COMPONENT ${PACKAGE_COMPONENT})
     ENDIF(WIN32)
 ENDMACRO(SETUP_PLUGIN)
 
--- OpenSceneGraph-3.6.5/src/osgPlugins/CMakeLists.txt.orig	2020-01-31 11:03:07.000000000 +0000
+++ OpenSceneGraph-3.6.5/src/osgPlugins/CMakeLists.txt	2022-11-20 14:54:23.540648945 +0000
@@ -209,9 +209,9 @@
     ADD_PLUGIN_DIRECTORY(txp)
 ENDIF()
 
-IF(FFMPEG_FOUND AND OSG_CPP_EXCEPTIONS_AVAILABLE)
-    ADD_PLUGIN_DIRECTORY(ffmpeg)
-ENDIF()
+#IF(FFMPEG_FOUND AND OSG_CPP_EXCEPTIONS_AVAILABLE)
+#    ADD_PLUGIN_DIRECTORY(ffmpeg)
+#ENDIF()
 
 IF(GSTREAMER_FOUND AND GLIB_FOUND)
     ADD_PLUGIN_DIRECTORY(gstreamer)
