--- openjpeg-v1_3/libopenjpeg/CMakeLists.txt.orig	2007-09-25 15:54:49.000000000 +0100
+++ openjpeg-v1_3/libopenjpeg/CMakeLists.txt	2011-02-26 21:09:55.336000258 +0000
@@ -37,7 +37,7 @@
 
 # Install library
 INSTALL(TARGETS ${OPENJPEG_LIBRARY_NAME}
-	DESTINATION lib)
+	DESTINATION lib${LIB_SUFFIX})
 
 # Install includes files
 INSTALL(FILES openjpeg.h
--- openjpeg-v1_3/codec/convert.c.orig	2007-10-12 16:04:34.000000000 +0100
+++ openjpeg-v1_3/codec/convert.c	2008-10-11 07:21:09.000000000 +0100
@@ -33,7 +33,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include "openjpeg.h"
-#include "../libs/libtiff/tiffio.h"
+#include <tiffio.h>
 #include "convert.h"
 
 /*
--- openjpeg-v1_3/Makefile.orig	2007-12-21 10:39:41.000000000 +0000
+++ openjpeg-v1_3/Makefile	2008-10-11 07:07:03.000000000 +0100
@@ -58,12 +58,11 @@
 
 install: OpenJPEG
 	install -d '$(DESTDIR)$(INSTALL_LIBDIR)' '$(DESTDIR)$(INSTALL_INCLUDE)'
-	install -m 644 -o root -g root $(STATICLIB) '$(DESTDIR)$(INSTALL_LIBDIR)'
+	install -m 644 $(STATICLIB) '$(DESTDIR)$(INSTALL_LIBDIR)'
 	ranlib '$(DESTDIR)$(INSTALL_LIBDIR)/$(STATICLIB)'
-	install -m 755 -o root -g root $(SHAREDLIB) '$(DESTDIR)$(INSTALL_LIBDIR)'
+	install -m 755 $(SHAREDLIB) '$(DESTDIR)$(INSTALL_LIBDIR)'
 	ln -sf $(SHAREDLIB) '$(DESTDIR)$(INSTALL_LIBDIR)/$(LIBNAME)'
-	install -m 644 -o root -g root libopenjpeg/openjpeg.h '$(DESTDIR)$(INSTALL_INCLUDE)'
-	-ldconfig
+	install -m 644 libopenjpeg/openjpeg.h '$(DESTDIR)$(INSTALL_INCLUDE)'
 
 clean:
 	rm -rf core dist/ u2dtmp* $(MODULES) $(STATICLIB) $(SHAREDLIB) $(LIBNAME)
