--- openvrml-0.17.9/src/libopenvrml/openvrml/node_impl_util.h.orig	2008-10-08 07:35:05.000000000 +0100
+++ openvrml-0.17.9/src/libopenvrml/openvrml/node_impl_util.h	2008-10-12 08:23:34.000000000 +0100
@@ -1029,9 +1029,9 @@
 
         template <typename BindableNode>
         class bound_node_stack : private std::stack<BindableNode *> {
-            BOOST_CLASS_REQUIRE(BindableNode,
-                                openvrml::node_impl_util,
-                                BindableNodeConcept);
+//            BOOST_CLASS_REQUIRE(BindableNode,
+//                                openvrml::node_impl_util,
+//                                BindableNodeConcept);
 
             typedef typename std::stack<BindableNode *>::container_type
                 container_type;
