--- openswan-2.6.16/lib/liblwres/context.c.orig	2008-08-18 18:07:48.000000000 +0100
+++ openswan-2.6.16/lib/liblwres/context.c	2008-09-13 22:48:26.000000000 +0100
@@ -370,7 +370,7 @@
  again:
 	OSW_FD_ZERO(&readfds);
 	OSW_FD_SET(ctx->sock, &readfds);
-	ret2 = osw_select(ctx->sock + 1, &readfds, NULL, NULL, &timeout);
+	ret2 = osw_select(ctx->sock + 1, (void *) &readfds, NULL, NULL, &timeout);
 	
 	/*
 	 * What happened with select?
--- openswan-2.6.16/lib/liblwres/getrrset2.c.orig	2008-08-18 18:07:48.000000000 +0100
+++ openswan-2.6.16/lib/liblwres/getrrset2.c	2008-09-13 22:48:08.000000000 +0100
@@ -67,7 +67,7 @@
 
 	OSW_FD_ZERO(&readfds);
 	OSW_FD_SET(sock, &readfds);
-	ret2 = osw_select(sock + 1, &readfds, NULL, NULL, &timeout);
+	ret2 = osw_select(sock + 1, (void *) &readfds, NULL, NULL, &timeout);
 	
 	/*
 	 * What happened with select?
