--- openmovieeditor-0.0.20080523/src/AudioFileFfmpeg.cxx.orig	2008-03-26 08:16:58.000000000 +0000
+++ openmovieeditor-0.0.20080523/src/AudioFileFfmpeg.cxx	2008-10-06 07:54:35.000000000 +0100
@@ -132,7 +132,7 @@
 			uint8_t *ptr = m_packet.data;
 			int data_size;
 			while ( ptr != NULL && ret >= 0 && m_packet.stream_index == m_audioStream && len > 0 ) {
-				ret = avcodec_decode_audio( m_codecContext, m_tmpBuffer, &data_size, ptr, len );
+				ret = avcodec_decode_audio2( m_codecContext, m_tmpBuffer, &data_size, ptr, len );
 				if ( ret < 0 ) {
 					ret = 0;
 					break;
