--- OpenColorIO-2.3.0/share/cmake/modules/Findyaml-cpp.cmake.orig	2023-09-01 02:38:17.000000000 +0100
+++ OpenColorIO-2.3.0/share/cmake/modules/Findyaml-cpp.cmake	2023-09-01 20:13:46.889620989 +0100
@@ -50,7 +50,8 @@
     endif()
 
     if(yaml-cpp_FOUND)
-        get_target_property(yaml-cpp_LIBRARY yaml-cpp LOCATION)
+        get_target_property(yaml-cpp_LIBRARY yaml-cpp::yaml-cpp LOCATION)
+        set(yaml-cpp_INCLUDE_DIR ${YAML_CPP_INCLUDE_DIR})
     else()
 
         # As yaml-cpp-config.cmake search fails, search an installed library
@@ -146,6 +147,7 @@
 ### Configure target ###
 
 if(_yaml-cpp_TARGET_CREATE)
+    message("IMPORTED_LOCATION ${yaml-cpp_LIBRARY} INTERFACE_INCLUDE_DIRECTORIES ${yaml-cpp_INCLUDE_DIR}")
     set_target_properties(yaml-cpp PROPERTIES
         IMPORTED_LOCATION ${yaml-cpp_LIBRARY}
         INTERFACE_INCLUDE_DIRECTORIES ${yaml-cpp_INCLUDE_DIR}
