--- opencore-amr-0.1.5/amrwb/opencore-amrwb.pc.in.orig	2016-03-02 11:14:43.000000000 +0000
+++ opencore-amr-0.1.5/amrwb/opencore-amrwb.pc.in	2019-07-16 20:34:12.934132581 +0100
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-includedir=@includedir@
+includedir=@includedir@/opencore-amrwb
 
 Name: OpenCORE AMR-WB
 Description: Adaptive Multi-Rate Wideband speech codec library
--- opencore-amr-0.1.5/amrnb/opencore-amrnb.pc.in.orig	2016-03-02 11:14:43.000000000 +0000
+++ opencore-amr-0.1.5/amrnb/opencore-amrnb.pc.in	2019-07-16 20:34:57.544130453 +0100
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-includedir=@includedir@
+includedir=@includedir@/opencore-amrnb
 
 Name: OpenCORE AMR-NB
 Description: Adaptive Multi-Rate speech codec library
