--- openmovieeditor-0.0.20090105/src/VideoViewGL.H.orig	2008-09-25 13:07:05.000000000 +0100
+++ openmovieeditor-0.0.20090105/src/VideoViewGL.H	2010-06-16 06:46:56.000000000 +0100
@@ -25,6 +25,7 @@
 
 #include "IVideoWriter.H"
 #include <sstream>
+#include <sys/types.h>
 
 namespace nle
 {
--- openmovieeditor-0.0.20090105/src/MediaBrowser.cxx.orig	2008-12-22 20:10:21.000000000 +0000
+++ openmovieeditor-0.0.20090105/src/MediaBrowser.cxx	2010-06-16 06:52:22.000000000 +0100
@@ -20,6 +20,7 @@
 #include <iostream>
 #include <cstring>
 #include <cstdlib>
+#include <cstdio>
 
 #include <FL/Fl.H>
 #include <FL/filename.H>
--- openmovieeditor-0.0.20090105/src/fl_font_browser.cpp.orig	2008-03-26 08:17:01.000000000 +0000
+++ openmovieeditor-0.0.20090105/src/fl_font_browser.cpp	2010-06-16 06:53:25.000000000 +0100
@@ -30,6 +30,7 @@
 #include "fl_font_browser.h"
 #include <cstdlib>
 #include <cstring>
+#include <cstdio>
 using namespace std;
 
 void Fl_Font_Browser::callback(void (*cb)(Fl_Widget *, void *), void *d ) 
--- openmovieeditor-0.0.20090105/src/WaveForm.H.orig	2009-01-01 16:07:25.000000000 +0000
+++ openmovieeditor-0.0.20090105/src/WaveForm.H	2010-06-16 06:50:35.000000000 +0100
@@ -21,6 +21,7 @@
 #include "JobManager.H"
 
 #include <string>
+#include <sys/types.h>
 
 namespace nle
 {
--- openmovieeditor-0.0.20090105/src/AddCommand.cxx.orig	2008-10-03 17:58:49.000000000 +0100
+++ openmovieeditor-0.0.20090105/src/AddCommand.cxx	2010-06-16 06:52:40.000000000 +0100
@@ -17,7 +17,7 @@
  *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
  */
 
-
+#include <cstdio>
 #include "AddCommand.H"
 #include "timeline/Clip.H"
 #include "Timeline.H"
