Index: b/OpenCTL/OpenCTL/Program.cpp
===================================================================
--- a/OpenCTL/OpenCTL/Program.cpp
+++ b/OpenCTL/OpenCTL/Program.cpp
@@ -28,7 +28,7 @@
 // Passes
 #include <llvm/PassManager.h>
 #include <llvm/Analysis/LoopPass.h>
-#include <llvm/Analysis/Verifier.h>
+#include <llvm/IR/Verifier.h>
 #include <llvm/IR/GlobalVariable.h>
 
 // GTLCore
Index: b/OpenGTL/GTLCore/Debug.cpp
===================================================================
--- a/OpenGTL/GTLCore/Debug.cpp
+++ b/OpenGTL/GTLCore/Debug.cpp
@@ -25,6 +25,7 @@
 #include <llvm/IR/Module.h>
 #include <llvm/IR/Value.h>
 #include <llvm/IR/Type.h>
+#include <llvm/Support/FileSystem.h>
 #include <llvm/Support/Path.h>
 #include <llvm/Support/raw_ostream.h>
 
Index: b/OpenGTL/GTLCore/ModuleData_p.cpp
===================================================================
--- a/OpenGTL/GTLCore/ModuleData_p.cpp
+++ b/OpenGTL/GTLCore/ModuleData_p.cpp
@@ -24,9 +24,9 @@
 #include <llvm/IR/Module.h>
 #include <llvm/PassManager.h>
 #include <llvm/Target/TargetMachine.h>
-#include <llvm/Analysis/Verifier.h>
+#include <llvm/IR/Verifier.h>
 #include <llvm/Transforms/Utils/Cloning.h>
-#include <llvm/Linker.h>
+#include <llvm/Linker/Linker.h>
 #ifdef LLVM_27_OR_28
 #include <llvm/System/Path.h>
 #else
