--- oregano-0.84.43/src/engines/ngspice.c.orig	2021-05-12 20:18:09.581787032 +0100
+++ oregano-0.84.43/src/engines/ngspice.c	2021-05-12 20:15:07.459795722 +0100
@@ -277,7 +277,7 @@
 		                        sim_settings_get_ac_npoints (output.settings),
 		                        sim_settings_get_ac_start (output.settings),
 		                        sim_settings_get_ac_stop (output.settings));
-	                g_string_append_printf (buffer, ".print ac %s\n",
+	                g_string_append_printf (buffer, ".print ac VM(%s)\n",
 	                                        sim_settings_get_ac_vout (output.settings));
 		}
 	}
--- oregano-0.84.43/src/sim-settings-gui.c.orig	2019-11-27 20:39:11.000000000 +0000
+++ oregano-0.84.43/src/sim-settings-gui.c	2021-05-12 20:11:51.277805082 +0100
@@ -461,7 +461,7 @@
 	s->ac_vout = NULL;
 	tmp = gtk_combo_box_text_get_active_text (GTK_COMBO_BOX_TEXT (s_gui->w_ac_vout));
 	if (tmp) {
-		node_ids = g_strsplit (tmp, "V(", 0);
+		node_ids = g_strsplit (tmp, "VM(", 0);
 		tmp = g_strdup (node_ids[1]);
 		g_strfreev (node_ids);
 		if (tmp) {
