--- oyranos-0.9.6/src/examples/image_display/Oy_Fl_Shader_Box.h.orig	2016-12-09 10:36:53.000000000 +0000
+++ oyranos-0.9.6/src/examples/image_display/Oy_Fl_Shader_Box.h	2018-12-21 10:48:56.294365990 +0000
@@ -7,6 +7,8 @@
 #  include <OpenGL/glu.h>
 #else
 #  include <GL/gl.h>
+#define GL_GLEXT_PROTOTYPES
+#  include <GL/glext.h>
 #  include <GL/glu.h>
 #endif
 
--- oyranos-0.9.6/src/examples/image_display/image_display.cpp.orig	2016-12-09 10:36:53.000000000 +0000
+++ oyranos-0.9.6/src/examples/image_display/image_display.cpp	2019-07-13 22:47:27.350994746 +0100
@@ -17,6 +17,8 @@
 #include <cstdio>
 #include <cstring>
 
+#define GL_GLEXT_PROTOTYPES
+
 #include "oyranos_config_internal.h" /* I18N */
 #include "oyranos_i18n.h"
 #include "oyranos_sentinel.h" /* oyInit_() */
--- oyranos-0.9.6/src/tools/qcmsevents/qcmsevents.h.orig	2016-12-09 10:36:53.000000000 +0000
+++ oyranos-0.9.6/src/tools/qcmsevents/qcmsevents.h	2020-07-15 22:26:02.497564862 +0100
@@ -28,6 +28,7 @@
 
 #include <X11/Xcm/XcmEvents.h>
 #include <X11/Xcm/Xcm.h>
+#undef None
 
 class QcmseDialog : public QDialog
 {
--- oyranos-0.9.6/src/modules/color/modules/devices/oyranos_cmm_oyRE.cpp.orig	2016-12-09 10:36:53.000000000 +0000
+++ oyranos-0.9.6/src/modules/color/modules/devices/oyranos_cmm_oyRE.cpp	2023-10-28 19:18:41.714487771 +0100
@@ -140,7 +140,7 @@
 extern oyCMMapi8_s_ _api8;
 
 
-bool is_raw( int id );
+bool is_raw( Exiv2::ImageType id );
 int DeviceFromContext(oyConfig_s **config, libraw_output_params_t *params);
 int DeviceFromHandle_opt(oyConfig_s *device, oyOption_s *option);
 
@@ -482,7 +482,7 @@
       oyOptions_s * result = 0;
 
       oyOptions_SetFromInt( &opts, "///icc_profile_flags", icc_profile_flags, 0, OY_CREATE_NEW );
-      const char * reg = "//"OY_TYPE_STD"/create_profile.color_matrix.icc";
+      const char * reg = "//" OY_TYPE_STD "/create_profile.color_matrix.icc";
       oyOptions_Handle( reg, opts, "create_profile.icc_profile.color_matrix",
                         &result );
 
@@ -558,7 +558,7 @@
  *
  * \todo { Untested }
  */
-int DeviceFromHandle(oyOptions_s **options, Exiv2::Image::AutoPtr image)
+int DeviceFromHandle(oyOptions_s **options, Exiv2::Image::UniquePtr &image)
 {
    int error = 0;
 
@@ -1447,14 +1447,14 @@
    DFC_OPT_ADD_INT_ARR(greybox,1) //4
    DFC_OPT_ADD_INT_ARR(greybox,2) //4
    DFC_OPT_ADD_INT_ARR(greybox,3) //4
-   DFC_OPT_ADD_INT(shot_select)
+//   DFC_OPT_ADD_INT(shot_select)
 
    return error;
 }
 
 int DeviceFromHandle_opt(oyConfig_s *device, oyOption_s *handle_opt)
 {
-   Exiv2::Image::AutoPtr device_handle;
+   Exiv2::Image::UniquePtr device_handle;
    oyAlloc_f allocateFunc = malloc;
    if (handle_opt) {
       char * filename = NULL;
@@ -1502,21 +1502,21 @@
    return 0;
 }
 
-bool is_raw( int id )
+bool is_raw( Exiv2::ImageType id )
 {
    //using namespace Exiv2::ImageType;
    switch (id) {
-      case 3: //crw:
-      case 4: //tiff
-      case 5: //mrw:
-      case 7: //cr2:
-      case 8: //raf:
-      case 9: //orf:
-      case 16: //rw2:
+      case Exiv2::ImageType::crw: //crw:
+      case Exiv2::ImageType::tiff: //tiff
+      case Exiv2::ImageType::mrw: //mrw:
+      case Exiv2::ImageType::cr2: //cr2:
+      case Exiv2::ImageType::raf: //raf:
+      case Exiv2::ImageType::orf: //orf:
+      case Exiv2::ImageType::rw2: //rw2:
          return true;
          break;
       default:
-         if(id == 0)
+         if(id == Exiv2::ImageType::none)
            return false;
          else
            return true;
--- oyranos-0.9.6/src/tools/qcmsevents/CMakeLists.txt.orig	2016-12-09 10:36:53.000000000 +0000
+++ oyranos-0.9.6/src/tools/qcmsevents/CMakeLists.txt	2023-12-03 11:58:24.045179176 +0000
@@ -1,5 +1,5 @@
 # X Color Management event observer
-ADD_SUBDIRECTORY( qt4 )
+#ADD_SUBDIRECTORY( qt4 )
 
 FIND_PACKAGE(Qt5 COMPONENTS Widgets X11Extras Xml Svg)
 IF(QT_FOUND OR Qt5Widgets_FOUND)
--- oyranos-0.9.6/src/modules/color/modules/devices/oyranos_cmm_CUPS.c.orig	2016-12-09 10:36:53.000000000 +0000
+++ oyranos-0.9.6/src/modules/color/modules/devices/oyranos_cmm_CUPS.c	2025-06-03 14:42:44.877157515 +0100
@@ -321,7 +321,7 @@
       system_port = device_name;
 
       host = cupsServer();
-      attrs = ppdFindAttr(ppd, "cupsICCProfile", 0);
+      attrs = ppdFindAttr(ppd, "cupsICCProfile", (const char *)0);
 
       if(attrs && attrs->text)
         device_settings = attrs->text;
--- oyranos-0.9.6/src/modules/color/modules/oyranos_cmm_oPNG.c.orig	2016-12-09 10:36:53.000000000 +0000
+++ oyranos-0.9.6/src/modules/color/modules/oyranos_cmm_oPNG.c	2025-06-03 14:41:52.840965284 +0100
@@ -882,7 +882,7 @@
   {
 #if defined(PNG_iCCP_SUPPORTED)
     png_charp name = 0;
-    png_charp profile = 0;
+    png_bytep profile = 0;
     png_uint_32 proflen = 0;
     int compression = 0;
 
