--- oyranos-0.9.6/src/examples/image_display/Oy_Fl_Shader_Box.h.orig	2016-12-09 10:36:53.000000000 +0000
+++ oyranos-0.9.6/src/examples/image_display/Oy_Fl_Shader_Box.h	2018-12-21 10:48:56.294365990 +0000
@@ -7,6 +7,8 @@
 #  include <OpenGL/glu.h>
 #else
 #  include <GL/gl.h>
+#define GL_GLEXT_PROTOTYPES
+#  include <GL/glext.h>
 #  include <GL/glu.h>
 #endif
 
--- oyranos-0.9.6/src/examples/image_display/image_display.cpp.orig	2016-12-09 10:36:53.000000000 +0000
+++ oyranos-0.9.6/src/examples/image_display/image_display.cpp	2019-07-13 22:47:27.350994746 +0100
@@ -17,6 +17,8 @@
 #include <cstdio>
 #include <cstring>
 
+#define GL_GLEXT_PROTOTYPES
+
 #include "oyranos_config_internal.h" /* I18N */
 #include "oyranos_i18n.h"
 #include "oyranos_sentinel.h" /* oyInit_() */
--- oyranos-0.9.6/src/tools/qcmsevents/qcmsevents.h.orig	2016-12-09 10:36:53.000000000 +0000
+++ oyranos-0.9.6/src/tools/qcmsevents/qcmsevents.h	2020-07-15 22:26:02.497564862 +0100
@@ -28,6 +28,7 @@
 
 #include <X11/Xcm/XcmEvents.h>
 #include <X11/Xcm/Xcm.h>
+#undef None
 
 class QcmseDialog : public QDialog
 {
--- oyranos-0.9.6/src/modules/color/modules/devices/oyranos_cmm_oyRE.cpp.orig	2016-12-09 10:36:53.000000000 +0000
+++ oyranos-0.9.6/src/modules/color/modules/devices/oyranos_cmm_oyRE.cpp	2023-01-30 22:33:03.715812562 +0000
@@ -1447,7 +1447,7 @@
    DFC_OPT_ADD_INT_ARR(greybox,1) //4
    DFC_OPT_ADD_INT_ARR(greybox,2) //4
    DFC_OPT_ADD_INT_ARR(greybox,3) //4
-   DFC_OPT_ADD_INT(shot_select)
+//   DFC_OPT_ADD_INT(shot_select)
 
    return error;
 }
