--- pcl-1.12.0/PCLConfig.cmake.in.orig	2021-07-07 15:17:31.000000000 +0100
+++ pcl-1.12.0/PCLConfig.cmake.in	2021-11-07 13:31:22.082792859 +0000
@@ -389,7 +389,7 @@
   endif()
 else()
 # PCLConfig.cmake is installed to PCL_ROOT/share/pcl-x.y
-  get_filename_component(PCL_ROOT "${CMAKE_CURRENT_LIST_DIR}/../.." ABSOLUTE)
+  get_filename_component(PCL_ROOT "${CMAKE_CURRENT_LIST_DIR}/../../.." ABSOLUTE)
 endif()
 
 # check whether PCLConfig.cmake is found into a PCL installation or in a build tree
