--- pcl-1.15.1/PCLConfig.cmake.in.orig	2025-08-26 17:14:08.000000000 +0100
+++ pcl-1.15.1/PCLConfig.cmake.in	2026-02-13 17:23:13.732240858 +0000
@@ -413,7 +413,7 @@
   endif()
 else()
 # PCLConfig.cmake is installed to PCL_ROOT/share/pcl-x.y
-  get_filename_component(PCL_ROOT "${CMAKE_CURRENT_LIST_DIR}/../.." ABSOLUTE)
+  get_filename_component(PCL_ROOT "${CMAKE_CURRENT_LIST_DIR}/../../.." ABSOLUTE)
 endif()
 
 # check whether PCLConfig.cmake is found into a PCL installation or in a build tree
--- pcl-1.15.1/outofcore/CMakeLists.txt.orig	2025-08-26 17:14:08.000000000 +0100
+++ pcl-1.15.1/outofcore/CMakeLists.txt	2026-02-13 18:35:02.510367185 +0000
@@ -76,10 +76,10 @@
 
 if(BUILD_visualization)
   PCL_ADD_LIBRARY(${LIB_NAME} COMPONENT ${SUBSYS_NAME} SOURCES ${srcs} ${incs} ${impl_incs} ${visualization_incs})
-  target_link_libraries("${LIB_NAME}" pcl_common pcl_visualization Boost::system Boost::filesystem)
+  target_link_libraries("${LIB_NAME}" pcl_common pcl_visualization Boost::filesystem)
 else()
   PCL_ADD_LIBRARY(${LIB_NAME} COMPONENT ${SUBSYS_NAME} SOURCES ${srcs} ${incs} ${impl_incs})
-  target_link_libraries("${LIB_NAME}" pcl_common Boost::system Boost::filesystem)
+  target_link_libraries("${LIB_NAME}" pcl_common Boost::filesystem)
 endif()
 if(HAVE_CJSON)
   target_link_libraries("${LIB_NAME}" ${CJSON_LIBRARIES})
