diff --git a/apps/modeler/src/points_actor_item.cpp b/apps/modeler/src/points_actor_item.cpp
index fbd94669b2..3f4d6fb320 100755
--- a/apps/modeler/src/points_actor_item.cpp
+++ b/apps/modeler/src/points_actor_item.cpp
@@ -87,7 +87,9 @@ pcl::modeler::PointsActorItem::initImpl()
   mapper->SetScalarModeToUsePointData();
   mapper->InterpolateScalarsBeforeMappingOn();
   mapper->ScalarVisibilityOn();
+#if VTK_RENDERING_BACKEND_OPENGL_VERSION < 2
   mapper->ImmediateModeRenderingOff();
+#endif
 
   vtkSmartPointer<vtkLODActor> actor = vtkSmartPointer<vtkLODActor>(dynamic_cast<vtkLODActor*>(actor_.GetPointer()));
   actor->SetMapper(mapper);
diff --git a/apps/modeler/src/surface_actor_item.cpp b/apps/modeler/src/surface_actor_item.cpp
index 6a8a925286..011d3d3c94 100755
--- a/apps/modeler/src/surface_actor_item.cpp
+++ b/apps/modeler/src/surface_actor_item.cpp
@@ -84,7 +84,9 @@ pcl::modeler::SurfaceActorItem::initImpl()
   mapper->SetScalarModeToUsePointData ();
   mapper->InterpolateScalarsBeforeMappingOn ();
   mapper->ScalarVisibilityOn ();
+#if VTK_RENDERING_BACKEND_OPENGL_VERSION < 2
   mapper->ImmediateModeRenderingOff ();
+#endif
 
   vtkSmartPointer<vtkLODActor> actor = vtkSmartPointer<vtkLODActor>(dynamic_cast<vtkLODActor*>(actor_.GetPointer()));
   actor->SetMapper(mapper);
