--- pcl-1.8.1/examples/segmentation/example_cpc_segmentation.cpp.orig	2017-08-08 07:04:14.000000000 +0100
+++ pcl-1.8.1/examples/segmentation/example_cpc_segmentation.cpp	2018-05-05 22:45:38.795827133 +0100
@@ -489,8 +489,8 @@
           color = concave_color;
 
         // two times since we add also two points per edge
-        colors->InsertNextTupleValue (color);
-        colors->InsertNextTupleValue (color);
+        colors->InsertNextTypedTuple (color);
+        colors->InsertNextTypedTuple (color);
 
         pcl::Supervoxel<PointT>::Ptr supervoxel = supervoxel_clusters.at (sv_label);
         pcl::PointXYZRGBA vert_curr = supervoxel->centroid_;
--- pcl-1.8.1/examples/segmentation/example_lccp_segmentation.cpp.orig	2017-08-08 07:04:14.000000000 +0100
+++ pcl-1.8.1/examples/segmentation/example_lccp_segmentation.cpp	2018-05-05 22:46:47.375823860 +0100
@@ -409,8 +409,8 @@
           color = concave_color;
         
         // two times since we add also two points per edge
-        colors->InsertNextTupleValue (color);
-        colors->InsertNextTupleValue (color);
+        colors->InsertNextTypedTuple (color);
+        colors->InsertNextTypedTuple (color);
         
         pcl::Supervoxel<PointT>::Ptr supervoxel = supervoxel_clusters.at (sv_label);
         pcl::PointXYZRGBA vert_curr = supervoxel->centroid_;
