--- pcl-1.14.0/PCLConfig.cmake.in.orig	2024-01-03 16:25:47.000000000 +0000
+++ pcl-1.14.0/PCLConfig.cmake.in	2024-04-19 13:17:02.320023721 +0100
@@ -411,7 +411,7 @@
   endif()
 else()
 # PCLConfig.cmake is installed to PCL_ROOT/share/pcl-x.y
-  get_filename_component(PCL_ROOT "${CMAKE_CURRENT_LIST_DIR}/../.." ABSOLUTE)
+  get_filename_component(PCL_ROOT "${CMAKE_CURRENT_LIST_DIR}/../../.." ABSOLUTE)
 endif()
 
 # check whether PCLConfig.cmake is found into a PCL installation or in a build tree
--- pcl-1.14.0/io/include/pcl/io/ply/ply_parser.h.orig	2024-01-03 16:25:47.000000000 +0000
+++ pcl-1.14.0/io/include/pcl/io/ply/ply_parser.h	2024-04-19 13:26:12.915060688 +0100
@@ -49,6 +49,7 @@
 #include <string>
 #include <tuple>
 #include <vector>
+#include <functional>
 #include <boost/lexical_cast.hpp> // for lexical_cast
 #include <boost/mpl/fold.hpp> // for fold
 #include <boost/mpl/inherit.hpp> // for inherit
--- pcl-1.14.0/outofcore/include/pcl/outofcore/octree_base.h.orig	2024-01-03 16:25:47.000000000 +0000
+++ pcl-1.14.0/outofcore/include/pcl/outofcore/octree_base.h	2024-04-19 13:17:14.180024518 +0100
@@ -63,6 +63,7 @@
 #include <pcl/PCLPointCloud2.h>
 
 #include <shared_mutex>
+#include <list>
 
 namespace pcl
 {
--- pcl-1.14.0/outofcore/include/pcl/outofcore/octree_base_node.h.orig	2024-01-03 16:25:47.000000000 +0000
+++ pcl-1.14.0/outofcore/include/pcl/outofcore/octree_base_node.h	2024-04-19 13:17:14.180024518 +0100
@@ -42,6 +42,7 @@
 #include <memory>
 #include <mutex>
 #include <random>
+#include <list>
 
 #include <pcl/common/io.h>
 #include <pcl/PCLPointCloud2.h>
--- pcl-1.14.0/test/outofcore/test_outofcore.cpp.orig	2024-01-03 16:25:47.000000000 +0000
+++ pcl-1.14.0/test/outofcore/test_outofcore.cpp	2024-04-19 13:17:14.180024518 +0100
@@ -44,6 +44,7 @@
 
 #include <pcl/test/gtest.h>
 
+#include <list>
 #include <vector>
 #include <iostream>
 #include <random>
--- pcl-1.14.0/recognition/include/pcl/recognition/face_detection/face_detector_data_provider.h.orig	2024-01-03 16:25:47.000000000 +0000
+++ pcl-1.14.0/recognition/include/pcl/recognition/face_detection/face_detector_data_provider.h	2024-04-19 13:17:14.180024518 +0100
@@ -12,7 +12,7 @@
 #include <pcl/recognition/face_detection/face_common.h>
 
 #include <boost/algorithm/string.hpp>
-#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem.hpp>
 
 #include <fstream>
 #include <string>
