--- pcl-1.15.1/PCLConfig.cmake.in.orig	2025-08-26 17:14:08.000000000 +0100
+++ pcl-1.15.1/PCLConfig.cmake.in	2026-09-08 10:46:31.415630804 +0100
@@ -413,7 +413,7 @@
   endif()
 else()
 # PCLConfig.cmake is installed to PCL_ROOT/share/pcl-x.y
-  get_filename_component(PCL_ROOT "${CMAKE_CURRENT_LIST_DIR}/../.." ABSOLUTE)
+  get_filename_component(PCL_ROOT "${CMAKE_CURRENT_LIST_DIR}/../../.." ABSOLUTE)
 endif()
 
 # check whether PCLConfig.cmake is found into a PCL installation or in a build tree
--- pcl-1.15.1/outofcore/CMakeLists.txt.orig	2025-08-26 17:14:08.000000000 +0100
+++ pcl-1.15.1/outofcore/CMakeLists.txt	2026-09-08 10:46:31.415714831 +0100
@@ -76,10 +76,10 @@
 
 if(BUILD_visualization)
   PCL_ADD_LIBRARY(${LIB_NAME} COMPONENT ${SUBSYS_NAME} SOURCES ${srcs} ${incs} ${impl_incs} ${visualization_incs})
-  target_link_libraries("${LIB_NAME}" pcl_common pcl_visualization Boost::system Boost::filesystem)
+  target_link_libraries("${LIB_NAME}" pcl_common pcl_visualization Boost::filesystem)
 else()
   PCL_ADD_LIBRARY(${LIB_NAME} COMPONENT ${SUBSYS_NAME} SOURCES ${srcs} ${incs} ${impl_incs})
-  target_link_libraries("${LIB_NAME}" pcl_common Boost::system Boost::filesystem)
+  target_link_libraries("${LIB_NAME}" pcl_common Boost::filesystem)
 endif()
 if(HAVE_CJSON)
   target_link_libraries("${LIB_NAME}" ${CJSON_LIBRARIES})
--- pcl-1.15.1/outofcore/include/pcl/outofcore/visualization/geometry.h.orig	2025-08-26 17:14:08.000000000 +0100
+++ pcl-1.15.1/outofcore/include/pcl/outofcore/visualization/geometry.h	2026-09-08 11:09:08.021887176 +0100
@@ -1,6 +1,7 @@
 #pragma once
 
 // C++
+#include <iostream>
 #include <string>
 
 // PCL
--- pcl-1.15.1/visualization/src/histogram_visualizer.cpp.orig	2025-08-26 17:14:08.000000000 +0100
+++ pcl-1.15.1/visualization/src/histogram_visualizer.cpp	2026-09-08 10:54:42.785906887 +0100
@@ -170,7 +170,7 @@
 pcl::visualization::PCLHistogramVisualizer::reCreateActor (
     const vtkSmartPointer<vtkDoubleArray> &xy_array, RenWinInteract* renwinupd, const int hsize)
 {
-  renwinupd->ren_->RemoveActor2D (renwinupd->xy_plot_);
+  renwinupd->ren_->RemoveViewProp (renwinupd->xy_plot_);
   renwinupd->xy_plot_->RemoveAllDataSetInputConnections ();
   
   double min_max[2];
