--- pdf417.h.orig	2005-01-31 00:33:09.000000000 +0000
+++ pdf417.h	2005-08-05 08:53:55.000000000 +0100
@@ -17,4 +17,4 @@
 extern void pdf417_en_new (int in_rval, int in_cval, int in_ec_level,
                        int output_type, int use_default, int raw_in_mode,
                        int X, int Y, int QZ );
-extern void pdf417_prep_to_raw(const char* buffer, int buffer_len);
+extern int pdf417_prep_to_raw(const char* buffer, int buffer_len);
