--- pfstools-2.1.0/cmake/FindOpenEXR.cmake.orig	2017-11-13 17:38:18.000000000 +0000
+++ pfstools-2.1.0/cmake/FindOpenEXR.cmake	2018-10-24 19:45:56.743398232 +0100
@@ -111,7 +111,7 @@
     OPENEXR_HALF_LIBRARY OPENEXR_IEX_LIBRARY OPENEXR_ILMIMF_LIBRARY OPENEXR_ILMTHREAD_LIBRARY OPENEXR_IMATH_LIBRARY ZLIB_LIBRARY OPENEXR_INCLUDE_DIR)
  	
 IF(OPENEXR_FOUND)
-  SET(OPENEXR_LIBRARIES ${OPENEXR_ILMTHREAD_LIBRARY} ${OPENEXR_IEX_LIBRARY} ${OPENEXR_ILMIMF_LIBRARY} ${OPENEXR_IMATH_LIBRARY} ${OPENEXR_HALF_LIBRARY} ${ZLIB_LIBRARY})
+  SET(OPENEXR_LIBRARIES ${OPENEXR_ILMTHREAD_LIBRARY} ${OPENEXR_IEX_LIBRARY} ${OPENEXR_ILMIMF_LIBRARY} ${OPENEXR_IMATH_LIBRARY} ${OPENEXR_HALF_LIBRARY} ${ZLIB_LIBRARY} pthread)
   # Both include paths are needed because of dummy OSL headers mixing #include <OpenEXR/foo.h> and #include <foo.h> :(
   SET(OPENEXR_INCLUDE_DIRS ${OPENEXR_INCLUDE_DIR} ${OPENEXR_INCLUDE_DIR}/OpenEXR ${ZLIB_INCLUDE_DIR})
 ENDIF()
--- pfstools-2.1.0/src/camera/pfsalign.cpp.orig	2020-10-30 20:41:37.873386631 +0000
+++ pfstools-2.1.0/src/camera/pfsalign.cpp	2020-10-30 20:48:48.015576270 +0000
@@ -230,7 +230,7 @@
 
     vector<uchar> inliners(matches_sym.size(), 0);
     //        affine = findHomography( pp1, pp2, inliners, CV_RANSAC, 1. );
-    homography = findHomography( pp2, pp1, CV_RANSAC, 1., inliners );
+    homography = findHomography( pp2, pp1, RANSAC, 1., inliners );
 
     //    solve( pp1, pp2, affine, DECOMP_SVD );
     //    Mat affine = (Mat_<float>(2,3) << 1, 0, 0, 0, 1, 10);
--- pfstools-2.1.0/src/octave/pfsclose.cpp.orig	2017-11-13 17:38:28.000000000 +0000
+++ pfstools-2.1.0/src/octave/pfsclose.cpp	2020-12-08 20:18:46.040966257 +0000
@@ -47,7 +47,7 @@
 
   // Get arguments and check if they are legal
     
-  if( nargin != 1 || !args(0).is_map() )
+  if( nargin != 1 || !args(0).isstruct() )
   {
     error( SCRIPT_NAME ": Improper usage!");
     return retval;
--- pfstools-2.1.0/src/octave/pfsget.cpp.orig	2017-11-13 17:38:28.000000000 +0000
+++ pfstools-2.1.0/src/octave/pfsget.cpp	2020-12-08 20:18:46.040966257 +0000
@@ -46,7 +46,7 @@
 	
   int nargin = args.length();
 
-  if( nargin != 1 || !args(0).is_map() )
+  if( nargin != 1 || !args(0).isstruct() )
   {
     error( SCRIPT_NAME ": Improper usage!");
     return retval;
--- pfstools-2.1.0/src/octave/pfsput.cpp.orig	2017-11-13 17:38:28.000000000 +0000
+++ pfstools-2.1.0/src/octave/pfsput.cpp	2020-12-08 20:18:46.041966257 +0000
@@ -46,7 +46,7 @@
 	
   int nargin = args.length();
 
-  if( nargin != 1 || !args(0).is_map() )
+  if( nargin != 1 || !args(0).isstruct() )
   {
     error( SCRIPT_NAME ": Improper usage!");
     return retval;
@@ -98,7 +98,7 @@
   {
     octave_scalar_map::const_iterator itChannels = pfsStream.seek( "channels" );
     if( itChannels == pfsStream.end() ||
-      !pfsStream.contents( itChannels )(0).is_map() )
+      !pfsStream.contents( itChannels )(0).isstruct() )
     {
       error( SCRIPT_NAME ": 'channels' field missing in the structure or it has wrong type");
       return retval;
@@ -137,7 +137,7 @@
     {
       octave_scalar_map::const_iterator itTags = pfsStream.seek( "tags" );
       if( itTags != pfsStream.end() ) {
-        if( !pfsStream.contents( itTags )(0).is_map() )
+        if( !pfsStream.contents( itTags )(0).isstruct() )
         {
           throw pfs::Exception( "'tags' field must be a structure" );  
         }
@@ -158,14 +158,14 @@
     {
       octave_scalar_map::const_iterator itChTags = pfsStream.seek( "channelTags" );
       if( itChTags != pfsStream.end() ) {
-        if( !pfsStream.contents( itChTags )(0).is_map() )
+        if( !pfsStream.contents( itChTags )(0).isstruct() )
         {
           throw pfs::Exception( "'channelTags' field must be a structure" );  
         }
         octave_map tagChannels = pfsStream.contents( itChTags )(0).map_value();
         for( octave_scalar_map::iterator itCh = tagChannels.begin(); itCh != tagChannels.end(); itCh++ ) {
           std::string channelName = tagChannels.key(itCh);
-          if( !tagChannels.contents( itCh )(0).is_map() ) {
+          if( !tagChannels.contents( itCh )(0).isstruct() ) {
             throw pfs::Exception( "each channelTags file must be a structure" );  
           }
           pfs::Channel *pfsChannel = frame->getChannel( channelName.c_str() );
