--- pfstools-2.1.0/cmake/FindOpenEXR.cmake.orig	2017-11-13 17:38:18.000000000 +0000
+++ pfstools-2.1.0/cmake/FindOpenEXR.cmake	2018-10-24 19:45:56.743398232 +0100
@@ -111,7 +111,7 @@
     OPENEXR_HALF_LIBRARY OPENEXR_IEX_LIBRARY OPENEXR_ILMIMF_LIBRARY OPENEXR_ILMTHREAD_LIBRARY OPENEXR_IMATH_LIBRARY ZLIB_LIBRARY OPENEXR_INCLUDE_DIR)
  	
 IF(OPENEXR_FOUND)
-  SET(OPENEXR_LIBRARIES ${OPENEXR_ILMTHREAD_LIBRARY} ${OPENEXR_IEX_LIBRARY} ${OPENEXR_ILMIMF_LIBRARY} ${OPENEXR_IMATH_LIBRARY} ${OPENEXR_HALF_LIBRARY} ${ZLIB_LIBRARY})
+  SET(OPENEXR_LIBRARIES ${OPENEXR_ILMTHREAD_LIBRARY} ${OPENEXR_IEX_LIBRARY} ${OPENEXR_ILMIMF_LIBRARY} ${OPENEXR_IMATH_LIBRARY} ${OPENEXR_HALF_LIBRARY} ${ZLIB_LIBRARY} pthread)
   # Both include paths are needed because of dummy OSL headers mixing #include <OpenEXR/foo.h> and #include <foo.h> :(
   SET(OPENEXR_INCLUDE_DIRS ${OPENEXR_INCLUDE_DIR} ${OPENEXR_INCLUDE_DIR}/OpenEXR ${ZLIB_INCLUDE_DIR})
 ENDIF()
--- pfstools-2.1.0/src/camera/pfsalign.cpp.orig	2020-10-30 20:41:37.873386631 +0000
+++ pfstools-2.1.0/src/camera/pfsalign.cpp	2020-10-30 20:48:48.015576270 +0000
@@ -230,7 +230,7 @@
 
     vector<uchar> inliners(matches_sym.size(), 0);
     //        affine = findHomography( pp1, pp2, inliners, CV_RANSAC, 1. );
-    homography = findHomography( pp2, pp1, CV_RANSAC, 1., inliners );
+    homography = findHomography( pp2, pp1, RANSAC, 1., inliners );
 
     //    solve( pp1, pp2, affine, DECOMP_SVD );
     //    Mat affine = (Mat_<float>(2,3) << 1, 0, 0, 0, 1, 10);
