--- pgf-6.11.24/Makefile.am.orig	2010-05-14 19:54:16.000000000 +0100
+++ pgf-6.11.24/Makefile.am	2011-12-07 09:18:08.000000000 +0000
@@ -16,8 +16,8 @@
 RPMRELEASE=$(PACKAGE_RELEASE)
 
 install-data-local:: $(PACKAGE).pc
-	$(mkinstalldirs) $(DESTDIR)$(bindir)/pkgconfig
-	$(INSTALL) -m 644 $(PACKAGE).pc $(DESTDIR)$(bindir)/pkgconfig/$(PACKAGE).pc
+	$(mkinstalldirs) $(DESTDIR)$(libdir)/pkgconfig
+	$(INSTALL) -m 644 $(PACKAGE).pc $(DESTDIR)$(libdir)/pkgconfig/$(PACKAGE).pc
 
 
 .PHONY: ChangeLog
