--- piave-0.2.4/plugins/iostream/vorbisfile/vorbisfile.cc.orig	2003-09-15 11:52:25.000000000 +0100
+++ piave-0.2.4/plugins/iostream/vorbisfile/vorbisfile.cc	2007-09-25 11:55:35.000000000 +0100
@@ -36,6 +36,7 @@
 #include <stdio.h>
 #include <sys/ioctl.h>
 #include <sys/soundcard.h>
+#include <math.h>
 
 // std c++
 #include <iostream>
--- piave-0.2.4/plugins/iostream/sndfile/sndfile.cc.orig	2003-11-28 05:31:38.000000000 +0000
+++ piave-0.2.4/plugins/iostream/sndfile/sndfile.cc	2007-09-25 11:56:21.000000000 +0100
@@ -36,6 +36,7 @@
 #include <stdio.h>
 #include <sys/ioctl.h>
 #include <sys/soundcard.h>
+#include <math.h>
 
 // std c++
 #include <iostream>
--- piave-0.2.4/libpiave/avccontroller.hh.orig	2003-09-09 12:01:36.000000000 +0100
+++ piave-0.2.4/libpiave/avccontroller.hh	2007-09-25 12:05:31.000000000 +0100
@@ -47,7 +47,7 @@
 // IEEE1394
 #include <libraw1394/raw1394.h>
 #include <libraw1394/csr.h>
-#include <libdv/dv1394.h>
+#include <libdv/dv.h>
 
 namespace PIAVE {
 
--- piave-0.2.4/libpiave/IEEE1394IO.hh.orig	2003-09-15 11:40:40.000000000 +0100
+++ piave-0.2.4/libpiave/IEEE1394IO.hh	2007-09-25 12:06:00.000000000 +0100
@@ -47,7 +47,7 @@
 // IEEE1394
 #include <libraw1394/raw1394.h>
 #include <libraw1394/csr.h>
-#include <libdv/dv1394.h>
+#include <libdv/dv.h>
 // channel is 63 (broadcast) for video
 #define  AVC1394_CHANNEL_BROADCAST 63
 
