--- pingus-0.7.4/Makefile.orig	2011-10-10 16:49:59.000000000 +0100
+++ pingus-0.7.4/Makefile	2011-10-23 17:24:02.000000000 +0100
@@ -39,7 +39,7 @@
 	install -d "$(DESTDIR)$(BINDIR)"
 
 	install -D build/pingus "$(DESTDIR)$(BINDIR)/pingus.bin"
-	echo "#!/bin/sh\nexec \"$(BINDIR)/pingus.bin\" --datadir \"$(DATADIR)\"" > "$(DESTDIR)$(BINDIR)/pingus"
+	echo -e "#!/bin/sh\nexec \"$(BINDIR)/pingus.bin\" --datadir \"$(DATADIR)\"" > "$(DESTDIR)$(BINDIR)/pingus"
 	chmod 755 "$(DESTDIR)$(BINDIR)/pingus"
 
 install-data:
@@ -66,7 +66,7 @@
         \) -exec install -D {} $(DESTDIR)$(DATADIR)/{} \;
 
 install-man:
-	install -D doc/man/pingus.6 "$(DESTDIR)$(MANDIR)/man1/pingus.6"
+	install -D doc/man/pingus.6 "$(DESTDIR)$(MANDIR)/man6/pingus.6"
 
 .PHONY : clean install install-exec install-data install-man
 
