--- pingus-0.7.2/SConstruct.orig	2007-10-30 09:23:37.000000000 +0000
+++ pingus-0.7.2/SConstruct	2007-11-09 12:12:12.000000000 +0000
@@ -539,6 +539,11 @@
     env = Environment(options = opts)
     Help(opts.GenerateHelpText(env))
 
+    if os.environ.has_key('PATH'):
+        env['ENV']['PATH'] = os.environ['PATH']
+    if os.environ.has_key('HOME'):
+       env['ENV']['HOME'] = os.environ['HOME']
+
     opts.Update(env)
     env['CPPPATH'] += ['.', 'src/']
     Default(env.Program('pingus', pingus_sources + env['optional_sources']))
