--- pingus-0.7.2/src/input/evdev_device.cpp.orig	2007-10-01 18:24:54.000000000 +0100
+++ pingus-0.7.2/src/input/evdev_device.cpp	2010-06-20 22:17:00.000000000 +0100
@@ -25,6 +25,8 @@
 #include <stdexcept>
 #include <iostream>
 #include <errno.h>
+#include <stdio.h>
+#include <string.h>
 
 #include "evdev_device.hpp"
 
--- pingus-0.7.2/src/tinygettext/po_file_reader.cpp.orig	2007-10-18 18:54:01.000000000 +0100
+++ pingus-0.7.2/src/tinygettext/po_file_reader.cpp	2010-06-20 22:13:07.000000000 +0100
@@ -24,6 +24,7 @@
 #include <sstream>
 #include <stdexcept>
 #include <string.h>
+#include <stdio.h>
 #include "iconv.hpp"
 #include "po_file_reader.hpp"
 
--- pingus-0.7.2/SConstruct.orig	2007-10-30 09:23:37.000000000 +0000
+++ pingus-0.7.2/SConstruct	2007-11-09 12:12:12.000000000 +0000
@@ -539,6 +539,11 @@
     env = Environment(options = opts)
     Help(opts.GenerateHelpText(env))
 
+    if os.environ.has_key('PATH'):
+        env['ENV']['PATH'] = os.environ['PATH']
+    if os.environ.has_key('HOME'):
+       env['ENV']['HOME'] = os.environ['HOME']
+
     opts.Update(env)
     env['CPPPATH'] += ['.', 'src/']
     Default(env.Program('pingus', pingus_sources + env['optional_sources']))
