--- pingus-0.6.0/configure.ac.orig	2003-04-16 00:56:40.000000000 +0100
+++ pingus-0.6.0/configure.ac	2006-12-08 09:57:39.000000000 +0000
@@ -64,6 +64,8 @@
 dnl -- Static Libaries --
 AC_PROG_RANLIB
 
+CPPFLAGS="$CPPFLAGS `pkg-config clanCore-0.8 --cflags`"
+
 AC_CHECK_HEADER(ClanLib/core.h,, [echo "Please get ClanLib from http://www.clanlib.org"; exit])
 
 dnl FIXME: we don't need pthread, but some system might need it to succeed at the final link
@@ -122,6 +124,8 @@
 
 AC_CHECK_LIB(Hermes,   main,,)
 
+LIBS="$LIBS `pkg-config clanCore-0.8 --libs`"
+
 AC_CHECK_LIB( clanCore,   main,, [
 	echo "ClanLib installed, but -lclanCore not found, run 'ldconfig' and try again,"
 	echo "if that doesn't help have a look at the file config.log and look into"
@@ -212,23 +216,23 @@
 #   	[echo "cross compiling detected, assuming that this is correct and continue"]
 #   )
 
-AC_CHECK_LIB( clanJPEG,   main,, [
-	echo "ClanLib is installed, but clanJPEG was not found. You probably"
-	echo "forgot to compile clanJPEG, which is part of the ClanLib package"
-	echo "or you need to run 'ldconfig'."
-	echo "If you think you have everything correctly installed have a look at"
-	echo "the file 'config.log' to see what went wrong"
-	exit
-	])
-
-AC_CHECK_LIB( clanPNG,   main,, [
-	echo "ClanLib is installed, but clanPNG was not found. You probably"
-	echo "forgot to compile clanPNG, which is part of the ClanLib package"
-	echo "or you need to run 'ldconfig'."
-	echo "If you think you have everything correctly installed have a look at"
-	echo "the file 'config.log' to see what went wrong"
-	exit
-	])
+#AC_CHECK_LIB( clanJPEG,   main,, [
+#	echo "ClanLib is installed, but clanJPEG was not found. You probably"
+#	echo "forgot to compile clanJPEG, which is part of the ClanLib package"
+#	echo "or you need to run 'ldconfig'."
+#	echo "If you think you have everything correctly installed have a look at"
+#	echo "the file 'config.log' to see what went wrong"
+#	exit
+#	])
+
+#AC_CHECK_LIB( clanPNG,   main,, [
+#	echo "ClanLib is installed, but clanPNG was not found. You probably"
+#	echo "forgot to compile clanPNG, which is part of the ClanLib package"
+#	echo "or you need to run 'ldconfig'."
+#	echo "If you think you have everything correctly installed have a look at"
+#	echo "the file 'config.log' to see what went wrong"
+#	exit
+#	])
 
 AC_CHECK_LIB( clanGUI,   main,, [
 	echo "ClanLib is installed, but clanGUI was not found. You probably"
