--- pingus-20061208/data/Makefile.am.orig	2006-12-08 12:11:55.000000000 +0000
+++ pingus-20061208/data/Makefile.am	2006-12-08 13:27:17.000000000 +0000
@@ -15,7 +15,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
-SUBDIRS = controller images levels demos data music po sounds worldmaps themes prefabs
+SUBDIRS = controller images levels demos data music po sounds worldmaps themes prefabs metamap
 
 # Tweaking the `datadir' to install the files in the correct location
 pkgdatadir = $(datadir)/games/@PACKAGE@
--- pingus-20061208/configure.ac.orig	2006-12-08 12:12:05.000000000 +0000
+++ pingus-20061208/configure.ac	2006-12-08 13:42:14.000000000 +0000
@@ -73,7 +73,7 @@
   proto_iconv=""
 fi
 AC_SUBST([ICONV_LIBS], [$LIBICONV])
-AC_DEFINE([ICONV_CONST], $proto_iconv,
+AC_DEFINE_UNQUOTED([ICONV_CONST], $proto_iconv,
   [Define as const if the declaration of iconv() needs const.])
 
 AC_SUBST(PINGUS_CFLAGS)
@@ -103,6 +103,8 @@
 AC_TYPE_SIGNAL
 AC_CHECK_FUNCS(select strdup)
 
+AM_GNU_GETTEXT
+
 dnl Checking for the install prefix and then use it to define PINGUS_DATADIR
 PINGUS_EXPAND_DIR(pingus_datadir, "$datadir/games/$PACKAGE")
 AC_DEFINE_UNQUOTED(PINGUS_DATADIR, "$pingus_datadir", [Location of Pingus data files])
@@ -141,6 +143,7 @@
            data/prefabs/Makefile
            data/themes/Makefile
            data/worldmaps/Makefile
+           data/metamap/Makefile
            data/demos/Makefile
            data/data/Makefile
            data/images/Makefile
