--- pioneer-20220203/data/models/ships/deneb/deneb_medhigh.dae.orig	2022-02-02 02:01:08.000000000 +0000
+++ pioneer-20220203/data/models/ships/deneb/deneb_medhigh.dae	2022-02-05 21:53:53.849485940 +0000
@@ -3824,7 +3824,7 @@
     </animation>
   </library_animations>
   <library_controllers>
-    <controller id="LandingGear_starboard_001-morph" name="LandingGear_starboard.001-morph">
+<!--    <controller id="LandingGear_starboard_001-morph" name="LandingGear_starboard.001-morph">
       <morph source="#Sphere_018-mesh" method="NORMALIZED">
         <source id="LandingGear_starboard_001-targets">
           <IDREF_array id="LandingGear_starboard_001-targets-array" count="0"/>
@@ -3847,7 +3847,7 @@
           <input semantic="MORPH_WEIGHT" source="#LandingGear_starboard_001-weights"/>
         </targets>
       </morph>
-    </controller>
+    </controller>-->
   </library_controllers>
   <library_visual_scenes>
     <visual_scene id="Scene" name="Scene">
@@ -3966,4 +3966,4 @@
   <scene>
     <instance_visual_scene url="#Scene"/>
   </scene>
-</COLLADA>
\ No newline at end of file
+</COLLADA>
--- pioneer-20220203/data/models/ships/deneb/deneb.dae.orig	2022-02-02 02:01:08.000000000 +0000
+++ pioneer-20220203/data/models/ships/deneb/deneb.dae	2022-02-05 21:53:20.040487553 +0000
@@ -3812,7 +3812,7 @@
     </animation>
   </library_animations>
   <library_controllers>
-    <controller id="LandingGear_starboard-morph" name="LandingGear_starboard-morph">
+<!--    <controller id="LandingGear_starboard-morph" name="LandingGear_starboard-morph">
       <morph source="#Sphere_010-mesh" method="NORMALIZED">
         <source id="LandingGear_starboard-targets">
           <IDREF_array id="LandingGear_starboard-targets-array" count="0"/>
@@ -3835,7 +3835,7 @@
           <input semantic="MORPH_WEIGHT" source="#LandingGear_starboard-weights"/>
         </targets>
       </morph>
-    </controller>
+    </controller>-->
   </library_controllers>
   <library_visual_scenes>
     <visual_scene id="Scene" name="Scene">
