--- pipepanic-0.1.0-source/Makefile.orig	2006-05-14 22:38:44.000000000 +0100
+++ pipepanic-0.1.0-source/Makefile	2006-05-15 13:16:13.000000000 +0100
@@ -4,10 +4,10 @@
 
 # Settings for x86 for desktop development.
 CC=gcc
-CFLAGS=-O2 -Wall -pedantic `sdl-config --cflags` -Dmain=SDL_main
+CFLAGS=-O2 -Wall -pedantic `sdl-config --cflags`
 LINK=gcc
 LDFLAGS=
-LIBS=-lSDL -lSDLmain
+LIBS=-lSDL
 
 all:
 	$(CC) $(CFLAGS) -c $(SOURCES)
