--- pixelstruct-20130906/src/triangulation.h.orig	2013-09-06 20:59:33.000000000 +0100
+++ pixelstruct-20130906/src/triangulation.h	2016-09-20 20:08:10.538635823 +0100
@@ -23,7 +23,7 @@
 #include <vector>
 
 #include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
-#include <CGAL/Triangulation_euclidean_traits_xy_3.h>
+#include <CGAL/Projection_traits_xy_3.h>
 #include <CGAL/Delaunay_triangulation_2.h>
 
 class Triangulation {
@@ -31,7 +31,7 @@
     typedef CGAL::Exact_predicates_inexact_constructions_kernel::Point_3
         CGAL_Point;
     typedef CGAL::Delaunay_triangulation_2<
-        CGAL::Triangulation_euclidean_traits_xy_3<
+        CGAL::Projection_traits_xy_3<
         CGAL::Exact_predicates_inexact_constructions_kernel> >
         DelaunayTriangulation;
     
--- pixelstruct-20130906/src/CMakeLists.txt.orig	2013-09-06 20:59:33.000000000 +0100
+++ pixelstruct-20130906/src/CMakeLists.txt	2016-09-20 20:05:47.114635295 +0100
@@ -52,4 +52,5 @@
     ${OPENGL_gl_LIBRARY}
     ${OPENGL_glu_LIBRARY}
     ${CGAL_LIBRARIES}
+    gmp
 )
--- pixelstruct-20130906/src/glwidget.cpp.orig	2013-09-06 20:59:33.000000000 +0100
+++ pixelstruct-20130906/src/glwidget.cpp	2016-09-20 20:05:47.114635295 +0100
@@ -27,6 +27,8 @@
 #include <QKeyEvent>
 #include <QImage>
 
+#include <GL/glu.h>
+
 using std::cerr;
 using std::endl;
 
