--- pixelstruct-20130906/src/glwidget.cpp.orig	2013-09-06 20:59:33.000000000 +0100
+++ pixelstruct-20130906/src/glwidget.cpp	2013-09-06 21:00:33.867436217 +0100
@@ -27,6 +27,8 @@
 #include <QKeyEvent>
 #include <QImage>
 
+#include <GL/glu.h>
+
 using std::cerr;
 using std::endl;
 
--- pixelstruct-20130906/src/CMakeLists.txt.orig	2013-09-06 20:59:33.000000000 +0100
+++ pixelstruct-20130906/src/CMakeLists.txt	2013-09-06 21:03:16.334419340 +0100
@@ -52,4 +52,5 @@
     ${OPENGL_gl_LIBRARY}
     ${OPENGL_glu_LIBRARY}
     ${CGAL_LIBRARIES}
+    gmp
 )
