--- pl-241/prefabs/heatmap.pl.orig	2007-11-23 21:36:57.000000000 +0000
+++ pl-241/prefabs/heatmap.pl	2010-02-10 21:27:26.000000000 +0000
@@ -212,9 +212,10 @@
   #include $chunk_doheatmap
 #endif
 
-
-#setifnotgiven legend = "max+0.7 min+0.8"
-#proc legend
+#if @legend != no
+ // legend != no added scg 2/10/10
+ #setifnotgiven legend = "max+0.7 min+0.8"
+ #proc legend
   location: @legend
   reverseorder: @reverseleg
   #ifspec legendfmt format
@@ -226,7 +227,8 @@
   #ifspec legframe frame
   #ifspec legtextdet textdetails
 
-#endproc
+ #endproc
+#endif
 
 
 //// title - added scg 8/8/05
