--- podofo-0.9.8/test/unit/StringTest.cpp.orig	2019-01-15 13:04:40.000000000 +0000
+++ podofo-0.9.8/test/unit/StringTest.cpp	2022-11-20 09:10:37.281652796 +0000
@@ -179,19 +179,19 @@
     CPPUNIT_ASSERT_EQUAL_MESSAGE( "Comparing UTF8 and UTF16 string converted to UTF8", 
                                   strUtf8.GetStringUtf8(), strUtf16.GetStringUtf8() );
 
-    CPPUNIT_ASSERT_EQUAL_MESSAGE( "Comparing UTF8 and UTF16 string", strUtf8, strUtf16 );
+    CPPUNIT_ASSERT_EQUAL_MESSAGE( "Comparing UTF8 and UTF16 string", strUtf8.GetString(), strUtf16.GetString() );
 
     // Compare two UTF16 strings
     CPPUNIT_ASSERT_EQUAL( strUtf16.GetCharacterLength(), strUtf16b.GetCharacterLength() );
     CPPUNIT_ASSERT_EQUAL( strUtf16.GetStringUtf8(), strUtf16b.GetStringUtf8() );
-    CPPUNIT_ASSERT_EQUAL( strUtf16, strUtf16b );
+    CPPUNIT_ASSERT_EQUAL( strUtf16.GetString(), strUtf16b.GetString() );
 
 }
 
 void StringTest::testWCharConstructor()
 {
-    CPPUNIT_ASSERT_EQUAL( PdfString("Hallo World"), PdfString(L"Hallo World") );
-    CPPUNIT_ASSERT_EQUAL( PdfString(L"Hallo World"), PdfString(L"Hallo World") );
+    CPPUNIT_ASSERT_EQUAL( PdfString("Hallo World").GetStringUtf8(), PdfString(L"Hallo World").GetStringUtf8() );
+    CPPUNIT_ASSERT_EQUAL( PdfString(L"Hallo World").GetStringUtf8(), PdfString(L"Hallo World").GetStringUtf8() );
 }
 
 void StringTest::testEscapeBrackets()
