--- podofo-0.9.6/src/CMakeLists.txt.orig	2018-03-11 11:42:20.000000000 +0000
+++ podofo-0.9.6/src/CMakeLists.txt	2020-10-03 21:53:38.037101090 +0100
@@ -314,10 +314,10 @@
         MESSAGE("Pkg-config found, creating a pkg-config file for linking against shared library.")
         CONFIGURE_FILE(
           "libpodofo.pc.in"
-          "${PoDoFo_BINARY_DIR}/libpodofo-${PODOFO_VERSION_MAJOR}.pc"
+          "${PoDoFo_BINARY_DIR}/libpodofo.pc"
           @ONLY)
         INSTALL(
-          FILES "${PoDoFo_BINARY_DIR}/libpodofo-${PODOFO_VERSION_MAJOR}.pc"
+          FILES "${PoDoFo_BINARY_DIR}/libpodofo.pc"
           DESTINATION "${LIBDIRNAME}/pkgconfig")
       ELSE(PKG_CONFIG_FOUND)
         MESSAGE("Pkg-config not found. No pkg-config file will be created.")
--- podofo-0.9.6/src/libpodofo.pc.in.orig	2016-09-19 16:38:03.000000000 +0100
+++ podofo-0.9.6/src/libpodofo.pc.in	2020-10-03 21:49:58.554111562 +0100
@@ -6,5 +6,5 @@
 Name: @CMAKE_PROJECT_NAME@
 Description: A C++ library to work with the PDF file format
 Version: @PODOFO_VERSION@
-Libs: -L${libdir} -lpodofo-@PODOFO_VERSION_MAJOR@
+Libs: -L${libdir} -lpodofo
 Cflags: -I${includedir}
