--- polkit-0.112/src/polkitagent/polkitagentsession.c.orig	2013-04-29 18:28:57.000000000 +0100
+++ polkit-0.112/src/polkitagent/polkitagentsession.c	2013-10-20 10:18:09.311673641 +0100
@@ -620,7 +620,7 @@
       goto error;
     }
 
-  helper_argv[0] = PACKAGE_PREFIX "/lib/polkit-1/polkit-agent-helper-1";
+  helper_argv[0] = PACKAGE_PREFIX "/usr/lib64/polkit-1/polkit-agent-helper-1";
   helper_argv[1] = passwd->pw_name;
   helper_argv[2] = session->cookie;
   helper_argv[3] = NULL;
