--- poppler-0.5.4/qt/Makefile.in.orig	2006-09-22 01:54:24.000000000 +0100
+++ poppler-0.5.4/qt/Makefile.in	2006-10-06 12:07:19.000000000 +0100
@@ -262,7 +262,8 @@
 	$(cairo_includes)			\
 	$(POPPLER_QT_CXXFLAGS)			\
 	$(SPLASH_CFLAGS)			\
-	$(FREETYPE_CFLAGS)			
+	$(FREETYPE_CFLAGS)			\
+	$(FONTCONFIG_CFLAGS)			
 
 poppler_includedir = $(includedir)/poppler
 poppler_include_HEADERS = \
--- poppler-0.5.4/utils/Makefile.in.orig	2006-09-22 01:54:29.000000000 +0100
+++ poppler-0.5.4/utils/Makefile.in	2006-10-06 12:07:49.000000000 +0100
@@ -288,7 +288,8 @@
 	-I$(top_srcdir)				\
 	-I$(top_srcdir)/utils			\
 	-I$(top_srcdir)/poppler			\
-	$(UTILS_CFLAGS)
+	$(UTILS_CFLAGS)				\
+	$(FONTCONFIG_CFLAGS)
 
 LDADD = \
 	$(top_builddir)/poppler/libpoppler.la	\
