--- poppler-0.61.0/poppler-glib.pc.cmake.orig	2017-11-03 10:25:02.000000000 +0000
+++ poppler-0.61.0/poppler-glib.pc.cmake	2017-11-04 19:25:20.299667551 +0000
@@ -1,6 +1,6 @@
 prefix=@CMAKE_INSTALL_PREFIX@
-libdir=@CMAKE_INSTALL_LIBDIR@
-includedir=@CMAKE_INSTALL_INCLUDEDIR@
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
 
 Name: poppler-glib
 Description: GLib wrapper for poppler
--- poppler-0.61.0/poppler-qt4.pc.cmake.orig	2017-11-03 10:25:02.000000000 +0000
+++ poppler-0.61.0/poppler-qt4.pc.cmake	2017-11-04 19:25:29.115667584 +0000
@@ -1,6 +1,6 @@
 prefix=@CMAKE_INSTALL_PREFIX@
-libdir=@CMAKE_INSTALL_LIBDIR@
-includedir=@CMAKE_INSTALL_INCLUDEDIR@
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
 
 Name: poppler-qt4
 Description: Qt4 bindings for poppler
--- poppler-0.61.0/poppler-cairo.pc.cmake.orig	2017-11-03 10:25:02.000000000 +0000
+++ poppler-0.61.0/poppler-cairo.pc.cmake	2017-11-04 19:24:55.112667458 +0000
@@ -1,6 +1,6 @@
 prefix=@CMAKE_INSTALL_PREFIX@
-libdir=@CMAKE_INSTALL_LIBDIR@
-includedir=@CMAKE_INSTALL_INCLUDEDIR@
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
 
 Name: poppler-cairo
 Description: Cairo backend for Poppler PDF rendering library
--- poppler-0.61.0/poppler/SplashOutputDev.cc.orig	2017-11-03 10:25:02.000000000 +0000
+++ poppler-0.61.0/poppler/SplashOutputDev.cc	2017-11-04 19:11:59.590664601 +0000
@@ -1573,6 +1573,7 @@
 
 void SplashOutputDev::restoreState(GfxState *state) {
   splash->restoreState();
+  updateCTM(state, 0, 0, 0, 0, 0, 0);
   needFontUpdate = gTrue;
   if (t3GlyphStack && !t3GlyphStack->haveDx) {
     t3GlyphStack->doNotCache = gTrue;
--- poppler-0.61.0/poppler-splash.pc.cmake.orig	2017-11-03 10:25:02.000000000 +0000
+++ poppler-0.61.0/poppler-splash.pc.cmake	2017-11-04 19:26:36.828667833 +0000
@@ -1,6 +1,6 @@
 prefix=@CMAKE_INSTALL_PREFIX@
-libdir=@CMAKE_INSTALL_LIBDIR@
-includedir=@CMAKE_INSTALL_INCLUDEDIR@
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
 
 Name: poppler-splash
 Description: Splash backend for Poppler PDF rendering library
--- poppler-0.61.0/poppler-cpp.pc.cmake.orig	2017-11-03 10:25:02.000000000 +0000
+++ poppler-0.61.0/poppler-cpp.pc.cmake	2017-11-04 19:25:08.586667508 +0000
@@ -1,6 +1,6 @@
 prefix=@CMAKE_INSTALL_PREFIX@
-libdir=@CMAKE_INSTALL_LIBDIR@
-includedir=@CMAKE_INSTALL_INCLUDEDIR@
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
 
 Name: poppler-cpp
 Description: cpp backend for Poppler PDF rendering library
--- poppler-0.61.0/poppler.pc.cmake.orig	2017-11-03 10:25:02.000000000 +0000
+++ poppler-0.61.0/poppler.pc.cmake	2017-11-04 19:26:46.984667871 +0000
@@ -1,6 +1,6 @@
 prefix=@CMAKE_INSTALL_PREFIX@
-libdir=@CMAKE_INSTALL_LIBDIR@
-includedir=@CMAKE_INSTALL_INCLUDEDIR@
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
 
 Name: poppler
 Description: PDF rendering library
--- poppler-0.61.0/poppler-qt5.pc.cmake.orig	2017-11-03 10:25:02.000000000 +0000
+++ poppler-0.61.0/poppler-qt5.pc.cmake	2017-11-04 19:25:59.500667696 +0000
@@ -1,6 +1,6 @@
 prefix=@CMAKE_INSTALL_PREFIX@
-libdir=CMAKE_INSTALL_LIBDIR@
-includedir=@CMAKE_INSTALL_INCLUDEDIR@
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
 
 Name: poppler-qt5
 Description: Qt5 bindings for poppler
