--- poppler-24.11.0/CMakeLists.txt.orig	2024-11-02 22:33:19.000000000 +0000
+++ poppler-24.11.0/CMakeLists.txt	2024-11-04 13:54:51.958965616 +0000
@@ -729,6 +729,9 @@
     poppler/BBoxOutputDev.h
     poppler/UTF.h
     poppler/Sound.h
+    poppler/ViewerPreferences.h
+    poppler/CharCodeToUnicode.h
+    poppler/FontEncodingTables.h
     ${CMAKE_CURRENT_BINARY_DIR}/poppler_private_export.h
     poppler/SplashOutputDev.h
     )
