--- poppler-0.71.0/poppler/SplashOutputDev.cc.orig	2018-11-01 18:21:20.691798195 +0000
+++ poppler-0.71.0/poppler/SplashOutputDev.cc	2018-11-01 19:12:11.883513995 +0000
@@ -1576,6 +1576,7 @@
 
 void SplashOutputDev::restoreState(GfxState *state) {
   splash->restoreState();
+  updateCTM(state, 0, 0, 0, 0, 0, 0);
   needFontUpdate = true;
   if (t3GlyphStack && !t3GlyphStack->haveDx) {
     t3GlyphStack->doNotCache = true;
