--- poppler-24.12.0/CMakeLists.txt.orig	2024-12-04 16:31:49.309056216 +0000
+++ poppler-24.12.0/CMakeLists.txt	2024-12-04 16:32:52.354060449 +0000
@@ -725,6 +725,9 @@
     poppler/BBoxOutputDev.h
     poppler/UTF.h
     poppler/Sound.h
+    poppler/ViewerPreferences.h
+    poppler/CharCodeToUnicode.h
+    poppler/FontEncodingTables.h
     poppler/SplashOutputDev.h
     )
   set(poppler_goo_installed_headers
