--- ppracer-0.3.1/src/ppgltk/FT/FTTextureGlyph.h.orig	2005-02-18 10:16:04.000000000 +0000
+++ ppracer-0.3.1/src/ppgltk/FT/FTTextureGlyph.h	2008-12-16 08:57:55.000000000 +0000
@@ -52,7 +52,7 @@
          * Reset the currently active texture to zero to get into a known state before
          * drawing a string. This is to get round possible threading issues.
          */
-        static void FTTextureGlyph::ResetActiveTexture(){ activeTextureID = 0;}
+        static void ResetActiveTexture(){ activeTextureID = 0;}
         
     private:
         /**
