--- presage-0.9.1/apps/gtk/gpresagemate/Makefile.in.orig	2015-04-21 12:40:55.000000000 +0100
+++ presage-0.9.1/apps/gtk/gpresagemate/Makefile.in	2015-08-21 09:45:45.289086521 +0100
@@ -270,7 +270,7 @@
 top_srcdir = @top_srcdir@
 @BUILD_GTK_PRESAGE_MATE_TRUE@gpresagemate_SOURCES = gpresagemate.cpp
 @BUILD_GTK_PRESAGE_MATE_TRUE@gpresagemate_CPPFLAGS = $(GNOME_CFLAGS) $(GTHREAD_CFLAGS) -I$(top_srcdir)/src/lib
-@BUILD_GTK_PRESAGE_MATE_TRUE@gpresagemate_LDADD = $(GNOME_LIBS) $(GTHREAD_LIBS) -lXevie ../../../src/lib/libpresage.la
+@BUILD_GTK_PRESAGE_MATE_TRUE@gpresagemate_LDADD = $(GNOME_LIBS) $(GTHREAD_LIBS) -lXevie -lX11 ../../../src/lib/libpresage.la
 all: all-am
 
 .SUFFIXES:
