--- procps-3.2.7/sysctl.c.old	2007-04-25 16:38:30.000000000 +0200
+++ procps-3.2.7/sysctl.c	2007-04-25 16:39:12.000000000 +0200
@@ -174,7 +174,7 @@
    }
 
    /* used to open the file */
-   tmpname = malloc(strlen(name)+strlen(PROC_PATH)+1);
+   tmpname = malloc(strlen(name)+strlen(PROC_PATH)+2);
    strcpy(tmpname, PROC_PATH);
    strcat(tmpname, name); 
    slashdot(tmpname+strlen(PROC_PATH),'.','/'); /* change . to / */
