--- procinfo-17/Makefile.mandir	Sun Jun 18 16:19:08 2000
+++ procinfo-17/Makefile	Sun Jun 18 16:19:49 2000
@@ -1,6 +1,7 @@
 ### Makefile for procinfo.
 
 prefix=/usr
+mandir=$(prefix)/man
 
 CC = gcc -Wall -Wstrict-prototypes
 
@@ -33,13 +34,13 @@
 
 install: procinfo procinfo.8 lsdev.pl socklist.pl lsdev.8
 	-mkdir -p $(prefix)/bin
-	install -m 755 -o 0 -g 0 procinfo $(prefix)/bin/procinfo
-	install -m 755 -o 0 -g 0 lsdev.pl $(prefix)/bin/lsdev
-	install -m 755 -o 0 -g 0 socklist.pl $(prefix)/bin/socklist
-	-mkdir -p $(prefix)/man/man8
-	install -m 644  procinfo.8 $(prefix)/man/man8/procinfo.8
-	install -m 644  lsdev.8 $(prefix)/man/man8/lsdev.8
-	install -m 644  socklist.8 $(prefix)/man/man8/socklist.8
+	install -m 755 procinfo $(prefix)/bin/procinfo
+	install -m 755 lsdev.pl $(prefix)/bin/lsdev
+	install -m 755 socklist.pl $(prefix)/bin/socklist
+	-mkdir -p $(mandir)
+	install -m 644  procinfo.8 $(mandir)/man8/procinfo.8
+	install -m 644  lsdev.8 $(mandir)/man8/lsdev.8
+	install -m 644  socklist.8 $(mandir)/man8/socklist.8
 
 clean:
 	rm -f procinfo procinfo.0 *.o *~ out
