--- projectM-2.1.0/src/projectM-test/CMakeLists.txt.orig	2012-05-20 04:43:26.000000000 +0100
+++ projectM-2.1.0/src/projectM-test/CMakeLists.txt	2012-06-29 21:56:44.000000000 +0100
@@ -38,8 +38,8 @@
 INCLUDE_DIRECTORIES(${PROJECTM_INCLUDE} ${SDL_INCLUDE_DIR} ${GTK_INCLUDE_DIR})
 LINK_DIRECTORIES(${PROJECTM_LINK})
 
-TARGET_LINK_LIBRARIES(projectM-test projectM  ${SDL_LIBRARY})
-TARGET_LINK_LIBRARIES(projectM-test-memleak projectM  ${SDL_LIBRARY} )
-TARGET_LINK_LIBRARIES(projectM-test-texture projectM  ${SDL_LIBRARY} )
+TARGET_LINK_LIBRARIES(projectM-test projectM  ${SDL_LIBRARY} ${OPENGL_gl_LIBRARY} )
+TARGET_LINK_LIBRARIES(projectM-test-memleak projectM  ${SDL_LIBRARY} ${OPENGL_gl_LIBRARY} )
+TARGET_LINK_LIBRARIES(projectM-test-texture projectM  ${SDL_LIBRARY} ${OPENGL_gl_LIBRARY} )
 
 INSTALL(TARGETS projectM-test projectM-test-texture DESTINATION ${CMAKE_INSTALL_PREFIX}/bin )
--- projectM-2.1.0/src/libprojectM/CMakeLists.txt.orig	2012-05-20 04:43:26.000000000 +0100
+++ projectM-2.1.0/src/libprojectM/CMakeLists.txt	2012-06-30 18:06:36.000000000 +0100
@@ -33,7 +33,7 @@
 set(PROJECTM_ABI_LEVEL 2)
 #
 
-ADD_DEFINITIONS(-DCMAKE_INSTALL_PREFIX="\\\"${CMAKE_INSTALL_PREFIX}\\\"")
+ADD_DEFINITIONS(-DCMAKE_INSTALL_PREFIX="${CMAKE_INSTALL_PREFIX}")
 
 SET(LIB_INSTALL_DIR "\${exec_prefix}/lib"
   CACHE PATH "The subdirectory relative to the install prefix where libraries will be installed (default is /lib)"
