--- pstoedit-3.45/src/drvswf.cpp.orig	2007-07-22 12:27:36.000000000 +0100
+++ pstoedit-3.45/src/drvswf.cpp	2008-12-17 20:47:33.000000000 +0000
@@ -373,7 +373,7 @@
 		printf("s->end();\n" "SWFDisplayItem * d = movie->add(s);\n" "d->move(0, 0); }\n");
 
 // delete s;
-	delete d;
+	//delete d;
 
 }
 
@@ -474,7 +474,7 @@
 	SWFDisplayItem_setMatrix(d->item, ma, mb, mc, md, mx, my);
 
 #endif
-	delete d;
+	//delete d;
 	// delete f;  // causes memory problems otherwise - at the cost of leaks - Hmmm
 	// delete t;
 
