--- pstoedit-3.70/src/pstoedit.cpp.orig	2014-09-04 19:21:45.000000000 +0100
+++ pstoedit-3.70/src/pstoedit.cpp	2015-01-02 15:21:15.231934275 +0000
@@ -275,7 +275,7 @@
 		}
 	}
 	// now try also $exepath/../lib/pstoedit
-	strcat_s(szExePath,1000,"/../lib/pstoedit");
+	strcat_s(szExePath,1000,"/../lib64/pstoedit");
 	if (!strequal(szExePath, plugindir.c_str())) {
     	loadPlugInDrivers(szExePath, errstream,verbose);
 		pluginsloaded = true;
