--- pstoedit-3.50/src/psimage.cpp.orig	2009-06-21 16:10:31.000000000 +0100
+++ pstoedit-3.50/src/psimage.cpp	2009-08-12 07:03:27.000000000 +0100
@@ -20,10 +20,6 @@
     Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
 */
-#ifdef HAVE_LIBGD
-#include <gd.h>
-#endif
-
 #include <cmath>
 
 #include "drvbase.h"
@@ -32,6 +28,10 @@
  
 #include I_iomanip
 
+#ifdef HAVE_LIBGD
+#include <gd.h>
+#endif
+
 #if 0
 // should be obsolete - since now PNG images are written directly by ghostscript
 
--- pstoedit-3.50/src/drvswf.cpp.orig	2009-08-12 06:49:59.000000000 +0100
+++ pstoedit-3.50/src/drvswf.cpp	2009-08-12 07:31:54.000000000 +0100
@@ -32,7 +32,7 @@
 //static bool cubic = false;
 
 #include "mingpp.h"
-#include "ming_config.h"
+//#include "ming_config.h"
 
 #if defined(_WIN32)
 extern "C" {
