--- pulseaudio-1.0/src/pulse/context.c.orig	2011-09-01 09:36:46.000000000 +0100
+++ pulseaudio-1.0/src/pulse/context.c	2011-10-19 19:33:09.000000000 +0100
@@ -1281,7 +1281,7 @@
 }
 
 const char* pa_get_library_version(void) {
-    return PACKAGE_VERSION;
+    return pa_get_headers_version();
 }
 
 const char* pa_context_get_server(pa_context *c) {
--- pulseaudio-1.0/src/pulse/version.h.in.orig	2011-06-24 00:29:11.000000000 +0100
+++ pulseaudio-1.0/src/pulse/version.h.in	2011-10-19 19:33:09.000000000 +0100
@@ -35,7 +35,7 @@
 /** Return the version of the header files. Keep in mind that this is
 a macro and not a function, so it is impossible to get the pointer of
 it. */
-#define pa_get_headers_version() ("@PACKAGE_VERSION@")
+#define pa_get_headers_version() ("@PA_MAJOR@.@PA_MINOR@.0")
 
 /** Return the version of the library the current application is
  * linked to. */
