--- pulseeffects-4.5.9/src/application_ui.cpp.orig	2019-04-13 19:28:02.000000000 +0100
+++ pulseeffects-4.5.9/src/application_ui.cpp	2019-04-21 22:05:13.010934700 +0100
@@ -59,13 +59,13 @@
 
   connections.push_back(app->pm->new_default_sink.connect([&](auto name) {
     if (stack->get_visible_child_name() == "sink_inputs") {
-      update_headerbar_subtitle(0);
+      this->update_headerbar_subtitle(0);
     }
   }));
 
   connections.push_back(app->pm->new_default_source.connect([&](auto name) {
     if (stack->get_visible_child_name() == "source_outputs") {
-      update_headerbar_subtitle(1);
+      this->update_headerbar_subtitle(1);
     }
   }));
 
--- pulseeffects-4.5.9/src/blacklist_settings_ui.cpp.orig	2019-04-13 19:28:02.000000000 +0100
+++ pulseeffects-4.5.9/src/blacklist_settings_ui.cpp	2019-04-21 22:06:59.183929634 +0100
@@ -52,12 +52,12 @@
 
   connections.push_back(
       settings->signal_changed("blacklist-in").connect([&](auto key) {
-        populate_blacklist_in_listbox();
+        this->populate_blacklist_in_listbox();
       }));
 
   connections.push_back(
       settings->signal_changed("blacklist-out").connect([&](auto key) {
-        populate_blacklist_out_listbox();
+        this->populate_blacklist_out_listbox();
       }));
 
   populate_blacklist_in_listbox();
--- pulseeffects-4.5.9/src/general_settings_ui.cpp.orig	2019-04-13 19:28:02.000000000 +0100
+++ pulseeffects-4.5.9/src/general_settings_ui.cpp	2019-04-21 22:04:52.668935671 +0100
@@ -72,7 +72,7 @@
 
   connections.push_back(
       settings->signal_changed("priority-type").connect([&](auto key) {
-        set_priority_controls_visibility();
+        this->set_priority_controls_visibility();
 
         app->sie->set_null_pipeline();
         app->soe->set_null_pipeline();
--- pulseeffects-4.5.9/src/convolver_ui.cpp.orig	2019-04-13 19:28:02.000000000 +0100
+++ pulseeffects-4.5.9/src/convolver_ui.cpp	2019-04-21 22:08:38.561924893 +0100
@@ -121,7 +121,7 @@
       settings->signal_changed("kernel-path").connect([=](auto key) {
         auto f = [=]() {
           std::lock_guard<std::mutex> lock(lock_guard_irs_info);
-          get_irs_info();
+          this->get_irs_info();
         };
 
         auto future = std::async(std::launch::async, f);
--- pulseeffects-4.5.9/src/equalizer_ui.cpp.orig	2019-04-13 19:28:02.000000000 +0100
+++ pulseeffects-4.5.9/src/equalizer_ui.cpp	2019-04-21 22:07:57.270926863 +0100
@@ -227,10 +227,10 @@
         bool split = settings->get_boolean("split-channels");
 
         if (split) {
-          build_bands(bands_grid_left, settings_left, nbands->get_value());
-          build_bands(bands_grid_right, settings_right, nbands->get_value());
+          this->build_bands(bands_grid_left, settings_left, nbands->get_value());
+          this->build_bands(bands_grid_right, settings_right, nbands->get_value());
         } else {
-          build_unified_bands(nbands->get_value());
+          this->build_unified_bands(nbands->get_value());
         }
       }));
 
--- pulseeffects-4.5.9/src/spectrum_ui.cpp.orig	2019-04-13 19:28:02.000000000 +0100
+++ pulseeffects-4.5.9/src/spectrum_ui.cpp	2019-04-21 22:06:22.138931402 +0100
@@ -24,16 +24,16 @@
 
   connections.push_back(
       settings->signal_changed("use-custom-color").connect([&](auto key) {
-        init_color();
-        init_gradient_color();
+        this->init_color();
+        this->init_gradient_color();
       }));
 
   connections.push_back(settings->signal_changed("color").connect(
-      [&](auto key) { init_color(); }));
+      [&](auto key) { this->init_color(); }));
 
   connections.push_back(
       settings->signal_changed("gradient-color").connect([&](auto key) {
-        init_gradient_color();
+        this->init_gradient_color();
       }));
 
   connections.push_back(
--- pulseeffects-4.5.9/include/pulse_manager.hpp.orig	2019-04-13 19:28:02.000000000 +0100
+++ pulseeffects-4.5.9/include/pulse_manager.hpp	2019-04-21 22:03:32.281939506 +0100
@@ -115,7 +115,7 @@
   pa_mainloop_api* main_loop_api = nullptr;
   pa_context* context = nullptr;
 
-  std::array<std::string, 10> blacklist_apps = {"PulseEffects",
+  std::array<std::string, 10> blacklist_apps = {{"PulseEffects",
                                                 "pulseeffects",
                                                 "PulseEffectsWebrtcProbe",
                                                 "gsd-media-keys",
@@ -124,13 +124,13 @@
                                                 "gnome-pomodoro",
                                                 "PulseAudio Volume Control",
                                                 "Screenshot",
-                                                "speech-dispatcher"};
+                                                "speech-dispatcher"}};
 
-  std::array<std::string, 5> blacklist_media_name = {
+  std::array<std::string, 5> blacklist_media_name = {{
       "pulsesink probe", "bell-window-system", "audio-volume-change",
-      "Peak detect", "screen-capture"};
+      "Peak detect", "screen-capture"}};
 
-  std::array<std::string, 1> blacklist_media_role = {"event"};
+  std::array<std::string, 1> blacklist_media_role = {{"event"}};
 
   static void context_state_cb(pa_context* ctx, void* data);
 
