--- ./waflib/Tools/python.py.orig	2011-03-30 07:17:51.000000000 +0100
+++ ./waflib/Tools/python.py	2016-05-19 21:09:56.591112985 +0100
@@ -169,7 +169,7 @@
 		conf.find_program('python-config-%s'%num,var='PYTHON_CONFIG',mandatory=False)
 	includes=[]
 	if conf.env.PYTHON_CONFIG:
-		for incstr in conf.cmd_and_log(conf.env.PYTHON+[conf.env.PYTHON_CONFIG,'--includes']).strip().split():
+		for incstr in conf.cmd_and_log([conf.env.PYTHON_CONFIG,'--includes']).strip().split():
 			if(incstr.startswith('-I')or incstr.startswith('/I')):
 				incstr=incstr[2:]
 			if incstr not in includes:
--- ./waflib/Build.py.orig	2016-05-19 21:10:17.759113063 +0100
+++ ./waflib/Build.py	2016-05-19 21:07:23.357112420 +0100
@@ -151,6 +151,7 @@
 				f.close()
 		self.init_dirs()
 	def store(self):
+		return
 		data={}
 		for x in SAVED_ATTRS:
 			data[x]=getattr(self,x)
