--- qantenna-0.3.0/src/src.pro.orig	2014-03-09 01:47:05.000000000 +0000
+++ qantenna-0.3.0/src/src.pro	2024-08-24 14:41:23.491628547 +0100
@@ -122,6 +122,7 @@
 
 TEMPLATE = app
 QT += opengl
+QT += core
 LIBS += -lGLU
 
 # Sub-directories
--- qantenna-0.3.0/src/point4f.h.orig	2014-03-09 01:47:05.000000000 +0000
+++ qantenna-0.3.0/src/point4f.h	2024-08-24 14:44:03.694639303 +0100
@@ -81,6 +81,7 @@
   Point4F stream functions
  *****************************************************************************/
 #ifndef QT_NO_DATASTREAM
+#include <QDataStream>
 Q_CORE_EXPORT QDataStream &operator<<(QDataStream &, const Point4F &);
 Q_CORE_EXPORT QDataStream &operator>>(QDataStream &, Point4F &);
 #endif
--- qantenna-0.3.0/src/camera.h.orig	2014-03-09 01:47:05.000000000 +0000
+++ qantenna-0.3.0/src/camera.h	2024-08-24 14:42:24.382632635 +0100
@@ -140,7 +140,7 @@
 	/// Moving speed
 	GLfloat step;
 	bool run;
-	static const float runDelta= 0.04;
+	static constexpr float runDelta= 0.04;
 
 	/** Camera orbitation around origin, these coords are applied to
 		{x,y,z}Rot with commitSpherical()
--- qantenna-0.3.0/src/glwidget.h.orig	2014-03-09 01:47:05.000000000 +0000
+++ qantenna-0.3.0/src/glwidget.h	2024-08-24 14:44:46.534642179 +0100
@@ -84,7 +84,7 @@
 	Camera camera;
 
 	/// Step for changing camera.rhoOrbit (with the mouse wheel)
-	static const float zoomStep= 0.1;
+	static constexpr float zoomStep= 0.1;
 
 	/// sceneTimer is used to simulate a fast auto-repeat event and
 	/// as a scene timer
@@ -95,7 +95,7 @@
 	bool movingTo[movingDirections];
 
 	/// Mouse Grabbing information
-	static const float sensibility= 0.1;
+	static constexpr float sensibility= 0.1;
 	bool mouseGrabbed;
 	/// Did the camera moved while the input was grabbed
 	bool movedWhileGrabbed;
