--- qca-2.0.3/src/botantools/botan/botan/secmem.h.orig	2007-04-19 22:26:13.000000000 +0100
+++ qca-2.0.3/src/botantools/botan/botan/secmem.h	2012-06-10 14:33:43.000000000 +0100
@@ -214,7 +214,7 @@
 
       SecureVector(u32bit n = 0) { MemoryRegion<T>::init(true, n); }
       SecureVector(const T in[], u32bit n)
-         { MemoryRegion<T>::init(true); set(in, n); }
+         { MemoryRegion<T>::init(true); this->set(in, n); }
       SecureVector(const MemoryRegion<T>& in)
          { MemoryRegion<T>::init(true); set(in); }
       SecureVector(const MemoryRegion<T>& in1, const MemoryRegion<T>& in2)
