--- qcad-3.26.4.7/src/3rdparty/qt-labs-qtscriptgenerator-5.15.3/qt-labs-qtscriptgenerator-5.15.3.pro.orig	2021-09-02 22:14:16.378965930 +0100
+++ qcad-3.26.4.7/src/3rdparty/qt-labs-qtscriptgenerator-5.15.3/qt-labs-qtscriptgenerator-5.15.3.pro	2021-08-24 14:43:23.000000000 +0100
@@ -0,0 +1,4 @@
+include( ../../../shared.pri )
+
+SUBDIRS = ../qt-labs-qtscriptgenerator-5.5.0/qtbindings
+TEMPLATE = subdirs
