--- qcad-2.0.5.0/qcadactions/src/rs_actionzoompan.cpp.orig	2005-11-22 11:51:46.000000000 +0000
+++ qcad-2.0.5.0/qcadactions/src/rs_actionzoompan.cpp	2010-06-20 09:12:27.000000000 +0100
@@ -27,7 +27,7 @@
 #include "rs_actionzoompan.h"
 #include "rs_snapper.h"
 #include "rs_point.h"
-
+#include <cstdlib>
 
 RS_ActionZoomPan::RS_ActionZoomPan(RS_EntityContainer& container,
                                    RS_GraphicView& graphicView)
--- qcad-2.0.5.0/qcadlib/src/information/rs_information.cpp.orig	2005-11-22 11:52:39.000000000 +0000
+++ qcad-2.0.5.0/qcadlib/src/information/rs_information.cpp	2010-06-20 09:04:48.000000000 +0100
@@ -28,6 +28,7 @@
 
 #include "rs_constructionline.h"
 
+#include <cstdlib>
 
 /**
  * Default constructor.
