--- qpdfview-0.5/qpdfview.pri.orig	2023-01-28 10:40:47.229000000 +0000
+++ qpdfview-0.5/qpdfview.pri	2024-03-20 18:13:14.777464987 +0000
@@ -12,4 +12,4 @@
 win32:include(qpdfview_win32.pri)
 os2:include(qpdfview_os2.pri)
 
-CONFIG += c++11
+CONFIG += c++17
--- qpdfview-0.5/sources/fitzmodel.cpp.orig	2023-01-28 10:40:47.229000000 +0000
+++ qpdfview-0.5/sources/fitzmodel.cpp	2024-03-20 18:17:06.272480529 +0000
@@ -34,6 +34,7 @@
 #include <mupdf/fitz/bidi.h>
 #include <mupdf/fitz/output.h>
 #include <mupdf/fitz/display-list.h>
+#include <mupdf/fitz/archive.h>
 #include <mupdf/fitz/document.h>
 #include <mupdf/fitz/pool.h>
 #include <mupdf/fitz/structured-text.h>
--- qpdfview-0.5/fitz-plugin.pro.orig	2024-03-20 18:18:27.685485995 +0000
+++ qpdfview-0.5/fitz-plugin.pro	2024-03-20 18:19:54.857491848 +0000
@@ -21,7 +21,7 @@
 DEFINES += $$FITZ_PLUGIN_DEFINES
 INCLUDEPATH += $$FITZ_PLUGIN_INCLUDEPATH
 
-isEmpty(FITZ_PLUGIN_LIBS):FITZ_PLUGIN_LIBS = -lmupdf -lmupdf-third -ljpeg
+isEmpty(FITZ_PLUGIN_LIBS):FITZ_PLUGIN_LIBS = -lmupdf -lmujs -ljpeg
 LIBS += $$FITZ_PLUGIN_LIBS
 
 !static_fitz_plugin {
