--- qt-creator-14.0.2/src/plugins/clangformat/clangformatutils.cpp.orig	2024-09-27 02:17:48.000000000 +0100
+++ qt-creator-14.0.2/src/plugins/clangformat/clangformatutils.cpp	2024-10-18 15:56:17.695326273 +0100
@@ -117,7 +117,7 @@
     style.IndentWrappedFunctionNames = false;
     style.JavaScriptQuotes = FormatStyle::JSQS_Leave;
     style.JavaScriptWrapImports = true;
-    style.KeepEmptyLinesAtTheStartOfBlocks = false;
+    style.KeepEmptyLines.AtStartOfBlock = false;
     // Do not add QT_BEGIN_NAMESPACE/QT_END_NAMESPACE as this will indent lines in between.
     style.MacroBlockBegin = "";
     style.MacroBlockEnd = "";
