--- qt-creator-11.0.2/cmake/Findyaml-cpp.cmake.orig	2023-08-14 02:53:08.000000000 +0100
+++ qt-creator-11.0.2/cmake/Findyaml-cpp.cmake	2023-08-14 22:32:00.735893470 +0100
@@ -15,7 +15,7 @@
     unset(yaml_cpp_include_dir CACHE)
     find_path(yaml_cpp_include_dir yaml-cpp/yaml.h)
   endif()
-  set_target_properties(yaml-cpp PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${yaml_cpp_include_dir}")
+#  set_target_properties(yaml-cpp PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${yaml_cpp_include_dir}")
 else()
   if(TARGET yaml-cpp)
     return()
