--- qt-creator-4.14.0/src/shared/qbs/cmake/QbsAPI.cmake.orig	2020-12-17 08:52:28.000000000 +0000
+++ qt-creator-4.14.0/src/shared/qbs/cmake/QbsAPI.cmake	2021-02-19 14:30:16.374275461 +0000
@@ -1,6 +1,6 @@
 set(QBS_APP_INSTALL_DIR "bin" CACHE STRING "Relative install location for Qbs binaries.")
 # default paths
-set(QBS_LIBDIR_NAME "lib")
+set(QBS_LIBDIR_NAME "lib${LIB_SUFFIX}")
 if(WIN32)
     set(_DEFAULT_LIB_INSTALL_DIR ${QBS_APP_INSTALL_DIR})
     set(_DEFAULT_LIBEXEC_INSTALL_DIR ${QBS_APP_INSTALL_DIR})
--- qt-creator-4.14.0/cmake/QtCreatorAPIInternal.cmake.orig	2020-12-17 08:52:31.000000000 +0000
+++ qt-creator-4.14.0/cmake/QtCreatorAPIInternal.cmake	2021-02-19 14:38:32.167251805 +0000
@@ -51,17 +51,17 @@
   set(_IDE_APP_PATH "bin")
   set(_IDE_APP_TARGET "${IDE_ID}")
 
-  set(_IDE_LIBRARY_BASE_PATH "lib")
-  set(_IDE_LIBRARY_PATH "lib/qtcreator")
-  set(_IDE_PLUGIN_PATH "lib/qtcreator/plugins")
+  set(_IDE_LIBRARY_BASE_PATH "lib${LIB_SUFFIX}")
+  set(_IDE_LIBRARY_PATH "lib${LIB_SUFFIX}/qtcreator")
+  set(_IDE_PLUGIN_PATH "lib${LIB_SUFFIX}/qtcreator/plugins")
   if (WIN32)
     set(_IDE_LIBEXEC_PATH "bin")
     set(QT_DEST_PLUGIN_PATH "bin/plugins")
     set(QT_DEST_QML_PATH "bin/qml")
   else ()
     set(_IDE_LIBEXEC_PATH "libexec/qtcreator")
-    set(QT_DEST_PLUGIN_PATH  "lib/Qt/plugins")
-    set(QT_DEST_QML_PATH "lib/Qt/qml")
+    set(QT_DEST_PLUGIN_PATH  "lib${LIB_SUFFIX}/Qt/plugins")
+    set(QT_DEST_QML_PATH "lib${LIB_SUFFIX}/Qt/qml")
   endif ()
   set(_IDE_DATA_PATH "share/qtcreator")
   set(_IDE_DOC_PATH "share/doc/qtcreator")
