--- qt-creator-4.10.0/src/plugins/clangformat/clangformatutils.cpp.orig	2019-09-03 15:36:38.000000000 +0100
+++ qt-creator-4.10.0/src/plugins/clangformat/clangformatutils.cpp	2019-10-03 21:40:39.623754140 +0100
@@ -60,7 +60,7 @@
     style.AllowShortBlocksOnASingleLine = false;
     style.AllowShortCaseLabelsOnASingleLine = false;
     style.AllowShortFunctionsOnASingleLine = FormatStyle::SFS_Inline;
-    style.AllowShortIfStatementsOnASingleLine = false;
+    style.AllowShortIfStatementsOnASingleLine = FormatStyle::SIS_Never;
     style.AllowShortLoopsOnASingleLine = false;
     style.AlwaysBreakAfterReturnType = FormatStyle::RTBS_None;
     style.AlwaysBreakBeforeMultilineStrings = false;
