--- qt-creator-7.0.2/src/plugins/texteditor/textdocumentlayout.h.orig	2022-05-23 03:22:02.000000000 +0100
+++ qt-creator-7.0.2/src/plugins/texteditor/textdocumentlayout.h	2022-06-29 20:30:32.571481720 +0100
@@ -32,7 +32,7 @@
 
 #include <utils/id.h>
 
-#include <State>
+#include <KSyntaxHighlighting/State>
 
 #include <QTextBlockUserData>
 #include <QPlainTextDocumentLayout>
--- qt-creator-7.0.2/src/plugins/texteditor/highlighter.h.orig	2022-05-23 03:22:02.000000000 +0100
+++ qt-creator-7.0.2/src/plugins/texteditor/highlighter.h	2022-06-29 20:29:15.196485412 +0100
@@ -29,8 +29,8 @@
 
 #include <utils/fileutils.h>
 
-#include <AbstractHighlighter>
-#include <Definition>
+#include <KSyntaxHighlighting/AbstractHighlighter>
+#include <KSyntaxHighlighting/Definition>
 
 namespace TextEditor {
 class TextDocument;
