--- qt-everywhere-src-6.4.0/qttools/cmake/FindWrapLibClang.cmake.orig	2022-09-23 12:50:40.000000000 +0100
+++ qt-everywhere-src-6.4.0/qttools/cmake/FindWrapLibClang.cmake	2022-10-29 22:40:33.053961966 +0100
@@ -58,7 +58,7 @@
         qt_internal_disable_find_package_global_promotion(Threads::Threads)
         # lupdate must also link to LLVM when using clang-cpp
         set(__qt_clang_genex_condition "$<AND:$<TARGET_EXISTS:clang-cpp>,$<TARGET_EXISTS:LLVM>>")
-        set(__qt_clang_genex "$<IF:${__qt_clang_genex_condition},clang-cpp;LLVM,clangHandleCXX>")
+        set(__qt_clang_genex "$<IF:${__qt_clang_genex_condition},clang-cpp;LLVM,clangHandleCXX;clangTooling;clangFrontend;clangSerialization;clangAST;clangLex;clangBasic;LLVMSupport>")
         target_link_libraries(WrapLibClang::WrapLibClang
             INTERFACE libclang
             "${__qt_clang_genex}"
