--- qt-everywhere-opensource-src-5.4.0/qtwebengine/src/3rdparty/chromium/build/common.gypi.orig	2014-12-05 16:25:33.000000000 +0000
+++ qt-everywhere-opensource-src-5.4.0/qtwebengine/src/3rdparty/chromium/build/common.gypi	2014-12-24 08:15:22.307253261 +0000
@@ -3309,7 +3309,7 @@
     ['os_posix==1 and disable_fatal_linker_warnings==0 and use_evdev_gestures==0 and (chromeos==0 or target_arch!="arm")', {
       'target_defaults': {
         'ldflags': [
-          '-Wl,--fatal-warnings',
+          '-Wl,--no-fatal-warnings',
         ],
       },
     }],
