--- qt-everywhere-src-5.10.0/gnuwin32/lib/libiconv.la.orig	2017-12-01 19:24:55.000000000 +0000
+++ qt-everywhere-src-5.10.0/gnuwin32/lib/libiconv.la	2017-12-01 19:24:55.000000000 +0000
@@ -32,4 +32,4 @@
 dlpreopen=''
 
 # Directory that this library needs to be installed in:
-libdir='c:/progra~1/LibIconv/lib'
+libdir='C:/Users/marius/DOWNLO~1/WEBKIT~1/install/lib'
--- qt-everywhere-src-5.10.0/gnuwin32/lib/libcharset.la.orig	2017-12-01 19:24:55.000000000 +0000
+++ qt-everywhere-src-5.10.0/gnuwin32/lib/libcharset.la	2017-12-01 19:24:55.000000000 +0000
@@ -32,4 +32,4 @@
 dlpreopen=''
 
 # Directory that this library needs to be installed in:
-libdir='c:/progra~1/LibIconv/lib'
+libdir='C:/Users/marius/DOWNLO~1/WEBKIT~1/install/lib'
--- qt-everywhere-src-5.10.0/qtmultimedia/src/gsttools/gsttools.pro.orig	2017-11-28 10:54:28.000000000 +0000
+++ qt-everywhere-src-5.10.0/qtmultimedia/src/gsttools/gsttools.pro	2017-12-09 09:35:22.524562569 +0000
@@ -10,6 +10,9 @@
 qtConfig(alsa): \
     QMAKE_USE += alsa
 
+qtConfig(alsa): \
+    LIBS += -lasound
+
 QMAKE_USE += gstreamer
 
 qtConfig(resourcepolicy): \
