+++ qt-everywhere-src-5.10.1/qtmultimedia/src/gsttools/gsttools.pro	2018-06-17 12:30:36.751398221 +0100
@@ -10,6 +10,9 @@
 qtConfig(alsa): \
     QMAKE_USE += alsa
 
+qtConfig(alsa): \
+    LIBS += -lasound
+
 QMAKE_USE += gstreamer
 
 qtConfig(resourcepolicy): \
--- qt-everywhere-src-5.10.1/qtwebengine/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/aec3/aec_state.cc.orig	2018-02-02 10:39:52.000000000 +0000
+++ qt-everywhere-src-5.10.1/qtwebengine/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/aec3/aec_state.cc	2018-06-17 14:10:52.439009361 +0100
@@ -10,6 +10,7 @@
 
 #include "webrtc/modules/audio_processing/aec3/aec_state.h"
 
+#include <cmath>
 #include <math.h>
 #include <numeric>
 #include <vector>
