--- qt-x11-free-3.3.8/src/tools/qdatetime.cpp.orig	2007-02-02 14:01:04.000000000 +0000
+++ qt-x11-free-3.3.8/src/tools/qdatetime.cpp	2007-05-23 17:03:15.000000000 +0100
@@ -2485,7 +2485,7 @@
 	    return dt;
 	}
 #endif
-	int day = s.mid( firstSpace + 4, 2 ).simplifyWhiteSpace().toInt();
+	int day = s.mid( firstSpace + 5, 2 ).simplifyWhiteSpace().toInt();
 	int year = s.right( 4 ).toInt();
 	QDate date( year, month, day );
 	QTime time;
