--- qt-x11-free-3.3.6/mkspecs/linux-g++/qmake.conf.orig	2006-03-08 16:48:20.000000000 +0000
+++ qt-x11-free-3.3.6/mkspecs/linux-g++/qmake.conf	2006-05-03 14:40:30.000000000 +0100
@@ -21,7 +21,7 @@
 QMAKE_CFLAGS_DEPS	= -M
 QMAKE_CFLAGS_WARN_ON	= -Wall -W
 QMAKE_CFLAGS_WARN_OFF	= -w
-QMAKE_CFLAGS_RELEASE	= -O2
+QMAKE_CFLAGS_RELEASE	= -O2 -march=i486 -mtune=i686
 QMAKE_CFLAGS_DEBUG	= -g
 QMAKE_CFLAGS_SHLIB	= -fPIC
 QMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
