--- qt-everywhere-src-5.15.2/gnuwin32/lib/libiconv.la.orig	2020-11-12 15:27:08.000000000 +0000
+++ qt-everywhere-src-5.15.2/gnuwin32/lib/libiconv.la	2020-11-12 15:27:08.000000000 +0000
@@ -32,4 +32,4 @@
 dlpreopen=''
 
 # Directory that this library needs to be installed in:
-libdir='c:/progra~1/LibIconv/lib'
+libdir='C:/Users/marius/DOWNLO~1/WEBKIT~1/install/lib'
--- qt-everywhere-src-5.15.2/gnuwin32/lib/libcharset.la.orig	2020-11-12 15:27:08.000000000 +0000
+++ qt-everywhere-src-5.15.2/gnuwin32/lib/libcharset.la	2020-11-12 15:27:08.000000000 +0000
@@ -32,4 +32,4 @@
 dlpreopen=''
 
 # Directory that this library needs to be installed in:
-libdir='c:/progra~1/LibIconv/lib'
+libdir='C:/Users/marius/DOWNLO~1/WEBKIT~1/install/lib'
--- qt-everywhere-src-5.15.2/qtbase/src/gui/image/qimagereader.cpp.orig	2020-10-27 08:02:11.000000000 +0000
+++ qt-everywhere-src-5.15.2/qtbase/src/gui/image/qimagereader.cpp	2021-03-14 15:58:11.126676845 +0000
@@ -409,6 +409,7 @@
                 break;
 #endif
 #ifndef QT_NO_IMAGEFORMAT_PPM
+            case _qt_PnmFormat:
             case _qt_PbmFormat:
             case _qt_PgmFormat:
             case _qt_PpmFormat:
--- qt-everywhere-src-5.15.2/qtbase/src/gui/image/qimagereaderwriterhelpers_p.h.orig	2020-10-27 08:02:11.000000000 +0000
+++ qt-everywhere-src-5.15.2/qtbase/src/gui/image/qimagereaderwriterhelpers_p.h	2021-03-14 15:57:09.368679791 +0000
@@ -72,6 +72,7 @@
     _qt_PpmFormat,
     _qt_PgmFormat,
     _qt_PbmFormat,
+    _qt_PnmFormat,
 #endif
 #ifndef QT_NO_IMAGEFORMAT_XBM
     _qt_XbmFormat,
@@ -110,6 +111,7 @@
     {"ppm", "x-portable-pixmap"},
     {"pgm", "x-portable-graymap"},
     {"pbm", "x-portable-bitmap"},
+    {"pnm", "x-portable-anymap"},
 #endif
 #ifndef QT_NO_IMAGEFORMAT_XBM
     {"xbm", "x-xbitmap"},
