--- qt-everywhere-src-5.15.2/qtbase/src/gui/image/qimagewriter.cpp.orig	2020-10-27 08:02:11.000000000 +0000
+++ qt-everywhere-src-5.15.2/qtbase/src/gui/image/qimagewriter.cpp	2021-03-29 20:26:51.108478152 +0100
@@ -202,8 +202,8 @@
             handler->setOption(QImageIOHandler::SubType, testFormat);
 #endif
 #ifndef QT_NO_IMAGEFORMAT_PPM
-        } else if (testFormat == "pbm" || testFormat == "pbmraw" || testFormat == "pgm"
-                 || testFormat == "pgmraw" || testFormat == "ppm" || testFormat == "ppmraw") {
+        } else if (testFormat == "pbm" || testFormat == "pbmraw" || testFormat == "pgm" || testFormat == "pnm"
+                 || testFormat == "pgmraw" || testFormat == "ppm" || testFormat == "ppmraw"  || testFormat == "pnmraw") {
             handler = new QPpmHandler;
             handler->setOption(QImageIOHandler::SubType, testFormat);
 #endif
--- qt-everywhere-src-5.15.2/qtbase/src/gui/image/qimagereader.cpp.orig	2020-10-27 08:02:11.000000000 +0000
+++ qt-everywhere-src-5.15.2/qtbase/src/gui/image/qimagereader.cpp	2021-03-14 15:58:11.126676845 +0000
@@ -409,6 +409,7 @@
                 break;
 #endif
 #ifndef QT_NO_IMAGEFORMAT_PPM
+            case _qt_PnmFormat:
             case _qt_PbmFormat:
             case _qt_PgmFormat:
             case _qt_PpmFormat:
--- qt-everywhere-src-5.15.2/qtbase/src/gui/image/qppmhandler.cpp.orig	2020-10-27 08:02:11.000000000 +0000
+++ qt-everywhere-src-5.15.2/qtbase/src/gui/image/qppmhandler.cpp	2021-03-29 20:25:22.372482386 +0100
@@ -301,6 +301,17 @@
     QByteArray format = sourceFormat;
 
     format = format.left(3);                        // ignore RAW part
+    if (format == "ppm") {
+        switch (image.format()) {
+            case QImage::Format_Mono: format = "pbm";
+                break;
+            case QImage::QImage::Format_Grayscale8: format = "pgm";
+                break;
+            default: format = "ppm";
+                break;
+        }
+    }
+
     bool gray = format == "pgm";
 
     if (format == "pbm") {
--- qt-everywhere-src-5.15.2/qtbase/src/gui/image/qimagereaderwriterhelpers_p.h.orig	2020-10-27 08:02:11.000000000 +0000
+++ qt-everywhere-src-5.15.2/qtbase/src/gui/image/qimagereaderwriterhelpers_p.h	2021-03-14 15:57:09.368679791 +0000
@@ -72,6 +72,7 @@
     _qt_PpmFormat,
     _qt_PgmFormat,
     _qt_PbmFormat,
+    _qt_PnmFormat,
 #endif
 #ifndef QT_NO_IMAGEFORMAT_XBM
     _qt_XbmFormat,
@@ -110,6 +111,7 @@
     {"ppm", "x-portable-pixmap"},
     {"pgm", "x-portable-graymap"},
     {"pbm", "x-portable-bitmap"},
+    {"pnm", "x-portable-anymap"},
 #endif
 #ifndef QT_NO_IMAGEFORMAT_XBM
     {"xbm", "x-xbitmap"},
