--- qt-everywhere-opensource-src-5.8.0/qtbase/src/plugins/sqldrivers/mysql/qsql_mysql_p.h.orig	2017-01-18 14:20:58.000000000 +0000
+++ qt-everywhere-opensource-src-5.8.0/qtbase/src/plugins/sqldrivers/mysql/qsql_mysql_p.h	2017-01-29 13:35:22.979220434 +0000
@@ -57,7 +57,7 @@
 #include <QtCore/qt_windows.h>
 #endif
 
-#include <mysql.h>
+#include <mysql/mysql.h>
 
 #ifdef QT_PLUGIN
 #define Q_EXPORT_SQLDRIVER_MYSQL
