From: "Steve M. Robbins" <smr@sumost.ca>
Date: Sun, 27 Aug 2017 21:14:42 -0500
Subject: Rename qtav-private-dev.install ==> libqtav-private-dev.install.

---
 src/libQtAV.pro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libQtAV.pro b/src/libQtAV.pro
index 29015ff..9b1ff97 100644
--- a/src/libQtAV.pro
+++ b/src/libQtAV.pro
@@ -634,7 +634,7 @@ target.depends *= $${qtav_dev.target}
 
 DEB_INSTALL_LIST = $$join(SDK_PRIVATE_HEADERS, \\n.$$[QT_INSTALL_HEADERS]/QtAV/*/, .$$[QT_INSTALL_HEADERS]/QtAV/*/)
 DEB_INSTALL_LIST += .$${MKSPECS_DIR}/modules/qt_lib_av_private.pri
-qtav_private_dev.target = qtav-private-dev.install
+qtav_private_dev.target = libqtav-private-dev.install
 qtav_private_dev.commands = echo \"$$join(DEB_INSTALL_LIST, \\n)\" >$$PROJECTROOT/debian/$${qtav_private_dev.target}
 QMAKE_EXTRA_TARGETS += qtav_private_dev
 target.depends *= $${qtav_private_dev.target}
