From: "Steve M. Robbins" <smr@sumost.ca>
Date: Sun, 27 Aug 2017 22:53:55 -0500
Subject: Disable warnings with -Wno-expansion-to-defined -Wno-unused-parameter

---
 src/libQtAV.pro            | 1 +
 widgets/libQtAVWidgets.pro | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/libQtAV.pro b/src/libQtAV.pro
index 9b1ff97..beaa28c 100644
--- a/src/libQtAV.pro
+++ b/src/libQtAV.pro
@@ -617,6 +617,7 @@ icon.files = $$PWD/$${TARGET}.svg
 icon.path = /usr/share/icons/hicolor/scalable/apps
 INSTALLS += icon
 #debian
+QMAKE_CXXFLAGS += -Wno-expansion-to-defined -Wno-unused-parameter
 DEB_INSTALL_LIST = .$$[QT_INSTALL_LIBS]/libQt*AV.so.*
 libqtav.target = libqtav1.install
 libqtav.commands = echo \"$$join(DEB_INSTALL_LIST, \\n)\" >$$PROJECTROOT/debian/$${libqtav.target}
diff --git a/widgets/libQtAVWidgets.pro b/widgets/libQtAVWidgets.pro
index 3176d09..958b041 100644
--- a/widgets/libQtAVWidgets.pro
+++ b/widgets/libQtAVWidgets.pro
@@ -142,6 +142,7 @@ mac {
 
 unix:!android:!mac {
 #debian
+QMAKE_CXXFLAGS += -Wno-expansion-to-defined -Wno-unused-parameter
 DEB_INSTALL_LIST = .$$[QT_INSTALL_LIBS]/libQt*AVWidgets.so.*
 libqtavwidgets.target = libqtavwidgets1.install
 libqtavwidgets.commands = echo \"$$join(DEB_INSTALL_LIST, \\n)\" >$$PROJECTROOT/debian/$${libqtavwidgets.target}
