--- qtfeedback-opensource-src-5.0~git20180903.a14bd0b/src/imports/feedback/feedback.pro.orig	2018-09-03 10:16:11.000000000 +0100
+++ qtfeedback-opensource-src-5.0~git20180903.a14bd0b/src/imports/feedback/feedback.pro	2023-04-21 19:48:36.582710022 +0100
@@ -14,3 +14,5 @@
            qdeclarativefeedbackeffect.cpp
 
 load(qml_plugin)
+
+INCLUDEPATH += "../../feedback"
--- qtfeedback-opensource-src-5.0~git20180903.a14bd0b/src/plugins/feedback/immersion/immersion.pro.orig	2018-09-03 10:16:11.000000000 +0100
+++ qtfeedback-opensource-src-5.0~git20180903.a14bd0b/src/plugins/feedback/immersion/immersion.pro	2023-04-21 19:47:27.337353323 +0100
@@ -8,3 +8,5 @@
 SOURCES += qfeedback.cpp
 
 LIBS += -limmvibe
+
+INCLUDEPATH += "../../../feedback"
--- qtfeedback-opensource-src-5.0~git20180903.a14bd0b/src/plugins/feedback/meegotouch/meegotouch.pro.orig	2018-09-03 10:16:11.000000000 +0100
+++ qtfeedback-opensource-src-5.0~git20180903.a14bd0b/src/plugins/feedback/meegotouch/meegotouch.pro	2023-04-21 19:47:37.515261337 +0100
@@ -8,3 +8,5 @@
 SOURCES += qfeedback.cpp
 
 LIBS += -lmeegotouchcore
+
+INCLUDEPATH += "../../../feedback"
--- qtfeedback-opensource-src-5.0~git20180903.a14bd0b/src/plugins/feedback/mmk/mmk.pro.orig	2018-09-03 10:16:11.000000000 +0100
+++ qtfeedback-opensource-src-5.0~git20180903.a14bd0b/src/plugins/feedback/mmk/mmk.pro	2023-04-21 19:46:26.137029690 +0100
@@ -6,3 +6,5 @@
 
 HEADERS += qfeedback.h
 SOURCES += qfeedback.cpp
+
+INCLUDEPATH += "../../../feedback"
\ No newline at end of file
