--- qtiplot-0.9.7.3/qtiplot/qtiplot.pro.orig	2008-10-22 18:57:19.000000000 +0100
+++ qtiplot-0.9.7.3/qtiplot/qtiplot.pro	2008-10-24 21:46:52.000000000 +0100
@@ -48,17 +48,19 @@
 ##################### Linux (Mac OS X) ######################################
 
 # statically link against libraries in 3rdparty
-unix:LIBS         += ../3rdparty/muparser/lib/libmuparser.a
-unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
-unix:LIBS         += ../3rdparty/gsl/lib/libgsl.a
-unix:LIBS         += ../3rdparty/gsl/lib/libgslcblas.a
-unix:LIBS         += /usr/local/lib/libboost_date_time-gcc41-mt.a
-unix:LIBS         += /usr/local/lib/libboost_thread-gcc41-mt.a
+#unix:LIBS         += ../3rdparty/muparser/lib/libmuparser.a
+#unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
+#unix:LIBS         += ../3rdparty/gsl/lib/libgsl.a
+#unix:LIBS         += ../3rdparty/gsl/lib/libgslcblas.a
+#unix:LIBS         += /usr/local/lib/libboost_date_time-gcc41-mt.a
+#unix:LIBS         += /usr/local/lib/libboost_thread-gcc41-mt.a
 
 # dynamically link against dependencies if they are installed system-wide
-#unix:LIBS         += -lmuparser
-#unix:LIBS         += -lqwt
-#unix:LIBS         += -lgsl -lgslcblas
+unix:LIBS         += -lmuparser
+unix:LIBS         += -lqwt
+unix:LIBS         += -lgsl -lgslcblas
+unix:LIBS         += -lboost_date_time-mt
+unix:LIBS         += -lboost_thread-mt
 
 ##################### Windows ###############################################
 
--- qtiplot-0.9.7.3/fitPlugins/fitRational0/fitRational0.pro.orig	2008-10-13 17:14:52.000000000 +0100
+++ qtiplot-0.9.7.3/fitPlugins/fitRational0/fitRational0.pro	2008-10-24 20:49:03.000000000 +0100
@@ -12,12 +12,12 @@
 DESTDIR           = ../
  
 # statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
+#INCLUDEPATH += ../../3rdparty/gsl/include/
+#LIBS        += ../../3rdparty/gsl/lib/libgsl.a
+#LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
 
 #dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
 
 target.path=/usr/lib$${libsuff}/qtiplot/plugins
 INSTALLS += target
--- qtiplot-0.9.7.3/fitPlugins/fitRational1/fitRational1.pro.orig	2007-10-28 23:44:06.000000000 +0000
+++ qtiplot-0.9.7.3/fitPlugins/fitRational1/fitRational1.pro	2008-10-24 20:49:24.000000000 +0100
@@ -12,12 +12,12 @@
 DESTDIR           = ../
  
 # statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
+#INCLUDEPATH += ../../3rdparty/gsl/include/
+#LIBS        += ../../3rdparty/gsl/lib/libgsl.a
+#LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
 
 #dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
 
 target.path=/usr/lib$${libsuff}/qtiplot/plugins
 INSTALLS += target
--- qtiplot-0.9.7.3/fitPlugins/explin/explin.pro.orig	2008-10-13 08:43:31.000000000 +0100
+++ qtiplot-0.9.7.3/fitPlugins/explin/explin.pro	2008-10-24 20:47:57.000000000 +0100
@@ -12,12 +12,12 @@
 DESTDIR           = ../
  
 # statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
+#INCLUDEPATH += ../../3rdparty/gsl/include/
+#LIBS        += ../../3rdparty/gsl/lib/libgsl.a
+#LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
 
 #dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
 
 target.path=/usr/lib$${libsuff}/qtiplot/plugins
 INSTALLS += target
--- qtiplot-0.9.7.3/fitPlugins/exp_saturation/exp_saturation.pro.orig	2008-10-13 08:43:31.000000000 +0100
+++ qtiplot-0.9.7.3/fitPlugins/exp_saturation/exp_saturation.pro	2008-10-24 20:48:25.000000000 +0100
@@ -12,12 +12,12 @@
 DESTDIR           = ../
  
 # statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
+#INCLUDEPATH += ../../3rdparty/gsl/include/
+#LIBS        += ../../3rdparty/gsl/lib/libgsl.a
+#LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
 
 #dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
 
 target.path=/usr/lib$${libsuff}/qtiplot/plugins
 INSTALLS += target
--- qtiplot-0.9.7.3/qtiplot.pro.orig	2008-10-13 08:43:31.000000000 +0100
+++ qtiplot-0.9.7.3/qtiplot.pro	2008-10-24 20:51:27.000000000 +0100
@@ -1,5 +1,4 @@
 TEMPLATE = subdirs
 
 SUBDIRS = fitPlugins \
-	      manual \
           qtiplot
--- qtiplot-0.9.7.4/qtiplot/src/scripting/qti.sip.orig	2008-10-20 11:32:35.000000000 +0100
+++ qtiplot-0.9.7.4/qtiplot/src/scripting/qti.sip	2008-11-30 19:53:59.000000000 +0000
@@ -1072,6 +1072,10 @@
     Graph3D* plotParametricSurface(const QString&, const QString&, const QString&,
             double, double, double, double, int=40, int=40, bool=true, bool=true) /PyName=plot3D/;
 
+%TypeCode
+typedef QList<MdiSubWindow*> MDIWindowList;
+%End
+
   MDIWindowList windowsList() /PyName=windows/;
 
   // folders
@@ -1867,6 +1871,10 @@
 #include "src/core/Folder.h"
 %End
 public:
+%TypeCode
+typedef QList<MdiSubWindow*> MDIWindowList;
+%End
+
   MDIWindowList windowsList() /PyName=windows/;
 // TODO: implement signal Folder::nameChanged and make it update the project explorer; adjust renaming from GUI accordingly
 //  void setFolderName(const QString&) /PyName=setName/;
