--- qtiplot-0.9.7.2/qtiplot/qtiplot.pro.orig	2008-09-28 14:34:40.000000000 +0100
+++ qtiplot-0.9.7.2/qtiplot/qtiplot.pro	2008-09-28 15:40:32.000000000 +0100
@@ -17,7 +17,7 @@
 # Uncomment the following line if you want to perform a custom installation using
 # the *.path variables defined bellow.
 ######################################################################################
-#CONFIG          += CustomInstall
+CONFIG          += CustomInstall
 
 CONFIG          += release
 #CONFIG          += debug
@@ -42,17 +42,17 @@
 ##################### Linux (Mac OS X) ######################################
 
 # statically link against libraries in 3rdparty
-unix:LIBS         += ../3rdparty/muparser/lib/libmuparser.a
-unix:LIBS         += ../3rdparty/qwtplot3d/lib/libqwtplot3d.a
-unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
-unix:LIBS         += ../3rdparty/gsl/lib/libgsl.a
-unix:LIBS         += ../3rdparty/gsl/lib/libgslcblas.a
+#unix:LIBS         += ../3rdparty/muparser/lib/libmuparser.a
+#unix:LIBS         += ../3rdparty/qwtplot3d/lib/libqwtplot3d.a
+#unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
+#unix:LIBS         += ../3rdparty/gsl/lib/libgsl.a
+#unix:LIBS         += ../3rdparty/gsl/lib/libgslcblas.a
 
 # dynamically link against dependencies if they are installed system-wide
-#unix:LIBS         += -lmuparser
-#unix:LIBS         += -lqwtplot3d
-#unix:LIBS         += -lqwt
-#unix:LIBS         += -lgsl -lgslcblas
+unix:LIBS         += -lmuparser
+unix:LIBS         += -lqwtplot3d
+unix:LIBS         += -lqwt
+unix:LIBS         += -lgsl -lgslcblas
 
 ##################### Windows ###############################################
 
@@ -87,9 +87,9 @@
 
 	unix: target.path = $$INSTALLBASE/bin
 	unix: translations.path = $$INSTALLBASE/share/qtiplot/translations
-	unix: manual.path = $$INSTALLBASE/share/doc/qtiplot/manual
-	unix: documentation.path = $$INSTALLBASE/share/doc/qtiplot
-	unix: man.path = $$INSTALLBASE/share/man/man1/
+	unix: manual.path = $$INSTALLBASE/share/qtiplot/manual
+	unix: documentation.path = $$INSTALLBASE/doc/qtiplot
+	unix: man.path = $$INSTALLBASE/man/man1/
 
 	win32: target.path = $$INSTALLBASE
 	win32: translations.path = $$INSTALLBASE/translations
@@ -495,7 +495,7 @@
   						qtiUtil.py \
   						qti_wordlist.txt \
 
-  	unix: pythonconfig.path = /usr/local/qtiplot
+  	unix: pythonconfig.path = /usr/share/qtiplot
   	win32: pythonconfig.path = $$INSTALLBASE
   	DEFINES += PYTHON_CONFIG_PATH="\\\"$$replace(pythonconfig.path," ","\ ")\\\"
   }
--- qtiplot-0.9.7.2/fitPlugins/fitRational0/fitRational0.pro.orig	2008-09-28 14:34:40.000000000 +0100
+++ qtiplot-0.9.7.2/fitPlugins/fitRational0/fitRational0.pro	2008-09-28 14:34:49.000000000 +0100
@@ -12,12 +12,12 @@
 DESTDIR           = ../
  
 # statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
+#INCLUDEPATH += ../../3rdparty/gsl/include/
+#LIBS        += ../../3rdparty/gsl/lib/libgsl.a
+#LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
 
 #dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
 
 target.path=/usr/lib$${libsuff}/qtiplot/plugins
 INSTALLS += target
--- qtiplot-0.9.7.2/fitPlugins/fitRational1/fitRational1.pro.orig	2008-09-28 14:34:40.000000000 +0100
+++ qtiplot-0.9.7.2/fitPlugins/fitRational1/fitRational1.pro	2008-09-28 14:34:49.000000000 +0100
@@ -12,12 +12,12 @@
 DESTDIR           = ../
  
 # statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
+#INCLUDEPATH += ../../3rdparty/gsl/include/
+#LIBS        += ../../3rdparty/gsl/lib/libgsl.a
+#LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
 
 #dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
 
 target.path=/usr/lib$${libsuff}/qtiplot/plugins
 INSTALLS += target
--- qtiplot-0.9.7.2/qtiplot.pro.orig	2008-09-28 14:34:40.000000000 +0100
+++ qtiplot-0.9.7.2/qtiplot.pro	2008-09-28 14:34:49.000000000 +0100
@@ -2,6 +2,7 @@
 
 SUBDIRS = fitPlugins/fitRational0\
           fitPlugins/fitRational1\
-          3rdparty/qwt\
-	      manual\
           qtiplot
+
+#          3rdparty/qwt\
+#	      manual\
