Description: Adds missing include on GL/glu.h
Author: Scott Howard <showard@debian.org>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/qtiplot/+bug/925652

Index: qtiplot-0.9.8.8/3rdparty/qwtplot3d/include/qwt3d_global.h
===================================================================
--- qtiplot-0.9.8.8.orig/3rdparty/qwtplot3d/include/qwt3d_global.h	2012-02-02 18:30:45.508825870 -0500
+++ qtiplot-0.9.8.8/3rdparty/qwtplot3d/include/qwt3d_global.h	2012-02-02 18:31:15.364825258 -0500
@@ -2,6 +2,7 @@
 #define QWT3D_GLOBAL_H
 
 #include <qglobal.h>
+#include <GL/glu.h>
 #if QT_VERSION < 0x040000
 #include <qmodules.h>
 #endif
Index: qtiplot-0.9.8.8/3rdparty/qwtplot3d/include/qwt3d_openglhelper.h
===================================================================
--- qtiplot-0.9.8.8.orig/3rdparty/qwtplot3d/include/qwt3d_openglhelper.h	2012-02-02 18:30:45.532825870 -0500
+++ qtiplot-0.9.8.8/3rdparty/qwtplot3d/include/qwt3d_openglhelper.h	2012-02-02 18:31:15.364825258 -0500
@@ -2,6 +2,7 @@
 #define __openglhelper_2003_06_06_15_49__
 
 #include "qglobal.h"
+#include <GL/glu.h>
 #if QT_VERSION < 0x040000
 #include <qgl.h>
 #else
