--- qtiplot-0.9.7.8/fitPlugins/explin/explin.pro.orig	2009-04-01 15:16:14.000000000 +0100
+++ qtiplot-0.9.7.8/fitPlugins/explin/explin.pro	2009-06-24 06:35:56.000000000 +0100
@@ -12,12 +12,12 @@
 DESTDIR           = ../
  
 # statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
+#INCLUDEPATH += ../../3rdparty/gsl/include/
+#LIBS        += ../../3rdparty/gsl/lib/libgsl.a
+#LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
 
 #dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
 
 target.path=/usr/lib$${libsuff}/qtiplot/plugins
 INSTALLS += target
--- qtiplot-0.9.7.8/fitPlugins/exp_saturation/exp_saturation.pro.orig	2009-04-01 15:16:14.000000000 +0100
+++ qtiplot-0.9.7.8/fitPlugins/exp_saturation/exp_saturation.pro	2009-06-24 06:35:56.000000000 +0100
@@ -12,12 +12,12 @@
 DESTDIR           = ../
  
 # statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
+#INCLUDEPATH += ../../3rdparty/gsl/include/
+#LIBS        += ../../3rdparty/gsl/lib/libgsl.a
+#LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
 
 #dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
 
 target.path=/usr/lib$${libsuff}/qtiplot/plugins
 INSTALLS += target
--- qtiplot-0.9.7.8/fitPlugins/fitRational0/fitRational0.pro.orig	2009-04-01 15:16:14.000000000 +0100
+++ qtiplot-0.9.7.8/fitPlugins/fitRational0/fitRational0.pro	2009-06-24 06:35:56.000000000 +0100
@@ -12,12 +12,12 @@
 DESTDIR           = ../
  
 # statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
+#INCLUDEPATH += ../../3rdparty/gsl/include/
+#LIBS        += ../../3rdparty/gsl/lib/libgsl.a
+#LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
 
 #dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
 
 target.path=/usr/lib$${libsuff}/qtiplot/plugins
 INSTALLS += target
--- qtiplot-0.9.7.8/fitPlugins/fitRational1/fitRational1.pro.orig	2009-04-01 15:16:14.000000000 +0100
+++ qtiplot-0.9.7.8/fitPlugins/fitRational1/fitRational1.pro	2009-06-24 06:35:56.000000000 +0100
@@ -12,12 +12,12 @@
 DESTDIR           = ../
  
 # statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
+#INCLUDEPATH += ../../3rdparty/gsl/include/
+#LIBS        += ../../3rdparty/gsl/lib/libgsl.a
+#LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
 
 #dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
 
 target.path=/usr/lib$${libsuff}/qtiplot/plugins
 INSTALLS += target
--- qtiplot-0.9.7.8/fitPlugins/planck_wavelength/planck_wavelength.pro.orig	2009-06-01 13:11:01.000000000 +0100
+++ qtiplot-0.9.7.8/fitPlugins/planck_wavelength/planck_wavelength.pro	2009-06-24 06:40:58.000000000 +0100
@@ -11,12 +11,12 @@
 INSTALLS += target
  
 # statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
+#INCLUDEPATH += ../../3rdparty/gsl/include/
+#LIBS        += ../../3rdparty/gsl/lib/libgsl.a
+#LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
 
 #dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
 
 target.path=/usr/lib$${libsuff}/qtiplot/plugins
 
--- qtiplot-0.9.7.8/qtiplot/qtiplot.pro.orig	2009-06-22 15:43:17.000000000 +0100
+++ qtiplot-0.9.7.8/qtiplot/qtiplot.pro	2009-06-24 06:35:56.000000000 +0100
@@ -21,7 +21,7 @@
 #CONFIG          += HAVE_EMF
 
 # Uncomment the following line if you want to perform a custom installation using the *.path variables defined bellow.
-#CONFIG          += CustomInstall
+CONFIG          += CustomInstall
 
 CONFIG          += release
 #CONFIG          += debug
@@ -31,13 +31,13 @@
 #!!! Warning: You must modify these paths according to your computer settings
 #############################################################################
 
-INCLUDEPATH       += ../3rdparty/muparser/include
+#INCLUDEPATH       += ../3rdparty/muparser/include
 INCLUDEPATH       += ../3rdparty/qwtplot3d/include
 INCLUDEPATH       += ../3rdparty/qwt/src
 INCLUDEPATH       += ../3rdparty/liborigin
-INCLUDEPATH       += ../3rdparty/gsl/include
-INCLUDEPATH       += ../3rdparty/zlib
-INCLUDEPATH       += ../3rdparty/boost_1_36_0
+#INCLUDEPATH       += ../3rdparty/gsl/include
+#INCLUDEPATH       += ../3rdparty/zlib
+#INCLUDEPATH       += ../3rdparty/boost_1_36_0
 
 ##################### 3rd PARTY LIBRARIES SECTION ###########################
 #!!! Warning: You must modify these paths according to your computer settings
@@ -46,17 +46,19 @@
 ##################### Linux (Mac OS X) ######################################
 
 # statically link against libraries in 3rdparty
-unix:LIBS         += ../3rdparty/muparser/lib/libmuparser.a
-unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
-unix:LIBS         += ../3rdparty/gsl/lib/libgsl.a
-unix:LIBS         += ../3rdparty/gsl/lib/libgslcblas.a
-unix:LIBS         += ../3rdparty/boost_1_36_0/lib/libboost_date_time-gcc43-mt-1_36.a
-unix:LIBS         += ../3rdparty/boost_1_36_0/lib/libboost_thread-gcc43-mt-1_36.a
+#unix:LIBS         += ../3rdparty/muparser/lib/libmuparser.a
+#unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
+#unix:LIBS         += ../3rdparty/gsl/lib/libgsl.a
+#unix:LIBS         += ../3rdparty/gsl/lib/libgslcblas.a
+#unix:LIBS         += ../3rdparty/boost_1_36_0/lib/libboost_date_time-gcc43-mt-1_36.a
+#unix:LIBS         += ../3rdparty/boost_1_36_0/lib/libboost_thread-gcc43-mt-1_36.a
 
 # dynamically link against dependencies if they are installed system-wide
-#unix:LIBS         += -lmuparser
-#unix:LIBS         += -lqwt
-#unix:LIBS         += -lgsl -lgslcblas
+unix:LIBS         += -lmuparser
+unix:LIBS         += -lqwt
+unix:LIBS         += -lgsl -lgslcblas
+unix:LIBS         += -lboost_date_time-mt
+unix:LIBS         += -lboost_thread-mt
 
 ##################### Windows ###############################################
 
@@ -190,7 +192,7 @@
   				    qtiUtil.py \
   				    qti_wordlist.txt \
 
-  	unix: pythonconfig.path = /usr/local/qtiplot
+  	unix: pythonconfig.path = /usr/share/qtiplot
   	win32: pythonconfig.path = $$INSTALLBASE
   	DEFINES += PYTHON_CONFIG_PATH="\\\"$$replace(pythonconfig.path," ","\ ")\\\"
   }
@@ -214,8 +216,8 @@
 
 contains(CONFIG, HAVE_LIBPNG){
 	DEFINES += GL2PS_HAVE_LIBPNG
-	INCLUDEPATH += ../3rdparty/libpng/
-	LIBS        += ../3rdparty/libpng/libpng.a
+#	INCLUDEPATH += ../3rdparty/libpng/
+	LIBS        += -lpng 
 }
 
 ###############################################################
--- qtiplot-0.9.7.8/qtiplot.pro.orig	2009-06-15 12:24:46.000000000 +0100
+++ qtiplot-0.9.7.8/qtiplot.pro	2009-06-24 06:39:24.000000000 +0100
@@ -1,6 +1,5 @@
 TEMPLATE = subdirs
 
 SUBDIRS = fitPlugins \
-	      manual \
 	      3rdparty/qwt \
           qtiplot
