--- qtiplot-0.6.7/qtiplot-0.6.7.pro.orig	2005-08-07 14:48:24.000000000 +0100
+++ qtiplot-0.6.7/qtiplot-0.6.7.pro	2005-08-17 15:02:15.000000000 +0100
@@ -15,7 +15,7 @@
 win32:LIBS     += C:/Qt/qwtplot3d/lib/qwtplot3d.lib 
 win32:LIBS     += C:/Qt/WinGsl/Lib/WinGsl.lib 
 
-unix:INCLUDEPATH +=  /home/ion/qt/qwtplot3d/include
+unix:INCLUDEPATH +=  /usr/include/qwtplot3d
 unix:INCLUDEPATH += ../3rdparty/qwt/include
 
 unix:LIBS += -L../3rdparty/qwt/lib -lqwt
