--- QtZeitgeist-0.7.0/CMakeLists.txt.orig	2011-03-28 17:16:34.000000000 +0100
+++ QtZeitgeist-0.7.0/CMakeLists.txt	2011-05-21 06:09:45.274000072 +0100
@@ -39,7 +39,7 @@
 
 # Generate pkg-config file
 configure_file(${CMAKE_SOURCE_DIR}/QtZeitgeist.pc.in ${CMAKE_BINARY_DIR}/QtZeitgeist.pc)
-install(FILES ${CMAKE_BINARY_DIR}/QtZeitgeist.pc DESTINATION lib/pkgconfig)
+install(FILES ${CMAKE_BINARY_DIR}/QtZeitgeist.pc DESTINATION lib${LIB_SUFFIX}/pkgconfig)
 
 add_subdirectory(include)
 add_subdirectory(src)
--- QtZeitgeist-0.7.0/src/CMakeLists.txt.orig	2011-03-28 17:16:34.000000000 +0100
+++ QtZeitgeist-0.7.0/src/CMakeLists.txt	2011-05-21 06:09:30.554000060 +0100
@@ -47,5 +47,5 @@
 add_library(qzeitgeist SHARED ${library_SRCS} ${library_MOC_SRCS})
 target_link_libraries(qzeitgeist ${QT_LIBRARIES} ${QDBUS_LDFLAGS})
 set_target_properties(qzeitgeist PROPERTIES VERSION "${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}" SOVERSION ${CPACK_PACKAGE_VERSION_MAJOR})
-install(TARGETS qzeitgeist EXPORT QZeitgeistExport DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
+install(TARGETS qzeitgeist EXPORT QZeitgeistExport DESTINATION ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX})
 install(EXPORT QZeitgeistExport DESTINATION ${CMAKE_INSTALL_PREFIX}/share/qzeitgeist/cmake NAMESPACE QZEITGEIST_)
