--- quazip-0.5.1/qztest/testquagzipfile.cpp.orig	2012-09-05 16:24:35.000000000 +0100
+++ quazip-0.5.1/qztest/testquagzipfile.cpp	2013-08-18 07:56:54.652482064 +0100
@@ -8,7 +8,7 @@
 {
     QDir curDir;
     curDir.mkpath("tmp");
-    voidp gzFile = gzopen("tmp/test.gz", "wb");
+    gzFile gzFile = gzopen("tmp/test.gz", "wb");
     gzwrite(gzFile, "test", 4);
     gzclose(gzFile);
     QuaGzipFile testFile("tmp/test.gz");
@@ -32,7 +32,7 @@
     QCOMPARE(testFile.write("test", 4), static_cast<qint64>(4));
     testFile.close();
     QVERIFY(!testFile.isOpen());
-    voidp gzFile = gzopen("tmp/test.gz", "rb");
+    gzFile gzFile = gzopen("tmp/test.gz", "rb");
     char buf[5];
     buf[4] = '\0';
     QCOMPARE(gzread(gzFile, buf, 5), 4);
