--- qwtplot3d-0.2.7+svn191/qwtplot3d.pro.orig	2007-08-23 11:20:57.000000000 +0100
+++ qwtplot3d-0.2.7+svn191/qwtplot3d.pro	2016-04-06 23:02:43.086699360 +0100
@@ -1,7 +1,8 @@
 # pro file for building the makefile for qwtplot3d
 #
 
-TARGET            = qwtplot3d
+TARGET            = qwtplot3d-qt4
+INSTALLS          = doc headers target
 TEMPLATE          = lib
 CONFIG           += qt warn_on opengl thread zlib debug
 MOC_DIR           = tmp
@@ -20,6 +21,8 @@
 # Comment the next line, if you have zlib on your windows system
 win32:CONFIG -= zlib
 
+unix:LIBS  += -lGLU
+
 linux-g++:TMAKE_CXXFLAGS += -fno-exceptions
 unix:VERSION = 0.2.7
 
@@ -92,4 +95,12 @@
   DEFINES += GL2PS_HAVE_ZLIB
   win32:LIBS += zlib.lib
 	unix:LIBS  += -lz
-}
\ No newline at end of file
+}
+
+# Install directives
+INSTALLBASE      = /usr
+headers.path      = $$INSTALLBASE/include/qwtplot3d-qt4
+headers.files     = $$HEADERS
+target.path       = $$INSTALLBASE/lib64/qt4/lib
+doc.path          = $$INSTALLBASE/usr/doc
+
--- qwtplot3d-0.2.7+svn191/include/qwt3d_openglhelper.h.orig	2007-08-23 11:20:56.000000000 +0100
+++ qwtplot3d-0.2.7+svn191/include/qwt3d_openglhelper.h	2016-04-06 23:01:02.766698990 +0100
@@ -7,6 +7,7 @@
 #else
 #include <QtOpenGL/qgl.h>
 #endif
+#include <GL/glu.h>
 
 namespace Qwt3D
 {
