--- rapidsvn-0.10.0/src/tests/Makefile.am.orig	2009-07-18 22:56:28.000000000 +0100
+++ rapidsvn-0.10.0/src/tests/Makefile.am	2009-07-18 09:58:03.000000000 +0100
@@ -1,2 +1,2 @@
-DIST_SUBDIRS=svncpp
+DIST_SUBDIRS=
 
--- rapidsvn-0.10.0/configure.in.orig	2009-03-20 12:26:12.000000000 +0000
+++ rapidsvn-0.10.0/configure.in	2009-07-18 23:03:12.000000000 +0100
@@ -460,6 +460,5 @@
           src/res/Makefile \
           src/svncpp/Makefile \
 	  src/tests/Makefile \
-	  src/tests/svncpp/Makefile \
 	  tools/Makefile)
 
