--- reed-solomon-3.1.1/makefile.in.orig	2002-01-02 05:19:55.000000000 +0000
+++ reed-solomon-3.1.1/makefile.in	2008-03-27 15:06:07.000000000 +0000
@@ -24,11 +24,12 @@
 	gcc -g -o $@ $^
 
 install: all
-	install -D -m 644 -p librs.a librs.so.@SO_VERSION@ @libdir@
-	(cd @libdir@;ln -f -s librs.so.@SO_VERSION@ librs.so)
-	ldconfig
-	install -m 644 -p rs.h @includedir@
-	install -m 644 rs.3 @mandir@/man3
+	install -d $(DESTDIR)@libdir@ $(DESTDIR)@includedir@ $(DESTDIR)@mandir@/man3
+	install -D -m 644 -p librs.a librs.so.@SO_VERSION@ $(DESTDIR)@libdir@
+	(cd $(DESTDIR)@libdir@;ln -f -s librs.so.@SO_VERSION@ librs.so)
+	-ldconfig
+	install -D -m 644 -p rs.h $(DESTDIR)@includedir@
+	install -D -m 644 rs.3 $(DESTDIR)@mandir@/man3
 
 librs.a: $(LIB)
 	ar rv $@ $^
