--- ring-20181001.4.a99aaec/daemon/contrib/src/opendht/rules.mak.orig	2018-10-01 22:25:18.000000000 +0100
+++ ring-20181001.4.a99aaec/daemon/contrib/src/opendht/rules.mak	2018-09-20 18:50:28.000000000 +0100
@@ -31,9 +31,11 @@
 
 opendht: opendht-$(OPENDHT_VERSION).tar.gz
 	$(UNPACK)
+	$(UPDATE_AUTOCONFIG) && cd $(UNPACK_DIR)
 	$(MOVE)
 
 .opendht: opendht .sum-opendht
-	cd $< && $(HOSTVARS) $(CMAKE) -DOPENDHT_STATIC=On -DOPENDHT_SHARED=Off -DOPENDHT_TOOLS=Off -DOPENDHT_PROXY_CLIENT=On -DOPENDHT_PUSH_NOTIFICATIONS=On -DOPENDHT_DOCUMENTATION=Off .
+	mkdir -p $</m4 && $(RECONF)
+	cd $< && $(HOSTVARS) ./configure --enable-static --disable-shared --disable-tools --disable-python --disable-doc --enable-proxy-server --enable-proxy-client --enable-push-notifications $(HOSTCONF)
 	cd $< && $(MAKE) install
 	touch $@
