--- ring-20180830.1.2b1e63a/client-gnome/src/accountcreationwizard.cpp.orig	2018-09-04 16:05:06.000000000 +0100
+++ ring-20180830.1.2b1e63a/client-gnome/src/accountcreationwizard.cpp	2018-09-09 06:52:18.321561961 +0100
@@ -267,7 +267,7 @@
     }
 
     g_signal_emit(G_OBJECT(view), account_creation_wizard_signals[ACCOUNT_CREATION_COMPLETED], 0);
-    g_object_unref(view);
+//    g_object_unref(view);
 }
 
 static gboolean
@@ -431,7 +431,7 @@
 {
     lrc::api::NewAccountModel::createNewAccount(lrc::api::profile::Type::SIP, "SIP");
     g_signal_emit(G_OBJECT(view), account_creation_wizard_signals[ACCOUNT_CREATION_COMPLETED], 0);
-    g_object_unref(view);
+//    g_object_unref(view);
 }
 
 static void
