--- rosegarden-4-1.0/base/Colour.C.orig	2005-02-08 14:55:22.000000000 +0000
+++ rosegarden-4-1.0/base/Colour.C	2006-01-04 15:31:58.000000000 +0000
@@ -163,7 +163,7 @@
 // Generic Colour routines:
 
 Colour
-Rosegarden::getCombinationColour(const Colour &input1, const Colour &input2)
+getCombinationColour(const Colour &input1, const Colour &input2)
 {
     Colour ret((input1.getRed()+input2.getRed())/2,
                 (input1.getGreen()+input2.getGreen())/2,
--- rosegarden-4-1.0/sequencer/Makefile.in.orig	2005-02-14 19:12:35.000000000 +0000
+++ rosegarden-4-1.0/sequencer/Makefile.in	2006-01-06 14:31:50.000000000 +0000
@@ -68,8 +68,8 @@
 @KDE_USE_FINAL_TRUE@rosegardensequencer_OBJECTS = $(rosegardensequencer_final_OBJECTS)
 am__DEPENDENCIES_1 =
 rosegardensequencer_DEPENDENCIES = $(am__DEPENDENCIES_1) \
-	$(am__DEPENDENCIES_1) ../sound/libRosegardenSound.la \
-	../sound/libRosegardenSequencer.la ../base/libbase.la \
+	$(am__DEPENDENCIES_1) \
+	../sound/libRosegardenSequencer.la \
 	$(am__DEPENDENCIES_1)
 DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir)
 depcomp = $(SHELL) $(top_srcdir)/admin/depcomp
@@ -370,7 +370,7 @@
                               sequencermapper.cpp \
 			       rosegardensequenceriface_skel.cpp
 
-rosegardensequencer_LDADD = -lkdeui -lkdecore $(LIB_QT) $(LIBSOCKET) ../sound/libRosegardenSound.la ../sound/libRosegardenSequencer.la ../base/libbase.la $(LIB_KFILE)
+rosegardensequencer_LDADD = -lkdeui -lkdecore $(LIB_QT) $(LIBSOCKET) ../sound/libRosegardenSequencer.la $(LIB_KFILE)
 
 # this 10 paths are KDE specific. Use them:
 # kde_htmldir       Where your docs should go to. (contains lang subdirs)
