--- rrdtool-1.5.3/bindings/lua/rrdlua.c.orig	2015-04-20 15:12:18.000000000 +0100
+++ rrdtool-1.5.3/bindings/lua/rrdlua.c	2015-05-03 09:56:26.329542353 +0100
@@ -350,7 +350,7 @@
 
 /**********************************************************/
 
-static const struct luaL_reg rrd[] = {
+static const struct luaL_Reg rrd[] = {
   {"create", lua_rrd_create},
   {"dump", lua_rrd_dump},
   {"fetch", lua_rrd_fetch},
@@ -383,7 +383,8 @@
   /* luaL_module is defined in compat-5.1.c */
   luaL_module (L, "rrd", rrd, 0);
 #else
-  luaL_register (L, "rrd", rrd);
+  //luaL_register (L, "rrd", rrd);
+  luaL_setfuncs(L, rrd, 0);
 #endif
   set_info (L);
   return 1;
