--- rtmidi-4.0.0/CMakeLists.txt.orig	2019-04-18 02:01:54.000000000 +0100
+++ rtmidi-4.0.0/CMakeLists.txt	2020-11-28 19:22:12.255838550 +0000
@@ -106,7 +106,7 @@
   check_symbol_exists(jack_port_rename ${jack_INCLUDEDIR}/jack/jack.h JACK_HAS_PORT_RENAME)
   set(CMAKE_REQUIRED_LIBRARIES ${tmp_CMAKE_REQUIRED_LIBRARIES})
   if(JACK_HAS_PORT_RENAME)
-    list(APPEND API_DEFS "JACK_HAS_PORT_RENAME")
+    list(APPEND API_DEFS "-DJACK_HAS_PORT_RENAME")
   endif()
 endif()
 
