--- rttr-0.9.6/3rd_party/rapidjson-1.1.0/rapidjson/document.h.orig	2018-03-26 22:30:31.000000000 +0100
+++ rttr-0.9.6/3rd_party/rapidjson-1.1.0/rapidjson/document.h	2019-07-13 20:14:08.055839123 +0100
@@ -1950,7 +1950,7 @@
         if (count) {
             GenericValue* e = static_cast<GenericValue*>(allocator.Malloc(count * sizeof(GenericValue)));
             SetElementsPointer(e);
-            std::memcpy(e, values, count * sizeof(GenericValue));
+            std::memcpy(static_cast<void*>(e), values, count * sizeof(GenericValue));
         }
         else
             SetElementsPointer(0);
@@ -1963,7 +1963,7 @@
         if (count) {
             Member* m = static_cast<Member*>(allocator.Malloc(count * sizeof(Member)));
             SetMembersPointer(m);
-            std::memcpy(m, members, count * sizeof(Member));
+            std::memcpy(static_cast<void*>(m), members, count * sizeof(Member));
         }
         else
             SetMembersPointer(0);
