--- rubberband-1.8.1/Makefile.in.orig	2012-10-28 10:30:09.000000000 +0000
+++ rubberband-1.8.1/Makefile.in	2015-10-22 23:03:08.846537666 +0100
@@ -28,11 +28,11 @@
 
 INSTALL_BINDIR		:= @prefix@/bin
 INSTALL_INCDIR		:= @prefix@/include/rubberband
-INSTALL_LIBDIR		:= @prefix@/lib
-INSTALL_VAMPDIR		:= @prefix@/lib/vamp
-INSTALL_LADSPADIR	:= @prefix@/lib/ladspa
+INSTALL_LIBDIR		:= @prefix@/lib64
+INSTALL_VAMPDIR		:= @prefix@/lib64/vamp
+INSTALL_LADSPADIR	:= @prefix@/lib64/ladspa
 INSTALL_LRDFDIR		:= @prefix@/share/ladspa/rdf
-INSTALL_PKGDIR		:= @prefix@/lib/pkgconfig
+INSTALL_PKGDIR		:= @prefix@/lib64/pkgconfig
 
 all:	bin lib $(PROGRAM_TARGET) $(STATIC_TARGET) $(DYNAMIC_TARGET) $(VAMP_TARGET) $(LADSPA_TARGET)
 
--- rubberband-1.8.1/rubberband.pc.in.orig	2012-10-28 10:30:09.000000000 +0000
+++ rubberband-1.8.1/rubberband.pc.in	2015-10-22 23:02:43.260537571 +0100
@@ -1,6 +1,6 @@
 prefix=%PREFIX%
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=${exec_prefix}/lib64
 includedir=${prefix}/include
 
 Name: rubberband
