--- rubberband-1.0.1/Makefile.in.orig	2007-12-13 22:15:03.000000000 +0000
+++ rubberband-1.0.1/Makefile.in	2008-01-22 16:10:13.000000000 +0000
@@ -20,12 +20,12 @@
 VAMP_TARGET    		:= lib/vamp-rubberband$(DYNAMIC_EXTENSION)
 LADSPA_TARGET  		:= lib/ladspa-rubberband$(DYNAMIC_EXTENSION)
 
-INSTALL_BINDIR		:= @prefix@/bin
-INSTALL_INCDIR		:= @prefix@/include/rubberband
-INSTALL_LIBDIR		:= @prefix@/lib
-INSTALL_VAMPDIR		:= @prefix@/lib/vamp
-INSTALL_LADSPADIR	:= @prefix@/lib/ladspa
-INSTALL_PKGDIR		:= @prefix@/lib/pkgconfig
+INSTALL_BINDIR		:= $(DESTDIR)@prefix@/bin
+INSTALL_INCDIR		:= $(DESTDIR)@prefix@/include/rubberband
+INSTALL_LIBDIR		:= $(DESTDIR)@prefix@/lib
+INSTALL_VAMPDIR		:= $(DESTDIR)@prefix@/lib/vamp
+INSTALL_LADSPADIR	:= $(DESTDIR)@prefix@/lib/ladspa
+INSTALL_PKGDIR		:= $(DESTDIR)@prefix@/lib/pkgconfig
 
 all:	bin lib $(PROGRAM_TARGET) $(STATIC_TARGET) $(DYNAMIC_TARGET) $(VAMP_TARGET) $(LADSPA_TARGET)
 
