--- scidavis-1.25/config.pri.orig	2019-03-04 23:49:55.000000000 +0000
+++ scidavis-1.25/config.pri	2019-03-16 22:12:14.036786681 +0000
@@ -286,6 +286,9 @@
         exists(/usr/include/qt5/qwt5-qt5)      {INCLUDEPATH+=/usr/include/qt5/qwt5-qt5}
         exists(/usr/include/qt5/qwtplot3d-qt5) {INCLUDEPATH+=/usr/include/qt5/qwtplot3d-qt5}
 
+        exists(/usr/lib64/qt5/include/qwt5-qt5)      {INCLUDEPATH+=/usr/lib64/qt5/include/qwt5-qt5}
+        exists(/usr/lib64/qt5/include/qwtplot3d-qt5) {INCLUDEPATH+=/usr/lib64/qt5/include/qwtplot3d-qt5}
+
         system (ls /usr/lib*/libqwt5-qt5.so) {
             LIBS+=-lqwt5-qt5
         }
