--- scidavis-2.8/scidavis/CMakeLists.txt.orig	2022-04-14 17:55:22.121350974 +0100
+++ scidavis-2.8/scidavis/CMakeLists.txt	2022-04-14 17:55:11.466351482 +0100
@@ -52,6 +52,7 @@
   )
 
 # SciDAVis expects qm files to be in the translations directory. qt5_create_translation puts them somewhere else
+add_custom_command(TARGET scidavis COMMAND mkdir -p translations)
 add_custom_command(TARGET scidavis COMMAND cp ${QM_FILES} translations)
 
 target_link_libraries( scidavis libscidavis )
