--- scidavis-2.4.0/fitPlugins/CMakeLists.txt.orig	2021-05-05 02:51:49.000000000 +0100
+++ scidavis-2.4.0/fitPlugins/CMakeLists.txt	2021-05-05 21:38:16.708543154 +0100
@@ -19,6 +19,6 @@
     )
 else()
   install( TARGETS exp_saturation explin fitRational0 fitRational1 planck_wavelength
-    LIBRARY DESTINATION lib/scidavis/plugins
+    LIBRARY DESTINATION lib${LIB_SUFFIX}/scidavis/plugins
     )
 endif()
