Index: scidavis-1.D9/libscidavis/src/Graph3D.h
===================================================================
--- scidavis-1.D9.orig/libscidavis/src/Graph3D.h
+++ scidavis-1.D9/libscidavis/src/Graph3D.h
@@ -29,8 +29,8 @@
 #ifndef GRAPH3D_H
 #define GRAPH3D_H
 
-#include <qwtplot3d/qwt3d_surfaceplot.h>
-#include <qwtplot3d/qwt3d_function.h>
+#include <qwtplot3d-qt4/qwt3d_surfaceplot.h>
+#include <qwtplot3d-qt4/qwt3d_function.h>
 
 #include <QTimer>
 #include <QVector>
Index: scidavis-1.D9/libscidavis/src/Bar.h
===================================================================
--- scidavis-1.D9.orig/libscidavis/src/Bar.h
+++ scidavis-1.D9/libscidavis/src/Bar.h
@@ -29,7 +29,7 @@
 #ifndef BARS_H
 #define BARS_H
 
-#include <qwtplot3d/qwt3d_plot.h>
+#include <qwtplot3d-qt4/qwt3d_plot.h>
 
 //! 3D bars (modifed enrichment from QwtPlot3D)
 class Bar : public Qwt3D::VertexEnrichment
Index: scidavis-1.D9/libscidavis/src/Cone3D.h
===================================================================
--- scidavis-1.D9.orig/libscidavis/src/Cone3D.h
+++ scidavis-1.D9/libscidavis/src/Cone3D.h
@@ -29,7 +29,7 @@
 #ifndef MYCONES_H
 #define MYCONES_H
 
-#include <qwtplot3d/qwt3d_plot.h>
+#include <qwtplot3d-qt4/qwt3d_plot.h>
 
 //! 3D cone class (based on QwtPlot3D)
 class Cone3D : public Qwt3D::VertexEnrichment
Index: scidavis-1.D9/libscidavis/src/Graph3D.cpp
===================================================================
--- scidavis-1.D9.orig/libscidavis/src/Graph3D.cpp
+++ scidavis-1.D9/libscidavis/src/Graph3D.cpp
@@ -43,8 +43,8 @@
 #include <QCursor>
 #include <QImageWriter>
 
-#include <qwtplot3d/qwt3d_io_gl2ps.h>
-#include <qwtplot3d/qwt3d_coordsys.h>
+#include <qwtplot3d-qt4/qwt3d_io_gl2ps.h>
+#include <qwtplot3d-qt4/qwt3d_coordsys.h>
 
 #include <gsl/gsl_vector.h>
 #include <fstream>
Index: scidavis-1.D9/libscidavis/src/Plot3DDialog.cpp
===================================================================
--- scidavis-1.D9.orig/libscidavis/src/Plot3DDialog.cpp
+++ scidavis-1.D9/libscidavis/src/Plot3DDialog.cpp
@@ -48,7 +48,7 @@
 #include <QFontDialog>
 #include <QColorDialog>
 
-#include <qwtplot3d/qwt3d_color.h>
+#include <qwtplot3d-qt4/qwt3d_color.h>
 
 Plot3DDialog::Plot3DDialog( QWidget* parent, Qt::WFlags fl )
     : QDialog( parent, fl )
Index: scidavis-1.D9/libscidavis/src/Bar.cpp
===================================================================
--- scidavis-1.D9.orig/libscidavis/src/Bar.cpp
+++ scidavis-1.D9/libscidavis/src/Bar.cpp
@@ -28,8 +28,8 @@
  ***************************************************************************/
 #include <qbitmap.h>
 
-#include <qwtplot3d/qwt3d_color.h>
-#include <qwtplot3d/qwt3d_plot.h>
+#include <qwtplot3d-qt4/qwt3d_color.h>
+#include <qwtplot3d-qt4/qwt3d_plot.h>
 #include "Bar.h"
 
 using namespace Qwt3D;
Index: scidavis-1.D9/libscidavis/src/Cone3D.cpp
===================================================================
--- scidavis-1.D9.orig/libscidavis/src/Cone3D.cpp
+++ scidavis-1.D9/libscidavis/src/Cone3D.cpp
@@ -27,8 +27,8 @@
  *                                                                         *
  ***************************************************************************/
 #include <math.h>
-#include <qwtplot3d/qwt3d_color.h>
-#include <qwtplot3d/qwt3d_plot.h>
+#include <qwtplot3d-qt4/qwt3d_color.h>
+#include <qwtplot3d-qt4/qwt3d_plot.h>
 #include "Cone3D.h"
 
 using namespace Qwt3D;
