--- scons-0.98.1/engine/SCons/Tool/g++.py.orig	2008-04-19 03:20:38.000000000 +0100
+++ scons-0.98.1/engine/SCons/Tool/g++.py	2008-04-20 19:47:09.000000000 +0100
@@ -72,6 +72,7 @@
     else:
         # Original line from Christian Engel added -DPIC:
         #env['SHCXXFLAGS'] = SCons.Util.CLVar('$CXXFLAGS -DPIC')
+        env['SHCXXFLAGS'] = SCons.Util.CLVar('$CXXFLAGS')
         pass
     # determine compiler version
     if env['CXX']:
