--- scorched/src/engine/ModFileEntry.cpp~	2005-06-06 00:54:01.000000000 +0300
+++ scorched/src/engine/ModFileEntry.cpp	2005-10-08 02:14:29.000000000 +0300
@@ -20,7 +20,7 @@
 
 #include <engine/ModFileEntry.h>
 #include <engine/ModFiles.h>
-#include <zlib/zlib.h>
+#include <zlib.h>
 #include <wx/dir.h>
 #include <wx/utils.h>
 
--- scorched/src/landscape/LandscapeMaps.cpp~	2005-08-10 15:39:31.000000000 +0300
+++ scorched/src/landscape/LandscapeMaps.cpp	2005-10-08 02:14:45.000000000 +0300
@@ -31,7 +31,7 @@
 #include <common/RandomGenerator.h>
 #include <common/Defines.h>
 #include <common/Logger.h>
-#include <zlib/zlib.h>
+#include <zlib.h>
 #include <float.h>
 #include <stdlib.h>
 
--- scorched/src/scorched/Makefile.am~	2005-08-30 21:04:22.000000000 +0300
+++ scorched/src/scorched/Makefile.am	2005-10-08 02:15:02.000000000 +0300
@@ -690,39 +690,6 @@
 	../XML/XMLParser.h \
 	../XML/XMLStringBuffer.cpp \
 	../XML/XMLStringBuffer.h \
-	../XML/expat.h \
-	../XML/expat_config.h \
-	../XML/winconfig.h \
-	../XML/xmlparse.c \
-	../XML/xmlrole.c \
-	../XML/xmlrole.h \
-	../XML/xmltok.c \
-	../XML/xmltok.h \
-	../zlib/adler32.c \
-	../zlib/compress.c \
-	../zlib/crc32.c \
-	../zlib/deflate.c \
-	../zlib/deflate.h \
-	../zlib/gzio.c \
-	../zlib/infblock.c \
-	../zlib/infblock.h \
-	../zlib/infcodes.c \
-	../zlib/infcodes.h \
-	../zlib/inffast.c \
-	../zlib/inffast.h \
-	../zlib/inffixed.h \
-	../zlib/inflate.c \
-	../zlib/inftrees.c \
-	../zlib/inftrees.h \
-	../zlib/infutil.c \
-	../zlib/infutil.h \
-	../zlib/trees.c \
-	../zlib/trees.h \
-	../zlib/uncompr.c \
-	../zlib/zconf.h \
-	../zlib/zlib.h \
-	../zlib/zutil.c \
-	../zlib/zutil.h \
 	../actions/ActionVector.cpp \
 	../actions/ActionVector.h \
 	../actions/Animation.cpp \
--- scorched/src/coms/NetServerProtocol.cpp~	2005-06-13 20:31:06.000000000 +0300
+++ scorched/src/coms/NetServerProtocol.cpp	2005-10-08 02:14:04.000000000 +0300
@@ -22,7 +22,7 @@
 #include <coms/NetMessagePool.h>
 #include <coms/NetServer.h>
 #include <common/Logger.h>
-#include <zlib/zlib.h>
+#include <zlib.h>
 
 NetServerProtocol::NetServerProtocol()
 {
--- scorched/src/coms/ComsMessageSender.cpp~	2005-07-20 00:45:36.000000000 +0300
+++ scorched/src/coms/ComsMessageSender.cpp	2005-10-08 02:13:38.000000000 +0300
@@ -27,7 +27,7 @@
 #include <tank/TankContainer.h>
 #include <client/ScorchedClient.h>
 #include <server/ScorchedServer.h>
-#include <zlib/zlib.h>
+#include <zlib.h>
 #include <set>
 
 bool ComsMessageSender::sendToServer(ComsMessage &message)
--- scorched/src/tank/TankAvatar.cpp~	2005-03-22 19:45:49.000000000 +0200
+++ scorched/src/tank/TankAvatar.cpp	2005-10-08 02:15:15.000000000 +0300
@@ -22,7 +22,7 @@
 #include <GLEXT/GLTexture.h>
 #include <GLEXT/GLGif.h>
 #include <common/Defines.h>
-#include <zlib/zlib.h>
+#include <zlib.h>
 #include <stdio.h>
 
 GLTexture *TankAvatar::defaultTexture_ = 0;
--- scorched/src/XML/XMLParser.h~	2005-06-11 19:53:41.000000000 +0300
+++ scorched/src/XML/XMLParser.h	2005-10-08 02:15:26.000000000 +0300
@@ -23,7 +23,7 @@
 
 #include <string>
 #include <list>
-#include <XML/expat.h>
+#include <expat.h>
 #include <common/FileLines.h>
 #include <common/Vector.h>
 
