--- src/sound/Sound.cpp.orig	2005-08-06 12:57:39.000000000 +0100
+++ src/sound/Sound.cpp	2005-08-11 13:40:15.000000000 +0100
@@ -89,7 +89,7 @@
 bool Sound::init(int channels)
 {
 	const char *deviceName = 0;
-	ALCdevice *soundDevice = alcOpenDevice((const ALubyte *) deviceName);
+	ALCdevice *soundDevice = alcOpenDevice((const ALchar *) deviceName);
 	if (!soundDevice)
 	{
 		dialogExit("Failed to find sound device %s",
