--- scorched/src/scorched/NetLan.cpp~	2006-07-16 12:50:44.000000000 +0200
+++ scorched/src/scorched/NetLan.cpp	2006-07-16 12:50:44.000000000 +0200
@@ -96,9 +96,9 @@
 	// Radio Buttons
 	wxFlexGridSizer *radioSizer = new wxFlexGridSizer(1, 8, 5, 5);
 	IDC_RADIO_NET_CTRL = new wxRadioButton(parent, IDC_RADIO_NET, 
-		"Internet", wxDefaultPosition, wxDefaultSize, wxRB_GROUP);
+		wxT("Internet"), wxDefaultPosition, wxDefaultSize, wxRB_GROUP);
 	IDC_RADIO_LAN_CTRL = new wxRadioButton(parent, IDC_RADIO_LAN, 
-		"LAN", wxDefaultPosition, wxDefaultSize, 0);
+		wxT("LAN"), wxDefaultPosition, wxDefaultSize, 0);
 	radioSizer->Add(IDC_RADIO_NET_CTRL, 0, wxRIGHT, 10);
 	radioSizer->Add(IDC_RADIO_LAN_CTRL, 0, wxRIGHT, 0);
 	sizer->Add(radioSizer, 0, wxALIGN_CENTER | wxALL, 5);
--- scorched/src/scorched/OptionEntrySetter.cpp~	2006-07-16 13:04:56.000000000 +0200
+++ scorched/src/scorched/OptionEntrySetter.cpp	2006-07-16 13:04:56.000000000 +0200
@@ -122,7 +122,7 @@
 		{
 			sizer->Add(control = 
 				new wxCheckBox(parent, -1,
-					"",
+					wxT(""),
 					wxDefaultPosition, wxDefaultSize),
 					0, wxALIGN_LEFT);
 			control->SetToolTip(
--- scorched/src/scorched/NetLan.cpp~	2006-07-16 20:36:19.000000000 +0200
+++ scorched/src/scorched/NetLan.cpp	2006-07-16 20:36:19.000000000 +0200
@@ -71,17 +71,11 @@
 		new wxTextCtrl(parent, IDC_EDIT_PASSWORD,
 		wxT(""), wxDefaultPosition, wxSize(285, -1), wxTE_PASSWORD);
 	IDC_CLEAR_CTRL = 
-		new wxButton(parent, IDC_CLEAR,
-		wxT("Clear"),
-		wxDefaultPosition, wxSize(-1, 20));
+		new wxButton(parent, IDC_CLEAR, wxT("Clear"));
 	IDC_CLEAR_NAME_CTRL = 
-		new wxButton(parent, IDC_CLEAR_NAME,
-		wxT("Clear"),
-		wxDefaultPosition, wxSize(-1, 20));
+		new wxButton(parent, IDC_CLEAR_NAME, wxT("Clear"));
 	IDC_CLEAR_PASSWORD_CTRL = 
-		new wxButton(parent, IDC_CLEAR_PASSWORD,
-		wxT("Clear"),
-		wxDefaultPosition, wxSize(-1, 20));
+		new wxButton(parent, IDC_CLEAR_PASSWORD, wxT("Clear"));
 	nameSizer->Add(nameText, 0, wxALIGN_CENTER_VERTICAL);
 	nameSizer->Add(IDC_EDIT_SERVER_CTRL, 0, 0);
 	nameSizer->Add(IDC_CLEAR_CTRL, 0, 0);
