diff -ur scorched.orig/src/scorched/MainDialog.cpp scorched/src/scorched/MainDialog.cpp
--- scorched.orig/src/scorched/MainDialog.cpp	2006-05-14 17:00:11.000000000 +0200
+++ scorched/src/scorched/MainDialog.cpp	2006-07-16 11:05:27.000000000 +0200
@@ -122,8 +122,8 @@
 #ifdef __DARWIN__
 	system(formatString("open %s", url));
 #else
-	system(formatString("mozilla %s", url));
-	dialogMessage("Web site location", formatString("%s", url));
+	system(formatString("htmlview %s&", url));
+//	dialogMessage("Web site location", formatString("%s", url));
 #endif // __DARWIN__
 #endif // _WIN32
 }
