diff -ur scorched.orig/configure scorched/configure
--- scorched.orig/configure	2006-07-15 17:15:30.000000000 +0200
+++ scorched/configure	2006-07-16 11:00:26.000000000 +0200
@@ -3946,7 +3946,7 @@
 	if test x"$use_static_openal" = x"yes"; then
 		LIBS="$LIBS /usr/local/lib/libopenal.a"
 	else
-		LIBS="$LIBS `$OPENAL_CONFIG --libs`"
+		LIBS="$LIBS `$OPENAL_CONFIG --libs` -lalut"
 	fi
 
 	CFLAGS="$CFLAGS `$OPENAL_CONFIG --cflags`"
diff -ur scorched.orig/configure.ac scorched/configure.ac
--- scorched.orig/configure.ac	2006-07-15 14:01:30.000000000 +0200
+++ scorched/configure.ac	2006-07-16 10:59:21.000000000 +0200
@@ -114,7 +114,7 @@
 	if test x"$use_static_openal" = x"yes"; then
 		LIBS="$LIBS /usr/local/lib/libopenal.a"
 	else
-		LIBS="$LIBS `$OPENAL_CONFIG --libs`"
+		LIBS="$LIBS `$OPENAL_CONFIG --libs` -lalut"
 	fi
 	
 	CFLAGS="$CFLAGS `$OPENAL_CONFIG --cflags`"
Only in scorched: configure~
