--- screem-0.17.1/src/screem-preview.c-	2006-02-04 13:26:47.000000000 -0500
+++ screem-0.17.1/src/screem-preview.c	2006-02-04 13:26:55.000000000 -0500
@@ -587,12 +587,14 @@
 	return document;	
 }
 
+#ifdef SCREEM_MOZ
 static gint screem_preview_moz_open_uri( GtkMozEmbed *embed, const gchar *aURI )
 {
 	/* stop links being followed */
 	g_print( "OPEN URI: %s\n", aURI );
 	return 1;
 }
+#endif
 
 /* G Object stuff */
 
