--- scribus-1.6.0/scribus/plugins/import/pdf/slaoutput.cpp.orig	2023-12-31 15:58:24.000000000 +0000
+++ scribus-1.6.0/scribus/plugins/import/pdf/slaoutput.cpp	2024-03-04 17:04:22.256309929 +0000
@@ -1845,7 +1845,7 @@
 	VGradient FillGradient = VGradient(VGradient::linear);
 	FillGradient.clearStops();
 	GfxColorSpace *color_space = shading->getColorSpace();
-	if (func->getType() == 3)
+	if (func->getType() == Function::Type::Stitching)
 	{
 		StitchingFunction *stitchingFunc = (StitchingFunction*)func;
 		const double *bounds = stitchingFunc->getBounds();
@@ -1867,7 +1867,7 @@
 			FillGradient.addStop( ScColorEngine::getShadeColor(m_doc->PageColors[stopColor], m_doc, shade), stopPoint, 0.5, 1.0, stopColor, shade );
 		}
 	}
-	else if ((func->getType() == 2) || (func->getType() == 0))
+	else if ((func->getType() == Function::Type::Exponential) || (func->getType() == Function::Type::Sampled))
 	{
 		GfxColor stop1;
 		shading->getColor(0.0, &stop1);
@@ -1979,7 +1979,7 @@
 	VGradient FillGradient = VGradient(VGradient::linear);
 	FillGradient.clearStops();
 	GfxColorSpace *color_space = shading->getColorSpace();
-	if (func->getType() == 3)
+	if (func->getType() == Function::Type::Stitching)
 	{
 		StitchingFunction *stitchingFunc = (StitchingFunction*)func;
 		const double *bounds = stitchingFunc->getBounds();
@@ -2001,7 +2001,7 @@
 			FillGradient.addStop( ScColorEngine::getShadeColor(m_doc->PageColors[stopColor], m_doc, shade), stopPoint, 0.5, 1.0, stopColor, shade );
 		}
 	}
-	else if ((func->getType() == 2) || (func->getType() == 0))
+	else if ((func->getType() == Function::Type::Exponential) || (func->getType() == Function::Type::Sampled))
 	{
 		GfxColor stop1;
 		shading->getColor(0.0, &stop1);
