--- scribus-1.5.8/cmake/modules/FindLIBPODOFO.cmake.orig	2022-01-23 16:16:42.000000000 +0000
+++ scribus-1.5.8/cmake/modules/FindLIBPODOFO.cmake	2023-05-27 16:56:08.356389840 +0100
@@ -12,23 +12,23 @@
 	endif(NOT DEFINED LIBPODOFO_SHARED)
 endif(WIN32)
 
-find_path(LIBPODOFO_INCLUDE_DIR
-	NAMES podofo/podofo.h
-	PATHS
-	"${LIBPODOFO_DIR}/include"
-	"${LIBPODOFO_DIR}/src"
-	"${LIBPODOFO_DIR}"
-	)
+#find_path(LIBPODOFO_INCLUDE_DIR
+#	NAMES podofo/podofo.h
+#	PATHS
+#	"${LIBPODOFO_DIR}/include"
+#	"${LIBPODOFO_DIR}/src"
+#	"${LIBPODOFO_DIR}"
+#	)
 
-set(LIBPODOFO_FIND_QUIETLY ON)
+#set(LIBPODOFO_FIND_QUIETLY ON)
 
-find_library(LIBPODOFO_LIBRARY
-	NAMES libpodofo podofo
-	PATHS
-	"${LIBPODOFO_DIR}/lib"
-	"${LIBPODOFO_DIR}/src"
-	"${LIBPODOFO_DIR}"
-	)
+#find_library(LIBPODOFO_LIBRARY
+#	NAMES libpodofo podofo
+#	PATHS
+#	"${LIBPODOFO_DIR}/lib"
+#	"${LIBPODOFO_DIR}/src"
+#	"${LIBPODOFO_DIR}"
+#	)
 
 if(LIBPODOFO_INCLUDE_DIR AND LIBPODOFO_LIBRARY)
 	if(NOT LIBPODOFO_CONFIG_H)
