--- scribus-1.6.5/scribus/plugins/import/pdf/slaoutput.cpp.orig	2025-12-14 22:43:57.000000000 +0000
+++ scribus-1.6.5/scribus/plugins/import/pdf/slaoutput.cpp	2026-01-07 12:18:52.602453109 +0000
@@ -2496,7 +2496,11 @@
 //	qDebug() << "Draw Image Mask";
 	auto imgStr = std::make_shared<ImageStream>(str, width, 1, 1);
 #if POPPLER_ENCODED_VERSION >= POPPLER_VERSION_ENCODE(25, 02, 0)
+#if POPPLER_ENCODED_VERSION >= POPPLER_VERSION_ENCODE(26, 01, 0)
+	bool resetDone = imgStr->rewind();
+#else
 	bool resetDone = imgStr->reset();
+#endif
 	if (!resetDone)
 		return;
 #else
@@ -2577,7 +2581,11 @@
 //	qDebug() << "SlaOutputDev::drawSoftMaskedImage Masked Image Components" << colorMap->getNumPixelComps();
 	auto imgStr = std::make_shared<ImageStream>(str, width, colorMap->getNumPixelComps(), colorMap->getBits());
 #if POPPLER_ENCODED_VERSION >= POPPLER_VERSION_ENCODE(25, 02, 0)
+#if POPPLER_ENCODED_VERSION >= POPPLER_VERSION_ENCODE(26, 01, 0)
+	bool resetDone = imgStr->rewind();
+#else
 	bool resetDone = imgStr->reset();
+#endif
 	if (!resetDone)
 		return;
 #else
@@ -2599,7 +2607,11 @@
 
 	auto mskStr = std::make_shared<ImageStream>(maskStr, maskWidth, maskColorMap->getNumPixelComps(), maskColorMap->getBits());
 #if POPPLER_ENCODED_VERSION >= POPPLER_VERSION_ENCODE(25, 02, 0)
+#if POPPLER_ENCODED_VERSION >= POPPLER_VERSION_ENCODE(26, 01, 0)
+	bool mskResetDone = mskStr->rewind();
+#else
 	bool mskResetDone = mskStr->reset();
+#endif
 	if (!mskResetDone)
 		return;
 #else
@@ -2661,7 +2673,11 @@
 //	qDebug() << "SlaOutputDev::drawMaskedImage";
 	auto imgStr = std::make_shared<ImageStream>(str, width, colorMap->getNumPixelComps(), colorMap->getBits());
 #if POPPLER_ENCODED_VERSION >= POPPLER_VERSION_ENCODE(25, 02, 0)
+#if POPPLER_ENCODED_VERSION >= POPPLER_VERSION_ENCODE(26, 01, 0)
+	bool resetDone = imgStr->rewind();
+#else
 	bool resetDone = imgStr->reset();
+#endif
 	if (resetDone)
 		return;
 #else
@@ -2683,7 +2699,11 @@
 
 	auto mskStr = std::make_shared<ImageStream>(maskStr, maskWidth, 1, 1);
 #if POPPLER_ENCODED_VERSION >= POPPLER_VERSION_ENCODE(25, 02, 0)
+#if POPPLER_ENCODED_VERSION >= POPPLER_VERSION_ENCODE(26, 01, 0)
+	bool mskResetDone = mskStr->rewind();
+#else
 	bool mskResetDone = mskStr->reset();
+#endif
 	if (!mskResetDone)
 		return;
 #else
@@ -2733,7 +2753,11 @@
 {
 	auto imgStr = std::make_shared<ImageStream>(str, width, colorMap->getNumPixelComps(), colorMap->getBits());
 #if POPPLER_ENCODED_VERSION >= POPPLER_VERSION_ENCODE(25, 02, 0)
+#if POPPLER_ENCODED_VERSION >= POPPLER_VERSION_ENCODE(26, 01, 0)
+	bool resetDone = imgStr->rewind();
+#else
 	bool resetDone = imgStr->reset();
+#endif
 	if (!resetDone)
 		return;
 #else
@@ -3125,10 +3149,15 @@
 #endif
 	const double *textMat = nullptr;
 	double m11, m12, m21, m22, fontSize;
-	SplashCoord mat[4] = { 1.0, 0.0, 0.0, 1.0 };
 	int n = 0;
 	int faceIndex = 0;
+#if POPPLER_ENCODED_VERSION >= POPPLER_VERSION_ENCODE(26, 1, 0)
+	std::array<SplashCoord, 4> mat = { 1.0, 0.0, 0.0, 1.0 };
+	std::array<SplashCoord, 6> matrix = { 1.0, 0.0, 0.0, 1.0, 0.0, 0.0 };
+#else
+	SplashCoord mat[4] = { 1.0, 0.0, 0.0, 1.0 };
 	SplashCoord matrix[6] = { 1.0, 0.0, 0.0, 1.0, 0.0, 0.0 };
+#endif
 
 	m_font = nullptr;
 
