--- scribus-1.6.4/scribus/plugins/import/pdf/slaoutput.cpp.orig	2025-04-20 14:52:19.000000000 +0100
+++ scribus-1.6.4/scribus/plugins/import/pdf/slaoutput.cpp	2025-06-04 21:18:59.820811136 +0100
@@ -606,7 +606,11 @@
 		FormWidget *fm = m_formWidgets->getWidget(i);
 		if (!fm)
 			continue;
+#if POPPLER_ENCODED_VERSION >= POPPLER_VERSION_ENCODE(25, 6, 0)
+		AnnotWidget *ano = fm->getWidgetAnnotation().get();
+#else
 		AnnotWidget *ano = fm->getWidgetAnnotation();
+#endif
 		if (!ano)
 			continue;
 		if (ano != (AnnotWidget*) annota)
