--- sdcc-4.4.0/src/pic16/glue.c.orig	2023-12-05 19:44:11.000000000 +0000
+++ sdcc-4.4.0/src/pic16/glue.c	2024-11-04 19:27:31.698797407 +0000
@@ -1802,7 +1802,7 @@
       unsigned long ramsize = pic16 ? pic16->RAMsize : 0x200;
       symbol *sym;
 
-      reg = newReg (REG_SFR, PO_SFR_REGISTER, ramsize-1, "_sram_end", 0, 0, NULL);
+      reg_info *reg = newReg (REG_SFR, PO_SFR_REGISTER, ramsize-1, "_sram_end", 0, 0, NULL);
       addSet (&pic16_fix_udata, reg);
 
       sym = newSymbol ("sram_end", 0);
