--- sdcc-4.5.0/sdas/linksrc/aslink.h.orig	2023-05-15 18:14:52.000000000 +0100
+++ sdcc-4.5.0/sdas/linksrc/aslink.h	2025-09-12 14:08:40.908593360 +0100
@@ -1312,7 +1312,7 @@
 extern  VOID            sflush(void);
 
 /* EEP: lkelf.c */
-extern  VOID            elf();
+extern  VOID            elf(int i);
 
 /* JCF: lkmem.c */
 extern int summary(struct area * xp);
--- sdcc-4.5.0/debugger/mcs51/sdcdb.h.orig	2024-02-04 18:33:46.000000000 +0000
+++ sdcc-4.5.0/debugger/mcs51/sdcdb.h	2025-09-12 14:14:37.754516587 +0100
@@ -60,7 +60,7 @@
 #define TRUE 1
 #define FALSE !TRUE
 
-typedef short bool;
+// typedef short bool;
 
 #ifndef max
 #define max(a,b) (a > b ? a : b)
