--- SDL_mixer-1.2.8/mikmod/mlreg.c.orig	2007-07-02 03:03:59.000000000 +0100
+++ SDL_mixer-1.2.8/mikmod/mlreg.c	2007-11-11 21:55:23.000000000 +0000
@@ -35,29 +35,29 @@
 void MikMod_RegisterAllLoaders_internal(void)
 {
 #if 1
-	_mm_registerloader(&load_it);
-	_mm_registerloader(&load_mod);
-	_mm_registerloader(&load_s3m);
-	_mm_registerloader(&load_xm);
+	MikMod_RegisterLoader(&load_it);
+	MikMod_RegisterLoader(&load_mod);
+	MikMod_RegisterLoader(&load_s3m);
+	MikMod_RegisterLoader(&load_xm);
 #else
-	_mm_registerloader(&load_669);
-	_mm_registerloader(&load_amf);
-	_mm_registerloader(&load_dsm);
-	_mm_registerloader(&load_far);
-	_mm_registerloader(&load_gdm);
-	_mm_registerloader(&load_it);
-	_mm_registerloader(&load_imf);
-	_mm_registerloader(&load_mod);
-	_mm_registerloader(&load_med);
-	_mm_registerloader(&load_mtm);
-	_mm_registerloader(&load_s3m);
-	_mm_registerloader(&load_stm);
-	_mm_registerloader(&load_stx);
-	_mm_registerloader(&load_ult);
-	_mm_registerloader(&load_uni);
-	_mm_registerloader(&load_xm);
+	MikMod_RegisterLoader(&load_669);
+	MikMod_RegisterLoader(&load_amf);
+	MikMod_RegisterLoader(&load_dsm);
+	MikMod_RegisterLoader(&load_far);
+	MikMod_RegisterLoader(&load_gdm);
+	MikMod_RegisterLoader(&load_it);
+	MikMod_RegisterLoader(&load_imf);
+	MikMod_RegisterLoader(&load_mod);
+	MikMod_RegisterLoader(&load_med);
+	MikMod_RegisterLoader(&load_mtm);
+	MikMod_RegisterLoader(&load_s3m);
+	MikMod_RegisterLoader(&load_stm);
+	MikMod_RegisterLoader(&load_stx);
+	MikMod_RegisterLoader(&load_ult);
+	MikMod_RegisterLoader(&load_uni);
+	MikMod_RegisterLoader(&load_xm);
 
-	_mm_registerloader(&load_m15);
+	MikMod_RegisterLoader(&load_m15);
 #endif
 }
 
