--- semantik-1.0.3/wscript.orig	2018-01-06 11:42:47.000000000 +0000
+++ semantik-1.0.3/wscript	2018-06-19 22:21:59.880782495 +0100
@@ -63,7 +63,7 @@
 src/fig/filetree.h
 src/fig/diagram_document.h
 ''',
-		use = 'KDECORE QT5CORE QT5WIDGETS QT5GUI QT5XML QT5PRINTSUPPORT QT5SVG semlib',
+		use = 'KDECORE QT5CORE QT5WIDGETS QT5GUI QT5XML QT5PRINTSUPPORT QT5SVG QT5DBUS semlib',
 		target = 'src/semantik-d',
 		includes = '. src src/fig src/base')
 
