--- semantik-0.5.1/wscript.orig	2007-08-08 06:14:45.000000000 +0100
+++ semantik-0.5.1/wscript	2007-08-13 12:35:36.000000000 +0100
@@ -83,7 +83,6 @@
 
 	install_files('SEMANTIK_DIR', 'flags', 'src/flags/*.svg')
 	install_files('TEMPLATE_DIR', '', rt+'*.py')
-	Common.do_install(os.path.abspath('waf'), bld.env_of_name('default')['TEMPLATE_DIR']+'waf')
 	install_files('TEMPLATE_DIR', 'pdflatex', rt+'pdflatex/*')
 	for x in 'html odt odp java'.split():
 		install_files('TEMPLATE_DIR', x, rt+'%s/*' % x)
