--- sge-030809/sge_tt_text.cpp.orig	2003-08-08 13:48:08.000000000 +0100
+++ sge-030809/sge_tt_text.cpp	2008-10-08 22:41:43.000000000 +0100
@@ -39,7 +39,7 @@
 #include <freetype/freetype.h>
 #include <freetype/ftoutln.h>
 #include <freetype/ttnameid.h>
-#include <freetype/internal/ftobjs.h>
+//#include <freetype/internal/ftobjs.h>
 
 /* The structure used to hold glyph information (cached) */
 struct glyph {
@@ -294,7 +294,7 @@
 	}
 	memset(stream, 0, sizeof(*stream));
 
-	stream->memory = _sge_library->memory;
+	stream->memory = NULL;
 	stream->read = RWread;
 	stream->descriptor.pointer = src;
 	stream->pos = (unsigned long)position;
