--- simgear-2018.3.2/CMakeModules/BoostTestTargets.cmake.orig	2019-01-29 13:17:36.000000000 +0000
+++ simgear-2018.3.2/CMakeModules/BoostTestTargets.cmake	2019-07-23 21:58:12.949316090 +0100
@@ -46,7 +46,7 @@
 if(NOT Boost_FOUND)
 	find_package(Boost 1.34.0 QUIET)
 endif()
-if("${Boost_VERSION}0" LESS "1034000")
+if("${Boost_VERSION_MACRO}0" LESS "1034000")
 	set(_shared_msg
 		"NOTE: boost::test-based targets and tests cannot "
 		"be added: boost >= 1.34.0 required but not found. "
--- simgear-2018.3.2/3rdparty/udns/CMakeLists.txt.orig	2019-01-29 13:17:36.000000000 +0000
+++ simgear-2018.3.2/3rdparty/udns/CMakeLists.txt	2019-07-24 19:33:39.061708839 +0100
@@ -12,8 +12,8 @@
 # somehow it is not, if somebody know a better way to define WINDOWS, please fix
 check_include_file(windows.h  WINDOWS)
 
-CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/config.h.in ${CMAKE_BINARY_DIR}/config.h)
-include_directories(${CMAKE_BINARY_DIR})
+CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/config.h.in ${CMAKE_BINARY_DIR}/simgear/config.h)
+include_directories(${CMAKE_BINARY_DIR}/simgear)
 
 set(HEADERS
 )
