--- simulavrxx-20060512/src/Makefile.am.orig	2005-07-14 04:09:33.000000000 +0100
+++ simulavrxx-20060512/src/Makefile.am	2006-05-12 10:24:06.000000000 +0100
@@ -194,7 +194,7 @@
 	$(swig_path) -c++ -python -outdir python/ -o $@ $?
 
 python/_simulavr.so: python/simulavr_wrap.cpp simulavr
-	g++ -I$(srcdir) -I/usr/include/python2.3 $< $(patsubst %.lo,%.o,$(libavrsim_pp_la_OBJECTS)) $(bfd_a_location)/libbfd.a $(libiberty_location)/libiberty.a -lc -lm -lncurses -shared -o $@ $(NonPortableDarwin_GXX_Flags)
+	g++ -I$(srcdir) -I/usr/include/python2.4 $< $(patsubst %.lo,%.o,$(libavrsim_pp_la_OBJECTS)) $(bfd_a_location)/libbfd.a $(libiberty_location)/libiberty.a -lc -lm -lncurses -shared -o $@ $(NonPortableDarwin_GXX_Flags)
 
 clean_python_simulavr:
 	$(RM) python/_simulavr.so python/simulavr.py python/simulavr_wrap.cpp
