--- smb4k-2.0.0/CMakeLists.txt.orig	2017-03-11 09:03:54.000000000 +0000
+++ smb4k-2.0.0/CMakeLists.txt	2017-03-12 09:30:28.084514987 +0000
@@ -105,7 +105,7 @@
 ########### install files ###############
 
 feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
-find_package(KF5I18n CONFIG REQUIRED)
-ki18n_install(po)
-find_package(KF5DocTools CONFIG REQUIRED)
-kdoctools_install(po)
+#find_package(KF5I18n CONFIG REQUIRED)
+#ki18n_install(po)
+#find_package(KF5DocTools CONFIG REQUIRED)
+#kdoctools_install(po)
