--- soundkonverter-0.3.11/admin/cvs.sh.orig	2008-03-02 10:34:18.000000000 +0000
+++ soundkonverter-0.3.11/admin/cvs.sh	2010-03-26 17:01:15.000000000 +0000
@@ -68,7 +68,7 @@
     echo "*** KDE requires automake $required_automake_version"
     exit 1
     ;;
-  automake*1.6.* | automake*1.[7-9]* | automake*1.10*)
+  automake*1.6.* | automake*1.[7-9]* | automake*1.1[0-9]*)
     echo "*** $AUTOMAKE_STRING found."
     UNSERMAKE=no
     ;;
