--- sox-14.1.0/src/ffmpeg.c.orig	2008-07-21 04:44:13.000000000 +0100
+++ sox-14.1.0/src/ffmpeg.c	2008-10-05 22:26:38.000000000 +0100
@@ -86,7 +86,7 @@
 
   codec = avcodec_find_decoder(enc->codec_id);
   enc->workaround_bugs = 1;
-  enc->error_resilience = 1;
+  enc->error_recognition = 1;
   if (!codec || avcodec_open(enc, codec) < 0)
     return -1;
   if (enc->codec_type != CODEC_TYPE_AUDIO) {
