--- stunnel-5.08/configure.orig	2014-12-03 10:35:21.000000000 +0000
+++ stunnel-5.08/configure	2014-12-14 12:04:41.746434527 +0000
@@ -13474,7 +13474,7 @@
 
         for main_dir in usr/local usr/lib usr/pkg opt/local opt usr ""; do
             for sub_dir in /ssl /openssl ""; do
-                check_ssl_dir "$SYSROOT$main_dir$sub_dir" && break 2
+                check_ssl_dir "$SYSROOT/$main_dir$sub_dir" && break 2
             done
         done
 
