--- SuiteSparse-2.3.1/UFconfig/UFconfig.mk.orig	2006-11-29 11:18:52.000000000 +0000
+++ SuiteSparse-2.3.1/UFconfig/UFconfig.mk	2007-01-09 13:31:19.000000000 +0000
@@ -73,8 +73,9 @@
 # These settings will probably not work, since there is no fixed convention for
 # naming the BLAS and LAPACK library (*.a or *.so) files.  Assume the Goto
 # BLAS are available.
-BLAS = -lgoto -lgfortran -lgfortranbegin
-LAPACK = -llapack
+#BLAS = -lgoto -lgfortran -lgfortranbegin
+BLAS = -lf77blas -latlas -lg2c
+LAPACK = -llapack -lcblas
 
 # The BLAS might not contain xerbla, an error-handling routine for LAPACK and
 # the BLAS.  Also, the standard xerbla requires the Fortran I/O library, and
