--- SuperTux-v0.5.1/src/video/glutil.hpp.orig	2016-11-05 14:15:47.000000000 +0000
+++ SuperTux-v0.5.1/src/video/glutil.hpp	2018-12-14 12:16:19.022645463 +0000
@@ -34,6 +34,9 @@
 
 #ifndef GL_VERSION_ES_CM_1_0
 #  include <GL/glew.h>
+// This is a workaround for mesa drivers that now use __gl_glext_h_ instead of
+// __glext_h_ in header file which is not defined in current glew version
+#   define __gl_glext_h_
 #endif
 
 #if defined(MACOSX)
