--- SuperTux-v0.6.3/external/partio_zip/zip_manager.hpp.orig	2021-12-23 11:37:26.000000000 +0000
+++ SuperTux-v0.6.3/external/partio_zip/zip_manager.hpp	2022-08-15 16:36:45.076621053 +0100
@@ -44,6 +44,7 @@
 #include <fstream>
 #include <iostream>
 #include <map>
+#include <memory>
 #include <string>
 #include <stdexcept>
 #include <vector>
--- SuperTux-v0.6.3/external/squirrel/sq/CMakeLists.txt.orig	2021-12-23 11:37:39.000000000 +0000
+++ SuperTux-v0.6.3/external/squirrel/sq/CMakeLists.txt	2025-08-17 21:46:04.681589844 +0100
@@ -13,7 +13,7 @@
 if(NOT DEFINED DISABLE_STATIC)
   add_executable(sq_static sq.c)
   set_target_properties(sq_static PROPERTIES LINKER_LANGUAGE C)
-  target_link_libraries(sq_static squirrel_static sqstdlib_static)
+  target_link_libraries(sq_static squirrel_static sqstdlib_static m stdc++)
   if(NOT DEFINED SQ_DISABLE_INSTALLER)
     install(TARGETS sq_static RUNTIME DESTINATION ${INSTALL_BIN_DIR})
   endif()
--- SuperTux-v0.6.3/CMakeLists.txt.orig	2021-12-23 11:37:24.000000000 +0000
+++ SuperTux-v0.6.3/CMakeLists.txt	2025-08-17 21:34:59.261236198 +0100
@@ -171,7 +171,7 @@
 else(ENABLE_BOOST_STATIC_LIBS)
   set(Boost_USE_STATIC_LIBS FALSE)
 endif(ENABLE_BOOST_STATIC_LIBS)
-find_package(Boost REQUIRED COMPONENTS filesystem system date_time locale)
+find_package(Boost REQUIRED COMPONENTS filesystem date_time locale)
 include_directories(SYSTEM ${Boost_INCLUDE_DIR})
 link_directories(${Boost_LIBRARY_DIRS})
 
