--- superlu-4.3/make.inc.orig	2011-10-27 20:49:47.000000000 +0100
+++ superlu-4.3/make.inc	2011-12-07 10:21:55.000000000 +0000
@@ -21,19 +21,19 @@
 #
 #  The name of the libraries to be created/linked to
 #
-SuperLUroot	= $(HOME)/Codes/SuperLU/SuperLU_4.2
-SUPERLULIB   	= $(SuperLUroot)/lib/libsuperlu_4.2.a
+SuperLUroot	= $(CURDIR)/..
+SUPERLULIB   	= $(SuperLUroot)/lib/libsuperlu_4.3.a
 TMGLIB       	= libtmglib.a
 
 ## BLASLIB   	= $(SuperLUroot)/lib/libblas.a
 
 BLASDEF 	= -DUSE_VENDOR_BLAS
-BLASLIB 	= -L$(HOME)/lib/GotoBLAS -lgoto
+#BLASLIB 	= -L$(HOME)/lib/GotoBLAS -lgoto
 
 ## ATLAS BLAS causes single-precision to fail
 #BLASLIB   	= -L/usr/lib/atlas -lblas
 ## This BLAS causes single-precision to fail the test in SuperLU
-#BLASLIB 	= -L/usr/lib -lblas
+BLASLIB 	= -L/usr/lib -lf77blas -latlas
 
 LIBS		= $(SUPERLULIB) $(BLASLIB)
 
