--- swig-2.0.8/Source/Modules/php.cxx.orig	2012-05-23 03:58:24.000000000 +0100
+++ swig-2.0.8/Source/Modules/php.cxx	2012-11-28 21:20:58.000000000 +0000
@@ -416,6 +416,7 @@
     Append(s_header, "static void SWIG_FAIL() __attribute__ ((__noreturn__));\n");
     Append(s_header, "#endif\n\n");
     Append(s_header, "static void SWIG_FAIL() {\n");
+    Append(s_header, "  TSRMLS_FETCH();\n");
     Append(s_header, "    zend_error(SWIG_ErrorCode(), \"%s\", SWIG_ErrorMsg());\n");
     // zend_error() should never return with the parameters we pass, but if it
     // does, we really don't want to let SWIG_FAIL() return.  This also avoids
