--- synfig-core-20070530/src/modules/mod_libavcodec/trgt_av.cpp.orig	2007-05-30 20:23:39.000000000 +0100
+++ synfig-core-20070530/src/modules/mod_libavcodec/trgt_av.cpp	2007-05-31 15:03:37.000000000 +0100
@@ -23,6 +23,7 @@
 /* === H E A D E R S ======================================================= */
 
 #define SYNFIG_NO_ANGLE
+#define __STDC_CONSTANT_MACROS
 
 #ifdef USING_PCH
 #	include "pch.h"
