--- synfigstudio-1.0.2/src/gui/states/state_zoom.cpp.orig	2016-07-09 06:27:40.482896333 +0100
+++ synfigstudio-1.0.2/src/gui/states/state_zoom.cpp	2016-07-09 06:29:20.282896700 +0100
@@ -31,7 +31,6 @@
 #endif
 
 #include <sigc++/signal.h>
-#include <sigc++/object.h>
 
 #include <ETL/handle>
 #include <synfig/vector.h>
--- synfigstudio-1.0.2/src/gui/states/state_stroke.h.orig	2016-07-09 06:27:40.479896333 +0100
+++ synfigstudio-1.0.2/src/gui/states/state_stroke.h	2016-07-09 06:29:14.645896680 +0100
@@ -29,7 +29,6 @@
 
 #include "canvasview.h"
 #include "workarea.h"
-#include <sigc++/object.h>
 #include "duckmatic.h"
 #include <synfig/blinepoint.h>
 #include <list>
--- synfigstudio-1.0.2/src/gui/duckmatic.cpp.orig	2015-07-21 11:01:51.000000000 +0100
+++ synfigstudio-1.0.2/src/gui/duckmatic.cpp	2016-07-09 06:48:37.111824465 +0100
@@ -68,8 +68,6 @@
 #include <synfig/layers/layer_pastecanvas.h>
 
 #include <sigc++/retype_return.h>
-#include <sigc++/retype.h>
-#include <sigc++/hide.h>
 #include <sigc++/bind.h>
 
 #include "ducktransform_matrix.h"
--- synfigstudio-1.0.2/src/gui/workarea.cpp.orig	2015-07-21 11:01:51.000000000 +0100
+++ synfigstudio-1.0.2/src/gui/workarea.cpp	2016-07-09 06:47:32.354219226 +0100
@@ -45,8 +45,6 @@
 #include <gtkmm/scrollbar.h>
 #include <cmath>
 #include <sigc++/retype_return.h>
-#include <sigc++/retype.h>
-#include <sigc++/hide.h>
 #include <ETL/misc>
 
 #include <synfig/target_scanline.h>
--- synfigstudio-1.0.2/src/gui/instance.h.orig	2016-07-09 06:27:40.493896333 +0100
+++ synfigstudio-1.0.2/src/gui/instance.h	2016-07-09 06:28:43.893896566 +0100
@@ -33,7 +33,6 @@
 #include <gtkmm/treeview.h>
 #include <gtkmm/treestore.h>
 #include <synfigapp/instance.h>
-#include <sigc++/object.h>
 #include <synfigapp/value_desc.h>
 #include "trees/historytreestore.h"
 #include <synfig/canvas.h>
--- synfigstudio-1.0.2/src/gui/docks/dock_metadata.cpp.orig	2015-07-09 08:33:04.000000000 +0100
+++ synfigstudio-1.0.2/src/gui/docks/dock_metadata.cpp	2016-07-09 06:39:53.615899034 +0100
@@ -37,7 +37,6 @@
 #include <cassert>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
 #include <sigc++/slot.h>
 #include "trees/metadatatreestore.h"
 #include "trees/metadatatree.h"
--- synfigstudio-1.0.2/src/gui/docks/dock_layergroups.cpp.orig	2015-07-09 08:33:04.000000000 +0100
+++ synfigstudio-1.0.2/src/gui/docks/dock_layergroups.cpp	2016-07-09 06:39:31.356898952 +0100
@@ -36,7 +36,6 @@
 #include <cassert>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
 #include <sigc++/retype_return.h>
 #include <sigc++/slot.h>
 #include "canvasview.h"
--- synfigstudio-1.0.2/src/gui/docks/dock_history.cpp.orig	2015-07-21 11:01:51.000000000 +0100
+++ synfigstudio-1.0.2/src/gui/docks/dock_history.cpp	2016-07-09 06:38:47.064898789 +0100
@@ -39,7 +39,6 @@
 #include <cassert>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
 #include <sigc++/slot.h>
 #include <synfigapp/action.h>
 #include "trees/historytreestore.h"
--- synfigstudio-1.0.2/src/gui/docks/dock_keyframes.cpp.orig	2015-07-09 08:33:04.000000000 +0100
+++ synfigstudio-1.0.2/src/gui/docks/dock_keyframes.cpp	2016-07-09 06:39:13.722898887 +0100
@@ -36,7 +36,6 @@
 #include <cassert>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
 #include <sigc++/slot.h>
 #include "trees/keyframetreestore.h"
 #include "trees/keyframetree.h"
--- synfigstudio-1.0.2/src/gui/docks/dock_toolbox.cpp.orig	2015-07-09 08:33:04.000000000 +0100
+++ synfigstudio-1.0.2/src/gui/docks/dock_toolbox.cpp	2016-07-09 06:41:50.415899464 +0100
@@ -38,10 +38,8 @@
 #include <gtkmm/accelmap.h>
 
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
 #include <sigc++/slot.h>
 #include <sigc++/retype_return.h>
-#include <sigc++/retype.h>
 
 #include "general.h"
 #include <sstream>
--- synfigstudio-1.0.2/src/gui/docks/dock_params.cpp.orig	2015-07-09 08:33:04.000000000 +0100
+++ synfigstudio-1.0.2/src/gui/docks/dock_params.cpp	2016-07-09 06:40:03.514899071 +0100
@@ -36,7 +36,6 @@
 #include <cassert>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
 #include <sigc++/slot.h>
 #include "canvasview.h"
 #include "trees/layerparamtreestore.h"
--- synfigstudio-1.0.2/src/gui/docks/dock_children.cpp.orig	2015-07-09 08:33:04.000000000 +0100
+++ synfigstudio-1.0.2/src/gui/docks/dock_children.cpp	2016-07-09 06:37:52.732898589 +0100
@@ -37,7 +37,6 @@
 #include <cassert>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
 #include <sigc++/retype_return.h>
 #include <sigc++/slot.h>
 #include "trees/childrentreestore.h"
--- synfigstudio-1.0.2/src/gui/docks/dock_timetrack.cpp.orig	2015-08-17 17:28:00.000000000 +0100
+++ synfigstudio-1.0.2/src/gui/docks/dock_timetrack.cpp	2016-07-09 06:40:15.602899115 +0100
@@ -38,7 +38,6 @@
 #include <cassert>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
 #include <sigc++/slot.h>
 #include "canvasview.h"
 #include "trees/layerparamtreestore.h"
--- synfigstudio-1.0.2/src/gui/docks/dock_layers.cpp.orig	2015-07-21 11:01:51.000000000 +0100
+++ synfigstudio-1.0.2/src/gui/docks/dock_layers.cpp	2016-07-09 06:39:41.449898989 +0100
@@ -39,7 +39,6 @@
 #include <cassert>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
 #include <sigc++/retype_return.h>
 #include <sigc++/slot.h>
 #include "trees/layertreestore.h"
--- synfigstudio-1.0.2/src/gui/docks/dock_curves.cpp.orig	2015-07-09 08:33:04.000000000 +0100
+++ synfigstudio-1.0.2/src/gui/docks/dock_curves.cpp	2016-07-09 06:38:23.255898701 +0100
@@ -36,7 +36,6 @@
 #include <cassert>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
 #include <sigc++/slot.h>
 #include "canvasview.h"
 #include "trees/layerparamtreestore.h"
--- synfigstudio-1.0.2/src/gui/docks/dockmanager.h.orig	2016-07-09 06:27:40.490896333 +0100
+++ synfigstudio-1.0.2/src/gui/docks/dockmanager.h	2016-07-09 06:29:00.971896629 +0100
@@ -32,7 +32,6 @@
 #include <map>
 #include <synfig/string.h>
 #include <sigc++/signal.h>
-#include <sigc++/object.h>
 #include <ETL/smart_ptr>
 
 #include <gtkmm/widget.h>
--- synfigstudio-1.0.2/src/gui/docks/dockable.cpp.orig	2015-07-09 08:33:04.000000000 +0100
+++ synfigstudio-1.0.2/src/gui/docks/dockable.cpp	2016-07-09 06:40:40.822899208 +0100
@@ -30,7 +30,6 @@
 #endif
 
 #include "app.h"
-#include <sigc++/hide.h>
 
 #include "docks/dockable.h"
 #include "docks/dockmanager.h"
--- synfigstudio-1.0.2/src/gui/docks/dockdialog.cpp.orig	2015-07-09 08:33:04.000000000 +0100
+++ synfigstudio-1.0.2/src/gui/docks/dockdialog.cpp	2016-07-09 06:42:52.635899694 +0100
@@ -42,10 +42,8 @@
 #include <synfig/general.h>
 #include <synfig/uniqueid.h>
 #include <gtkmm/table.h>
-#include <sigc++/hide.h>
 #include <sigc++/slot.h>
 #include <sigc++/retype_return.h>
-#include <sigc++/retype.h>
 #include "canvasview.h"
 #include <gtkmm/paned.h>
 #include <gtkmm/box.h>
--- synfigstudio-1.0.2/src/gui/compview.cpp.orig	2015-07-09 08:33:04.000000000 +0100
+++ synfigstudio-1.0.2/src/gui/compview.cpp	2016-07-09 06:48:58.773988045 +0100
@@ -39,7 +39,6 @@
 #include <iostream>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
 #include <sigc++/slot.h>
 #include "canvasview.h"
 #include <synfigapp/action.h>
--- synfigstudio-1.0.2/src/gui/canvasview.cpp.orig	2015-07-21 11:01:51.000000000 +0100
+++ synfigstudio-1.0.2/src/gui/canvasview.cpp	2016-07-09 06:45:50.074900348 +0100
@@ -92,7 +92,6 @@
 //#include <synfigapp/action_deletewaypoint.h>
 
 #include <sigc++/retype_return.h>
-#include <sigc++/retype.h>
 //#include <sigc++/hide.h>
 
 #include "canvasview.h"
--- synfigstudio-1.0.2/src/gui/widgets/widget_curves.cpp.orig	2016-07-09 06:27:40.495896333 +0100
+++ synfigstudio-1.0.2/src/gui/widgets/widget_curves.cpp	2016-07-09 06:29:30.534896738 +0100
@@ -38,7 +38,6 @@
 #include <map>
 #include <vector>
 #include <ETL/misc>
-#include <sigc++/object.h>
 
 #include "general.h"
 
--- synfigstudio-1.0.2/src/gui/instance.cpp.orig	2015-07-21 11:01:51.000000000 +0100
+++ synfigstudio-1.0.2/src/gui/instance.cpp	2016-07-09 06:47:18.492153675 +0100
@@ -54,7 +54,6 @@
 
 #include "autorecover.h"
 #include <sigc++/retype_return.h>
-#include <sigc++/retype.h>
 //#include <sigc++/hide.h>
 #include <synfig/valuenodes/valuenode_composite.h>
 #include <synfig/valuenodes/valuenode_duplicate.h>
--- synfigstudio-1.0.2/src/gui/duck.h.orig	2016-07-09 06:27:40.487896333 +0100
+++ synfigstudio-1.0.2/src/gui/duck.h	2016-07-09 06:28:22.367896487 +0100
@@ -38,7 +38,6 @@
 #include <synfig/string.h>
 #include <synfig/real.h>
 #include <sigc++/signal.h>
-#include <sigc++/object.h>
 #include <synfig/time.h>
 #include <ETL/smart_ptr>
 #include <synfigapp/value_desc.h>
--- synfigstudio-1.0.2/src/gui/duckmatic.h.orig	2016-07-09 06:27:40.485896333 +0100
+++ synfigstudio-1.0.2/src/gui/duckmatic.h	2016-07-09 06:28:33.603896528 +0100
@@ -40,7 +40,6 @@
 #include <synfig/string.h>
 #include <synfig/real.h>
 #include <sigc++/signal.h>
-#include <sigc++/object.h>
 #include <synfig/time.h>
 #include <synfig/color.h>
 #include <ETL/smart_ptr>
--- synfigstudio-1.0.2/src/gui/workarearenderer/workarearenderer.h.orig	2016-07-09 06:27:40.498896333 +0100
+++ synfigstudio-1.0.2/src/gui/workarearenderer/workarearenderer.h	2016-07-09 06:29:41.272896778 +0100
@@ -29,7 +29,6 @@
 
 #include <ETL/handle>
 #include <sigc++/signal.h>
-#include <sigc++/object.h>
 #include <synfig/vector.h>
 #include <gdkmm/window.h>
 #include <gdkmm/rectangle.h>
--- synfigstudio-1.0.2/src/synfigapp/canvasinterface.h.orig	2016-07-09 06:27:40.506896333 +0100
+++ synfigstudio-1.0.2/src/synfigapp/canvasinterface.h	2016-07-09 06:30:04.486896863 +0100
@@ -31,7 +31,6 @@
 //#include <synfig/canvas.h>
 #include <synfig/value.h>
 #include <sigc++/signal.h>
-#include <sigc++/object.h>
 #include <list>
 #include "selectionmanager.h"
 #include "uimanager.h"
--- synfigstudio-1.0.2/src/synfigapp/instance.h.orig	2016-07-09 06:27:40.503896333 +0100
+++ synfigstudio-1.0.2/src/synfigapp/instance.h	2016-07-09 06:30:11.842896890 +0100
@@ -36,7 +36,6 @@
 #include <list>
 #include <set>
 #include <sigc++/signal.h>
-#include <sigc++/object.h>
 #include "action_system.h"
 #include "selectionmanager.h"
 #include "cvs.h"
--- synfigstudio-1.0.2/src/synfigapp/action_system.h.orig	2016-07-09 06:27:40.509896333 +0100
+++ synfigstudio-1.0.2/src/synfigapp/action_system.h	2016-07-09 06:29:58.550896841 +0100
@@ -29,7 +29,6 @@
 
 #include "action.h"
 #include <sigc++/signal.h>
-#include <sigc++/object.h>
 #include <ETL/handle>
 #include <synfig/canvas.h>
 #include "uimanager.h"
--- synfigstudio-1.0.2/src/synfigapp/main.h.orig	2016-07-09 06:27:40.501896333 +0100
+++ synfigstudio-1.0.2/src/synfigapp/main.h	2016-07-09 06:30:18.665896915 +0100
@@ -35,7 +35,6 @@
 #include <synfig/real.h>
 #include <synfig/waypoint.h>
 #include <sigc++/signal.h>
-#include <sigc++/object.h>
 #include "inputdevice.h"
 #include "settings.h"
 
--- synfigstudio-1.0.2/src/synfigapp/uimanager.h.orig	2016-07-09 06:27:40.512896333 +0100
+++ synfigstudio-1.0.2/src/synfigapp/uimanager.h	2016-07-09 06:32:56.844897498 +0100
@@ -30,7 +30,7 @@
 #include <ETL/handle>
 #include <synfig/general.h>
 #include <synfig/string.h>
-#include <sigc++/object.h>
+#include <sigc++/signal.h>
 
 /* === M A C R O S ========================================================= */
 
