--- tagtool-0.12.3/src/Makefile.in.orig	2007-02-27 20:34:13.000000000 +0000
+++ tagtool-0.12.3/src/Makefile.in	2007-03-16 23:08:07.000000000 +0000
@@ -189,7 +189,7 @@
 @ENABLE_VORBIS_FALSE@vorbis_cflags = 
 @ENABLE_VORBIS_TRUE@vorbis_cflags = -DENABLE_VORBIS
 
-tagtool_CFLAGS = -Wall -Wno-pointer-sign $(GTK_CFLAGS) ${mp3_cflags} 	\
+tagtool_CFLAGS = -Wall $(GTK_CFLAGS) ${mp3_cflags} 	\
 	${vorbis_cflags} -DDATADIR='"$(datadir)/${PACKAGE_NAME}"'
 
 
