--- tapioca-qt-0.14.1/include/QtTapioca/CMakeLists.txt.orig	2007-05-23 19:20:42.000000000 +0100
+++ tapioca-qt-0.14.1/include/QtTapioca/CMakeLists.txt	2009-03-26 21:31:05.000000000 +0000
@@ -31,7 +31,15 @@
     TextChannel
     textchannel.h
     StreamChannel
+    streamchannel.h
     Stream
+    stream.h
+    VideoStream
+    videostream.h
+    AudioStream
+    audiostream.h
+    ContactGroup
+    contactgroup.h
 )
 
 install(FILES ${library_HDRS} DESTINATION include/QtTapioca)
