--- tcpreplay-4.3.4/configure.orig	2021-05-02 01:25:04.000000000 +0100
+++ tcpreplay-4.3.4/configure	2021-07-21 18:16:09.477945627 +0100
@@ -21458,11 +21458,11 @@
 #ifdef __cplusplus
 extern "C"
 #endif
-char nl_handle_alloc ();
+char nl_handle_allocX ();
 int
 main ()
 {
-return nl_handle_alloc ();
+return nl_handle_allocX ();
   ;
   return 0;
 }
@@ -21523,11 +21523,11 @@
 #ifdef __cplusplus
 extern "C"
 #endif
-char nl_cache_alloc ();
+char nl_cache_allocX ();
 int
 main ()
 {
-return nl_cache_alloc ();
+return nl_cache_allocX ();
   ;
   return 0;
 }
