Index: tcptrace-6.6.1/Makefile.in
===================================================================
--- tcptrace-6.6.1.orig/Makefile.in
+++ tcptrace-6.6.1/Makefile.in
@@ -215,26 +215,11 @@ YACC_VAL=@YACC@
 LEX_VAL=@LEX@
 
 filt_parser.c: filt_parser.y filter.h
-	if test "${YACC_VAL}" = "bison -y" ; then \
-		@YACC@ -vd -p filtyy -o filt_parser.c $(srcdir)/filt_parser.y ;\
-		cp filt_parser.c flex_bison ;\
-		cp filt_parser.h flex_bison ;\
-	else \
-		echo "Could not find BISON on this system";\
-		echo "Copying the BISON output files generated at our place" ;\
-		cp flex_bison/filt_parser.c . ;\
-		cp flex_bison/filt_parser.h . ;\
-	fi
+	cp $(top_srcdir)/flex_bison/filt_parser.c .
+	cp $(top_srcdir)/flex_bison/filt_parser.h .
 
 filt_scanner.c: filt_scanner.l filter.h filt_parser.h
-	if test ${LEX_VAL} = "flex" ; then \
-		@LEX@ -t -Pfiltyy $(srcdir)/filt_scanner.l > filt_scanner.c ;\
-		cp filt_scanner.c flex_bison ;\
-	else \
-		echo "Could not find FLEX on this system" ;\
-		echo "Copying the FLEX output files generated at our place" ;\
-		cp flex_bison/filt_scanner.c . ;\
-	fi
+	cp $(top_srcdir)/flex_bison/filt_scanner.c .
 
 # filt_parser.h created as a side effect of running yacc...
 filt_parser.h: filt_parser.c
