--- tesseract-1.02/textord/makerow.cpp.orig	2006-06-16 23:17:09.000000000 +0100
+++ tesseract-1.02/textord/makerow.cpp	2007-01-03 13:44:21.000000000 +0000
@@ -660,13 +660,13 @@
           "Bad y coord of bottom, " INT32FORMAT "(" INT32FORMAT ","
           INT32FORMAT ")\n", (INT32) floor (bottom), min_y, max_y);
                                  //count transitions
-      deltas[(INT32) floor (bottom) - min_y] += width;
+      deltas[(INT32) floor (bottom) - min_y] = deltas[(INT32) floor (bottom) - min_y] + width;
       if ((INT32) floor (top) < min_y
         || (INT32) floor (top) - min_y >= line_count)
         fprintf (stderr,
           "Bad y coord of top, " INT32FORMAT "(" INT32FORMAT ","
           INT32FORMAT ")\n", (INT32) floor (top), min_y, max_y);
-      deltas[(INT32) floor (top) - min_y] -= width;
+      deltas[(INT32) floor (top) - min_y] = deltas[(INT32) floor (top) - min_y] - width;
     }
   }
   occupation[0] = deltas[0];
