--- tesseract-5.5.0/src/training/CMakeLists.txt.orig	2024-11-10 19:23:34.000000000 +0000
+++ tesseract-5.5.0/src/training/CMakeLists.txt	2024-11-10 22:39:36.928007781 +0000
@@ -442,6 +442,11 @@
     generate_export_header(pango_training EXPORT_MACRO_NAME
                            TESS_PANGO_TRAINING_API)
     project_group(pango_training "Training Tools")
+    install(
+    TARGETS pango_training
+    RUNTIME DESTINATION bin
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
     # ##########################################################################
     # EXECUTABLE text2image
