--- tesseract-5.1.0/CMakeLists.txt.orig	2022-03-01 16:25:45.000000000 +0000
+++ tesseract-5.1.0/CMakeLists.txt	2022-04-07 21:03:09.923604115 +0100
@@ -424,7 +424,7 @@
 endif()
 
 if(NOT DEFINED CMAKE_INSTALL_LIBDIR)
-  set(CMAKE_INSTALL_LIBDIR lib)
+  set(CMAKE_INSTALL_LIBDIR lib${LIB_SUFFIX})
 endif(NOT DEFINED CMAKE_INSTALL_LIBDIR)
 
 set(AUTOCONFIG_SRC ${CMAKE_CURRENT_BINARY_DIR}/config_auto.h.in)
--- tesseract-5.1.0/src/training/CMakeLists.txt.orig	2022-03-01 16:25:45.000000000 +0000
+++ tesseract-5.1.0/src/training/CMakeLists.txt	2022-04-07 21:46:10.998480964 +0100
@@ -116,6 +116,14 @@
                        TESS_COMMON_TRAINING_API)
 project_group(common_training "Training Tools")
 
+if(NOT DISABLED_LEGACY_ENGINE)
+  install(
+    TARGETS common_training
+    RUNTIME DESTINATION bin
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX})
+endif()
+
 # ##############################################################################
 # EXECUTABLE ambiguous_words
 # ##############################################################################
@@ -127,8 +135,8 @@
   install(
     TARGETS ambiguous_words
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX})
 endif()
 
 # ##############################################################################
@@ -142,8 +150,8 @@
   install(
     TARGETS classifier_tester
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX})
 endif()
 
 # ##############################################################################
@@ -156,8 +164,8 @@
 install(
   TARGETS combine_tessdata
   RUNTIME DESTINATION bin
-  LIBRARY DESTINATION lib
-  ARCHIVE DESTINATION lib)
+  LIBRARY DESTINATION lib${LIB_SUFFIX}
+  ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
 # ##############################################################################
 # EXECUTABLE cntraining
@@ -170,8 +178,8 @@
   install(
     TARGETS cntraining
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX})
 endif()
 
 # ##############################################################################
@@ -184,8 +192,8 @@
 install(
   TARGETS dawg2wordlist
   RUNTIME DESTINATION bin
-  LIBRARY DESTINATION lib
-  ARCHIVE DESTINATION lib)
+  LIBRARY DESTINATION lib${LIB_SUFFIX}
+  ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
 # ##############################################################################
 # EXECUTABLE mftraining
@@ -198,8 +206,8 @@
   install(
     TARGETS mftraining
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX})
 endif()
 
 # ##############################################################################
@@ -213,8 +221,8 @@
   install(
     TARGETS shapeclustering
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX})
 endif()
 
 # ##############################################################################
@@ -227,8 +235,8 @@
 install(
   TARGETS wordlist2dawg
   RUNTIME DESTINATION bin
-  LIBRARY DESTINATION lib
-  ARCHIVE DESTINATION lib)
+  LIBRARY DESTINATION lib${LIB_SUFFIX}
+  ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
 if(ICU_FOUND)
 
@@ -261,6 +269,14 @@
                          TESS_UNICHARSET_TRAINING_API)
   project_group(unicharset_training "Training Tools")
 
+  if(NOT DISABLED_LEGACY_ENGINE)
+    install(
+      TARGETS unicharset_training
+      RUNTIME DESTINATION bin
+      LIBRARY DESTINATION lib${LIB_SUFFIX}
+      ARCHIVE DESTINATION lib${LIB_SUFFIX})
+  endif()
+
   # ############################################################################
   # EXECUTABLE combine_lang_model
   # ############################################################################
@@ -271,8 +287,8 @@
   install(
     TARGETS combine_lang_model
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
   # ############################################################################
   # EXECUTABLE lstmeval
@@ -284,8 +300,8 @@
   install(
     TARGETS lstmeval
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
   # ############################################################################
   # EXECUTABLE lstmtraining
@@ -297,8 +313,8 @@
   install(
     TARGETS lstmtraining
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
   # ############################################################################
   # EXECUTABLE merge_unicharsets
@@ -310,8 +326,8 @@
   install(
     TARGETS merge_unicharsets
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
   # ############################################################################
   # EXECUTABLE set_unicharset_properties
@@ -323,8 +339,8 @@
   install(
     TARGETS set_unicharset_properties
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
   # ############################################################################
   # EXECUTABLE unicharset_extractor
@@ -336,8 +352,8 @@
   install(
     TARGETS unicharset_extractor
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
   # ############################################################################
 
@@ -384,6 +400,14 @@
                            TESS_PANGO_TRAINING_API)
     project_group(pango_training "Training Tools")
 
+    if(NOT DISABLED_LEGACY_ENGINE)
+      install(
+        TARGETS pango_training
+        RUNTIME DESTINATION bin
+        LIBRARY DESTINATION lib${LIB_SUFFIX}
+        ARCHIVE DESTINATION lib${LIB_SUFFIX})
+    endif()
+
     # ##########################################################################
     # EXECUTABLE text2image
     # ##########################################################################
@@ -396,8 +420,8 @@
     install(
       TARGETS text2image
       RUNTIME DESTINATION bin
-      LIBRARY DESTINATION lib
-      ARCHIVE DESTINATION lib)
+      LIBRARY DESTINATION lib${LIB_SUFFIX}
+      ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
   endif()
 endif(ICU_FOUND)
