--- tesseract-5.5.0/src/training/CMakeLists.txt.orig	2024-11-10 19:23:34.000000000 +0000
+++ tesseract-5.5.0/src/training/CMakeLists.txt	2025-03-30 09:27:22.486168014 +0100
@@ -441,6 +441,11 @@
                                PUBLIC pango ${CMAKE_CURRENT_BINARY_DIR})
     generate_export_header(pango_training EXPORT_MACRO_NAME
                            TESS_PANGO_TRAINING_API)
+    install(
+      TARGETS pango_training
+      RUNTIME DESTINATION bin
+      LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+      ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
     project_group(pango_training "Training Tools")
 
     # ##########################################################################
