--- tesseract-1.03/textord/makerow.cpp.orig	2006-12-20 18:03:32.000000000 +0000
+++ tesseract-1.03/textord/makerow.cpp	2007-04-15 22:06:39.000000000 +0100
@@ -816,13 +816,13 @@
           "Bad y coord of bottom, " INT32FORMAT "(" INT32FORMAT ","
           INT32FORMAT ")\n", (INT32) floor (bottom), min_y, max_y);
                                  //count transitions
-      deltas[(INT32) floor (bottom) - min_y] += width;
+      deltas[(INT32) floor (bottom) - min_y] = deltas[(INT32) floor (bottom) - min_y] + width;
       if ((INT32) floor (top) < min_y
         || (INT32) floor (top) - min_y >= line_count)
         fprintf (stderr,
           "Bad y coord of top, " INT32FORMAT "(" INT32FORMAT ","
           INT32FORMAT ")\n", (INT32) floor (top), min_y, max_y);
-      deltas[(INT32) floor (top) - min_y] -= width;
+      deltas[(INT32) floor (top) - min_y] = deltas[(INT32) floor (top) - min_y] - width;
     }
   }
   occupation[0] = deltas[0];
