--- tesseract-4.0.0/CMakeLists.txt.orig	2018-10-29 08:53:12.000000000 +0000
+++ tesseract-4.0.0/CMakeLists.txt	2018-10-29 20:30:31.131683983 +0000
@@ -316,9 +316,9 @@
 get_target_property(tesseract_VERSION libtesseract VERSION)
 get_target_property(tesseract_OUTPUT_NAME libtesseract OUTPUT_NAME)
 configure_file(tesseract.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/tesseract.pc @ONLY)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/tesseract.pc DESTINATION lib/pkgconfig)
-install(TARGETS tesseract RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
-install(TARGETS libtesseract EXPORT TesseractTargets RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/tesseract.pc DESTINATION lib${LIB_SUFFIX}/pkgconfig)
+install(TARGETS tesseract RUNTIME DESTINATION bin LIBRARY DESTINATION lib${LIB_SUFFIX} ARCHIVE DESTINATION lib${LIB_SUFFIX})
+install(TARGETS libtesseract EXPORT TesseractTargets RUNTIME DESTINATION bin LIBRARY DESTINATION lib${LIB_SUFFIX} ARCHIVE DESTINATION lib${LIB_SUFFIX})
 install(EXPORT TesseractTargets DESTINATION cmake)
 install(FILES
     ${CMAKE_BINARY_DIR}/TesseractConfig.cmake
