--- tidy-5.0.0/CMakeLists.txt.orig	2015-09-04 13:43:28.000000000 +0100
+++ tidy-5.0.0/CMakeLists.txt	2016-10-01 18:34:30.959127439 +0100
@@ -147,8 +147,8 @@
 list ( APPEND add_LIBS ${name} )
 install(TARGETS ${name}
         RUNTIME DESTINATION bin
-        ARCHIVE DESTINATION lib
-        LIBRARY DESTINATION lib
+        ARCHIVE DESTINATION lib${LIB_SUFFIX}
+        LIBRARY DESTINATION lib${LIB_SUFFIX}
         )
 install( FILES ${HFILES} DESTINATION include )
 ########################################
@@ -167,8 +167,8 @@
                                    COMPILE_FLAGS "-DBUILDING_SHARED_LIB" )
     install(TARGETS ${name}
         RUNTIME DESTINATION bin
-        ARCHIVE DESTINATION lib
-        LIBRARY DESTINATION lib
+        ARCHIVE DESTINATION lib${LIB_SUFFIX}
+        LIBRARY DESTINATION lib${LIB_SUFFIX}
         )
 endif ()                               
 
