--- tintii-2.5.2/config.py.orig	2011-02-23 11:33:01.000000000 +0000
+++ tintii-2.5.2/config.py	2011-05-17 22:28:04.000000000 +0100
@@ -12,6 +12,6 @@
 
 # Linux/g++ compiler options
 CXX='g++'
-CXXFLAGS='-O3 -Wall -funroll-loops -fomit-frame-pointer `wx-config --cxxflags` -fopenmp -ffast-math -g'
-LINKFLAGS='`wx-config --libs` `wx-config --libs aui` -fopenmp -g'
+CXXFLAGS='-O3 -Wall -funroll-loops -fomit-frame-pointer `wx-config-2.9 --cxxflags` -fopenmp -ffast-math -g'
+LINKFLAGS='`wx-config-2.9 --libs` `wx-config-2.9 --libs aui` -fopenmp -g'
 #NDEBUG='yes'
