--- tintii-2.10.0/src/indii/gui/PreviewPane.cpp.orig	2015-10-18 16:59:13.000000000 +0100
+++ tintii-2.10.0/src/indii/gui/PreviewPane.cpp	2020-11-01 22:24:09.829214639 +0000
@@ -26,7 +26,7 @@
   preview = new PreviewImage(this, res, model, control);
 
   wxBoxSizer* sizer = new wxBoxSizer(wxVERTICAL);
-  sizer->Add(preview, 1, wxALIGN_CENTER|wxEXPAND);
+  sizer->Add(preview, 1, wxEXPAND);
   SetSizer(sizer);
 }
 
--- tintii-2.10.0/src/indii/tint/gui/ThumbPane.cpp.orig	2015-10-18 16:59:13.000000000 +0100
+++ tintii-2.10.0/src/indii/tint/gui/ThumbPane.cpp	2020-11-01 22:19:57.014226701 +0000
@@ -104,6 +104,9 @@
     thumbs[i]->Show(model != NULL && i < model->getNumClusters());
   }
 
+  butAdd->GetContainingSizer()->Detach(butAdd);
+  butSub->GetContainingSizer()->Detach(butSub);
+
   wxGridBagSizer* buttonSizer = new wxGridBagSizer(0,0);
   buttonSizer->Add(butAdd, wxGBPosition(0,0), wxDefaultSpan, wxTOP|wxLEFT|wxBOTTOM, PADDING);
   buttonSizer->Add(butSub, wxGBPosition(0,1), wxDefaultSpan, wxTOP|wxLEFT|wxBOTTOM, PADDING);
