--- tkgate-1.8.7/src/tkgate/concat.c.orig	2003-12-02 17:33:47.000000000 +0000
+++ tkgate-1.8.7/src/tkgate/concat.c	2007-01-27 22:51:52.000000000 +0000
@@ -1,5 +1,5 @@
 /****************************************************************************
-    Copyright (C) 1987-2001 by Jeffery P. Hansen
+    Copyright (C) 1987-2007 by Jeffery P. Hansen
 
     This program is free software; you can redistribute it and/or modify
     it under the terms of the GNU General Public License as published by
@@ -15,7 +15,7 @@
     along with this program; if not, write to the Free Software
     Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
-    Last edit by hansen on Tue Dec  2 12:33:47 2003
+    Last edit by hansen on Thu Jan 25 20:54:20 2007
 ****************************************************************************/
 #include "tkgate.h"
 
@@ -109,6 +109,7 @@
   Concat_HitDistance,
   Concat_Draw,
   Generic_Move,
+  Generic_Rotate,
   Concat_Replicate,
   Concat_AddInput,
   Concat_AddInput,
@@ -122,7 +123,7 @@
   Concat_SetProp
 };
 
-static void Concat_adjustWires(GCElement *g)
+void Concat_adjustWires(GCElement *g)
 {
   int N,L,i,wx,wy,dx,dy;
   GWire *w;
@@ -224,8 +225,8 @@
   case 3 :
     *minx = gx-L/2-5;
     *maxx = gx+L/2-5;
-    *minx = gy - 5;
-    *maxx = gy + 5;
+    *miny = gy - 5;
+    *maxy = gy + 5;
     break;
   }
 }
@@ -352,10 +353,10 @@
 
   switch (e1->orient) {
   case D_RIGHT :
-    e2->nodes->x += d*10;
+    e2->nodes->x -= d*10;
     break;
   case D_UP :
-    e2->nodes->y -= d*10;
+    e2->nodes->y += d*10;
     break;
   case D_LEFT :
     e2->nodes->x -= d*10;
@@ -394,35 +395,58 @@
   int N = wire_numOnPad(g->wires[CONCAT_IN]);
   int len = (N > 0) ? (CONCAT_WIRESPACE*(N+1)) : 2*CONCAT_WIRESPACE;
   char buf[STRMAX];
+  int *wire_nbits;
+  int nwires = 0;
   int nbits = 0;
+  int ncount;
   GWire *w;
 
+  /*
+  ** We need to collect bit-widths of wires into an array, since we don't
+  ** know it which order we will be going through them.
+  */
   for (w = g->wires[CONCAT_IN];w;w = w->next)
+  {
     nbits += w->net->nbits;
+    nwires++;
+  }
+
+  wire_nbits = malloc(nwires * sizeof(int));
 
+  nwires = 0;
+  for (w = g->wires[CONCAT_IN];w;w = w->next) {
+    wire_nbits[nwires++] = w->net->nbits;
+  }
+
+  /*
+  ** Now use this information to print the wires.
+  */
   fprintf(f,"[\n");
   if (g->u.cat.direction) {
     int lsb = 0;
-    for (w = g->wires[CONCAT_IN];w;w = w->next) {
-      if (w->net->nbits == 1)
+    for (ncount = 0; ncount < nwires; ncount++) {
+      if (wire_nbits[ncount] == 1)
 	sprintf(buf,"%d",lsb);
       else
-	sprintf(buf,"%d:%d",lsb+w->net->nbits-1,lsb);
+	sprintf(buf,"%d:%d",lsb+wire_nbits[ncount]-1,lsb);
       fprintf(f,"  (%s)\n",buf);
-      lsb += w->net->nbits;
+      lsb += wire_nbits[ncount];
     }
   } else {
     int msb = nbits-1;
-    for (w = g->wires[CONCAT_IN];w;w = w->next) {
-      if (w->net->nbits == 1)
+    int ncount = nwires;
+    for (ncount = nwires - 1;ncount >= 0; ncount--) {
+      if (wire_nbits[ncount] == 1)
 	sprintf(buf,"%d",msb);
       else
-	sprintf(buf,"%d:%d",msb,msb-w->net->nbits+1);
+	sprintf(buf,"%d:%d",msb,msb-wire_nbits[ncount]+1);
       fprintf(f,"  (%s)\n",buf);
-      msb -= w->net->nbits;
+      msb -= wire_nbits[ncount];
     }
   }
+
   fprintf(f,"] %d %d %d %d psconcat\n",len,g->xpos,g->ypos,-g->orient*90);
+  free(wire_nbits);
 }
 
 void init_concat()
--- tkgate-1.8.7/src/tkgate/mux.c.orig	2007-01-27 20:36:16.000000000 +0000
+++ tkgate-1.8.7/src/tkgate/mux.c	2007-01-29 03:16:12.000000000 +0000
@@ -1,5 +1,5 @@
 /****************************************************************************
-    Copyright (C) 1987-2005 by Jeffery P. Hansen
+    Copyright (C) 1987-2007 by Jeffery P. Hansen
 
     This program is free software; you can redistribute it and/or modify
     it under the terms of the GNU General Public License as published by
@@ -15,7 +15,7 @@
     along with this program; if not, write to the Free Software
     Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
-    Last edit by hansen on Thu Jan 25 21:06:04 2007
+    Last edit by hansen on Sun Jan 28 22:16:12 2007
 ****************************************************************************/
 #include "tkgate.h"
 
@@ -23,14 +23,24 @@
 #define MUX_SEL 1
 #define MUX_OUT 2
 
-#define F_IN 0x1
-
+/*
+ * Some of these are shared by the Demux element.  These have a MuxDemux_ prefix. 
+ */
 void Mux_Delete(GCElement *g,GModuleDef *env,int drawp);
 void Mux_Draw(GCElement *g,int md);
+void Mux_Rotate(GCElement *g, int centX, int centY,int rdir);
 GCElement *Mux_Make(EditState **es,GModuleDef *env,int GType,
 		    int x,int y,int r,const char *Name,int noWires,const char**,int);
 void Mux_PSWrite(FILE *f,GModLayout*,GCElement *g);
-void Mux_Rotate(GCElement *g, int centX, int centY,int rdir);
+void MuxDemux_VerSave(FILE*,GCElement*);
+void Mux_SetProp(GCElement*,const char*,const void*);
+void Mux_AddInput(EditState *es,GCElement *g);
+void MuxDemux_EditProps(GCElement *g,int isLoadDialog);
+GWireNode *Mux_wireSnap(GCElement *g,GWire *w,int *mod,int retry);
+void Mux_VersionDelta(GCElement *g,Version *s);
+//GCElement *MuxDemux_Copy(GModuleDef *M,GCElement *g,int x,int y,unsigned flags);
+
+void Demux_adjustWires(GCElement *g);
 
 static iconDimensions mux_iconDims[] = {
   {0,  0,  59, 29, 30, 15},
@@ -40,23 +50,31 @@
 };
 static int mux_iconBoldOffset = 89;
 
-struct locate mux_in_loc[] = {
+GPadLoc mux_in_loc[] = {
 	{-30,-16,30,-16,D_UP},
 	{-16,-30,-16,30,D_LEFT},
 	{-30,16,30,16,D_DOWN},
 	{16,-30,16,30,D_RIGHT}};
 
-struct locate mux_out_loc[] = {
+GPadLoc mux_out_loc[] = {
 	{0,13,0,13,D_DOWN},
 	{13,0,13,0,D_RIGHT},
 	{0,-13,0,-13,D_UP},
 	{-13,0,-13,0,D_LEFT}};
 
-struct locate mux_sel_loc[] = {
+GPadLoc mux_sel_loc[] = {
 	{-23,0,-23,0,D_LEFT},
+	{0,23,0,23,D_DOWN},
+	{23,0,23,0,D_RIGHT},
 	{0,-23,0,-23,D_UP},
+};
+
+GPadLoc mux_sel_altloc[] = {
 	{23,0,23,0,D_RIGHT},
-	{0,23,0,23,D_DOWN}};
+	{0,-23,0,-23,D_UP},
+	{-23,0,-23,0,D_LEFT},
+	{0,23,0,23,D_DOWN},
+};
 
 static char *psMux[] = {
   "%",
@@ -131,10 +149,14 @@
   Nop_SimStateFunc,
   Nop_SimHitFunc,
   Mux_PSWrite,
-  Generic_EditProps,
-  Generic_VerSave
+  MuxDemux_EditProps,
+  MuxDemux_VerSave,
+  Mux_SetProp,
+  Mux_VersionDelta,
+  Mux_wireSnap,
 };
 
+
 /*
  * Adjust the wires on a mux to conform to the current setting of data_order
  */
@@ -156,39 +178,31 @@
   switch (g->orient) {
   case 0 :
     dx = (double)L/(double)(N+1);
-#if 0
     if (g->u.mux.data_order) {
       wx = g->xpos + pd->x2;
       dx = -dx;
     }
-#endif
     break;
   case 1 :
     dy = (double)L/(double)(N+1);
-#if 0
     if (!g->u.mux.data_order) {
       wy = g->ypos + pd->y2;
       dy = -dy;
     }
-#endif
     break;
   case 2 :
     dx = (double)L/(double)(N+1);
-#if 0
     if (!g->u.mux.data_order) {
       wx = g->xpos + pd->x2;
       dx = -dx;
     }
-#endif
     break;
   case 3 :
     dy = (double)L/(double)(N+1);
-#if 0
     if (g->u.mux.data_order) {
       wy = g->ypos + pd->y2;
       dy = -dy;
     }
-#endif
     break;
   }
   
@@ -207,41 +221,11 @@
   int wx,wy;
   GPadLoc *pd = &g->typeinfo->Pad[MUX_SEL].Loc[g->orient];
   GWire *w = g->wires[MUX_SEL];
-  GWire *ow = wire_other(w);
 
-#if 0
   if (g->u.mux.select_side)
     pd = &mux_sel_altloc[g->orient];
-#endif
 
-  wx = g->xpos + pd->x2;
-  wy = g->ypos + pd->y2;
-
-  ob_touch(w);
-  wire_move(w->nodes,wx-w->nodes->x,wy-w->nodes->y,VERTICAL|HORIZONTAL);
-
-  /*
-   * If wire end is not attached, compute mirror image.  Otherwise use brute force
-   * to fix the wire.
-   */
-  if (w->orient != pd->dir) {
-    w->orient = pd->dir;
-    if (!ow->gate) {
-      int cx = w->nodes->x;
-      int cy = w->nodes->y;
-      GWireNode *n;
-
-
-      for (n = w->driver->nodes;n;n = n->out) {
-	ob_touch(n);
-	if (g->orient == 0 || g->orient == 2)
-	  n->x = cx - (n->x - cx);
-	else
-	  n->y = cy - (n->y - cy);
-      }
-    } else
-      wire_snap(w->driver->nodes);
-  }
+  wire_rePort(w,g->xpos+pd->x2,g->ypos+pd->y2,pd->dir);
 }
 
 /*
@@ -256,14 +240,46 @@
 GCElement *Mux_Make(EditState **es,GModuleDef *env,int GType,
 		    int x,int y,int r,const char *Name,int noWires,const char **options,int nOptions)
 {
-  int i,nbits;
+  int nbits;
   struct celemnt *g;
   struct wire *w;
 
-  if (!(g = Generic_Make(es,env,GType,x,y,r,Name,noWires,options,nOptions)))
+  if (!(g = Generic_Make(es,env,GType,x,y,r,Name,1,options,nOptions)))
     return NULL;
 
+  g->u.mux.select_side = 0;
+  g->u.mux.data_order = 0;
+
   if (!noWires) {
+    int i,j;
+    const char *Invert,*Pins;
+    int N = GCElement_numPads(g);
+
+    Invert = seekOption("-invert",options,nOptions);
+    Pins = seekOption("-pins",options,nOptions);
+
+    for (i = 0;i < N;i++) {
+      int Num = 0;
+
+      if (Pins) {
+	char buf[STRMAX];
+	int N;
+	if (sscanf(Pins,"%[^=]=%d",buf,&N) == 2 && strcmp(buf,g->typeinfo->Pad[i].Name) == 0)
+	  Num = N;
+      }
+      if (!Num)
+	Num = GCElement_getPadNum(g,i);
+
+      for (j = 0;j < Num;j++) {
+	if (Invert && strcmp(Invert,g->typeinfo->Pad[i].Name) == 0)
+	  wire_addToGate(g,i,env,1);
+	else
+	  wire_addToGate(g,i,env,0);
+      }
+    }
+
+    Mux_adjustWires(g);
+
     for (i = 0,w = g->wires[MUX_IN];w;w = w->next,i++);
 
     nbits = required_bits(i);
@@ -273,6 +289,41 @@
   return g;
 }
 
+#if 0
+GCElement *MuxDemux_Copy(GModuleDef *M,GCElement *g,int x,int y,unsigned flags)
+{
+  GCElement *ng = Generic_Copy(M,g,x,y,flags);
+
+  ob_touch(ng);
+  ng->u.mux.select_side = g->u.mux.select_side;
+  ng->u.mux.data_order = g->u.mux.data_order;
+
+  return ng;
+}
+#endif
+
+
+void Mux_AddInput(EditState *es,GCElement *g)
+{
+  int i;
+  int N = GCElement_numPads(g);
+
+  for (i = 0;i < N;i++)
+    if (GCElement_getPadDir(g,i) == IN && g->typeinfo->Pad[i].CanAdd)
+      break;
+  
+  if (i == N)
+    return;
+  
+  if (es) SetModified();
+  
+  gate_draw(g,GD_NORMAL);
+  wire_addToGate(g,i,es->env,0);
+  Mux_adjustWires(g);
+  gate_draw(g,GD_NORMAL);
+}
+
+
 void Mux_Draw_Special(GCElement *g)
 {
   GWire *w,*w0;
@@ -361,6 +412,112 @@
 	NIn,g->xpos,g->ypos,-g->orient*90,g->typeinfo->psprint);
 }
 
+void MuxDemux_VerSave(FILE *f,GCElement *g)
+{
+  VerilogBasicGateCall(f,g);
+  VerilogBasicGateParmList(f,g);
+  VerilogBasicGateComment(f,g,1);
+
+  fprintf(f," /ss:%d /do:%d",g->u.mux.select_side,g->u.mux.data_order);
+
+  fprintf(f,"\n");
+}
+
+void MuxDemux_EditProps(GCElement *g,int isLoadDialog)
+{
+  Tcl_Interp *tcl = XGate.tcl;
+
+  Generic_EditProps(g,isLoadDialog);
+  if (isLoadDialog) {
+    DoTcl("set ::edgat_dataOrder %d",g->u.mux.data_order);
+    DoTcl("set ::edgat_selectSide %d",g->u.mux.select_side);
+  } else {
+    const char *p;
+    int new_data_order = g->u.mux.data_order;
+    int new_select_side = g->u.mux.select_side;
+
+    if ((p = Tcl_GetVar(tcl,"edgat_dataOrder",TCL_GLOBAL_ONLY))) {
+      sscanf(p,"%d",&new_data_order);
+    }
+    if ((p = Tcl_GetVar(tcl,"edgat_selectSide",TCL_GLOBAL_ONLY))) {
+      sscanf(p,"%d",&new_select_side);
+    }
+
+    if (g->u.mux.data_order != new_data_order || g->u.mux.select_side != new_select_side) {
+      ob_touch(g);
+      gate_draw(g,0);
+      g->u.mux.data_order = new_data_order;
+      g->u.mux.select_side = new_select_side;
+
+      if (strcmp(g->typeinfo->englishName,"MUX") == 0)
+	Mux_adjustWires(g);
+      else
+	Demux_adjustWires(g);
+      gate_draw(g,0);
+    }
+  }
+}
+
+void Mux_SetProp(GCElement *g,const char *prop,const void *value)
+{
+  if (strcmp(prop,"/ss") == 0) g->u.mux.select_side = *((int*)value);
+  if (strcmp(prop,"/do") == 0) g->u.mux.data_order = *((int*)value);
+
+  if (g->wires[MUX_SEL])
+    g->wires[MUX_SEL]->orient = (g->orient + 2 + (g->u.mux.select_side ? 2 : 0) ) % 4;
+}
+
+/*
+ * This function is called when reading a save file from a previous version of tkgate.
+ * Default pin placement of muxes changed in version 1.8.7.  This function sets flags on
+ * muxes to make it compatable with the new vesion, but only if the save file is older
+ * than 1.8.7.
+ */
+void Mux_VersionDelta(GCElement *g,Version *V)
+{
+  Version x1 = {1, 8, 7};
+
+  if (VersionCmp(V,&x1) >= 0)
+    return;
+  
+  switch (g->orient) {
+  case D_RIGHT :
+    g->u.mux.select_side = 0;
+    g->u.mux.data_order = 0;
+    g->wires[MUX_SEL]->orient = D_LEFT;
+    break;
+  case D_UP :
+    g->u.mux.select_side = 1;
+    g->u.mux.data_order = 1;
+    g->wires[MUX_SEL]->orient = D_DOWN;
+    break;
+  case D_LEFT :
+    g->u.mux.select_side = 0;
+    g->u.mux.data_order = 1;
+    g->wires[MUX_SEL]->orient = D_RIGHT;
+    break;
+  case D_DOWN :
+    g->u.mux.select_side = 1;
+    g->u.mux.data_order = 0;
+    g->wires[MUX_SEL]->orient = D_UP;
+    break;
+  }
+}
+
+GWireNode *Mux_wireSnap(GCElement *g,GWire *w,int *mod,int retry)
+{
+  int p,n;
+
+  if (posongate(w,w->gate,&p,&n) == 0) {
+    GPadLoc *pd = &g->typeinfo->Pad[p].Loc[g->orient];
+    if (p == MUX_SEL && g->u.mux.select_side)
+      pd = &mux_sel_altloc[g->orient];
+  
+    *mod = wire_force(w,pd->dir,retry);
+  }
+  return w->nodes;
+}
+
 void init_mux()
 {
   Pixmap P;
--- tkgate-1.8.7/src/tkgate/demux.c.orig	2007-01-27 21:44:02.000000000 +0000
+++ tkgate-1.8.7/src/tkgate/demux.c	2007-01-27 22:51:52.000000000 +0000
@@ -1,5 +1,5 @@
 /****************************************************************************
-    Copyright (C) 1987-2005 by Jeffery P. Hansen
+    Copyright (C) 1987-2007 by Jeffery P. Hansen
 
     This program is free software; you can redistribute it and/or modify
     it under the terms of the GNU General Public License as published by
@@ -15,7 +15,7 @@
     along with this program; if not, write to the Free Software
     Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
-    Last edit by hansen on Thu Jan 25 21:07:15 2007
+    Last edit by hansen on Sat Jan 27 17:43:49 2007
 ****************************************************************************/
 #include "tkgate.h"
 
@@ -23,14 +23,20 @@
 #define DEMUX_EN 1
 #define DEMUX_OUT 2
 
-#define F_IN 0x1
-
 void Demux_Delete(GCElement *g,GModuleDef *env,int drawp);
 void Demux_Draw(GCElement *g,int md);
 GCElement *Demux_Make(EditState **es,GModuleDef *env,int GType,
 		    int x,int y,int r,const char *Name,int noWires,const char**,int);
 void Demux_PSWrite(FILE *f,GModLayout*,GCElement *g);
 void Demux_Rotate(GCElement *g, int centX, int centY,int rdir);
+GCElement *MuxDemux_Copy(GModuleDef *M,GCElement *g,int x,int y,unsigned flags);
+void Demux_SetProp(GCElement*,const char*,const void*);
+void MuxDemux_EditProps(GCElement *g,int isLoadDialog);
+void Demux_VersionDelta(GCElement*,Version*); 
+GWireNode *Demux_wireSnap(GCElement *g,GWire *w,int *mod,int retry);
+void Demux_AddOutput(EditState *es,GCElement *g);
+void MuxDemux_VerSave(FILE*,GCElement*);
+GWireNode *Demux_wireSnap(GCElement *g,GWire *w,int *mod,int retry);
 
 static iconDimensions demux_iconDims[] = {
   {29, 60, 59, 29, 30, 12},
@@ -40,24 +46,31 @@
 };
 static int demux_iconBoldOffset = 89;
 
-struct locate demux_out_loc[] = {
+GPadLoc demux_out_loc[] = {
 	{-30,16,30,16,D_DOWN},
 	{16,-30,16,30,D_RIGHT},
 	{-30,-16,30,-16,D_UP},
 	{-16,-30,-16,30,D_LEFT},
 };
 
-struct locate demux_in_loc[] = {
+GPadLoc demux_in_loc[] = {
 	{0,-13,0,-13,D_UP},
 	{-13,0,-13,0,D_LEFT},
 	{0,13,0,13,D_DOWN},
 	{13,0,13,0,D_RIGHT},
 };
 
-struct locate demux_en_loc[] = {
+GPadLoc demux_en_loc[] = {
 	{-24,0,-24,0,D_LEFT},
+	{0,22,0,22,D_DOWN},
+	{22,0,22,0,D_RIGHT},
 	{0,-24,0,-24,D_UP},
+};
+
+GPadLoc demux_en_altloc[] = {
 	{22,0,22,0,D_RIGHT},
+	{0,-24,0,-24,D_UP},
+	{-24,0,-24,0,D_LEFT},
 	{0,22,0,22,D_DOWN},
 };
 
@@ -127,8 +140,11 @@
   Nop_SimStateFunc,
   Nop_SimHitFunc,
   Demux_PSWrite,
-  Generic_EditProps,
-  Generic_VerSave
+  MuxDemux_EditProps,
+  Generic_VerSave,
+  Demux_SetProp,
+  Demux_VersionDelta,
+  Demux_wireSnap
 };
 
 /*
@@ -152,39 +168,31 @@
   switch (g->orient) {
   case 0 :
     dx = (double)L/(double)(N+1);
-#if 0
     if (g->u.mux.data_order) {
       wx = g->xpos + pd->x2;
       dx = -dx;
     }
-#endif
     break;
   case 1 :
     dy = (double)L/(double)(N+1);
-#if 0
     if (!g->u.mux.data_order) {
       wy = g->ypos + pd->y2;
       dy = -dy;
     }
-#endif
     break;
   case 2 :
     dx = (double)L/(double)(N+1);
-#if 0
     if (!g->u.mux.data_order) {
       wx = g->xpos + pd->x2;
       dx = -dx;
     }
-#endif
     break;
   case 3 :
     dy = (double)L/(double)(N+1);
-#if 0
     if (g->u.mux.data_order) {
       wy = g->ypos + pd->y2;
       dy = -dy;
     }
-#endif
     break;
   }
   
@@ -205,10 +213,8 @@
   GWire *w = g->wires[DEMUX_EN];
   GWire *ow = wire_other(w);
 
-#if 0
   if (g->u.mux.select_side)
     pd = &demux_en_altloc[g->orient];
-#endif
 
   wx = g->xpos + pd->x2;
   wy = g->ypos + pd->y2;
@@ -249,17 +255,52 @@
   Demux_adjustWiresSelector(g);
 }
 
+
 GCElement *Demux_Make(EditState **es,GModuleDef *env,int GType,
 		    int x,int y,int r,const char *Name,int noWires,const char **options,int nOptions)
 {
-  int i,nbits;
-  struct celemnt *g;
-  struct wire *w;
+  int nbits;
+  GCElement *g;
+  GWire *w;
 
-  if (!(g = Generic_Make(es,env,GType,x,y,r,Name,noWires,options,nOptions)))
+  if (!(g = Generic_Make(es,env,GType,x,y,r,Name,1,options,nOptions)))
     return NULL;
 
+  g->u.mux.select_side = 0;
+  g->u.mux.data_order = 0;
+
   if (!noWires) {
+    int i,j;
+    const char *Invert,*Pins;
+    int N = GCElement_numPads(g);
+
+    
+    Invert = seekOption("-invert",options,nOptions);
+    Pins = seekOption("-pins",options,nOptions);
+
+    for (i = 0;i < N;i++) {
+      int Num = 0;
+
+      if (Pins) {
+	char buf[STRMAX];
+	int N;
+	if (sscanf(Pins,"%[^=]=%d",buf,&N) == 2 && strcmp(buf,GCElement_getPadName(g,i)) == 0)
+	  Num = N;
+      }
+      if (!Num)
+	Num = GCElement_getPadNum(g,i);
+
+      for (j = 0;j < Num;j++) {
+	if (Invert && strcmp(Invert,GCElement_getPadName(g,i)) == 0)
+	  wire_addToGate(g,i,env,1);
+	else
+	  wire_addToGate(g,i,env,0);
+      }
+    }
+
+    Demux_adjustWires(g);
+
+
     for (i = 0,w = g->wires[DEMUX_OUT];w;w = w->next,i++);
 
     nbits = required_bits(i);
@@ -339,6 +380,7 @@
   Demux_Draw_Special(g);
 }
 
+
 void Demux_Rotate(GCElement *g, int centX, int centY,int rdir)
 {
   int x = g->xpos;
@@ -352,6 +394,7 @@
   Demux_adjustWires(g);
 }
 
+
 void Demux_PSWrite(FILE *f,GModLayout *L,GCElement *g)
 {
     int N;
@@ -365,6 +408,80 @@
 	N,g->xpos,g->ypos,-g->orient*90,g->typeinfo->psprint);
 }
 
+void Demux_SetProp(GCElement *g,const char *prop,const void *value)
+{
+  if (strcmp(prop,"/ss") == 0) g->u.mux.select_side = *((int*)value);
+  if (strcmp(prop,"/do") == 0) g->u.mux.data_order = *((int*)value);
+
+  if (g->wires[DEMUX_EN])
+    g->wires[DEMUX_EN]->orient = (g->orient + 2 + (g->u.mux.select_side ? 2 : 0) ) % 4;
+}
+
+void Demux_VersionDelta(GCElement *g,Version *V)
+{
+  Version x1 = {1, 8, 7};
+
+  if (VersionCmp(V,&x1) >= 0)
+    return;
+
+  switch (g->orient) {
+  case D_RIGHT :
+    g->u.mux.select_side = 0;
+    g->u.mux.data_order = 0;
+    g->wires[DEMUX_EN]->orient = D_LEFT;
+    break;
+  case D_UP :
+    g->u.mux.select_side = 1;
+    g->u.mux.data_order = 1;
+    g->wires[DEMUX_EN]->orient = D_UP;
+    break;
+  case D_LEFT :
+    g->u.mux.select_side = 0;
+    g->u.mux.data_order = 1;
+    g->wires[DEMUX_EN]->orient = D_RIGHT;
+    break;
+  case D_DOWN :
+    g->u.mux.select_side = 1;
+    g->u.mux.data_order = 0;
+    g->wires[DEMUX_EN]->orient = D_DOWN;
+    break;
+  }
+}
+
+void Demux_AddOutput(EditState *es,GCElement *g)
+{
+  int i;
+  int N = GCElement_numPads(g);
+
+  for (i = 0;i < N;i++)
+    if (GCElement_getPadDir(g,i) == OUT && GCElement_getPadCanAdd(g,i))
+      break;
+  
+  if (i == N)
+    return;
+  
+  if (es) SetModified();
+  
+  gate_draw(g,GD_NORMAL);
+  wire_addToGate(g,i,es->env,0);
+  Demux_adjustWires(g);
+  gate_draw(g,GD_NORMAL);
+}
+
+GWireNode *Demux_wireSnap(GCElement *g,GWire *w,int *mod,int retry)
+{
+  int p,n;
+
+  if (posongate(w,w->gate,&p,&n) == 0) {
+    GPadLoc *pd = &g->typeinfo->Pad[p].Loc[g->orient];
+    if (p == DEMUX_EN && g->u.mux.select_side)
+      pd = &demux_en_altloc[g->orient];
+  
+    *mod = wire_force(w,pd->dir,retry);
+  }
+  return w->nodes;
+}
+
 void init_demux()
 {
   Pixmap P;
--- tkgate-1.8.7/locale/en/tutorials/create_tut.v.orig	2004-02-18 20:13:32.000000000 +0000
+++ tkgate-1.8.7/locale/en/tutorials/create_tut.v	2004-03-18 15:21:45.000000000 +0000
@@ -1,4 +1,5 @@
-//: version "1.6"
+//: version "1.8.3"
+//: property useExtBars = 0
 //: property discardChanges = 1
 
 module main;    //: root_module
@@ -26,7 +27,7 @@
   //: /end
   //: comment g10 /dolink:0 /link:"" @(12,236) /anc:1
   //: /line:"5) Create the two switches the same way as the AND and OR gates, by setting a position and selecting"
-  //: /line:"\"Switch\" from the \"Switch\" submenu of the \"Make\" menu, or by typing an \"s\"."
+  //: /line:"\"Switch\" from the \"I/O\" submenu of the \"Make\" menu, or by typing an \"s\"."
   //: /line:""
   //: /line:"6) Repeat step 4 to connect the output of the upper switch to the upper input of the AND gate."
   //: /line:""
@@ -42,8 +43,8 @@
   //: comment g9 /dolink:0 /link:"" @(12,9) /anc:1
   //: /line:"1) In this example, you will build a circuit like the one shown below. Build your"
   //: /line:"circuit in the frame box area in the lower right corner of this window.  Start by"
-  //: /line:"setting a position for the AND gate by clicking the right mouse button.  An \"X\""
-  //: /line:"symbol will appear to mark the position."
+  //: /line:"setting a position for the AND gate by clicking and holding the right mouse"
+  //: /line:"button.  An \"X\" symbol will appear to mark the position."
   //: /line:""
   //: /line:"2) Now select \"AND\" from the \"Gate\" submenu of the \"Make\" popup (or from the menu"
   //: /line:"bar) to create an AND gate at the position you selected. Alternatively, you can use"
--- tkgate-1.8.7/locale/ja/tutorials/create_tut.v.orig	2001-06-11 21:44:36.000000000 +0100
+++ tkgate-1.8.7/locale/ja/tutorials/create_tut.v	2002-01-11 21:24:29.000000000 +0000
@@ -1,4 +1,5 @@
-//: version "1.6f"
+//: version "1.6i"
+//: property useExtBars = 0
 //: property discardChanges = 1
 
 module main;    //: root_module
@@ -10,72 +11,72 @@
 wire w5;    //: /sn:0 {0}(232,497)(218,497)(218,464)(208,464){1}
 //: enddecls
 
-  //: comment g61 /dolink:1 /link:"@T/edwire_tut.v" @(622,100) /sn:0
-  //: /line:"ワイヤの編集"
-  //: /end
+  or g4 (.I0(w5), .I1(w6), .Z(w7));   //: @(243,500) /sn:0 /anc:1 /w:[ 0 0 0 ]
   //: comment g8 /dolink:0 /link:"" @(248,476) /anc:1
   //: /line:"ORゲート"
   //: /end
-  or g4 (.I0(w5), .I1(w6), .Z(w7));   //: @(243,500) /sn:0 /anc:1 /w:[ 0 0 0 ]
+  //: comment g61 /dolink:1 /link:"@T/edwire_tut.v" @(622,100) /sn:0
+  //: /line:"ワイヤの編集"
+  //: /end
+  //: frame g13 @(412,372) /sn:0 /wi:381 /ht:177 /tx:"回路をこの中に作成して下さい"
   and g3 (.I0(w2), .I1(w6), .Z(w5));   //: @(198,464) /sn:0 /anc:1 /w:[ 0 3 1 ]
-  //: frame g13 @(412,372) /sn:0 /wi:381 /ht:177 /tx:"回路をこの中に作成してください"
   //: switch g2 (w6) @(124,502) /sn:0 /anc:1 /w:[ 5 ] /st:0
   //: switch g1 (w2) @(124,461) /sn:0 /anc:1 /w:[ 1 ] /st:0
   //: comment g16 /dolink:1 /link:"@T/module_tut.v" @(622,115) /sn:0
   //: /line:"モジュールの使い方"
   //: /end
   //: comment g10 /dolink:0 /link:"" @(11,205) /anc:1
-  //: /line:"5) ANDゲートとORゲートと同じ様に二つのスイッチを作成して下さい。［作成］メニューの［スイッチ］"
+  //: /line:"5) ANDゲートとORゲートと同じ様に二つのスイッチを作成して下さい。［作成］メニューの［I/O］"
   //: /line:"サブメニューの［スイッチ］を選んで、又は、キーボード・ショートカットの\"s\"を押して下さい。"
   //: /line:""
-  //: /line:"6) ステップ4を繰り返して、上のスイッチの出力をANDゲートの上の出力に接続してください"
+  //: /line:"6) ステップ4を繰り返して、上のスイッチの出力をANDゲートの上の出力に接続して下さい。"
   //: /line:""
-  //: /line:"7) ステップ4を繰り返して、下のスイッチの出力をORゲートの下の出力に接続してください"
+  //: /line:"7) ステップ4を繰り返して、下のスイッチの出力をORゲートの下の出力に接続して下さい。"
   //: /line:""
-  //: /line:"8) ANDゲートの下の入力にクリックして、はんだごてを出して、下のスイッチとORゲートの間の"
-  //: /line:"ワイヤの真中にドラッグして、マウス・ボタンを話してください。接続点が作成される。"
+  //: /line:"8) ANDゲートの下の入力をクリックして、はんだごてを出し、下のスイッチとORゲートの間の"
+  //: /line:"ワイヤの真中にドラッグして、マウスボタンを離して下さい。接続点が作成されます。"
   //: /end
   //: comment g6 /dolink:0 /link:"" @(101,427) /anc:1
   //: /line:"スイッチ"
   //: /end
+  //: comment g7 /dolink:0 /link:"" @(192,436) /anc:1
+  //: /line:"ANDゲート"
+  //: /end
   //: comment g9 /dolink:0 /link:"" @(12,9) /anc:1
-  //: /line:"1) このチュートリアルでは、以下に示してあるのと同じ様な回路を作成する。右下の"
-  //: /line:"点の箱の中に作成して下さい。まず、右のマウスボタンでANDゲートの箇所を選んで"
-  //: /line:"下さい。クリックした箇所にX印と［作成］メニューが現れる。"
+  //: /line:"1) このチュートリアルでは、下に示してあるのと同じ様な回路を作成する。右下の"
+  //: /line:"点の箱の中に作成して下さい。まず、マウスの右ボタンでANDゲートの位置を選んで"
+  //: /line:"下さい。クリックした位置にX印と［作成］メニューが現れます。"
   //: /line:""
   //: /line:"2) それから、［作成］メニューの［ゲート］サブメニューの［AND］を選んで下さい。"
-  //: /line:"メニュー・バー、又はキーボード・ショートカットの\"a\"でも使える。"
+  //: /line:"メニュー・バー、又はキーボード・ショートカットの\"a\"も使えます。"
   //: /line:""
-  //: /line:"3) ステップ1、2を繰り返して, ORゲートをANDゲートの右下の箇所に作成して下さい。"
-  //: /line:"メニュー、又はキーボードショート・カットの\"o\"でも使える。"
+  //: /line:"3) ステップ1、2を繰り返して, ORゲートをANDゲートの右下の位置に作成して下さい。"
+  //: /line:"メニュー・バー、又はキーボードショート・カットの\"o\"も使えます。"
   //: /line:""
-  //: /line:"4) ANDゲートの出力のワイヤの先端に左マウス・ボタンでクリックして、ボタンを"
-  //: /line:"押したままにする。マウス・カーソルははんだごてに変わる。ワイヤをORゲートの"
-  //: /line:"上の入力までドラッグしてマウス・ボタンを話すと、ワイヤが接続される。"
-  //: /end
-  //: comment g7 /dolink:0 /link:"" @(192,436) /anc:1
-  //: /line:"ANDゲート"
+  //: /line:"4) ANDゲートの出力のワイヤの先端をマウスの左ボタンでクリックして、ボタンを押し"
+  //: /line:"たままにします。マウス・カーソルがはんだごてに変わります。ワイヤをORゲートの"
+  //: /line:"上の入力までドラッグしてマウス・ボタンを離すと、ワイヤが接続されます。"
   //: /end
+  //: frame g15 @(600,25) /sn:0 /wi:224 /ht:149 /tx:"チュートリアル"
   //: comment g20 /dolink:0 /link:"@T/create_tut.v" @(622,55) /sn:0 /anc:1
   //: /line:"回路の作成"
   //: /end
-  //: frame g15 @(600,25) /sn:0 /wi:224 /ht:149 /tx:"チュートリアル"
   //: comment g17 /dolink:1 /link:"@T/sim_tut.v" @(622,130) /sn:0
   //: /line:"組合せ回路のシミュレーション"
   //: /end
+  //: joint g5 (w6) @(175, 502) /anc:1 /w:[ 1 2 4 -1 ]
   //: comment g14 /dolink:1 /link:"@T/seqsim_tut.v" @(622,145) /sn:0
   //: /line:"順序回路のシミュレーション"
   //: /end
-  //: joint g5 (w6) @(175, 502) /anc:1 /w:[ 1 2 4 -1 ]
   //: comment g21 /dolink:1 /link:"@T/edit2_tut.v" @(622,85) /sn:0
   //: /line:"グループの編集機能"
   //: /end
-  //: comment g22 /dolink:0 /link:"" @(605,55) /sn:0
-  //: /line:"->"
-  //: /end
   //: comment g0 /dolink:1 /link:"@T/edit1_tut.v" @(622,70) /sn:0
   //: /line:"基本の編集モード"
   //: /end
+  //: comment g22 /dolink:0 /link:"" @(605,55) /sn:0
+  //: /line:"->"
+  //: /end
   //: comment g18 /dolink:1 /link:"@T/welcome_tut.v" @(622,40) /sn:0
   //: /line:"TkGateの紹介"
   //: /end
--- tkgate-1.8.7/locale/ja/tutorials/edit1_tut.v.orig	2001-06-11 21:44:36.000000000 +0100
+++ tkgate-1.8.7/locale/ja/tutorials/edit1_tut.v	2002-01-11 21:24:32.000000000 +0000
@@ -1,4 +1,5 @@
-//: version "1.6f"
+//: version "1.6i"
+//: property useExtBars = 0
 //: property discardChanges = 1
 
 module main;    //: root_module
@@ -11,14 +12,14 @@
 wire c1;    //: {0}(700,396)(700,433){1}
 //: {2}(702,435)(725,435){3}
 //: {4}(700,437)(700,447){5}
-wire s1;    //: /dp:1 {0}(746,433)(781,433){1}
 wire a1;    //: {0}(598,427)(638,427){1}
 //: {2}(642,427)(650,427){3}
 //: {4}(640,429)(640,483)(657,483){5}
+wire s1;    //: /dp:1 {0}(746,433)(781,433){1}
+wire w8;    //: /sn:0 {0}(678,486)(693,486)(693,498){1}
 wire b1;    //: {0}(598,462)(632,462){1}
 //: {2}(634,460)(634,432)(650,432){3}
 //: {4}(634,464)(634,488)(657,488){5}
-wire w8;    //: /sn:0 {0}(678,486)(693,486)(693,498){1}
 wire a0;    //: /dp:1 {0}(598,304)(642,304){1}
 //: {2}(646,304)(654,304){3}
 //: {4}(644,306)(644,360)(661,360){5}
@@ -35,96 +36,96 @@
 //: {4}(699,309)(699,324){5}
 //: enddecls
 
-  //: joint g8 (c1) @(700, 435) /w:[ 2 1 -1 4 ]
-  nand g4 (.I0(w8), .I1(w5), .Z(c2));   //: @(696,509) /sn:0 /R:3 /w:[ 1 1 1 ]
   //: comment g61 /dolink:1 /link:"@T/edwire_tut.v" @(622,100) /sn:0
   //: /line:"ワイヤの編集"
   //: /end
-  led g37 (.I(s0));   //: @(787,310) /sn:0 /R:3 /w:[ 1 ] /type:0
-  xor g34 (.I0(w10), .I1(c1), .Z(s1));   //: @(736,433) /sn:0 /w:[ 0 3 0 ]
+  nand g4 (.I0(w8), .I1(w5), .Z(c2));   //: @(696,509) /sn:0 /R:3 /w:[ 1 1 1 ]
+  //: joint g8 (c1) @(700, 435) /w:[ 2 1 -1 4 ]
   //: comment g3 /dolink:0 /link:"" @(29,249) /anc:1
-  //: /line:"5) 「ツール」メニューの「ビット幅」を選択して、リボンケーブルのツールを"
-  //: /line:"出して下さい。そうすると、ウィンドの右下に「ビット幅」の入力ボックスも表す。"
-  //: /line:"ボックスにクリックして、1〜32の数字を入力して下さい。それからワイヤに"
+  //: /line:"5) 「ツール」メニューの「ビット幅」を選択して、リボンケーブルのツールを出し"
+  //: /line:"て下さい。そうすると、ウィンドウ右下に「ビット幅」の入力ボックスが出ます。"
+  //: /line:"ボックスをクリックして、1〜32の数字を入力して下さい。それからワイヤを"
   //: /line:"クリックして、示しているビット幅に設定してみて下さい。"
   //: /line:""
-  //: /line:"6) 「ツール」メニューの「移動／接続」を選択しおいて、ワイヤにダブル・クリック"
-  //: /line:"すると、特徴ボックスが開く。ネット名、ネット名が表示するかどうか、ビット幅などの"
-  //: /line:"特徴が設定できる。右のマウスボタンでも利用して開ける"
-  //: /line:""
-  //: /line:"7) ORゲートなどにダブル・クリックすると、ゲートの特徴ボックスが開く。"
-  //: /line:"ゲート名、ゲート名が表示するかどうか、ゲートが移動出来るかどうかなどの"
-  //: /line:"特徴が設定出来る。ポートの追加、削除、編集なども出来る。"
+  //: /line:"6) 「ツール」メニューの「移動／接続」を選択し、ワイヤをダブル・クリックする"
+  //: /line:"と、属性ボックスが開きます。ネット名、ネット名を表示するかどうか、ビット幅"
+  //: /line:"などの属性が設定できます。マウスの右ボタンのメニューでも設定できます。"
+  //: /line:""
+  //: /line:"7) ORゲートなどをダブル・クリックすると、ゲートの属性ボックスが開きます。"
+  //: /line:"ゲート名、ゲート名を表示するかどうか、ゲートが移動出来るかどうかなどの"
+  //: /line:"属性が設定出来ます。ポートの追加、削除、編集なども出来ます。"
   //: /line:""
-  //: /line:"8) ゲートに右マウスボタンでクリックして、「移動不可」のフラグをセットして"
-  //: /line:"みて下さい。このフラグがセットしてあれば、ゲートの移動は出来ない。"
+  //: /line:"8) ゲートをマウス右ボタンでクリックして、「移動不可」のフラグをセットして"
+  //: /line:"みて下さい。このフラグがセットしてあれば、ゲートの移動は出来ません。"
   //: /end
+  xor g34 (.I0(w10), .I1(c1), .Z(s1));   //: @(736,433) /sn:0 /w:[ 0 3 0 ]
+  led g37 (.I(s0));   //: @(787,310) /sn:0 /R:3 /w:[ 1 ] /type:0
   //: comment g2 /dolink:0 /link:"" @(29,38) /anc:1
-  //: /line:"1) 左のマウスボタンでゲートにクリックして、別の箇所に移動してみて下さい。"
-  //: /line:"ゲートに繋いでいるワイヤもついて移動される。"
+  //: /line:"1) マウスの左ボタンでゲートをクリックし、ボタンを押したまま別の場所に移動"
+  //: /line:"してみて下さい。ゲートに繋いでいるワイヤも一緒に移動します。"
   //: /line:""
   //: /line:"2) 「ツール」メニューの「ワイヤの切断」を選択して、カッターのツールを"
-  //: /line:"出し、ワイヤのどこかにクリックしてみて下さい。何にも繋いでいないワイヤの"
-  //: /line:"部分は自動的に削除される。右マウスボタンで開くメニューも利用して"
-  //: /line:"カッターを出せる。"
+  //: /line:"出し、ワイヤのどこかをクリックしてみて下さい。何にも繋いでいないワイヤの"
+  //: /line:"部分は自動的に削除されます。マウス右ボタンで開くメニューを利用しても"
+  //: /line:"カッターを出せます。"
   //: /line:""
   //: /line:"3) 「ツール」メニューの「否定」を選択して、否定ツールを出し、ゲートの"
-  //: /line:"入力や出力にクリックしてみて下さい。否定の丸が表したり、消えたりする。"
+  //: /line:"入力や出力をクリックしてみて下さい。否定の丸が現れたり、消えたりします。"
   //: /line:""
-  //: /line:"4) 「ツール」メニューの「ゲートの削除」を選択して、ゲートにクリックして"
-  //: /line:"みて下さい。ゲートが削除される。ゲートを選択して、デリート・キーを押したり、"
-  //: /line:"右マウスボタンで「削除」を選択したりすることでもゲートが削除できる。"
+  //: /line:"4) 「ツール」メニューの「ゲートの削除」を選択し、ゲートをクリックしてみて"
+  //: /line:"下さい。ゲートが削除されます。ゲートを選択して、デリート・キーを押したり、"
+  //: /line:"マウス右ボタンで「削除」を選択したりすることでもゲートが削除できます。"
   //: /end
   nand g1 (.I0(w7), .I1(w4), .Z(c1));   //: @(700,386) /sn:0 /R:3 /w:[ 1 1 0 ]
-  //: switch g11 (a1) @(581,427) /sn:0 /w:[ 0 ] /st:0
   //: comment g16 /dolink:1 /link:"@T/module_tut.v" @(622,115) /sn:0
   //: /line:"モジュールの使い方"
   //: /end
-  //: joint g10 (a1) @(640, 427) /w:[ 2 -1 1 4 ]
+  //: switch g11 (a1) @(581,427) /sn:0 /w:[ 0 ] /st:0
   nand g28 (.I0(a0), .I1(b0), .Z(w7));   //: @(672,363) /sn:0 /w:[ 5 5 0 ]
-  led g19 (.I(s1));   //: @(788,433) /sn:0 /R:3 /w:[ 1 ] /type:0
-  //: switch g27 (c0) @(581,268) /sn:0 /w:[ 0 ] /st:0
+  //: joint g10 (a1) @(640, 427) /w:[ 2 -1 1 4 ]
   //: joint g32 (w9) @(699, 307) /w:[ 1 -1 2 4 ]
-  //: joint g38 (b1) @(634, 462) /w:[ -1 2 1 4 ]
+  //: switch g27 (c0) @(581,268) /sn:0 /w:[ 0 ] /st:0
+  led g19 (.I(s1));   //: @(788,433) /sn:0 /R:3 /w:[ 1 ] /type:0
   nand g6 (.I0(a1), .I1(b1), .Z(w8));   //: @(668,486) /sn:0 /w:[ 5 5 0 ]
-  nand g9 (.I0(w10), .I1(c1), .Z(w5));   //: @(698,458) /sn:0 /R:3 /w:[ 5 5 0 ]
+  //: joint g38 (b1) @(634, 462) /w:[ -1 2 1 4 ]
   //: joint g7 (w10) @(695, 430) /w:[ 1 -1 2 4 ]
-  nand g31 (.I0(w9), .I1(c0), .Z(w4));   //: @(702,335) /sn:0 /R:3 /w:[ 5 5 0 ]
-  //: frame g15 @(600,25) /sn:0 /wi:224 /ht:149 /tx:"チュートリアル"
+  nand g9 (.I0(w10), .I1(c1), .Z(w5));   //: @(698,458) /sn:0 /R:3 /w:[ 5 5 0 ]
   //: comment g20 /dolink:1 /link:"@T/create_tut.v" @(622,55) /sn:0
   //: /line:"回路の作成"
   //: /end
+  //: frame g15 @(600,25) /sn:0 /wi:224 /ht:149 /tx:"チュートリアル"
+  nand g31 (.I0(w9), .I1(c0), .Z(w4));   //: @(702,335) /sn:0 /R:3 /w:[ 5 5 0 ]
   led g39 (.I(c2));   //: @(790,530) /sn:0 /R:3 /w:[ 0 ] /type:0
-  //: switch g25 (a0) @(581,304) /sn:0 /w:[ 0 ] /st:0
-  //: joint g29 (a0) @(644, 304) /w:[ 2 -1 1 4 ]
   //: comment g17 /dolink:1 /link:"@T/sim_tut.v" @(622,130) /sn:0
   //: /line:"組合せ回路のシミュレーション"
   //: /end
-  //: comment g5 /dolink:0 /link:"@T/edit1_tut.v" @(622,70) /sn:0 /anc:1
-  //: /line:"基本の編集モード"
-  //: /end
+  //: joint g29 (a0) @(644, 304) /w:[ 2 -1 1 4 ]
+  //: switch g25 (a0) @(581,304) /sn:0 /w:[ 0 ] /st:0
   //: comment g14 /dolink:1 /link:"@T/seqsim_tut.v" @(622,145) /sn:0
   //: /line:"順序回路のシミュレーション"
   //: /end
-  //: switch g36 (b1) @(581,462) /sn:0 /w:[ 0 ] /st:0
-  xor g24 (.I0(w9), .I1(c0), .Z(s0));   //: @(740,310) /sn:0 /w:[ 0 3 0 ]
+  //: comment g5 /dolink:0 /link:"@T/edit1_tut.v" @(622,70) /sn:0 /anc:1
+  //: /line:"基本の編集モード"
+  //: /end
   //: comment g21 /dolink:1 /link:"@T/edit2_tut.v" @(622,85) /sn:0
   //: /line:"グループの編集機能"
   //: /end
+  xor g24 (.I0(w9), .I1(c0), .Z(s0));   //: @(740,310) /sn:0 /w:[ 0 3 0 ]
+  //: switch g36 (b1) @(581,462) /sn:0 /w:[ 0 ] /st:0
   xor g23 (.I0(a0), .I1(b0), .Z(w9));   //: @(665,307) /sn:0 /w:[ 3 3 3 ]
-  xor g35 (.I0(a1), .I1(b1), .Z(w10));   //: @(661,430) /sn:0 /w:[ 3 3 3 ]
+  //: comment g22 /dolink:0 /link:"" @(605,70) /sn:0
+  //: /line:"->"
+  //: /end
+  //: switch g26 (b0) @(581,339) /sn:0 /w:[ 0 ] /st:0
   //: comment g0 /dolink:0 /link:"" @(8,10) /anc:1
   //: /line:"このチュートリアルでは、TkGateの基本編集機能を紹介する。"
   //: /line:""
   //: /end
-  //: switch g26 (b0) @(581,339) /sn:0 /w:[ 0 ] /st:0
-  //: comment g22 /dolink:0 /link:"" @(605,70) /sn:0
-  //: /line:"->"
-  //: /end
+  xor g35 (.I0(a1), .I1(b1), .Z(w10));   //: @(661,430) /sn:0 /w:[ 3 3 3 ]
   //: comment g18 /dolink:1 /link:"@T/welcome_tut.v" @(622,40) /sn:0
   //: /line:"TkGateの紹介"
   //: /end
-  //: joint g30 (b0) @(638, 339) /w:[ -1 2 1 4 ]
   //: joint g33 (c0) @(704, 312) /w:[ 2 1 -1 4 ]
+  //: joint g30 (b0) @(638, 339) /w:[ -1 2 1 4 ]
 
 endmodule
--- tkgate-1.8.7/locale/ja/tutorials/edit2_tut.v.orig	2001-06-11 21:44:36.000000000 +0100
+++ tkgate-1.8.7/locale/ja/tutorials/edit2_tut.v	2002-01-11 21:24:34.000000000 +0000
@@ -1,4 +1,5 @@
-//: version "1.6f"
+//: version "1.6i"
+//: property useExtBars = 0
 //: property discardChanges = 1
 
 module main;    //: root_module
@@ -8,68 +9,68 @@
 wire w5;    //: /sn:0 {0}(777,262)(787,262){1}
 //: enddecls
 
-  and g4 (.I0(w1), .I1(w3), .Z(w5));   //: @(767,262) /sn:0 /w:[ 0 0 0 ]
   //: comment g61 /dolink:1 /link:"@T/edwire_tut.v" @(622,100) /sn:0
   //: /line:"ワイヤの編集"
   //: /end
+  and g4 (.I0(w1), .I1(w3), .Z(w5));   //: @(767,262) /sn:0 /w:[ 0 0 0 ]
   //: switch g3 (w2) @(662,302) /sn:0 /w:[ 0 ] /st:0
   //: switch g2 (w3) @(662,264) /sn:0 /w:[ 1 ] /st:0
   //: switch g1 (w1) @(662,224) /sn:0 /w:[ 1 ] /st:0
-  //: frame g11 @(434,347) /sn:0 /anc:1 /wi:373 /ht:230 /tx:"ゲートをこの中にコピーする"
   //: comment g16 /dolink:1 /link:"@T/module_tut.v" @(622,115) /sn:0
   //: /line:"モジュールの使い方"
   //: /end
+  //: frame g11 @(434,347) /sn:0 /anc:1 /wi:373 /ht:230 /tx:"ゲートをこの中にコピーする"
   //: frame g10 @(22,347) /sn:0 /anc:1 /wi:394 /ht:227 /tx:"ゲートをこの中に移動する"
   //: comment g6 /dolink:1 /link:"@T/edit1_tut.v" @(622,70) /sn:0
   //: /line:"基本の編集モード"
   //: /end
-  //: comment g9 /dolink:0 /link:"" @(607,452) /sn:0
+  //: comment g7 /dolink:0 /link:"" @(634,193) /sn:0
   //: /line:""
   //: /end
-  //: comment g7 /dolink:0 /link:"" @(634,193) /sn:0
+  //: comment g9 /dolink:0 /link:"" @(607,452) /sn:0
   //: /line:""
   //: /end
-  //: frame g15 @(600,25) /sn:0 /wi:224 /ht:149 /tx:"チュートリアル"
   //: comment g20 /dolink:1 /link:"@T/create_tut.v" @(622,55) /sn:0
   //: /line:"回路の作成"
   //: /end
+  //: frame g15 @(600,25) /sn:0 /wi:224 /ht:149 /tx:"チュートリアル"
   //: comment g17 /dolink:1 /link:"@T/sim_tut.v" @(622,130) /sn:0
   //: /line:"組合せ回路のシミュレーション"
   //: /end
+  //: comment g14 /dolink:1 /link:"@T/seqsim_tut.v" @(622,145) /sn:0
+  //: /line:"順序回路のシミュレーション"
+  //: /end
   //: comment g5 /dolink:0 /link:"" @(38,36) /anc:1
-  //: /line:"1) 左マウス・ボタンを右にある回路の左上にクリックして、箱が回路を"
-  //: /line:"完全に含むまで、マウスを右下へドラッグして、ボタンを放して下さい。"
-  //: /line:"ゲートが選択され、太線で表示される。選択されているゲートの集合は"
-  //: /line:"セレクションという。"
-  //: /line:""
-  //: /line:"2) ゲートのどれかにクリックして「ゲートをこの中に移動する」と書いてある"
-  //: /line:"点線の箱の中に移動してみて下さい。ゲートやワイヤのない所にクリックすると"
-  //: /line:"セレクションがなくなる。"
+  //: /line:"1) マウス左ボタンで右にある回路の左上をクリックして、箱が回路を完全に"
+  //: /line:"含むまでマウスを右下へドラッグして、ボタンを離して下さい。"
+  //: /line:"ゲートが選択され、太線で表示されます。選択されているゲートの集合を"
+  //: /line:"セレクションといいます。"
+  //: /line:""
+  //: /line:"2) ゲートのどれかをクリックして「ゲートをこの中に移動する」と書いてある"
+  //: /line:"点線の箱の中に移動してみて下さい。ゲートやワイヤのない所をクリックすると"
+  //: /line:"セレクションは解除されます。"
   //: /line:""
-  //: /line:"3) ANDゲートを選択して、右マウス・ボタンを押して、「入力を追加」を選択して"
-  //: /line:"下さい。それからANDゲートに追加されたワイヤを下のスイッチを接続して下さい。"
+  //: /line:"3) ANDゲートを選択して、マウス右ボタンを押し、「入力を追加」を選択して"
+  //: /line:"下さい。それからANDゲートに追加されたワイヤを下のスイッチに接続して下さい。"
   //: /line:""
   //: /line:"4) ステップ1に書いてある様に全てのゲートを選択しておいて下さい。それから"
-  //: /line:"「編集」メニューの「コピー」を選択して、左マウス・ボタンを「ゲートをこの中に"
-  //: /line:"コピーする」と書いてある点線の箱の真中にクリックして、「編集」メニューの"
-  //: /line:"「張り付け」を選択して下さい。ゲートのセレクションがコピーされる。セレクションが"
-  //: /line:"ある時、デリート・キーを押すと、セレクションのゲートが削除される。"
+  //: /line:"「編集」メニューの「コピー」を選択し、マウス左ボタンで「ゲートをこの中に"
+  //: /line:"コピーする」と書いてある点線の箱の真中をクリックして、「編集」メニューの"
+  //: /line:"「張り付け」を選択して下さい。ゲートのセレクションがコピーされます。セレク"
+  //: /line:"ションがある時、デリート・キーを押すと、セレクションのゲートが削除されます。"
   //: /line:""
-  //: /line:"5) コントロール・キーを押しながら、ゲートにクリックすると、ゲートを一つずつ"
-  //: /line:"セレクションに追加することが出来る。"
-  //: /end
-  //: comment g14 /dolink:1 /link:"@T/seqsim_tut.v" @(622,145) /sn:0
-  //: /line:"順序回路のシミュレーション"
+  //: /line:"5) コントロール・キーを押しながら、ゲートをクリックすると、ゲートを一つずつ"
+  //: /line:"セレクションに追加することが出来ます。"
   //: /end
   //: comment g21 /dolink:0 /link:"@T/edit2_tut.v" @(622,85) /sn:0 /anc:1
   //: /line:"グループの編集機能"
   //: /end
-  //: comment g0 /dolink:0 /link:"" @(26,17) /anc:1
-  //: /line:"このチュートリアルはグループ編集の機能を紹介する。"
-  //: /end
   //: comment g22 /dolink:0 /link:"" @(605,85) /sn:0
   //: /line:"->"
   //: /end
+  //: comment g0 /dolink:0 /link:"" @(26,17) /anc:1
+  //: /line:"このチュートリアルはグループ編集の機能を紹介する。"
+  //: /end
   //: comment g18 /dolink:1 /link:"@T/welcome_tut.v" @(622,40) /sn:0
   //: /line:"TkGateの紹介"
   //: /end
--- tkgate-1.8.7/locale/ja/tutorials/welcome_tut.v.orig	2001-06-11 21:44:36.000000000 +0100
+++ tkgate-1.8.7/locale/ja/tutorials/welcome_tut.v	2004-06-09 18:03:38.000000000 +0100
@@ -1,14 +1,16 @@
-//: version "1.6"
+//: version "1.8.4"
+//: property useExtBars = 0
 //: property discardChanges = 1
 
 module ALU(F, A, B, Z);
-//: interface  /sz:(89, 65) /bd:[ Ti0>A[7:0](20/89) Ti1>B[7:0](67/89) Li0>F[1:0](28/65) Bo0<Z[7:0](44/89) ]
+//: interface  /sz:(89, 65) /bd:[ Ti0>B[7:0](67/89) Ti1>A[7:0](20/89) Li0>F[1:0](28/65) Bo0<Z[7:0](44/89) ]
 input [7:0] B;    //: /sn:0 {0}(342,147)(342,184){1}
 //: {2}(344,186)(367,186){3}
 //: {4}(371,186)(425,186)(425,258){5}
 //: {6}(369,188)(369,261){7}
 //: {8}(342,188)(342,258){9}
 input [1:0] F;    //: /sn:0 /dp:1 {0}(343,353)(214,353){1}
+supply0 w0;    //: /sn:0 {0}(491,295)(491,272)(465,272){1}
 input [7:0] A;    //: /sn:0 {0}(265,147)(265,215){1}
 //: {2}(267,217)(345,217){3}
 //: {4}(349,217)(372,217){5}
@@ -16,7 +18,6 @@
 //: {8}(374,219)(374,261){9}
 //: {10}(347,219)(347,258){11}
 //: {12}(265,219)(265,322)(348,322)(348,337){13}
-supply0 w0;    //: /sn:0 {0}(491,295)(491,272)(465,272){1}
 output [7:0] Z;    //: /sn:0 {0}(366,404)(366,366){1}
 wire [7:0] w14;    //: /sn:0 {0}(345,279)(345,310)(360,310)(360,337){1}
 wire w4;    //: /sn:0 {0}(417,272)(407,272){1}
@@ -24,17 +25,17 @@
 wire [7:0] w10;    //: /sn:0 /dp:1 {0}(384,337)(384,307)(441,307)(441,287){1}
 //: enddecls
 
-  //: joint g8 (A) @(347, 217) /w:[ 4 -1 3 10 ]
   add g4 (.A(B), .B(A), .S(w10), .CI(w0), .CO(w4));   //: @(441,274) /sn:0 /w:[ 5 7 1 1 0 ]
-  //: supply0 g13 (w0) @(491,301) /sn:0 /w:[ 0 ]
+  //: joint g8 (A) @(347, 217) /w:[ 4 -1 3 10 ]
   //: input g3 (F) @(212,353) /sn:0 /w:[ 1 ]
+  //: supply0 g13 (w0) @(491,301) /sn:0 /w:[ 0 ]
   //: input g2 (A) @(265,145) /sn:0 /R:3 /w:[ 0 ]
   //: input g1 (B) @(342,145) /sn:0 /R:3 /w:[ 0 ]
   //: joint g11 (B) @(369, 186) /w:[ 4 -1 3 6 ]
   //: joint g10 (A) @(374, 217) /w:[ 6 -1 5 8 ]
   and g6 (.I0(B), .I1(A), .Z(w11));   //: @(372,272) /sn:0 /R:3 /w:[ 7 9 0 ]
-  //: joint g9 (A) @(265, 217) /w:[ 2 1 -1 12 ]
   or g7 (.I0(B), .I1(A), .Z(w14));   //: @(345,269) /sn:0 /R:3 /w:[ 9 11 0 ]
+  //: joint g9 (A) @(265, 217) /w:[ 2 1 -1 12 ]
   //: output g5 (Z) @(366,401) /sn:0 /R:3 /w:[ 0 ]
   mux g0 (.I0(A), .I1(w14), .I2(w11), .I3(w10), .S(F), .Z(Z));   //: @(366,353) /sn:0 /w:[ 13 1 1 0 0 1 ]
   //: joint g12 (B) @(342, 186) /w:[ 2 1 -1 8 ]
@@ -42,7 +43,7 @@
 endmodule
 
 module REGS(SB, SA, BOUT, AOUT, ck, AIN);
-//: interface  /sz:(94, 80) /bd:[ Ti0>AIN[7:0](46/94) Li0>SA[1:0](14/80) Li1>SB[1:0](34/80) Li2>ck(56/80) Bi0>BOUT[7:0](79/94) Bi1>AOUT[7:0](16/94) ]
+//: interface  /sz:(94, 80) /bd:[ Ti0>AIN[7:0](46/94) Li0>ck(56/80) Li1>SB[1:0](34/80) Li2>SA[1:0](14/80) Bo0<BOUT[7:0](79/94) Bo1<AOUT[7:0](16/94) ]
 supply1 w0;    //: /sn:0 /dp:1 {0}(361,358)(373,358)(373,339){1}
 supply1 w21;    //: /sn:0 /dp:1 {0}(260,211)(181,211)(181,173){1}
 input [1:0] SB;    //: /sn:0 {0}(507,507)(570,507){1}
@@ -58,6 +59,7 @@
 //: {6}(320,287)(180,287)(180,352){7}
 //: {8}(322,289)(322,353){9}
 //: {10}(467,289)(467,353){11}
+output [7:0] AOUT;    //: /sn:0 /dp:1 {0}(405,520)(405,549){1}
 input ck;    //: /sn:0 {0}(51,392)(124,392){1}
 //: {2}(128,392)(273,392){3}
 //: {4}(277,392)(412,392){5}
@@ -65,64 +67,63 @@
 //: {8}(414,390)(414,363)(430,363){9}
 //: {10}(275,390)(275,363)(285,363){11}
 //: {12}(126,390)(126,362)(143,362){13}
-output [7:0] AOUT;    //: /sn:0 /dp:1 {0}(405,520)(405,549){1}
 supply1 w5;    //: /sn:0 /dp:1 {0}(665,358)(680,358)(680,340){1}
 wire w4;    //: /sn:0 {0}(290,227)(290,266)(555,266)(555,368)(506,368){1}
+wire w3;    //: /sn:0 {0}(278,227)(278,281)(404,281)(404,368)(361,368){1}
 wire [7:0] R2;    //: /dp:4 {0}(411,491)(411,432)(465,432){1}
 //: {2}(469,432)(599,432)(599,491){3}
 //: {4}(467,430)(467,374){5}
-wire w3;    //: /sn:0 {0}(278,227)(278,281)(404,281)(404,368)(361,368){1}
 wire w20;    //: /sn:0 {0}(302,227)(302,248)(712,248)(712,368)(665,368){1}
-wire [7:0] R3;    //: /dp:3 {0}(423,491)(423,447)(609,447){1}
-//: {2}(613,447)(626,447)(626,374){3}
-//: {4}(611,449)(611,491){5}
 wire [7:0] R1;    //: /dp:5 {0}(399,491)(399,422){1}
 //: {2}(401,420)(587,420)(587,491){3}
 //: {4}(397,420)(322,420)(322,374){5}
+wire [7:0] R3;    //: /dp:3 {0}(423,491)(423,447)(609,447){1}
+//: {2}(613,447)(626,447)(626,374){3}
+//: {4}(611,449)(611,491){5}
 wire w8;    //: /sn:0 /dp:1 {0}(219,367)(266,367)(266,227){1}
 wire [7:0] R0;    //: /dp:3 {0}(575,491)(575,460)(389,460){1}
 //: {2}(385,460)(180,460)(180,373){3}
 //: {4}(387,462)(387,491){5}
 //: enddecls
 
-  register g8 (.Q(R3), .D(AIN), .EN(w20), .CLR(w5), .CK(ck));   //: @(626,363) /sn:0 /w:[ 3 3 1 0 7 ]
   //: input g4 (SB) @(505,507) /sn:0 /w:[ 0 ]
-  mux g13 (.I0(R0), .I1(R1), .I2(R2), .I3(R3), .S(SB), .Z(BOUT));   //: @(593,507) /sn:0 /w:[ 0 3 3 5 1 1 ]
+  register g8 (.Q(R3), .D(AIN), .EN(w20), .CLR(w5), .CK(ck));   //: @(626,363) /sn:0 /w:[ 3 3 1 0 7 ]
   //: input g3 (SA) @(45,136) /sn:0 /w:[ 5 ]
+  mux g13 (.I0(R0), .I1(R1), .I2(R2), .I3(R3), .S(SB), .Z(BOUT));   //: @(593,507) /sn:0 /w:[ 0 3 3 5 1 1 ]
   //: output g2 (BOUT) @(593,546) /sn:0 /R:3 /w:[ 0 ]
   //: output g1 (AOUT) @(405,546) /sn:0 /R:3 /w:[ 1 ]
-  //: joint g16 (R0) @(387, 460) /w:[ 1 -1 2 4 ]
   //: joint g11 (ck) @(275, 392) /w:[ 4 10 3 -1 ]
+  //: joint g16 (R0) @(387, 460) /w:[ 1 -1 2 4 ]
   //: supply1 g10 (w21) @(192,173) /sn:0 /w:[ 1 ]
-  //: joint g27 (ck) @(126, 392) /w:[ 2 12 1 -1 ]
   //: joint g19 (R1) @(399, 420) /w:[ 2 -1 4 1 ]
+  //: joint g27 (ck) @(126, 392) /w:[ 2 12 1 -1 ]
   register g6 (.Q(R1), .D(AIN), .EN(w3), .CLR(w0), .CK(ck));   //: @(322,363) /sn:0 /w:[ 5 9 1 0 11 ]
-  demux g9 (.I(SA), .E(w21), .Z0(!w8), .Z1(!w3), .Z2(!w4), .Z3(!w20));   //: @(284,211) /sn:0 /w:[ 3 0 1 0 0 0 ]
   register g7 (.Q(R2), .D(AIN), .EN(w4), .CLR(w1), .CK(ck));   //: @(467,363) /sn:0 /w:[ 5 11 1 0 9 ]
-  register g15 (.Q(R0), .D(AIN), .EN(w8), .CLR(w2), .CK(ck));   //: @(180,362) /sn:0 /w:[ 3 7 0 0 13 ]
+  demux g9 (.I(SA), .E(w21), .Z0(!w8), .Z1(!w3), .Z2(!w4), .Z3(!w20));   //: @(284,211) /sn:0 /w:[ 3 0 1 0 0 0 ]
   //: joint g20 (SA) @(92, 136) /w:[ 2 -1 4 1 ]
-  //: supply1 g25 (w2) @(242,340) /sn:0 /w:[ 1 ]
+  register g15 (.Q(R0), .D(AIN), .EN(w8), .CLR(w2), .CK(ck));   //: @(180,362) /sn:0 /w:[ 3 7 0 0 13 ]
   //: joint g17 (R3) @(611, 447) /w:[ 2 -1 1 4 ]
-  mux g14 (.I0(R0), .I1(R1), .I2(R2), .I3(R3), .S(SA), .Z(AOUT));   //: @(405,507) /sn:0 /w:[ 5 0 0 0 0 0 ]
+  //: supply1 g25 (w2) @(242,340) /sn:0 /w:[ 1 ]
   //: input g5 (ck) @(49,392) /sn:0 /w:[ 0 ]
-  //: joint g24 (AIN) @(467, 287) /w:[ 2 1 4 10 ]
+  mux g14 (.I0(R0), .I1(R1), .I2(R2), .I3(R3), .S(SA), .Z(AOUT));   //: @(405,507) /sn:0 /w:[ 5 0 0 0 0 0 ]
   //: supply1 g21 (w0) @(384,339) /sn:0 /w:[ 1 ]
+  //: joint g24 (AIN) @(467, 287) /w:[ 2 1 4 10 ]
   //: supply1 g23 (w5) @(691,340) /sn:0 /w:[ 1 ]
-  //: joint g26 (AIN) @(322, 287) /w:[ 5 -1 6 8 ]
-  //: supply1 g22 (w1) @(530,339) /sn:0 /w:[ 1 ]
   //: input g0 (AIN) @(467,142) /sn:0 /R:3 /w:[ 0 ]
-  //: joint g18 (R2) @(467, 432) /w:[ 2 4 1 -1 ]
+  //: supply1 g22 (w1) @(530,339) /sn:0 /w:[ 1 ]
+  //: joint g26 (AIN) @(322, 287) /w:[ 5 -1 6 8 ]
   //: joint g12 (ck) @(414, 392) /w:[ 6 8 5 -1 ]
+  //: joint g18 (R2) @(467, 432) /w:[ 2 4 1 -1 ]
 
 endmodule
 
 module main;    //: root_module
 supply1 w7;    //: /sn:0 {0}(256,325)(256,312){1}
+supply0 w11;    //: /sn:0 {0}(496,395)(496,373)(480,373){1}
+supply0 w12;    //: /sn:0 {0}(478,499)(478,481)(460,481){1}
 supply0 w2;    //: /sn:0 {0}(295,430)(295,445)(258,445){1}
 //: {2}(254,445)(223,445)(223,429){3}
 //: {4}(256,447)(256,457){5}
-supply0 w12;    //: /sn:0 {0}(478,499)(478,481)(460,481){1}
-supply0 w11;    //: /sn:0 {0}(496,395)(496,373)(480,373){1}
 supply1 w10;    //: /sn:0 /dp:1 {0}(460,471)(474,471)(474,453){1}
 supply0 [7:0] w9;    //: /sn:0 {0}(387,395)(387,385)(411,385)(411,414){1}
 wire [7:0] w6;    //: /sn:0 /dp:1 {0}(711,277)(711,265)(771,265){1}
@@ -131,168 +132,165 @@
 wire b;    //: /dp:1 {0}(62,491)(98,491){1}
 //: {2}(100,489)(100,461)(116,461){3}
 //: {4}(100,493)(100,537)(128,537){5}
-wire [7:0] w14;    //: /sn:0 /dp:1 {0}(440,359)(440,332){1}
 wire [7:0] w16;    //: /sn:0 {0}(431,414)(431,400)(456,400)(456,388){1}
+wire [7:0] w14;    //: /sn:0 /dp:1 {0}(440,359)(440,332){1}
+wire w19;    //: /sn:0 /dp:1 {0}(398,430)(361,430){1}
+wire w4;    //: /sn:0 /dp:1 {0}(194,535)(149,535){1}
 wire [7:0] w15;    //: /sn:0 /dp:1 {0}(748,364)(748,335)(771,335){1}
 //: {2}(775,335)(798,335)(798,552)(758,552)(758,538){3}
 //: {4}(773,333)(773,320){5}
-wire w4;    //: /sn:0 /dp:1 {0}(194,535)(149,535){1}
-wire w19;    //: /sn:0 /dp:1 {0}(398,430)(361,430){1}
+wire w3;    //: /sn:0 /dp:1 {0}(194,530)(164,530)(164,514){1}
 wire w0;    //: /sn:0 {0}(157,333)(229,333){1}
 //: {2}(233,333)(242,333){3}
 //: {4}(231,331)(231,323){5}
 //: {6}(231,335)(231,421)(281,421){7}
-wire w3;    //: /sn:0 /dp:1 {0}(194,530)(164,530)(164,514){1}
 wire [1:0] w21;    //: /sn:0 {0}(635,453)(652,453)(652,471)(678,471){1}
 wire w28;    //: /sn:0 {0}(637,293)(678,293){1}
-wire [1:0] w23;    //: /sn:0 /dp:1 {0}(678,491)(651,491)(651,503)(637,503){1}
 wire [7:0] w20;    //: /sn:0 {0}(421,443)(421,466){1}
+wire [1:0] w23;    //: /sn:0 /dp:1 {0}(678,491)(651,491)(651,503)(637,503){1}
 wire w1;    //: /sn:0 {0}(156,367)(192,367){1}
 //: {2}(196,367)(242,367){3}
 //: {4}(194,365)(194,358){5}
 //: {6}(194,369)(194,420)(209,420){7}
 wire co;    //: /dp:1 {0}(215,533)(228,533)(228,522){1}
 wire [7:0] w25;    //: /sn:0 {0}(658,233)(691,233)(691,277){1}
+wire w18;    //: /sn:0 {0}(432,373)(422,373){1}
 wire w8;    //: /sn:0 /dp:1 {0}(191,459)(163,459){1}
 //: {2}(159,459)(137,459){3}
 //: {4}(161,461)(161,493){5}
-wire w18;    //: /sn:0 {0}(432,373)(422,373){1}
 wire [7:0] w17;    //: /sn:0 {0}(725,456)(725,431){1}
+wire clk;    //: {0}(323,476)(384,476){1}
 wire ci;    //: {0}(62,420)(166,420)(166,462){1}
 //: {2}(168,464)(191,464){3}
 //: {4}(166,466)(166,493){5}
-wire clk;    //: {0}(323,476)(384,476){1}
+wire ck;    //: /sn:0 {0}(611,559)(665,559)(665,513)(678,513){1}
+wire s;    //: /dp:1 {0}(212,462)(230,462)(230,479){1}
 wire w13;    //: /sn:0 {0}(223,412)(223,393)(254,393){1}
 //: {2}(258,393)(293,393){3}
 //: {4}(295,391)(295,381){5}
 //: {6}(295,395)(295,413){7}
 //: {8}(256,391)(256,376){9}
-wire s;    //: /dp:1 {0}(212,462)(230,462)(230,479){1}
-wire ck;    //: /sn:0 {0}(611,559)(665,559)(665,513)(678,513){1}
-wire w5;    //: /sn:0 /dp:1 {0}(256,342)(256,359){1}
 wire a;    //: {0}(62,456)(104,456){1}
 //: {2}(108,456)(116,456){3}
 //: {4}(106,458)(106,532)(128,532){5}
-wire [7:0] w29;    //: /sn:0 /dp:1 {0}(701,364)(701,306){1}
+wire w5;    //: /sn:0 /dp:1 {0}(256,342)(256,359){1}
 wire [7:0] reg_out;    //: {0}(421,487)(421,523)(466,523){1}
 //: {2}(470,523)(520,523)(520,348)(472,348)(472,359){3}
 //: {4}(468,525)(468,542){5}
+wire [7:0] w29;    //: /sn:0 /dp:1 {0}(701,364)(701,306){1}
 wire [1:0] w26;    //: /sn:0 /dp:1 {0}(637,356)(658,356)(658,393)(680,393){1}
 //: enddecls
 
+  //: supply1 g8 (w7) @(267,312) /sn:0 /w:[ 1 ]
+  nmos g4 (.Z(w2), .S0(w13), .G0(w0));   //: @(289,421) /sn:0 /w:[ 0 7 7 ]
   //: comment g61 /dolink:1 /link:"@T/edwire_tut.v" @(622,100) /sn:0 /anc:1
   //: /line:"ワイヤの編集"
   //: /end
-  nmos g4 (.Z(w2), .S0(w13), .G0(w0));   //: @(289,421) /sn:0 /w:[ 0 7 7 ]
-  //: supply1 g8 (w7) @(267,312) /sn:0 /w:[ 1 ]
-  clock g51 (.Z(ck));   //: @(598,559) /sn:0 /w:[ 0 ] /omega:200 /phi:0 /duty:50
-  //: switch g3 (w0) @(140,333) /sn:0 /w:[ 0 ] /st:0
-  //: joint g13 (w1) @(194, 367) /w:[ 2 4 1 6 ]
-  or g34 (.I0(w3), .I1(w4), .Z(co));   //: @(205,533) /sn:0 /w:[ 0 0 0 ]
-  clock g37 (.Z(clk));   //: @(310,476) /sn:0 /w:[ 0 ] /omega:120 /phi:0 /duty:50
-  led g55 (.I(w0));   //: @(231,316) /sn:0 /w:[ 5 ] /type:0
   //: joint g58 (w6) @(773, 265) /w:[ 2 4 1 -1 ]
-  //: comment g2 /dolink:0 /link:"" @(-2,217) /sn:0 /anc:1
-  //: /line:""
-  //: /end
+  led g55 (.I(w0));   //: @(231,316) /sn:0 /w:[ 5 ] /type:0
+  clock g37 (.Z(clk));   //: @(310,476) /sn:0 /w:[ 0 ] /omega:120 /phi:0 /duty:50
+  or g34 (.I0(w3), .I1(w4), .Z(co));   //: @(205,533) /sn:0 /w:[ 0 0 0 ]
+  //: joint g13 (w1) @(194, 367) /w:[ 2 4 1 6 ]
+  //: switch g3 (w0) @(140,333) /sn:0 /w:[ 0 ] /st:0
+  clock g51 (.Z(ck));   //: @(598,559) /sn:0 /w:[ 0 ] /omega:200 /phi:0 /duty:50
   led g59 (.I(w15));   //: @(773,313) /sn:0 /w:[ 5 ] /type:2
   //: comment g1 /dolink:1 /link:"@T/edit1_tut.v" @(622,70) /sn:0 /anc:1
   //: /line:"基本の編集モード"
   //: /end
+  //: joint g11 (w2) @(256, 445) /w:[ 1 -1 2 4 ]
   //: comment g16 /dolink:1 /link:"@T/module_tut.v" @(622,115) /sn:0 /anc:1
   //: /line:"モジュールの使い方"
   //: /end
-  //: joint g11 (w2) @(256, 445) /w:[ 1 -1 2 4 ]
-  led g50 (.I(co));   //: @(228,515) /sn:0 /w:[ 1 ] /type:0
-  //: supply0 g10 (w2) @(256,463) /sn:0 /w:[ 5 ]
   and g28 (.I0(a), .I1(b), .Z(w4));   //: @(139,535) /sn:0 /w:[ 5 5 1 ]
-  //: joint g19 (w0) @(231, 333) /w:[ 2 4 1 6 ]
-  //: switch g27 (ci) @(45,420) /sn:0 /w:[ 0 ] /st:0
+  //: supply0 g10 (w2) @(256,463) /sn:0 /w:[ 5 ]
+  led g50 (.I(co));   //: @(228,515) /sn:0 /w:[ 1 ] /type:0
   //: joint g32 (w8) @(161, 459) /w:[ 1 -1 2 4 ]
-  //: dip sa (w21) @(597,453) /R:1 /w:[ 0 ] /st:0
-  nmos g6 (.Z(w2), .S0(w13), .G0(w1));   //: @(217,420) /sn:0 /w:[ 3 0 7 ]
+  //: switch g27 (ci) @(45,420) /sn:0 /w:[ 0 ] /st:0
+  //: joint g19 (w0) @(231, 333) /w:[ 2 4 1 6 ]
   //: supply0 g38 (w12) @(478,505) /sn:0 /w:[ 0 ]
-  mux g53 (.I0(w25), .I1(w6), .S(w28), .Z(w29));   //: @(701,293) /sn:0 /w:[ 1 0 1 1 ]
-  pmos g7 (.Z(w7), .S0(w5), .G0(w0));   //: @(250,333) /sn:0 /w:[ 0 0 3 ]
-  //: joint g9 (w13) @(256, 393) /w:[ 2 8 1 -1 ]
+  nmos g6 (.Z(w2), .S0(w13), .G0(w1));   //: @(217,420) /sn:0 /w:[ 3 0 7 ]
+  //: dip sa (w21) @(597,453) /R:1 /w:[ 0 ] /st:0
   //: joint g57 (w13) @(295, 393) /w:[ -1 4 3 6 ]
-  //: frame g15 @(600,25) /sn:0 /anc:1 /wi:224 /ht:149 /tx:"チュートリアル"
+  //: joint g9 (w13) @(256, 393) /w:[ 2 8 1 -1 ]
+  pmos g7 (.Z(w7), .S0(w5), .G0(w0));   //: @(250,333) /sn:0 /w:[ 0 0 3 ]
+  mux g53 (.I0(w25), .I1(w6), .S(w28), .Z(w29));   //: @(701,293) /sn:0 /w:[ 1 0 1 1 ]
+  and g31 (.I0(w8), .I1(ci), .Z(w3));   //: @(164,504) /sn:0 /R:3 /tech:unit /w:[ 5 5 1 ]
   //: comment g20 /dolink:1 /link:"@T/create_tut.v" @(622,55) /sn:0 /anc:1
   //: /line:"回路の作成"
   //: /end
-  and g31 (.I0(w8), .I1(ci), .Z(w3));   //: @(164,504) /sn:0 /R:3 /w:[ 5 5 1 ]
+  //: frame g15 @(600,25) /sn:0 /anc:1 /wi:224 /ht:149 /tx:"チュートリアル"
   //: dip sb (w23) @(599,503) /R:1 /w:[ 1 ] /st:1
   //: supply1 g39 (w10) @(485,453) /sn:0 /w:[ 1 ]
-  led g48 (.I(reg_out));   //: @(468,549) /sn:0 /R:2 /w:[ 5 ] /type:1
-  //: dip func (w26) @(599,356) /R:1 /w:[ 0 ] /st:0
-  //: supply0 g43 (w11) @(496,401) /sn:0 /w:[ 0 ]
-  REGS regs (.AIN(w17), .ck(ck), .SB(w23), .SA(w21), .AOUT(w6), .BOUT(w15));   //: @(679, 457) /sz:(94, 80) /p:[ Ti0>0 Li0>1 Li1>0 Li2>1 Bi0>3 Bi1>3 ]
   ALU alu (.B(w15), .A(w29), .F(w26), .Z(w17));   //: @(681, 365) /sz:(89, 65) /p:[ Ti0>0 Ti1>0 Li0>1 Bo0<1 ]
+  REGS regs (.AIN(w17), .ck(ck), .SB(w23), .SA(w21), .BOUT(w15), .AOUT(w6));   //: @(679, 457) /sz:(94, 80) /p:[ Ti0>0 Li0>1 Li1>0 Li2>1 Bo0<3 Bo1<3 ]
+  //: supply0 g43 (w11) @(496,401) /sn:0 /w:[ 0 ]
+  //: dip func (w26) @(599,356) /R:1 /w:[ 0 ] /st:0
+  led g48 (.I(reg_out));   //: @(468,549) /sn:0 /R:2 /w:[ 5 ] /type:1
+  led g62 (.I(w6));   //: @(773,245) /sn:0 /w:[ 5 ] /type:2
+  //: joint g29 (a) @(106, 456) /w:[ 2 -1 1 4 ]
+  //: switch g25 (a) @(45,456) /sn:0 /w:[ 0 ] /st:0
   //: comment g17 /dolink:1 /link:"@T/sim_tut.v" @(622,130) /sn:0 /anc:1
   //: /line:"組合せ回路のシミュレーション"
   //: /end
-  //: switch g25 (a) @(45,456) /sn:0 /w:[ 0 ] /st:0
-  //: joint g29 (a) @(106, 456) /w:[ 2 -1 1 4 ]
-  led g62 (.I(w6));   //: @(773,245) /sn:0 /w:[ 5 ] /type:2
-  led g52 (.I(s));   //: @(230,486) /sn:0 /R:2 /w:[ 1 ] /type:0
   //: dip g42 (w14) @(440,322) /sn:0 /w:[ 1 ] /st:248
+  led g52 (.I(s));   //: @(230,486) /sn:0 /R:2 /w:[ 1 ] /type:0
   //: dip data_in (w25) @(620,233) /R:1 /w:[ 0 ] /st:11
-  pmos g5 (.Z(w5), .S0(w13), .G0(w1));   //: @(250,367) /sn:0 /w:[ 1 9 3 ]
-  led g56 (.I(w13));   //: @(295,374) /sn:0 /w:[ 5 ] /type:0
   //: comment g14 /dolink:1 /link:"@T/seqsim_tut.v" @(622,145) /sn:0 /anc:1
   //: /line:"順序回路のシミュレーション"
   //: /end
-  //: switch g44 (w19) @(344,430) /sn:0 /w:[ 1 ] /st:1
+  led g56 (.I(w13));   //: @(295,374) /sn:0 /w:[ 5 ] /type:0
+  pmos g5 (.Z(w5), .S0(w13), .G0(w1));   //: @(250,367) /sn:0 /w:[ 1 9 3 ]
   //: comment g47 /dolink:0 /link:"" @(106,285) /sn:0
-  //: /line:"にクリックして、始めよう!"
+  //: /line:"をクリックして、始めよう!"
   //: /end
+  //: switch g44 (w19) @(344,430) /sn:0 /w:[ 1 ] /st:1
+  add g36 (.A(w14), .B(reg_out), .S(w16), .CI(w11), .CO(w18));   //: @(456,375) /sn:0 /w:[ 0 3 1 1 0 ]
+  xor g24 (.I0(w8), .I1(ci), .Z(s));   //: @(202,462) /sn:0 /w:[ 0 3 0 ]
   //: comment g21 /dolink:1 /link:"@T/edit2_tut.v" @(622,85) /sn:0 /anc:1
   //: /line:"グループの編集機能"
   //: /end
-  xor g24 (.I0(w8), .I1(ci), .Z(s));   //: @(202,462) /sn:0 /w:[ 0 3 0 ]
-  add g36 (.A(w14), .B(reg_out), .S(w16), .CI(w11), .CO(w18));   //: @(456,375) /sn:0 /w:[ 0 3 1 1 0 ]
-  xor g23 (.I0(a), .I1(b), .Z(w8));   //: @(127,459) /sn:0 /w:[ 3 3 3 ]
   //: supply0 g41 (w9) @(387,401) /sn:0 /w:[ 0 ]
-  mux g40 (.I0(w9), .I1(w16), .S(w19), .Z(w20));   //: @(421,430) /sn:0 /w:[ 1 0 0 0 ]
-  led g54 (.I(w1));   //: @(194,351) /sn:0 /w:[ 5 ] /type:0
+  xor g23 (.I0(a), .I1(b), .Z(w8));   //: @(127,459) /sn:0 /w:[ 3 3 3 ]
   //: joint g60 (w15) @(773, 335) /w:[ 2 4 1 -1 ]
-  //: switch data_select (w28) @(620,293) /w:[ 0 ] /st:0
-  //: comment g22 /dolink:0 /link:"" @(605,40) /sn:0 /anc:1
-  //: /line:"->"
+  led g54 (.I(w1));   //: @(194,351) /sn:0 /w:[ 5 ] /type:0
+  mux g40 (.I0(w9), .I1(w16), .S(w19), .Z(w20));   //: @(421,430) /sn:0 /w:[ 1 0 0 0 ]
+  //: comment g46 /dolink:1 /link:"@T/create_tut.v" @(11,285) /sn:0
+  //: /line:"「回路の作成」"
   //: /end
+  register g35 (.Q(reg_out), .D(w20), .EN(w12), .CLR(w10), .CK(clk));   //: @(421,476) /sn:0 /w:[ 0 1 1 0 1 ]
+  //: switch g26 (b) @(45,491) /sn:0 /w:[ 0 ] /st:1
   //: comment g0 /dolink:0 /link:"" @(13,16) /anc:1
-  //: /line:"TkGateへようこそ!"
+  //: /line:"TkGate <value-of name=\"tkgate-version\">へようこそ!"
   //: /line:""
-  //: /line:"Copyright (C) 1987-2000 by Jeffery P. Hansen"
+  //: /line:"<value-of name=\"tkgate-copyright\">"
   //: /line:"  TKGate comes with ABSOLUTELY NO WARRANTY; see 'Help...License' menu"
-  //: /line:"  for license and warranty details.  Report problems to hansen@cmu.edu"
+  //: /line:"  for license and warranty details.  Report problems to <value-of name=\"tkgate-mailcontact\">"
   //: /line:""
-  //: /line:"   * TkGateを実行する時、このチュートリアルが自動的に読み込まれる。この機能を"
-  //: /line:"     無効にする為、「ファイル」の下の「設定...」を選択して、「初心者モード」の"
+  //: /line:"   * TkGateを実行する時、このチュートリアルが自動的に読み込まれます。この機能を"
+  //: /line:"     無効にするには、「ファイル」の下の「設定...」を選択して、「初心者モード」の"
   //: /line:"     チェック・ボタンを消して下さい。"
   //: /line:""
-  //: /line:"    * 新しい回路を作成する為、「ファイル」の下の「新規作成」を選択してください。"
-  //: /line:"     存在している回路を編集する為、「ファイル」の下の「開く...」を選択してください。"
+  //: /line:"   * 新しい回路を作成するには、「ファイル」の下の「新規作成」を選択して下さい。"
+  //: /line:"     存在している回路を編集するには、「ファイル」の下の「開く...」を選択して下さい。"
   //: /line:""
-  //: /line:"   * TkGateのドキュメントなどはhttp://www.cs.cmu.edu/~hansen/tkgateにある。"
+  //: /line:"   * TkGateのドキュメントなどは<value-of name=\"tkgate-homepage\">にある。"
   //: /line:""
   //: /line:"このチュートリアルでは、TkGateの基本の使い方を紹介する。右上のハイパーリンク、"
-  //: /line:"又は、「ヘルプ」のメニューで、それぞれのチュートリアルを見てください。"
+  //: /line:"又は、「ヘルプ」のメニューで、それぞれのチュートリアルを見て下さい。"
   //: /line:"各チュートリアルの説明に従って、TkGateの操作を練習して下さい。全てのチュート"
-  //: /line:"リアルを終ったら、「ヘルプ」メニューで幾つかの例題回路も読み込められる。"
+  //: /line:"リアルを終ったら、「ヘルプ」メニューで幾つかの例題回路も読み込めます。"
   //: /line:""
   //: /end
-  //: switch g26 (b) @(45,491) /sn:0 /w:[ 0 ] /st:1
-  register g35 (.Q(reg_out), .D(w20), .EN(w12), .CLR(w10), .CK(clk));   //: @(421,476) /sn:0 /w:[ 0 1 1 0 1 ]
-  //: comment g46 /dolink:1 /link:"@T/create_tut.v" @(11,285) /sn:0
-  //: /line:"「回路の作成」"
+  //: comment g22 /dolink:0 /link:"" @(605,40) /sn:0 /anc:1
+  //: /line:"->"
   //: /end
+  //: switch data_select (w28) @(620,293) /w:[ 0 ] /st:0
+  //: switch g12 (w1) @(139,367) /sn:0 /w:[ 0 ] /st:0
   //: comment g18 /dolink:0 /link:"@T/welcome_tut.v" @(622,40) /sn:0 /anc:1
   //: /line:"TkGateの紹介"
   //: /end
-  //: switch g12 (w1) @(139,367) /sn:0 /w:[ 0 ] /st:0
-  //: joint g30 (b) @(100, 491) /w:[ -1 2 1 4 ]
   //: joint g33 (ci) @(166, 464) /w:[ 2 1 -1 4 ]
+  //: joint g30 (b) @(100, 491) /w:[ -1 2 1 4 ]
   //: joint g49 (reg_out) @(468, 523) /w:[ 2 -1 1 4 ]
 
 endmodule
--- tkgate-1.8.7/Imakefile.orig	2004-06-09 18:16:06.000000000 +0100
+++ tkgate-1.8.7/Imakefile	2007-02-02 10:55:01.000000000 +0000
@@ -99,4 +99,4 @@
 	@echo "All files from ${HOMEDIR}"
 
 install::
-	cp -r locale ${HOMEDIR}
+	cp -r locale ${DESTDIR}${HOMEDIR}
