--- tkgate-1.8.7/config.h.orig	2007-01-30 03:45:09.000000000 +0000
+++ tkgate-1.8.7/config.h	2007-07-19 22:45:01.000000000 +0100
@@ -52,13 +52,13 @@
   double quotes and should be all one one line.  This list is used to
   find X11, tcl and tk.
 */
-#define TKGATE_INCDIRS		"/usr/X11R6/include /usr/X11/include /pkgs/include /usr/local/include /usr/openwin/include /usr/X/include /usr/include /sw/include"
+#define TKGATE_INCDIRS		"/pkgs/include /usr/local/include /usr/openwin/include /usr/X/include /usr/include /sw/include"
 
 /*
   List of directories to search for tcl library.  The list must be in 
   double quotes and should be all one one line.
 */
-#define TKGATE_LIBDIRS		"/usr/X11R6/lib /usr/X11/lib /pkgs/lib /usr/local/lib /usr/lib /sw/lib"
+#define TKGATE_LIBDIRS		"/pkgs/lib /usr/local/lib /usr/lib /sw/lib"
 
 /*
   Required libraries.  TkGate will not compile unless these libraries are found.
@@ -79,19 +79,19 @@
   version number to this path.  You can override the default by
   uncommenting the defintion of TKGATE_HOMEDIR below.
  */
-#define TKGATE_HOMEDIRBASE	"/usr/X11R6/lib"
+#define TKGATE_HOMEDIRBASE	"/usr/lib"
 
 /*
   The location of the tkgate and gmac executables.  If you change this, you will
   need to run the configure script again.
  */
-#define TKGATE_BINDIR		/usr/X11R6/bin
+#define TKGATE_BINDIR		/usr/bin
 
 /*
   The location of the tkgate and gmac man pages.  If you change this, you will
   need to run the configure script again.
  */
-#define TKGATE_MANDIR		/usr/X11R6/man/man1
+#define TKGATE_MANDIR		/usr/man/man1
 
 /*
   Digital circuit editor and simulator.
@@ -101,7 +101,7 @@
 /*
   URL for local documentation
  */
-#define TKGATE_LOCALDOC		"file:/usr/X11R6/lib/tkgate-" TKGATE_VERSION "/doc/index.html"
+#define TKGATE_LOCALDOC		"file:/usr/lib/tkgate-" TKGATE_VERSION "/doc/index.html"
 
 /*
   The TkGate Copyright message.
@@ -200,13 +200,13 @@
 /*
  * Set this to the list of library directories to search
  *
- * #define TKGATE_LPATH		-L/usr/local/lib -L/usr/X11R6/lib -L/usr/lib
+ * #define TKGATE_LPATH		-L/usr/local/lib -L/usr/lib
  */
 
 /*
  * Set this to the list of include file directories to search
  *
- * #define TKGATE_IPATH		-I/usr/local/include/tk8.0 -I/usr/local/include/tcl8.0 -I/usr/X11R6/include
+ * #define TKGATE_IPATH		-I/usr/local/include/tk8.0 -I/usr/local/include/tcl8.0 -I/usr/include
  */
 
 /*
@@ -323,17 +323,17 @@
 #endif
 
 #ifndef TKGATE_LPATH
-#define TKGATE_LPATH		 -L/usr/local/lib -L/usr/X11R6/lib -L/usr/lib
+#define TKGATE_LPATH		 -L/usr/local/lib -L/usr/lib
 #endif
 #ifndef TKGATE_IPATH
-#define TKGATE_IPATH		 -I/usr/local/include/tk8.4 -I/usr/local/include/tcl8.4 -I/usr/X11R6/include
+#define TKGATE_IPATH		 -I/usr/local/include/tk8.4 -I/usr/local/include/tcl8.4 -I/usr/include
 #endif
 #ifndef TKGATE_LIBS
 #define TKGATE_LIBS		 -ltk84 -ltcl84 -lX11 -lm
 #endif
 
 #ifndef TKGATE_HOMEDIR
-#define TKGATE_HOMEDIR		"/usr/X11R6/lib/tkgate-1.8.7"
+#define TKGATE_HOMEDIR		"/usr/lib/tkgate-1.8.7"
 #endif
 
 #ifndef TKGATE_SECONDARYHOME
--- tkgate-1.8.7/Imakefile.orig	2004-06-09 18:16:06.000000000 +0100
+++ tkgate-1.8.7/Imakefile	2007-02-02 10:55:01.000000000 +0000
@@ -99,4 +99,4 @@
 	@echo "All files from ${HOMEDIR}"
 
 install::
-	cp -r locale ${HOMEDIR}
+	cp -r locale ${DESTDIR}${HOMEDIR}
