--- TLP-1.10.2/tlp-pd.orig	2026-10-05 09:06:56.823123777 +0100
+++ TLP-1.10.2/tlp-pd	2026-10-06 19:42:38.480808375 +0100
@@ -547,7 +547,7 @@
 
         self._logger.info(f"Run detached TLP to apply profile '{profile}'")
         _ = subprocess.Popen(
-            ["tlp", f"{profile}"],
+            ["/usr/sbin/tlp", f"{profile}"],
             start_new_session=True,
             stdin=subprocess.DEVNULL,
             stdout=subprocess.DEVNULL,
