--- tttrigger-1.1/src/playlist.cpp.orig	2007-10-08 21:27:44.000000000 +0100
+++ tttrigger-1.1/src/playlist.cpp	2007-10-11 13:43:57.000000000 +0100
@@ -386,7 +386,7 @@
 		entry = (PlaylistEntry *) n;
 
 		dialog->setSelection(entry->filename);
-		dialog->setFilter("Sounds (*.wav)");
+		dialog->setFilter("Sounds (*.wav *.ogg *.flac)");
 		dialog->setCaption("Please select file...");
 		dialog->setMode(QFileDialog::ExistingFile);
 
@@ -417,7 +417,7 @@
 		if((doInsert = (n != 0)))
 			insertId = n->index;
 
-		dialog->setFilter("Sounds (*.wav)");
+		dialog->setFilter("Sounds (*.wav *.ogg *.flac)");
 		dialog->setCaption("Please select one or more files...");
 		dialog->setMode(QFileDialog::ExistingFiles);
 
