--- udev-177/rules/50-udev-default.rules.orig	2012-01-10 00:43:22.000000000 +0000
+++ udev-177/rules/50-udev-default.rules	2012-01-15 08:34:20.000000000 +0000
@@ -8,9 +8,9 @@
 KERNEL=="vcs|vcs[0-9]*|vcsa|vcsa[0-9]*", GROUP="tty"
 
 # serial
-KERNEL=="tty[A-Z]*[0-9]|pppox[0-9]*|ircomm[0-9]*|noz[0-9]*|rfcomm[0-9]*", GROUP="dialout"
-KERNEL=="mwave", GROUP="dialout"
-KERNEL=="hvc*|hvsi*", GROUP="dialout"
+KERNEL=="tty[A-Z]*[0-9]|pppox[0-9]*|ircomm[0-9]*|noz[0-9]*|rfcomm[0-9]*", GROUP="uucp"
+KERNEL=="mwave", GROUP="uucp"
+KERNEL=="hvc*|hvsi*", GROUP="uucp"
 
 # virtio serial / console ports
 KERNEL=="vport*", ATTR{name}=="?*", SYMLINK+="virtio-ports/$attr{name}"
