--- udev-165/extras/ata_id/ata_id.c.orig	2010-11-10 00:30:53.000000000 +0000
+++ udev-165/extras/ata_id/ata_id.c	2011-01-02 23:05:06.000000000 +0000
@@ -400,8 +400,10 @@
 	if (peripheral_device_type == 0x05)
 	  {
 	    is_packet_device = 1;
+#if 0
 	    ret = disk_identify_packet_device_command(fd, out_identify, 512);
 	    goto check_nul_bytes;
+#endif
 	  }
 	if (peripheral_device_type != 0x00) {
 		ret = -1;
