diff -u -r udev-141.orig/rules/gentoo/40-gentoo.rules udev-141/rules/gentoo/40-gentoo.rules
--- udev-141.orig/rules/gentoo/40-gentoo.rules	2009-01-27 04:55:41.000000000 -0600
+++ udev-141/rules/gentoo/40-gentoo.rules	2009-04-21 13:52:48.000000000 -0500
@@ -15,7 +15,7 @@
 SUBSYSTEM=="usb", ENV{DEVTYPE}=="usb_device", GROUP="usb"
 
 # dialout devices
-KERNEL=="ippp*|isdn*|dcbri*|rfcomm*|ttyACM[0-9]*|pilot", GROUP="dialout"
+KERNEL=="ippp*|isdn*|dcbri*|rfcomm*|ttyACM[0-9]*|pilot", GROUP="uucp"
 
 # sound devices (oss)
 SUBSYSTEM=="snd",	GROUP="audio"
diff -u -r udev-141.orig/rules/rules.d/50-udev-default.rules udev-141/rules/rules.d/50-udev-default.rules
--- udev-155/rules/rules.d/50-udev-default.rules.orig	2010-05-20 14:34:43.000000000 +0100
+++ udev-155/rules/rules.d/50-udev-default.rules	2010-05-24 21:44:19.000000000 +0100
@@ -11,9 +11,9 @@
 KERNEL=="vcs|vcs[0-9]*|vcsa|vcsa[0-9]*", GROUP="tty"
 
 # serial
-KERNEL=="tty[A-Z]*[0-9]|pppox[0-9]*|ircomm[0-9]*|noz[0-9]*|rfcomm[0-9]*", GROUP="dialout"
-KERNEL=="mwave",		GROUP="dialout"
-KERNEL=="hvc*|hvsi*",		GROUP="dialout"
+KERNEL=="tty[A-Z]*[0-9]|pppox[0-9]*|ircomm[0-9]*|noz[0-9]*|rfcomm[0-9]*", GROUP="uucp"
+KERNEL=="mwave",		GROUP="uucp"
+KERNEL=="hvc*|hvsi*",		GROUP="uucp"
 
 # virtio serial / console ports
 KERNEL=="vport*", ATTR{name}=="?*", SYMLINK+="virtio-ports/$attr{name}"
