--- uget-2.0.8/ui-gtk/UgtkApp.c.orig	2016-05-15 22:11:58.000000000 +0100
+++ uget-2.0.8/ui-gtk/UgtkApp.c	2016-05-16 20:56:26.214604089 +0100
@@ -987,6 +987,11 @@
 			gdk_display_get_device_manager (gdk_window_get_display (gdk_win)));
 	gdk_window_get_device_position (gdk_win, dev_pointer, NULL, NULL, &mask);
 
+	if (!app || !app->traveler.download.cursor.node) {
+		printf("app or node is NULL\n");
+		return;
+	}
+
 	cursor = app->traveler.download.cursor.node->data;
 	list = ugtk_traveler_get_selected (&app->traveler);
 	for (link = list;  link;  link = link->next) {
