--- ugrep-3.12.1/Makefile.in.orig	2023-06-04 18:47:56.000000000 +0100
+++ ugrep-3.12.1/Makefile.in	2023-06-04 21:26:19.676454835 +0100
@@ -908,7 +908,7 @@
 		@echo
 
 install-exec-hook:
-		cp -f $(top_builddir)/bin/ugrep+ $(top_builddir)/bin/ug+ $(DESTDIR)$(bindir)
+		cp -f $(top_srcdir)/bin/ugrep+ $(top_srcdir)/bin/ug+ $(DESTDIR)$(bindir)
 		rm -f $(DESTDIR)$(bindir)/ug$(EXEEXT)
 		cd $(DESTDIR)$(bindir) && cp -f ugrep$(EXEEXT) ug$(EXEEXT)
 
