--- ugrep-5.1.3/Makefile.in.orig	2024-04-08 16:43:04.265115160 +0100
+++ ugrep-5.1.3/Makefile.in	2024-04-08 16:46:02.991127160 +0100
@@ -1017,7 +1017,7 @@
 		@echo
 
 install-exec-hook:
-		@cp -f $(top_builddir)/bin/ugrep+ $(top_builddir)/bin/ug+ $(DESTDIR)$(bindir)
+		@cp -f $(top_srcdir)/bin/ugrep+ $(top_srcdir)/bin/ug+ $(DESTDIR)$(bindir)
 		@rm -f $(DESTDIR)$(bindir)/ug$(EXEEXT)
 		@cd $(DESTDIR)$(bindir) && cp -f ugrep$(EXEEXT) ug$(EXEEXT)
 
--- ugrep-5.1.3/src/Makefile.in.orig	2024-04-08 16:43:04.266115160 +0100
+++ ugrep-5.1.3/src/Makefile.in	2024-04-08 16:47:54.602134653 +0100
@@ -328,6 +328,7 @@
 top_srcdir = @top_srcdir@
 ugrep_CPPFLAGS = \
         -I$(top_srcdir)/include \
+        -I$(top_srcdir)/lzma/C \
         $(EXTRA_CFLAGS) \
         $(SIMD_FLAGS) \
         $(PTHREAD_CFLAGS) \
