--- ugrep-4.4.0/Makefile.in.orig	2023-12-18 21:41:11.000000000 +0000
+++ ugrep-4.4.0/Makefile.in	2023-12-19 16:20:44.070972782 +0000
@@ -1013,7 +1013,7 @@
 		@echo
 
 install-exec-hook:
-		cp -f $(top_builddir)/bin/ugrep+ $(top_builddir)/bin/ug+ $(DESTDIR)$(bindir)
+		cp -f $(top_srcdir)/bin/ugrep+ $(top_srcdir)/bin/ug+ $(DESTDIR)$(bindir)
 		rm -f $(DESTDIR)$(bindir)/ug$(EXEEXT)
 		cd $(DESTDIR)$(bindir) && cp -f ugrep$(EXEEXT) ug$(EXEEXT)
 
