--- ugrep-6.0.0/Makefile.in.orig	2024-05-06 21:32:01.000000000 +0100
+++ ugrep-6.0.0/Makefile.in	2024-05-08 16:21:30.632167754 +0100
@@ -1016,7 +1016,7 @@
 		@echo
 
 install-exec-hook:
-		@cp -f $(top_builddir)/bin/ugrep+ $(top_builddir)/bin/ug+ $(DESTDIR)$(bindir)
+		@cp -f $(top_srcdir)/bin/ugrep+ $(top_srcdir)/bin/ug+ $(DESTDIR)$(bindir)
 		@rm -f $(DESTDIR)$(bindir)/ug$(EXEEXT)
 		@cd $(DESTDIR)$(bindir) && cp -f ugrep$(EXEEXT) ug$(EXEEXT)
 
--- ugrep-6.0.0/src/Makefile.in.orig	2024-05-08 16:21:30.632167754 +0100
+++ ugrep-6.0.0/src/Makefile.in	2024-05-08 16:31:44.464208966 +0100
@@ -337,6 +337,7 @@
 top_srcdir = @top_srcdir@
 ugrep_CPPFLAGS = \
 	-I$(top_srcdir)/include \
+	-I$(top_srcdir)/lzma/C \
 	$(EXTRA_CFLAGS) \
 	$(SIMD_FLAGS) \
 	$(PTHREAD_CFLAGS) \
@@ -391,6 +392,7 @@
 
 ugrep_indexer_CPPFLAGS = \
 	-I$(top_srcdir)/include \
+	-I$(top_srcdir)/lzma/C \
 	$(EXTRA_CFLAGS) \
 	$(PTHREAD_CFLAGS) \
 	-DPLATFORM=\"$(PLATFORM)\"
