--- vamp-plugin-sdk-2.6/Makefile.in.orig	2015-06-22 09:28:58.000000000 +0100
+++ vamp-plugin-sdk-2.6/Makefile.in	2015-10-22 22:56:44.046536248 +0100
@@ -74,8 +74,8 @@
 INSTALL_API_HEADERS	  = $(INSTALL_PREFIX)/include/vamp
 INSTALL_SDK_HEADERS	  = $(INSTALL_PREFIX)/include/vamp-sdk
 INSTALL_HOSTSDK_HEADERS	  = $(INSTALL_PREFIX)/include/vamp-hostsdk
-INSTALL_SDK_LIBS	  = $(INSTALL_PREFIX)/lib
-INSTALL_PLUGINS		  = $(INSTALL_PREFIX)/lib/vamp
+INSTALL_SDK_LIBS	  = $(INSTALL_PREFIX)/lib64
+INSTALL_PLUGINS		  = $(INSTALL_PREFIX)/lib64/vamp
 INSTALL_BINARIES	  = $(INSTALL_PREFIX)/bin 
 
 INSTALL_SDK_LIBNAME	  = libvamp-sdk.so.2.6.0
@@ -90,7 +90,7 @@
 INSTALL_HOSTSDK_STATIC    = libvamp-hostsdk.a
 INSTALL_HOSTSDK_LA        = libvamp-hostsdk.la
 
-INSTALL_PKGCONFIG	  = $(INSTALL_PREFIX)/lib/pkgconfig
+INSTALL_PKGCONFIG	  = $(INSTALL_PREFIX)/lib64/pkgconfig
 
 # Flags required to tell the compiler to create a dynamically loadable object
 #
--- vamp-plugin-sdk-2.6/pkgconfig/vamp.pc.in.orig	2015-06-22 09:28:58.000000000 +0100
+++ vamp-plugin-sdk-2.6/pkgconfig/vamp.pc.in	2015-10-22 22:56:11.381536127 +0100
@@ -1,6 +1,6 @@
 prefix=%PREFIX%
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=${exec_prefix}/lib64
 includedir=${prefix}/include
 
 Name: vamp
--- vamp-plugin-sdk-2.6/pkgconfig/vamp-hostsdk.pc.in.orig	2015-06-22 09:28:58.000000000 +0100
+++ vamp-plugin-sdk-2.6/pkgconfig/vamp-hostsdk.pc.in	2015-10-22 22:55:54.306536065 +0100
@@ -1,6 +1,6 @@
 prefix=%PREFIX%
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=${exec_prefix}/lib64
 includedir=${prefix}/include
 
 Name: vamp-hostsdk
--- vamp-plugin-sdk-2.6/pkgconfig/vamp-sdk.pc.in.orig	2015-06-22 09:28:58.000000000 +0100
+++ vamp-plugin-sdk-2.6/pkgconfig/vamp-sdk.pc.in	2015-10-22 22:56:02.744536096 +0100
@@ -1,6 +1,6 @@
 prefix=%PREFIX%
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=${exec_prefix}/lib64
 includedir=${prefix}/include
 
 Name: vamp-sdk
