--- Vc-20130205/cmake/VcConfig.cmake.in.orig	2013-02-05 23:11:40.000000000 +0000
+++ Vc-20130205/cmake/VcConfig.cmake.in	2013-02-07 21:29:19.437527810 +0000
@@ -6,9 +6,9 @@
 
 set(Vc_INSTALL_DIR "@CMAKE_INSTALL_PREFIX@")
 
-set(Vc_LIB_DIR              "@CMAKE_INSTALL_PREFIX@/lib")
+set(Vc_LIB_DIR              "@CMAKE_INSTALL_PREFIX@/lib@LIB_SUFFIX@")
 set(Vc_INCLUDE_DIR          "@CMAKE_INSTALL_PREFIX@/include")
-set(Vc_CMAKE_MODULES_DIR    "@CMAKE_INSTALL_PREFIX@/lib/cmake/Vc")
+set(Vc_CMAKE_MODULES_DIR    "@CMAKE_INSTALL_PREFIX@/lib@LIB_SUFFIX@/cmake/Vc")
 
 find_library(Vc_LIBRARIES Vc PATHS "${Vc_LIB_DIR}" NO_DEFAULT_PATH)
 
--- Vc-20130205/CMakeLists.txt.orig	2013-02-05 23:11:40.000000000 +0000
+++ Vc-20130205/CMakeLists.txt	2013-02-06 06:35:01.118091674 +0000
@@ -137,7 +137,7 @@
    add_target_property(Vc LABELS "other")
    add_dependencies(other Vc)
 
-   install(TARGETS Vc DESTINATION lib/)
+   install(TARGETS Vc DESTINATION lib${LIB_SUFFIX}/)
 else()
    # libVc should be compiled in the AliRoot tree, so we need to install the sources
    #
@@ -202,7 +202,7 @@
       ${CMAKE_CURRENT_BINARY_DIR}/cmake/VcConfigVersion.cmake
       cmake/OptimizeForArchitecture.cmake
       cmake/FindVc.cmake
-      DESTINATION lib/cmake/Vc
+      DESTINATION lib${LIB_SUFFIX}/cmake/Vc
       )
 endif()
 
