--- vdrift-2008-08-05/data/textures/effects/SConscript.orig	2008-08-12 22:24:39.000000000 +0100
+++ vdrift-2008-08-05/data/textures/effects/SConscript	2008-08-12 22:24:33.000000000 +0100
@@ -0,0 +1,25 @@
+#-------------#
+# Import Vars #
+#-------------#
+Import('*')
+
+#---------#
+# Sources #
+#---------#
+src = Split("""noise.png
+	       racingline.png
+	       white.png""")
+
+#---------------------------------#
+# Distribute to src_dir & bin_dir #
+#---------------------------------#
+dist_files = ['SConscript'] + src
+
+env.Distribute (src_dir, dist_files)
+env.Distribute (bin_dir, src)
+
+#--------------------#
+# Install data files #
+#--------------------#
+install = env.Install(Dir(Dir(env.subst('$data_directory/textures/effects'))), src)
+env.Alias('install', install)
--- vdrift-2008-08-05/data/textures/SConscript.orig	2007-04-18 10:05:23.000000000 +0100
+++ vdrift-2008-08-05/data/textures/SConscript	2008-08-12 22:25:08.000000000 +0100
@@ -13,6 +13,7 @@
 #----------------#
 Export(['env', 'src_dir', 'bin_dir'])
 
+SConscript('effects/SConscript')
 SConscript('hud/SConscript')
 SConscript('icons/SConscript')
 SConscript('smoke/SConscript')
--- vdrift-2008-08-05/data/skins/simple/textures/widgets/SConscript.orig	2007-04-18 10:05:23.000000000 +0100
+++ vdrift-2008-08-05/data/skins/simple/textures/widgets/SConscript	2008-08-12 22:09:56.000000000 +0100
@@ -15,6 +15,7 @@
                tog_off_down.png
                btn_up.png
                btn_down.png
+	       tab_page_background.png
                text_box.png
                text_cursor.png
                text_highlight.png
