--- verilog-10.2/vpi/Makefile.in.orig	2017-08-24 16:22:49.000000000 +0100
+++ verilog-10.2/vpi/Makefile.in	2019-07-21 22:22:22.152496597 +0100
@@ -139,6 +139,7 @@
 # Build this in two steps to avoid parallel build issues (see pr3462585)
 sdf_parse.c: $(srcdir)/sdf_parse.y
 	$(YACC) --verbose -t -p sdf -d -o $@ $<
+	touch sdf_parse.h sdf_parse.c
 sdf_parse.h: sdf_parse.c
 
 table_mod_lexor.o: table_mod_lexor.c table_mod_parse.h
@@ -149,6 +150,7 @@
 # Build this in two steps to avoid parallel build issues (see pr3462585)
 table_mod_parse.c: $(srcdir)/table_mod_parse.y
 	$(YACC) --verbose -t -p tblmod -d -o $@ $<
+	touch table_mod_parse.h table_mod_parse.c
 table_mod_parse.h: table_mod_parse.c
 
 v2005_math.vpi: $M ../vvp/libvpi.a
--- verilog-10.2/Makefile.in.orig	2017-08-24 16:22:49.000000000 +0100
+++ verilog-10.2/Makefile.in	2019-07-21 22:20:39.601501490 +0100
@@ -255,6 +255,7 @@
 # Build this in two steps to avoid parallel build issues (see pr3462585)
 parse.cc: $(srcdir)/parse.y
 	$(YACC) --verbose -t -p VL -d -o $@ $<
+	touch parse.hh parse.cc
 parse.h: parse.cc
 	mv parse.cc.h $@ 2>/dev/null || mv parse.hh $@
 
--- verilog-10.2/driver/Makefile.in.orig	2017-08-24 16:22:49.000000000 +0100
+++ verilog-10.2/driver/Makefile.in	2019-07-21 22:22:56.960494937 +0100
@@ -88,6 +88,7 @@
 # Build this in two steps to avoid parallel build issues (see pr3462585)
 cfparse.c: $(srcdir)/cfparse.y
 	$(YACC) --verbose -t -p cf -d -o $@ $<
+	touch cfparse.h cfparse.c
 cfparse.h: cfparse.c
 
 %.o: %.c
--- verilog-10.2/vhdlpp/Makefile.in.orig	2017-08-24 16:22:49.000000000 +0100
+++ verilog-10.2/vhdlpp/Makefile.in	2019-07-21 22:23:32.905493222 +0100
@@ -118,6 +118,7 @@
 # Build this in two steps to avoid parallel build issues (see pr3462585)
 parse.cc: $(srcdir)/parse.y
 	$(YACC) --verbose -t -d -o $@ $<
+	touch parse.hh parse.cc
 parse.h: parse.cc
 	mv parse.cc.h $@ 2>/dev/null || mv parse.hh $@
 
--- verilog-10.2/tgt-pcb/Makefile.in.orig	2017-08-24 16:22:49.000000000 +0100
+++ verilog-10.2/tgt-pcb/Makefile.in	2019-07-21 22:24:43.839489837 +0100
@@ -89,6 +89,7 @@
 
 fp.cc: $(srcdir)/fp.y
 	$(YACC) --verbose -t -p fp -d -o $@ $<
+	touch fp.cc fp.hh
 fp.h: fp.cc
 	mv fp.cc.h $@ 2>/dev/null || mv fp.hh $@
 
--- verilog-10.2/vvp/Makefile.in.orig	2017-08-24 16:22:49.000000000 +0100
+++ verilog-10.2/vvp/Makefile.in	2019-07-21 22:24:10.533491426 +0100
@@ -166,6 +166,7 @@
 # Build this in two steps to avoid parallel build issues (see pr3462585)
 parse.cc: $(srcdir)/parse.y
 	$(YACC) --verbose -t -d -o $@ $<
+	touch parse.cc parse.hh
 parse.h: parse.cc
 	mv parse.cc.h $@ 2>/dev/null || mv parse.hh $@
 
