--- videoporama-0.7.5/videoporama.py.orig	2009-05-24 06:20:46.000000000 +0100
+++ videoporama-0.7.5/videoporama.py	2009-05-31 16:32:22.000000000 +0100
@@ -41,7 +41,7 @@
 class videoporama :
     index=-1
     def __init__(self,args="o"):
-      z=self.checkfirsttime(args)
+      z=self.checkfirsttime("o")
       self.home=os.path.expanduser("~")
       self.lastdir=self.home
       # Qt application 
