--- viking-0.9.8/src/vikgpslayer.c.orig	2009-02-08 13:47:01.000000000 +0000
+++ viking-0.9.8/src/vikgpslayer.c	2009-05-31 20:58:33.000000000 +0100
@@ -234,7 +234,7 @@
   GpsFix realtime_fix;
   GpsFix last_fix;
 
-  enum unit realtime_gpsd_unit;
+//  enum unit realtime_gpsd_unit;
 
   VikTrack *realtime_track;
   gchar *realtime_track_name;
@@ -470,7 +470,7 @@
   vgl->realtime_track_pt1_gc = vik_viewport_new_gc ( vp, "red", 2 );
   vgl->realtime_track_pt2_gc = vik_viewport_new_gc ( vp, "green", 2 );
   vgl->realtime_track_pt_gc = vgl->realtime_track_pt1_gc;
-  vgl->realtime_gpsd_unit = gpsd_units();
+  //vgl->realtime_gpsd_unit = gpsd_units();
   // fprintf(stderr, "DEBUG: gpsd_unit = %d\n", vgl->realtime_gpsd_unit);
   vgl->realtime_track = NULL;
 
