--- vlc-2.1.5/configure.orig	2014-02-27 15:20:10.000000000 +0000
+++ vlc-2.1.5/configure	2014-11-14 10:44:52.586036350 +0000
@@ -35411,8 +35411,8 @@
 $as_echo "yes" >&6; }
 
     if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libavcodec < 56\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "libavcodec < 56") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libavcodec < 57\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "libavcodec < 57") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
