--- VTK-8.2.0/Rendering/Qt/vtkQtLabelRenderStrategy.cxx.orig	2019-01-30 17:15:13.000000000 +0000
+++ VTK-8.2.0/Rendering/Qt/vtkQtLabelRenderStrategy.cxx	2021-11-06 22:14:10.323121478 +0000
@@ -41,6 +41,7 @@
 #include <QImage>
 #include <QMap>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPair>
 #include <QPixmap>
 #include <QTextDocument>
--- VTK-8.2.0/Rendering/Qt/vtkQtStringToImage.cxx.orig	2019-01-30 17:15:13.000000000 +0000
+++ VTK-8.2.0/Rendering/Qt/vtkQtStringToImage.cxx	2021-11-06 22:14:28.243120623 +0000
@@ -31,6 +31,7 @@
 #include <QFontMetrics>
 #include <QImage>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPixmap>
 #include <QTextDocument>
 #include <QTextStream>
--- VTK-8.2.0/ThirdParty/exodusII/vtkexodusII/src/ex_create_par.c.orig	2019-01-30 17:15:13.000000000 +0000
+++ VTK-8.2.0/ThirdParty/exodusII/vtkexodusII/src/ex_create_par.c	2021-11-06 21:59:34.728163255 +0000
@@ -614,5 +614,5 @@
  * Prevent warning in some versions of ranlib(1) because the object
  * file has no symbols.
  */
-const char exodus_unused_symbol_dummy_1;
+const char exodus_unused_symbol_dummy_2;
 #endif
