--- VTK-9.2.6/IO/Image/vtkSEPReader.h.orig	2023-02-15 04:03:53.000000000 +0000
+++ VTK-9.2.6/IO/Image/vtkSEPReader.h	2023-07-07 19:30:25.304964344 +0100
@@ -27,6 +27,7 @@
 
 #include <array>  // for std::array
 #include <string> // for std::string
+#include <cstdint>
 
 namespace details
 {
--- VTK-9.2.6/ThirdParty/libproj/vtklibproj/src/proj_json_streaming_writer.hpp.orig	2023-02-15 04:03:53.000000000 +0000
+++ VTK-9.2.6/ThirdParty/libproj/vtklibproj/src/proj_json_streaming_writer.hpp	2023-07-07 20:11:25.706260352 +0100
@@ -33,6 +33,7 @@
 
 #include <vector>
 #include <string>
+#include <cstdint>
 
 #define CPL_DLL
 
