--- wacomtablet-1.2.3b/src/kded/devicehandler.cpp.orig	2010-10-08 08:51:00.000000000 +0100
+++ wacomtablet-1.2.3b/src/kded/devicehandler.cpp	2011-01-06 22:46:07.000000000 +0000
@@ -166,7 +166,7 @@
 
     for (int i = 0; i < ndevices; i++)
     {
-        if (info[i].use == IsXPointer || info[i].use == IsXKeyboard || info[i].use == IsXExtensionPointer)
+        if (info[i].use == IsXPointer || info[i].use == IsXKeyboard)
             continue;
 
         uint wacom_prop = XInternAtom(QX11Info::display(), "Wacom Tool Type", True);
--- wacomtablet-1.2.3b/src/kded/tabletdaemon.cpp.orig	2010-10-07 19:12:57.000000000 +0100
+++ wacomtablet-1.2.3b/src/kded/tabletdaemon.cpp	2011-01-06 22:34:36.000000000 +0000
@@ -252,7 +252,7 @@
     XDeviceInfo	*info = XListInputDevices(QX11Info::display(), &ndevices);
 
     for (int i = 0; i < ndevices; i++) {
-        if (info[i].use == IsXPointer || info[i].use == IsXKeyboard || info[i].use == IsXExtensionPointer)
+        if (info[i].use == IsXPointer || info[i].use == IsXKeyboard)
             continue;
 
         uint wacom_prop = XInternAtom(QX11Info::display(), "Wacom Tool Type", True);
