--- warzone2100-4.1.1/src/CMakeLists.txt.orig	2021-07-25 18:09:07.000000000 +0100
+++ warzone2100-4.1.1/src/CMakeLists.txt	2021-07-28 16:31:53.122090766 +0100
@@ -618,8 +618,10 @@
 # Install
 install(TARGETS warzone2100 COMPONENT Core DESTINATION "${WZ_APP_INSTALL_DEST}")
 
+if(CMAKE_SYSTEM_NAME MATCHES "Windows")
 # For Portable packages only, copy the ".portable" file that triggers portable mode (Windows-only)
 install(FILES "${CMAKE_SOURCE_DIR}/pkg/portable.in" COMPONENT PortableConfig DESTINATION "${WZ_APP_INSTALL_DEST}" RENAME ".portable")
+endif()
 
 #####################
 # Installing Required Runtime Dependencies
