--- warzone2100-4.3.2/src/CMakeLists.txt.orig	2022-11-10 17:05:49.693477573 +0000
+++ warzone2100-4.3.2/src/CMakeLists.txt	2022-11-10 17:06:31.906475559 +0000
@@ -625,8 +625,10 @@
 # Install
 install(TARGETS warzone2100 COMPONENT Core DESTINATION "${WZ_APP_INSTALL_DEST}")
 
+if(CMAKE_SYSTEM_NAME MATCHES "Windows")
 # For Portable packages only, copy the ".portable" file that triggers portable mode (Windows-only)
 install(FILES "${CMAKE_SOURCE_DIR}/pkg/portable.in" COMPONENT PortableConfig DESTINATION "${WZ_APP_INSTALL_DEST}" RENAME ".portable" EXCLUDE_FROM_ALL)
+endif()
 
 #####################
 # Installing Required Runtime Dependencies
