--- WebKit-r37126/WebCore/platform/gtk/ScrollViewGtk.cpp.orig	2008-10-01 01:28:31.000000000 +0100
+++ WebKit-r37126/WebCore/platform/gtk/ScrollViewGtk.cpp	2008-10-02 22:11:50.000000000 +0100
@@ -180,7 +180,7 @@
 
     // Now update the window (which should do nothing but a blit of the backing store's updateRect and so should
     // be very fast).
-    view->update();
+    // view->update();
 }
 
 void ScrollView::ScrollViewPrivate::adjustmentChanged(GtkAdjustment* adjustment, gpointer _that)
--- WebKit-r37126/WebKit/gtk/WebCoreSupport/ChromeClientGtk.cpp.orig	2008-10-01 01:28:33.000000000 +0100
+++ WebKit-r37126/WebKit/gtk/WebCoreSupport/ChromeClientGtk.cpp	2008-10-02 20:49:06.000000000 +0100
@@ -269,7 +269,7 @@
         return;
 
     GdkRectangle rect = windowRect;
-    GdkWindow* window = GTK_WINDOW(windowWidget);
+    GdkWindow* window = GDK_WINDOW(GTK_WINDOW(windowWidget));
 
     if (window) {
         // No double buffer.
