--- wesnoth-1.16.10/src/gettext.cpp.orig	2023-08-06 04:18:30.000000000 +0100
+++ wesnoth-1.16.10/src/gettext.cpp	2023-08-13 10:56:26.223334748 +0100
@@ -169,7 +169,7 @@
 			return msg;
 		}
 
-		const char* get(int domain_id, const char* ctx, const char* sid, int n) const override
+		const char* get(int domain_id, const char* ctx, const char* sid, bl::count_type n) const override
 		{
 			auto& base = get_base();
 			const char* msg = base.get(domain_id, ctx, sid, n);
