--- wireshark-0.99.3a/configure.orig	2006-08-24 18:04:01.000000000 +0100
+++ wireshark-0.99.3a/configure	2006-09-20 13:54:17.000000000 +0100
@@ -26058,10 +26058,10 @@
 				# the beginning of CFLAGS, CPPFLAGS,
 				# LDFLAGS, and LIBS.
 				#
-				LUA_LIBS="-L$lua_dir/lib -llua"
+				LUA_LIBS="-L$lua_dir/lib -llua -lm"
 				LUA_INCLUDES="-I$lua_dir/include"
 			else
-				LUA_LIBS="-llua"
+				LUA_LIBS="-llua -lm"
 				LUA_INCLUDES=""
 			fi
 
