--- wireshark-0.99.3a/gtk/packet_list.c.orig	2006-08-24 17:57:17.000000000 +0100
+++ wireshark-0.99.3a/gtk/packet_list.c	2006-09-22 15:27:12.000000000 +0100
@@ -837,6 +837,7 @@
 		/* even after move still invisible (happens with empty list) -> give up */
 		if(eth_clist_row_is_visible(ETH_CLIST(packet_list), row) !=
 			GTK_VISIBILITY_FULL) {
+			eth_clist_thaw(ETH_CLIST(packet_list));
 			return;
 		}
 
