--- wxPython-2.8.9.2/wxPython/contrib/gizmos/gtk/gizmos_wrap.cpp.orig	2009-02-16 20:58:04.000000000 +0000
+++ wxPython-2.8.9.2/wxPython/contrib/gizmos/gtk/gizmos_wrap.cpp	2009-03-18 19:59:20.000000000 +0000
@@ -10321,7 +10321,7 @@
   PyObject *resultobj = 0;
   wxPyTreeListCtrl *arg1 = (wxPyTreeListCtrl *) 0 ;
   wxTreeItemId *arg2 = 0 ;
-  wxTreeItemId const &arg3_defvalue = NULL ;
+  wxTreeItemId const &arg3_defvalue = (void *) NULL ;
   wxTreeItemId *arg3 = (wxTreeItemId *) &arg3_defvalue ;
   bool arg4 = (bool) true ;
   void *argp1 = 0 ;
