--- wxSVG-1.0b11/src/mediadec_ffmpeg.cpp.orig	2008-07-17 15:34:50.000000000 +0100
+++ wxSVG-1.0b11/src/mediadec_ffmpeg.cpp	2008-09-02 07:47:18.000000000 +0100
@@ -23,8 +23,8 @@
 extern "C" {
 #define __STDC_CONSTANT_MACROS
 #define __STDC_LIMIT_MACROS
-#include <avformat.h>
-#include <swscale.h>
+#include <libavformat/avformat.h>
+#include <libswscale/swscale.h>
 }
 
 wxFfmpegMediaDecoder::wxFfmpegMediaDecoder():
