--- x264-snapshot-20150807-2245/configure.orig	2015-08-07 21:45:03.000000000 +0100
+++ x264-snapshot-20150807-2245/configure	2015-08-09 10:41:39.081154195 +0100
@@ -1074,7 +1074,7 @@
 
 if [ "$gpac" = "auto" -a "$lsmash" != "yes" ] ; then
     gpac="no"
-    GPAC_LIBS="-lgpac_static"
+    GPAC_LIBS="-lgpac_static -lss -lcryptl"
     cc_check "" -lz && GPAC_LIBS="$GPAC_LIBS -lz"
     if [ "$SYS" = "WINDOWS" ] ; then
         cc_check "" -lws2_32 && GPAC_LIBS="$GPAC_LIBS -lws2_32"
