--- x265-1.5/source/CMakeLists.txt.orig	2015-02-10 21:15:13.000000000 +0000
+++ x265-1.5/source/CMakeLists.txt	2015-02-27 22:14:41.476338712 +0000
@@ -169,7 +169,7 @@
 endif()
 
 # Build options
-set(LIB_INSTALL_DIR lib CACHE STRING "Install location of libraries")
+set(LIB_INSTALL_DIR lib${LIB_SUFFIX} CACHE STRING "Install location of libraries")
 set(BIN_INSTALL_DIR bin CACHE STRING "Install location of executables")
 
 if(X64)
