--- xawtv-3.94/console/showriff.c.orig	2005-01-07 22:31:36.000000000 +0000
+++ xawtv-3.94/console/showriff.c	2005-01-07 22:30:11.000000000 +0000
@@ -371,7 +371,7 @@
    ProcessChunk prints out information of the chunk to stdout
    and returns FALSE, if an error occured. */
 
-static boolean ProcessChunk(FILE* f, size_t filepos, size_t filesize,
+static boolean ProcessChunk(FILE* f, off_t filepos, off_t filesize,
 			    FOURCC DesiredTag, int RekDepth,
 			    DWORD* chunksize)
 {
@@ -379,7 +379,7 @@
     int    buflen;
     char   tagstr[5];          /* FOURCC of chunk converted to string */
     FOURCC chunkid;            /* read FOURCC of chunk                */
-    size_t datapos;            /* position of data in file to process */
+    off_t datapos;            /* position of data in file to process */
     
     if (filepos>filesize-1) {  /* Oops. Must be something wrong!      */
 	printf("  *****  Error: Data would be behind end of file!\n");
