--- xawtv-3.95/x11/vbi-gui.c.orig	2003-04-14 19:06:25.000000000 +0100
+++ xawtv-3.95/x11/vbi-gui.c	2007-11-17 22:56:16.000000000 +0000
@@ -44,6 +44,7 @@
 #include "vbi-x11.h"
 #include "vbi-gui.h"
 
+#include "grab-ng.h"
 #include "channel.h"
 
 static int tt_debug = 1;
--- xawtv-3.95/console/matrox.c.orig	2005-02-03 10:40:41.000000000 +0000
+++ xawtv-3.95/console/matrox.c	2010-06-28 07:24:28.000000000 +0100
@@ -9,7 +9,6 @@
 #include <sys/ioctl.h>
 #include <sys/mman.h>
 
-#include <asm/page.h> /* PAGE_SIZE */
 #include <linux/fb.h>
 
 #include "byteswap.h"
@@ -226,7 +225,7 @@
 	return -1;
     }
     off = (unsigned long)fb_fix.mmio_start -
-	((unsigned long)fb_fix.mmio_start & ~(PAGE_SIZE-1));
+	((unsigned long)fb_fix.mmio_start & ~(sysconf(_SC_PAGE_SIZE)-1));
     bmmio += off;
     mmio = (uint32_t*)bmmio;
     return 0;
--- xawtv-3.95/console/fbtools.c.orig	2003-02-14 14:14:04.000000000 +0000
+++ xawtv-3.95/console/fbtools.c	2010-06-28 07:23:32.000000000 +0100
@@ -21,8 +21,6 @@
 #include <linux/vt.h>
 #include <linux/fb.h>
 
-#include <asm/page.h>
-
 #include "fbtools.h"
 
 /* -------------------------------------------------------------------- */
@@ -424,7 +422,7 @@
 	goto err;
     }
 #endif
-    fb_mem_offset = (unsigned long)(fb_fix.smem_start) & (~PAGE_MASK);
+    fb_mem_offset = (unsigned long)(fb_fix.smem_start) & ~(sysconf(_SC_PAGE_SIZE)-1);
     fb_mem = mmap(NULL,fb_fix.smem_len+fb_mem_offset,
 		  PROT_READ|PROT_WRITE,MAP_SHARED,fb,0);
     if (-1L == (long)fb_mem) {
