--- xine-lib-1.1.11.1/src/demuxers/demux_qt.c.orig	2008-03-28 22:50:47.000000000 +0000
+++ xine-lib-1.1.11.1/src/demuxers/demux_qt.c	2008-04-11 10:33:33.000000000 +0100
@@ -896,11 +896,6 @@
     current_atom_size = _X_BE_32(&trak_atom[i - 4]);	
     current_atom = _X_BE_32(&trak_atom[i]);
 
-    if (current_atom_size > trak_atom_size - i) {
-      last_error = QT_NOT_A_VALID_FILE;
-      goto free_trak;
-    }
-
     if (current_atom == TKHD_ATOM) {
       trak->flags = _X_BE_16(&trak_atom[i + 6]);
     } else if (current_atom == ELST_ATOM) {
--- xine-lib-1.1.11.1/src/demuxers/ebml.c.orig	2008-03-28 14:24:50.000000000 +0000
+++ xine-lib-1.1.11.1/src/demuxers/ebml.c	2008-04-11 10:33:39.000000000 +0100
@@ -327,7 +327,7 @@
   if (text)
   {
     text[elem->len] = '\0';
-    if (ebml_read_ascii (ebml, &elem, text))
+    if (ebml_read_ascii (ebml, elem, text))
       return text;
     free (text);
   }
