--- xine-lib-1.1.7/src/demuxers/demux_ogg.c.orig	2007-04-17 02:00:50.000000000 +0100
+++ xine-lib-1.1.7/src/demuxers/demux_ogg.c	2007-06-09 22:16:49.000000000 +0100
@@ -240,7 +241,7 @@
     buffer = ogg_sync_buffer(&this->oy, CHUNKSIZE);
     bytes  = this->input->read(this->input, buffer, CHUNKSIZE);
     ogg_sync_wrote(&this->oy, bytes);
-    if (bytes < CHUNKSIZE/2) {
+    if (bytes <= 0) {
       return 0;
     }
   }
