diff -rc xpaint-2.9.0/fontOp.c xpaint-2.9.0a/fontOp.c
*** xpaint-2.9.0/fontOp.c	2010-05-13 16:04:52.000000000 +0200
--- xpaint-2.9.0a/fontOp.c	2010-05-24 23:18:30.837257498 +0200
***************
*** 79,87 ****
      int zoom; 
      static int x, y, xp, yp;
      double c, s;
-     if (l->w == None)
- 	return;
  
      XtVaGetValues(l->w, XtNzoom, &zoom, NULL);
      if (zoom != zoom0) l->state = False;
      zoom0 = zoom;
--- 79,86 ----
      int zoom; 
      static int x, y, xp, yp;
      double c, s;
  
+     if (!l->w)return;
      XtVaGetValues(l->w, XtNzoom, &zoom, NULL);
      if (zoom != zoom0) l->state = False;
      zoom0 = zoom;
***************
*** 112,117 ****
--- 111,117 ----
  static void 
  flash(LocalInfo * l)
  {
+     if (!l->w) return;
      cursor(l, True);
      l->id = XtAppAddTimeOut(XtWidgetToApplicationContext(l->w),
  		      DELAY, (XtTimerCallbackProc) flash, (XtPointer) l);
***************
*** 546,553 ****
  #endif
  
  
!     if (l->w == None)
! 	return;
  
  #ifdef XAW3DXFT
      if (XftEncoding==0) {
--- 546,552 ----
  #endif
  
  
!     if (!l->w) return;
  
  #ifdef XAW3DXFT
      if (XftEncoding==0) {
