--- xpdf-chinese-simplified/add-to-xpdfrc.orig	2019-01-08 21:14:07.516029282 +0000
+++ xpdf-chinese-simplified/add-to-xpdfrc	2019-01-08 21:18:50.531015779 +0000
@@ -1,9 +1,9 @@
 #----- begin Chinese Simplified support package (2011-sep-02)
-cidToUnicode	Adobe-GB1	/usr/local/share/xpdf/chinese-simplified/Adobe-GB1.cidToUnicode
-unicodeMap	ISO-2022-CN	/usr/local/share/xpdf/chinese-simplified/ISO-2022-CN.unicodeMap
-unicodeMap	EUC-CN		/usr/local/share/xpdf/chinese-simplified/EUC-CN.unicodeMap
-unicodeMap	GBK		/usr/local/share/xpdf/chinese-simplified/GBK.unicodeMap
-cMapDir		Adobe-GB1	/usr/local/share/xpdf/chinese-simplified/CMap
-toUnicodeDir			/usr/local/share/xpdf/chinese-simplified/CMap
-#fontFileCC	Adobe-GB1	/usr/..../NotoSansCJKsc-Regular.otf
+cidToUnicode	Adobe-GB1	/usr/share/xpdf/chinese-simplified/Adobe-GB1.cidToUnicode
+unicodeMap	ISO-2022-CN	/usr/share/xpdf/chinese-simplified/ISO-2022-CN.unicodeMap
+unicodeMap	EUC-CN		/usr/share/xpdf/chinese-simplified/EUC-CN.unicodeMap
+unicodeMap	GBK		/usr/share/xpdf/chinese-simplified/GBK.unicodeMap
+cMapDir		Adobe-GB1	/usr/share/xpdf/chinese-simplified/CMap
+toUnicodeDir			/usr/share/xpdf/chinese-simplified/CMap
+fontFileCC	Adobe-GB1	/usr/share/fonts/truetype/Noto/NotoSansCJKsc-Regular.otf
 #----- end Chinese Simplified support package
