--- xpdf-korean/add-to-xpdfrc.orig	2019-01-08 21:22:27.146005443 +0000
+++ xpdf-korean/add-to-xpdfrc	2019-01-08 21:23:02.963003734 +0000
@@ -1,7 +1,7 @@
 #----- begin Korean support package (2011-sep-02)
-cidToUnicode	Adobe-Korea1	/usr/local/share/xpdf/korean/Adobe-Korea1.cidToUnicode
-unicodeMap	ISO-2022-KR	/usr/local/share/xpdf/korean/ISO-2022-KR.unicodeMap
-cMapDir		Adobe-Korea1	/usr/local/share/xpdf/korean/CMap
-toUnicodeDir			/usr/local/share/xpdf/korean/CMap
-#fontFileCC	Adobe-Korea1	/usr/..../NotoSansCJKkr-Regular.otf
+cidToUnicode	Adobe-Korea1	/usr/share/xpdf/korean/Adobe-Korea1.cidToUnicode
+unicodeMap	ISO-2022-KR	/usr/share/xpdf/korean/ISO-2022-KR.unicodeMap
+cMapDir		Adobe-Korea1	/usr/share/xpdf/korean/CMap
+toUnicodeDir			/usr/share/xpdf/korean/CMap
+#fontFileCC	Adobe-Korea1	/usr/share/fonts/truetype/Noto/NotoSansCJKkr-Regular.otf
 #----- end Korean support package
