--- ./add-to-xpdfrc.orig	2003-07-07 22:43:40.000000000 -0700
+++ ./add-to-xpdfrc	2003-09-21 15:15:27.000000000 -0700
@@ -1,4 +1,4 @@
 #----- begin Greek support package (2003-jun-28)
-nameToUnicode			/usr/local/share/xpdf/greek/Greek.nameToUnicode
-unicodeMap	ISO-8859-7	/usr/local/share/xpdf/greek/ISO-8859-7.unicodeMap
+nameToUnicode			/usr/X11R6/share/xpdf/greek/Greek.nameToUnicode
+unicodeMap	ISO-8859-7	/usr/X11R6/share/xpdf/greek/ISO-8859-7.unicodeMap
 #----- end Greek support package
