--- xplanet-1.2.1/src/Separation.cpp.orig	2008-11-14 00:19:05.000000000 +0000
+++ xplanet-1.2.1/src/Separation.cpp	2010-06-19 09:55:44.000000000 +0100
@@ -1,4 +1,5 @@
 #include <cmath>
+#include <cstdio>
 #include <iostream>
 
 #include "Separation.h"
--- xplanet-1.2.1/src/Satellite.cpp.orig	2006-03-26 00:34:21.000000000 +0000
+++ xplanet-1.2.1/src/Satellite.cpp	2010-06-19 09:54:17.000000000 +0100
@@ -1,4 +1,5 @@
 #include <clocale>
+#include <cstdio>
 #include <cstring>
 #include <iostream>
 using namespace std;
