--- yafaray-0.1.5/include/core_api/color.h.orig	2013-09-03 18:03:17.000000000 +0100
+++ yafaray-0.1.5/include/core_api/color.h	2019-07-13 19:07:42.397469287 +0100
@@ -53,8 +53,8 @@
 		color_t(CFLOAT g) { R=G=B=g; }
 		color_t(CFLOAT af[3]) { R=af[0];  G=af[1];  B=af[2]; }
 		bool isBlack() const { return ((R==0) && (G==0) && (B==0)); }
-		bool isNaN() const { return (isnan(R) || isnan(G) || isnan(B)); }
-		bool isInf() const { return (isinf(R) || isinf(G) || isinf(B)); }
+		bool isNaN() const { return (std::isnan(R) || std::isnan(G) || std::isnan(B)); }
+		bool isInf() const { return (std::isinf(R) || std::isinf(G) || std::isinf(B)); }
 		~color_t() {}
 		void set(CFLOAT r, CFLOAT g, CFLOAT b) { R=r;  G=g;  B=b; }
 
--- yafaray-0.1.5/CMakeModules/FindOpenEXR.cmake.orig	2013-09-03 18:03:17.000000000 +0100
+++ yafaray-0.1.5/CMakeModules/FindOpenEXR.cmake	2022-05-05 14:36:14.767966618 +0100
@@ -26,10 +26,13 @@
 		/sw/include
 		/opt/local/include)
 
-	FIND_LIBRARY(OPENEXR_HALF_LIBRARY 
-		NAMES Half
-		PATHS ${LIBRARY_DIRS})
-  
+	FIND_PATH(IMATH_INCLUDE_PATH half.h
+		PATH_SUFFIXES Imath
+		/usr/include
+		/usr/local/include
+		/sw/include
+		/opt/local/include)
+
 	FIND_LIBRARY(OPENEXR_IEX_LIBRARY 
 		NAMES Iex
 		PATHS ${LIBRARY_DIRS})
@@ -38,22 +41,18 @@
 		NAMES Imath
 		PATHS ${LIBRARY_DIRS})
   
-	FIND_LIBRARY(OPENEXR_ILMIMF_LIBRARY
-		NAMES IlmImf
-		PATHS ${LIBRARY_DIRS})
-
 	FIND_LIBRARY(OPENEXR_ILMTHREAD_LIBRARY
 		NAMES IlmThread
 		PATHS ${LIBRARY_DIRS})
 
 ENDIF(ZLIB_FOUND)
 
-#MESSAGE(STATUS ${OPENEXR_IMATH_LIBRARY} ${OPENEXR_ILMIMF_LIBRARY} ${OPENEXR_IEX_LIBRARY} ${OPENEXR_HALF_LIBRARY} ${OPENEXR_ILMTHREAD_LIBRARY} ${ZLIB_LIBRARY})
+#MESSAGE(STATUS ${OPENEXR_IMATH_LIBRARY} ${OPENEXR_IEX_LIBRARY} ${OPENEXR_ILMTHREAD_LIBRARY} ${ZLIB_LIBRARY})
 
-IF (OPENEXR_INCLUDE_PATH AND OPENEXR_IMATH_LIBRARY AND OPENEXR_ILMIMF_LIBRARY AND OPENEXR_IEX_LIBRARY AND OPENEXR_HALF_LIBRARY)
+IF (OPENEXR_INCLUDE_PATH AND OPENEXR_IMATH_LIBRARY AND OPENEXR_IEX_LIBRARY)
 	SET(OPENEXR_FOUND TRUE)
-	SET(OPENEXR_INCLUDE_DIRS ${OPENEXR_INCLUDE_PATH} CACHE STRING "The include paths needed to use OpenEXR")
-	SET(OPENEXR_LIBRARIES ${OPENEXR_IMATH_LIBRARY} ${OPENEXR_ILMIMF_LIBRARY} ${OPENEXR_IEX_LIBRARY} ${OPENEXR_HALF_LIBRARY} ${ZLIB_LIBRARY} CACHE STRING "The libraries needed to use OpenEXR")
+	SET(OPENEXR_INCLUDE_DIRS ${OPENEXR_INCLUDE_PATH} ${IMATH_INCLUDE_PATH} CACHE STRING "The include paths needed to use OpenEXR")
+	SET(OPENEXR_LIBRARIES ${OPENEXR_IMATH_LIBRARY} ${OPENEXR_IEX_LIBRARY} ${ZLIB_LIBRARY} CACHE STRING "The libraries needed to use OpenEXR")
 
 	IF(OPENEXR_ILMTHREAD_LIBRARY)
 		SET(OPENEXR_LIBRARIES ${OPENEXR_LIBRARIES} ${OPENEXR_ILMTHREAD_LIBRARY})
