--- yafray-0.0.9/src/loader/mlex.cc.orig	2004-12-22 09:26:24.000000000 +0000
+++ yafray-0.0.9/src/loader/mlex.cc	2010-07-03 17:27:26.000000000 +0100
@@ -21,6 +21,7 @@
  */
 
 #include<stdio.h>
+#include<cstring>
 #include<iostream>
 #include "mlex.h"
 using namespace std;
--- yafray-0.0.9/src/shaders/basictex.cc.orig	2006-06-10 01:41:26.000000000 +0100
+++ yafray-0.0.9/src/shaders/basictex.cc	2010-07-03 17:29:49.000000000 +0100
@@ -23,6 +23,7 @@
 #include "basictex.h"
 #include "object3d.h"
 #include <iostream>
+#include <cstring>
 
 #include "targaIO.h"
 #include "HDR_io.h"
@@ -259,7 +260,7 @@
 		intp_type = BICUBIC;
 	
 	// Load image, try to determine from extensions first
-	char *ext = strrchr(filename, '.');
+	const char *ext = strrchr(filename, '.');
 	bool jpg_tried = false;
 	bool tga_tried = false;
 	bool hdr_tried = false;
--- yafray-0.0.9/src/yafraycore/buffer.h.orig	2005-06-27 15:11:30.000000000 +0100
+++ yafray-0.0.9/src/yafraycore/buffer.h	2010-07-03 17:13:43.000000000 +0100
@@ -28,6 +28,7 @@
 #endif
 
 #include <cstdio>
+#include <cstdlib>
 #include <iostream>
 #include "color.h"
 
--- yafray-0.0.9/src/yafraycore/targaIO.cc.orig	2006-05-18 16:36:08.000000000 +0100
+++ yafray-0.0.9/src/yafraycore/targaIO.cc	2010-07-03 17:23:10.000000000 +0100
@@ -3,6 +3,7 @@
 //--------------------------------------------------------------------------------
 #include "targaIO.h"
 #include "vector3d.h"
+#include <cstring>
 
 //--------------------------------------------------------------------------------
 // Save uncompressed 24 bit targa
--- yafray-0.0.9/src/yafraycore/ccthreads.h.orig	2010-07-03 17:16:45.000000000 +0100
+++ yafray-0.0.9/src/yafraycore/ccthreads.h	2010-07-03 17:16:17.000000000 +0100
@@ -7,6 +7,7 @@
 #endif
 
 #include<errno.h>
+#include <cstdlib>
 
 #if HAVE_PTHREAD
 #include<pthread.h>
--- yafray-0.0.9/src/yafraycore/HDR_io.cc.orig	2010-07-03 17:15:41.000000000 +0100
+++ yafray-0.0.9/src/yafraycore/HDR_io.cc	2010-07-03 17:15:13.000000000 +0100
@@ -5,6 +5,8 @@
 
 #include "HDR_io.h"
 
+#include<cstring>
+
 #ifdef HAVE_CONFIG_H
 #include<config.h>
 #endif
--- yafray-0.0.9/src/yafraycore/renderblock.cc.orig	2005-05-30 07:28:19.000000000 +0100
+++ yafray-0.0.9/src/yafraycore/renderblock.cc	2010-07-03 17:19:03.000000000 +0100
@@ -1,5 +1,6 @@
 
 #include "renderblock.h"
+#include <cstdlib>
 
 using namespace std;
 
--- yafray-0.0.9/src/yafraycore/kdtree.cc.orig	2006-07-14 10:58:05.000000000 +0100
+++ yafray-0.0.9/src/yafraycore/kdtree.cc	2010-07-03 17:17:53.000000000 +0100
@@ -4,6 +4,7 @@
 #include <math.h>
 #include <limits>
 #include <time.h>
+#include <cstring>
 
 __BEGIN_YAFRAY
 
--- yafray-0.0.9/src/yafraycore/geometree.h.orig	2004-12-22 09:26:38.000000000 +0000
+++ yafray-0.0.9/src/yafraycore/geometree.h	2010-07-03 17:22:27.000000000 +0100
@@ -2,6 +2,7 @@
 #define __GEOMETREE_H
 
 #include"bound.h"
+#include <limits>
 
 __BEGIN_YAFRAY
 
