--- zam-plugins-3.10/plugins/Makefile.mk.orig	2018-02-18 01:26:44.000000000 +0000
+++ zam-plugins-3.10/plugins/Makefile.mk	2018-12-30 20:52:14.293016404 +0000
@@ -62,7 +62,7 @@
 DISTRHO_PLUGIN_FILES = ../../dpf/distrho/DistrhoPluginMain.cpp
 
 ifeq ($(HAVE_DGL),true)
-DISTRHO_UI_FILES     = ../../dpf/distrho/DistrhoUIMain.cpp ../../dpf/libdgl.a
+DISTRHO_UI_FILES     = ../../dpf/distrho/DistrhoUIMain.cpp ../../dpf/build/libdgl.a
 endif
 
 # --------------------------------------------------------------
