--- zeitgeist-1.0.3/libzeitgeist/where-clause.vala.orig	2020-10-15 07:18:22.000000000 +0100
+++ zeitgeist-1.0.3/libzeitgeist/where-clause.vala	2021-08-20 22:02:12.950824637 +0100
@@ -216,7 +216,7 @@
 #else
             long[] pointers = new long[gptrarr.length + 1];
 #endif
-            Memory.copy(pointers, ((PtrArray *) gptrarr)->pdata,
+            Memory.copy(pointers, (void *) gptrarr.data,
                 gptrarr.length * sizeof (void *));
             return (T[]) pointers;
         }
