--- zxing-cpp-20180818/CMakeLists.txt.orig	2018-08-18 11:48:25.000000000 +0100
+++ zxing-cpp-20180818/CMakeLists.txt	2018-08-18 11:58:21.285304490 +0100
@@ -86,13 +86,13 @@
 target_link_libraries(zxing libzxing)
 
 install(TARGETS zxing libzxing EXPORT zxing-targets
-	LIBRARY DESTINATION lib
+	LIBRARY DESTINATION lib${LIB_SUFFIX}
 	RUNTIME DESTINATION bin
-	ARCHIVE DESTINATION lib
+	ARCHIVE DESTINATION lib${LIB_SUFFIX}
     INCLUDES DESTINATION include
 )
 
-install(EXPORT zxing-targets DESTINATION lib/zxing/cmake NAMESPACE zxing::)
+install(EXPORT zxing-targets DESTINATION lib${LIB_SUFFIX}/zxing/cmake NAMESPACE zxing::)
 
 install(
     DIRECTORY core/src/zxing/
@@ -109,7 +109,7 @@
 endif()
 
 configure_file(cmake/zxing-config.cmake.in zxing-config.cmake @ONLY)
-install(FILES ${CMAKE_BINARY_DIR}/zxing-config.cmake DESTINATION lib/zxing/cmake)
+install(FILES ${CMAKE_BINARY_DIR}/zxing-config.cmake DESTINATION lib${LIB_SUFFIX}/zxing/cmake)
 
 if(BUILD_TESTING)
     # Add testrunner executable.
