--- zynaddsubfx-3.0.5/src/Nio/MidiIn.h.orig	2019-02-23 15:43:37.000000000 +0000
+++ zynaddsubfx-3.0.5/src/Nio/MidiIn.h	2025-07-29 08:54:53.007297720 +0100
@@ -17,6 +17,7 @@
 #define MIDI_IN_H
 
 #include "Engine.h"
+#include <cstdint>
 
 namespace zyn {
 
--- zynaddsubfx-3.0.5/src/UI/MasterUI.fl.orig	2019-03-10 16:34:32.000000000 +0000
+++ zynaddsubfx-3.0.5/src/UI/MasterUI.fl	2025-07-29 09:02:36.065009934 +0100
@@ -17,6 +17,9 @@
 decl {\#include <string.h>} {public local
 } 
 
+decl {\#include <FL/platform.H>} {public local
+}
+
 decl {\#if ! defined(PLUGINVERSION) && HAS_X11
 \#include "zynaddsubfx.xpm"
 \#endif} {private local
--- zynaddsubfx-3.0.5/src/Misc/Bank.h.orig	2017-05-18 14:58:31.000000000 +0100
+++ zynaddsubfx-3.0.5/src/Misc/Bank.h	2025-07-29 09:03:31.982063810 +0100
@@ -14,6 +14,7 @@
 #ifndef BANK_H
 #define BANK_H
 
+#include <cstdint>
 #include <string>
 #include <vector>
 #include "../globals.h"
