--- libXaw3d-1.6/src/XawIm.c.orig	2011-10-09 19:01:20.000000000 +0100
+++ libXaw3d-1.6/src/XawIm.c	2012-01-22 20:17:10.496186711 +0000
@@ -141,14 +141,14 @@
     return(NULL);
 }
 
-static XContext extContext = (XContext)NULL;
+static XContext extContext = (XContext)0;
 
 static XawVendorShellExtPart *
 SetExtPart(VendorShellWidget w, XawVendorShellExtWidget vew)
 {
     contextDataRec *contextData;
 
-    if (extContext == (XContext)NULL) extContext = XUniqueContext();
+    if (extContext == (XContext)0) extContext = XUniqueContext();
 
     contextData = XtNew(contextDataRec);
     contextData->parent = (Widget)w;
@@ -252,14 +252,14 @@
     }
 }
 
-static XContext errContext = (XContext)NULL;
+static XContext errContext = (XContext)0;
 
 static Widget
 SetErrCnxt(Widget w, XIM xim)
 {
     contextErrDataRec *contextErrData;
 
-    if (errContext == (XContext)NULL) errContext = XUniqueContext();
+    if (errContext == (XContext)0) errContext = XUniqueContext();
 
     contextErrData = XtNew(contextErrDataRec);
     contextErrData->widget = w;
@@ -1321,12 +1321,12 @@
 	return;
     XtFree( (char*) ve->im.resources );
 
-    if (extContext != (XContext)NULL &&
+    if (extContext != (XContext)0 &&
 	!XFindContext (XtDisplay (w), (Window)w,
 		       extContext, (XPointer*)&contextData))
         XtFree( (char*) contextData );
 
-    if (errContext != (XContext)NULL &&
+    if (errContext != (XContext)0 &&
 	!XFindContext (XDisplayOfIM( ve->im.xim ), (Window) ve->im.xim,
 		       errContext, (XPointer*) &contextErrData))
         XtFree( (char*) contextErrData );
