--- Mesa-10.2.7/src/gallium/auxiliary/gallivm/lp_bld_debug.cpp.orig	2014-09-06 00:35:25.000000000 +0100
+++ Mesa-10.2.7/src/gallium/auxiliary/gallivm/lp_bld_debug.cpp	2014-09-06 16:04:39.681611815 +0100
@@ -62,8 +62,12 @@
 #endif /* HAVE_LLVM >= 0x0301 */
 
 #if HAVE_LLVM >= 0x0303
+#if HAVE_LLVM >= 0x0305
+  #define OwningPtr std::unique_ptr
+#else
 #include <llvm/ADT/OwningPtr.h>
 #endif
+#endif
 
 #if HAVE_LLVM >= 0x0305
 #include <llvm/MC/MCContext.h>
