--- xcm-0.5.0/scripts/90-xcm-i2c.rules.orig	2011-01-02 17:58:25.000000000 +0000
+++ xcm-0.5.0/scripts/90-xcm-i2c.rules	2012-03-21 06:26:20.000000000 +0000
@@ -9,4 +9,4 @@
 KERNEL=="i2c-[0-9]*", ACTION=="add", GROUP="video", MODE="0666"
 
 # FIXME: match only DDC devices
-KERNEL=="i2c-[0-9]*", ACTION=="add", PROGRAM=="xcmddc --i2c /dev/%k --identify", RESULT=="0", GROUP="video"
+KERNEL=="i2c-[0-9]*", ACTION=="add", PROGRAM=="/usr/bin/xcmddc --i2c /dev/%k --identify", RESULT=="0", GROUP="video"
