--- xf86-video-avivo-20070913/xorg/avivo_output.c.orig	2007-09-13 20:05:47.000000000 +0100
+++ xf86-video-avivo-20070913/xorg/avivo_output.c	2007-09-14 22:50:53.000000000 +0100
@@ -357,7 +357,7 @@
     xf86MonPtr edid_mon;
 
     if (!xf86I2CProbeAddress(avivo_output->i2c, 0x00A0))
-        return XF86OutputStatusUnknown;
+        return XF86OutputStatusDisconnected;
     edid_mon = xf86OutputGetEDID(output, avivo_output->i2c);
     if (edid_mon == NULL) {
         return XF86OutputStatusUnknown;
@@ -381,7 +381,7 @@
     xf86MonPtr edid_mon;
 
     if (!xf86I2CProbeAddress(avivo_output->i2c, 0x00A0))
-        return XF86OutputStatusUnknown;
+        return XF86OutputStatusDisconnected;
     edid_mon = xf86OutputGetEDID(output, avivo_output->i2c);
     if (edid_mon == NULL) {
         return XF86OutputStatusUnknown;
