--- xf86-video-ati-20090819/src/radeon_dri2.c.orig	2009-08-19 21:57:21.000000000 +0100
+++ xf86-video-ati-20090819/src/radeon_dri2.c	2009-08-19 22:53:17.000000000 +0100
@@ -127,13 +127,13 @@
     return buffers;
 }
 #else
-static DRI2BufferPtr
+static DRI2Buffer2Ptr
 radeon_dri2_create_buffer(DrawablePtr drawable,
                           unsigned int attachment,
                           unsigned int format)
 {
     ScreenPtr pScreen = drawable->pScreen;
-    DRI2BufferPtr buffers;
+    DRI2Buffer2Ptr buffers;
     struct dri2_buffer_priv *privates;
     PixmapPtr pixmap, depth_pixmap;
     struct radeon_exa_pixmap_priv *driver_priv;
@@ -228,7 +228,7 @@
 }
 #else
 static void
-radeon_dri2_destroy_buffer(DrawablePtr drawable, DRI2BufferPtr buffers)
+radeon_dri2_destroy_buffer(DrawablePtr drawable, DRI2Buffer2Ptr buffers)
 {
     if(buffers)
     {
