--- xorg-server-1.14.0/dix/window.c.orig	2013-03-06 19:37:39.315304812 +0000
+++ xorg-server-1.14.0/dix/window.c	2013-03-06 19:38:41.277307824 +0000
@@ -145,8 +145,8 @@
 
 Bool bgNoneRoot = FALSE;
 
-static unsigned char _back_lsb[4] = { 0x88, 0x22, 0x44, 0x11 };
-static unsigned char _back_msb[4] = { 0x11, 0x44, 0x22, 0x88 };
+static unsigned char _back_lsb[4] = { 0x00, 0x00, 0x00, 0x00 };
+static unsigned char _back_msb[4] = { 0x00, 0x00, 0x00, 0x00 };
 
 static Bool WindowParentHasDeviceCursor(WindowPtr pWin,
                                         DeviceIntPtr pDev, CursorPtr pCurs);
