--- ./hw/xfree86/utils/xorgconfig/xorgconfig.c.orig	2006-11-30 19:40:11.000000000 -0600
+++ ./hw/xfree86/utils/xorgconfig/xorgconfig.c	2007-03-29 17:22:25.000000000 -0500
@@ -170,7 +170,7 @@
 #else
 # define TREEROOTFONT		TREEROOTLX "/fonts"
 #endif
-#define MODULEPATH		TREEROOT "/lib/modules"
+#define MODULEPATH		TREEROOT "/lib/xorg/modules"
 
 #ifndef __UNIXOS2__
 #define XSERVERNAME_FOR_PROBE "X"
@@ -1949,7 +1949,7 @@
 "#    Load        \"xtt\"\n"
 "\n"
 "# This loads the GLX module\n"
-"#    Load       \"glx\"\n"
+"    Load       \"glx\"\n"
 "# This loads the DRI module\n"
 "#    Load       \"dri\"\n"
 "\n"
@@ -1988,8 +1988,9 @@
 	"/100dpi/:unscaled",
 	"/Speedo/",
 	"/Type1/",
-	"/TrueType/",
-	"/freefont/",
+	"/TTF/",
+	"/OTF/",
+	"/cyrillic/",
 	"/75dpi/",
 	"/100dpi/",
 	0 /* end of fontpaths */
@@ -2897,7 +2898,7 @@
 static char *pathnote_text =	
 "Note that the X binary directory in your path may be a symbolic link.\n"
 "In that case you could modify the symbolic link to point to the new binaries.\n"
-"Example: 'rm -f /usr/bin/X11; ln -s /usr/X11R6/bin /usr/bin/X11', if the\n"
+"Example: 'rm -f /usr/bin/X11; ln -s /usr/bin /usr/bin/X11', if the\n"
 "link is '/usr/bin/X11'.\n"
 "\n"
 "Make sure the path is OK before continuing.\n";
