--- kdepim-3.5.10/configure.orig	2008-08-20 16:25:12.000000000 +0100
+++ kdepim-3.5.10/configure	2008-08-31 07:30:22.000000000 +0100
@@ -35259,7 +35259,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lgnokii $all_libraries -lintl -lXpm -lX11 $LIB_BLUETOOTH  $LIBS"
+LIBS="-lgnokii $all_libraries -lXpm -lX11 $LIB_BLUETOOTH  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -35315,7 +35315,7 @@
 { echo "$as_me:$LINENO: result: $ac_cv_lib_gnokii_gn_cfg_get" >&5
 echo "${ECHO_T}$ac_cv_lib_gnokii_gn_cfg_get" >&6; }
 if test $ac_cv_lib_gnokii_gn_cfg_get = yes; then
-   LIB_GNOKII="-lgnokii -lintl $LIB_BLUETOOTH"
+   LIB_GNOKII="-lgnokii $LIB_BLUETOOTH"
 fi
 
 
