--- orage-4.7.5/src/ical-expimp.c.orig	2010-03-04 07:34:55.000000000 +0000
+++ orage-4.7.5/src/ical-expimp.c	2010-07-04 15:53:10.000000000 +0100
@@ -51,8 +51,8 @@
 #include <gtk/gtk.h>
 #include <glib/gprintf.h>
 #include <glib/gstdio.h>
-#include <ical.h>
-#include <icalss.h>
+#include <libical/ical.h>
+#include <libical/icalss.h>
 
 #include "orage-i18n.h"
 #include "functions.h"
--- orage-4.7.5/src/ical-code.c.orig	2010-03-04 07:34:55.000000000 +0000
+++ orage-4.7.5/src/ical-code.c	2010-07-04 15:52:28.000000000 +0100
@@ -51,8 +51,8 @@
 #include <gtk/gtk.h>
 #include <glib/gprintf.h>
 #include <glib/gstdio.h>
-#include <ical.h>
-#include <icalss.h>
+#include <libical/ical.h>
+#include <libical/icalss.h>
 
 #include "orage-i18n.h"
 #include "functions.h"
--- orage-4.7.5/src/ical-archive.c.orig	2010-03-04 07:34:55.000000000 +0000
+++ orage-4.7.5/src/ical-archive.c	2010-07-04 15:51:36.000000000 +0100
@@ -51,8 +51,8 @@
 #include <gtk/gtk.h>
 #include <glib/gprintf.h>
 #include <glib/gstdio.h>
-#include <ical.h>
-#include <icalss.h>
+#include <libical/ical.h>
+#include <libical/icalss.h>
 
 #include "orage-i18n.h"
 #include "functions.h"
