--- Thunar-1.0.2/Makefile.in.orig	2010-05-21 18:11:02.000000000 +0100
+++ Thunar-1.0.2/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -1040,7 +1040,7 @@
 
 ThunarHelp: ThunarHelp.in Makefile
 	rm -f ThunarHelp.gen ThunarHelp
-	sed -e "s,\@datadir\@,$(datadir),g" 				\
+	sed -e "s,\@htmldir\@,$(htmldir),g" 				\
 		< $(srcdir)/ThunarHelp.in				\
 		> ThunarHelp.gen
 	mv ThunarHelp.gen ThunarHelp
--- Thunar-1.0.2/docs/Makefile.in.orig	2010-05-21 18:10:57.000000000 +0100
+++ Thunar-1.0.2/docs/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -322,7 +322,7 @@
 builddir = @builddir@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(datadir)/doc/Thunar
+docdir = @docdir@
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
--- Thunar-1.0.2/docs/manual/it/images/Makefile.in.orig	2010-05-21 18:10:58.000000000 +0100
+++ Thunar-1.0.2/docs/manual/it/images/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -314,7 +314,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/it/images
+imagesdir = $(htmldir)/it/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
--- Thunar-1.0.2/docs/manual/it/Makefile.in.orig	2010-05-21 18:10:58.000000000 +0100
+++ Thunar-1.0.2/docs/manual/it/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -334,7 +334,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/it
+TARGET_DIR = $(htmldir)/it
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
--- Thunar-1.0.2/docs/manual/tr/images/Makefile.in.orig	2010-05-21 18:10:59.000000000 +0100
+++ Thunar-1.0.2/docs/manual/tr/images/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -314,7 +314,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/tr/images
+imagesdir = $(htmldir)/tr/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
--- Thunar-1.0.2/docs/manual/tr/Makefile.in.orig	2010-05-21 18:10:59.000000000 +0100
+++ Thunar-1.0.2/docs/manual/tr/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -334,7 +334,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/tr
+TARGET_DIR = $(htmldir)/tr
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
--- Thunar-1.0.2/docs/manual/zh_TW/images/Makefile.in.orig	2010-05-21 18:10:59.000000000 +0100
+++ Thunar-1.0.2/docs/manual/zh_TW/images/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/zh_TW/images
+imagesdir = $(htmldir)/zh_TW/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
--- Thunar-1.0.2/docs/manual/zh_TW/Makefile.in.orig	2010-05-21 18:10:59.000000000 +0100
+++ Thunar-1.0.2/docs/manual/zh_TW/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/zh_TW
+TARGET_DIR = $(htmldir)/zh_TW
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
--- Thunar-1.0.2/docs/manual/C/images/Makefile.in.orig	2010-05-21 18:10:57.000000000 +0100
+++ Thunar-1.0.2/docs/manual/C/images/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/C/images
+imagesdir = $(htmldir)/C/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
--- Thunar-1.0.2/docs/manual/C/Makefile.in.orig	2010-05-21 18:10:57.000000000 +0100
+++ Thunar-1.0.2/docs/manual/C/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/C
+TARGET_DIR = $(htmldir)/C
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
--- Thunar-1.0.2/docs/manual/es/images/Makefile.in.orig	2010-05-21 18:10:58.000000000 +0100
+++ Thunar-1.0.2/docs/manual/es/images/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/es/images
+imagesdir = $(htmldir)/es/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
--- Thunar-1.0.2/docs/manual/es/Makefile.in.orig	2010-05-21 18:10:58.000000000 +0100
+++ Thunar-1.0.2/docs/manual/es/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/es
+TARGET_DIR = $(htmldir)/es
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
--- Thunar-1.0.2/docs/manual/eu/images/Makefile.in.orig	2010-05-21 18:10:58.000000000 +0100
+++ Thunar-1.0.2/docs/manual/eu/images/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/eu/images
+imagesdir = $(htmldir)/eu/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
--- Thunar-1.0.2/docs/manual/eu/Makefile.in.orig	2010-05-21 18:10:58.000000000 +0100
+++ Thunar-1.0.2/docs/manual/eu/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/eu
+TARGET_DIR = $(htmldir)/eu
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
--- Thunar-1.0.2/docs/manual/fr/images/Makefile.in.orig	2010-05-21 18:10:58.000000000 +0100
+++ Thunar-1.0.2/docs/manual/fr/images/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/fr/images
+imagesdir = $(htmldir)/fr/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
--- Thunar-1.0.2/docs/manual/fr/Makefile.in.orig	2010-05-21 18:10:58.000000000 +0100
+++ Thunar-1.0.2/docs/manual/fr/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/fr
+TARGET_DIR = $(htmldir)/fr
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
--- Thunar-1.0.2/docs/manual/Makefile.in.orig	2010-05-21 18:10:57.000000000 +0100
+++ Thunar-1.0.2/docs/manual/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -370,7 +370,7 @@
 	tr								\
 	zh_TW
 
-cssdir = $(datadir)/doc/Thunar/html
+cssdir = $(htmldir)
 css_DATA = \
 	thunar.css
 
--- Thunar-1.0.2/docs/manual/nl/images/Makefile.in.orig	2010-05-21 18:10:59.000000000 +0100
+++ Thunar-1.0.2/docs/manual/nl/images/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/nl/images
+imagesdir = $(htmldir)/nl/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
--- Thunar-1.0.2/docs/manual/nl/Makefile.in.orig	2010-05-21 18:10:59.000000000 +0100
+++ Thunar-1.0.2/docs/manual/nl/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/nl
+TARGET_DIR = $(htmldir)/nl
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
--- Thunar-1.0.2/docs/manual/gl/images/Makefile.in.orig	2010-05-21 18:10:58.000000000 +0100
+++ Thunar-1.0.2/docs/manual/gl/images/Makefile.in	2010-07-04 14:12:59.000000000 +0100
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/gl/images
+imagesdir = $(htmldir)/C/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
--- Thunar-1.0.2/docs/manual/gl/Makefile.in.orig	2010-05-21 18:10:58.000000000 +0100
+++ Thunar-1.0.2/docs/manual/gl/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/gl
+TARGET_DIR = $(htmldir)/gl
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
--- Thunar-1.0.2/docs/manual/ru/images/Makefile.in.orig	2010-05-21 18:10:59.000000000 +0100
+++ Thunar-1.0.2/docs/manual/ru/images/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/ru/images
+imagesdir = $(htmldir)/ru/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
--- Thunar-1.0.2/docs/manual/ru/Makefile.in.orig	2010-05-21 18:10:59.000000000 +0100
+++ Thunar-1.0.2/docs/manual/ru/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/ru
+TARGET_DIR = $(htmldir)/ru
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
--- Thunar-1.0.2/docs/manual/am/images/Makefile.in.orig	2010-05-21 18:10:58.000000000 +0100
+++ Thunar-1.0.2/docs/manual/am/images/Makefile.in	2010-07-04 14:28:57.000000000 +0100
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/am/images
+imagesdir = $(htmldir)/am/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
--- Thunar-1.0.2/docs/manual/am/Makefile.in.orig	2010-05-21 18:10:58.000000000 +0100
+++ Thunar-1.0.2/docs/manual/am/Makefile.in	2010-07-04 14:27:49.000000000 +0100
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/am
+TARGET_DIR = $(htmldir)/am
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
--- Thunar-1.0.2/docs/manual/da/images/Makefile.in.orig	2010-05-21 18:10:58.000000000 +0100
+++ Thunar-1.0.2/docs/manual/da/images/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -314,7 +314,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/da/images
+imagesdir = $(htmldir)/da/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png					\
--- Thunar-1.0.2/docs/manual/da/Makefile.in.orig	2010-05-21 18:10:58.000000000 +0100
+++ Thunar-1.0.2/docs/manual/da/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -334,7 +334,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/da
+TARGET_DIR = $(htmldir)/da
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
--- Thunar-1.0.2/docs/manual/pl/images/Makefile.in.orig	2010-05-21 18:10:59.000000000 +0100
+++ Thunar-1.0.2/docs/manual/pl/images/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/pl/images
+imagesdir = $(htmldir)/pl/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
--- Thunar-1.0.2/docs/manual/pl/Makefile.in.orig	2010-05-21 18:10:59.000000000 +0100
+++ Thunar-1.0.2/docs/manual/pl/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/pl
+TARGET_DIR = $(htmldir)/pl
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
--- Thunar-1.0.2/docs/manual/ja/images/Makefile.in.orig	2010-05-21 18:10:59.000000000 +0100
+++ Thunar-1.0.2/docs/manual/ja/images/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/ja/images
+imagesdir = $(htmldir)/ja/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
--- Thunar-1.0.2/docs/manual/ja/Makefile.in.orig	2010-05-21 18:10:59.000000000 +0100
+++ Thunar-1.0.2/docs/manual/ja/Makefile.in	2010-07-04 14:11:45.000000000 +0100
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/ja
+TARGET_DIR = $(htmldir)/ja
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
--- Thunar-1.0.2/ThunarHelp.in.orig	2010-05-21 18:10:52.000000000 +0100
+++ Thunar-1.0.2/ThunarHelp.in	2010-07-04 14:11:45.000000000 +0100
@@ -19,7 +19,7 @@
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 #
 
-HELPDIR="@datadir@/doc/Thunar/html/"
+HELPDIR="@htmldir@/"
 
 if test -n "$LC_ALL"; then
   LC=$LC_ALL
