--- calibre-2.0.0/src/unrar/pathfn.cpp.orig	2014-08-22 04:10:28.000000000 +0100
+++ calibre-2.0.0/src/unrar/pathfn.cpp	2014-08-23 16:29:16.450862047 +0100
@@ -448,7 +448,7 @@
   return(true);
 #elif defined(_UNIX)
   static const char *AltPath[]={
-    "/etc","/etc/rar","/usr/lib","/usr/local/lib","/usr/local/etc"
+    "/etc","/etc/rar","/usr/lib64","/usr/local/lib64","/usr/local/etc"
   };
   if (Number==0)
   {
--- calibre-2.0.0/setup/build_environment.py.orig	2014-08-22 04:10:28.000000000 +0100
+++ calibre-2.0.0/setup/build_environment.py	2014-08-23 16:52:45.563231316 +0100
@@ -114,7 +114,7 @@
 jpg_libs = []
 jpg_lib_dirs = []
 podofo_inc = '/usr/include/podofo'
-podofo_lib = '/usr/lib'
+podofo_lib = '/usr/lib64'
 chmlib_inc_dirs = chmlib_lib_dirs = []
 sqlite_inc_dirs = []
 icu_inc_dirs = []
