--- Coherence-0.6.6.2.x/coherence/upnp/core/utils.py.orig	2010-01-02 15:10:21.000000000 +0000
+++ Coherence-0.6.6.2.x/coherence/upnp/core/utils.py	2015-12-22 21:43:35.579256795 +0000
@@ -6,6 +6,7 @@
 
 import urlparse
 from urlparse import urlsplit
+from urlparse import urlparse
 
 from coherence.extern.et import parse_xml as et_parse_xml
 
@@ -585,7 +586,10 @@
 
     See HTTPClientFactory to see what extra args can be passed.
     """
-    scheme, host, port, path = client._parse(url)
+    parsed = urlparse(url)
+    scheme, host= parsed[:2]
+    port = parsed.port or (443 if scheme == 'https' else 80)
+    path = parsed.path or '/'
     factory = HeaderAwareHTTPClientFactory(url, *args, **kwargs)
     if scheme == 'https':
         from twisted.internet import ssl
@@ -604,7 +608,10 @@
 
     See HTTPDownloader to see what extra args can be passed.
     """
-    scheme, host, port, path = client._parse(url)
+    parsed = urlparse(url)
+    scheme, host= parsed[:2]
+    port = parsed.port or (443 if scheme == 'https' else 80)
+    path = parsed.path or '/'
     factory = HeaderAwareHTTPDownloader(url, file, *args, **kwargs)
     factory.noisy = False
     if scheme == 'https':
