--- gramps-3.4.5/src/plugins/lib/libgedcom.py.orig	2013-05-22 10:58:26.000000000 +0100
+++ gramps-3.4.5/src/plugins/lib/libgedcom.py	2013-06-02 06:46:57.705842180 +0100
@@ -6273,9 +6273,9 @@
             (file_ok, filename) = self.__find_file(line.data, self.dir_path)
             if state.media != "URL":
                 if not file_ok:
-                    self.__add_msg(_("Could not import %s") % filename[0], line,
+                    self.__add_msg(_("Could not import %s") % filename, line,
                                    state)
-            path = filename[0].replace('\\', os.path.sep)
+            path = filename.replace('\\', os.path.sep)
         else:
             path = line.data
             
