--- pygame-1.9.1release/Setup.in.orig	2009-07-02 05:41:56.000000000 +0100
+++ pygame-1.9.1release/Setup.in	2010-06-20 06:47:33.000000000 +0100
@@ -16,7 +16,7 @@
 JPEG = -ljpeg
 SCRAP = -lX11
 PORTMIDI = -lportmidi
-PORTTIME = -lporttime
+PORTTIME = -lportmidi
 #--EndConfig
 
 #DEBUG = -C-W -C-Wall
--- pygame-1.9.1release/config_unix.py.orig	2009-05-26 22:15:24.000000000 +0100
+++ pygame-1.9.1release/config_unix.py	2010-06-20 06:45:25.000000000 +0100
@@ -146,7 +146,7 @@
         Dependency('JPEG', 'jpeglib.h', 'libjpeg', ['jpeg']),
         Dependency('SCRAP', '', 'libX11', ['X11']),
         Dependency('PORTMIDI', 'portmidi.h', 'libportmidi.so', ['portmidi']),
-        Dependency('PORTTIME', 'porttime.h', 'libporttime.so', ['porttime']),
+        Dependency('PORTTIME', 'porttime.h', 'libportmidi.so', ['portmidi']),
         #Dependency('GFX', 'SDL_gfxPrimitives.h', 'libSDL_gfx.so', ['SDL_gfx']),
     ]
     if not DEPS[0].found:
