--- Shiboken-1.2.2/sources/shiboken/tests/libsample/simplefile.cpp.orig	2014-04-24 08:28:02.000000000 +0100
+++ Shiboken-1.2.2/sources/shiboken/tests/libsample/simplefile.cpp	2019-07-13 20:03:29.049869612 +0100
@@ -90,13 +90,13 @@
 SimpleFile::exists() const
 {
     std::ifstream ifile(p->m_filename);
-    return ifile;
+    return !!ifile;
 }
 
 bool
 SimpleFile::exists(const char* filename)
 {
     std::ifstream ifile(filename);
-    return ifile;
+    return !!ifile;
 }
 
