--- linphoneqt-4.1.1/CMakeLists.txt.orig	2017-07-21 14:14:18.000000000 +0100
+++ linphoneqt-4.1.1/CMakeLists.txt	2017-07-22 16:19:19.238985290 +0100
@@ -429,7 +429,7 @@
   "Qml" REGULAR_EXPRESSION ".+\.qml$"
 )
 source_group(
-  "Js" REGULAR_EXPRESSION ".+\.js)$"
+  "Js" REGULAR_EXPRESSION ".+\.js$"
 )
 source_group(
   "Svg" REGULAR_EXPRESSION ".+\.svg$"
--- linphoneqt-4.1.1/src/app/AppController.cpp.orig	2017-07-21 13:25:50.000000000 +0100
+++ linphoneqt-4.1.1/src/app/AppController.cpp	2017-07-22 16:28:56.293942373 +0100
@@ -28,6 +28,10 @@
 
 #include "AppController.hpp"
 
+#ifndef LINPHONE_QT_GIT_VERSION
+#define LINPHONE_QT_GIT_VERSION "unknown"
+#endif
+
 // Must be unique. Used by `SingleApplication` and `Paths`.
 #define APPLICATION_NAME "linphone"
 #define APPLICATION_VERSION LINPHONE_QT_GIT_VERSION
