--- linphone-desktop-5.2.4/linphone-sdk/external/openldap/libraries/libldap/thr_posix.c.orig	2024-06-18 22:16:41.000000000 +0100
+++ linphone-desktop-5.2.4/linphone-sdk/external/openldap/libraries/libldap/thr_posix.c	2024-06-19 16:44:46.788238193 +0100
@@ -38,6 +38,7 @@
 #define LDAP_THREAD_RDWR_IMPLEMENTATION
 #include "ldap_thr_debug.h"	 /* May rename the symbols defined below */
 #include <signal.h>			 /* For pthread_kill() */
+#include <unistd.h>
 
 extern int ldap_int_stackguard;
 
