--- linphone-desktop-6.0.1/external/linphone-sdk/external/gsm/inc/gsm.h.orig	2025-10-22 11:44:18.000000000 +0100
+++ linphone-desktop-6.0.1/external/linphone-sdk/external/gsm/inc/gsm.h	2025-10-22 11:44:18.000000000 +0100
@@ -4,7 +4,7 @@
  * details.  THERE IS ABSOLUTELY NO WARRANTY FOR THIS SOFTWARE.
  */
 
-/*$Header: /tmp_amd/presto/export/kbs/jutta/src/gsm/RCS/gsm.h,v 1.10 1996/07/02 10:15:22 jutta Exp $*/
+/*$Header: /home/kbs/jutta/src/gsm/gsm-1.0/inc/RCS/gsm.h,v 1.11 1996/07/05 18:02:56 jutta Exp $*/
 
 #ifndef	GSM_H
 #define	GSM_H
@@ -43,7 +43,7 @@
 
 #define	GSM_MAGIC		0xD		  	/* 13 kbit/s RPE-LTP */
 
-#define	GSM_PATCHLEVEL		9
+#define	GSM_PATCHLEVEL		10
 #define	GSM_MINOR		0
 #define	GSM_MAJOR		1
 
--- linphone-desktop-6.0.1/Linphone/CMakeLists.txt.orig	2025-10-22 11:43:12.000000000 +0100
+++ linphone-desktop-6.0.1/Linphone/CMakeLists.txt	2025-10-22 15:40:01.049403939 +0100
@@ -42,7 +42,7 @@
 #						CONFIGS
 ################################################################
 if(NOT WIN32)
-	add_compile_options(-Werror=deprecated-declarations)
+#	add_compile_options(-Werror=deprecated-declarations)
 endif()
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -DNDEBUG -DQT_NO_DEBUG")
@@ -73,7 +73,7 @@
 
 set(LINPHONESDK_VERSION)
 execute_process(
-    COMMAND git describe
+    COMMAND git describe --always
     WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/external/linphone-sdk
     OUTPUT_VARIABLE LINPHONESDK_VERSION
     OUTPUT_STRIP_TRAILING_WHITESPACE
