--- linphone-desktop-5.0.12/linphone-sdk/bzrtp/src/bzrtp.c.orig	2023-03-12 14:33:46.975813057 +0000
+++ linphone-desktop-5.0.12/linphone-sdk/bzrtp/src/bzrtp.c	2023-03-12 15:17:58.513882862 +0000
@@ -740,6 +740,9 @@
  *
  * @return number of supported types, 0 on error
  */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic warning "-Wstringop-overflow"
+
 uint8_t bzrtp_getSupportedCryptoTypes(bzrtpContext_t *zrtpContext, uint8_t algoType, uint8_t supportedTypes[7])
 {
 	if (zrtpContext==NULL) {
@@ -762,6 +765,8 @@
 	}
 }
 
+#pragma GCC diagnostic pop
+
 /**
  * @brief set the supported crypto types
  *
