--- mediastreamer-2.16.1/tools/mediastream.c.orig	2020-09-04 21:41:57.290322541 +0100
+++ mediastreamer-2.16.1/tools/mediastream.c	2020-09-04 21:49:47.333441930 +0100
@@ -1294,7 +1294,7 @@
 		if (*end != '\0' || end == addr) {
 			return FALSE;
 		}
-		strncpy(ip,localhost, MIN(len, strlen(localhost)));
+		strncpy(ip,localhost, MIN(len-1, strlen(localhost)+1));
 		ip[len-1] = '\0';
 		return TRUE;
 	}
--- mediastreamer-2.16.1/src/audiofilters/pulseaudio.c.orig	2021-10-27 09:47:10.760169268 +0100
+++ mediastreamer-2.16.1/src/audiofilters/pulseaudio.c	2021-10-27 09:52:55.294205343 +0100
@@ -227,7 +227,7 @@
  * output: sink and bidirectionnal cards into sink list, source only card into source list
  * merging is based on pulse audio card description */
 int pulse_card_compare(pa_device_t *sink, pa_device_t *source) {
-	return strncmp(sink->description, source->description, 512);
+	return strncmp(sink->description, source->description, PA_STRING_SIZE);
 }
 
 static void pulse_card_merge_lists(pa_device_t *pa_device, bctbx_list_t **pa_source_list) {
