--- mediastreamer-2.16.1/tools/mediastream.c.orig	2020-09-04 21:41:57.290322541 +0100
+++ mediastreamer-2.16.1/tools/mediastream.c	2020-09-04 21:49:47.333441930 +0100
@@ -1294,7 +1294,7 @@
 		if (*end != '\0' || end == addr) {
 			return FALSE;
 		}
-		strncpy(ip,localhost, MIN(len, strlen(localhost)));
+		strncpy(ip,localhost, MIN(len-1, strlen(localhost)+1));
 		ip[len-1] = '\0';
 		return TRUE;
 	}
