--- mediastreamer-2.16.1/src/base/msfactory.c.orig	2017-07-20 14:56:05.000000000 +0100
+++ mediastreamer-2.16.1/src/base/msfactory.c	2017-07-22 14:03:20.104214815 +0100
@@ -20,13 +20,12 @@
 #ifdef HAVE_CONFIG_H
 #include "mediastreamer-config.h"
 #include "gitversion.h"
-#else
-#   ifndef MEDIASTREAMER_VERSION
-#   define MEDIASTREAMER_VERSION "unknown"
-#   endif
-#	ifndef MS2_GIT_VERSION
-#	define MS2_GIT_VERSION "unknown"
-#	endif
+#endif
+#ifndef MEDIASTREAMER_VERSION
+#define MEDIASTREAMER_VERSION "unknown"
+#endif
+#ifndef MS2_GIT_VERSION
+#define MS2_GIT_VERSION "unknown"
 #endif
 
 #include "mediastreamer2/msfilter.h"
