--- calligra-25.08.2/filters/karbon/pdf/SvgOutputDev.cpp.orig	2025-10-07 04:21:16.000000000 +0100
+++ calligra-25.08.2/filters/karbon/pdf/SvgOutputDev.cpp	2025-10-09 19:21:13.678869396 +0100
@@ -390,7 +390,7 @@
         return;
 
     // ignore empty strings
-    if (s->getLength() == 0)
+    if (s->size() == 0)
         return;
 
 #if POPPLER_VERSION_MACRO < QT_VERSION_CHECK(22, 04, 0)
@@ -402,7 +402,7 @@
     QString str;
 
     const char *p = s->c_str();
-    int len = s->getLength();
+    int len = s->size();
     CharCode code;
     const Unicode *u = nullptr;
     int uLen;
