--- cantor-26.04.0/src/backends/CMakeLists.txt.orig	2026-04-10 04:53:48.000000000 +0100
+++ cantor-26.04.0/src/backends/CMakeLists.txt	2026-05-04 11:28:07.882311110 +0100
@@ -47,7 +47,7 @@
 
 find_package(R)
 if(R_FOUND)
-  add_subdirectory(R)
+#  add_subdirectory(R)
 endif(R_FOUND)
 
 set_package_properties(Qalculate PROPERTIES DESCRIPTION "A multi-purpose desktop calculator with support for customizable functions, units and arbitrary precision."
