--- cantor-26.04.0/src/backends/julia/juliaserver/CMakeLists.txt.orig	2026-04-10 04:53:48.000000000 +0100
+++ cantor-26.04.0/src/backends/julia/juliaserver/CMakeLists.txt	2026-04-17 14:32:52.205569751 +0100
@@ -11,6 +11,7 @@
     ${JULIA_INTERNAL_LIBRARY}
     Qt6::Widgets
     Qt6::DBus
+    pcre2-8
 )
 
 install(TARGETS cantor_juliaserver ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
--- cantor-26.04.0/src/backends/R/rserver/rserver.cpp.orig	2026-04-10 04:53:48.000000000 +0100
+++ cantor-26.04.0/src/backends/R/rserver/rserver.cpp	2026-05-04 11:48:20.308309319 +0100
@@ -144,7 +144,7 @@
 
     /* delayedAssign call*/
     PROTECT(da = Rf_findFun(Rf_install("delayedAssign"), R_GlobalEnv));
-    PROTECT(AutoloadEnv = Rf_findVar(Rf_install(".AutoloadEnv"), R_GlobalEnv));
+    PROTECT(AutoloadEnv = R_getVar(Rf_install(".AutoloadEnv"), R_GlobalEnv, TRUE));
     if (AutoloadEnv == R_NilValue){
         qDebug()<<"RServer: "<<"Cannot find .AutoloadEnv";
         //exit(1);
@@ -202,7 +202,9 @@
     // if (completion is needed) // TODO: discuss how to pass parameter
     {
         /* First check if namespace is loaded */
-        if  (findVarInFrame(R_NamespaceRegistry,install("utils"))==R_UnboundValue)
+//        if  (findVarInFrame(R_NamespaceRegistry,install("utils"))==R_UnboundValue)
+        SEXP ut = install("utils");
+        if  (R_getVarEx(ut, R_getRegisteredNamespace("utils"), FALSE, R_UnboundValue)==R_UnboundValue)
         { /* Then try to load it */
             SEXP cmdSexp, cmdexpr;
             ParseStatus status;
@@ -218,7 +220,8 @@
             }
             UNPROTECT(2);
             /* Completion is available if the namespace is correctly loaded */
-            m_isCompletionAvailable= (findVarInFrame(R_NamespaceRegistry,install("utils"))!=R_UnboundValue);
+//            m_isCompletionAvailable= (findVarInFrame(R_NamespaceRegistry,install("utils"))!=R_UnboundValue);
+            m_isCompletionAvailable= (R_getVarEx(ut, R_getRegisteredNamespace("utils"), FALSE, R_UnboundValue)!=R_UnboundValue);
         }
     }
 }
@@ -444,7 +447,7 @@
     {
         SEXP object = STRING_ELT(usr,i);
         const QString& name = QString::fromUtf8(translateCharUTF8(object));
-        SEXP value = findVar(installChar(object), R_GlobalEnv);
+        SEXP value = R_getVar(installChar(object), R_GlobalEnv, TRUE);
 
         if (Rf_isFunction(value))
             funcs << name;
--- cantor-26.08.1/src/backends/julia/juliaserver/juliaserver.cpp.orig	2026-09-08 04:38:02.000000000 +0100
+++ cantor-26.08.1/src/backends/julia/juliaserver/juliaserver.cpp	2026-09-12 12:18:55.860394044 +0100
@@ -60,7 +60,7 @@
     );
 
     jl_module_t* jl_repl_module = (jl_module_t*)(jl_eval_string("REPL"));
-    jl_function_t* jl_ends_func = jl_get_function(jl_repl_module, "ends_with_semicolon");
+    jl_value_t* jl_ends_func = jl_get_function(jl_repl_module, "ends_with_semicolon");
     bool isEndsWithSemicolon = jl_unbox_bool(jl_call1(jl_ends_func, jl_cstr_to_string(command.toStdString().c_str())));
 
     // Run command
@@ -78,7 +78,7 @@
         jl_value_t *ex = jl_exception_in_transit;
 #endif
         jl_printf(JL_STDERR, "error during run:\n");
-        jl_function_t *showerror =
+        jl_value_t *showerror =
             jl_get_function(jl_base_module, "showerror");
         jl_value_t *bt = static_cast<jl_value_t *>(
             jl_eval_string("catch_backtrace()")
@@ -91,7 +91,7 @@
         m_was_exception = true;
     } else if (val && !isEndsWithSemicolon) { // no exception occurred
         // If last result is not nothing, show it
-        jl_function_t *equality = jl_get_function(jl_base_module, "==");
+        jl_value_t *equality = jl_get_function(jl_base_module, "==");
         jl_value_t *nothing =
             static_cast<jl_value_t *>(jl_eval_string("nothing"));
         bool is_nothing = jl_unbox_bool(
@@ -99,7 +99,7 @@
         );
         if (!is_nothing) {
             jl_value_t *out_display = static_cast<jl_value_t *>(jl_eval_string("TextDisplay(stdout)"));
-            jl_function_t *display = jl_get_function(jl_base_module, "display");
+            jl_value_t *display = jl_get_function(jl_base_module, "display");
             jl_call2(display, out_display, val);
         }
         m_was_exception = false;
@@ -154,8 +154,8 @@
 
 void JuliaServer::parseJlModule(jl_module_t* module, bool parseValue)
 {
-    jl_function_t* jl_string_function = jl_get_function(jl_base_module, "string");
-    jl_function_t* jl_sizeof_function = jl_get_function(jl_base_module, "sizeof");
+    jl_value_t* jl_string_function = jl_get_function(jl_base_module, "string");
+    jl_value_t* jl_sizeof_function = jl_get_function(jl_base_module, "sizeof");
 
     if (module != JL_MAIN_MODULE)
     {
@@ -166,7 +166,7 @@
             parsedModules.append(moduleName);
     }
 
-    jl_function_t* jl_names_function = jl_get_function(jl_base_module, "names");
+    jl_value_t* jl_names_function = jl_get_function(jl_base_module, "names");
     jl_value_t* names = jl_call1(jl_names_function, (jl_value_t*)module);
 #if QT_VERSION_CHECK(JULIA_VERSION_MAJOR, JULIA_VERSION_MINOR, 0) >= QT_VERSION_CHECK(1, 11, 0)
     jl_value_t **data = (jl_value_t**)jl_array_data_(names);
