diff --git a/core/libs/database/coredb/coredburl.cpp b/core/libs/database/coredb/coredburl.cpp
index cb76c14438601334cf1db81b72dc60575ccbd9cd..a06008e61cc254f5c0c3542ad56c53a984d7dd72 100644
--- a/core/libs/database/coredb/coredburl.cpp
+++ b/core/libs/database/coredb/coredburl.cpp
@@ -248,7 +248,16 @@ CoreDbUrl& CoreDbUrl::operator=(const CoreDbUrl& url)
 
 bool CoreDbUrl::operator==(const QUrl& digikamalbumsUrl) const
 {
+#if (QT_VERSION >= QT_VERSION_CHECK(6, 8, 0))
+
+    return (this->matches(digikamalbumsUrl, QUrl::None));
+
+#else
+
     return (QUrl::operator==(digikamalbumsUrl));
+
+#endif
+
 }
 
 /*
