--- dragon-25.08.1/src/main.cpp.orig	2025-09-09 04:26:17.000000000 +0100
+++ dragon-25.08.1/src/main.cpp	2025-10-08 13:29:57.176756535 +0100
@@ -23,7 +23,7 @@
 
     if (Renderer::isAMD()) {
         // https://bugreports.qt.io/browse/QTBUG-138679
-#if QT_VERSION >= QT_VERSION_CHECK(6, 10, 0)
+#if QT_VERSION >= QT_VERSION_CHECK(6, 10, 1)
 #error "Reevaluate the use of QT_DISABLE_HW_TEXTURES_CONVERSION at a later point in time"
 #endif
         qWarning() << "Detected AMD GPU, disabling HW Texture Conversion renderer as it is known to cause issues.";
