--- heaptrack-20241110/CMakeLists.txt.orig	2024-11-10 16:02:17.000000000 +0000
+++ heaptrack-20241110/CMakeLists.txt	2025-08-17 16:20:53.159289244 +0100
@@ -32,7 +32,7 @@
 endif()
 
 include(FeatureSummary)
-find_package(Boost 1.60.0 ${REQUIRED_IN_APPIMAGE} COMPONENTS system filesystem iostreams container)
+find_package(Boost 1.60.0 ${REQUIRED_IN_APPIMAGE} COMPONENTS filesystem iostreams container)
 set_package_properties(Boost PROPERTIES TYPE RECOMMENDED PURPOSE "Boost container libraries can greatly improve performance (via pmr allocators)")
 find_package(Threads REQUIRED)
 find_package(ZLIB REQUIRED)
--- heaptrack-20241110/src/analyze/CMakeLists.txt.orig	2024-11-10 16:02:17.000000000 +0000
+++ heaptrack-20241110/src/analyze/CMakeLists.txt	2025-08-17 16:22:09.010059247 +0100
@@ -2,7 +2,7 @@
     include(ECMEnableSanitizers)
 endif()
 
-find_package(Boost 1.41.0 REQUIRED COMPONENTS iostreams program_options system filesystem)
+find_package(Boost 1.41.0 REQUIRED COMPONENTS iostreams program_options filesystem)
 
 configure_file(analyze_config.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/analyze_config.h)
 
