--- krita-6.0.2.1/cmake/modules/SIPMacros.cmake.orig	2026-05-29 22:17:03.000000000 +0100
+++ krita-6.0.2.1/cmake/modules/SIPMacros.cmake	2026-06-16 19:08:15.210502881 +0100
@@ -152,7 +152,7 @@
         endif()
 
         if (QT_MAJOR_VERSION STREQUAL "6")
-            set(abi_version "13.0")
+            set(abi_version "13.8")
             set(sip_disabled_features "[\"Krita_Qt5\"]")
         else()
             set(abi_version "12.8")
