--- ./cmake/modules/SIPMacros.cmake.orig	2026-05-21 05:41:34.000000000 -0500
+++ ./cmake/modules/SIPMacros.cmake	2026-05-31 15:52:30.251732269 -0500
@@ -152,7 +152,7 @@
         endif()
 
         if (QT_MAJOR_VERSION STREQUAL "6")
-            set(abi_version "13.0")
+            set(abi_version "13.8")
             set(sip_disabled_features "[\"Krita_Qt5\"]")
         else()
             set(abi_version "12.8")
