--- kstars-3.7.6/kstars/ekos/manager.cpp.orig	2025-03-31 15:05:54.000000000 +0100
+++ kstars-3.7.6/kstars/ekos/manager.cpp	2025-03-31 22:32:32.741183070 +0100
@@ -1200,7 +1200,7 @@
         profileScripts = doc.array();
         for (const auto &oneRule : qAsConst(profileScripts))
         {
-            auto totalDelay = (oneRule["PreDelay"].toDouble(0) + oneRule["PostDelay"].toDouble(0)) * 1000;
+            auto totalDelay = (oneRule.toObject()["PreDelay"].toDouble(0) + oneRule.toObject()["PostDelay"].toDouble(0)) * 1000;
             if (totalDelay >= maxTimeout)
                 maxTimeout = totalDelay + MAX_LOCAL_INDI_TIMEOUT;
         }
