--- kstars-3.7.4/kstars/data/fr/CMakeLists.txt.orig	2024-12-03 06:23:48.000000000 +0000
+++ kstars-3.7.4/kstars/data/fr/CMakeLists.txt	2024-12-16 22:26:57.579083559 +0000
@@ -1,5 +1,11 @@
 ########### install files ###############
+if(APPLE)
+    set(KSTARS_DATADIR "${BUNDLED_DATA_LOCATION}")
+else(APPLE)
+        set(KSTARS_DATADIR "${KDE_INSTALL_DATADIR}")
+endif(APPLE)
 
-install( FILES info_url.dat DESTINATION  ${DATA_INSTALL_DIR}/kstars/fr
+
+install( FILES info_url.dat DESTINATION  ${KSTARS_DATADIR}/kstars/fr
 )
 
--- kstars-3.7.4/kstars/data/nds/CMakeLists.txt.orig	2024-12-03 06:23:48.000000000 +0000
+++ kstars-3.7.4/kstars/data/nds/CMakeLists.txt	2024-12-16 22:27:15.763084780 +0000
@@ -1,5 +1,10 @@
 ########### install files ###############
+if(APPLE)
+    set(KSTARS_DATADIR "${BUNDLED_DATA_LOCATION}")
+else(APPLE)
+        set(KSTARS_DATADIR "${KDE_INSTALL_DATADIR}")
+endif(APPLE)
 
-install( FILES info_url.dat DESTINATION  ${DATA_INSTALL_DIR}/kstars/nds
+install( FILES info_url.dat DESTINATION  ${KSTARS_DATADIR}/kstars/nds
 )
 
