--- plasma-workspace-6.2.3/CMakeLists.txt.orig	2024-11-05 12:37:02.000000000 +0000
+++ plasma-workspace-6.2.3/CMakeLists.txt	2024-11-06 19:11:18.367007169 +0000
@@ -78,7 +78,7 @@
     TYPE RUNTIME
 )
 
-find_package(KF6KirigamiAddons2 0.10.0 CONFIG)
+find_package(KF6KirigamiAddons 0.10.0 CONFIG)
 set_package_properties(KF6KirigamiAddons2 PROPERTIES
     DESCRIPTION "Extra controls for Kirigami applications"
     PURPOSE "Required at runtime for the User KCM"
