--- attrs-23.1.0/pyproject.toml.orig	2023-04-16 11:22:15.000000000 +0100
+++ attrs-23.1.0/pyproject.toml	2023-04-18 19:10:26.213966257 +0100
@@ -8,7 +8,7 @@
 [project]
 name = "attrs"
 authors = [{ name = "Hynek Schlawack", email = "hs@ox.cx" }]
-license = "MIT"
+license = {file = "LICENSE.md"}
 requires-python = ">=3.7"
 description = "Classes Without Boilerplate"
 keywords = ["class", "attribute", "boilerplate"]
@@ -196,3 +196,6 @@
 [tool.mypy]
 disallow_untyped_defs = true
 check_untyped_defs = true
+
+[tool.setuptools.dynamic]
+readme = {file = ["README.md"]}
