--- calibre-6.2.1/setup/build_environment.py.orig	2022-07-29 09:42:04.000000000 +0100
+++ calibre-6.2.1/setup/build_environment.py	2022-07-30 15:36:47.855035350 +0100
@@ -119,7 +119,7 @@
 ft_libs = []
 ft_inc_dirs = []
 podofo_inc = '/usr/include/podofo'
-podofo_lib = '/usr/lib'
+podofo_lib = '/usr/lib64'
 chmlib_inc_dirs = chmlib_lib_dirs = []
 sqlite_inc_dirs = []
 icu_inc_dirs = []
@@ -183,10 +183,10 @@
     hunspell_lib_dirs = pkgconfig_lib_dirs('hunspell', 'HUNSPELL_LIB_DIR', '/usr/lib')
     sw = os.environ.get('SW', os.path.expanduser('~/sw'))
     podofo_inc = '/usr/include/podofo'
-    podofo_lib = '/usr/lib'
+    podofo_lib = '/usr/lib64'
     if not os.path.exists(podofo_inc + '/podofo.h'):
         podofo_inc = os.path.join(sw, 'include', 'podofo')
-        podofo_lib = os.path.join(sw, 'lib')
+        podofo_lib = os.path.join(sw, 'lib64')
     uchardet_inc_dirs = pkgconfig_include_dirs('uchardet', '', '/usr/include/uchardet')
     uchardet_lib_dirs = pkgconfig_lib_dirs('uchardet', '', '/usr/lib')
     uchardet_libs = pkgconfig_libs('uchardet', '', '')
