--- calibre-8.16.2/src/calibre/utils/ffmpeg.c.orig	2025-12-06 03:52:49.000000000 +0000
+++ calibre-8.16.2/src/calibre/utils/ffmpeg.c	2026-08-09 17:26:49.949625244 +0100
@@ -240,7 +240,7 @@
     av_channel_layout_default(&t->enc_ctx->ch_layout, t->dec_ctx->ch_layout.nb_channels);
     t->enc_ctx->sample_rate = t->dec_ctx->sample_rate;
     int ret;
-#if LIBAVCODEC_VERSION_MAJOR >= 61 && LIBAVCODEC_VERSION_MINOR >= 14
+#if LIBAVCODEC_VERSION_MAJOR >= 62 || (LIBAVCODEC_VERSION_MAJOR >= 61 && LIBAVCODEC_VERSION_MINOR >= 14)
     const enum AVSampleFormat *sample_fmts = NULL;
     ret = avcodec_get_supported_config(t->dec_ctx, output_codec, AV_CODEC_CONFIG_SAMPLE_FORMAT, 0, (const void**)&sample_fmts, NULL);
     t->enc_ctx->sample_fmt = (ret >= 0 && sample_fmts) ? sample_fmts[0] : t->dec_ctx->sample_fmt;
