--- calibre-6.16.0/src/calibre/utils/copy_files.py.orig	2023-04-21 03:50:25.000000000 +0100
+++ calibre-6.16.0/src/calibre/utils/copy_files.py	2023-04-23 22:19:38.016662676 +0100
@@ -127,7 +127,7 @@
             winutil.delete_file(make_long_path_useable(src_path))
 
 
-def get_copier() -> Union[UnixFileCopier | WindowsFileCopier]:
+def get_copier() -> Union[UnixFileCopier , WindowsFileCopier]:
     return WindowsFileCopier() if iswindows else UnixFileCopier()
 
 
