--- gramps-4.0.3/gramps/plugins/lib/libgedcom.py.orig	2014-01-27 14:26:49.000000000 +0000
+++ gramps-4.0.3/gramps/plugins/lib/libgedcom.py	2014-03-26 19:37:59.645108176 +0000
@@ -6347,7 +6347,8 @@
                 if not file_ok:
                     self.__add_msg(_("Could not import %s") % filename[0], line,
                                    state)
-            path = filename[0].replace('\\', os.path.sep)
+                    filename = line.data
+            path = filename.replace('\\', os.path.sep)
         else:
             path = line.data
             
