--- matplotlib-3.11.0/pyproject.toml.orig	2026-06-11 21:12:06.000000000 +0100
+++ matplotlib-3.11.0/pyproject.toml	2026-06-14 18:12:55.125827504 +0100
@@ -63,7 +63,7 @@
     "pybind11>=2.13.2,!=2.13.3",
     # setuptools_scm 10 breaks versioning in editable installs. You can remove this pin
     # if you're a downstream distributor just building wheels or your equivalent.
-    "setuptools_scm>=7,<10",
+    "setuptools_scm>=7,<11",
 ]
 
 [dependency-groups]
