--- mypaint-2.0.0/lib/meta.py.orig	2020-02-15 15:05:22.000000000 +0000
+++ mypaint-2.0.0/lib/meta.py	2020-05-13 19:56:29.346672399 +0100
@@ -264,7 +264,7 @@
         # If release information from release.sh exists, use that
         relinfo = {}
         with open("release_info", "rb") as relinfo_fp:
-            exec(relinfo_fp, relinfo)
+            exec(relinfo_fp.read(), relinfo)
         base_version = relinfo.get(
             "MYPAINT_VERSION_BASE",
             base_version,
--- mypaint-2.0.0/setup.cfg.orig	2020-02-15 15:05:23.000000000 +0000
+++ mypaint-2.0.0/setup.cfg	2020-02-16 11:28:53.833533177 +0000
@@ -8,8 +8,8 @@
 [install]
 verbose=1
 install-scripts=$base/bin
-install-platlib=$base/lib/mypaint
-install-purelib=$base/lib/mypaint
+install-platlib=$base/lib64/mypaint
+install-purelib=$base/lib64/mypaint
 install-data=$base/share
 install-headers=$base/include
 
--- mypaint-2.0.0/mypaint.py.orig	2020-02-15 15:05:22.000000000 +0000
+++ mypaint-2.0.0/mypaint.py	2020-02-16 11:29:00.951532838 +0000
@@ -168,7 +168,7 @@
     if os.path.exists(libpath) and os.path.exists(iconspath):
         # This is a normal POSIX-like installation.
         # The Windows standalone distribution works like this too.
-        libpath_compiled = join(prefix, 'lib', 'mypaint')  # or lib64?
+        libpath_compiled = join(prefix, 'lib64', 'mypaint')  # or lib64?
         sys.path.insert(0, libpath)
         sys.path.insert(0, libpath_compiled)
         sys.path.insert(0, join(prefix, 'share'))  # for libmypaint
