--- ocrmypdf-16.0.0/src/ocrmypdf/hocrtransform/_hocr.py.orig	2023-12-18 00:31:52.000000000 +0000
+++ ocrmypdf-16.0.0/src/ocrmypdf/hocrtransform/_hocr.py	2023-12-21 19:40:17.243741621 +0000
@@ -12,7 +12,7 @@
 import re
 import unicodedata
 from dataclasses import dataclass
-from itertools import pairwise
+from more_itertools import pairwise
 from math import atan, cos, pi
 from pathlib import Path
 from xml.etree import ElementTree
