--- picard-2.8/picard/ui/coverartbox.py.orig	2022-05-24 22:23:54.278349476 +0100
+++ picard-2.8/picard/ui/coverartbox.py	2022-05-24 22:25:20.959345340 +0100
@@ -481,7 +481,7 @@
                                        priority=True, important=True)
         elif url.scheme() == 'file':
             path = normpath(url.toLocalFile().rstrip("\0"))
-            if path and os.path.exists(path):
+            if path and os.path.exists(path) and os.path.isfile(path):
                 with open(path, 'rb') as f:
                     data = f.read()
                 self.load_remote_image(url, data)
