--- picard-2.6.2/picard/ui/coverartbox.py.orig	2021-04-27 10:13:00.000000000 +0100
+++ picard-2.6.2/picard/ui/coverartbox.py	2021-05-25 21:33:20.296240937 +0100
@@ -415,7 +415,7 @@
                                        priority=True, important=True)
         elif url.scheme() == 'file':
             path = os.path.normpath(os.path.realpath(url.toLocalFile().rstrip("\0")))
-            if path and os.path.exists(path):
+            if path and os.path.exists(path) and os.path.isfile(path):
                 with open(path, 'rb') as f:
                     data = f.read()
                 self.load_remote_image(url, data)
