--- plasma-disassembler-20191029/setup.py.orig	2019-10-29 21:30:37.000000000 +0000
+++ plasma-disassembler-20191029/setup.py	2020-10-26 12:19:33.431936742 +0000
@@ -22,8 +22,8 @@
         links.append(str(item.url))
     if getattr(item, 'link', None): # newer pip has link
         links.append(str(item.link))
-    if item.req:
-        requires.append(str(item.req))
+#    if item.req:
+#        requires.append(str(item.req))
 
 
 x86_analyzer = Extension('plasma.lib.arch.x86.analyzer',
