--- pygame-1.9.6/src_c/pypm.c.orig	2019-04-25 07:14:17.000000000 +0100
+++ pygame-1.9.6/src_c/pypm.c	2020-10-26 12:05:28.177864158 +0000
@@ -6416,7 +6416,9 @@
   /*--- Function export code ---*/
   /*--- Type init code ---*/
   if (PyType_Ready(&__pyx_type_4pypm_Output) < 0) __PYX_ERR(0, 248, __pyx_L1_error)
+  #if PY_VERSION_HEX < 0x03080000
   __pyx_type_4pypm_Output.tp_print = 0;
+  #endif
   #if CYTHON_COMPILING_IN_CPYTHON
   {
     PyObject *wrapper = PyObject_GetAttrString((PyObject *)&__pyx_type_4pypm_Output, "__init__"); if (unlikely(!wrapper)) __PYX_ERR(0, 248, __pyx_L1_error)
@@ -6431,7 +6433,9 @@
   if (__Pyx_setup_reduce((PyObject*)&__pyx_type_4pypm_Output) < 0) __PYX_ERR(0, 248, __pyx_L1_error)
   __pyx_ptype_4pypm_Output = &__pyx_type_4pypm_Output;
   if (PyType_Ready(&__pyx_type_4pypm_Input) < 0) __PYX_ERR(0, 509, __pyx_L1_error)
+  #if PY_VERSION_HEX < 0x03080000
   __pyx_type_4pypm_Input.tp_print = 0;
+  #endif
   #if CYTHON_COMPILING_IN_CPYTHON
   {
     PyObject *wrapper = PyObject_GetAttrString((PyObject *)&__pyx_type_4pypm_Input, "__init__"); if (unlikely(!wrapper)) __PYX_ERR(0, 509, __pyx_L1_error)
