--- pymupdf-1.28.2/src/extra.i.orig	2026-08-06 11:45:29.906516800 +0100
+++ pymupdf-1.28.2/src/extra.i	2026-08-15 10:43:29.188636459 +0100
@@ -225,7 +225,7 @@
     static PyObject* message_fn = PyObject_GetAttrString(pymupdf_module, "message");
     char* text;
     vasprintf(&text, format, va);
-    PyObject* text_py = PyString_FromString(text);
+    PyObject* text_py = PyUnicode_FromString(text);
     PyObject* args = PyTuple_Pack(1, text_py);
     PyObject* ret = PyObject_CallObject(message_fn, args);
     Py_XDECREF(ret);
