--- PyOpenGL-accelerate-3.1.5/src/vbo.c.orig	2020-01-04 02:07:10.000000000 +0000
+++ PyOpenGL-accelerate-3.1.5/src/vbo.c	2021-10-11 15:38:43.549813458 +0100
@@ -11128,9 +11128,9 @@
   {
     PyObject *etype, *eval, *etb;
     PyErr_Fetch(&etype, &eval, &etb);
-    ++Py_REFCNT(o);
+    Py_INCREF(o);
     __pyx_pw_17OpenGL_accelerate_3vbo_3VBO_3__dealloc__(o);
-    --Py_REFCNT(o);
+    Py_DECREF(o);
     PyErr_Restore(etype, eval, etb);
   }
   if (p->__weakref__) PyObject_ClearWeakRefs(o);
@@ -11616,9 +11616,9 @@
   {
     PyObject *etype, *eval, *etb;
     PyErr_Fetch(&etype, &eval, &etb);
-    ++Py_REFCNT(o);
+    Py_INCREF(o);
     __pyx_pw_17OpenGL_accelerate_3vbo_9VBOOffset_3__dealloc__(o);
-    --Py_REFCNT(o);
+    Py_DECREF(o);
     PyErr_Restore(etype, eval, etb);
   }
   Py_CLEAR(p->vbo);
