--- pypdf-3.9.0/setup.cfg.orig	2023-06-03 11:25:27.235825688 +0100
+++ pypdf-3.9.0/setup.cfg	2023-06-03 11:25:05.843826708 +0100
@@ -1,5 +1,6 @@
 [metadata]
 name = pypdf
+version = attr: pypdf.__version__
 
 author = Mathieu Fenniak
 author_email = biziqe@mathieu.fenniak.net
@@ -35,6 +36,7 @@
 packages =
     pypdf
     pypdf._codecs
+    pypdf._text_extraction
     pypdf.generic
 python_requires = >=3.6
 install_requires =
