--- pypdf-3.8.1/setup.cfg.orig	2023-02-05 10:26:02.030123000 +0000
+++ pypdf-3.8.1/setup.cfg	2023-05-03 22:09:30.348498990 +0100
@@ -1,5 +1,6 @@
 [metadata]
 name = pypdf
+version = attr: pypdf.__version__
 
 author = Mathieu Fenniak
 author_email = biziqe@mathieu.fenniak.net
@@ -35,6 +36,7 @@
 packages =
     pypdf
     pypdf._codecs
+    pypdf._text_extraction
     pypdf.generic
 python_requires = >=3.6
 install_requires =
