--- pyside-setup-opensource-src-5.15.17/sources/shiboken2/libshiboken/CMakeLists.txt.orig	2025-05-20 06:48:26.000000000 +0100
+++ pyside-setup-opensource-src-5.15.17/sources/shiboken2/libshiboken/CMakeLists.txt	2025-09-03 12:09:01.960057196 +0100
@@ -111,7 +111,7 @@
 )
 
 if (NOT "${PYTHON_NUMPY_LOCATION}" STREQUAL "")
-    target_include_directories(libshiboken PRIVATE ${PYTHON_NUMPY_LOCATION}/core/include)
+    target_include_directories(libshiboken PRIVATE ${PYTHON_NUMPY_LOCATION}/_core/include)
     target_compile_definitions(libshiboken PRIVATE -DHAVE_NUMPY
                                            PRIVATE -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION)
 
