--- pyside-setup-opensource-src-5.15.16/sources/pyside2-tools/CMakeLists.txt.orig	2020-01-16 13:58:11.000000000 +0000
+++ pyside-setup-opensource-src-5.15.16/sources/pyside2-tools/CMakeLists.txt	2025-03-09 11:24:56.645365479 +0000
@@ -46,7 +46,7 @@
     set(EXE_EXT "")
 endif()
 
-set(TOOLS_PATH "${_qt5Core_install_prefix}/bin")
+set(TOOLS_PATH "${_qt5Core_install_prefix}/lib$ENV{LIB_SUFFIX}/qt5/bin")
 set(UIC_PATH "${TOOLS_PATH}/uic${EXE_EXT}")
 set(RCC_PATH "${TOOLS_PATH}/rcc${EXE_EXT}")
 if (APPLE)
