--- scipy-1.11.3/tools/cythonize.py.orig	2023-09-28 20:49:45.388114300 +0100
+++ scipy-1.11.3/tools/cythonize.py	2023-09-28 20:49:59.187113642 +0100
@@ -70,9 +70,6 @@
                 if "cython" not in line.lower():
                     continue
 
-                if ">=" not in line:
-                    continue
-
                 # parse a string such as 'Cython>=x.y.z,<=a.b.c'
                 # Assume always >= and not a singleton >
                 # hence split at the first "=" and discard "Cython>" part
