--- beagle-0.3.8/search/Beagle.Search.Tiles/Tile.cs.orig	2008-06-12 06:41:19.000000000 +0100
+++ beagle-0.3.8/search/Beagle.Search.Tiles/Tile.cs	2008-09-27 19:22:13.000000000 +0100
@@ -348,7 +348,7 @@
 #else
 			MimeApplication app;
 			app = Mime.GetDefaultApplication (mimetype);
-			bool expect_uris = app.SupportUris ();
+			bool expect_uris = app.SupportsUris ();
 			path = hit.Path;
 
 			if (app == null) {
