--- beagle-0.2.15.1/tools/beagle-crawl-system.in.orig	2007-01-24 17:52:32.000000000 +0000
+++ beagle-0.2.15.1/tools/beagle-crawl-system.in	2007-01-26 22:00:57.000000000 +0000
@@ -67,7 +67,7 @@
                 IONICE="$IONICE -c 3"
             fi
 
-	    eval $IONICE su -s /bin/bash $CRAWL_USER -c \"MONO_SHARED_DIR=$MONO_SHARED_DIR @sbindir@/beagle-build-index --target @localstatedir@/cache/beagle/indexes/$CRAWL_INDEX_NAME $OPTIONS $CRAWL_PATHS\" > /dev/null 2>&1
+	    eval $IONICE su $CRAWL_USER -c \"MONO_SHARED_DIR=$MONO_SHARED_DIR @sbindir@/beagle-build-index --target @localstatedir@/cache/beagle/indexes/$CRAWL_INDEX_NAME $OPTIONS $CRAWL_PATHS\" > /dev/null 2>&1
 	fi
     fi
 done
