--- brasero-0.7.1/data/brasero.desktop.in.in.orig	2008-01-27 15:25:17.000000000 +0000
+++ brasero-0.7.1/data/brasero.desktop.in.in	2008-05-17 20:39:23.000000000 +0100
@@ -6,7 +6,7 @@
 Encoding=UTF-8
 Exec=brasero %U
 Icon=brasero
-MimeType=application/x-cd-image;application/x-cdrdao-toc;application/x-toc;application/x-cue;application/x-brasero;x-content/blank-cd:x-content/blank-dvd;x-content/blank-bd;x-content/blank-hddvd;
+MimeType=application/x-cd-image;application/x-cdrdao-toc;application/x-toc;application/x-cue;application/x-brasero;x-content/blank-cd;x-content/blank-dvd;x-content/blank-bd;x-content/blank-hddvd;
 StartupNotify=true
 Terminal=false
 Type=Application
