--- dates-0.4.11/Makefile.in.orig	2009-10-16 18:56:07.000000000 +0100
+++ dates-0.4.11/Makefile.in	2011-05-10 07:19:07.000000000 +0100
@@ -651,7 +651,7 @@
 
 
 snapshot:
-        $(MAKE) dist distdir=$(PACKAGE)-snap`date +"%Y%m%d"`
+	$(MAKE) dist distdir=$(PACKAGE)-snap`date +"%Y%m%d"`
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
