--- dia-0.97.2/objects/custom/shape_typeinfo.c.orig	2009-11-07 14:28:34.000000000 +0000
+++ dia-0.97.2/objects/custom/shape_typeinfo.c	2012-06-23 00:40:39.000000000 +0100
@@ -27,7 +27,7 @@
 #include "custom_util.h"
 #include <string.h>
 #include <stdarg.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <glib/gstdio.h>
 #include <libxml/parser.h>
 
--- dia-0.97.2/tests/test-objects.c.orig	2009-11-07 14:28:35.000000000 +0000
+++ dia-0.97.2/tests/test-objects.c	2012-06-23 01:05:28.000000000 +0100
@@ -29,7 +29,6 @@
 #include <glib-object.h>
 
 #if GLIB_CHECK_VERSION(2,16,0)
-#include <glib/gtestutils.h>
 #endif
 
 #include "object.h"
--- dia-0.97.2/tests/test-boundingbox.c.orig	2009-11-07 14:28:35.000000000 +0000
+++ dia-0.97.2/tests/test-boundingbox.c	2012-06-23 00:54:36.000000000 +0100
@@ -29,7 +29,7 @@
 #include <glib-object.h>
 
 #if GLIB_CHECK_VERSION(2,16,0)
-#include <glib/gtestutils.h>
+#include <glib.h>
 #endif
 #include "dialib.h"
 
