--- dia-20240523/meson.build.orig	2024-05-23 20:02:04.000000000 +0100
+++ dia-20240523/meson.build	2025-09-03 08:43:53.714063321 +0100
@@ -6,7 +6,7 @@
                              'warning_level=1',
                              'buildtype=debugoptimized',
                              'c_std=gnu17',
-                             'cpp_std=gnu++17',
+                             'cpp_std=gnu++20',
                          ],
 )
 
--- dia-20240523/plug-ins/pdf/pdf-import.cpp.orig	2024-05-23 20:02:04.000000000 +0100
+++ dia-20240523/plug-ins/pdf/pdf-import.cpp	2025-10-06 21:05:15.399073477 +0100
@@ -365,7 +365,11 @@
     if ((pf = strstr (family, " Oblique")) != NULL)
       *pf = 0;
 
+#if POPPLER_VERSION_MAJOR > 25 || (POPPLER_VERSION_MAJOR == 25 && POPPLER_VERSION_MINOR >= 9)
+    const double *fm = f->getFontMatrix().data();
+#else
     const double *fm = f->getFontMatrix();
+#endif
     double fsize = state->getTransformedFontSize();
     if (fm[0] != 0)
       fsize *= fabs(fm[3] / fm[0]);
@@ -725,7 +729,7 @@
 DiaOutputDev::drawString(GfxState *state, GooString *s)
 {
   Color text_color = this->fill_color;
-  int len = s->getLength();
+  int len = s->size();
   DiaObject *obj;
   gchar *utf8 = NULL;
   DiaFont *font;
