--- dia-20201021/plug-ins/pdf/pdf-import.cpp.orig	2020-10-21 21:20:33.000000000 +0100
+++ dia-20201021/plug-ins/pdf/pdf-import.cpp	2021-03-02 15:18:18.457183986 +0000
@@ -867,7 +867,7 @@
   // without this we will get strange crashes (at least with /O2 build)
   globalParams = std::make_unique<GlobalParams>();
 
-  doc = PDFDocFactory().createPDFDoc(*fileName, ownerPW, userPW);
+  doc = PDFDocFactory().createPDFDoc(*fileName, ownerPW, userPW).get();
   if (!doc->isOk()) {
     dia_context_add_message (ctx, _("PDF document not OK.\n%s"),
 			     dia_context_get_filename (ctx));
