--- gcab-1.5/meson.build.orig	2022-08-05 07:58:25.000000000 +0100
+++ gcab-1.5/meson.build	2022-09-22 16:14:53.696695439 +0100
@@ -8,7 +8,7 @@
 git_version = []
 git = find_program('git', required: false)
 if git.found()
-  git_version = run_command(git, 'describe', '--abbrev=4', '--dirty', check: true).stdout().strip().split('-')
+#  git_version = run_command(git, 'describe', '--abbrev=4', '--dirty', check: true).stdout().strip().split('-')
 endif
 
 # libtool versioning
