--- gir-repository-0.6.5/gir/Makefile.in.orig	2009-09-09 20:31:46.000000000 +0100
+++ gir-repository-0.6.5/gir/Makefile.in	2010-02-05 19:52:31.000000000 +0000
@@ -1479,6 +1479,7 @@
 @BUILD_GSTREAMER_PLUGINS_BASE_TRUE@             --include=Gst-0.10 \
 @BUILD_GSTREAMER_PLUGINS_BASE_TRUE@             --include=GstBase-0.10 \
 @BUILD_GSTREAMER_PLUGINS_BASE_TRUE@	     --library=gstvideo-0.10 \
+@BUILD_GSTREAMER_PLUGINS_BASE_TRUE@	     --library=gstbasevideo-0.10 \
 @BUILD_GSTREAMER_PLUGINS_BASE_TRUE@	     --libtool="$(LIBTOOL)" \
 @BUILD_GSTREAMER_PLUGINS_BASE_TRUE@             --pkg gobject-2.0 \
 @BUILD_GSTREAMER_PLUGINS_BASE_TRUE@             --pkg gstreamer-0.10 \
