--- glabels-3.4.1/src/template-history.h.orig	2016-03-04 04:01:15.000000000 +0000
+++ glabels-3.4.1/src/template-history.h	2021-08-17 21:38:22.752917960 +0100
@@ -28,7 +28,7 @@
 G_BEGIN_DECLS
 
 
-glTemplateHistoryModel *gl_template_history;
+extern glTemplateHistoryModel *gl_template_history;
 
 
 void            gl_template_history_init (void);
--- glabels-3.4.1/src/font-history.h.orig	2016-03-04 04:01:15.000000000 +0000
+++ glabels-3.4.1/src/font-history.h	2021-08-17 21:39:04.257915980 +0100
@@ -28,7 +28,7 @@
 G_BEGIN_DECLS
 
 
-glFontHistoryModel *gl_font_history;
+extern glFontHistoryModel *gl_font_history;
 
 
 void            gl_font_history_init (void);
