--- glade-3.10.3/configure.orig	2012-09-19 18:33:25.000000000 +0100
+++ glade-3.10.3/configure	2015-02-15 09:01:12.782966015 +0000
@@ -15631,10 +15631,10 @@
 $as_echo_n "checking for libraries required to embed python... " >&6; }
 py_exec_prefix=`$PYTHON -c "import sys; print sys.exec_prefix"`
 if test "x$PYTHON_LIBS" == x; then
-	PYTHON_LIBS="-L${py_prefix}/lib -lpython${PYTHON_VERSION}"
+	PYTHON_LIBS="-L${py_prefix}/lib64 -lpython${PYTHON_VERSION}"
 fi
 if test "x$PYTHON_LIB_LOC" == x; then
-	PYTHON_LIB_LOC="${py_prefix}/lib"
+	PYTHON_LIB_LOC="${py_prefix}/lib64"
 fi
 
 
