--- glib-2.38.2/gobject/Makefile.am.orig	2013-11-07 15:29:13.000000000 +0000
+++ glib-2.38.2/gobject/Makefile.am	2013-12-30 21:01:04.924277023 +0000
@@ -46,6 +46,7 @@
 libgobjectincludedir = $(includedir)/glib-2.0/gobject
 libgobject_2_0_la_LDFLAGS = $(GLIB_LINK_FLAGS) \
   $(gobject_win32_res_ldflag) \
+  -Wl,-z,nodelete \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
   -export-dynamic $(no_undefined)
 
