--- gnome-vfs-2.24.4/configure.in.orig	2010-07-01 16:36:09.000000000 +0100
+++ gnome-vfs-2.24.4/configure.in	2020-10-22 14:55:32.157642029 +0100
@@ -1241,7 +1241,7 @@
 dnl *** Checks for gtk-doc ***
 dnl **************************
 
-GTK_DOC_CHECK([1.0])
+GTK_DOC_CHECK([1.14],[--flavour no-tmpl])
 
 dnl ******************************
 dnl *** End checks for gtk-doc ***
--- gnome-vfs-2.24.4/libgnomevfs/gnome-vfs-job-queue.h.orig	2010-02-09 12:16:14.000000000 +0000
+++ gnome-vfs-2.24.4/libgnomevfs/gnome-vfs-job-queue.h	2020-10-22 15:01:51.388623935 +0100
@@ -20,7 +20,7 @@
    write to the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
    Boston, MA 02111-1307, USA.
 
-   Author: László Péter <laca@ireland.sun.com>
+   Author: Laszlo Peter <laca@ireland.sun.com>
 */
 
 #ifndef GNOME_VFS_JOB_QUEUE_H
--- gnome-vfs-2.24.4/libgnomevfs/gnome-vfs-ssl.c.orig	2010-02-09 12:16:14.000000000 +0000
+++ gnome-vfs-2.24.4/libgnomevfs/gnome-vfs-ssl.c	2020-10-21 22:24:18.937793768 +0100
@@ -400,8 +400,8 @@
 			}
 		}
 
-                if (ssl->private->ssl->ctx)
-                        SSL_CTX_free (ssl->private->ssl->ctx);
+//                if (ssl->private->ssl->ctx)
+//                        SSL_CTX_free (ssl->private->ssl->ctx);
 
                 SSL_free (ssl->private->ssl);
 		g_free (ssl->private);
@@ -705,7 +705,7 @@
 		}
 	}
 	
-	SSL_CTX_free (ssl->private->ssl->ctx);
+//	SSL_CTX_free (ssl->private->ssl->ctx);
 	SSL_free (ssl->private->ssl);
 	close (ssl->private->sockfd);
 	if (ssl->private->timeout)
