--- gnome-phone-manager-0.7/src/e-contact-entry.c-	2006-08-14 08:59:16.000000000 -0400
+++ gnome-phone-manager-0.7/src/e-contact-entry.c	2006-08-14 08:59:43.000000000 -0400
@@ -255,8 +255,8 @@
 
       loader = gdk_pixbuf_loader_new ();
 
-      if (gdk_pixbuf_loader_write (loader, (guchar *)photo->data,
-			      photo->length, NULL))
+      if (gdk_pixbuf_loader_write (loader, (guchar *)photo->data.inlined.data,
+			      photo->data.inlined.length, NULL))
         pixbuf = gdk_pixbuf_loader_get_pixbuf (loader);
 
       if (pixbuf) {
