--- gnome-power-manager-2.14.3/man/Makefile.in.orig	2006-04-25 10:45:38.000000000 +0100
+++ gnome-power-manager-2.14.3/man/Makefile.in	2006-04-27 14:35:00.000000000 +0100
@@ -433,10 +433,10 @@
 
 
 @HAVE_DOCBOOK2MAN_TRUE@gnome-power-manager.1: gnome-power-manager.sgml
-@HAVE_DOCBOOK2MAN_TRUE@	docbook2man $? > $@
+@HAVE_DOCBOOK2MAN_TRUE@	docbook2man $? 
 
 @HAVE_DOCBOOK2MAN_TRUE@gnome-power-preferences.1: gnome-power-preferences.sgml
-@HAVE_DOCBOOK2MAN_TRUE@	docbook2man $? > $@
+@HAVE_DOCBOOK2MAN_TRUE@	docbook2man $? 
 
 clean-local :
 	rm -f *~
