--- gst-ffmpeg-0.10.13/ext/libswscale/gstffmpegscale.c.orig	2011-11-02 13:04:05.000000000 +0000
+++ gst-ffmpeg-0.10.13/ext/libswscale/gstffmpegscale.c	2020-05-02 09:47:03.169894913 +0100
@@ -638,7 +638,7 @@
   swsflags = (mmx_flags & ORC_TARGET_MMX_MMX ? SWS_CPU_CAPS_MMX : 0)
       | (mmx_flags & ORC_TARGET_MMX_MMXEXT ? SWS_CPU_CAPS_MMX2 : 0)
       | (mmx_flags & ORC_TARGET_MMX_3DNOW ? SWS_CPU_CAPS_3DNOW : 0)
-      | (altivec_flags & ORC_TARGET_ALTIVEC_ALTIVEC ? SWS_CPU_CAPS_ALTIVEC : 0);
+      | (altivec_flags & ORC_TARGET_POWERPC_ALTIVEC ? SWS_CPU_CAPS_ALTIVEC : 0);
 #else
   mmx_flags = 0;
   altivec_flags = 0;
--- gst-ffmpeg-0.10.13/ext/libpostproc/gstpostproc.c.orig	2011-07-12 15:35:27.000000000 +0100
+++ gst-ffmpeg-0.10.13/ext/libpostproc/gstpostproc.c	2020-05-02 09:47:14.999894493 +0100
@@ -299,7 +299,7 @@
     ppflags = (mmx_flags & ORC_TARGET_MMX_MMX ? PP_CPU_CAPS_MMX : 0)
         | (mmx_flags & ORC_TARGET_MMX_MMXEXT ? PP_CPU_CAPS_MMX2 : 0)
         | (mmx_flags & ORC_TARGET_MMX_3DNOW ? PP_CPU_CAPS_3DNOW : 0)
-        | (altivec_flags & ORC_TARGET_ALTIVEC_ALTIVEC ? PP_CPU_CAPS_ALTIVEC :
+        | (altivec_flags & ORC_TARGET_POWERPC_ALTIVEC ? PP_CPU_CAPS_ALTIVEC :
         0);
 #else
     mmx_flags = 0;
