--- gst-plugins-bad-1.20.5/ext/opencv/meson.build.orig	2022-12-19 23:34:46.000000000 +0000
+++ gst-plugins-bad-1.20.5/ext/opencv/meson.build	2023-01-04 22:11:30.472862900 +0000
@@ -71,7 +71,7 @@
 endif
 
 if not opencv_found
-  opencv_dep = dependency('opencv4', version : ['>= 4.0.0', '< 4.7.0'], required : false)
+  opencv_dep = dependency('opencv4', version : ['>= 4.0.0', '< 4.8.0'], required : false)
   opencv_found = opencv_dep.found()
   if opencv_found
     foreach h : libopencv4_headers
