--- gst-plugins-bad-0.10.23/ext/vp8/gstvp8utils.h.orig	2011-12-11 18:49:11.000000000 +0000
+++ gst-plugins-bad-0.10.23/ext/vp8/gstvp8utils.h	2015-04-04 10:56:48.502991073 +0100
@@ -26,9 +26,9 @@
 G_BEGIN_DECLS
 
 /* Some compatibility defines for older libvpx versions */
-#ifndef VPX_IMG_FMT_I420
-#define VPX_IMG_FMT_I420 IMG_FMT_I420
-#endif
+//#ifndef VPX_IMG_FMT_I420
+//#define VPX_IMG_FMT_I420 IMG_FMT_I420
+//#endif
 
 #ifndef VPX_PLANE_Y
 #define VPX_PLANE_Y PLANE_Y
--- gst-plugins-bad-0.10.23/ext/opus/gstopusdec.c.orig	2011-12-30 18:24:28.000000000 +0000
+++ gst-plugins-bad-0.10.23/ext/opus/gstopusdec.c	2019-07-16 20:45:41.924099707 +0100
@@ -357,7 +357,7 @@
 
     GST_DEBUG_OBJECT (dec, "Creating decoder with %d channels, %d Hz",
         dec->n_channels, dec->sample_rate);
-#ifndef GST_DISABLE_DEBUG
+#ifndef GST_DISABLE_GST_DEBUG
     gst_opus_common_log_channel_mapping_table (GST_ELEMENT (dec), opusdec_debug,
         "Mapping table", dec->n_channels, dec->channel_mapping);
 #endif
--- gst-plugins-bad-0.10.23/ext/opus/gstopusenc.c.orig	2012-01-18 12:32:49.000000000 +0000
+++ gst-plugins-bad-0.10.23/ext/opus/gstopusenc.c	2019-07-16 20:47:47.258956702 +0100
@@ -589,7 +589,7 @@
       }
     }
 
-#ifndef GST_DISABLE_DEBUG
+#ifndef GST_DISABLE_GST_DEBUG
     GST_INFO_OBJECT (enc,
         "Mapping tables built: %d channels, %d stereo streams", enc->n_channels,
         enc->n_stereo_streams);
@@ -654,7 +654,7 @@
 {
   int error = OPUS_OK;
 
-#ifndef GST_DISABLE_DEBUG
+#ifndef GST_DISABLE_GST_DEBUG
   GST_DEBUG_OBJECT (enc,
       "setup: %d Hz, %d channels, %d stereo streams, family %d",
       enc->sample_rate, enc->n_channels, enc->n_stereo_streams,
--- gst-plugins-bad-0.10.23/ext/modplug/gstmodplug.cc.orig      2012-01-18 12:32:55.000000000 +0000
+++ gst-plugins-bad-0.10.23/ext/modplug/gstmodplug.cc   2019-08-13 19:56:16.293414139 +0100
@@ -50,7 +50,7 @@
 #define WORDS_BIGENDIAN 0
 #endif

-#include <stdafx.h>
+#include <libmodplug/stdafx.h>
 #include <libmodplug/sndfile.h>

 #include "gstmodplug.h"
