--- gst-plugins-good-1.14.4/ext/vpx/gstvpxdec.c.orig	2018-03-23 20:44:48.000000000 +0000
+++ gst-plugins-good-1.14.4/ext/vpx/gstvpxdec.c	2019-02-09 16:07:38.309262956 +0000
@@ -62,6 +62,7 @@
     {C_FLAGS (VP8_DEBLOCK), "Deblock", "deblock"},
     {C_FLAGS (VP8_DEMACROBLOCK), "Demacroblock", "demacroblock"},
     {C_FLAGS (VP8_ADDNOISE), "Add noise", "addnoise"},
+#if 0
     {C_FLAGS (VP8_DEBUG_TXT_FRAME_INFO),
           "Print frame information",
         "visualize-frame-info"},
@@ -74,6 +75,7 @@
     {C_FLAGS (VP8_DEBUG_TXT_RATE_INFO),
           "Print video rate info",
         "visualize-rate-info"},
+#endif
     {C_FLAGS (VP8_MFQE), "Multi-frame quality enhancement", "mfqe"},
     {0, NULL, NULL}
   };
