--- gst-plugins-bad-1.18.1/ext/opencv/meson.build.orig	2020-10-26 11:14:46.528234700 +0000
+++ gst-plugins-bad-1.18.1/ext/opencv/meson.build	2020-10-27 22:04:56.580728749 +0000
@@ -56,7 +56,7 @@
 gstopencv_cargs = ['-DGST_HAAR_CASCADES_DIR="@0@"']
 
 opencv_dep = dependency('opencv', version : ['>= 3.0.0', '< 3.5.0'], required : false)
-opencv_found = opencv_dep.found()
+opencv_found = false # opencv_dep.found()
 
 if opencv_found
   foreach h : libopencv_headers
@@ -66,7 +66,7 @@
     endif
   endforeach
 else
-  opencv_dep = dependency('opencv4', version : ['>= 4.0.0', '< 4.5.0'], required : false)
+  opencv_dep = dependency('opencv4', version : ['>= 4.0.0', '< 4.6.0'], required : false)
   opencv_found = opencv_dep.found()
   if opencv_found
     foreach h : libopencv4_headers
