--- gtk+-2.10.14/modules/printbackends/cups/gtkcupsutils.c.orig	2007-07-16 20:45:18.000000000 +0100
+++ gtk+-2.10.14/modules/printbackends/cups/gtkcupsutils.c	2007-08-23 07:21:38.000000000 +0100
@@ -628,7 +628,7 @@
   httpSetField(request->http, HTTP_FIELD_CONTENT_LENGTH, length);
   httpSetField(request->http, HTTP_FIELD_CONTENT_TYPE, "application/ipp");
 #ifdef HAVE_HTTP_AUTHSTRING
-  httpSetField(request->http, HTTP_FIELD_AUTHORIZATION, request->http->authstring);
+  httpSetField(request->http, HTTP_FIELD_AUTHORIZATION, httpGetAuthString(request->http));
 #endif
 
   if (httpPost(request->http, request->resource))
@@ -967,7 +967,7 @@
 
   httpClearFields(request->http);
 #ifdef HAVE_HTTP_AUTHSTRING
-  httpSetField(request->http, HTTP_FIELD_AUTHORIZATION, request->http->authstring);
+  httpSetField(request->http, HTTP_FIELD_AUTHORIZATION, httpGetAuthString(request->http));
 #endif
 
   if (httpGet(request->http, request->resource))
