--- gtk+-3.24.2/docs/tools/shooter.c.orig	2018-12-12 17:46:09.000000000 +0000
+++ gtk+-3.24.2/docs/tools/shooter.c	2018-12-15 15:35:28.571577547 +0000
@@ -170,7 +170,7 @@
       y_orig = 0;
     }
 
-  if (x_orig + width > gdk_screen_get_width (gdk_screen_get_dfeault ()))
+  if (x_orig + width > gdk_screen_get_width (gdk_screen_get_default ()))
     width = gdk_screen_width () - x_orig;
 
   if (y_orig + height > gdk_screen_get_height (gdk_screen_get_default ()))
