--- gtk+-2.20.1/gtk/gtktoolbutton.c.orig	2010-05-03 00:49:03.000000000 +0100
+++ gtk+-2.20.1/gtk/gtktoolbutton.c	2010-08-24 23:21:31.000000000 +0100
@@ -470,11 +470,11 @@
 	  gtk_widget_show (icon);
 	}
 
-      if (icon && text_orientation == GTK_ORIENTATION_HORIZONTAL)
+      if (GTK_IS_MISC (icon) && text_orientation == GTK_ORIENTATION_HORIZONTAL)
 	gtk_misc_set_alignment (GTK_MISC (icon),
 				1.0 - gtk_tool_item_get_text_alignment (GTK_TOOL_ITEM (button)),
 				0.5);
-      else if (icon)
+      else if (GTK_IS_MISC (icon))
 	gtk_misc_set_alignment (GTK_MISC (icon),
 				0.5,
 				gtk_tool_item_get_text_alignment (GTK_TOOL_ITEM (button)));
--- gtk+-2.20.1/configure.orig	2010-05-03 00:55:05.000000000 +0100
+++ gtk+-2.20.1/configure	2010-06-05 21:48:16.000000000 +0100
@@ -21748,14 +21748,14 @@
 fi
 
   if test x$with_libpng != xno && test -z "$LIBPNG"; then
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libpng12" >&5
-$as_echo_n "checking for libpng12... " >&6; }
-    if $PKG_CONFIG --exists libpng12 ; then
+    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libpng14" >&5
+$as_echo_n "checking for libpng14... " >&6; }
+    if $PKG_CONFIG --exists libpng14 ; then
         { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
         PNG='png'
-	PNG_DEP_CFLAGS_PACKAGES=libpng12
-	LIBPNG=`$PKG_CONFIG --libs libpng12`
+	PNG_DEP_CFLAGS_PACKAGES=libpng14
+	LIBPNG=`$PKG_CONFIG --libs libpng14`
     else
       { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
