--- gtk+-2.22.0/docs/faq/Makefile.in.orig	2010-10-10 10:52:16.000000000 +0100
+++ gtk+-2.22.0/docs/faq/Makefile.in	2010-10-10 10:44:33.000000000 +0100
@@ -563,7 +563,7 @@
 @HAVE_DOCBOOK_TRUE@html:
 @HAVE_DOCBOOK_TRUE@	if test -w $(srcdir); then \
 @HAVE_DOCBOOK_TRUE@	  (cd $(srcdir); \
-@HAVE_DOCBOOK_TRUE@	   db2html gtk-faq.sgml; \
+@HAVE_DOCBOOK_TRUE@	   db2html gtk-faq.sgml -o gtk-faq; \
 @HAVE_DOCBOOK_TRUE@	   test -d html && rm -r html; \
 @HAVE_DOCBOOK_TRUE@	   mv gtk-faq html); \
 @HAVE_DOCBOOK_TRUE@	fi
--- gtk+-2.22.0/docs/tutorial/Makefile.in.orig	2010-09-23 15:27:22.000000000 +0100
+++ gtk+-2.22.0/docs/tutorial/Makefile.in	2010-10-10 10:21:07.000000000 +0100
@@ -576,7 +576,7 @@
 @HAVE_DOCBOOK_TRUE@html: gtk-tut.sgml
 @HAVE_DOCBOOK_TRUE@	if test -w $(srcdir); then \
 @HAVE_DOCBOOK_TRUE@	  (cd $(srcdir); \
-@HAVE_DOCBOOK_TRUE@	   db2html gtk-tut.sgml; \
+@HAVE_DOCBOOK_TRUE@	   db2html gtk-tut.sgml -o gtk-tut; \
 @HAVE_DOCBOOK_TRUE@	   test -d html && rm -r html; \
 @HAVE_DOCBOOK_TRUE@	   mv gtk-tut html; \
 @HAVE_DOCBOOK_TRUE@	   mkdir html/images; \
