--- gwget-0.98.1/src/eggtrayicon.c.orig	2006-04-10 20:49:18.000000000 +0100
+++ gwget-0.98.1/src/eggtrayicon.c	2006-05-07 21:44:14.000000000 +0100
@@ -569,7 +569,7 @@
 
   if (pixbuf)
     {
-#if (LIBNOTIFY_VERSION_MICRO < 2)
+#if (LIBNOTIFY_VERSION_MINOR < 4 && LIBNOTIFY_VERSION_MICRO < 2)
       notify_notification_set_icon_data_from_pixbuf (icon->notify->handle, pixbuf);
 #else
       notify_notification_set_icon_from_pixbuf (icon->notify->handle, pixbuf);
