--- gwget-0.98.1/src/main_window_cb.c-	2006-08-14 12:10:28.000000000 -0400
+++ gwget-0.98.1/src/main_window_cb.c	2006-08-14 12:10:46.000000000 -0400
@@ -24,7 +24,7 @@
 #include <libgnome/gnome-init.h>
 #include <libgnomevfs/gnome-vfs-utils.h>
 #include <libgnomevfs/gnome-vfs-mime-utils.h>
-#include <gnome-vfs-module-2.0/libgnomevfs/gnome-vfs-mime.h>
+#include <libgnomevfs/gnome-vfs-mime.h>
 
 #include "main_window.h"
 #include "main_window_cb.h"
