--- libgda-3.1.5/providers/sqlite/Makefile.am.orig	2008-07-20 10:51:16.000000000 +0100
+++ libgda-3.1.5/providers/sqlite/Makefile.am	2010-02-11 22:23:30.000000000 +0000
@@ -22,7 +22,6 @@
 xmldir   = $(datadir)/libgda-3.0
 xml_in_files = sqlite_specs_create_db.xml.in \
 	sqlite_specs_dsn.xml.in \
-	sqlite_specs_create_db.xml.in \
 	sqlite_specs_drop_db.xml.in \
 	sqlite_specs_create_table.xml.in \
 	sqlite_specs_drop_table.xml.in \
