--- libgnomedb-3.1.2/libgnomedb-goo/gnome-db-goo.c.orig	2007-09-01 13:29:00.000000000 +0100
+++ libgnomedb-3.1.2/libgnomedb-goo/gnome-db-goo.c	2009-05-31 20:12:16.000000000 +0100
@@ -935,8 +935,8 @@
 
 	for (list = nodes_list; list; list = list->next) {
 		NodeLayout *nl = (NodeLayout*) list->data;
-		nl->end_x = ND_coord_i (nl->node).x - (nl->width / 2.);
-		nl->end_y = - ND_coord_i (nl->node).y - (nl->height / 2.);
+		nl->end_x = ND_coord (nl->node).x - (nl->width / 2.);
+		nl->end_y = - ND_coord (nl->node).y - (nl->height / 2.);
 		nl->dx = fabs (nl->end_x - nl->start_x);
 		nl->dy = fabs (nl->end_y - nl->start_y);
 		nl->stop = FALSE;
--- libgnomedb-3.1.2/libgnomedb-graph/gnome-db-canvas.c.orig	2007-09-01 13:28:14.000000000 +0100
+++ libgnomedb-3.1.2/libgnomedb-graph/gnome-db-canvas.c	2009-05-31 20:10:58.000000000 +0100
@@ -853,8 +853,8 @@
 
 	for (list = nodes_list; list; list = list->next) {
 		NodeLayout *nl = (NodeLayout*) list->data;
-		nl->end_x = ND_coord_i (nl->node).x - (nl->width / 2.);
-		nl->end_y = - ND_coord_i (nl->node).y - (nl->height / 2.);
+		nl->end_x = ND_coord (nl->node).x - (nl->width / 2.);
+		nl->end_y = - ND_coord (nl->node).y - (nl->height / 2.);
 		/*g_print ("After: Node %p: HxW: %.3f %.3f XxY = %d, %d\n", nl->node, 
 			 ND_height (nl->node), ND_width (nl->node),
 			 ND_coord_i (nl->node).x, - ND_coord_i (nl->node).y);*/
