--- libgnomedb-3.99.7/libgnomedb-graph/gnome-db-canvas.c.orig	2008-12-30 10:22:49.000000000 +0000
+++ libgnomedb-3.99.7/libgnomedb-graph/gnome-db-canvas.c	2009-05-31 20:25:35.000000000 +0100
@@ -922,8 +922,8 @@
 
 	for (list = nodes_list; list; list = list->next) {
 		NodeLayout *nl = (NodeLayout*) list->data;
-		nl->end_x = ND_coord_i (nl->node).x - (nl->width / 2.);
-		nl->end_y = - ND_coord_i (nl->node).y - (nl->height / 2.);
+		nl->end_x = ND_coord (nl->node).x - (nl->width / 2.);
+		nl->end_y = - ND_coord (nl->node).y - (nl->height / 2.);
 		nl->dx = fabs (nl->end_x - nl->start_x);
 		nl->dy = fabs (nl->end_y - nl->start_y);
 		nl->stop = FALSE;
