--- libgnomekbd-2.28.2/libgnomekbd/gkbd-indicator.c.orig	2009-12-13 12:14:59.000000000 +0000
+++ libgnomekbd-2.28.2/libgnomekbd/gkbd-indicator.c	2010-02-07 19:41:17.000000000 +0000
@@ -577,7 +577,8 @@
 static void
 gkbd_indicator_stop_listen (void)
 {
-	xkl_engine_stop_listen (globals.engine);
+	xkl_engine_stop_listen (globals.engine,
+				XKLL_TRACK_KEYBOARD_STATE);
 
 	gdk_window_remove_filter (NULL, (GdkFilterFunc)
 				  gkbd_indicator_filter_x_evt, NULL);
