--- libgda-3.1.5/libgda/gda-quark-list.c.orig	2008-07-20 10:51:23.000000000 +0100
+++ libgda-3.1.5/libgda/gda-quark-list.c	2012-11-17 19:29:36.000000000 +0000
@@ -22,10 +22,7 @@
  */
 
 #include <libgda/gda-quark-list.h>
-#include <glib/ghash.h>
-#include <glib/gmem.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 
 struct _GdaQuarkList {
 	GHashTable *hash_table;
--- libgda-3.1.5/libgda/gda-server-operation.c.orig	2008-08-04 12:16:26.000000000 +0100
+++ libgda-3.1.5/libgda/gda-server-operation.c	2012-11-17 19:31:05.000000000 +0000
@@ -21,7 +21,7 @@
  */
 
 #include <stdlib.h>
-#include <glib/gmessages.h>
+#include <glib.h>
 #include <libgda/gda-marshal.h>
 #include <libgda/gda-dict.h>
 #include <libgda/gda-dict-type.h>
--- libgda-3.1.5/libgda/sqlite/gda-sqlite.h.orig	2008-07-20 10:51:22.000000000 +0100
+++ libgda-3.1.5/libgda/sqlite/gda-sqlite.h	2012-11-17 18:56:51.000000000 +0000
@@ -25,8 +25,7 @@
 #ifndef __GDA_SQLITE_H__
 #define __GDA_SQLITE_H__
 
-#include <glib/gmacros.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <libgda/gda-value.h>
 #include <libgda/gda-connection.h>
 
--- libgda-3.1.5/libgda/gda-column.c.orig	2008-07-20 10:51:23.000000000 +0100
+++ libgda-3.1.5/libgda/gda-column.c	2012-11-17 19:08:49.000000000 +0000
@@ -23,8 +23,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <libgda/gda-column.h>
 #include <string.h>
 #include "gda-marshal.h"
--- libgda-3.1.5/libgda/gda-server-provider.h.orig	2008-07-20 10:51:23.000000000 +0100
+++ libgda-3.1.5/libgda/gda-server-provider.h	2012-11-17 19:01:08.000000000 +0000
@@ -25,6 +25,7 @@
 #ifndef __GDA_SERVER_PROVIDER_H__
 #define __GDA_SERVER_PROVIDER_H__
 
+//#include <glib.h>
 #include <libgda/gda-command.h>
 #include <libgda/gda-server-operation.h>
 #include <libgda/gda-connection.h>
--- libgda-3.1.5/libgda/gda-data-model-filter-sql.c.orig	2008-07-20 10:51:23.000000000 +0100
+++ libgda-3.1.5/libgda/gda-data-model-filter-sql.c	2012-11-17 19:23:22.000000000 +0000
@@ -21,7 +21,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gdataset.h>
+#include <glib.h>
 #include <glib-object.h>
 #include <libsql/sql_parser.h>
 #include <glib/gi18n-lib.h>
--- libgda-3.1.5/libgda/gda-value.h.orig	2008-07-20 10:51:23.000000000 +0100
+++ libgda-3.1.5/libgda/gda-value.h	2012-11-17 19:05:04.000000000 +0000
@@ -27,8 +27,7 @@
 #define __GDA_VALUE_H__
 
 #include <time.h>
-#include <glib/glist.h>
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib-object.h>
 #include <libxml/tree.h>
 #include <libgda/gda-decl.h>
--- libgda-3.1.5/libgda/gda-row.h.orig	2008-07-20 10:43:15.000000000 +0100
+++ libgda-3.1.5/libgda/gda-row.h	2012-11-17 19:04:32.000000000 +0000
@@ -25,9 +25,9 @@
 #ifndef __GDA_ROW_H__
 #define __GDA_ROW_H__
 
+#include <glib.h>
 #include <glib-object.h>
 #include <libgda/gda-column.h>
-#include <glib/gmacros.h>
 #include <libgda/gda-decl.h>
 
 G_BEGIN_DECLS
--- libgda-3.1.5/libgda/gda-util.c.orig	2008-07-20 10:51:23.000000000 +0100
+++ libgda-3.1.5/libgda/gda-util.c	2012-11-17 19:34:51.000000000 +0000
@@ -27,10 +27,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <string.h>
-#include <glib/gfileutils.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
-#include <glib/gprintf.h>
+#include <glib.h>
 #include <libsql/sql_parser.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-log.h>
--- libgda-3.1.5/libgda/gda-value.c.orig	2008-07-20 10:51:23.000000000 +0100
+++ libgda-3.1.5/libgda/gda-value.c	2012-11-17 19:36:12.000000000 +0000
@@ -28,10 +28,7 @@
 #include <string.h>
 #include <stdlib.h>
 #include <time.h>
-#include <glib/gdate.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
-#include <glib/gstring.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-value.h>
 #include <libgda/gda-blob-op.h>
--- libgda-3.1.5/libgda/gda-log.h.orig	2008-07-20 10:43:15.000000000 +0100
+++ libgda-3.1.5/libgda/gda-log.h	2012-11-17 18:57:58.000000000 +0000
@@ -20,8 +20,7 @@
 #if !defined(__gda_log_h__)
 #  define __gda_log_h__
 
-#include <glib/gmacros.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <time.h>
 
 G_BEGIN_DECLS
--- libgda-3.1.5/libgda/gda-data-model-array.c.orig	2008-07-20 10:51:23.000000000 +0100
+++ libgda-3.1.5/libgda/gda-data-model-array.c	2012-11-17 19:19:50.000000000 +0000
@@ -22,7 +22,7 @@
  */
 
 #include <string.h>
-#include <glib/garray.h>
+#include <glib.h>
 #include <libgda/gda-data-model-array.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-row.h>
--- libgda-3.1.5/libgda/gda-init.c.orig	2008-07-20 10:51:23.000000000 +0100
+++ libgda-3.1.5/libgda/gda-init.c	2012-11-17 19:28:28.000000000 +0000
@@ -17,7 +17,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gmain.h>
+#include <glib.h>
 #include <gmodule.h>
 #include <libgda/libgda.h>
 #include <glib/gi18n-lib.h>
--- libgda-3.1.5/libgda/gda-log.c.orig	2008-07-20 10:43:15.000000000 +0100
+++ libgda-3.1.5/libgda/gda-log.c	2012-11-17 19:28:57.000000000 +0000
@@ -24,10 +24,7 @@
 #include <syslog.h>
 #endif
 #include <time.h>
-#include <glib/gmem.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
-#include <glib/gutils.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-log.h>
 
--- libgda-3.1.5/libgda/gda-config.h.orig	2008-07-20 10:51:23.000000000 +0100
+++ libgda-3.1.5/libgda/gda-config.h	2012-11-17 18:57:17.000000000 +0000
@@ -23,7 +23,7 @@
 #ifndef __GDA_CONFIG_H__
 #define __GDA_CONFIG_H__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-data-model.h>
 #include <libgda/gda-parameter.h>
 
--- libgda-3.1.5/libgda/gda-command.h.orig	2008-07-20 10:51:23.000000000 +0100
+++ libgda-3.1.5/libgda/gda-command.h	2012-11-17 19:03:14.000000000 +0000
@@ -24,9 +24,8 @@
 #ifndef __GDA_COMMAND_H__
 #define __GDA_COMMAND_H__
 
+#include <glib.h>
 #include <glib-object.h>
-#include <glib/gmacros.h>
-#include <glib/gtypes.h>
 
 G_BEGIN_DECLS
 
--- libgda-3.1.5/libgda/gda-config.c.orig	2008-07-20 10:51:23.000000000 +0100
+++ libgda-3.1.5/libgda/gda-config.c	2012-11-17 19:10:56.000000000 +0000
@@ -38,7 +38,6 @@
 #include <sys/stat.h>
 #ifdef HAVE_FAM
 #include <fam.h>
-#include <glib/giochannel.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
--- libgda-3.1.5/libgda/gda-command.c.orig	2008-07-20 10:51:23.000000000 +0100
+++ libgda-3.1.5/libgda/gda-command.c	2012-11-17 19:10:00.000000000 +0000
@@ -21,9 +21,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gmem.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <libgda/gda-command.h>
 
 GType
--- libgda-3.1.5/libgda/gda-server-provider.c.orig	2008-07-20 10:51:23.000000000 +0100
+++ libgda-3.1.5/libgda/gda-server-provider.c	2012-11-17 19:31:34.000000000 +0000
@@ -22,7 +22,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gmessages.h>
+#include <glib.h>
 #include <libgda/gda-server-provider.h>
 #include <libgda/gda-server-provider-extra.h>
 #include <libgda/gda-server-provider-private.h>
--- libgda-3.1.5/libgda/gda-data-model-hash.c.orig	2008-07-20 10:51:23.000000000 +0100
+++ libgda-3.1.5/libgda/gda-data-model-hash.c	2012-11-17 19:23:47.000000000 +0000
@@ -23,7 +23,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/ghash.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-data-model-hash.h>
 #include <libgda/gda-data-model-extra.h>
--- libgda-3.1.5/libgda/gda-util.h.orig	2008-07-20 10:51:23.000000000 +0100
+++ libgda-3.1.5/libgda/gda-util.h	2012-11-17 18:58:56.000000000 +0000
@@ -24,8 +24,7 @@
 #ifndef __GDA_UTIL_H__
 #define __GDA_UTIL_H__
 
-#include <glib/ghash.h>
-#include <glib/glist.h>
+#include <glib.h>
 #include "gda-parameter.h"
 #include "gda-row.h"
 #include "gda-connection.h"
--- libgda-3.1.5/libgda/gda-column.h.orig	2008-07-20 10:43:15.000000000 +0100
+++ libgda-3.1.5/libgda/gda-column.h	2012-11-17 19:03:50.000000000 +0000
@@ -26,9 +26,9 @@
 #ifndef __GDA_COLUMN_H__
 #define __GDA_COLUMN_H__
 
+#include <glib.h>
 #include <glib-object.h>
 #include <libgda/gda-value.h>
-#include <glib/gmacros.h>
 #include <libgda/gda-decl.h>
 
 G_BEGIN_DECLS
--- libgda-3.1.5/libgda/gda-quark-list.h.orig	2008-07-20 10:43:15.000000000 +0100
+++ libgda-3.1.5/libgda/gda-quark-list.h	2012-11-17 19:04:20.000000000 +0000
@@ -24,8 +24,7 @@
 #ifndef __GDA_QUARK_LIST_H__
 #define __GDA_QUARK_LIST_H__
 
-#include <glib/gmacros.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <glib-object.h>
 
 G_BEGIN_DECLS
--- libgda-3.1.5/providers/postgres/gda-postgres.h.orig	2008-07-20 10:51:16.000000000 +0100
+++ libgda-3.1.5/providers/postgres/gda-postgres.h	2012-11-17 19:49:28.000000000 +0000
@@ -24,7 +24,7 @@
 #ifndef __GDA_POSTGRES_H__
 #define __GDA_POSTGRES_H__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
 #include "gda-postgres-provider.h"
--- libgda-3.1.5/providers/ldap/gda-ldap.h.orig	2007-10-01 16:51:37.000000000 +0100
+++ libgda-3.1.5/providers/ldap/gda-ldap.h	2012-11-17 19:51:28.000000000 +0000
@@ -26,7 +26,7 @@
 #if !defined(__gda_ldap_h__)
 #  define __gda_ldap_h__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-connection.h>
 #include <libgda/gda-server-provider.h>
 #include <glib/gi18n-lib.h>
--- libgda-3.1.5/providers/bdb/gda-bdb.h.orig	2008-07-20 10:51:15.000000000 +0100
+++ libgda-3.1.5/providers/bdb/gda-bdb.h	2012-11-17 19:37:26.000000000 +0000
@@ -25,7 +25,7 @@
 #if defined(HAVE_CONFIG_H)
 #endif
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
 #include <db.h>
--- libgda-3.1.5/providers/odbc/gda-odbc.h.orig	2007-10-01 16:51:41.000000000 +0100
+++ libgda-3.1.5/providers/odbc/gda-odbc.h	2012-11-17 19:48:30.000000000 +0000
@@ -25,7 +25,7 @@
 #if !defined(__gda_odbc_h__)
 #  define __gda_odbc_h__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib-object.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
--- libgda-3.1.5/providers/mdb/gda-mdb-provider.c.orig	2008-07-20 10:51:16.000000000 +0100
+++ libgda-3.1.5/providers/mdb/gda-mdb-provider.c	2012-11-17 19:42:10.000000000 +0000
@@ -22,7 +22,7 @@
  */
 
 #include <stdlib.h>
-#include <glib/gbacktrace.h>
+#include <glib.h>
 #include <virtual/gda-vconnection-data-model.h>
 #include <libgda/gda-data-model-array.h>
 #include <libgda/gda-data-model-private.h>
--- libgda-3.1.5/providers/mdb/gda-mdb.h.orig	2008-07-20 10:51:16.000000000 +0100
+++ libgda-3.1.5/providers/mdb/gda-mdb.h	2012-11-17 19:42:59.000000000 +0000
@@ -24,7 +24,7 @@
 #ifndef __GDA_MDB_H__
 #define __GDA_MDB_H__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-connection.h>
 #include <libgda/gda-server-provider.h>
 #include <glib/gi18n-lib.h>
--- libgda-3.1.5/providers/mysql/gda-mysql.h.orig	2008-07-20 10:51:16.000000000 +0100
+++ libgda-3.1.5/providers/mysql/gda-mysql.h	2012-11-17 19:47:00.000000000 +0000
@@ -25,7 +25,7 @@
 #ifndef __GDA_MYSQL_H__
 #define __GDA_MYSQL_H__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-connection.h>
 #include <libgda/gda-server-provider.h>
 #include <glib/gi18n-lib.h>
--- libgda-3.1.5/providers/sqlite/Makefile.am.orig	2008-07-20 10:51:16.000000000 +0100
+++ libgda-3.1.5/providers/sqlite/Makefile.am	2010-02-11 22:23:30.000000000 +0000
@@ -22,7 +22,6 @@
 xmldir   = $(datadir)/libgda-3.0
 xml_in_files = sqlite_specs_create_db.xml.in \
 	sqlite_specs_dsn.xml.in \
-	sqlite_specs_create_db.xml.in \
 	sqlite_specs_drop_db.xml.in \
 	sqlite_specs_create_table.xml.in \
 	sqlite_specs_drop_table.xml.in \
