--- meld-1.2/help/fr/GNUmakefile.orig	2008-03-31 21:44:53.000000000 +0100
+++ meld-1.2/help/fr/GNUmakefile	2008-06-28 21:41:04.000000000 +0100
@@ -17,7 +17,6 @@
 install : $(OMF_NAME).install
 	-mkdir -m 755 -p $(OMF_DIR_) $(XML_DIR_) $(XML_DIR_)/figures
 	install -m 644 meld.$(LANG).xml $(XML_DIR_)/meld.xml
-	install -m 644 figures/*.png $(XML_DIR_)/figures
 	-install -m 644 $< $(OMF_DIR_)/$(OMF_NAME)
 	-scrollkeeper-update -p $(OMF_STATE) -o $(OMF_DIR_)
 
