--- nautilus-actions-3.0.7/src/plugin-menu/nautilus-actions.c.orig	2011-01-19 20:47:57.000000000 +0000
+++ nautilus-actions-3.0.7/src/plugin-menu/nautilus-actions.c	2011-05-21 08:05:54.000000000 +0100
@@ -395,7 +395,9 @@
 
 	iface->get_file_items = menu_provider_get_file_items;
 	iface->get_background_items = menu_provider_get_background_items;
+#if 0
 	iface->get_toolbar_items = menu_provider_get_toolbar_items;
+#endif
 }
 
 /*
--- nautilus-actions-3.0.7/src/plugin-tracker/na-tracker.c.orig	2011-01-01 12:50:12.000000000 +0000
+++ nautilus-actions-3.0.7/src/plugin-tracker/na-tracker.c	2011-05-21 08:06:33.000000000 +0100
@@ -259,7 +259,9 @@
 
 	iface->get_background_items = menu_provider_get_background_items;
 	iface->get_file_items = menu_provider_get_file_items;
+#if 0
 	iface->get_toolbar_items = NULL;
+#endif
 }
 
 static GList *
