--- NetworkManager-1.10.4/src/devices/ovs/nm-device-ovs-interface.c.orig	2018-02-05 09:58:33.000000000 +0000
+++ NetworkManager-1.10.4/src/devices/ovs/nm-device-ovs-interface.c	2018-02-06 22:22:46.208569753 +0000
@@ -50,7 +50,7 @@
 
 G_DEFINE_TYPE (NMDeviceOvsInterface, nm_device_ovs_interface, NM_TYPE_DEVICE)
 
-#define NM_DEVICE_OVS_INTERFACE_GET_PRIVATE(self) _NM_GET_PRIVATE (self, NMDeviceOvsInterface, NM_IS_DEVICE_OVS_INTERFACE)
+#define NM_DEVICE_OVS_INTERFACE_GET_PRIVATE(self) _NM_GET_PRIVATE (self, NMDeviceOvsInterface, NM_IS_DEVICE_OVS_INTERFACE, NMDevice)
 
 /*****************************************************************************/
 
