--- NetworkManager-1.12.2/libnm-core/nm-version.h.orig	2018-07-24 19:18:00.000000000 +0100
+++ NetworkManager-1.12.2/libnm-core/nm-version.h	2018-08-19 14:14:40.254756829 +0100
@@ -161,7 +161,7 @@
 #endif
 
 #if NM_VERSION_MAX_ALLOWED < NM_VERSION_1_12_2
-# define NM_AVAILABLE_IN_1_12_2          G_UNAVAILABLE(1,12,2)
+# define NM_AVAILABLE_IN_1_12_2          G_UNAVAILABLE(1.12,2)
 #else
 # define NM_AVAILABLE_IN_1_12_2
 #endif
