--- ocrfeeder-0.8.5/m4/m4_ax_python_module.m4.orig	2020-01-04 20:14:03.000000000 +0000
+++ ocrfeeder-0.8.5/m4/m4_ax_python_module.m4	2025-07-30 16:25:32.874013957 +0100
@@ -31,7 +31,7 @@
     fi
     PYTHON_NAME=`basename $PYTHON`
     AC_MSG_CHECKING($PYTHON_NAME module: $1)
-	$PYTHON -c "import imp; imp.find_module(\"$1\")" 2>/dev/null
+	$PYTHON -c "import $1" 2>/dev/null
 	if test $? -eq 0;
 	then
 		AC_MSG_RESULT(yes)
