--- pango-1.32.6/pango/pangocairo-fontmap.c.orig	2013-01-07 19:07:35.000000000 +0000
+++ pango-1.32.6/pango/pangocairo-fontmap.c	2013-02-03 08:11:16.020490125 +0000
@@ -66,6 +66,10 @@
 PangoFontMap *
 pango_cairo_font_map_new (void)
 {
+#if !GLIB_CHECK_VERSION (2, 35, 3)
+  g_type_init();
+#endif
+
 #if defined(HAVE_CORE_TEXT) && defined (HAVE_CAIRO_QUARTZ)
   return g_object_new (PANGO_TYPE_CAIRO_CORE_TEXT_FONT_MAP, NULL);
 #elif defined(HAVE_CAIRO_WIN32)
@@ -99,6 +103,10 @@
 PangoFontMap *
 pango_cairo_font_map_new_for_font_type (cairo_font_type_t fonttype)
 {
+#if !GLIB_CHECK_VERSION (2, 35, 3)
+  g_type_init();
+#endif
+
   switch ((int) fonttype)
   {
 #if defined(HAVE_CORE_TEXT) && defined (HAVE_CAIRO_QUARTZ)
