--- PyKDE-3.16.0/sip/kio/kmimetype.sip	2006-09-22 18:41:53.000000000 +0530
+++ PyKDE-3.16.0/sip/kio/kmimetype.sip	2006-11-06 05:27:39.000000000 +0530
@@ -101,7 +101,7 @@
 %If ( KDE_3_2_0 -  )
     static KMimeType::Ptr defaultMimeTypePtr ();
     QString              parentMimeType () const;
-    bool                 is (const QString&) const;
+    bool                 is (const QString&) const /PyName=is_/; //'is' is a reserved keyword in python
     static KMimeType::Ptr diagnoseFileName (const QString&, QString&);
 %End
 
