--- recommonmark-0.4.0/recommonmark/parser.py.orig	2016-01-05 21:59:38.000000000 +0000
+++ recommonmark-0.4.0/recommonmark/parser.py	2018-09-11 19:42:38.470696674 +0100
@@ -3,7 +3,7 @@
 
 from docutils import parsers, nodes
 
-from CommonMark import DocParser, HTMLRenderer
+from commonmark import Parser, HtmlRenderer
 from warnings import warn
 
 __all__ = ['CommonMarkParser']
@@ -80,7 +80,7 @@
         self.current_node = document
         self.section_handler = _SectionHandler(document)
 
-        parser = DocParser()
+        parser = Parser()
 
         ast = parser.parse(inputstring + '\n')
 
@@ -220,7 +220,7 @@
 
 
 def inline_entity(inline):
-    val = HTMLRenderer().renderInline(inline)
+    val = HtmlRenderer().renderInline(inline)
     entity_node = nodes.paragraph('', val, format='html')
     return entity_node
 
