--- qt-everywhere-opensource-src-4.8.6/src/gui/image/qppmhandler.cpp.orig	2014-04-10 19:37:12.000000000 +0100
+++ qt-everywhere-opensource-src-4.8.6/src/gui/image/qppmhandler.cpp	2015-05-06 22:49:32.011872755 +0100
@@ -172,19 +172,39 @@
                         *p++ = qRgb(b[0],b[1],b[2]);
                         b += 3;
                     } else {
-                        *p++ = qRgb(((int(b[0]) * 256 + int(b[1]) + 1) * 256) / (mcc + 1) - 1,
-                                    ((int(b[2]) * 256 + int(b[3]) + 1) * 256) / (mcc + 1) - 1,
-                                    ((int(b[4]) * 256 + int(b[5]) + 1) * 256) / (mcc + 1) - 1);
+                        *p++ = qRgb(((int(b[0]) * 256 + int(b[1])) * 256) / (mcc + 1),
+                                    ((int(b[2]) * 256 + int(b[3])) * 256) / (mcc + 1),
+                                    ((int(b[4]) * 256 + int(b[5])) * 256) / (mcc + 1));
                         b += 6;
                     }
                 }
             }
             delete[] buf24;
         } else {                                // type 4,5
-            for (y=0; y<h; y++) {
-                if (device->read((char *)outImage->scanLine(y), pbm_bpl)
-                        != pbm_bpl)
-                    return false;
+            if (mcc < 256) {
+                for (y=0; y<h; y++) {
+                    if (device->read((char *)outImage->scanLine(y), pbm_bpl)
+                            != pbm_bpl)
+                        return false;
+                }
+            } else {
+                pbm_bpl *= 2;
+                uchar *buf16 = new uchar[pbm_bpl], *b;
+                char *p, *end;
+                for (y=0; y<h; y++) {
+                    if (device->read((char *)buf16, pbm_bpl) != pbm_bpl) {
+                        delete[] buf16;
+                        return false;
+                    }
+                    p = (char *)outImage->scanLine(y);
+                    end = p + w;
+                    b = buf16;
+                    while (p < end) {
+                        *p++ = ((int(b[0]) * 256 + int(b[1])) * 256) / (mcc + 1);
+                        b += 2;
+                    }
+                }
+                delete[] buf16;
             }
         }
     } else {                                        // read ascii data
