--- ExtUtils-Install-1.43/lib/ExtUtils/Install.pm.orig	2007-07-20 23:08:52.000000000 +0100
+++ ExtUtils-Install-1.43/lib/ExtUtils/Install.pm	2007-07-25 13:50:16.000000000 +0100
@@ -676,7 +676,7 @@
 
         if ( defined $inc_uninstall ) {
             inc_uninstall($sourcefile,$ffd, $verbose,
-                          $nonono,
+                          $nonono || $inc_uninstall == 0,
                           $realtarget ne $targetfile ? $realtarget : "");
         }
 
