--- demos/widget.orig	2005-06-16 22:57:11.000000000 +0200
+++ demos/widget	2005-06-16 23:43:23.000000000 +0200
@@ -6,7 +6,7 @@
 use 5.008;
 use Config;
 use Tk 804.000;
-use lib Tk->findINC( 'demos/widget_lib' );
+use lib "@demopath@/widget_lib";
 use Tk::widgets qw/ DialogBox ErrorDialog LabEntry ROText /;
 use WidgetDemo;
 use subs qw/
@@ -44,9 +44,9 @@
 $MW->title('Perl/Tk Widget Demonstration');
 $MW->iconname('widget');
 $FONT = 'Helvetica 12';
-my $widget_lib = Tk->findINC('demos/widget_lib');
+my $widget_lib = "@demopath@/widget_lib";
 my $wd = "$widget_lib/WidgetDemo.pm";
-$WIDTRIB = Tk->findINC('demos/widtrib');
+$WIDTRIB = "@demopath@/widtrib";
 unless (Tk::tainting) {
     $WIDTRIB = $ENV{WIDTRIB} if defined $ENV{WIDTRIB};
     $WIDTRIB = $ARGV[0] if defined $ARGV[0];
