--- xmltv-0.5.46/grab/uk_rt/tv_grab_uk_rt.in.orig	2006-11-30 19:36:30.000000000 +0000
+++ xmltv-0.5.46/grab/uk_rt/tv_grab_uk_rt.in	2007-07-26 22:41:51.000000000 +0100
@@ -193,7 +193,7 @@
     /^(\d+)\|(.+)/ or die "bad line in channel list: $_";
     my ($rt_id, $name) = ($1, $2);
     $seen_rt_id{$rt_id}++ && die "channel with RT id $rt_id seen twice";
-    $seen_name{$name}++ && die "channel named '$name' seen twice";
+    $seen_name{$name}++ ;# && die "channel named '$name' seen twice";
     my $xmltv_id = $rt_to_xmltv{$rt_id};
     if (not defined $xmltv_id) {
         if (not $opt_quiet) {
