--- akonadi-17.04.1/src/shared/akdebug.cpp.orig	2017-04-28 20:28:35.000000000 +0100
+++ akonadi-17.04.1/src/shared/akdebug.cpp	2017-05-17 19:19:43.306098261 +0100
@@ -192,7 +192,7 @@
         QFile fileOld(infoOld.absoluteFilePath());
         const bool success = fileOld.remove();
         if (!success) {
-            qFatal("Cannot remove old log file - running on a readonly filesystem maybe?");
+            qWarning("Cannot remove old log file - running on a readonly filesystem maybe?");
         }
     }
     QFileInfo info(sInstance()->errorLogFileName());
@@ -200,7 +200,7 @@
         QFile file(info.absoluteFilePath());
         const bool success = file.rename(sInstance()->errorLogFileName() + QLatin1String(".old"));
         if (!success) {
-            qFatal("Cannot rename log file - running on a readonly filesystem maybe?");
+            qWarning("Cannot rename log file - running on a readonly filesystem maybe?");
         }
     }
 
