--- alkimia-8.0.3/CMakeLists.txt.orig	2020-01-10 16:43:45.000000000 +0000
+++ alkimia-8.0.3/CMakeLists.txt	2020-01-18 00:01:55.997347580 +0000
@@ -121,6 +121,9 @@
         -DKDELIBS4SUPPORT_DEPRECATED_EXPORT=Q_DECL_IMPORT
         -DKDELIBS4SUPPORT_EXPORT=Q_DECL_IMPORT
     )
+    if(NOT SHARE_INSTALL_DIR)
+        set(SHARE_INSTALL_DIR ${DATA_INSTALL_DIR})
+    endif()
 endif()
 
 set(TARGET_SUFFIX ${PC_TARGET_SUFFIX})
--- alkimia-8.0.3/src/LibAlkimiaConfig.cmake.in.orig	2020-01-10 16:43:04.000000000 +0000
+++ alkimia-8.0.3/src/LibAlkimiaConfig.cmake.in	2020-01-20 22:08:30.731822271 +0000
@@ -9,6 +9,9 @@
 else()
     find_dependency(Qt5Core)
     find_dependency(Qt5DBus)
+    if("@BUILD_WITH_WEBKIT@" STREQUAL "ON" OR "@BUILD_WITH_WEBKIT@" STREQUAL "1")
+        find_dependency(Qt5WebKitWidgets)
+    endif()
 endif()
 find_dependency(@MP_CMAKE_MODULE@)
 
