--- audiocd-kio-17.04.2/data/solid_audiocd.desktop.orig	2016-12-20 12:07:06.000000000 +0000
+++ audiocd-kio-17.04.2/data/solid_audiocd.desktop	2017-07-04 20:40:56.048310142 +0100
@@ -7,7 +7,7 @@
 [Desktop Action open]
 Exec=kioclient exec audiocd:/?device=%d
 Icon=system-file-manager
-Name=Open with File Manager
+Name=Open with File Manager (audio cd)
 Name[ar]=افتح بمدير الملفات
 Name[bg]=Отваряне с файлов мениджър
 Name[bs]=Otvori menadžerom datoteka
@@ -17,7 +17,7 @@
 Name[da]=Åbn med filhåndtering
 Name[de]=Mit Dateiverwaltung öffnen
 Name[el]=Άνοιγμα με διαχειριστή αρχείων
-Name[en_GB]=Open with File Manager
+Name[en_GB]=Open with File Manager (audio cd)
 Name[es]=Abrir con el gestor de archivos
 Name[et]=Avamine failihalduriga
 Name[fi]=Avaa tiedostonhallinnassa
