--- calligra-3.1.0/stage/part/KPrPresentationTool.cpp.orig	2018-01-26 09:24:44.000000000 +0000
+++ calligra-3.1.0/stage/part/KPrPresentationTool.cpp	2018-09-22 17:24:54.932144884 +0100
@@ -24,6 +24,7 @@
 
 #include "KPrPresentationTool.h"
 
+#include <QFrame>
 #include <QWidget>
 #include <QVBoxLayout>
 #include <QPainter>
--- calligra-3.1.0/sheets/dialogs/LayoutDialog.cpp.orig	2018-01-26 09:24:44.000000000 +0000
+++ calligra-3.1.0/sheets/dialogs/LayoutDialog.cpp	2018-09-22 17:42:15.491095236 +0100
@@ -35,6 +35,7 @@
 #include <stdlib.h>
 #include <math.h>
 
+#include <QButtonGroup>
 #include <QIntValidator>
 #include <QCheckBox>
 #include <QFrame>
--- calligra-3.1.0/plugins/formulashape/FormulaToolWidget.cpp.orig	2018-01-26 09:24:44.000000000 +0000
+++ calligra-3.1.0/plugins/formulashape/FormulaToolWidget.cpp	2018-09-22 19:02:01.575551358 +0100
@@ -27,6 +27,7 @@
 #include "FormulaCursor.h"
 #include "FormulaDebug.h"
 
+#include <QHeaderView>
 #include <QWidgetAction>
 #include <QTableWidget>
 #include <QAction>
--- calligra-3.1.0/plugins/chartshape/dialogs/TableEditorDialog.cpp.orig	2018-01-26 09:24:44.000000000 +0000
+++ calligra-3.1.0/plugins/chartshape/dialogs/TableEditorDialog.cpp	2018-09-22 18:57:53.374563200 +0100
@@ -24,6 +24,7 @@
 
 // Qt
 #include <QAbstractItemModel>
+#include <QAction>
 
 // Calligra
 #include <KoIcon.h>
--- calligra-3.1.0/libs/widgets/KoPageLayoutWidget.cpp.orig	2018-01-26 09:24:44.000000000 +0000
+++ calligra-3.1.0/libs/widgets/KoPageLayoutWidget.cpp	2018-09-22 17:04:50.897202332 +0100
@@ -19,6 +19,8 @@
 
 #include "KoPageLayoutWidget.h"
 
+#include <QButtonGroup>
+
 #include <ui_KoPageLayoutWidget.h>
 
 #include <KoUnit.h>
--- calligra-3.1.0/libs/widgets/KoCsvImportDialog.cpp.orig	2018-01-26 09:24:44.000000000 +0000
+++ calligra-3.1.0/libs/widgets/KoCsvImportDialog.cpp	2018-09-22 17:03:30.521206166 +0100
@@ -27,6 +27,7 @@
 #include <QTableWidget>
 #include <QTableWidgetSelectionRange>
 
+#include <QButtonGroup>
 // KF5
 #include <kcharsets.h>
 #include <kconfig.h>
--- calligra-3.1.0/words/part/dialogs/KWRunAroundProperties.cpp.orig	2018-01-26 09:24:44.000000000 +0000
+++ calligra-3.1.0/words/part/dialogs/KWRunAroundProperties.cpp	2018-09-22 17:19:31.999160292 +0100
@@ -23,6 +23,8 @@
 #include "frames/KWFrame.h"
 #include "frames/KWTextFrameSet.h"
 
+#include <QButtonGroup>
+
 #include <commands/KoShapeRunAroundCommand.h>
 #include <KoUnit.h>
 
--- calligra-3.1.0/words/part/dialogs/KWAnchoringProperties.cpp.orig	2018-01-26 09:24:44.000000000 +0000
+++ calligra-3.1.0/words/part/dialogs/KWAnchoringProperties.cpp	2018-09-22 17:20:28.557157593 +0100
@@ -35,6 +35,7 @@
 
 #include <kundo2command.h>
 
+#include <QButtonGroup>
 #include <QComboBox>
 
 const int KWAnchoringProperties::vertRels[4][20] = {
--- calligra-3.1.0/filters/karbon/pdf/PdfImport.cpp.orig	2018-01-26 09:24:44.000000000 +0000
+++ calligra-3.1.0/filters/karbon/pdf/PdfImport.cpp	2018-09-22 19:07:43.480535044 +0100
@@ -102,9 +102,6 @@
     delete globalParams;
     globalParams = 0;
 
-    // check for memory leaks
-    Object::memCheck(stderr);
-
     return KoFilter::OK;
 }
 
