--- cantor-22.04.2/cmake/FindJulia.cmake.orig	2022-06-07 04:43:15.000000000 +0100
+++ cantor-22.04.2/cmake/FindJulia.cmake	2022-06-09 13:09:03.527611229 +0100
@@ -46,6 +46,7 @@
 )
 if(RESULT EQUAL 0)
     string(REGEX REPLACE "\"" "" JULIA_INCLUDE_DIRS ${JULIA_INCLUDE_DIRS})
+    string(REGEX REPLACE "\n" "" JULIA_INCLUDE_DIRS ${JULIA_INCLUDE_DIRS})
     set(JULIA_INCLUDE_DIRS ${JULIA_INCLUDE_DIRS}
         CACHE PATH "Location of Julia include files")
 endif()
@@ -57,6 +58,13 @@
     return()
 endif()
 
+execute_process(
+    COMMAND ${JULIA_EXECUTABLE} /usr/share/julia/julia-config.jl --ldflags --ldlibs
+    OUTPUT_VARIABLE JULIA_LIBRARY
+)
+
+set( JULIA_LIBRARY "${JULIA_LIBRARY} -L/usr/lib64/julia -lLLVM-12jl -ljulia-internal" )
+string(REGEX REPLACE "\n" " " JULIA_LIBRARY ${JULIA_LIBRARY})
 
 #
 # Julia library location
