--- cantor-23.08.5/src/backends/julia/juliaserver/juliaserver.cpp.orig	2024-02-09 22:24:29.000000000 +0000
+++ cantor-23.08.5/src/backends/julia/juliaserver/juliaserver.cpp	2024-10-09 22:30:15.647437127 +0100
@@ -188,7 +188,7 @@
 
     jl_function_t* jl_names_function = jl_get_function(jl_base_module, "names");
     jl_value_t* names = jl_call1(jl_names_function, (jl_value_t*)module);
-    jl_value_t **data = (jl_value_t**)jl_array_data(names);
+    jl_value_t **data = (jl_value_t**)jl_array_data_(names);
     for (size_t i = 0; i < jl_array_len(names); i++)
     {
         bool isBindingResolved = (bool)jl_binding_resolved_p(module, (jl_sym_t*)(data[i]));
