--- clazy-20250309/src/FixItExporter.cpp.orig	2025-03-09 12:08:14.000000000 +0000
+++ clazy-20250309/src/FixItExporter.cpp	2025-03-09 17:03:33.307130135 +0000
@@ -191,7 +191,7 @@
     // clear out any current diagnostic, and let the downstream client
     // format the diagnostic.
     DiagEngine.setClient(Client, false);
-    DiagEngine.Clear();
+//    DiagEngine.Clear();
     DiagEngine.Report(Loc, DiagID);
     DiagEngine.setClient(this, false);
 }
