--- digikam-6.4.0/core/libs/facesengine/recognition/dlib-dnn/dnnface/dnn_base/cv_image.h.orig	2019-11-03 09:04:00.000000000 +0000
+++ digikam-6.4.0/core/libs/facesengine/recognition/dlib-dnn/dnnface/dnn_base/cv_image.h	2019-12-24 23:01:48.262234206 +0000
@@ -52,7 +52,7 @@
                      << "\n\t img.channels(): " << img.channels()
                      << "\n\t img.pixel_traits<pixel_type>::num: " << pixel_traits<pixel_type>::num
                      );
-        IplImage temp = img;
+        IplImage temp = cvIplImage(img);
         init(&temp);
     }
 
--- digikam-6.4.0/core/libs/facesengine/detection/opencvfacedetector.cpp.orig	2019-11-03 09:04:00.000000000 +0000
+++ digikam-6.4.0/core/libs/facesengine/detection/opencvfacedetector.cpp	2019-12-24 21:46:55.577994455 +0000
@@ -166,7 +166,7 @@
      * of the region of interest of this cascade (still relative to whole image).
      * For frontal face cascades, returns the given parameter unchanged.
      */
-    cv::Rect faceROI(const CvRect& faceRect) const
+    cv::Rect faceROI(const cv::Rect& faceRect) const
     {
         return cv::Rect(lround(faceRect.x + roi.x()      * faceRect.width),
                         lround(faceRect.y + roi.y()      * faceRect.height),
