--- digikam-6.3.0/core/libs/dplugins/setup/dpluginloader_p.cpp.orig	2019-09-04 19:05:18.000000000 +0100
+++ digikam-6.3.0/core/libs/dplugins/setup/dpluginloader_p.cpp	2019-09-30 22:31:30.990428410 +0100
@@ -77,7 +77,8 @@
 
     qCDebug(DIGIKAM_GENERAL_LOG) << "Parsing plugins from" << path;
 
-    QDirIterator  it(path, QDir::Files |
+    QDirIterator  it(path, QStringList() << QStringLiteral("*.so"),
+                           QDir::Files |
                            QDir::NoDotAndDotDot,
                            QDirIterator::Subdirectories);
     QFileInfoList allFiles;
