--- grantlee-0.5.1/CMakeLists.txt.orig	2014-09-22 12:01:25.000000000 +0100
+++ grantlee-0.5.1/CMakeLists.txt	2015-09-30 11:39:48.376636663 +0100
@@ -96,7 +96,7 @@
 set( LIB_INSTALL_DIR lib${LIB_SUFFIX} )
 set( PLUGIN_INSTALL_DIR lib${LIB_SUFFIX}/grantlee/${Grantlee_MAJOR_MINOR_VERSION_STRING} )
 set( BIN_INSTALL_DIR bin )
-set( INCLUDE_INSTALL_DIR include )
+set( INCLUDE_INSTALL_DIR include/grantlee0 )
 set( DATA_INSTALL_DIR share/apps )
 
 # set up RPATH/install_name_dir
