--- gwenview-23.08.5/CMakeLists.txt.orig	2024-02-13 03:59:31.000000000 +0000
+++ gwenview-23.08.5/CMakeLists.txt	2024-02-15 18:03:29.598574344 +0000
@@ -166,7 +166,7 @@
 endif()
 
 if (QT_MAJOR_VERSION STREQUAL "5")
-    find_package(kImageAnnotator)
+    find_package(kImageAnnotator-Qt5)
     set_package_properties(kImageAnnotator PROPERTIES URL "https://github.com/ksnip/kImageAnnotator" DESCRIPTION "The kImageAnnotator library provides tools to annotate" TYPE REQUIRED)
     if(kImageAnnotator_FOUND)
         set(KIMAGEANNOTATOR_FOUND 1)
