--- gwenview-18.12.1/lib/CMakeLists.txt.orig	2019-01-03 21:49:09.000000000 +0000
+++ gwenview-18.12.1/lib/CMakeLists.txt	2019-01-10 22:17:22.408429414 +0000
@@ -224,6 +224,54 @@
     exiv2imageloader.cpp
     imagemetainfomodel.cpp
     timeutils.cpp
+    cms/cmsprofile.cpp
+    contextmanager.cpp 
+    crop/cropimageoperation.cpp
+    crop/croptool.cpp
+    crop/cropwidget.cpp
+    document/abstractdocumentimpl.cpp
+    document/documentjob.cpp
+    document/animateddocumentloadedimpl.cpp
+    document/document.cpp
+    document/documentfactory.cpp
+    document/documentloadedimpl.cpp
+    document/emptydocumentimpl.cpp
+    document/jpegdocumentloadedimpl.cpp
+    document/loadingdocumentimpl.cpp
+    document/loadingjob.cpp
+    document/savejob.cpp
+    document/svgdocumentloadedimpl.cpp
+    document/videodocumentloadedimpl.cpp
+    documentview/abstractdocumentviewadapter.cpp
+    documentview/abstractimageview.cpp
+    documentview/abstractrasterimageviewtool.cpp
+    documentview/birdeyeview.cpp
+    documentview/documentview.cpp
+    documentview/documentviewcontroller.cpp
+    documentview/documentviewsynchronizer.cpp
+    documentview/messageviewadapter.cpp
+    documentview/rasterimageview.cpp
+    documentview/rasterimageviewadapter.cpp
+    documentview/svgviewadapter.cpp
+    documentview/videoviewadapter.cpp
+    abstractimageoperation.cpp
+    documentview/documentviewcontainer.cpp
+    imagescaler.cpp
+    jpegcontent.cpp
+    mimetypeutils.cpp
+    print/printhelper.cpp
+    print/printoptionspage.cpp
+    slideshow.cpp
+    redeyereduction/redeyereductionimageoperation.cpp
+    redeyereduction/redeyereductiontool.cpp
+    resize/resizeimageoperation.cpp
+    thumbnailprovider/thumbnailgenerator.cpp
+    thumbnailview/thumbnailbarview.cpp
+    transformimageoperation.cpp
+    mpris2/mpris2service.cpp
+    mpris2/mprismediaplayer2player.cpp
+    gwenviewconfig.cpp
+    gwenviewlib_autogen/mocs_compilation.cpp
     )
 
 ki18n_wrap_ui(gwenviewlib_SRCS
--- gwenview-18.12.1/lib/jpegcontent.cpp.orig	2019-01-03 21:49:09.000000000 +0000
+++ gwenview-18.12.1/lib/jpegcontent.cpp	2019-01-10 22:16:28.203432000 +0000
@@ -20,6 +20,8 @@
 */
 #include "jpegcontent.h"
 
+#include <exiv2/version.hpp>
+
 // System
 #include <math.h>
 #include <stdio.h>
--- gwenview-18.12.1/part/CMakeLists.txt.orig	2019-01-10 22:22:03.040416024 +0000
+++ gwenview-18.12.1/part/CMakeLists.txt	2019-01-10 22:21:49.953416649 +0000
@@ -10,6 +10,12 @@
     gvpart.cpp
     )
 
+kde_source_files_enable_exceptions(
+    gvbrowserextension.cpp
+    gvpart_autogen/mocs_compilation.cpp
+    gvpart.cpp
+    )
+
 add_library(gvpart MODULE ${gvpart_SRCS})
 
 target_link_libraries(gvpart KF5::Parts gwenviewlib)
--- gwenview-18.12.1/app/CMakeLists.txt.orig	2019-01-03 21:49:09.000000000 +0000
+++ gwenview-18.12.1/app/CMakeLists.txt	2019-01-10 22:28:30.741397526 +0000
@@ -64,6 +64,28 @@
 
 qt5_add_resources(gwenview_SRCS gwenview.qrc)
 
+kde_source_files_enable_exceptions(
+    viewmainpage.cpp
+    documentinfoprovider.cpp
+    gvcore.cpp
+    configdialog.cpp
+    fileoperations.cpp
+    fullscreencontent.cpp
+    infocontextmanageritem.cpp
+    imageopscontextmanageritem.cpp
+    main.cpp
+    mainwindow.cpp
+    preloader.cpp
+    saveallhelper.cpp
+    savebar.cpp
+    sidebar.cpp
+    startmainpage.cpp
+    thumbnailviewhelper.cpp
+    browsemainpage.cpp
+    semanticinfocontextmanageritem.cpp
+    kipiexportaction.cpp
+    gwenview_autogen/mocs_compilation.cpp
+    )
 
 if (NOT GWENVIEW_SEMANTICINFO_BACKEND_NONE)
     ki18n_wrap_ui(gwenview_SRCS
