--- gwenview-18.12.3/lib/CMakeLists.txt.orig	2019-02-28 01:40:15.000000000 +0000
+++ gwenview-18.12.3/lib/CMakeLists.txt	2019-03-08 20:47:23.975867420 +0000
@@ -226,6 +226,8 @@
     timeutils.cpp
     )
 
+kde_enable_exceptions()
+
 ki18n_wrap_ui(gwenviewlib_SRCS
     crop/cropwidget.ui
     documentview/messageview.ui
--- gwenview-18.12.3/lib/jpegcontent.cpp.orig	2019-02-28 01:40:15.000000000 +0000
+++ gwenview-18.12.3/lib/jpegcontent.cpp	2019-03-08 20:49:12.630507737 +0000
@@ -20,6 +20,8 @@
 */
 #include "jpegcontent.h"
 
+#include <exiv2/version.hpp>
+
 // System
 #include <math.h>
 #include <stdio.h>
--- gwenview-18.12.3/part/CMakeLists.txt.orig	2019-02-28 01:40:15.000000000 +0000
+++ gwenview-18.12.3/part/CMakeLists.txt	2019-03-08 20:51:35.572906328 +0000
@@ -10,6 +10,8 @@
     gvpart.cpp
     )
 
+kde_enable_exceptions()
+
 add_library(gvpart MODULE ${gvpart_SRCS})
 
 target_link_libraries(gvpart KF5::Parts gwenviewlib)
--- gwenview-18.12.3/importer/CMakeLists.txt.orig	2019-02-28 01:40:15.000000000 +0000
+++ gwenview-18.12.3/importer/CMakeLists.txt	2019-03-08 20:51:20.822907032 +0000
@@ -20,6 +20,8 @@
     thumbnailpage.cpp
     )
 
+kde_enable_exceptions()
+
 ki18n_wrap_ui(importer_SRCS
     dialogpage.ui
     importerconfigdialog.ui
--- gwenview-18.12.3/app/CMakeLists.txt.orig	2019-02-28 01:40:15.000000000 +0000
+++ gwenview-18.12.3/app/CMakeLists.txt	2019-03-08 20:51:50.932905595 +0000
@@ -34,6 +34,8 @@
     browsemainpage.cpp
     )
 
+kde_enable_exceptions()
+
 if (NOT GWENVIEW_SEMANTICINFO_BACKEND_NONE)
     set (gwenview_SRCS
         ${gwenview_SRCS}
