diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3215e94..d24e848 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -93,7 +93,11 @@ add_subdirectory(icons)
 add_subdirectory(profiles)
 add_subdirectory(src)
 if (IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/po")
-  add_subdirectory(po)
+  if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/po/CMakeLists.txt" )
+    add_subdirectory(po)
+  else()
+    ki18n_install(po)
+  endif()
 endif()
 
 if(KF5DocTools_VERSION)
