--- kalgebra-16.04.3/calgebra/CMakeLists.txt.orig	2016-04-06 18:57:19.000000000 +0100
+++ kalgebra-16.04.3/calgebra/CMakeLists.txt	2016-07-18 21:02:28.447686541 +0100
@@ -13,7 +13,7 @@
 endif()
 
 add_executable(calgebra main.cpp)
-target_link_libraries(calgebra Qt5::Core ${READLINE_LIBRARY} KF5::Analitza)
+target_link_libraries(calgebra Qt5::Core ${READLINE_LIBRARY} ${CURSES_LIBRARIES} KF5::Analitza)
 
 install(TARGETS calgebra ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
 
