--- kamera-17.04.2/solid_camera.desktop.orig	2017-06-05 19:57:45.000000000 +0100
+++ kamera-17.04.2/solid_camera.desktop	2017-07-04 20:41:54.436305800 +0100
@@ -6,7 +6,7 @@
 [Desktop Action open]
 Exec=kioclient5 exec camera:/
 Icon=system-file-manager
-Name=Open with File Manager
+Name=Open with File Manager (camera)
 Name[ar]=افتح بمدير الملفات
 Name[ast]=Abrir col xestor de ficheros
 Name[bg]=Отваряне в файлов браузър
@@ -17,7 +17,7 @@
 Name[da]=Åbn med filhåndtering
 Name[de]=Mit Dateiverwaltung öffnen
 Name[el]=Άνοιγμα με τον διαχειριστή αρχείων
-Name[en_GB]=Open with File Manager
+Name[en_GB]=Open with File Manager (camera)
 Name[es]=Abrir con el gestor de archivos
 Name[et]=Avamine failihalduriga
 Name[eu]=Ireki fitxategi kudeatzailearekin
