--- kate-19.08.1/addons/kate-ctags/kate_ctags_plugin.cpp.orig	2019-09-01 07:22:28.000000000 +0100
+++ kate-19.08.1/addons/kate-ctags/kate_ctags_plugin.cpp	2019-09-16 20:01:00.326988683 +0100
@@ -216,7 +216,7 @@
         if (target.endsWith(QLatin1Char('/')) || target.endsWith(QLatin1Char('\\'))) {
             target = target.left(target.size() - 1);
         }
-        targets += target + QLatin1Char(' ');
+        targets += QLatin1Char('"') + target + QLatin1String("\" ");
     }
 
     QString file = QStandardPaths::writableLocation(QStandardPaths::DataLocation) + QLatin1String("/katectags");
--- kate-19.08.1/addons/kate-ctags/kate_ctags_view.cpp.orig	2019-09-01 07:22:28.000000000 +0100
+++ kate-19.08.1/addons/kate-ctags/kate_ctags_view.cpp	2019-09-16 19:58:03.602997115 +0100
@@ -520,7 +520,7 @@
       if (target.endsWith(QLatin1Char('/')) || target.endsWith(QLatin1Char('\\'))) {
         target = target.left(target.size() - 1);
       }
-      targets += target + QLatin1Char(' ');
+      targets += QLatin1Char('"') + target + QLatin1String("\" ");
     }
 
     QString pluginFolder = QStandardPaths::writableLocation(QStandardPaths::DataLocation) + QLatin1String("/katectags");
