--- kde-runtime-15.08.3/solid-networkstatus/kded/CMakeLists.txt.orig	2015-11-10 22:19:41.604660070 +0000
+++ kde-runtime-15.08.3/solid-networkstatus/kded/CMakeLists.txt	2015-11-10 22:19:19.569659988 +0000
@@ -18,9 +18,10 @@
 if(NETWORKMANAGER_FOUND)
     macro_ensure_version("0.7.0" ${NETWORKMANAGER_VERSION} NM_0_7)
     if(NM_0_7)
+        find_package(GLIB2)
         MESSAGE(STATUS "Adding in-process NetworkManager service-wart to kded module")
         set(kded_networkstatus_PART_SRCS ${kded_networkstatus_PART_SRCS} networkmanagerstatus.cpp)
-        include_directories(${NETWORKMANAGER_INCLUDE_DIRS})
+        include_directories(${NETWORKMANAGER_INCLUDE_DIRS} ${GLIB2_MAIN_INCLUDE_DIR} ${GLIB2_INTERNAL_INCLUDE_DIR})
         add_definitions(-DNM_BACKEND_ENABLED)
     endif(NM_0_7)
 endif(NETWORKMANAGER_FOUND)
