--- kde-runtime-15.08.2/solid-networkstatus/kded/CMakeLists.txt.orig	2015-09-27 07:57:45.000000000 +0100
+++ kde-runtime-15.08.2/solid-networkstatus/kded/CMakeLists.txt	2015-10-14 13:03:35.668447766 +0100
@@ -20,7 +20,7 @@
     if(NM_0_7)
         MESSAGE(STATUS "Adding in-process NetworkManager service-wart to kded module")
         set(kded_networkstatus_PART_SRCS ${kded_networkstatus_PART_SRCS} networkmanagerstatus.cpp)
-        include_directories(${NETWORKMANAGER_INCLUDE_DIRS})
+        include_directories(${NETWORKMANAGER_INCLUDE_DIRS} ${GLIB2_MAIN_INCLUDE_DIR} ${GLIB2_INTERNAL_INCLUDE_DIR})
         add_definitions(-DNM_BACKEND_ENABLED)
     endif(NM_0_7)
 endif(NETWORKMANAGER_FOUND)
