--- kdoctools-5.40.0/src/customization/pt-BR/user.entities.orig	2017-11-05 08:29:48.000000000 +0000
+++ kdoctools-5.40.0/src/customization/pt-BR/user.entities	2017-12-02 15:10:48.047311789 +0000
@@ -11,6 +11,11 @@
 <!ENTITY Backspace	"<keycap>Backspace</keycap>">
 <!ENTITY cdrom	'<hardware>leitor de CD-ROM</hardware>'>
 <!ENTITY Ctrl	"<keycap>Ctrl</keycap>">
+<!ENTITY PgUp   "<keycap>PgUp</keycap>">
+<!ENTITY PgDn   "<keycap>PgDn</keycap>">
+<!ENTITY Space  "<keycap>Space</keycap>">
+<!ENTITY Left   "<keycap>Left</keycap>">
+<!ENTITY Right  "<keycap>Right</keycap>">
 <!ENTITY dpi	'<acronym>ppp</acronym>'>
 <!ENTITY eg     "por exemplo">
 <!ENTITY Enter	"<keycap>Enter</keycap>">
