--- kexi-20241102/CMakeLists.txt.orig	2024-11-02 09:44:25.626206909 +0000
+++ kexi-20241102/CMakeLists.txt	2024-11-02 10:10:56.616029515 +0000
@@ -110,7 +110,7 @@
                        PURPOSE "Used to provide crash reporting on Linux")
 
 find_package(Qt5 ${QT_MIN_VERSION} REQUIRED COMPONENTS Core Gui Widgets Xml Network PrintSupport Svg Test)
-find_package(Qt5 ${QT_MIN_VERSION} COMPONENTS UiTools WebKit WebKitWidgets)
+find_package(Qt5 ${QT_MIN_VERSION} COMPONENTS UiTools)
 
 # use sane compile flags
 add_definitions(
