--- kid3-3.4.5/src/core/config/importconfig.cpp.orig	2017-02-06 21:53:19.000000000 +0000
+++ kid3-3.4.5/src/core/config/importconfig.cpp	2017-02-21 19:54:41.724014209 +0000
@@ -89,7 +89,7 @@
 
   m_importFormatNames.append(QLatin1String("CSV quoted"));
   m_importFormatHeaders.append(QLatin1String(""));
-  m_importFormatTracks.append(QLatin1String("\"?%{track}(\\d+)\"?\\t\"?%{title}([^\\r\\n\\t\"]*)\"?\\t\"?%{artist}([^\\r\\n\\t\"]*)\"?\\t\"?%{album}([^\\r\\n\\t\"]*)\"?\\t\"?%{year}([^\\r\\n\\t]*)\"?\\t\"?%{genre}([^\\r\\n\\t\"]*)\"?\\t\"?%{comment}([^\\r\\n\\t\"]*)\"?\\t\"?(?:\\d+:)?%{duration}(\\d+:\\d+)"));
+  m_importFormatTracks.append(QLatin1String("\"?%{track}(\\d+)\"?\\t\"?%{title}([^\\r\\n\\t\"]*)\"?\\t\"?%{artist}([^\\r\\n\\t\"]*)\"?\\t\"?%{album}([^\\r\\n\\t\"]*)\"?\\t\"?%{year}([^\\r\\n\\t\"]*)\"?\\t\"?%{genre}([^\\r\\n\\t\"]*)\"?\\t\"?%{comment}([^\\r\\n\\t\"]*)\"?\\t\"?(?:\\d+:)?%{duration}(\\d+:\\d+)"));
 
   m_importFormatNames.append(QLatin1String("CSV more unquoted"));
   m_importFormatHeaders.append(QLatin1String(""));
@@ -114,7 +114,7 @@
   m_importFormatTracks.append(QLatin1String(
     "\"?%{track}(\\d+)\"?\\t\"?%{title}([^\\r\\n\\t\"]*)\"?\\t"
     "\"?%{artist}([^\\r\\n\\t\"]*)\"?\\t\"?%{album}([^\\r\\n\\t\"]*)\"?\\t"
-    "\"?%{year}([^\\r\\n\\t]*)\"?\\t\"?%{genre}([^\\r\\n\\t\"]*)\"?\\t"
+    "\"?%{year}([^\\r\\n\\t\"]*)\"?\\t\"?%{genre}([^\\r\\n\\t\"]*)\"?\\t"
     "\"?%{comment}([^\\r\\n\\t\"]*)\"?\\t"
     "\"?(?:\\d+:)?%{duration}(\\d+:\\d+)(?:\\.\\d+)?\"?\\t"
     "\"?%{album artist}([^\\r\\n\\t\"]*)\"?\\t\"?%{arranger}([^\\r\\n\\t\"]*)\"?\\t"
