--- kig-19.04.0/cmake/modules/FindBoostPython.cmake.orig	2019-01-26 02:05:47.000000000 +0000
+++ kig-19.04.0/cmake/modules/FindBoostPython.cmake	2019-04-21 22:11:58.873915335 +0100
@@ -101,6 +101,9 @@
 
 cmake_pop_check_state()
 
+set(BoostPython_INCLUDE_DIRS boost_headers_DIR ${BoostPython_INCLUDE_DIRS} ${PYTHON_INCLUDE_DIRS})
+set(BoostPython_LIBRARIES boost_python27 python2.7 ${BoostPython_LIBRARIES})
+
 find_package_handle_standard_args(BoostPython
                                   FOUND_VAR BoostPython_FOUND
                                   REQUIRED_VARS BoostPython_LIBRARIES BoostPython_INCLUDE_DIRS
