--- kio-extras-17.04.2/mtp/solid_mtp.desktop.orig	2017-05-22 04:30:06.000000000 +0100
+++ kio-extras-17.04.2/mtp/solid_mtp.desktop	2017-07-04 20:42:52.159301507 +0100
@@ -6,7 +6,7 @@
 [Desktop Action open]
 Exec=kioclient5 exec mtp:udi=%i/
 Icon=system-file-manager
-Name=Open with File Manager
+Name=Open with File Manager (mtp)
 Name[ar]=افتح بمدير الملفّات
 Name[ast]=Abrir col xestor de ficheros
 Name[bg]=Отваряне с файлов мениджър
@@ -17,7 +17,7 @@
 Name[da]=Åbn med filhåndtering
 Name[de]=Mit Dateiverwaltung öffnen
 Name[el]=Άνοιγμα με τον διαχειριστή αρχείων
-Name[en_GB]=Open with File Manager
+Name[en_GB]=Open with File Manager (mtp)
 Name[eo]=Malfermi per &defaŭlta retumilo
 Name[es]=Abrir con el gestor de archivos
 Name[et]=Ava failihalduriga
