--- kitinerary-21.04.3/src/lib/pdf/pdfdocument.cpp.orig	2021-07-05 20:24:08.000000000 +0100
+++ kitinerary-21.04.3/src/lib/pdf/pdfdocument.cpp	2021-08-01 22:27:23.570408368 +0100
@@ -205,8 +205,9 @@
 {
     int year, month, day, hour, min, sec, tzHours, tzMins;
     char tz;
+    GooString gstr = GooString(str);
 
-    if (!parseDateString(str, &year, &month, &day, &hour, &min, &sec, &tz, &tzHours, &tzMins)) {
+    if (!parseDateString(&gstr, &year, &month, &day, &hour, &min, &sec, &tz, &tzHours, &tzMins)) {
         return {};
     }
 
