--- kitinerary-20.04.3/CMakeLists.txt.orig	2020-07-03 22:33:53.000000000 +0100
+++ kitinerary-20.04.3/CMakeLists.txt	2020-08-03 19:36:37.444836786 +0100
@@ -72,6 +72,10 @@
     list(GET _poppler_version_components 0 POPPLER_VERSION_MAJOR)
     list(GET _poppler_version_components 1 POPPLER_VERSION_MINOR)
     list(GET _poppler_version_components 2 POPPLER_VERSION_PATCH)
+
+    string(REGEX REPLACE "^0*([0-9].*)" "\\1" POPPLER_VERSION_MAJOR ${POPPLER_VERSION_MAJOR})
+    string(REGEX REPLACE "^0*([0-9].*)" "\\1" POPPLER_VERSION_MINOR ${POPPLER_VERSION_MINOR})
+    string(REGEX REPLACE "^0*([0-9].*)" "\\1" POPPLER_VERSION_PATCH ${POPPLER_VERSION_PATCH})
 endif()
 
 if (TARGET ZXing::ZXing OR TARGET ZXing::Core)
