--- kitinerary-19.08.2/src/pdf/pdfextractoroutputdevice_p.h.orig	2019-09-30 06:37:37.000000000 +0100
+++ kitinerary-19.08.2/src/pdf/pdfextractoroutputdevice_p.h	2019-10-28 21:01:44.321029375 +0000
@@ -43,7 +43,7 @@
     void finalize();
 
     bool needNonText() override { return true; }
-    void drawImage(GfxState *state, Object *ref, Stream *str, int width, int height, GfxImageColorMap *colorMap, bool interpolate, int *maskColors, bool inlineImg) override;
+    void drawImage(GfxState *state, Object *ref, Stream *str, int width, int height, GfxImageColorMap *colorMap, bool interpolate, const int *maskColors, bool inlineImg) override;
 
     // operations used to detect vector barcodes
     void saveState(GfxState *state) override;
--- kitinerary-19.08.2/src/pdf/pdfextractoroutputdevice.cpp.orig	2019-09-30 06:37:37.000000000 +0100
+++ kitinerary-19.08.2/src/pdf/pdfextractoroutputdevice.cpp	2019-10-28 21:02:04.752028400 +0000
@@ -30,7 +30,7 @@
 {
 }
 
-void PdfExtractorOutputDevice::drawImage(GfxState* state, Object* ref, Stream* str, int width, int height, GfxImageColorMap* colorMap, bool interpolate, int* maskColors, bool inlineImg)
+void PdfExtractorOutputDevice::drawImage(GfxState* state, Object* ref, Stream* str, int width, int height, GfxImageColorMap* colorMap, bool interpolate, const int* maskColors, bool inlineImg)
 {
     Q_UNUSED(str);
     Q_UNUSED(interpolate);
