--- kpimtextedit-16.04.3/src/texttospeech/texttospeech.cpp.orig	2016-06-30 12:49:53.000000000 +0100
+++ kpimtextedit-16.04.3/src/texttospeech/texttospeech.cpp	2016-07-21 21:36:28.658872860 +0100
@@ -70,7 +70,8 @@
 #if QT_VERSION < QT_VERSION_CHECK(5, 7, 0)
         mTextToSpeech = new QTextToSpeech(this, engineName);
 #else
-        mTextToSpeech = new QTextToSpeech(engineName, QVariantMap(), this);
+        mTextToSpeech = new QTextToSpeech(this, engineName);
+//        mTextToSpeech = new QTextToSpeech(engineName, QVariantMap(), this);
 #endif
         connect(mTextToSpeech, &QTextToSpeech::stateChanged, this, &TextToSpeech::slotStateChanged);
     } else if (mDefaultEngine != engineName) {
@@ -79,7 +80,8 @@
 #if QT_VERSION < QT_VERSION_CHECK(5, 7, 0)
         mTextToSpeech = new QTextToSpeech(this, engineName);
 #else
-        mTextToSpeech = new QTextToSpeech(engineName, QVariantMap(), this);
+        mTextToSpeech = new QTextToSpeech(this, engineName);
+//        mTextToSpeech = new QTextToSpeech(engineName, QVariantMap(), this);
 #endif
         connect(mTextToSpeech, &QTextToSpeech::stateChanged, this, &TextToSpeech::slotStateChanged);
     }
--- kpimtextedit-16.04.3/src/texttospeech/texttospeechconfiginterface.cpp.orig	2016-06-30 12:49:53.000000000 +0100
+++ kpimtextedit-16.04.3/src/texttospeech/texttospeechconfiginterface.cpp	2016-07-21 21:36:58.435874975 +0100
@@ -80,7 +80,8 @@
 #if QT_VERSION < QT_VERSION_CHECK(5, 7, 0)
     mTextToSpeech = new QTextToSpeech(this, engineName);
 #else
-    mTextToSpeech = new QTextToSpeech(engineName, QVariantMap(), this);
+    mTextToSpeech = new QTextToSpeech(this, engineName);
+//    mTextToSpeech = new QTextToSpeech(engineName, QVariantMap(), this);
 #endif
 #endif
 }
