--- kpimtextedit-16.08.0/src/texttospeech/texttospeech.cpp.orig	2016-08-18 20:57:53.881744602 +0100
+++ kpimtextedit-16.08.0/src/texttospeech/texttospeech.cpp	2016-08-18 21:00:13.599745117 +0100
@@ -69,12 +69,12 @@
     KConfigGroup grp = config.group("Settings");
     const QString engineName = grp.readEntry("engine");
     if (!mTextToSpeech) {
-        mTextToSpeech = new QTextToSpeech(engineName, this);
+        mTextToSpeech = new QTextToSpeech(this, engineName);
         connect(mTextToSpeech, &QTextToSpeech::stateChanged, this, &TextToSpeech::slotStateChanged);
     } else if (mDefaultEngine != engineName) {
         disconnect(mTextToSpeech, &QTextToSpeech::stateChanged, this, &TextToSpeech::slotStateChanged);
         delete mTextToSpeech;
-        mTextToSpeech = new QTextToSpeech(engineName, this);
+        mTextToSpeech = new QTextToSpeech(this, engineName);
         connect(mTextToSpeech, &QTextToSpeech::stateChanged, this, &TextToSpeech::slotStateChanged);
     }
     mDefaultEngine = engineName;
--- kpimtextedit-16.08.0/src/texttospeech/texttospeechconfiginterface.cpp.orig	2016-08-18 20:57:53.882744602 +0100
+++ kpimtextedit-16.08.0/src/texttospeech/texttospeechconfiginterface.cpp	2016-08-18 21:00:50.170745252 +0100
@@ -79,6 +79,6 @@
 {
 #if KPIMTEXTEDIT_HAVE_TEXTTOSPEECH
     delete mTextToSpeech;
-    mTextToSpeech = new QTextToSpeech(engineName, this);
+    mTextToSpeech = new QTextToSpeech(this, engineName);
 #endif
 }
