--- krita-5.2.11/CMakeLists.txt.orig	2025-07-12 20:30:12.000000000 +0100
+++ krita-5.2.11/CMakeLists.txt	2025-08-16 17:21:53.729290494 +0100
@@ -813,7 +813,7 @@
     set(Boost_COMPILER "mgw${KRITA_boost_COMPILER_VERSION}")
 endif()
 
-find_package(Boost 1.65 REQUIRED COMPONENTS system)
+find_package(Boost 1.65 REQUIRED)
 set_package_properties(Boost PROPERTIES
     DESCRIPTION "Boost provides free peer-reviewed portable C++ source libraries."
     URL "https://www.boost.org"
--- krita-5.2.11/cmake/modules/pyproject.toml.in.orig	2025-07-12 20:30:12.000000000 +0100
+++ krita-5.2.11/cmake/modules/pyproject.toml.in	2025-07-16 09:17:15.496298509 +0100
@@ -9,7 +9,7 @@
 sip-module = "@sip_name@"
 sip-include-dirs = @sip_include_dirs@
 sip-files-dir = "@module_srcs@"
-abi-version = "12"
+abi-version = "12.8"
 
 [tool.sip.bindings.@module_name_toml@]
 pep484-pyi = true
