--- kscreenlocker-5.11.5/greeter/seccomp_filter.cpp.orig	2018-01-02 11:54:36.681806206 +0000
+++ kscreenlocker-5.11.5/greeter/seccomp_filter.cpp	2018-01-03 22:22:27.662294377 +0000
@@ -58,7 +58,7 @@
                 // BUG: 384005
                 writeSupported = false;
             }
-            else if (gl->isSoftwareEmulation() && KWindowSystem::isPlatformWayland()) {
+            else if ((gl->isSoftwareEmulation() && KWindowSystem::isPlatformWayland()) || gl->driver() == KWin::Driver_Intel) {
                 createSupported = writeSupported = false;
             }
         }
