--- labplot-2.8.2/src/kdefrontend/datasources/DatabaseManagerWidget.cpp.orig	2021-03-26 12:07:25.000000000 +0000
+++ labplot-2.8.2/src/kdefrontend/datasources/DatabaseManagerWidget.cpp	2022-05-19 10:48:56.762171655 +0100
@@ -42,9 +42,9 @@
 #include <QSqlError>
 
 #ifdef HAVE_KF5_SYNTAX_HIGHLIGHTING
-#include <KF5/KSyntaxHighlighting/SyntaxHighlighter>
-#include <KF5/KSyntaxHighlighting/Definition>
-#include <KF5/KSyntaxHighlighting/Theme>
+#include <KSyntaxHighlighting/SyntaxHighlighter>
+#include <KSyntaxHighlighting/Definition>
+#include <KSyntaxHighlighting/Theme>
 #endif
 
 /*!
--- labplot-2.8.2/src/kdefrontend/datasources/ImportSQLDatabaseWidget.h.orig	2021-03-26 12:07:25.000000000 +0000
+++ labplot-2.8.2/src/kdefrontend/datasources/ImportSQLDatabaseWidget.h	2022-05-19 10:47:26.245695886 +0100
@@ -35,7 +35,7 @@
 #include "ui_importsqldatabasewidget.h"
 
 #ifdef HAVE_KF5_SYNTAX_HIGHLIGHTING
-#include <repository.h>
+#include <KSyntaxHighlighting/repository.h>
 namespace KSyntaxHighlighting {
 	class SyntaxHighlighter;
 }
--- labplot-2.8.2/src/kdefrontend/datasources/DatabaseManagerWidget.h.orig	2021-03-26 12:07:25.000000000 +0000
+++ labplot-2.8.2/src/kdefrontend/datasources/DatabaseManagerWidget.h	2022-05-19 10:47:12.775435035 +0100
@@ -31,7 +31,7 @@
 #include "ui_databasemanagerwidget.h"
 
 #ifdef HAVE_KF5_SYNTAX_HIGHLIGHTING
-#include <repository.h>
+#include <KSyntaxHighlighting/repository.h>
 namespace KSyntaxHighlighting {
 	class SyntaxHighlighter;
 }
--- labplot-2.8.2/src/kdefrontend/datasources/ImportSQLDatabaseWidget.cpp.orig	2021-03-26 12:07:25.000000000 +0000
+++ labplot-2.8.2/src/kdefrontend/datasources/ImportSQLDatabaseWidget.cpp	2022-05-19 10:49:19.689170561 +0100
@@ -45,9 +45,9 @@
 #include <KMessageBox>
 #include <KSharedConfig>
 #ifdef HAVE_KF5_SYNTAX_HIGHLIGHTING
-#include <KF5/KSyntaxHighlighting/SyntaxHighlighter>
-#include <KF5/KSyntaxHighlighting/Definition>
-#include <KF5/KSyntaxHighlighting/Theme>
+#include <KSyntaxHighlighting/SyntaxHighlighter>
+#include <KSyntaxHighlighting/Definition>
+#include <KSyntaxHighlighting/Theme>
 #endif
 
 ImportSQLDatabaseWidget::ImportSQLDatabaseWidget(QWidget* parent) : QWidget(parent) {
--- labplot-2.8.2/src/kdefrontend/widgets/LabelWidget.h.orig	2021-03-26 12:07:25.000000000 +0000
+++ labplot-2.8.2/src/kdefrontend/widgets/LabelWidget.h	2022-05-19 10:47:01.385322887 +0100
@@ -35,7 +35,7 @@
 #include <KConfigGroup>
 
 #ifdef HAVE_KF5_SYNTAX_HIGHLIGHTING
-#include <repository.h>
+#include <KSyntaxHighlighting/repository.h>
 namespace KSyntaxHighlighting {
 	class SyntaxHighlighter;
 }
--- labplot-2.8.2/src/kdefrontend/widgets/LabelWidget.cpp.orig	2021-03-26 12:07:25.000000000 +0000
+++ labplot-2.8.2/src/kdefrontend/widgets/LabelWidget.cpp	2022-05-19 10:49:42.752169461 +0100
@@ -45,9 +45,9 @@
 #include <KCharSelect>
 #include <KLocalizedString>
 #ifdef HAVE_KF5_SYNTAX_HIGHLIGHTING
-#include <KF5/KSyntaxHighlighting/SyntaxHighlighter>
-#include <KF5/KSyntaxHighlighting/Definition>
-#include <KF5/KSyntaxHighlighting/Theme>
+#include <KSyntaxHighlighting/SyntaxHighlighter>
+#include <KSyntaxHighlighting/Definition>
+#include <KSyntaxHighlighting/Theme>
 #endif
 
 #include <cmath>
