--- phonon-4.8.3/cmake/FindPhononInternal.cmake.orig	2014-12-04 09:30:26.000000000 +0000
+++ phonon-4.8.3/cmake/FindPhononInternal.cmake	2016-03-25 14:33:09.982088572 +0000
@@ -409,7 +409,7 @@
       file(WRITE "${_source_file}" "${_source}")
       set(_include_dirs "-DINCLUDE_DIRECTORIES:STRING=${QT_INCLUDES}")
 
-      try_compile(_compile_result ${CMAKE_BINARY_DIR} ${_source_file} CMAKE_FLAGS "${_include_dirs}" OUTPUT_VARIABLE _compile_output_var)
+      try_compile(_compile_result ${CMAKE_BINARY_DIR} ${_source_file} CMAKE_FLAGS "${_include_dirs}" LINK_LIBRARIES "${Qt5Core_LIBRARIES}"  OUTPUT_VARIABLE _compile_output_var)
 
       if(NOT _compile_result)
          message("${_compile_output_var}")
