--- plasma-nm-5.5.0/vpn/openconnect/openconnectauth.cpp.orig	2015-12-03 12:30:03.000000000 +0000
+++ plasma-nm-5.5.0/vpn/openconnect/openconnectauth.cpp	2015-12-13 07:13:20.736657631 +0000
@@ -585,7 +585,7 @@
     Q_D(OpenconnectAuthWidget);
 
     const int lastIndex = d->ui.loginBoxLayout->count() - 1;
-    QLayout *layout = d->ui.loginBoxLayout->itemAt(lastIndex - 2)->layout();
+    QLayout *layout = d->ui.loginBoxLayout->itemAt(lastIndex - 1)->layout();
     struct oc_auth_form *form = (struct oc_auth_form *) d->ui.loginBoxLayout->itemAt(lastIndex)->widget()->property("openconnect_form").value<quintptr>();
 
     for (int i = 0; i < layout->count(); i++) {
