--- plasma-workspace-5.13.5/dataengines/geolocation/location_gps.cpp.orig	2018-09-04 11:22:03.000000000 +0100
+++ plasma-workspace-5.13.5/dataengines/geolocation/location_gps.cpp	2018-10-03 19:58:28.824866859 +0100
@@ -51,7 +51,9 @@
     while (!m_abort) {
         Plasma::DataEngine::Data d;
 
-#if GPSD_API_MAJOR_VERSION >= 5
+#if GPSD_API_MAJOR_VERSION >= 7
+	if (gps_read(m_gpsdata, NULL, 0) != -1) {
+#elif GPSD_API_MAJOR_VERSION >= 5
 	if (gps_read(m_gpsdata) != -1) {
 #else
         if (gps_poll(m_gpsdata) != -1) {
