--- sddm-0.14.0/data/scripts/Xsession.orig	2016-08-28 22:13:37.293002799 +0100
+++ sddm-0.14.0/data/scripts/Xsession	2016-08-29 22:03:52.267695444 +0100
@@ -5,6 +5,8 @@
 # This file is extracted from kde-workspace (kdm/kfrontend/genkdmconf.c)
 # Copyright (C) 2001-2005 Oswald Buddenhagen <ossi@kde.org>
 
+session=$1
+
 # Note that the respective logout scripts are not sourced.
 case $SHELL in
   */bash)
@@ -88,8 +90,18 @@
   . "$USERXSESSION"
 fi
 
-if [ -z "$@" ]; then
+#if [ -z "$@" ]; then
+#    exec xmessage -center -buttons OK:0 -default OK "Sorry, $DESKTOP_SESSION is no valid session."
+#else
+#    exec $@
+#fi
+
+case $session in
+  "")
     exec xmessage -center -buttons OK:0 -default OK "Sorry, $DESKTOP_SESSION is no valid session."
-else
-    exec $@
-fi
+    ;;
+  *)
+    eval exec ck-launch-session "$session"
+    ;;
+esac
+exec xmessage -center -buttons OK:0 -default OK "Sorry, cannot execute $session. Check $DESKTOP_SESSION.desktop."
