--- sddm-0.18.0/services/sddm.pam.orig	2018-07-18 11:31:40.000000000 +0100
+++ sddm-0.18.0/services/sddm.pam	2020-05-21 21:30:07.031477978 +0100
@@ -1,15 +1,15 @@
 #%PAM-1.0
 
-auth		include		system-login
+auth		substack	system-login
 -auth		optional	pam_gnome_keyring.so
 -auth   optional  pam_kwallet5.so
 
 account		include		system-login
 
-password	include		system-login
+password	substack	system-login
 -password	optional	pam_gnome_keyring.so use_authtok
 
 session		optional	pam_keyinit.so force revoke
-session		include		system-login
+session		substack	system-login
 -session		optional	pam_gnome_keyring.so auto_start
 -session  optional  pam_kwallet5.so auto_start
