--- tellico-3.3.1/CMakeLists.txt.orig	2020-06-01 01:57:12.000000000 +0100
+++ tellico-3.3.1/CMakeLists.txt	2020-08-03 20:49:09.033053145 +0100
@@ -128,7 +128,7 @@
     TYPE OPTIONAL)
 if(TAGLIB_FOUND)
     add_definitions(${TAGLIB_CFLAGS})
-    include_directories(${TAGLIB_INCLUDES})
+    include_directories(${TAGLIB_INCLUDES} ${Taglib_INCLUDE_DIRS})
 endif(TAGLIB_FOUND)
 
 find_package(Yaz 2.0)
--- tellico-3.3.1/src/tests/CMakeLists.txt.orig	2020-08-03 20:40:34.080086195 +0100
+++ tellico-3.3.1/src/tests/CMakeLists.txt	2020-08-03 20:38:35.851091836 +0100
@@ -224,7 +224,7 @@
 ecm_mark_nongui_executable(audiofiletest)
 add_test(audiofiletest audiofiletest)
 ecm_mark_as_test(audiofiletest)
-TARGET_LINK_LIBRARIES(audiofiletest ${TELLICO_TEST_LIBS} ${TAGLIB_LIBRARIES})
+TARGET_LINK_LIBRARIES(audiofiletest ${TELLICO_TEST_LIBS} ${TAGLIB_LIBRARIES} ${Taglib_LIBRARIES})
 
 ENDIF( TAGLIB_FOUND )
 
--- tellico-3.3.1/src/translators/CMakeLists.txt.orig	2020-06-01 01:56:03.000000000 +0100
+++ tellico-3.3.1/src/translators/CMakeLists.txt	2020-08-03 20:49:27.575052260 +0100
@@ -88,7 +88,7 @@
 ENDIF( Exempi_FOUND )
 
 IF( TAGLIB_FOUND )
-  TARGET_LINK_LIBRARIES(translators ${TAGLIB_LIBRARIES})
+  TARGET_LINK_LIBRARIES(translators ${TAGLIB_LIBRARIES} ${Taglib_LIBRARIES})
 ENDIF( TAGLIB_FOUND )
 
 IF( CDIO_FOUND )
