--- tellico-3.2.1/src/gui/datewidget.cpp.orig	2019-07-10 01:59:41.238344217 +0100
+++ tellico-3.2.1/src/gui/datewidget.cpp	2019-08-24 18:15:09.151724052 +0100
@@ -34,6 +34,7 @@
 
 #include <QPushButton>
 #include <QHBoxLayout>
+#include <QDate>
 #include <QFrame>
 #include <QEvent>
 #include <QMenu>
--- tellico-3.2.1/src/mainwindow.cpp.orig	2019-07-10 01:59:41.238344217 +0100
+++ tellico-3.2.1/src/mainwindow.cpp	2019-08-24 19:18:03.421820557 +0100
@@ -104,6 +104,7 @@
 #include <QSignalMapper>
 #include <QTimer>
 #include <QMetaObject> // needed for copy, cut, paste slots
+#include <QMetaMethod>
 #include <QMimeDatabase>
 #include <QMimeType>
 #include <QMenuBar>
