--- ./php.ini-recommended.orig	2010-08-15 19:26:39.000000000 -0500
+++ ./php.ini-recommended	2010-12-22 18:19:00.466000177 -0600
@@ -539,7 +539,7 @@
 user_dir =
 
 ; Directory in which the loadable extensions (modules) reside.
-extension_dir = "./"
+extension_dir = "/usr/lib/php/extensions/"
 
 ; Whether or not to enable the dl() function.  The dl() function does NOT work
 ; properly in multithreaded servers, such as IIS or Zeus, and is automatically
@@ -654,6 +654,42 @@
 ; needs to go here.  Specify the location of the extension with the
 ; extension_dir directive above.
 
+; These are the extensions that are available for use on this PHP build.
+; If you do NOT need any of these extensions, you should comment them out.
+; That will probably increase both your security and performance.
+
+extension=bcmath.so
+extension=bz2.so
+extension=calendar.so
+extension=ctype.so
+extension=curl.so
+extension=dba.so
+extension=dbase.so
+extension=exif.so
+extension=ftp.so
+extension=gettext.so
+extension=gd.so
+extension=gmp.so
+extension=iconv.so
+extension=ldap.so
+extension=mbstring.so
+extension=mhash.so
+extension=mysql.so
+extension=mysqli.so
+extension=openssl.so
+extension=pdo.so
+extension=pdo_mysql.so
+extension=pdo_sqlite.so
+extension=pspell.so
+extension=shmop.so
+extension=snmp.so
+extension=soap.so
+extension=sqlite.so
+extension=tokenizer.so
+extension=wddx.so
+extension=xsl.so
+extension=zip.so
+extension=zlib.so
 
 ; Windows Extensions
 ; Note that ODBC support is built in, so no dll is needed for it.
@@ -1044,6 +1080,7 @@
 ; where MODE is the octal representation of the mode. Note that this
 ; does not overwrite the process's umask.
 ;session.save_path = "/tmp"
+session.save_path = "/var/lib/php"
 
 ; Whether to use cookies.
 session.use_cookies = 1
