--- amarok-2.5.0/src/musicbrainz/MusicDNSAudioDecoder.cpp.orig	2011-12-15 07:36:48.000000000 +0000
+++ amarok-2.5.0/src/musicbrainz/MusicDNSAudioDecoder.cpp	2012-04-21 21:09:59.000000000 +0100
@@ -136,7 +136,7 @@
     foreach( Meta::TrackPtr track, m_tracks )
     {
         //TODO replace with "avformat_open_input" since av_open_input_file is deprecated
-        if( av_open_input_file( &pFormatCtx, ( const char * )track->playableUrl().toLocalFile().toAscii(), NULL, 0, NULL ) )
+        if( avformat_open_input( &pFormatCtx, ( const char * )track->playableUrl().toLocalFile().toAscii(), NULL, NULL ) )
         {
             warning() << QLatin1String( "Unable to open input file: " ) + track->playableUrl().toLocalFile();
             continue;
