--- amarok-2.8.0/src/musicbrainz/MusicDNSAudioDecoder.cpp.orig	2013-08-14 21:22:52.000000000 +0100
+++ amarok-2.8.0/src/musicbrainz/MusicDNSAudioDecoder.cpp	2014-11-14 11:55:23.579059752 +0000
@@ -23,7 +23,9 @@
 #include "core/meta/Meta.h"
 
 extern "C" {
-    typedef quint64 UINT64_C;
+//    typedef quint64 UINT64_C;
+    #define __STDC_CONSTANT_MACROS 1
+    #include <stdint.h>
     #include <libavcodec/avcodec.h>
     #include <libavformat/avformat.h>
     #include <libavutil/mathematics.h>
