--- amarok-2.7.0/CMakeLists.txt.orig	2013-01-17 13:12:00.000000000 +0000
+++ amarok-2.7.0/CMakeLists.txt	2013-02-08 22:40:01.419072436 +0000
@@ -64,9 +64,9 @@
 endif()
 
 # Needed to conditionally build tests and gui
-if(KDE4_BUILD_TESTS)
-   add_definitions(-DDEBUG)
-endif()
+#if(KDE4_BUILD_TESTS)
+#   add_definitions(-DDEBUG)
+#endif()
 
 if(WITH_DESKTOP_UI)
     add_definitions(-DDESKTOP_UI)
@@ -207,10 +207,10 @@
 
     include_directories( ${KDE4_INCLUDES} )
 
-    if( KDE4_BUILD_TESTS AND NOT WIN32 )
-        ENABLE_TESTING()
-        add_subdirectory( tests )
-    endif( KDE4_BUILD_TESTS AND NOT WIN32 )
+#    if( KDE4_BUILD_TESTS AND NOT WIN32 )
+#        ENABLE_TESTING()
+#        add_subdirectory( tests )
+#    endif( KDE4_BUILD_TESTS AND NOT WIN32 )
 
     add_subdirectory( src )
 
