--- calligra-2.4.3/plugins/reporting/maps/KoReportItemMaps.h.orig	2012-06-26 05:21:31.000000000 +0100
+++ calligra-2.4.3/plugins/reporting/maps/KoReportItemMaps.h	2012-08-05 20:49:17.331822525 +0100
@@ -29,7 +29,7 @@
 #include <kdebug.h>
 #include <klocalizedstring.h>
 #include <kglobalsettings.h>
-#include <MarbleWidget.h>
+#include <marble/MarbleWidget.h>
 #include <RdfForward.h>
 #include <QMap>
 
--- calligra-2.4.3/plugins/reporting/maps/KoReportItemMaps.cpp.orig	2012-06-26 05:21:31.000000000 +0100
+++ calligra-2.4.3/plugins/reporting/maps/KoReportItemMaps.cpp	2012-08-05 21:11:34.528843783 +0100
@@ -25,8 +25,8 @@
 #include <QBuffer>
 #include <kcodecs.h>
 #include <renderobjects.h>
-#include <MarbleWidget.h>
-#include <MarbleModel.h>
+#include <marble/MarbleWidget.h>
+#include <marble/MarbleModel.h>
 #include <QImage>
 #include <QPixmap>
 #include <sys/socket.h>
--- calligra-2.4.3/kexi/plugins/forms/widgets/mapbrowser/MapBrowserWidget.h.orig	2012-06-26 05:21:31.000000000 +0100
+++ calligra-2.4.3/kexi/plugins/forms/widgets/mapbrowser/MapBrowserWidget.h	2012-08-05 21:28:19.609859765 +0100
@@ -21,7 +21,7 @@
 #ifndef MAPBROWSERWIDGET_H
 #define MAPBROWSERWIDGET_H
 
-#include <MarbleWidget.h>
+#include <marble/MarbleWidget.h>
 
 #include "widgetfactory.h"
 #include "container.h"
--- calligra-2.4.3/kexi/plugins/forms/widgets/mapbrowser/MapBrowserWidget.cpp.orig	2012-06-26 05:21:31.000000000 +0100
+++ calligra-2.4.3/kexi/plugins/forms/widgets/mapbrowser/MapBrowserWidget.cpp	2012-08-05 21:28:06.557859551 +0100
@@ -19,7 +19,7 @@
 
 
 #include "MapBrowserWidget.h"
-#include <MarbleWidget.h>
+#include <marble/MarbleWidget.h>
 #include <QPointF>
 #include <KDebug>
 
