--- digikam-2.9.0/extra/kipi-plugins/printimages/tools/tphoto.cpp.orig	2012-09-02 11:37:39.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/printimages/tools/tphoto.cpp	2013-02-07 21:40:22.323458949 +0000
@@ -125,7 +125,7 @@
 
     // Check if RAW file.
     if (KPMetadata::isRawFile(filename))
-        KDcrawIface::KDcraw::loadDcrawPreview(photo, filename.path());
+        KDcrawIface::KDcraw::loadRawPreview(photo, filename.path());
     else
         photo.load(filename.path()); // PENDING(blackie) handle URL
 
--- digikam-2.9.0/extra/kipi-plugins/smug/smugwindow.cpp.orig	2012-09-02 11:37:39.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/smug/smugwindow.cpp	2013-02-07 21:41:38.681451017 +0000
@@ -694,7 +694,7 @@
     if (isRAW)
     {
         kDebug() << "Get RAW preview " << imgPath;
-        KDcrawIface::KDcraw::loadDcrawPreview(image, imgPath);
+        KDcrawIface::KDcraw::loadRawPreview(image, imgPath);
     }
     else
     {
--- digikam-2.9.0/extra/kipi-plugins/galleryexport/gallerytalker.cpp.orig	2012-09-02 11:37:42.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/galleryexport/gallerytalker.cpp	2013-02-07 21:38:59.628467539 +0000
@@ -251,7 +251,7 @@
 
     // Check if RAW file.
     if (KPMetadata::isRawFile(photoPath))
-        KDcrawIface::KDcraw::loadDcrawPreview(image, photoPath);
+        KDcrawIface::KDcraw::loadRawPreview(image, photoPath);
     else
         image.load(photoPath);
 
--- digikam-2.9.0/extra/kipi-plugins/rajceexport/rajcesession.cpp.orig	2012-09-02 11:37:40.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/rajceexport/rajcesession.cpp	2013-02-07 21:40:43.351456764 +0000
@@ -557,7 +557,7 @@
     if (isRaw)
     {
         kDebug() << "Get RAW preview " << path;
-        KDcrawIface::KDcraw::loadDcrawPreview(m_image, path);
+        KDcrawIface::KDcraw::loadRawPreview(m_image, path);
     }
     else
     {
--- digikam-2.9.0/extra/kipi-plugins/rawconverter/manager/actionthread.cpp.orig	2012-09-02 11:37:42.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/rawconverter/manager/actionthread.cpp	2013-02-07 21:40:58.345455207 +0000
@@ -174,7 +174,7 @@
             QImage image;
             {
                 KPFileReadLocker(m_d->iface, m_url.toLocalFile());
-                m_d->dcrawIface.loadDcrawPreview(image, m_url.toLocalFile());
+                m_d->dcrawIface.loadRawPreview(image, m_url.toLocalFile());
             }
 
             ActionData ad;
--- digikam-2.9.0/extra/kipi-plugins/shwup/swwindow.cpp.orig	2012-09-02 11:37:39.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/shwup/swwindow.cpp	2013-02-07 21:41:26.659452266 +0000
@@ -455,7 +455,7 @@
     if (isRAW)
     {
         kDebug() << "Get RAW preview " << imgPath;
-        KDcraw::loadDcrawPreview(image, imgPath);
+        KDcraw::loadRawPreview(image, imgPath);
     }
     else
     {
--- digikam-2.9.0/extra/kipi-plugins/debianscreenshots/dswindow.cpp.orig	2012-09-02 11:37:40.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/debianscreenshots/dswindow.cpp	2013-02-07 21:37:56.031474146 +0000
@@ -211,7 +211,7 @@
     if ( massage == DsWindow::ImageIsRaw )
     {
         kDebug() << "Get RAW preview " << imgPath;
-        KDcrawIface::KDcraw::loadDcrawPreview(image, imgPath);
+        KDcrawIface::KDcraw::loadRawPreview(image, imgPath);
     }
     else
     {
--- digikam-2.9.0/extra/kipi-plugins/advancedslideshow/plugin/slideshowloader.cpp.orig	2012-09-02 11:37:42.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/advancedslideshow/plugin/slideshowloader.cpp	2013-02-07 21:37:03.822479569 +0000
@@ -78,7 +78,7 @@
     if (KPMetadata::isRawFile(m_path))
     {
         // it's a RAW file, use the libkdcraw loader
-        KDcraw::loadDcrawPreview(newImage, m_path.toLocalFile());
+        KDcraw::loadRawPreview(newImage, m_path.toLocalFile());
     }
     else
     {
--- digikam-2.9.0/extra/kipi-plugins/advancedslideshow/plugin/imageloadthread.cpp.orig	2012-09-02 11:37:42.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/advancedslideshow/plugin/imageloadthread.cpp	2013-02-07 21:37:18.615478033 +0000
@@ -173,7 +173,7 @@
     if (KPMetadata::isRawFile(path))
     {
         // it's a RAW file, use the libkdcraw loader
-        KDcraw::loadDcrawPreview(image, path);
+        KDcraw::loadRawPreview(image, path);
     }
     else
     {
--- digikam-2.9.0/extra/kipi-plugins/flashexport/simpleviewer.cpp.orig	2012-09-02 11:37:40.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/flashexport/simpleviewer.cpp	2013-02-07 21:38:22.716471374 +0000
@@ -483,7 +483,7 @@
 
         // Check if RAW file.
         if (KPMetadata::isRawFile(url))
-            KDcrawIface::KDcraw::loadDcrawPreview(image, url.path());
+            KDcrawIface::KDcraw::loadRawPreview(image, url.path());
         else
             image.load(url.path());
 
--- digikam-2.9.0/extra/kipi-plugins/picasawebexport/picasawebwindow.cpp.orig	2012-09-02 11:37:40.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/picasawebexport/picasawebwindow.cpp	2013-02-07 21:39:54.780461810 +0000
@@ -553,7 +553,7 @@
     if (isRAW)
     {
         kDebug() << "Get RAW preview " << imgPath;
-        KDcrawIface::KDcraw::loadDcrawPreview(image, imgPath);
+        KDcrawIface::KDcraw::loadRawPreview(image, imgPath);
     }
     else
     {
--- digikam-2.9.0/extra/kipi-plugins/common/libkipiplugins/tools/kprawthumbthread.cpp.orig	2012-09-02 11:37:42.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/common/libkipiplugins/tools/kprawthumbthread.cpp	2013-02-07 21:36:11.777484976 +0000
@@ -109,7 +109,7 @@
         if (!url.isEmpty())
         {
             QImage img;
-            bool ret = KDcraw::loadDcrawPreview(img, url.path());
+            bool ret = KDcraw::loadRawPreview(img, url.path());
             if (ret)
             {
                 kDebug() << url << " :: processed as RAW file";
--- digikam-2.9.0/extra/kipi-plugins/common/libkipiplugins/widgets/kppreviewimage.cpp.orig	2012-09-02 11:37:42.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/common/libkipiplugins/widgets/kppreviewimage.cpp	2013-02-07 21:36:31.576482919 +0000
@@ -555,7 +555,7 @@
 
     if (KPMetadata::isRawFile(file))
     {
-        KDcraw::loadDcrawPreview(image, file);
+        KDcraw::loadRawPreview(image, file);
     }
     else
     {
--- digikam-2.9.0/extra/kipi-plugins/yandexfotki/yfwindow.cpp.orig	2012-09-02 11:37:39.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/yandexfotki/yfwindow.cpp	2013-02-07 21:41:55.848449233 +0000
@@ -771,7 +771,7 @@
             bool isRAW = KPMetadata::isRawFile(photo.originalUrl());
             if (isRAW)
             {
-                KDcraw::loadDcrawPreview(image, photo.originalUrl());
+                KDcraw::loadRawPreview(image, photo.originalUrl());
             }
             else
             {
--- digikam-2.9.0/extra/kipi-plugins/imageviewer/texture.cpp.orig	2012-09-02 11:37:40.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/imageviewer/texture.cpp	2013-02-07 21:39:32.064464170 +0000
@@ -117,7 +117,7 @@
     if (KPMetadata::isRawFile(d->filename))
     {
         // it's a RAW file, use the libkdcraw loader
-        KDcraw::loadDcrawPreview(d->qimage, d->filename);
+        KDcraw::loadRawPreview(d->qimage, d->filename);
     }
     else
     {
--- digikam-2.9.0/extra/kipi-plugins/facebook/fbwindow.cpp.orig	2012-09-02 11:37:42.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/facebook/fbwindow.cpp	2013-02-07 21:38:09.971472698 +0000
@@ -658,7 +658,7 @@
     if (isRAW)
     {
         kDebug() << "Get RAW preview " << imgPath;
-        KDcrawIface::KDcraw::loadDcrawPreview(image, imgPath);
+        KDcrawIface::KDcraw::loadRawPreview(image, imgPath);
     }
     else
     {
--- digikam-2.9.0/extra/kipi-plugins/flickrexport/flickrtalker.cpp.orig	2012-09-02 11:37:43.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/flickrexport/flickrtalker.cpp	2013-02-07 21:38:42.206469349 +0000
@@ -611,7 +611,7 @@
     // Check if RAW file.
     if (KPMetadata::isRawFile(photoPath))
     {
-        KDcrawIface::KDcraw::loadDcrawPreview(image, photoPath);
+        KDcrawIface::KDcraw::loadRawPreview(image, photoPath);
     }
     else
     {
--- digikam-2.9.0/extra/kipi-plugins/sendimages/imageresize.cpp.orig	2012-09-02 11:37:39.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/sendimages/imageresize.cpp	2013-02-07 21:41:12.211453766 +0000
@@ -125,7 +125,7 @@
 
     // Check if RAW file.
     if (KPMetadata::isRawFile(orgUrl))
-        KDcraw::loadDcrawPreview(img, orgUrl.path());
+        KDcraw::loadRawPreview(img, orgUrl.path());
     else
         img.load(orgUrl.path());
 
--- digikam-2.9.0/extra/kipi-plugins/piwigoexport/piwigotalker.cpp.orig	2012-09-02 11:37:40.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/piwigoexport/piwigotalker.cpp	2013-02-07 21:40:09.891460240 +0000
@@ -183,7 +183,7 @@
 
     // Check if RAW file.
     if (KPMetadata::isRawFile(photoPath))
-        KDcrawIface::KDcraw::loadDcrawPreview(image, photoPath);
+        KDcrawIface::KDcraw::loadRawPreview(image, photoPath);
     else
         image.load(photoPath);
 
--- digikam-2.9.0/extra/kipi-plugins/removeredeyes/detection/locators/haarclassifier/haarclassifierlocator.cpp.orig	2012-09-02 11:37:40.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/removeredeyes/detection/locators/haarclassifier/haarclassifierlocator.cpp	2013-02-01 19:20:36.163075928 +0000
@@ -209,7 +209,7 @@
                  0);
 
     // fill the mask
-    cvFillImage(i_mask, 0);
+    cvSet( i_mask, cvColorToScalar(0, cvGetElemType(i_mask)), 0 );
     d->red_eyes = 0;
 
     for (int i = 0; i < blobs.GetNumBlobs(); ++i)
@@ -234,8 +234,8 @@
     d->redMask  = cvCreateImage(cvGetSize(d->original), d->original->depth, 1);
 
     // reset masks
-    cvFillImage(d->aChannel, 0);
-    cvFillImage(d->redMask, 0);
+    cvSet( d->aChannel, cvColorToScalar(0, cvGetElemType(d->aChannel)), 0 );
+    cvSet( d->redMask, cvColorToScalar(0, cvGetElemType(d->redMask)), 0 );
 }
 
 void HaarClassifierLocator::clearBuffers()
--- digikam-2.9.0/extra/kipi-plugins/calendar/print/calpainter.cpp.orig	2012-09-02 11:37:39.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/calendar/print/calpainter.cpp	2013-02-07 21:37:39.131475901 +0000
@@ -316,7 +316,7 @@
     // Check if RAW file.
     if (KPMetadata::isRawFile(imagePath_))
     {
-        KDcrawIface::KDcraw::loadDcrawPreview(image_, imagePath_.path());
+        KDcrawIface::KDcraw::loadRawPreview(image_, imagePath_.path());
     }
     else
     {
--- digikam-2.9.0/extra/kipi-plugins/htmlexport/imagegenerationfunctor.cpp.orig	2012-09-02 11:37:42.000000000 +0100
+++ digikam-2.9.0/extra/kipi-plugins/htmlexport/imagegenerationfunctor.cpp	2013-02-07 21:39:16.146465823 +0000
@@ -106,7 +106,7 @@
     // Check if RAW file.
     if (KPMetadata::isRawFile(path))
     {
-        if (!KDcraw::loadDcrawPreview(originalImage, path))
+        if (!KDcraw::loadRawPreview(originalImage, path))
         {
             emitWarning(i18n("Error loading RAW image '%1'", path));
             return;
--- digikam-2.9.0/core/utilities/cameragui/devices/umscamera.cpp.orig	2012-09-02 11:37:54.000000000 +0100
+++ digikam-2.9.0/core/utilities/cameragui/devices/umscamera.cpp	2013-02-07 21:43:52.792437085 +0000
@@ -224,7 +224,7 @@
 
     // RAW files : try to extract embedded thumbnail using libkdcraw
 
-    KDcrawIface::KDcraw::loadDcrawPreview(thumbnail, path);
+    KDcrawIface::KDcraw::loadRawPreview(thumbnail, path);
 
     if (!thumbnail.isNull())
     {
