--- digikam-3.1.0/core/libs/dimg/dimg.cpp.orig	2013-03-11 08:24:28.000000000 +0000
+++ digikam-3.1.0/core/libs/dimg/dimg.cpp	2013-04-25 21:53:04.610855184 +0100
@@ -2265,6 +2265,7 @@
 
                 switchDims = true;
 
+                detach();
                 delete [] m_priv->data;
                 m_priv->data = (uchar*)newData;
             }
@@ -2288,6 +2289,7 @@
 
                 switchDims = true;
 
+                detach();
                 delete [] m_priv->data;
                 m_priv->data = (uchar*)newData;
             }
@@ -2400,6 +2402,7 @@
 
                 switchDims = true;
 
+                detach();
                 delete [] m_priv->data;
                 m_priv->data = (uchar*)newData;
             }
@@ -2423,6 +2426,7 @@
 
                 switchDims = true;
 
+                detach();
                 delete [] m_priv->data;
                 m_priv->data = (uchar*)newData;
             }
