--- digikam-2.1.0/extra/kipi-plugins/removeredeyes/detection/locators/haarclassifier/haarclassifierlocator.cpp.orig	2011-09-07 10:44:39.000000000 +0100
+++ digikam-2.1.0/extra/kipi-plugins/removeredeyes/detection/locators/haarclassifier/haarclassifierlocator.cpp	2011-09-12 20:18:37.000000000 +0100
@@ -212,7 +212,7 @@
                  0);
 
     // fill the mask
-    cvFillImage(i_mask, 0);
+    cvSet( i_mask, cvColorToScalar(0, cvGetElemType(i_mask)), 0 );
     d->red_eyes = 0;
 
     for (int i = 0; i < blobs.GetNumBlobs(); ++i)
@@ -237,8 +237,8 @@
     d->redMask  = cvCreateImage(cvGetSize(d->original), d->original->depth, 1);
 
     // reset masks
-    cvFillImage(d->aChannel, 0);
-    cvFillImage(d->redMask, 0);
+    cvSet( d->aChannel, cvColorToScalar(0, cvGetElemType(d->aChannel)), 0 );
+    cvSet( d->redMask, cvColorToScalar(0, cvGetElemType(d->redMask)), 0 );
 }
 
 void HaarClassifierLocator::clearBuffers()
