--- digikam-2.3.0/core/libs/widgets/common/ditemtooltip.cpp.orig	2011-11-07 07:50:35.000000000 +0000
+++ digikam-2.3.0/core/libs/widgets/common/ditemtooltip.cpp	2011-11-07 20:12:54.140022186 +0000
@@ -65,18 +65,18 @@
     headEnd     = QString("</b></center></font></nobr></td></tr>");
 
     cellBeg     = QString("<tr><td><nobr><font size=\"-1\" color=\"%1\" face=\"%2\">")
-                  .arg(kapp->palette().color(QPalette::Text).name())
+                  .arg(kapp->palette().color(QPalette::ToolTipText).name())
                   .arg(font.family());
     cellMid     = QString("</font></nobr></td><td><nobr><font size=\"-1\" color=\"%1\" face=\"%2\">")
-                  .arg(kapp->palette().color(QPalette::Text).name())
+                  .arg(kapp->palette().color(QPalette::ToolTipText).name())
                   .arg(font.family());
     cellEnd     = QString("</font></nobr></td></tr>");
 
     cellSpecBeg = QString("<tr><td><nobr><font size=\"-1\" color=\"%1\" face=\"%2\">")
-                  .arg(kapp->palette().color(QPalette::Text).name())
+                  .arg(kapp->palette().color(QPalette::ToolTipText).name())
                   .arg(font.family());
     cellSpecMid = QString("</font></nobr></td><td><nobr><font size=\"-1\" color=\"%1\" face=\"%2\"><i>")
-                  .arg(kapp->palette().color(QPalette::Text).name())
+                  .arg(kapp->palette().color(QPalette::ToolTipText).name())
                   .arg(font.family());
     cellSpecEnd = QString("</i></font></nobr></td></tr>");
 }
--- digikam-2.3.0/extra/kipi-plugins/removeredeyes/detection/locators/haarclassifier/haarclassifierlocator.cpp.orig	2011-11-07 07:50:26.000000000 +0000
+++ digikam-2.3.0/extra/kipi-plugins/removeredeyes/detection/locators/haarclassifier/haarclassifierlocator.cpp	2011-11-07 20:12:54.140022186 +0000
@@ -212,7 +212,7 @@
                  0);
 
     // fill the mask
-    cvFillImage(i_mask, 0);
+    cvSet( i_mask, cvColorToScalar(0, cvGetElemType(i_mask)), 0 );
     d->red_eyes = 0;
 
     for (int i = 0; i < blobs.GetNumBlobs(); ++i)
@@ -237,8 +237,8 @@
     d->redMask  = cvCreateImage(cvGetSize(d->original), d->original->depth, 1);
 
     // reset masks
-    cvFillImage(d->aChannel, 0);
-    cvFillImage(d->redMask, 0);
+    cvSet( d->aChannel, cvColorToScalar(0, cvGetElemType(d->aChannel)), 0 );
+    cvSet( d->redMask, cvColorToScalar(0, cvGetElemType(d->redMask)), 0 );
 }
 
 void HaarClassifierLocator::clearBuffers()
