--- guidance-power-manager-4.4.0/modules/CMakeLists.txt.orig	2010-02-11 22:06:45.000000000 +0000
+++ guidance-power-manager-4.4.0/modules/CMakeLists.txt	2011-02-27 22:27:05.212000004 +0000
@@ -4,7 +4,7 @@
 ADD_LIBRARY(ixf86misc SHARED ixf86misc.c)
 SET_TARGET_PROPERTIES(ixf86misc PROPERTIES PREFIX "")
 include_directories(${PYTHON_INCLUDE_PATH})
-TARGET_LINK_LIBRARIES(ixf86misc -L/usr/X11R6/lib -lX11 -lXxf86vm -lXext -lXrandr -lXrender -lXss ${PYTHON_LIBRARY})
+TARGET_LINK_LIBRARIES(ixf86misc -L/usr/X11R6/lib${LIB_SUFFIX} -lX11 -lXxf86vm -lXext -lXrandr -lXrender -lXss ${PYTHON_LIBRARY})
 
 INSTALL(TARGETS ixf86misc DESTINATION "${PYTHON_SITE_PACKAGES_DIR}")
 INSTALL(FILES "${CMAKE_CURRENT_SOURCE_DIR}/xf86misc.py" DESTINATION "${PYTHON_SITE_PACKAGES_DIR}")
