--- kamoso-20121020/cmake/FindGLIB2.cmake.orig	2012-10-20 16:17:55.000000000 +0100
+++ kamoso-20121020/cmake/FindGLIB2.cmake	2012-10-20 16:21:20.000000000 +0100
@@ -31,7 +31,7 @@
              PATHS ${PKG_GLIB_LIBRARY_DIRS} )
 
 find_path(GLIB2_INTERNAL_INCLUDE_DIR glibconfig.h
-          PATH_SUFFIXES glib-2.0/include ../lib/glib-2.0/include
+          PATH_SUFFIXES glib-2.0/include ../lib64/glib-2.0/include ../lib/glib-2.0/include
           PATHS ${PKG_GLIB_INCLUDE_DIRS} ${PKG_GLIB_LIBRARIES} ${CMAKE_SYSTEM_LIBRARY_PATH})
 
 set(GLIB2_INCLUDE_DIR ${GLIB2_MAIN_INCLUDE_DIR})
--- kamoso-20121020/src/webcamwidget.cpp.orig	2012-10-20 16:17:55.000000000 +0100
+++ kamoso-20121020/src/webcamwidget.cpp	2012-10-20 16:21:20.000000000 +0100
@@ -372,7 +372,8 @@
     " ! video/x-raw-yuv, width=640, height=480, framerate=15/1;"
     " video/x-raw-yuv, width=640, height=480, framerate=24/1;"
     " video/x-raw-yuv, width=640, height=480, framerate=30/1;"
-    " video/x-raw-yuv, width=352, height=288, framerate=15/1"
+    " video/x-raw-yuv, width=352, height=288, framerate=15/1;"
+    " video/x-raw-yuv, width=360, height=296, framerate=15/1"
 
     //Basic plug-in for video controls
     " ! gamma name=gamma"
