--- kamoso-2.0.2/src/webcamwidget.cpp.orig	2011-05-24 23:34:42.000000000 +0100
+++ kamoso-2.0.2/src/webcamwidget.cpp	2012-07-30 21:03:29.000000000 +0100
@@ -372,7 +372,8 @@
     " ! video/x-raw-yuv, width=640, height=480, framerate=15/1;"
     " video/x-raw-yuv, width=640, height=480, framerate=24/1;"
     " video/x-raw-yuv, width=640, height=480, framerate=30/1;"
-    " video/x-raw-yuv, width=352, height=288, framerate=15/1"
+    " video/x-raw-yuv, width=352, height=288, framerate=15/1;"
+    " video/x-raw-yuv, width=360, height=296, framerate=15/1"
 
     //Basic plug-in for video controls
     " ! gamma name=gamma"
