--- kamoso-2.0.2/src/webcamwidget.cpp.orig	2011-05-24 23:34:42.000000000 +0100
+++ kamoso-2.0.2/src/webcamwidget.cpp	2012-07-30 21:03:29.000000000 +0100
@@ -372,7 +372,8 @@
     " ! video/x-raw-yuv, width=640, height=480, framerate=15/1;"
     " video/x-raw-yuv, width=640, height=480, framerate=24/1;"
     " video/x-raw-yuv, width=640, height=480, framerate=30/1;"
-    " video/x-raw-yuv, width=352, height=288, framerate=15/1"
+    " video/x-raw-yuv, width=352, height=288, framerate=15/1;"
+    " video/x-raw-yuv, width=360, height=296, framerate=15/1"
 
     //Basic plug-in for video controls
     " ! gamma name=gamma"
--- kamoso-2.0.2/cmake/FindGLIB2.cmake.orig	2011-05-24 23:34:42.000000000 +0100
+++ kamoso-2.0.2/cmake/FindGLIB2.cmake	2012-08-31 22:08:42.065180316 +0100
@@ -31,7 +31,7 @@
              PATHS ${PKG_GLIB_LIBRARY_DIRS} )
 
 find_path(GLIB2_INTERNAL_INCLUDE_DIR glibconfig.h
-          PATH_SUFFIXES glib-2.0/include ../lib/glib-2.0/include
+          PATH_SUFFIXES glib-2.0/include ../lib64/glib-2.0/include ../lib/glib-2.0/include
           PATHS ${PKG_GLIB_INCLUDE_DIRS} ${PKG_GLIB_LIBRARIES} ${CMAKE_SYSTEM_LIBRARY_PATH})
 
 set(GLIB2_INCLUDE_DIR ${GLIB2_MAIN_INCLUDE_DIR})
--- kamoso-2.0.2/src/plugins/youtube/kipiplugin_youtube.cpp.orig	2011-05-24 23:34:42.000000000 +0100
+++ kamoso-2.0.2/src/plugins/youtube/kipiplugin_youtube.cpp	2012-08-31 22:31:36.707202170 +0100
@@ -36,6 +36,7 @@
 #include <kwallet.h>
 #include <KDebug>
 #include <libkipi/interface.h>
+#include <libkipi/imagecollection.h>
 
 using KWallet::Wallet;
 
