--- kde-runtime-4.8.0/nepomuk/ontologies/nrio.ontology.in.orig	2011-07-27 19:36:16.000000000 +0100
+++ kde-runtime-4.8.0/nepomuk/ontologies/nrio.ontology.in	2012-02-03 20:58:02.000000000 +0000
@@ -3,6 +3,6 @@
 Name=Nepomuk Resource Identification Ontology
 Comment=Required for backup and sync.
 Namespace=http://nepomuk.kde.org/ontologies/2010/11/11/nrio#
-Path=${ONTO_INSTALL_DIR}/nrio.trig
+Path=${CMAKE_INSTALL_PREFIX}/share/ontology/kde/nrio.trig
 MimeType=application/x-trig
 Type=Data
