--- kde-workspace-4.10.0/ksmserver/screenlocker/greeter/screensaverwindow.cpp.orig	2013-01-23 21:46:34.000000000 +0000
+++ kde-workspace-4.10.0/ksmserver/screenlocker/greeter/screensaverwindow.cpp	2013-02-12 19:59:13.474509260 +0000
@@ -131,6 +131,7 @@
     m_startMousePos = QPoint(-1, -1);
     //reappear in one minute
     m_reactivateTimer->start(1000 * 60);
+    stopXScreenSaver();
     hide();
     emit hidden();
 }
@@ -140,6 +141,7 @@
 {
     Q_UNUSED(event)
 
+    stopXScreenSaver();
     hide();
     emit hidden();
 }
@@ -155,6 +157,7 @@
     }
     else if ((event->globalPos() - m_startMousePos).manhattanLength() > QApplication::startDragDistance()) {
         m_startMousePos = QPoint(-1, -1);
+        stopXScreenSaver();
         hide();
         emit hidden();
         //reappear in one minute
