--- kde-workspace-4.9.1/plasma/generic/dataengines/soliddevice/soliddeviceengine.cpp.orig	2012-08-28 14:39:40.000000000 +0100
+++ kde-workspace-4.9.1/plasma/generic/dataengines/soliddevice/soliddeviceengine.cpp	2012-09-13 18:20:51.000000000 +0100
@@ -148,6 +148,9 @@
         setData(name, I18N_NOOP("Accessible"), storageaccess->isAccessible());
         setData(name, I18N_NOOP("File Path"), storageaccess->filePath());
 
+        setData(name, I18N_NOOP("Free Space"), 0);
+        setData(name, I18N_NOOP("Free Space Text"), "0 bytes");
+
         if (storageaccess->isAccessible()) {
             QVariant freeDiskVar;
             qlonglong freeDisk = freeDiskSpace(storageaccess->filePath());
