--- kdebase-4.3.5/apps/konqueror/kttsplugin/CMakeLists.txt.orig	2010-01-21 20:47:53.000000000 +0000
+++ kdebase-4.3.5/apps/konqueror/kttsplugin/CMakeLists.txt	2010-01-27 18:36:32.000000000 +0000
@@ -5,26 +5,13 @@
 
 include(MacroOptionalDependPackage)
 
-MACRO_OPTIONAL_DEPEND_PACKAGE(KdeWebKit "kdewebkit")
-if(DEPEND_PACKAGE_KdeWebKit)
-        macro_optional_find_package(KdeWebKit QUIET)
-else(DEPEND_PACKAGE_KdeWebKit)
-        set(KDEWEBKIT_FOUND FALSE)
-endif(DEPEND_PACKAGE_KdeWebKit)
-
-MACRO_OPTIONAL_DEPEND_PACKAGE(WebKitPart "kwebkitpart")
-if(DEPEND_PACKAGE_WebKitPart)
-        macro_optional_find_package(WebKitPart QUIET)
-else(DEPEND_PACKAGE_WebKitPart)
-        set(WEBKITPART_FOUND FALSE)
-endif(DEPEND_PACKAGE_WebKitPart)
+    set(KDEWEBKIT_FOUND TRUE)
+    set(WEBKITPART_FOUND TRUE)
 
-if( KDEWEBKIT_FOUND AND WEBKITPART_FOUND )
-    set( HAVE_WEBKITKDE true)
-    include_directories(${KDEWEBKIT_INCLUDE_DIR})
-    include_directories(${WEBKITPART_INCLUDE_DIR})
-endif( KDEWEBKIT_FOUND AND WEBKITPART_FOUND )
+    set(KDEWEBKIT_LIBRARIES kdewebkit)
+    set(WEBKITPART_LIBRARIES kwebkit)
 
+    set( HAVE_WEBKITKDE true)
 
 configure_file(config-kttsplugin.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-kttsplugin.h )
 
