--- kdelibs-4.4.2/kate/document/katedocument.cpp.orig	2010-03-26 17:24:26.000000000 +0000
+++ kdelibs-4.4.2/kate/document/katedocument.cpp	2010-04-17 09:15:22.000000000 +0100
@@ -812,9 +812,13 @@
   else
   {
     int startLine = qMax(0, range.start().line());
+    int vc1 = toVirtualColumn(range.start());
+    int vc2 = toVirtualColumn(range.end());
     for (int line = qMin(range.end().line(), lastLine()); line >= startLine; --line) {
-      KTextEditor::Range subRange = rangeOnLine(range, line);
-      editRemoveText(line, subRange.start().column(), subRange.end().column() - subRange.start().column());
+      KateTextLine::Ptr tl = const_cast<KateDocument*>(this)->kateTextLine(line);
+      int col1 = tl->fromVirtualColumn(vc1, config()->tabWidth());
+      int col2 = tl->fromVirtualColumn(vc2, config()->tabWidth());
+      editRemoveText(line, qMin(col1, col2), qAbs(col2 - col1));
     }
   }
 
