--- kdelibs-4.10.3/plasma/private/dialogshadows.cpp.orig	2013-05-03 05:41:11.000000000 +0100
+++ kdelibs-4.10.3/plasma/private/dialogshadows.cpp	2013-05-09 07:08:31.816746200 +0100
@@ -335,6 +335,9 @@
 void DialogShadows::Private::freeX11Pixmaps()
 {
 #ifdef Q_WS_X11
+    if (!QX11Info::display())
+        return;
+
     foreach (const QPixmap &pixmap, m_shadowPixmaps) {
         if (!pixmap.isNull()) {
             XFreePixmap(QX11Info::display(), pixmap.handle());
