--- kdemultimedia-4.8.2/ffmpegthumbs/ffmpegthumbnailer/moviedecoder.cpp.orig	2011-07-27 19:25:34.000000000 +0100
+++ kdemultimedia-4.8.2/ffmpegthumbs/ffmpegthumbnailer/moviedecoder.cpp	2012-04-06 19:24:15.000000000 +0100
@@ -52,12 +52,11 @@
 void MovieDecoder::initialize(const QString& filename)
 {
     av_register_all();
-    avcodec_init();
     avcodec_register_all();
 
     QFileInfo fileInfo(filename);
 
-    if ((!m_FormatContextWasGiven) && av_open_input_file(&m_pFormatContext, fileInfo.absoluteFilePath().toUtf8().data(), NULL, 0, NULL) != 0) {
+    if ((!m_FormatContextWasGiven) && avformat_open_input(&m_pFormatContext, fileInfo.absoluteFilePath().toUtf8().data(), NULL, NULL) != 0) {
         kDebug() <<  "Could not open input file: " << fileInfo.absoluteFilePath();
         return;
     }
