--- kdemultimedia-4.6.3/ffmpegthumbs/ffmpegthumbnailer/moviedecoder.cpp.orig	2011-01-19 22:23:19.000000000 +0000
+++ kdemultimedia-4.6.3/ffmpegthumbs/ffmpegthumbnailer/moviedecoder.cpp	2011-06-05 16:35:35.084873197 +0100
@@ -122,7 +122,7 @@
 void MovieDecoder::initializeVideo()
 {
     for (unsigned int i = 0; i < m_pFormatContext->nb_streams; i++) {
-        if (m_pFormatContext->streams[i]->codec->codec_type == CODEC_TYPE_VIDEO) {
+        if (m_pFormatContext->streams[i]->codec->codec_type == AVMEDIA_TYPE_VIDEO) {
             m_pVideoStream = m_pFormatContext->streams[i];
             m_VideoStream = i;
             break;
