--- kmymoney-4.6.2/libkdchart/src/KDChartBackgroundAttributes.cpp.orig	2012-02-04 08:52:59.000000000 +0000
+++ kmymoney-4.6.2/libkdchart/src/KDChartBackgroundAttributes.cpp	2012-02-05 15:06:50.000000000 +0000
@@ -150,7 +150,7 @@
 	<< "visible="<<ba.isVisible()
 	<< "brush="<<ba.brush()
 	<< "pixmapmode="<<ba.pixmapMode()
-	<< "pixmap="<<ba.pixmap()
+//	<< "pixmap="<<ba.pixmap()
 	<< ")";
     return dbg;
 }
