--- printer-applet-4.9.5/printer-applet.py.orig	2012-11-30 07:58:12.000000000 +0000
+++ printer-applet-4.9.5/printer-applet.py	2013-01-07 22:41:15.000000000 +0000
@@ -297,7 +297,7 @@
         try:
             import cupshelpers.ppds
             notification = NewPrinterNotification(bus, self)
-        except ImportError:
+        except:
             pass  # cupshelpers not installed, no new printer notification will be shown
 
     def cleanup (self):
