--- quanta-20130210/CMakeLists.txt.orig	2013-02-10 10:32:58.000000000 +0000
+++ quanta-20130210/CMakeLists.txt	2013-02-10 17:41:32.301981018 +0000
@@ -24,8 +24,8 @@
 
 add_subdirectory(src)
 add_subdirectory(data)
-add_subdirectory(plugins)
-add_subdirectory(debuggers)
-add_subdirectory(languages)
+#add_subdirectory(plugins)
+#add_subdirectory(debuggers)
+#add_subdirectory(languages)
 
 macro_display_feature_log()
