--- soundkonverter-0.3.11/src/metadata/audible/audibleproperties.h.orig	2007-12-17 15:53:19.000000000 +0000
+++ soundkonverter-0.3.11/src/metadata/audible/audibleproperties.h	2010-06-01 21:34:04.000000000 +0100
@@ -28,6 +28,7 @@
 
 #include <config.h>
 
+#include <stdio.h>
 #include <taglib/audioproperties.h>
 #include <taglib/tstring.h>
 
--- soundkonverter-0.3.11/src/metadata/wav/wavproperties.cpp.orig	2007-12-17 15:53:19.000000000 +0000
+++ soundkonverter-0.3.11/src/metadata/wav/wavproperties.cpp	2010-06-01 21:36:51.000000000 +0100
@@ -30,6 +30,7 @@
 #include "wavfile.h"
 
 #include <netinet/in.h> // ntohl
+#include <stdio.h>
 
 using namespace TagLib;
 
--- soundkonverter-0.3.11/src/metadata/wav/wavfiletyperesolver.cpp.orig	2007-12-17 15:53:19.000000000 +0000
+++ soundkonverter-0.3.11/src/metadata/wav/wavfiletyperesolver.cpp	2010-06-01 21:38:14.000000000 +0100
@@ -23,6 +23,7 @@
 #include "wavfile.h"
 
 #include <string.h>
+#include <stdio.h>
 
 TagLib::File *WavFileTypeResolver::createFile(const char *fileName,
         bool readProperties,
--- soundkonverter-0.3.11/src/metadata/wav/wavfile.cpp.orig	2007-12-17 15:53:19.000000000 +0000
+++ soundkonverter-0.3.11/src/metadata/wav/wavfile.cpp	2010-06-01 21:37:49.000000000 +0100
@@ -28,6 +28,7 @@
 #include <taglib/tfile.h>
 #include <taglib/audioproperties.h>
 #include <taglib/tag.h>
+#include <stdio.h>
 
 namespace TagLib {
 ////////////////////////////////////////////////////////////////////////////////
--- soundkonverter-0.3.11/admin/cvs.sh.orig	2008-03-02 10:34:18.000000000 +0000
+++ soundkonverter-0.3.11/admin/cvs.sh	2010-03-26 17:01:15.000000000 +0000
@@ -68,7 +68,7 @@
     echo "*** KDE requires automake $required_automake_version"
     exit 1
     ;;
-  automake*1.6.* | automake*1.[7-9]* | automake*1.10*)
+  automake*1.6.* | automake*1.[7-9]* | automake*1.1[0-9]*)
     echo "*** $AUTOMAKE_STRING found."
     UNSERMAKE=no
     ;;
